<!DOCTYPE html>

<html lang="en">
<head><meta charset="utf-8"/>
<meta content="width=device-width, initial-scale=1.0" name="viewport"/>
<title>TMEM175_ALL_GLM</title><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>
<style type="text/css">
    pre { line-height: 125%; }
td.linenos .normal { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }
span.linenos { color: inherit; background-color: transparent; padding-left: 5px; padding-right: 5px; }
td.linenos .special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }
span.linenos.special { color: #000000; background-color: #ffffc0; padding-left: 5px; padding-right: 5px; }
.highlight .hll { background-color: var(--jp-cell-editor-active-background) }
.highlight { background: var(--jp-cell-editor-background); color: var(--jp-mirror-editor-variable-color) }
.highlight .c { color: var(--jp-mirror-editor-comment-color); font-style: italic } /* Comment */
.highlight .err { color: var(--jp-mirror-editor-error-color) } /* Error */
.highlight .k { color: var(--jp-mirror-editor-keyword-color); font-weight: bold } /* Keyword */
.highlight .o { color: var(--jp-mirror-editor-operator-color); font-weight: bold } /* Operator */
.highlight .p { color: var(--jp-mirror-editor-punctuation-color) } /* Punctuation */
.highlight .ch { color: var(--jp-mirror-editor-comment-color); font-style: italic } /* Comment.Hashbang */
.highlight .cm { color: var(--jp-mirror-editor-comment-color); font-style: italic } /* Comment.Multiline */
.highlight .cp { color: var(--jp-mirror-editor-comment-color); font-style: italic } /* Comment.Preproc */
.highlight .cpf { color: var(--jp-mirror-editor-comment-color); font-style: italic } /* Comment.PreprocFile */
.highlight .c1 { color: var(--jp-mirror-editor-comment-color); font-style: italic } /* Comment.Single */
.highlight .cs { color: var(--jp-mirror-editor-comment-color); font-style: italic } /* Comment.Special */
.highlight .kc { color: var(--jp-mirror-editor-keyword-color); font-weight: bold } /* Keyword.Constant */
.highlight .kd { color: var(--jp-mirror-editor-keyword-color); font-weight: bold } /* Keyword.Declaration */
.highlight .kn { color: var(--jp-mirror-editor-keyword-color); font-weight: bold } /* Keyword.Namespace */
.highlight .kp { color: var(--jp-mirror-editor-keyword-color); font-weight: bold } /* Keyword.Pseudo */
.highlight .kr { color: var(--jp-mirror-editor-keyword-color); font-weight: bold } /* Keyword.Reserved */
.highlight .kt { color: var(--jp-mirror-editor-keyword-color); font-weight: bold } /* Keyword.Type */
.highlight .m { color: var(--jp-mirror-editor-number-color) } /* Literal.Number */
.highlight .s { color: var(--jp-mirror-editor-string-color) } /* Literal.String */
.highlight .ow { color: var(--jp-mirror-editor-operator-color); font-weight: bold } /* Operator.Word */
.highlight .pm { color: var(--jp-mirror-editor-punctuation-color) } /* Punctuation.Marker */
.highlight .w { color: var(--jp-mirror-editor-variable-color) } /* Text.Whitespace */
.highlight .mb { color: var(--jp-mirror-editor-number-color) } /* Literal.Number.Bin */
.highlight .mf { color: var(--jp-mirror-editor-number-color) } /* Literal.Number.Float */
.highlight .mh { color: var(--jp-mirror-editor-number-color) } /* Literal.Number.Hex */
.highlight .mi { color: var(--jp-mirror-editor-number-color) } /* Literal.Number.Integer */
.highlight .mo { color: var(--jp-mirror-editor-number-color) } /* Literal.Number.Oct */
.highlight .sa { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Affix */
.highlight .sb { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Backtick */
.highlight .sc { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Char */
.highlight .dl { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Delimiter */
.highlight .sd { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Doc */
.highlight .s2 { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Double */
.highlight .se { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Escape */
.highlight .sh { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Heredoc */
.highlight .si { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Interpol */
.highlight .sx { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Other */
.highlight .sr { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Regex */
.highlight .s1 { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Single */
.highlight .ss { color: var(--jp-mirror-editor-string-color) } /* Literal.String.Symbol */
.highlight .il { color: var(--jp-mirror-editor-number-color) } /* Literal.Number.Integer.Long */
  </style>
<style type="text/css">
/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*
 * Mozilla scrollbar styling
 */

/* use standard opaque scrollbars for most nodes */
[data-jp-theme-scrollbars='true'] {
  scrollbar-color: rgb(var(--jp-scrollbar-thumb-color))
    var(--jp-scrollbar-background-color);
}

/* for code nodes, use a transparent style of scrollbar. These selectors
 * will match lower in the tree, and so will override the above */
[data-jp-theme-scrollbars='true'] .CodeMirror-hscrollbar,
[data-jp-theme-scrollbars='true'] .CodeMirror-vscrollbar {
  scrollbar-color: rgba(var(--jp-scrollbar-thumb-color), 0.5) transparent;
}

/* tiny scrollbar */

.jp-scrollbar-tiny {
  scrollbar-color: rgba(var(--jp-scrollbar-thumb-color), 0.5) transparent;
  scrollbar-width: thin;
}

/* tiny scrollbar */

.jp-scrollbar-tiny::-webkit-scrollbar,
.jp-scrollbar-tiny::-webkit-scrollbar-corner {
  background-color: transparent;
  height: 4px;
  width: 4px;
}

.jp-scrollbar-tiny::-webkit-scrollbar-thumb {
  background: rgba(var(--jp-scrollbar-thumb-color), 0.5);
}

.jp-scrollbar-tiny::-webkit-scrollbar-track:horizontal {
  border-left: 0 solid transparent;
  border-right: 0 solid transparent;
}

.jp-scrollbar-tiny::-webkit-scrollbar-track:vertical {
  border-top: 0 solid transparent;
  border-bottom: 0 solid transparent;
}

/*
 * Lumino
 */

.lm-ScrollBar[data-orientation='horizontal'] {
  min-height: 16px;
  max-height: 16px;
  min-width: 45px;
  border-top: 1px solid #a0a0a0;
}

.lm-ScrollBar[data-orientation='vertical'] {
  min-width: 16px;
  max-width: 16px;
  min-height: 45px;
  border-left: 1px solid #a0a0a0;
}

.lm-ScrollBar-button {
  background-color: #f0f0f0;
  background-position: center center;
  min-height: 15px;
  max-height: 15px;
  min-width: 15px;
  max-width: 15px;
}

.lm-ScrollBar-button:hover {
  background-color: #dadada;
}

.lm-ScrollBar-button.lm-mod-active {
  background-color: #cdcdcd;
}

.lm-ScrollBar-track {
  background: #f0f0f0;
}

.lm-ScrollBar-thumb {
  background: #cdcdcd;
}

.lm-ScrollBar-thumb:hover {
  background: #bababa;
}

.lm-ScrollBar-thumb.lm-mod-active {
  background: #a0a0a0;
}

.lm-ScrollBar[data-orientation='horizontal'] .lm-ScrollBar-thumb {
  height: 100%;
  min-width: 15px;
  border-left: 1px solid #a0a0a0;
  border-right: 1px solid #a0a0a0;
}

.lm-ScrollBar[data-orientation='vertical'] .lm-ScrollBar-thumb {
  width: 100%;
  min-height: 15px;
  border-top: 1px solid #a0a0a0;
  border-bottom: 1px solid #a0a0a0;
}

.lm-ScrollBar[data-orientation='horizontal']
  .lm-ScrollBar-button[data-action='decrement'] {
  background-image: var(--jp-icon-caret-left);
  background-size: 17px;
}

.lm-ScrollBar[data-orientation='horizontal']
  .lm-ScrollBar-button[data-action='increment'] {
  background-image: var(--jp-icon-caret-right);
  background-size: 17px;
}

.lm-ScrollBar[data-orientation='vertical']
  .lm-ScrollBar-button[data-action='decrement'] {
  background-image: var(--jp-icon-caret-up);
  background-size: 17px;
}

.lm-ScrollBar[data-orientation='vertical']
  .lm-ScrollBar-button[data-action='increment'] {
  background-image: var(--jp-icon-caret-down);
  background-size: 17px;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-Widget {
  box-sizing: border-box;
  position: relative;
  overflow: hidden;
}

.lm-Widget.lm-mod-hidden {
  display: none !important;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

.lm-AccordionPanel[data-orientation='horizontal'] > .lm-AccordionPanel-title {
  /* Title is rotated for horizontal accordion panel using CSS */
  display: block;
  transform-origin: top left;
  transform: rotate(-90deg) translate(-100%);
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-CommandPalette {
  display: flex;
  flex-direction: column;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lm-CommandPalette-search {
  flex: 0 0 auto;
}

.lm-CommandPalette-content {
  flex: 1 1 auto;
  margin: 0;
  padding: 0;
  min-height: 0;
  overflow: auto;
  list-style-type: none;
}

.lm-CommandPalette-header {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.lm-CommandPalette-item {
  display: flex;
  flex-direction: row;
}

.lm-CommandPalette-itemIcon {
  flex: 0 0 auto;
}

.lm-CommandPalette-itemContent {
  flex: 1 1 auto;
  overflow: hidden;
}

.lm-CommandPalette-itemShortcut {
  flex: 0 0 auto;
}

.lm-CommandPalette-itemLabel {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.lm-close-icon {
  border: 1px solid transparent;
  background-color: transparent;
  position: absolute;
  z-index: 1;
  right: 3%;
  top: 0;
  bottom: 0;
  margin: auto;
  padding: 7px 0;
  display: none;
  vertical-align: middle;
  outline: 0;
  cursor: pointer;
}
.lm-close-icon:after {
  content: 'X';
  display: block;
  width: 15px;
  height: 15px;
  text-align: center;
  color: #000;
  font-weight: normal;
  font-size: 12px;
  cursor: pointer;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-DockPanel {
  z-index: 0;
}

.lm-DockPanel-widget {
  z-index: 0;
}

.lm-DockPanel-tabBar {
  z-index: 1;
}

.lm-DockPanel-handle {
  z-index: 2;
}

.lm-DockPanel-handle.lm-mod-hidden {
  display: none !important;
}

.lm-DockPanel-handle:after {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: '';
}

.lm-DockPanel-handle[data-orientation='horizontal'] {
  cursor: ew-resize;
}

.lm-DockPanel-handle[data-orientation='vertical'] {
  cursor: ns-resize;
}

.lm-DockPanel-handle[data-orientation='horizontal']:after {
  left: 50%;
  min-width: 8px;
  transform: translateX(-50%);
}

.lm-DockPanel-handle[data-orientation='vertical']:after {
  top: 50%;
  min-height: 8px;
  transform: translateY(-50%);
}

.lm-DockPanel-overlay {
  z-index: 3;
  box-sizing: border-box;
  pointer-events: none;
}

.lm-DockPanel-overlay.lm-mod-hidden {
  display: none !important;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-Menu {
  z-index: 10000;
  position: absolute;
  white-space: nowrap;
  overflow-x: hidden;
  overflow-y: auto;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lm-Menu-content {
  margin: 0;
  padding: 0;
  display: table;
  list-style-type: none;
}

.lm-Menu-item {
  display: table-row;
}

.lm-Menu-item.lm-mod-hidden,
.lm-Menu-item.lm-mod-collapsed {
  display: none !important;
}

.lm-Menu-itemIcon,
.lm-Menu-itemSubmenuIcon {
  display: table-cell;
  text-align: center;
}

.lm-Menu-itemLabel {
  display: table-cell;
  text-align: left;
}

.lm-Menu-itemShortcut {
  display: table-cell;
  text-align: right;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-MenuBar {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lm-MenuBar-content {
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: row;
  list-style-type: none;
}

.lm-MenuBar-item {
  box-sizing: border-box;
}

.lm-MenuBar-itemIcon,
.lm-MenuBar-itemLabel {
  display: inline-block;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-ScrollBar {
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lm-ScrollBar[data-orientation='horizontal'] {
  flex-direction: row;
}

.lm-ScrollBar[data-orientation='vertical'] {
  flex-direction: column;
}

.lm-ScrollBar-button {
  box-sizing: border-box;
  flex: 0 0 auto;
}

.lm-ScrollBar-track {
  box-sizing: border-box;
  position: relative;
  overflow: hidden;
  flex: 1 1 auto;
}

.lm-ScrollBar-thumb {
  box-sizing: border-box;
  position: absolute;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-SplitPanel-child {
  z-index: 0;
}

.lm-SplitPanel-handle {
  z-index: 1;
}

.lm-SplitPanel-handle.lm-mod-hidden {
  display: none !important;
}

.lm-SplitPanel-handle:after {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  content: '';
}

.lm-SplitPanel[data-orientation='horizontal'] > .lm-SplitPanel-handle {
  cursor: ew-resize;
}

.lm-SplitPanel[data-orientation='vertical'] > .lm-SplitPanel-handle {
  cursor: ns-resize;
}

.lm-SplitPanel[data-orientation='horizontal'] > .lm-SplitPanel-handle:after {
  left: 50%;
  min-width: 8px;
  transform: translateX(-50%);
}

.lm-SplitPanel[data-orientation='vertical'] > .lm-SplitPanel-handle:after {
  top: 50%;
  min-height: 8px;
  transform: translateY(-50%);
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-TabBar {
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.lm-TabBar[data-orientation='horizontal'] {
  flex-direction: row;
  align-items: flex-end;
}

.lm-TabBar[data-orientation='vertical'] {
  flex-direction: column;
  align-items: flex-end;
}

.lm-TabBar-content {
  margin: 0;
  padding: 0;
  display: flex;
  flex: 1 1 auto;
  list-style-type: none;
}

.lm-TabBar[data-orientation='horizontal'] > .lm-TabBar-content {
  flex-direction: row;
}

.lm-TabBar[data-orientation='vertical'] > .lm-TabBar-content {
  flex-direction: column;
}

.lm-TabBar-tab {
  display: flex;
  flex-direction: row;
  box-sizing: border-box;
  overflow: hidden;
  touch-action: none; /* Disable native Drag/Drop */
}

.lm-TabBar-tabIcon,
.lm-TabBar-tabCloseIcon {
  flex: 0 0 auto;
}

.lm-TabBar-tabLabel {
  flex: 1 1 auto;
  overflow: hidden;
  white-space: nowrap;
}

.lm-TabBar-tabInput {
  user-select: all;
  width: 100%;
  box-sizing: border-box;
}

.lm-TabBar-tab.lm-mod-hidden {
  display: none !important;
}

.lm-TabBar-addButton.lm-mod-hidden {
  display: none !important;
}

.lm-TabBar.lm-mod-dragging .lm-TabBar-tab {
  position: relative;
}

.lm-TabBar.lm-mod-dragging[data-orientation='horizontal'] .lm-TabBar-tab {
  left: 0;
  transition: left 150ms ease;
}

.lm-TabBar.lm-mod-dragging[data-orientation='vertical'] .lm-TabBar-tab {
  top: 0;
  transition: top 150ms ease;
}

.lm-TabBar.lm-mod-dragging .lm-TabBar-tab.lm-mod-dragging {
  transition: none;
}

.lm-TabBar-tabLabel .lm-TabBar-tabInput {
  user-select: all;
  width: 100%;
  box-sizing: border-box;
  background: inherit;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-TabPanel-tabBar {
  z-index: 1;
}

.lm-TabPanel-stackedPanel {
  z-index: 0;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-Collapse {
  display: flex;
  flex-direction: column;
  align-items: stretch;
}

.jp-Collapse-header {
  padding: 1px 12px;
  background-color: var(--jp-layout-color1);
  border-bottom: solid var(--jp-border-width) var(--jp-border-color2);
  color: var(--jp-ui-font-color1);
  cursor: pointer;
  display: flex;
  align-items: center;
  font-size: var(--jp-ui-font-size0);
  font-weight: 600;
  text-transform: uppercase;
  user-select: none;
}

.jp-Collapser-icon {
  height: 16px;
}

.jp-Collapse-header-collapsed .jp-Collapser-icon {
  transform: rotate(-90deg);
  margin: auto 0;
}

.jp-Collapser-title {
  line-height: 25px;
}

.jp-Collapse-contents {
  padding: 0 12px;
  background-color: var(--jp-layout-color1);
  color: var(--jp-ui-font-color1);
  overflow: auto;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/* This file was auto-generated by ensureUiComponents() in @jupyterlab/buildutils */

/**
 * (DEPRECATED) Support for consuming icons as CSS background images
 */

/* Icons urls */

:root {
  --jp-icon-add-above: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQiIGhlaWdodD0iMTQiIHZpZXdCb3g9IjAgMCAxNCAxNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPGcgY2xpcC1wYXRoPSJ1cmwoI2NsaXAwXzEzN18xOTQ5MikiPgo8cGF0aCBjbGFzcz0ianAtaWNvbjMiIGQ9Ik00Ljc1IDQuOTMwNjZINi42MjVWNi44MDU2NkM2LjYyNSA3LjAxMTkxIDYuNzkzNzUgNy4xODA2NiA3IDcuMTgwNjZDNy4yMDYyNSA3LjE4MDY2IDcuMzc1IDcuMDExOTEgNy4zNzUgNi44MDU2NlY0LjkzMDY2SDkuMjVDOS40NTYyNSA0LjkzMDY2IDkuNjI1IDQuNzYxOTEgOS42MjUgNC41NTU2NkM5LjYyNSA0LjM0OTQxIDkuNDU2MjUgNC4xODA2NiA5LjI1IDQuMTgwNjZINy4zNzVWMi4zMDU2NkM3LjM3NSAyLjA5OTQxIDcuMjA2MjUgMS45MzA2NiA3IDEuOTMwNjZDNi43OTM3NSAxLjkzMDY2IDYuNjI1IDIuMDk5NDEgNi42MjUgMi4zMDU2NlY0LjE4MDY2SDQuNzVDNC41NDM3NSA0LjE4MDY2IDQuMzc1IDQuMzQ5NDEgNC4zNzUgNC41NTU2NkM0LjM3NSA0Ljc2MTkxIDQuNTQzNzUgNC45MzA2NiA0Ljc1IDQuOTMwNjZaIiBmaWxsPSIjNjE2MTYxIiBzdHJva2U9IiM2MTYxNjEiIHN0cm9rZS13aWR0aD0iMC43Ii8+CjwvZz4KPHBhdGggY2xhc3M9ImpwLWljb24zIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiIGNsaXAtcnVsZT0iZXZlbm9kZCIgZD0iTTExLjUgOS41VjExLjVMMi41IDExLjVWOS41TDExLjUgOS41Wk0xMiA4QzEyLjU1MjMgOCAxMyA4LjQ0NzcyIDEzIDlWMTJDMTMgMTIuNTUyMyAxMi41NTIzIDEzIDEyIDEzTDIgMTNDMS40NDc3MiAxMyAxIDEyLjU1MjMgMSAxMlY5QzEgOC40NDc3MiAxLjQ0NzcxIDggMiA4TDEyIDhaIiBmaWxsPSIjNjE2MTYxIi8+CjxkZWZzPgo8Y2xpcFBhdGggaWQ9ImNsaXAwXzEzN18xOTQ5MiI+CjxyZWN0IGNsYXNzPSJqcC1pY29uMyIgd2lkdGg9IjYiIGhlaWdodD0iNiIgZmlsbD0id2hpdGUiIHRyYW5zZm9ybT0ibWF0cml4KC0xIDAgMCAxIDEwIDEuNTU1NjYpIi8+CjwvY2xpcFBhdGg+CjwvZGVmcz4KPC9zdmc+Cg==);
  --jp-icon-add-below: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQiIGhlaWdodD0iMTQiIHZpZXdCb3g9IjAgMCAxNCAxNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPGcgY2xpcC1wYXRoPSJ1cmwoI2NsaXAwXzEzN18xOTQ5OCkiPgo8cGF0aCBjbGFzcz0ianAtaWNvbjMiIGQ9Ik05LjI1IDEwLjA2OTNMNy4zNzUgMTAuMDY5M0w3LjM3NSA4LjE5NDM0QzcuMzc1IDcuOTg4MDkgNy4yMDYyNSA3LjgxOTM0IDcgNy44MTkzNEM2Ljc5Mzc1IDcuODE5MzQgNi42MjUgNy45ODgwOSA2LjYyNSA4LjE5NDM0TDYuNjI1IDEwLjA2OTNMNC43NSAxMC4wNjkzQzQuNTQzNzUgMTAuMDY5MyA0LjM3NSAxMC4yMzgxIDQuMzc1IDEwLjQ0NDNDNC4zNzUgMTAuNjUwNiA0LjU0Mzc1IDEwLjgxOTMgNC43NSAxMC44MTkzTDYuNjI1IDEwLjgxOTNMNi42MjUgMTIuNjk0M0M2LjYyNSAxMi45MDA2IDYuNzkzNzUgMTMuMDY5MyA3IDEzLjA2OTNDNy4yMDYyNSAxMy4wNjkzIDcuMzc1IDEyLjkwMDYgNy4zNzUgMTIuNjk0M0w3LjM3NSAxMC44MTkzTDkuMjUgMTAuODE5M0M5LjQ1NjI1IDEwLjgxOTMgOS42MjUgMTAuNjUwNiA5LjYyNSAxMC40NDQzQzkuNjI1IDEwLjIzODEgOS40NTYyNSAxMC4wNjkzIDkuMjUgMTAuMDY5M1oiIGZpbGw9IiM2MTYxNjEiIHN0cm9rZT0iIzYxNjE2MSIgc3Ryb2tlLXdpZHRoPSIwLjciLz4KPC9nPgo8cGF0aCBjbGFzcz0ianAtaWNvbjMiIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMi41IDUuNUwyLjUgMy41TDExLjUgMy41TDExLjUgNS41TDIuNSA1LjVaTTIgN0MxLjQ0NzcyIDcgMSA2LjU1MjI4IDEgNkwxIDNDMSAyLjQ0NzcyIDEuNDQ3NzIgMiAyIDJMMTIgMkMxMi41NTIzIDIgMTMgMi40NDc3MiAxMyAzTDEzIDZDMTMgNi41NTIyOSAxMi41NTIzIDcgMTIgN0wyIDdaIiBmaWxsPSIjNjE2MTYxIi8+CjxkZWZzPgo8Y2xpcFBhdGggaWQ9ImNsaXAwXzEzN18xOTQ5OCI+CjxyZWN0IGNsYXNzPSJqcC1pY29uMyIgd2lkdGg9IjYiIGhlaWdodD0iNiIgZmlsbD0id2hpdGUiIHRyYW5zZm9ybT0ibWF0cml4KDEgMS43NDg0NmUtMDcgMS43NDg0NmUtMDcgLTEgNCAxMy40NDQzKSIvPgo8L2NsaXBQYXRoPgo8L2RlZnM+Cjwvc3ZnPgo=);
  --jp-icon-add: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTE5IDEzaC02djZoLTJ2LTZINXYtMmg2VjVoMnY2aDZ2MnoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-bell: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDE2IDE2IiB2ZXJzaW9uPSIxLjEiPgogICA8cGF0aCBjbGFzcz0ianAtaWNvbjIganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjMzMzMzMzIgogICAgICBkPSJtOCAwLjI5Yy0xLjQgMC0yLjcgMC43My0zLjYgMS44LTEuMiAxLjUtMS40IDMuNC0xLjUgNS4yLTAuMTggMi4yLTAuNDQgNC0yLjMgNS4zbDAuMjggMS4zaDVjMC4wMjYgMC42NiAwLjMyIDEuMSAwLjcxIDEuNSAwLjg0IDAuNjEgMiAwLjYxIDIuOCAwIDAuNTItMC40IDAuNi0xIDAuNzEtMS41aDVsMC4yOC0xLjNjLTEuOS0wLjk3LTIuMi0zLjMtMi4zLTUuMy0wLjEzLTEuOC0wLjI2LTMuNy0xLjUtNS4yLTAuODUtMS0yLjItMS44LTMuNi0xLjh6bTAgMS40YzAuODggMCAxLjkgMC41NSAyLjUgMS4zIDAuODggMS4xIDEuMSAyLjcgMS4yIDQuNCAwLjEzIDEuNyAwLjIzIDMuNiAxLjMgNS4yaC0xMGMxLjEtMS42IDEuMi0zLjQgMS4zLTUuMiAwLjEzLTEuNyAwLjMtMy4zIDEuMi00LjQgMC41OS0wLjcyIDEuNi0xLjMgMi41LTEuM3ptLTAuNzQgMTJoMS41Yy0wLjAwMTUgMC4yOCAwLjAxNSAwLjc5LTAuNzQgMC43OS0wLjczIDAuMDAxNi0wLjcyLTAuNTMtMC43NC0wLjc5eiIgLz4KPC9zdmc+Cg==);
  --jp-icon-bug-dot: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyBqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiM2MTYxNjEiPgogICAgICAgIDxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMTcuMTkgOEgyMFYxMEgxNy45MUMxNy45NiAxMC4zMyAxOCAxMC42NiAxOCAxMVYxMkgyMFYxNEgxOC41SDE4VjE0LjAyNzVDMTUuNzUgMTQuMjc2MiAxNCAxNi4xODM3IDE0IDE4LjVDMTQgMTkuMjA4IDE0LjE2MzUgMTkuODc3OSAxNC40NTQ5IDIwLjQ3MzlDMTMuNzA2MyAyMC44MTE3IDEyLjg3NTcgMjEgMTIgMjFDOS43OCAyMSA3Ljg1IDE5Ljc5IDYuODEgMThINFYxNkg2LjA5QzYuMDQgMTUuNjcgNiAxNS4zNCA2IDE1VjE0SDRWMTJINlYxMUM2IDEwLjY2IDYuMDQgMTAuMzMgNi4wOSAxMEg0VjhINi44MUM3LjI2IDcuMjIgNy44OCA2LjU1IDguNjIgNi4wNEw3IDQuNDFMOC40MSAzTDEwLjU5IDUuMTdDMTEuMDQgNS4wNiAxMS41MSA1IDEyIDVDMTIuNDkgNSAxMi45NiA1LjA2IDEzLjQyIDUuMTdMMTUuNTkgM0wxNyA0LjQxTDE1LjM3IDYuMDRDMTYuMTIgNi41NSAxNi43NCA3LjIyIDE3LjE5IDhaTTEwIDE2SDE0VjE0SDEwVjE2Wk0xMCAxMkgxNFYxMEgxMFYxMloiIGZpbGw9IiM2MTYxNjEiLz4KICAgICAgICA8cGF0aCBkPSJNMjIgMTguNUMyMiAyMC40MzMgMjAuNDMzIDIyIDE4LjUgMjJDMTYuNTY3IDIyIDE1IDIwLjQzMyAxNSAxOC41QzE1IDE2LjU2NyAxNi41NjcgMTUgMTguNSAxNUMyMC40MzMgMTUgMjIgMTYuNTY3IDIyIDE4LjVaIiBmaWxsPSIjNjE2MTYxIi8+CiAgICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-bug: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIj4KICAgIDxwYXRoIGQ9Ik0yMCA4aC0yLjgxYy0uNDUtLjc4LTEuMDctMS40NS0xLjgyLTEuOTZMMTcgNC40MSAxNS41OSAzbC0yLjE3IDIuMTdDMTIuOTYgNS4wNiAxMi40OSA1IDEyIDVjLS40OSAwLS45Ni4wNi0xLjQxLjE3TDguNDEgMyA3IDQuNDFsMS42MiAxLjYzQzcuODggNi41NSA3LjI2IDcuMjIgNi44MSA4SDR2MmgyLjA5Yy0uMDUuMzMtLjA5LjY2LS4wOSAxdjFINHYyaDJ2MWMwIC4zNC4wNC42Ny4wOSAxSDR2MmgyLjgxYzEuMDQgMS43OSAyLjk3IDMgNS4xOSAzczQuMTUtMS4yMSA1LjE5LTNIMjB2LTJoLTIuMDljLjA1LS4zMy4wOS0uNjYuMDktMXYtMWgydi0yaC0ydi0xYzAtLjM0LS4wNC0uNjctLjA5LTFIMjBWOHptLTYgOGgtNHYtMmg0djJ6bTAtNGgtNHYtMmg0djJ6Ii8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-build: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIHZpZXdCb3g9IjAgMCAyNCAyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTE0LjkgMTcuNDVDMTYuMjUgMTcuNDUgMTcuMzUgMTYuMzUgMTcuMzUgMTVDMTcuMzUgMTMuNjUgMTYuMjUgMTIuNTUgMTQuOSAxMi41NUMxMy41NCAxMi41NSAxMi40NSAxMy42NSAxMi40NSAxNUMxMi40NSAxNi4zNSAxMy41NCAxNy40NSAxNC45IDE3LjQ1Wk0yMC4xIDE1LjY4TDIxLjU4IDE2Ljg0QzIxLjcxIDE2Ljk1IDIxLjc1IDE3LjEzIDIxLjY2IDE3LjI5TDIwLjI2IDE5LjcxQzIwLjE3IDE5Ljg2IDIwIDE5LjkyIDE5LjgzIDE5Ljg2TDE4LjA5IDE5LjE2QzE3LjczIDE5LjQ0IDE3LjMzIDE5LjY3IDE2LjkxIDE5Ljg1TDE2LjY0IDIxLjdDMTYuNjIgMjEuODcgMTYuNDcgMjIgMTYuMyAyMkgxMy41QzEzLjMyIDIyIDEzLjE4IDIxLjg3IDEzLjE1IDIxLjdMMTIuODkgMTkuODVDMTIuNDYgMTkuNjcgMTIuMDcgMTkuNDQgMTEuNzEgMTkuMTZMOS45NjAwMiAxOS44NkM5LjgxMDAyIDE5LjkyIDkuNjIwMDIgMTkuODYgOS41NDAwMiAxOS43MUw4LjE0MDAyIDE3LjI5QzguMDUwMDIgMTcuMTMgOC4wOTAwMiAxNi45NSA4LjIyMDAyIDE2Ljg0TDkuNzAwMDIgMTUuNjhMOS42NTAwMSAxNUw5LjcwMDAyIDE0LjMxTDguMjIwMDIgMTMuMTZDOC4wOTAwMiAxMy4wNSA4LjA1MDAyIDEyLjg2IDguMTQwMDIgMTIuNzFMOS41NDAwMiAxMC4yOUM5LjYyMDAyIDEwLjEzIDkuODEwMDIgMTAuMDcgOS45NjAwMiAxMC4xM0wxMS43MSAxMC44NEMxMi4wNyAxMC41NiAxMi40NiAxMC4zMiAxMi44OSAxMC4xNUwxMy4xNSA4LjI4OTk4QzEzLjE4IDguMTI5OTggMTMuMzIgNy45OTk5OCAxMy41IDcuOTk5OThIMTYuM0MxNi40NyA3Ljk5OTk4IDE2LjYyIDguMTI5OTggMTYuNjQgOC4yODk5OEwxNi45MSAxMC4xNUMxNy4zMyAxMC4zMiAxNy43MyAxMC41NiAxOC4wOSAxMC44NEwxOS44MyAxMC4xM0MyMCAxMC4wNyAyMC4xNyAxMC4xMyAyMC4yNiAxMC4yOUwyMS42NiAxMi43MUMyMS43NSAxMi44NiAyMS43MSAxMy4wNSAyMS41OCAxMy4xNkwyMC4xIDE0LjMxTDIwLjE1IDE1TDIwLjEgMTUuNjhaIi8+CiAgICA8cGF0aCBkPSJNNy4zMjk2NiA3LjQ0NDU0QzguMDgzMSA3LjAwOTU0IDguMzM5MzIgNi4wNTMzMiA3LjkwNDMyIDUuMjk5ODhDNy40NjkzMiA0LjU0NjQzIDYuNTA4MSA0LjI4MTU2IDUuNzU0NjYgNC43MTY1NkM1LjM5MTc2IDQuOTI2MDggNS4xMjY5NSA1LjI3MTE4IDUuMDE4NDkgNS42NzU5NEM0LjkxMDA0IDYuMDgwNzEgNC45NjY4MiA2LjUxMTk4IDUuMTc2MzQgNi44NzQ4OEM1LjYxMTM0IDcuNjI4MzIgNi41NzYyMiA3Ljg3OTU0IDcuMzI5NjYgNy40NDQ1NFpNOS42NTcxOCA0Ljc5NTkzTDEwLjg2NzIgNC45NTE3OUMxMC45NjI4IDQuOTc3NDEgMTEuMDQwMiA1LjA3MTMzIDExLjAzODIgNS4xODc5M0wxMS4wMzg4IDYuOTg4OTNDMTEuMDQ1NSA3LjEwMDU0IDEwLjk2MTYgNy4xOTUxOCAxMC44NTUgNy4yMTA1NEw5LjY2MDAxIDcuMzgwODNMOS4yMzkxNSA4LjEzMTg4TDkuNjY5NjEgOS4yNTc0NUM5LjcwNzI5IDkuMzYyNzEgOS42NjkzNCA5LjQ3Njk5IDkuNTc0MDggOS41MzE5OUw4LjAxNTIzIDEwLjQzMkM3LjkxMTMxIDEwLjQ5MiA3Ljc5MzM3IDEwLjQ2NzcgNy43MjEwNSAxMC4zODI0TDYuOTg3NDggOS40MzE4OEw2LjEwOTMxIDkuNDMwODNMNS4zNDcwNCAxMC4zOTA1QzUuMjg5MDkgMTAuNDcwMiA1LjE3MzgzIDEwLjQ5MDUgNS4wNzE4NyAxMC40MzM5TDMuNTEyNDUgOS41MzI5M0MzLjQxMDQ5IDkuNDc2MzMgMy4zNzY0NyA5LjM1NzQxIDMuNDEwNzUgOS4yNTY3OUwzLjg2MzQ3IDguMTQwOTNMMy42MTc0OSA3Ljc3NDg4TDMuNDIzNDcgNy4zNzg4M0wyLjIzMDc1IDcuMjEyOTdDMi4xMjY0NyA3LjE5MjM1IDIuMDQwNDkgNy4xMDM0MiAyLjA0MjQ1IDYuOTg2ODJMMi4wNDE4NyA1LjE4NTgyQzIuMDQzODMgNS4wNjkyMiAyLjExOTA5IDQuOTc5NTggMi4yMTcwNCA0Ljk2OTIyTDMuNDIwNjUgNC43OTM5M0wzLjg2NzQ5IDQuMDI3ODhMMy40MTEwNSAyLjkxNzMxQzMuMzczMzcgMi44MTIwNCAzLjQxMTMxIDIuNjk3NzYgMy41MTUyMyAyLjYzNzc2TDUuMDc0MDggMS43Mzc3NkM1LjE2OTM0IDEuNjgyNzYgNS4yODcyOSAxLjcwNzA0IDUuMzU5NjEgMS43OTIzMUw2LjExOTE1IDIuNzI3ODhMNi45ODAwMSAyLjczODkzTDcuNzI0OTYgMS43ODkyMkM3Ljc5MTU2IDEuNzA0NTggNy45MTU0OCAxLjY3OTIyIDguMDA4NzkgMS43NDA4Mkw5LjU2ODIxIDIuNjQxODJDOS42NzAxNyAyLjY5ODQyIDkuNzEyODUgMi44MTIzNCA5LjY4NzIzIDIuOTA3OTdMOS4yMTcxOCA0LjAzMzgzTDkuNDYzMTYgNC4zOTk4OEw5LjY1NzE4IDQuNzk1OTNaIi8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-caret-down-empty-thin: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIwIDIwIj4KCTxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSIgc2hhcGUtcmVuZGVyaW5nPSJnZW9tZXRyaWNQcmVjaXNpb24iPgoJCTxwb2x5Z29uIGNsYXNzPSJzdDEiIHBvaW50cz0iOS45LDEzLjYgMy42LDcuNCA0LjQsNi42IDkuOSwxMi4yIDE1LjQsNi43IDE2LjEsNy40ICIvPgoJPC9nPgo8L3N2Zz4K);
  --jp-icon-caret-down-empty: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDE4IDE4Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiIHNoYXBlLXJlbmRlcmluZz0iZ2VvbWV0cmljUHJlY2lzaW9uIj4KICAgIDxwYXRoIGQ9Ik01LjIsNS45TDksOS43bDMuOC0zLjhsMS4yLDEuMmwtNC45LDVsLTQuOS01TDUuMiw1Ljl6Ii8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-caret-down: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDE4IDE4Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiIHNoYXBlLXJlbmRlcmluZz0iZ2VvbWV0cmljUHJlY2lzaW9uIj4KICAgIDxwYXRoIGQ9Ik01LjIsNy41TDksMTEuMmwzLjgtMy44SDUuMnoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-caret-left: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDE4IDE4Ij4KCTxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSIgc2hhcGUtcmVuZGVyaW5nPSJnZW9tZXRyaWNQcmVjaXNpb24iPgoJCTxwYXRoIGQ9Ik0xMC44LDEyLjhMNy4xLDlsMy44LTMuOGwwLDcuNkgxMC44eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-caret-right: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDE4IDE4Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiIHNoYXBlLXJlbmRlcmluZz0iZ2VvbWV0cmljUHJlY2lzaW9uIj4KICAgIDxwYXRoIGQ9Ik03LjIsNS4yTDEwLjksOWwtMy44LDMuOFY1LjJINy4yeiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-caret-up-empty-thin: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIwIDIwIj4KCTxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSIgc2hhcGUtcmVuZGVyaW5nPSJnZW9tZXRyaWNQcmVjaXNpb24iPgoJCTxwb2x5Z29uIGNsYXNzPSJzdDEiIHBvaW50cz0iMTUuNCwxMy4zIDkuOSw3LjcgNC40LDEzLjIgMy42LDEyLjUgOS45LDYuMyAxNi4xLDEyLjYgIi8+Cgk8L2c+Cjwvc3ZnPgo=);
  --jp-icon-caret-up: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDE4IDE4Ij4KCTxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSIgc2hhcGUtcmVuZGVyaW5nPSJnZW9tZXRyaWNQcmVjaXNpb24iPgoJCTxwYXRoIGQ9Ik01LjIsMTAuNUw5LDYuOGwzLjgsMy44SDUuMnoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-case-sensitive: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIwIDIwIj4KICA8ZyBjbGFzcz0ianAtaWNvbjIiIGZpbGw9IiM0MTQxNDEiPgogICAgPHJlY3QgeD0iMiIgeT0iMiIgd2lkdGg9IjE2IiBoZWlnaHQ9IjE2Ii8+CiAgPC9nPgogIDxnIGNsYXNzPSJqcC1pY29uLWFjY2VudDIiIGZpbGw9IiNGRkYiPgogICAgPHBhdGggZD0iTTcuNiw4aDAuOWwzLjUsOGgtMS4xTDEwLDE0SDZsLTAuOSwySDRMNy42LDh6IE04LDkuMUw2LjQsMTNoMy4yTDgsOS4xeiIvPgogICAgPHBhdGggZD0iTTE2LjYsOS44Yy0wLjIsMC4xLTAuNCwwLjEtMC43LDAuMWMtMC4yLDAtMC40LTAuMS0wLjYtMC4yYy0wLjEtMC4xLTAuMi0wLjQtMC4yLTAuNyBjLTAuMywwLjMtMC42LDAuNS0wLjksMC43Yy0wLjMsMC4xLTAuNywwLjItMS4xLDAuMmMtMC4zLDAtMC41LDAtMC43LTAuMWMtMC4yLTAuMS0wLjQtMC4yLTAuNi0wLjNjLTAuMi0wLjEtMC4zLTAuMy0wLjQtMC41IGMtMC4xLTAuMi0wLjEtMC40LTAuMS0wLjdjMC0wLjMsMC4xLTAuNiwwLjItMC44YzAuMS0wLjIsMC4zLTAuNCwwLjQtMC41QzEyLDcsMTIuMiw2LjksMTIuNSw2LjhjMC4yLTAuMSwwLjUtMC4xLDAuNy0wLjIgYzAuMy0wLjEsMC41LTAuMSwwLjctMC4xYzAuMiwwLDAuNC0wLjEsMC42LTAuMWMwLjIsMCwwLjMtMC4xLDAuNC0wLjJjMC4xLTAuMSwwLjItMC4yLDAuMi0wLjRjMC0xLTEuMS0xLTEuMy0xIGMtMC40LDAtMS40LDAtMS40LDEuMmgtMC45YzAtMC40LDAuMS0wLjcsMC4yLTFjMC4xLTAuMiwwLjMtMC40LDAuNS0wLjZjMC4yLTAuMiwwLjUtMC4zLDAuOC0wLjNDMTMuMyw0LDEzLjYsNCwxMy45LDQgYzAuMywwLDAuNSwwLDAuOCwwLjFjMC4zLDAsMC41LDAuMSwwLjcsMC4yYzAuMiwwLjEsMC40LDAuMywwLjUsMC41QzE2LDUsMTYsNS4yLDE2LDUuNnYyLjljMCwwLjIsMCwwLjQsMCwwLjUgYzAsMC4xLDAuMSwwLjIsMC4zLDAuMmMwLjEsMCwwLjIsMCwwLjMsMFY5Ljh6IE0xNS4yLDYuOWMtMS4yLDAuNi0zLjEsMC4yLTMuMSwxLjRjMCwxLjQsMy4xLDEsMy4xLTAuNVY2Ljl6Ii8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-check: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIj4KICAgIDxwYXRoIGQ9Ik05IDE2LjE3TDQuODMgMTJsLTEuNDIgMS40MUw5IDE5IDIxIDdsLTEuNDEtMS40MXoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-circle-empty: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTEyIDJDNi40NyAyIDIgNi40NyAyIDEyczQuNDcgMTAgMTAgMTAgMTAtNC40NyAxMC0xMFMxNy41MyAyIDEyIDJ6bTAgMThjLTQuNDEgMC04LTMuNTktOC04czMuNTktOCA4LTggOCAzLjU5IDggOC0zLjU5IDgtOCA4eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-circle: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMTggMTgiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPGNpcmNsZSBjeD0iOSIgY3k9IjkiIHI9IjgiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-clear: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8bWFzayBpZD0iZG9udXRIb2xlIj4KICAgIDxyZWN0IHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgZmlsbD0id2hpdGUiIC8+CiAgICA8Y2lyY2xlIGN4PSIxMiIgY3k9IjEyIiByPSI4IiBmaWxsPSJibGFjayIvPgogIDwvbWFzaz4KCiAgPGcgY2xhc3M9ImpwLWljb24zIiBmaWxsPSIjNjE2MTYxIj4KICAgIDxyZWN0IGhlaWdodD0iMTgiIHdpZHRoPSIyIiB4PSIxMSIgeT0iMyIgdHJhbnNmb3JtPSJyb3RhdGUoMzE1LCAxMiwgMTIpIi8+CiAgICA8Y2lyY2xlIGN4PSIxMiIgY3k9IjEyIiByPSIxMCIgbWFzaz0idXJsKCNkb251dEhvbGUpIi8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-close: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbi1ub25lIGpwLWljb24tc2VsZWN0YWJsZS1pbnZlcnNlIGpwLWljb24zLWhvdmVyIiBmaWxsPSJub25lIj4KICAgIDxjaXJjbGUgY3g9IjEyIiBjeT0iMTIiIHI9IjExIi8+CiAgPC9nPgoKICA8ZyBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIGpwLWljb24tYWNjZW50Mi1ob3ZlciIgZmlsbD0iIzYxNjE2MSI+CiAgICA8cGF0aCBkPSJNMTkgNi40MUwxNy41OSA1IDEyIDEwLjU5IDYuNDEgNSA1IDYuNDEgMTAuNTkgMTIgNSAxNy41OSA2LjQxIDE5IDEyIDEzLjQxIDE3LjU5IDE5IDE5IDE3LjU5IDEzLjQxIDEyeiIvPgogIDwvZz4KCiAgPGcgY2xhc3M9ImpwLWljb24tbm9uZSBqcC1pY29uLWJ1c3kiIGZpbGw9Im5vbmUiPgogICAgPGNpcmNsZSBjeD0iMTIiIGN5PSIxMiIgcj0iNyIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-code-check: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIiBzaGFwZS1yZW5kZXJpbmc9Imdlb21ldHJpY1ByZWNpc2lvbiI+CiAgICA8cGF0aCBkPSJNNi41OSwzLjQxTDIsOEw2LjU5LDEyLjZMOCwxMS4xOEw0LjgyLDhMOCw0LjgyTDYuNTksMy40MU0xMi40MSwzLjQxTDExLDQuODJMMTQuMTgsOEwxMSwxMS4xOEwxMi40MSwxMi42TDE3LDhMMTIuNDEsMy40MU0yMS41OSwxMS41OUwxMy41LDE5LjY4TDkuODMsMTZMOC40MiwxNy40MUwxMy41LDIyLjVMMjMsMTNMMjEuNTksMTEuNTlaIiAvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-code: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjIiIGhlaWdodD0iMjIiIHZpZXdCb3g9IjAgMCAyOCAyOCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KCTxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CgkJPHBhdGggZD0iTTExLjQgMTguNkw2LjggMTRMMTEuNCA5LjRMMTAgOEw0IDE0TDEwIDIwTDExLjQgMTguNlpNMTYuNiAxOC42TDIxLjIgMTRMMTYuNiA5LjRMMTggOEwyNCAxNEwxOCAyMEwxNi42IDE4LjZWMTguNloiLz4KCTwvZz4KPC9zdmc+Cg==);
  --jp-icon-collapse-all: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGgKICAgICAgICAgICAgZD0iTTggMmMxIDAgMTEgMCAxMiAwczIgMSAyIDJjMCAxIDAgMTEgMCAxMnMwIDItMiAyQzIwIDE0IDIwIDQgMjAgNFMxMCA0IDYgNGMwLTIgMS0yIDItMnoiIC8+CiAgICAgICAgPHBhdGgKICAgICAgICAgICAgZD0iTTE4IDhjMC0xLTEtMi0yLTJTNSA2IDQgNnMtMiAxLTIgMmMwIDEgMCAxMSAwIDEyczEgMiAyIDJjMSAwIDExIDAgMTIgMHMyLTEgMi0yYzAtMSAwLTExIDAtMTJ6bS0yIDB2MTJINFY4eiIgLz4KICAgICAgICA8cGF0aCBkPSJNNiAxM3YyaDh2LTJ6IiAvPgogICAgPC9nPgo8L3N2Zz4K);
  --jp-icon-console: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIwMCAyMDAiPgogIDxnIGNsYXNzPSJqcC1jb25zb2xlLWljb24tYmFja2dyb3VuZC1jb2xvciBqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiMwMjg4RDEiPgogICAgPHBhdGggZD0iTTIwIDE5LjhoMTYwdjE1OS45SDIweiIvPgogIDwvZz4KICA8ZyBjbGFzcz0ianAtY29uc29sZS1pY29uLWNvbG9yIGpwLWljb24tc2VsZWN0YWJsZS1pbnZlcnNlIiBmaWxsPSIjZmZmIj4KICAgIDxwYXRoIGQ9Ik0xMDUgMTI3LjNoNDB2MTIuOGgtNDB6TTUxLjEgNzdMNzQgOTkuOWwtMjMuMyAyMy4zIDEwLjUgMTAuNSAyMy4zLTIzLjNMOTUgOTkuOSA4NC41IDg5LjQgNjEuNiA2Ni41eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-copy: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMTggMTgiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTExLjksMUgzLjJDMi40LDEsMS43LDEuNywxLjcsMi41djEwLjJoMS41VjIuNWg4LjdWMXogTTE0LjEsMy45aC04Yy0wLjgsMC0xLjUsMC43LTEuNSwxLjV2MTAuMmMwLDAuOCwwLjcsMS41LDEuNSwxLjVoOCBjMC44LDAsMS41LTAuNywxLjUtMS41VjUuNEMxNS41LDQuNiwxNC45LDMuOSwxNC4xLDMuOXogTTE0LjEsMTUuNWgtOFY1LjRoOFYxNS41eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-copyright: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGVuYWJsZS1iYWNrZ3JvdW5kPSJuZXcgMCAwIDI0IDI0IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCI+CiAgPGcgY2xhc3M9ImpwLWljb24zIiBmaWxsPSIjNjE2MTYxIj4KICAgIDxwYXRoIGQ9Ik0xMS44OCw5LjE0YzEuMjgsMC4wNiwxLjYxLDEuMTUsMS42MywxLjY2aDEuNzljLTAuMDgtMS45OC0xLjQ5LTMuMTktMy40NS0zLjE5QzkuNjQsNy42MSw4LDksOCwxMi4xNCBjMCwxLjk0LDAuOTMsNC4yNCwzLjg0LDQuMjRjMi4yMiwwLDMuNDEtMS42NSwzLjQ0LTIuOTVoLTEuNzljLTAuMDMsMC41OS0wLjQ1LDEuMzgtMS42MywxLjQ0QzEwLjU1LDE0LjgzLDEwLDEzLjgxLDEwLDEyLjE0IEMxMCw5LjI1LDExLjI4LDkuMTYsMTEuODgsOS4xNHogTTEyLDJDNi40OCwyLDIsNi40OCwyLDEyczQuNDgsMTAsMTAsMTBzMTAtNC40OCwxMC0xMFMxNy41MiwyLDEyLDJ6IE0xMiwyMGMtNC40MSwwLTgtMy41OS04LTggczMuNTktOCw4LThzOCwzLjU5LDgsOFMxNi40MSwyMCwxMiwyMHoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-cut: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTkuNjQgNy42NGMuMjMtLjUuMzYtMS4wNS4zNi0xLjY0IDAtMi4yMS0xLjc5LTQtNC00UzIgMy43OSAyIDZzMS43OSA0IDQgNGMuNTkgMCAxLjE0LS4xMyAxLjY0LS4zNkwxMCAxMmwtMi4zNiAyLjM2QzcuMTQgMTQuMTMgNi41OSAxNCA2IDE0Yy0yLjIxIDAtNCAxLjc5LTQgNHMxLjc5IDQgNCA0IDQtMS43OSA0LTRjMC0uNTktLjEzLTEuMTQtLjM2LTEuNjRMMTIgMTRsNyA3aDN2LTFMOS42NCA3LjY0ek02IDhjLTEuMSAwLTItLjg5LTItMnMuOS0yIDItMiAyIC44OSAyIDItLjkgMi0yIDJ6bTAgMTJjLTEuMSAwLTItLjg5LTItMnMuOS0yIDItMiAyIC44OSAyIDItLjkgMi0yIDJ6bTYtNy41Yy0uMjggMC0uNS0uMjItLjUtLjVzLjIyLS41LjUtLjUuNS4yMi41LjUtLjIyLjUtLjUuNXpNMTkgM2wtNiA2IDIgMiA3LTdWM3oiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-delete: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjE2cHgiIGhlaWdodD0iMTZweCI+CiAgICA8cGF0aCBkPSJNMCAwaDI0djI0SDB6IiBmaWxsPSJub25lIiAvPgogICAgPHBhdGggY2xhc3M9ImpwLWljb24zIiBmaWxsPSIjNjI2MjYyIiBkPSJNNiAxOWMwIDEuMS45IDIgMiAyaDhjMS4xIDAgMi0uOSAyLTJWN0g2djEyek0xOSA0aC0zLjVsLTEtMWgtNWwtMSAxSDV2MmgxNFY0eiIgLz4KPC9zdmc+Cg==);
  --jp-icon-download: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTE5IDloLTRWM0g5djZINWw3IDcgNy03ek01IDE4djJoMTR2LTJINXoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-duplicate: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQiIGhlaWdodD0iMTQiIHZpZXdCb3g9IjAgMCAxNCAxNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggY2xhc3M9ImpwLWljb24zIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiIGNsaXAtcnVsZT0iZXZlbm9kZCIgZD0iTTIuNzk5OTggMC44NzVIOC44OTU4MkM5LjIwMDYxIDAuODc1IDkuNDQ5OTggMS4xMzkxNCA5LjQ0OTk4IDEuNDYxOThDOS40NDk5OCAxLjc4NDgyIDkuMjAwNjEgMi4wNDg5NiA4Ljg5NTgyIDIuMDQ4OTZIMy4zNTQxNUMzLjA0OTM2IDIuMDQ4OTYgMi43OTk5OCAyLjMxMzEgMi43OTk5OCAyLjYzNTk0VjkuNjc5NjlDMi43OTk5OCAxMC4wMDI1IDIuNTUwNjEgMTAuMjY2NyAyLjI0NTgyIDEwLjI2NjdDMS45NDEwMyAxMC4yNjY3IDEuNjkxNjUgMTAuMDAyNSAxLjY5MTY1IDkuNjc5NjlWMi4wNDg5NkMxLjY5MTY1IDEuNDAzMjggMi4xOTA0IDAuODc1IDIuNzk5OTggMC44NzVaTTUuMzY2NjUgMTEuOVY0LjU1SDExLjA4MzNWMTEuOUg1LjM2NjY1Wk00LjE0MTY1IDQuMTQxNjdDNC4xNDE2NSAzLjY5MDYzIDQuNTA3MjggMy4zMjUgNC45NTgzMiAzLjMyNUgxMS40OTE3QzExLjk0MjcgMy4zMjUgMTIuMzA4MyAzLjY5MDYzIDEyLjMwODMgNC4xNDE2N1YxMi4zMDgzQzEyLjMwODMgMTIuNzU5NCAxMS45NDI3IDEzLjEyNSAxMS40OTE3IDEzLjEyNUg0Ljk1ODMyQzQuNTA3MjggMTMuMTI1IDQuMTQxNjUgMTIuNzU5NCA0LjE0MTY1IDEyLjMwODNWNC4xNDE2N1oiIGZpbGw9IiM2MTYxNjEiLz4KPHBhdGggY2xhc3M9ImpwLWljb24zIiBkPSJNOS40MzU3NCA4LjI2NTA3SDguMzY0MzFWOS4zMzY1QzguMzY0MzEgOS40NTQzNSA4LjI2Nzg4IDkuNTUwNzggOC4xNTAwMiA5LjU1MDc4QzguMDMyMTcgOS41NTA3OCA3LjkzNTc0IDkuNDU0MzUgNy45MzU3NCA5LjMzNjVWOC4yNjUwN0g2Ljg2NDMxQzYuNzQ2NDUgOC4yNjUwNyA2LjY1MDAyIDguMTY4NjQgNi42NTAwMiA4LjA1MDc4QzYuNjUwMDIgNy45MzI5MiA2Ljc0NjQ1IDcuODM2NSA2Ljg2NDMxIDcuODM2NUg3LjkzNTc0VjYuNzY1MDdDNy45MzU3NCA2LjY0NzIxIDguMDMyMTcgNi41NTA3OCA4LjE1MDAyIDYuNTUwNzhDOC4yNjc4OCA2LjU1MDc4IDguMzY0MzEgNi42NDcyMSA4LjM2NDMxIDYuNzY1MDdWNy44MzY1SDkuNDM1NzRDOS41NTM2IDcuODM2NSA5LjY1MDAyIDcuOTMyOTIgOS42NTAwMiA4LjA1MDc4QzkuNjUwMDIgOC4xNjg2NCA5LjU1MzYgOC4yNjUwNyA5LjQzNTc0IDguMjY1MDdaIiBmaWxsPSIjNjE2MTYxIiBzdHJva2U9IiM2MTYxNjEiIHN0cm9rZS13aWR0aD0iMC41Ii8+Cjwvc3ZnPgo=);
  --jp-icon-edit: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTMgMTcuMjVWMjFoMy43NUwxNy44MSA5Ljk0bC0zLjc1LTMuNzVMMyAxNy4yNXpNMjAuNzEgNy4wNGMuMzktLjM5LjM5LTEuMDIgMC0xLjQxbC0yLjM0LTIuMzRjLS4zOS0uMzktMS4wMi0uMzktMS40MSAwbC0xLjgzIDEuODMgMy43NSAzLjc1IDEuODMtMS44M3oiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-ellipses: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPGNpcmNsZSBjeD0iNSIgY3k9IjEyIiByPSIyIi8+CiAgICA8Y2lyY2xlIGN4PSIxMiIgY3k9IjEyIiByPSIyIi8+CiAgICA8Y2lyY2xlIGN4PSIxOSIgY3k9IjEyIiByPSIyIi8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-error: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KPGcgY2xhc3M9ImpwLWljb24zIiBmaWxsPSIjNjE2MTYxIj48Y2lyY2xlIGN4PSIxMiIgY3k9IjE5IiByPSIyIi8+PHBhdGggZD0iTTEwIDNoNHYxMmgtNHoiLz48L2c+CjxwYXRoIGZpbGw9Im5vbmUiIGQ9Ik0wIDBoMjR2MjRIMHoiLz4KPC9zdmc+Cg==);
  --jp-icon-expand-all: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGgKICAgICAgICAgICAgZD0iTTggMmMxIDAgMTEgMCAxMiAwczIgMSAyIDJjMCAxIDAgMTEgMCAxMnMwIDItMiAyQzIwIDE0IDIwIDQgMjAgNFMxMCA0IDYgNGMwLTIgMS0yIDItMnoiIC8+CiAgICAgICAgPHBhdGgKICAgICAgICAgICAgZD0iTTE4IDhjMC0xLTEtMi0yLTJTNSA2IDQgNnMtMiAxLTIgMmMwIDEgMCAxMSAwIDEyczEgMiAyIDJjMSAwIDExIDAgMTIgMHMyLTEgMi0yYzAtMSAwLTExIDAtMTJ6bS0yIDB2MTJINFY4eiIgLz4KICAgICAgICA8cGF0aCBkPSJNMTEgMTBIOXYzSDZ2MmgzdjNoMnYtM2gzdi0yaC0zeiIgLz4KICAgIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-extension: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTIwLjUgMTFIMTlWN2MwLTEuMS0uOS0yLTItMmgtNFYzLjVDMTMgMi4xMiAxMS44OCAxIDEwLjUgMVM4IDIuMTIgOCAzLjVWNUg0Yy0xLjEgMC0xLjk5LjktMS45OSAydjMuOEgzLjVjMS40OSAwIDIuNyAxLjIxIDIuNyAyLjdzLTEuMjEgMi43LTIuNyAyLjdIMlYyMGMwIDEuMS45IDIgMiAyaDMuOHYtMS41YzAtMS40OSAxLjIxLTIuNyAyLjctMi43IDEuNDkgMCAyLjcgMS4yMSAyLjcgMi43VjIySDE3YzEuMSAwIDItLjkgMi0ydi00aDEuNWMxLjM4IDAgMi41LTEuMTIgMi41LTIuNVMyMS44OCAxMSAyMC41IDExeiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-fast-forward: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTQgMThsOC41LTZMNCA2djEyem05LTEydjEybDguNS02TDEzIDZ6Ii8+CiAgICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-file-upload: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTkgMTZoNnYtNmg0bC03LTctNyA3aDR6bS00IDJoMTR2Mkg1eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-file: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8cGF0aCBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIiBkPSJNMTkuMyA4LjJsLTUuNS01LjVjLS4zLS4zLS43LS41LTEuMi0uNUgzLjljLS44LjEtMS42LjktMS42IDEuOHYxNC4xYzAgLjkuNyAxLjYgMS42IDEuNmgxNC4yYy45IDAgMS42LS43IDEuNi0xLjZWOS40Yy4xLS41LS4xLS45LS40LTEuMnptLTUuOC0zLjNsMy40IDMuNmgtMy40VjQuOXptMy45IDEyLjdINC43Yy0uMSAwLS4yIDAtLjItLjJWNC43YzAtLjIuMS0uMy4yLS4zaDcuMnY0LjRzMCAuOC4zIDEuMWMuMy4zIDEuMS4zIDEuMS4zaDQuM3Y3LjJzLS4xLjItLjIuMnoiLz4KPC9zdmc+Cg==);
  --jp-icon-filter-dot: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiNGRkYiPgogICAgPHBhdGggZD0iTTE0LDEyVjE5Ljg4QzE0LjA0LDIwLjE4IDEzLjk0LDIwLjUgMTMuNzEsMjAuNzFDMTMuMzIsMjEuMSAxMi42OSwyMS4xIDEyLjMsMjAuNzFMMTAuMjksMTguN0MxMC4wNiwxOC40NyA5Ljk2LDE4LjE2IDEwLDE3Ljg3VjEySDkuOTdMNC4yMSw0LjYyQzMuODcsNC4xOSAzLjk1LDMuNTYgNC4zOCwzLjIyQzQuNTcsMy4wOCA0Ljc4LDMgNSwzVjNIMTlWM0MxOS4yMiwzIDE5LjQzLDMuMDggMTkuNjIsMy4yMkMyMC4wNSwzLjU2IDIwLjEzLDQuMTkgMTkuNzksNC42MkwxNC4wMywxMkgxNFoiIC8+CiAgPC9nPgogIDxnIGNsYXNzPSJqcC1pY29uLWRvdCIgZmlsbD0iI0ZGRiI+CiAgICA8Y2lyY2xlIGN4PSIxOCIgY3k9IjE3IiByPSIzIj48L2NpcmNsZT4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-filter-list: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTEwIDE4aDR2LTJoLTR2MnpNMyA2djJoMThWNkgzem0zIDdoMTJ2LTJINnYyeiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-filter: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiNGRkYiPgogICAgPHBhdGggZD0iTTE0LDEyVjE5Ljg4QzE0LjA0LDIwLjE4IDEzLjk0LDIwLjUgMTMuNzEsMjAuNzFDMTMuMzIsMjEuMSAxMi42OSwyMS4xIDEyLjMsMjAuNzFMMTAuMjksMTguN0MxMC4wNiwxOC40NyA5Ljk2LDE4LjE2IDEwLDE3Ljg3VjEySDkuOTdMNC4yMSw0LjYyQzMuODcsNC4xOSAzLjk1LDMuNTYgNC4zOCwzLjIyQzQuNTcsMy4wOCA0Ljc4LDMgNSwzVjNIMTlWM0MxOS4yMiwzIDE5LjQzLDMuMDggMTkuNjIsMy4yMkMyMC4wNSwzLjU2IDIwLjEzLDQuMTkgMTkuNzksNC42MkwxNC4wMywxMkgxNFoiIC8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-folder-favorite: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjRweCIgdmlld0JveD0iMCAwIDI0IDI0IiB3aWR0aD0iMjRweCIgZmlsbD0iIzAwMDAwMCI+CiAgPHBhdGggZD0iTTAgMGgyNHYyNEgwVjB6IiBmaWxsPSJub25lIi8+PHBhdGggY2xhc3M9ImpwLWljb24zIGpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iIzYxNjE2MSIgZD0iTTIwIDZoLThsLTItMkg0Yy0xLjEgMC0yIC45LTIgMnYxMmMwIDEuMS45IDIgMiAyaDE2YzEuMSAwIDItLjkgMi0yVjhjMC0xLjEtLjktMi0yLTJ6bS0yLjA2IDExTDE1IDE1LjI4IDEyLjA2IDE3bC43OC0zLjMzLTIuNTktMi4yNCAzLjQxLS4yOUwxNSA4bDEuMzQgMy4xNCAzLjQxLjI5LTIuNTkgMi4yNC43OCAzLjMzeiIvPgo8L3N2Zz4K);
  --jp-icon-folder: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8cGF0aCBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIiBkPSJNMTAgNEg0Yy0xLjEgMC0xLjk5LjktMS45OSAyTDIgMThjMCAxLjEuOSAyIDIgMmgxNmMxLjEgMCAyLS45IDItMlY4YzAtMS4xLS45LTItMi0yaC04bC0yLTJ6Ii8+Cjwvc3ZnPgo=);
  --jp-icon-home: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjRweCIgdmlld0JveD0iMCAwIDI0IDI0IiB3aWR0aD0iMjRweCIgZmlsbD0iIzAwMDAwMCI+CiAgPHBhdGggZD0iTTAgMGgyNHYyNEgweiIgZmlsbD0ibm9uZSIvPjxwYXRoIGNsYXNzPSJqcC1pY29uMyBqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiM2MTYxNjEiIGQ9Ik0xMCAyMHYtNmg0djZoNXYtOGgzTDEyIDMgMiAxMmgzdjh6Ii8+Cjwvc3ZnPgo=);
  --jp-icon-html5: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDUxMiA1MTIiPgogIDxwYXRoIGNsYXNzPSJqcC1pY29uMCBqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiMwMDAiIGQ9Ik0xMDguNCAwaDIzdjIyLjhoMjEuMlYwaDIzdjY5aC0yM1Y0NmgtMjF2MjNoLTIzLjJNMjA2IDIzaC0yMC4zVjBoNjMuN3YyM0gyMjl2NDZoLTIzbTUzLjUtNjloMjQuMWwxNC44IDI0LjNMMzEzLjIgMGgyNC4xdjY5aC0yM1YzNC44bC0xNi4xIDI0LjgtMTYuMS0yNC44VjY5aC0yMi42bTg5LjItNjloMjN2NDYuMmgzMi42VjY5aC01NS42Ii8+CiAgPHBhdGggY2xhc3M9ImpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iI2U0NGQyNiIgZD0iTTEwNy42IDQ3MWwtMzMtMzcwLjRoMzYyLjhsLTMzIDM3MC4yTDI1NS43IDUxMiIvPgogIDxwYXRoIGNsYXNzPSJqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiNmMTY1MjkiIGQ9Ik0yNTYgNDgwLjVWMTMxaDE0OC4zTDM3NiA0NDciLz4KICA8cGF0aCBjbGFzcz0ianAtaWNvbi1zZWxlY3RhYmxlLWludmVyc2UiIGZpbGw9IiNlYmViZWIiIGQ9Ik0xNDIgMTc2LjNoMTE0djQ1LjRoLTY0LjJsNC4yIDQ2LjVoNjB2NDUuM0gxNTQuNG0yIDIyLjhIMjAybDMuMiAzNi4zIDUwLjggMTMuNnY0Ny40bC05My4yLTI2Ii8+CiAgPHBhdGggY2xhc3M9ImpwLWljb24tc2VsZWN0YWJsZS1pbnZlcnNlIiBmaWxsPSIjZmZmIiBkPSJNMzY5LjYgMTc2LjNIMjU1Ljh2NDUuNGgxMDkuNm0tNC4xIDQ2LjVIMjU1Ljh2NDUuNGg1NmwtNS4zIDU5LTUwLjcgMTMuNnY0Ny4ybDkzLTI1LjgiLz4KPC9zdmc+Cg==);
  --jp-icon-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8cGF0aCBjbGFzcz0ianAtaWNvbi1icmFuZDQganAtaWNvbi1zZWxlY3RhYmxlLWludmVyc2UiIGZpbGw9IiNGRkYiIGQ9Ik0yLjIgMi4yaDE3LjV2MTcuNUgyLjJ6Ii8+CiAgPHBhdGggY2xhc3M9ImpwLWljb24tYnJhbmQwIGpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iIzNGNTFCNSIgZD0iTTIuMiAyLjJ2MTcuNWgxNy41bC4xLTE3LjVIMi4yem0xMi4xIDIuMmMxLjIgMCAyLjIgMSAyLjIgMi4ycy0xIDIuMi0yLjIgMi4yLTIuMi0xLTIuMi0yLjIgMS0yLjIgMi4yLTIuMnpNNC40IDE3LjZsMy4zLTguOCAzLjMgNi42IDIuMi0zLjIgNC40IDUuNEg0LjR6Ii8+Cjwvc3ZnPgo=);
  --jp-icon-info: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDUwLjk3OCA1MC45NzgiPgoJPGcgY2xhc3M9ImpwLWljb24zIiBmaWxsPSIjNjE2MTYxIj4KCQk8cGF0aCBkPSJNNDMuNTIsNy40NThDMzguNzExLDIuNjQ4LDMyLjMwNywwLDI1LjQ4OSwwQzE4LjY3LDAsMTIuMjY2LDIuNjQ4LDcuNDU4LDcuNDU4CgkJCWMtOS45NDMsOS45NDEtOS45NDMsMjYuMTE5LDAsMzYuMDYyYzQuODA5LDQuODA5LDExLjIxMiw3LjQ1NiwxOC4wMzEsNy40NThjMCwwLDAuMDAxLDAsMC4wMDIsMAoJCQljNi44MTYsMCwxMy4yMjEtMi42NDgsMTguMDI5LTcuNDU4YzQuODA5LTQuODA5LDcuNDU3LTExLjIxMiw3LjQ1Ny0xOC4wM0M1MC45NzcsMTguNjcsNDguMzI4LDEyLjI2Niw0My41Miw3LjQ1OHoKCQkJIE00Mi4xMDYsNDIuMTA1Yy00LjQzMiw0LjQzMS0xMC4zMzIsNi44NzItMTYuNjE1LDYuODcyaC0wLjAwMmMtNi4yODUtMC4wMDEtMTIuMTg3LTIuNDQxLTE2LjYxNy02Ljg3MgoJCQljLTkuMTYyLTkuMTYzLTkuMTYyLTI0LjA3MSwwLTMzLjIzM0MxMy4zMDMsNC40NCwxOS4yMDQsMiwyNS40ODksMmM2LjI4NCwwLDEyLjE4NiwyLjQ0LDE2LjYxNyw2Ljg3MgoJCQljNC40MzEsNC40MzEsNi44NzEsMTAuMzMyLDYuODcxLDE2LjYxN0M0OC45NzcsMzEuNzcyLDQ2LjUzNiwzNy42NzUsNDIuMTA2LDQyLjEwNXoiLz4KCQk8cGF0aCBkPSJNMjMuNTc4LDMyLjIxOGMtMC4wMjMtMS43MzQsMC4xNDMtMy4wNTksMC40OTYtMy45NzJjMC4zNTMtMC45MTMsMS4xMS0xLjk5NywyLjI3Mi0zLjI1MwoJCQljMC40NjgtMC41MzYsMC45MjMtMS4wNjIsMS4zNjctMS41NzVjMC42MjYtMC43NTMsMS4xMDQtMS40NzgsMS40MzYtMi4xNzVjMC4zMzEtMC43MDcsMC40OTUtMS41NDEsMC40OTUtMi41CgkJCWMwLTEuMDk2LTAuMjYtMi4wODgtMC43NzktMi45NzljLTAuNTY1LTAuODc5LTEuNTAxLTEuMzM2LTIuODA2LTEuMzY5Yy0xLjgwMiwwLjA1Ny0yLjk4NSwwLjY2Ny0zLjU1LDEuODMyCgkJCWMtMC4zMDEsMC41MzUtMC41MDMsMS4xNDEtMC42MDcsMS44MTRjLTAuMTM5LDAuNzA3LTAuMjA3LDEuNDMyLTAuMjA3LDIuMTc0aC0yLjkzN2MtMC4wOTEtMi4yMDgsMC40MDctNC4xMTQsMS40OTMtNS43MTkKCQkJYzEuMDYyLTEuNjQsMi44NTUtMi40ODEsNS4zNzgtMi41MjdjMi4xNiwwLjAyMywzLjg3NCwwLjYwOCw1LjE0MSwxLjc1OGMxLjI3OCwxLjE2LDEuOTI5LDIuNzY0LDEuOTUsNC44MTEKCQkJYzAsMS4xNDItMC4xMzcsMi4xMTEtMC40MSwyLjkxMWMtMC4zMDksMC44NDUtMC43MzEsMS41OTMtMS4yNjgsMi4yNDNjLTAuNDkyLDAuNjUtMS4wNjgsMS4zMTgtMS43MywyLjAwMgoJCQljLTAuNjUsMC42OTctMS4zMTMsMS40NzktMS45ODcsMi4zNDZjLTAuMjM5LDAuMzc3LTAuNDI5LDAuNzc3LTAuNTY1LDEuMTk5Yy0wLjE2LDAuOTU5LTAuMjE3LDEuOTUxLTAuMTcxLDIuOTc5CgkJCUMyNi41ODksMzIuMjE4LDIzLjU3OCwzMi4yMTgsMjMuNTc4LDMyLjIxOHogTTIzLjU3OCwzOC4yMnYtMy40ODRoMy4wNzZ2My40ODRIMjMuNTc4eiIvPgoJPC9nPgo8L3N2Zz4K);
  --jp-icon-inspector: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8cGF0aCBjbGFzcz0ianAtaW5zcGVjdG9yLWljb24tY29sb3IganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIiBkPSJNMjAgNEg0Yy0xLjEgMC0xLjk5LjktMS45OSAyTDIgMThjMCAxLjEuOSAyIDIgMmgxNmMxLjEgMCAyLS45IDItMlY2YzAtMS4xLS45LTItMi0yem0tNSAxNEg0di00aDExdjR6bTAtNUg0VjloMTF2NHptNSA1aC00VjloNHY5eiIvPgo8L3N2Zz4K);
  --jp-icon-json: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8ZyBjbGFzcz0ianAtanNvbi1pY29uLWNvbG9yIGpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iI0Y5QTgyNSI+CiAgICA8cGF0aCBkPSJNMjAuMiAxMS44Yy0xLjYgMC0xLjcuNS0xLjcgMSAwIC40LjEuOS4xIDEuMy4xLjUuMS45LjEgMS4zIDAgMS43LTEuNCAyLjMtMy41IDIuM2gtLjl2LTEuOWguNWMxLjEgMCAxLjQgMCAxLjQtLjggMC0uMyAwLS42LS4xLTEgMC0uNC0uMS0uOC0uMS0xLjIgMC0xLjMgMC0xLjggMS4zLTItMS4zLS4yLTEuMy0uNy0xLjMtMiAwLS40LjEtLjguMS0xLjIuMS0uNC4xLS43LjEtMSAwLS44LS40LS43LTEuNC0uOGgtLjVWNC4xaC45YzIuMiAwIDMuNS43IDMuNSAyLjMgMCAuNC0uMS45LS4xIDEuMy0uMS41LS4xLjktLjEgMS4zIDAgLjUuMiAxIDEuNyAxdjEuOHpNMS44IDEwLjFjMS42IDAgMS43LS41IDEuNy0xIDAtLjQtLjEtLjktLjEtMS4zLS4xLS41LS4xLS45LS4xLTEuMyAwLTEuNiAxLjQtMi4zIDMuNS0yLjNoLjl2MS45aC0uNWMtMSAwLTEuNCAwLTEuNC44IDAgLjMgMCAuNi4xIDEgMCAuMi4xLjYuMSAxIDAgMS4zIDAgMS44LTEuMyAyQzYgMTEuMiA2IDExLjcgNiAxM2MwIC40LS4xLjgtLjEgMS4yLS4xLjMtLjEuNy0uMSAxIDAgLjguMy44IDEuNC44aC41djEuOWgtLjljLTIuMSAwLTMuNS0uNi0zLjUtMi4zIDAtLjQuMS0uOS4xLTEuMy4xLS41LjEtLjkuMS0xLjMgMC0uNS0uMi0xLTEuNy0xdi0xLjl6Ii8+CiAgICA8Y2lyY2xlIGN4PSIxMSIgY3k9IjEzLjgiIHI9IjIuMSIvPgogICAgPGNpcmNsZSBjeD0iMTEiIGN5PSI4LjIiIHI9IjIuMSIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-julia: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDMyNSAzMDAiPgogIDxnIGNsYXNzPSJqcC1icmFuZDAganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjY2IzYzMzIj4KICAgIDxwYXRoIGQ9Ik0gMTUwLjg5ODQzOCAyMjUgQyAxNTAuODk4NDM4IDI2Ni40MjE4NzUgMTE3LjMyMDMxMiAzMDAgNzUuODk4NDM4IDMwMCBDIDM0LjQ3NjU2MiAzMDAgMC44OTg0MzggMjY2LjQyMTg3NSAwLjg5ODQzOCAyMjUgQyAwLjg5ODQzOCAxODMuNTc4MTI1IDM0LjQ3NjU2MiAxNTAgNzUuODk4NDM4IDE1MCBDIDExNy4zMjAzMTIgMTUwIDE1MC44OTg0MzggMTgzLjU3ODEyNSAxNTAuODk4NDM4IDIyNSIvPgogIDwvZz4KICA8ZyBjbGFzcz0ianAtYnJhbmQwIGpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iIzM4OTgyNiI+CiAgICA8cGF0aCBkPSJNIDIzNy41IDc1IEMgMjM3LjUgMTE2LjQyMTg3NSAyMDMuOTIxODc1IDE1MCAxNjIuNSAxNTAgQyAxMjEuMDc4MTI1IDE1MCA4Ny41IDExNi40MjE4NzUgODcuNSA3NSBDIDg3LjUgMzMuNTc4MTI1IDEyMS4wNzgxMjUgMCAxNjIuNSAwIEMgMjAzLjkyMTg3NSAwIDIzNy41IDMzLjU3ODEyNSAyMzcuNSA3NSIvPgogIDwvZz4KICA8ZyBjbGFzcz0ianAtYnJhbmQwIGpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iIzk1NThiMiI+CiAgICA8cGF0aCBkPSJNIDMyNC4xMDE1NjIgMjI1IEMgMzI0LjEwMTU2MiAyNjYuNDIxODc1IDI5MC41MjM0MzggMzAwIDI0OS4xMDE1NjIgMzAwIEMgMjA3LjY3OTY4OCAzMDAgMTc0LjEwMTU2MiAyNjYuNDIxODc1IDE3NC4xMDE1NjIgMjI1IEMgMTc0LjEwMTU2MiAxODMuNTc4MTI1IDIwNy42Nzk2ODggMTUwIDI0OS4xMDE1NjIgMTUwIEMgMjkwLjUyMzQzOCAxNTAgMzI0LjEwMTU2MiAxODMuNTc4MTI1IDMyNC4xMDE1NjIgMjI1Ii8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-jupyter-favicon: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUyIiBoZWlnaHQ9IjE2NSIgdmlld0JveD0iMCAwIDE1MiAxNjUiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgPGcgY2xhc3M9ImpwLWp1cHl0ZXItaWNvbi1jb2xvciIgZmlsbD0iI0YzNzcyNiI+CiAgICA8cGF0aCB0cmFuc2Zvcm09InRyYW5zbGF0ZSgwLjA3ODk0NywgMTEwLjU4MjkyNykiIGQ9Ik03NS45NDIyODQyLDI5LjU4MDQ1NjEgQzQzLjMwMjM5NDcsMjkuNTgwNDU2MSAxNC43OTY3ODMyLDE3LjY1MzQ2MzQgMCwwIEM1LjUxMDgzMjExLDE1Ljg0MDY4MjkgMTUuNzgxNTM4OSwyOS41NjY3NzMyIDI5LjM5MDQ5NDcsMzkuMjc4NDE3MSBDNDIuOTk5Nyw0OC45ODk4NTM3IDU5LjI3MzcsNTQuMjA2NzgwNSA3NS45NjA1Nzg5LDU0LjIwNjc4MDUgQzkyLjY0NzQ1NzksNTQuMjA2NzgwNSAxMDguOTIxNDU4LDQ4Ljk4OTg1MzcgMTIyLjUzMDY2MywzOS4yNzg0MTcxIEMxMzYuMTM5NDUzLDI5LjU2Njc3MzIgMTQ2LjQxMDI4NCwxNS44NDA2ODI5IDE1MS45MjExNTgsMCBDMTM3LjA4Nzg2OCwxNy42NTM0NjM0IDEwOC41ODI1ODksMjkuNTgwNDU2MSA3NS45NDIyODQyLDI5LjU4MDQ1NjEgTDc1Ljk0MjI4NDIsMjkuNTgwNDU2MSBaIiAvPgogICAgPHBhdGggdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMC4wMzczNjgsIDAuNzA0ODc4KSIgZD0iTTc1Ljk3ODQ1NzksMjQuNjI2NDA3MyBDMTA4LjYxODc2MywyNC42MjY0MDczIDEzNy4xMjQ0NTgsMzYuNTUzNDQxNSAxNTEuOTIxMTU4LDU0LjIwNjc4MDUgQzE0Ni40MTAyODQsMzguMzY2MjIyIDEzNi4xMzk0NTMsMjQuNjQwMTMxNyAxMjIuNTMwNjYzLDE0LjkyODQ4NzggQzEwOC45MjE0NTgsNS4yMTY4NDM5IDkyLjY0NzQ1NzksMCA3NS45NjA1Nzg5LDAgQzU5LjI3MzcsMCA0Mi45OTk3LDUuMjE2ODQzOSAyOS4zOTA0OTQ3LDE0LjkyODQ4NzggQzE1Ljc4MTUzODksMjQuNjQwMTMxNyA1LjUxMDgzMjExLDM4LjM2NjIyMiAwLDU0LjIwNjc4MDUgQzE0LjgzMzA4MTYsMzYuNTg5OTI5MyA0My4zMzg1Njg0LDI0LjYyNjQwNzMgNzUuOTc4NDU3OSwyNC42MjY0MDczIEw3NS45Nzg0NTc5LDI0LjYyNjQwNzMgWiIgLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-jupyter: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzkiIGhlaWdodD0iNTEiIHZpZXdCb3g9IjAgMCAzOSA1MSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMTYzOCAtMjI4MSkiPgogICAgIDxnIGNsYXNzPSJqcC1qdXB5dGVyLWljb24tY29sb3IiIGZpbGw9IiNGMzc3MjYiPgogICAgICA8cGF0aCB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNjM5Ljc0IDIzMTEuOTgpIiBkPSJNIDE4LjI2NDYgNy4xMzQxMUMgMTAuNDE0NSA3LjEzNDExIDMuNTU4NzIgNC4yNTc2IDAgMEMgMS4zMjUzOSAzLjgyMDQgMy43OTU1NiA3LjEzMDgxIDcuMDY4NiA5LjQ3MzAzQyAxMC4zNDE3IDExLjgxNTIgMTQuMjU1NyAxMy4wNzM0IDE4LjI2OSAxMy4wNzM0QyAyMi4yODIzIDEzLjA3MzQgMjYuMTk2MyAxMS44MTUyIDI5LjQ2OTQgOS40NzMwM0MgMzIuNzQyNCA3LjEzMDgxIDM1LjIxMjYgMy44MjA0IDM2LjUzOCAwQyAzMi45NzA1IDQuMjU3NiAyNi4xMTQ4IDcuMTM0MTEgMTguMjY0NiA3LjEzNDExWiIvPgogICAgICA8cGF0aCB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNjM5LjczIDIyODUuNDgpIiBkPSJNIDE4LjI3MzMgNS45MzkzMUMgMjYuMTIzNSA1LjkzOTMxIDMyLjk3OTMgOC44MTU4MyAzNi41MzggMTMuMDczNEMgMzUuMjEyNiA5LjI1MzAzIDMyLjc0MjQgNS45NDI2MiAyOS40Njk0IDMuNjAwNEMgMjYuMTk2MyAxLjI1ODE4IDIyLjI4MjMgMCAxOC4yNjkgMEMgMTQuMjU1NyAwIDEwLjM0MTcgMS4yNTgxOCA3LjA2ODYgMy42MDA0QyAzLjc5NTU2IDUuOTQyNjIgMS4zMjUzOSA5LjI1MzAzIDAgMTMuMDczNEMgMy41Njc0NSA4LjgyNDYzIDEwLjQyMzIgNS45MzkzMSAxOC4yNzMzIDUuOTM5MzFaIi8+CiAgICA8L2c+CiAgICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgICA8cGF0aCB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNjY5LjMgMjI4MS4zMSkiIGQ9Ik0gNS44OTM1MyAyLjg0NEMgNS45MTg4OSAzLjQzMTY1IDUuNzcwODUgNC4wMTM2NyA1LjQ2ODE1IDQuNTE2NDVDIDUuMTY1NDUgNS4wMTkyMiA0LjcyMTY4IDUuNDIwMTUgNC4xOTI5OSA1LjY2ODUxQyAzLjY2NDMgNS45MTY4OCAzLjA3NDQ0IDYuMDAxNTEgMi40OTgwNSA1LjkxMTcxQyAxLjkyMTY2IDUuODIxOSAxLjM4NDYzIDUuNTYxNyAwLjk1NDg5OCA1LjE2NDAxQyAwLjUyNTE3IDQuNzY2MzMgMC4yMjIwNTYgNC4yNDkwMyAwLjA4MzkwMzcgMy42Nzc1N0MgLTAuMDU0MjQ4MyAzLjEwNjExIC0wLjAyMTIzIDIuNTA2MTcgMC4xNzg3ODEgMS45NTM2NEMgMC4zNzg3OTMgMS40MDExIDAuNzM2ODA5IDAuOTIwODE3IDEuMjA3NTQgMC41NzM1MzhDIDEuNjc4MjYgMC4yMjYyNTkgMi4yNDA1NSAwLjAyNzU5MTkgMi44MjMyNiAwLjAwMjY3MjI5QyAzLjYwMzg5IC0wLjAzMDcxMTUgNC4zNjU3MyAwLjI0OTc4OSA0Ljk0MTQyIDAuNzgyNTUxQyA1LjUxNzExIDEuMzE1MzEgNS44NTk1NiAyLjA1Njc2IDUuODkzNTMgMi44NDRaIi8+CiAgICAgIDxwYXRoIHRyYW5zZm9ybT0idHJhbnNsYXRlKDE2MzkuOCAyMzIzLjgxKSIgZD0iTSA3LjQyNzg5IDMuNTgzMzhDIDcuNDYwMDggNC4zMjQzIDcuMjczNTUgNS4wNTgxOSA2Ljg5MTkzIDUuNjkyMTNDIDYuNTEwMzEgNi4zMjYwNyA1Ljk1MDc1IDYuODMxNTYgNS4yODQxMSA3LjE0NDZDIDQuNjE3NDcgNy40NTc2MyAzLjg3MzcxIDcuNTY0MTQgMy4xNDcwMiA3LjQ1MDYzQyAyLjQyMDMyIDcuMzM3MTIgMS43NDMzNiA3LjAwODcgMS4yMDE4NCA2LjUwNjk1QyAwLjY2MDMyOCA2LjAwNTIgMC4yNzg2MSA1LjM1MjY4IDAuMTA1MDE3IDQuNjMyMDJDIC0wLjA2ODU3NTcgMy45MTEzNSAtMC4wMjYyMzYxIDMuMTU0OTQgMC4yMjY2NzUgMi40NTg1NkMgMC40Nzk1ODcgMS43NjIxNyAwLjkzMTY5NyAxLjE1NzEzIDEuNTI1NzYgMC43MjAwMzNDIDIuMTE5ODMgMC4yODI5MzUgMi44MjkxNCAwLjAzMzQzOTUgMy41NjM4OSAwLjAwMzEzMzQ0QyA0LjU0NjY3IC0wLjAzNzQwMzMgNS41MDUyOSAwLjMxNjcwNiA2LjIyOTYxIDAuOTg3ODM1QyA2Ljk1MzkzIDEuNjU4OTYgNy4zODQ4NCAyLjU5MjM1IDcuNDI3ODkgMy41ODMzOEwgNy40Mjc4OSAzLjU4MzM4WiIvPgogICAgICA8cGF0aCB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNjM4LjM2IDIyODYuMDYpIiBkPSJNIDIuMjc0NzEgNC4zOTYyOUMgMS44NDM2MyA0LjQxNTA4IDEuNDE2NzEgNC4zMDQ0NSAxLjA0Nzk5IDQuMDc4NDNDIDAuNjc5MjY4IDMuODUyNCAwLjM4NTMyOCAzLjUyMTE0IDAuMjAzMzcxIDMuMTI2NTZDIDAuMDIxNDEzNiAyLjczMTk4IC0wLjA0MDM3OTggMi4yOTE4MyAwLjAyNTgxMTYgMS44NjE4MUMgMC4wOTIwMDMxIDEuNDMxOCAwLjI4MzIwNCAxLjAzMTI2IDAuNTc1MjEzIDAuNzEwODgzQyAwLjg2NzIyMiAwLjM5MDUxIDEuMjQ2OTEgMC4xNjQ3MDggMS42NjYyMiAwLjA2MjA1OTJDIDIuMDg1NTMgLTAuMDQwNTg5NyAyLjUyNTYxIC0wLjAxNTQ3MTQgMi45MzA3NiAwLjEzNDIzNUMgMy4zMzU5MSAwLjI4Mzk0MSAzLjY4NzkyIDAuNTUxNTA1IDMuOTQyMjIgMC45MDMwNkMgNC4xOTY1MiAxLjI1NDYyIDQuMzQxNjkgMS42NzQzNiA0LjM1OTM1IDIuMTA5MTZDIDQuMzgyOTkgMi42OTEwNyA0LjE3Njc4IDMuMjU4NjkgMy43ODU5NyAzLjY4NzQ2QyAzLjM5NTE2IDQuMTE2MjQgMi44NTE2NiA0LjM3MTE2IDIuMjc0NzEgNC4zOTYyOUwgMi4yNzQ3MSA0LjM5NjI5WiIvPgogICAgPC9nPgogIDwvZz4+Cjwvc3ZnPgo=);
  --jp-icon-jupyterlab-wordmark: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyMDAiIHZpZXdCb3g9IjAgMCAxODYwLjggNDc1Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjIiIGZpbGw9IiM0RTRFNEUiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDQ4MC4xMzY0MDEsIDY0LjI3MTQ5MykiPgogICAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMC4wMDAwMDAsIDU4Ljg3NTU2NikiPgogICAgICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSgwLjA4NzYwMywgMC4xNDAyOTQpIj4KICAgICAgICA8cGF0aCBkPSJNLTQyNi45LDE2OS44YzAsNDguNy0zLjcsNjQuNy0xMy42LDc2LjRjLTEwLjgsMTAtMjUsMTUuNS0zOS43LDE1LjVsMy43LDI5IGMyMi44LDAuMyw0NC44LTcuOSw2MS45LTIzLjFjMTcuOC0xOC41LDI0LTQ0LjEsMjQtODMuM1YwSC00Mjd2MTcwLjFMLTQyNi45LDE2OS44TC00MjYuOSwxNjkuOHoiLz4KICAgICAgPC9nPgogICAgPC9nPgogICAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMTU1LjA0NTI5NiwgNTYuODM3MTA0KSI+CiAgICAgIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEuNTYyNDUzLCAxLjc5OTg0MikiPgogICAgICAgIDxwYXRoIGQ9Ik0tMzEyLDE0OGMwLDIxLDAsMzkuNSwxLjcsNTUuNGgtMzEuOGwtMi4xLTMzLjNoLTAuOGMtNi43LDExLjYtMTYuNCwyMS4zLTI4LDI3LjkgYy0xMS42LDYuNi0yNC44LDEwLTM4LjIsOS44Yy0zMS40LDAtNjktMTcuNy02OS04OVYwaDM2LjR2MTEyLjdjMCwzOC43LDExLjYsNjQuNyw0NC42LDY0LjdjMTAuMy0wLjIsMjAuNC0zLjUsMjguOS05LjQgYzguNS01LjksMTUuMS0xNC4zLDE4LjktMjMuOWMyLjItNi4xLDMuMy0xMi41LDMuMy0xOC45VjAuMmgzNi40VjE0OEgtMzEyTC0zMTIsMTQ4eiIvPgogICAgICA8L2c+CiAgICA8L2c+CiAgICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSgzOTAuMDEzMzIyLCA1My40Nzk2MzgpIj4KICAgICAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMS43MDY0NTgsIDAuMjMxNDI1KSI+CiAgICAgICAgPHBhdGggZD0iTS00NzguNiw3MS40YzAtMjYtMC44LTQ3LTEuNy02Ni43aDMyLjdsMS43LDM0LjhoMC44YzcuMS0xMi41LDE3LjUtMjIuOCwzMC4xLTI5LjcgYzEyLjUtNywyNi43LTEwLjMsNDEtOS44YzQ4LjMsMCw4NC43LDQxLjcsODQuNywxMDMuM2MwLDczLjEtNDMuNywxMDkuMi05MSwxMDkuMmMtMTIuMSwwLjUtMjQuMi0yLjItMzUtNy44IGMtMTAuOC01LjYtMTkuOS0xMy45LTI2LjYtMjQuMmgtMC44VjI5MWgtMzZ2LTIyMEwtNDc4LjYsNzEuNEwtNDc4LjYsNzEuNHogTS00NDIuNiwxMjUuNmMwLjEsNS4xLDAuNiwxMC4xLDEuNywxNS4xIGMzLDEyLjMsOS45LDIzLjMsMTkuOCwzMS4xYzkuOSw3LjgsMjIuMSwxMi4xLDM0LjcsMTIuMWMzOC41LDAsNjAuNy0zMS45LDYwLjctNzguNWMwLTQwLjctMjEuMS03NS42LTU5LjUtNzUuNiBjLTEyLjksMC40LTI1LjMsNS4xLTM1LjMsMTMuNGMtOS45LDguMy0xNi45LDE5LjctMTkuNiwzMi40Yy0xLjUsNC45LTIuMywxMC0yLjUsMTUuMVYxMjUuNkwtNDQyLjYsMTI1LjZMLTQ0Mi42LDEyNS42eiIvPgogICAgICA8L2c+CiAgICA8L2c+CiAgICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSg2MDYuNzQwNzI2LCA1Ni44MzcxMDQpIj4KICAgICAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMC43NTEyMjYsIDEuOTg5Mjk5KSI+CiAgICAgICAgPHBhdGggZD0iTS00NDAuOCwwbDQzLjcsMTIwLjFjNC41LDEzLjQsOS41LDI5LjQsMTIuOCw0MS43aDAuOGMzLjctMTIuMiw3LjktMjcuNywxMi44LTQyLjQgbDM5LjctMTE5LjJoMzguNUwtMzQ2LjksMTQ1Yy0yNiw2OS43LTQzLjcsMTA1LjQtNjguNiwxMjcuMmMtMTIuNSwxMS43LTI3LjksMjAtNDQuNiwyMy45bC05LjEtMzEuMSBjMTEuNy0zLjksMjIuNS0xMC4xLDMxLjgtMTguMWMxMy4yLTExLjEsMjMuNy0yNS4yLDMwLjYtNDEuMmMxLjUtMi44LDIuNS01LjcsMi45LTguOGMtMC4zLTMuMy0xLjItNi42LTIuNS05LjdMLTQ4MC4yLDAuMSBoMzkuN0wtNDQwLjgsMEwtNDQwLjgsMHoiLz4KICAgICAgPC9nPgogICAgPC9nPgogICAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoODIyLjc0ODEwNCwgMC4wMDAwMDApIj4KICAgICAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMS40NjQwNTAsIDAuMzc4OTE0KSI+CiAgICAgICAgPHBhdGggZD0iTS00MTMuNywwdjU4LjNoNTJ2MjguMmgtNTJWMTk2YzAsMjUsNywzOS41LDI3LjMsMzkuNWM3LjEsMC4xLDE0LjItMC43LDIxLjEtMi41IGwxLjcsMjcuN2MtMTAuMywzLjctMjEuMyw1LjQtMzIuMiw1Yy03LjMsMC40LTE0LjYtMC43LTIxLjMtMy40Yy02LjgtMi43LTEyLjktNi44LTE3LjktMTIuMWMtMTAuMy0xMC45LTE0LjEtMjktMTQuMS01Mi45IFY4Ni41aC0zMVY1OC4zaDMxVjkuNkwtNDEzLjcsMEwtNDEzLjcsMHoiLz4KICAgICAgPC9nPgogICAgPC9nPgogICAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoOTc0LjQzMzI4NiwgNTMuNDc5NjM4KSI+CiAgICAgIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKDAuOTkwMDM0LCAwLjYxMDMzOSkiPgogICAgICAgIDxwYXRoIGQ9Ik0tNDQ1LjgsMTEzYzAuOCw1MCwzMi4yLDcwLjYsNjguNiw3MC42YzE5LDAuNiwzNy45LTMsNTUuMy0xMC41bDYuMiwyNi40IGMtMjAuOSw4LjktNDMuNSwxMy4xLTY2LjIsMTIuNmMtNjEuNSwwLTk4LjMtNDEuMi05OC4zLTEwMi41Qy00ODAuMiw0OC4yLTQ0NC43LDAtMzg2LjUsMGM2NS4yLDAsODIuNyw1OC4zLDgyLjcsOTUuNyBjLTAuMSw1LjgtMC41LDExLjUtMS4yLDE3LjJoLTE0MC42SC00NDUuOEwtNDQ1LjgsMTEzeiBNLTMzOS4yLDg2LjZjMC40LTIzLjUtOS41LTYwLjEtNTAuNC02MC4xIGMtMzYuOCwwLTUyLjgsMzQuNC01NS43LDYwLjFILTMzOS4yTC0zMzkuMiw4Ni42TC0zMzkuMiw4Ni42eiIvPgogICAgICA8L2c+CiAgICA8L2c+CiAgICA8ZyB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxMjAxLjk2MTA1OCwgNTMuNDc5NjM4KSI+CiAgICAgIDxnIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEuMTc5NjQwLCAwLjcwNTA2OCkiPgogICAgICAgIDxwYXRoIGQ9Ik0tNDc4LjYsNjhjMC0yMy45LTAuNC00NC41LTEuNy02My40aDMxLjhsMS4yLDM5LjloMS43YzkuMS0yNy4zLDMxLTQ0LjUsNTUuMy00NC41IGMzLjUtMC4xLDcsMC40LDEwLjMsMS4ydjM0LjhjLTQuMS0wLjktOC4yLTEuMy0xMi40LTEuMmMtMjUuNiwwLTQzLjcsMTkuNy00OC43LDQ3LjRjLTEsNS43LTEuNiwxMS41LTEuNywxNy4ydjEwOC4zaC0zNlY2OCBMLTQ3OC42LDY4eiIvPgogICAgICA8L2c+CiAgICA8L2c+CiAgPC9nPgoKICA8ZyBjbGFzcz0ianAtaWNvbi13YXJuMCIgZmlsbD0iI0YzNzcyNiI+CiAgICA8cGF0aCBkPSJNMTM1Mi4zLDMyNi4yaDM3VjI4aC0zN1YzMjYuMnogTTE2MDQuOCwzMjYuMmMtMi41LTEzLjktMy40LTMxLjEtMy40LTQ4Ljd2LTc2IGMwLTQwLjctMTUuMS04My4xLTc3LjMtODMuMWMtMjUuNiwwLTUwLDcuMS02Ni44LDE4LjFsOC40LDI0LjRjMTQuMy05LjIsMzQtMTUuMSw1My0xNS4xYzQxLjYsMCw0Ni4yLDMwLjIsNDYuMiw0N3Y0LjIgYy03OC42LTAuNC0xMjIuMywyNi41LTEyMi4zLDc1LjZjMCwyOS40LDIxLDU4LjQsNjIuMiw1OC40YzI5LDAsNTAuOS0xNC4zLDYyLjItMzAuMmgxLjNsMi45LDI1LjZIMTYwNC44eiBNMTU2NS43LDI1Ny43IGMwLDMuOC0wLjgsOC0yLjEsMTEuOGMtNS45LDE3LjItMjIuNywzNC00OS4yLDM0Yy0xOC45LDAtMzQuOS0xMS4zLTM0LjktMzUuM2MwLTM5LjUsNDUuOC00Ni42LDg2LjItNDUuOFYyNTcuN3ogTTE2OTguNSwzMjYuMiBsMS43LTMzLjZoMS4zYzE1LjEsMjYuOSwzOC43LDM4LjIsNjguMSwzOC4yYzQ1LjQsMCw5MS4yLTM2LjEsOTEuMi0xMDguOGMwLjQtNjEuNy0zNS4zLTEwMy43LTg1LjctMTAzLjcgYy0zMi44LDAtNTYuMywxNC43LTY5LjMsMzcuNGgtMC44VjI4aC0zNi42djI0NS43YzAsMTguMS0wLjgsMzguNi0xLjcsNTIuNUgxNjk4LjV6IE0xNzA0LjgsMjA4LjJjMC01LjksMS4zLTEwLjksMi4xLTE1LjEgYzcuNi0yOC4xLDMxLjEtNDUuNCw1Ni4zLTQ1LjRjMzkuNSwwLDYwLjUsMzQuOSw2MC41LDc1LjZjMCw0Ni42LTIzLjEsNzguMS02MS44LDc4LjFjLTI2LjksMC00OC4zLTE3LjYtNTUuNS00My4zIGMtMC44LTQuMi0xLjctOC44LTEuNy0xMy40VjIwOC4yeiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-kernel: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICAgIDxwYXRoIGNsYXNzPSJqcC1pY29uMiIgZmlsbD0iIzYxNjE2MSIgZD0iTTE1IDlIOXY2aDZWOXptLTIgNGgtMnYtMmgydjJ6bTgtMlY5aC0yVjdjMC0xLjEtLjktMi0yLTJoLTJWM2gtMnYyaC0yVjNIOXYySDdjLTEuMSAwLTIgLjktMiAydjJIM3YyaDJ2MkgzdjJoMnYyYzAgMS4xLjkgMiAyIDJoMnYyaDJ2LTJoMnYyaDJ2LTJoMmMxLjEgMCAyLS45IDItMnYtMmgydi0yaC0ydi0yaDJ6bS00IDZIN1Y3aDEwdjEweiIvPgo8L3N2Zz4K);
  --jp-icon-keyboard: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8cGF0aCBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIiBkPSJNMjAgNUg0Yy0xLjEgMC0xLjk5LjktMS45OSAyTDIgMTdjMCAxLjEuOSAyIDIgMmgxNmMxLjEgMCAyLS45IDItMlY3YzAtMS4xLS45LTItMi0yem0tOSAzaDJ2MmgtMlY4em0wIDNoMnYyaC0ydi0yek04IDhoMnYySDhWOHptMCAzaDJ2Mkg4di0yem0tMSAySDV2LTJoMnYyem0wLTNINVY4aDJ2MnptOSA3SDh2LTJoOHYyem0wLTRoLTJ2LTJoMnYyem0wLTNoLTJWOGgydjJ6bTMgM2gtMnYtMmgydjJ6bTAtM2gtMlY4aDJ2MnoiLz4KPC9zdmc+Cg==);
  --jp-icon-launch: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMzIgMzIiIHdpZHRoPSIzMiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIj4KICAgIDxwYXRoIGQ9Ik0yNiwyOEg2YTIuMDAyNywyLjAwMjcsMCwwLDEtMi0yVjZBMi4wMDI3LDIuMDAyNywwLDAsMSw2LDRIMTZWNkg2VjI2SDI2VjE2aDJWMjZBMi4wMDI3LDIuMDAyNywwLDAsMSwyNiwyOFoiLz4KICAgIDxwb2x5Z29uIHBvaW50cz0iMjAgMiAyMCA0IDI2LjU4NiA0IDE4IDEyLjU4NiAxOS40MTQgMTQgMjggNS40MTQgMjggMTIgMzAgMTIgMzAgMiAyMCAyIi8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-launcher: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8cGF0aCBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIiBkPSJNMTkgMTlINVY1aDdWM0g1YTIgMiAwIDAwLTIgMnYxNGEyIDIgMCAwMDIgMmgxNGMxLjEgMCAyLS45IDItMnYtN2gtMnY3ek0xNCAzdjJoMy41OWwtOS44MyA5LjgzIDEuNDEgMS40MUwxOSA2LjQxVjEwaDJWM2gtN3oiLz4KPC9zdmc+Cg==);
  --jp-icon-line-form: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICAgIDxwYXRoIGZpbGw9IndoaXRlIiBkPSJNNS44OCA0LjEyTDEzLjc2IDEybC03Ljg4IDcuODhMOCAyMmwxMC0xMEw4IDJ6Ii8+Cjwvc3ZnPgo=);
  --jp-icon-link: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTMuOSAxMmMwLTEuNzEgMS4zOS0zLjEgMy4xLTMuMWg0VjdIN2MtMi43NiAwLTUgMi4yNC01IDVzMi4yNCA1IDUgNWg0di0xLjlIN2MtMS43MSAwLTMuMS0xLjM5LTMuMS0zLjF6TTggMTNoOHYtMkg4djJ6bTktNmgtNHYxLjloNGMxLjcxIDAgMy4xIDEuMzkgMy4xIDMuMXMtMS4zOSAzLjEtMy4xIDMuMWgtNFYxN2g0YzIuNzYgMCA1LTIuMjQgNS01cy0yLjI0LTUtNS01eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-list: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICAgIDxwYXRoIGNsYXNzPSJqcC1pY29uMiBqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiM2MTYxNjEiIGQ9Ik0xOSA1djE0SDVWNWgxNG0xLjEtMkgzLjljLS41IDAtLjkuNC0uOS45djE2LjJjMCAuNC40LjkuOS45aDE2LjJjLjQgMCAuOS0uNS45LS45VjMuOWMwLS41LS41LS45LS45LS45ek0xMSA3aDZ2MmgtNlY3em0wIDRoNnYyaC02di0yem0wIDRoNnYyaC02ek03IDdoMnYySDd6bTAgNGgydjJIN3ptMCA0aDJ2Mkg3eiIvPgo8L3N2Zz4K);
  --jp-icon-markdown: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8cGF0aCBjbGFzcz0ianAtaWNvbi1jb250cmFzdDAganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjN0IxRkEyIiBkPSJNNSAxNC45aDEybC02LjEgNnptOS40LTYuOGMwLTEuMy0uMS0yLjktLjEtNC41LS40IDEuNC0uOSAyLjktMS4zIDQuM2wtMS4zIDQuM2gtMkw4LjUgNy45Yy0uNC0xLjMtLjctMi45LTEtNC4zLS4xIDEuNi0uMSAzLjItLjIgNC42TDcgMTIuNEg0LjhsLjctMTFoMy4zTDEwIDVjLjQgMS4yLjcgMi43IDEgMy45LjMtMS4yLjctMi42IDEtMy45bDEuMi0zLjdoMy4zbC42IDExaC0yLjRsLS4zLTQuMnoiLz4KPC9zdmc+Cg==);
  --jp-icon-move-down: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQiIGhlaWdodD0iMTQiIHZpZXdCb3g9IjAgMCAxNCAxNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggY2xhc3M9ImpwLWljb24zIiBkPSJNMTIuNDcxIDcuNTI4OTlDMTIuNzYzMiA3LjIzNjg0IDEyLjc2MzIgNi43NjMxNiAxMi40NzEgNi40NzEwMVY2LjQ3MTAxQzEyLjE3OSA2LjE3OTA1IDExLjcwNTcgNi4xNzg4NCAxMS40MTM1IDYuNDcwNTRMNy43NSAxMC4xMjc1VjEuNzVDNy43NSAxLjMzNTc5IDcuNDE0MjEgMSA3IDFWMUM2LjU4NTc5IDEgNi4yNSAxLjMzNTc5IDYuMjUgMS43NVYxMC4xMjc1TDIuNTk3MjYgNi40NjgyMkMyLjMwMzM4IDYuMTczODEgMS44MjY0MSA2LjE3MzU5IDEuNTMyMjYgNi40Njc3NFY2LjQ2Nzc0QzEuMjM4MyA2Ljc2MTcgMS4yMzgzIDcuMjM4MyAxLjUzMjI2IDcuNTMyMjZMNi4yOTI4OSAxMi4yOTI5QzYuNjgzNDIgMTIuNjgzNCA3LjMxNjU4IDEyLjY4MzQgNy43MDcxMSAxMi4yOTI5TDEyLjQ3MSA3LjUyODk5WiIgZmlsbD0iIzYxNjE2MSIvPgo8L3N2Zz4K);
  --jp-icon-move-up: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTQiIGhlaWdodD0iMTQiIHZpZXdCb3g9IjAgMCAxNCAxNCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggY2xhc3M9ImpwLWljb24zIiBkPSJNMS41Mjg5OSA2LjQ3MTAxQzEuMjM2ODQgNi43NjMxNiAxLjIzNjg0IDcuMjM2ODQgMS41Mjg5OSA3LjUyODk5VjcuNTI4OTlDMS44MjA5NSA3LjgyMDk1IDIuMjk0MjYgNy44MjExNiAyLjU4NjQ5IDcuNTI5NDZMNi4yNSAzLjg3MjVWMTIuMjVDNi4yNSAxMi42NjQyIDYuNTg1NzkgMTMgNyAxM1YxM0M3LjQxNDIxIDEzIDcuNzUgMTIuNjY0MiA3Ljc1IDEyLjI1VjMuODcyNUwxMS40MDI3IDcuNTMxNzhDMTEuNjk2NiA3LjgyNjE5IDEyLjE3MzYgNy44MjY0MSAxMi40Njc3IDcuNTMyMjZWNy41MzIyNkMxMi43NjE3IDcuMjM4MyAxMi43NjE3IDYuNzYxNyAxMi40Njc3IDYuNDY3NzRMNy43MDcxMSAxLjcwNzExQzcuMzE2NTggMS4zMTY1OCA2LjY4MzQyIDEuMzE2NTggNi4yOTI4OSAxLjcwNzExTDEuNTI4OTkgNi40NzEwMVoiIGZpbGw9IiM2MTYxNjEiLz4KPC9zdmc+Cg==);
  --jp-icon-new-folder: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTIwIDZoLThsLTItMkg0Yy0xLjExIDAtMS45OS44OS0xLjk5IDJMMiAxOGMwIDEuMTEuODkgMiAyIDJoMTZjMS4xMSAwIDItLjg5IDItMlY4YzAtMS4xMS0uODktMi0yLTJ6bS0xIDhoLTN2M2gtMnYtM2gtM3YtMmgzVjloMnYzaDN2MnoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-not-trusted: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSJub25lIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI1IDI1Ij4KICAgIDxwYXRoIGNsYXNzPSJqcC1pY29uMiIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utd2lkdGg9IjIiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDMgMykiIGQ9Ik0xLjg2MDk0IDExLjQ0MDlDMC44MjY0NDggOC43NzAyNyAwLjg2Mzc3OSA2LjA1NzY0IDEuMjQ5MDcgNC4xOTkzMkMyLjQ4MjA2IDMuOTMzNDcgNC4wODA2OCAzLjQwMzQ3IDUuNjAxMDIgMi44NDQ5QzcuMjM1NDkgMi4yNDQ0IDguODU2NjYgMS41ODE1IDkuOTg3NiAxLjA5NTM5QzExLjA1OTcgMS41ODM0MSAxMi42MDk0IDIuMjQ0NCAxNC4yMTggMi44NDMzOUMxNS43NTAzIDMuNDEzOTQgMTcuMzk5NSAzLjk1MjU4IDE4Ljc1MzkgNC4yMTM4NUMxOS4xMzY0IDYuMDcxNzcgMTkuMTcwOSA4Ljc3NzIyIDE4LjEzOSAxMS40NDA5QzE3LjAzMDMgMTQuMzAzMiAxNC42NjY4IDE3LjE4NDQgOS45OTk5OSAxOC45MzU0QzUuMzMzMTkgMTcuMTg0NCAyLjk2OTY4IDE0LjMwMzIgMS44NjA5NCAxMS40NDA5WiIvPgogICAgPHBhdGggY2xhc3M9ImpwLWljb24yIiBzdHJva2U9IiMzMzMzMzMiIHN0cm9rZS13aWR0aD0iMiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoOS4zMTU5MiA5LjMyMDMxKSIgZD0iTTcuMzY4NDIgMEwwIDcuMzY0NzkiLz4KICAgIDxwYXRoIGNsYXNzPSJqcC1pY29uMiIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utd2lkdGg9IjIiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDkuMzE1OTIgMTYuNjgzNikgc2NhbGUoMSAtMSkiIGQ9Ik03LjM2ODQyIDBMMCA3LjM2NDc5Ii8+Cjwvc3ZnPgo=);
  --jp-icon-notebook: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8ZyBjbGFzcz0ianAtbm90ZWJvb2staWNvbi1jb2xvciBqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiNFRjZDMDAiPgogICAgPHBhdGggZD0iTTE4LjcgMy4zdjE1LjRIMy4zVjMuM2gxNS40bTEuNS0xLjVIMS44djE4LjNoMTguM2wuMS0xOC4zeiIvPgogICAgPHBhdGggZD0iTTE2LjUgMTYuNWwtNS40LTQuMy01LjYgNC4zdi0xMWgxMXoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-numbering: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjIiIGhlaWdodD0iMjIiIHZpZXdCb3g9IjAgMCAyOCAyOCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KCTxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CgkJPHBhdGggZD0iTTQgMTlINlYxOS41SDVWMjAuNUg2VjIxSDRWMjJIN1YxOEg0VjE5Wk01IDEwSDZWNkg0VjdINVYxMFpNNCAxM0g1LjhMNCAxNS4xVjE2SDdWMTVINS4yTDcgMTIuOVYxMkg0VjEzWk05IDdWOUgyM1Y3SDlaTTkgMjFIMjNWMTlIOVYyMVpNOSAxNUgyM1YxM0g5VjE1WiIvPgoJPC9nPgo8L3N2Zz4K);
  --jp-icon-offline-bolt: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjE2Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTEyIDIuMDJjLTUuNTEgMC05Ljk4IDQuNDctOS45OCA5Ljk4czQuNDcgOS45OCA5Ljk4IDkuOTggOS45OC00LjQ3IDkuOTgtOS45OFMxNy41MSAyLjAyIDEyIDIuMDJ6TTExLjQ4IDIwdi02LjI2SDhMMTMgNHY2LjI2aDMuMzVMMTEuNDggMjB6Ii8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-palette: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTE4IDEzVjIwSDRWNkg5LjAyQzkuMDcgNS4yOSA5LjI0IDQuNjIgOS41IDRINEMyLjkgNCAyIDQuOSAyIDZWMjBDMiAyMS4xIDIuOSAyMiA0IDIySDE4QzE5LjEgMjIgMjAgMjEuMSAyMCAyMFYxNUwxOCAxM1pNMTkuMyA4Ljg5QzE5Ljc0IDguMTkgMjAgNy4zOCAyMCA2LjVDMjAgNC4wMSAxNy45OSAyIDE1LjUgMkMxMy4wMSAyIDExIDQuMDEgMTEgNi41QzExIDguOTkgMTMuMDEgMTEgMTUuNDkgMTFDMTYuMzcgMTEgMTcuMTkgMTAuNzQgMTcuODggMTAuM0wyMSAxMy40MkwyMi40MiAxMkwxOS4zIDguODlaTTE1LjUgOUMxNC4xMiA5IDEzIDcuODggMTMgNi41QzEzIDUuMTIgMTQuMTIgNCAxNS41IDRDMTYuODggNCAxOCA1LjEyIDE4IDYuNUMxOCA3Ljg4IDE2Ljg4IDkgMTUuNSA5WiIvPgogICAgPHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik00IDZIOS4wMTg5NEM5LjAwNjM5IDYuMTY1MDIgOSA2LjMzMTc2IDkgNi41QzkgOC44MTU3NyAxMC4yMTEgMTAuODQ4NyAxMi4wMzQzIDEySDlWMTRIMTZWMTIuOTgxMUMxNi41NzAzIDEyLjkzNzcgMTcuMTIgMTIuODIwNyAxNy42Mzk2IDEyLjYzOTZMMTggMTNWMjBINFY2Wk04IDhINlYxMEg4VjhaTTYgMTJIOFYxNEg2VjEyWk04IDE2SDZWMThIOFYxNlpNOSAxNkgxNlYxOEg5VjE2WiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-paste: url(data:image/svg+xml;base64,PHN2ZyBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTE5IDJoLTQuMThDMTQuNC44NCAxMy4zIDAgMTIgMGMtMS4zIDAtMi40Ljg0LTIuODIgMkg1Yy0xLjEgMC0yIC45LTIgMnYxNmMwIDEuMS45IDIgMiAyaDE0YzEuMSAwIDItLjkgMi0yVjRjMC0xLjEtLjktMi0yLTJ6bS03IDBjLjU1IDAgMSAuNDUgMSAxcy0uNDUgMS0xIDEtMS0uNDUtMS0xIC40NS0xIDEtMXptNyAxOEg1VjRoMnYzaDEwVjRoMnYxNnoiLz4KICAgIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-pdf: url(data:image/svg+xml;base64,PHN2ZwogICB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyMiAyMiIgd2lkdGg9IjE2Ij4KICAgIDxwYXRoIHRyYW5zZm9ybT0icm90YXRlKDQ1KSIgY2xhc3M9ImpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iI0ZGMkEyQSIKICAgICAgIGQ9Im0gMjIuMzQ0MzY5LC0zLjAxNjM2NDIgaCA1LjYzODYwNCB2IDEuNTc5MjQzMyBoIC0zLjU0OTIyNyB2IDEuNTA4NjkyOTkgaCAzLjMzNzU3NiBWIDEuNjUwODE1NCBoIC0zLjMzNzU3NiB2IDMuNDM1MjYxMyBoIC0yLjA4OTM3NyB6IG0gLTcuMTM2NDQ0LDEuNTc5MjQzMyB2IDQuOTQzOTU0MyBoIDAuNzQ4OTIgcSAxLjI4MDc2MSwwIDEuOTUzNzAzLC0wLjYzNDk1MzUgMC42NzgzNjksLTAuNjM0OTUzNSAwLjY3ODM2OSwtMS44NDUxNjQxIDAsLTEuMjA0NzgzNTUgLTAuNjcyOTQyLC0xLjgzNDMxMDExIC0wLjY3Mjk0MiwtMC42Mjk1MjY1OSAtMS45NTkxMywtMC42Mjk1MjY1OSB6IG0gLTIuMDg5Mzc3LC0xLjU3OTI0MzMgaCAyLjIwMzM0MyBxIDEuODQ1MTY0LDAgMi43NDYwMzksMC4yNjU5MjA3IDAuOTA2MzAxLDAuMjYwNDkzNyAxLjU1MjEwOCwwLjg5MDAyMDMgMC41Njk4MywwLjU0ODEyMjMgMC44NDY2MDUsMS4yNjQ0ODAwNiAwLjI3Njc3NCwwLjcxNjM1NzgxIDAuMjc2Nzc0LDEuNjIyNjU4OTQgMCwwLjkxNzE1NTEgLTAuMjc2Nzc0LDEuNjM4OTM5OSAtMC4yNzY3NzUsMC43MTYzNTc4IC0wLjg0NjYwNSwxLjI2NDQ4IC0wLjY1MTIzNCwwLjYyOTUyNjYgLTEuNTYyOTYyLDAuODk1NDQ3MyAtMC45MTE3MjgsMC4yNjA0OTM3IC0yLjczNTE4NSwwLjI2MDQ5MzcgaCAtMi4yMDMzNDMgeiBtIC04LjE0NTg1NjUsMCBoIDMuNDY3ODIzIHEgMS41NDY2ODE2LDAgMi4zNzE1Nzg1LDAuNjg5MjIzIDAuODMwMzI0LDAuNjgzNzk2MSAwLjgzMDMyNCwxLjk1MzcwMzE0IDAsMS4yNzUzMzM5NyAtMC44MzAzMjQsMS45NjQ1NTcwNiBRIDkuOTg3MTk2MSwyLjI3NDkxNSA4LjQ0MDUxNDUsMi4yNzQ5MTUgSCA3LjA2MjA2ODQgViA1LjA4NjA3NjcgSCA0Ljk3MjY5MTUgWiBtIDIuMDg5Mzc2OSwxLjUxNDExOTkgdiAyLjI2MzAzOTQzIGggMS4xNTU5NDEgcSAwLjYwNzgxODgsMCAwLjkzODg2MjksLTAuMjkzMDU1NDcgMC4zMzEwNDQxLC0wLjI5ODQ4MjQxIDAuMzMxMDQ0MSwtMC44NDExNzc3MiAwLC0wLjU0MjY5NTMxIC0wLjMzMTA0NDEsLTAuODM1NzUwNzQgLTAuMzMxMDQ0MSwtMC4yOTMwNTU1IC0wLjkzODg2MjksLTAuMjkzMDU1NSB6IgovPgo8L3N2Zz4K);
  --jp-icon-python: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iLTEwIC0xMCAxMzEuMTYxMzYxNjk0MzM1OTQgMTMyLjM4ODk5OTkzODk2NDg0Ij4KICA8cGF0aCBjbGFzcz0ianAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjMzA2OTk4IiBkPSJNIDU0LjkxODc4NSw5LjE5Mjc0MjFlLTQgQyA1MC4zMzUxMzIsMC4wMjIyMTcyNyA0NS45NTc4NDYsMC40MTMxMzY5NyA0Mi4xMDYyODUsMS4wOTQ2NjkzIDMwLjc2MDA2OSwzLjA5OTE3MzEgMjguNzAwMDM2LDcuMjk0NzcxNCAyOC43MDAwMzUsMTUuMDMyMTY5IHYgMTAuMjE4NzUgaCAyNi44MTI1IHYgMy40MDYyNSBoIC0yNi44MTI1IC0xMC4wNjI1IGMgLTcuNzkyNDU5LDAgLTE0LjYxNTc1ODgsNC42ODM3MTcgLTE2Ljc0OTk5OTgsMTMuNTkzNzUgLTIuNDYxODE5OTgsMTAuMjEyOTY2IC0yLjU3MTAxNTA4LDE2LjU4NjAyMyAwLDI3LjI1IDEuOTA1OTI4Myw3LjkzNzg1MiA2LjQ1NzU0MzIsMTMuNTkzNzQ4IDE0LjI0OTk5OTgsMTMuNTkzNzUgaCA5LjIxODc1IHYgLTEyLjI1IGMgMCwtOC44NDk5MDIgNy42NTcxNDQsLTE2LjY1NjI0OCAxNi43NSwtMTYuNjU2MjUgaCAyNi43ODEyNSBjIDcuNDU0OTUxLDAgMTMuNDA2MjUzLC02LjEzODE2NCAxMy40MDYyNSwtMTMuNjI1IHYgLTI1LjUzMTI1IGMgMCwtNy4yNjYzMzg2IC02LjEyOTk4LC0xMi43MjQ3NzcxIC0xMy40MDYyNSwtMTMuOTM3NDk5NyBDIDY0LjI4MTU0OCwwLjMyNzk0Mzk3IDU5LjUwMjQzOCwtMC4wMjAzNzkwMyA1NC45MTg3ODUsOS4xOTI3NDIxZS00IFogbSAtMTQuNSw4LjIxODc1MDEyNTc5IGMgMi43Njk1NDcsMCA1LjAzMTI1LDIuMjk4NjQ1NiA1LjAzMTI1LDUuMTI0OTk5NiAtMmUtNiwyLjgxNjMzNiAtMi4yNjE3MDMsNS4wOTM3NSAtNS4wMzEyNSw1LjA5Mzc1IC0yLjc3OTQ3NiwtMWUtNiAtNS4wMzEyNSwtMi4yNzc0MTUgLTUuMDMxMjUsLTUuMDkzNzUgLTEwZS03LC0yLjgyNjM1MyAyLjI1MTc3NCwtNS4xMjQ5OTk2IDUuMDMxMjUsLTUuMTI0OTk5NiB6Ii8+CiAgPHBhdGggY2xhc3M9ImpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iI2ZmZDQzYiIgZD0ibSA4NS42Mzc1MzUsMjguNjU3MTY5IHYgMTEuOTA2MjUgYyAwLDkuMjMwNzU1IC03LjgyNTg5NSwxNi45OTk5OTkgLTE2Ljc1LDE3IGggLTI2Ljc4MTI1IGMgLTcuMzM1ODMzLDAgLTEzLjQwNjI0OSw2LjI3ODQ4MyAtMTMuNDA2MjUsMTMuNjI1IHYgMjUuNTMxMjQ3IGMgMCw3LjI2NjM0NCA2LjMxODU4OCwxMS41NDAzMjQgMTMuNDA2MjUsMTMuNjI1MDA0IDguNDg3MzMxLDIuNDk1NjEgMTYuNjI2MjM3LDIuOTQ2NjMgMjYuNzgxMjUsMCA2Ljc1MDE1NSwtMS45NTQzOSAxMy40MDYyNTMsLTUuODg3NjEgMTMuNDA2MjUsLTEzLjYyNTAwNCBWIDg2LjUwMDkxOSBoIC0yNi43ODEyNSB2IC0zLjQwNjI1IGggMjYuNzgxMjUgMTMuNDA2MjU0IGMgNy43OTI0NjEsMCAxMC42OTYyNTEsLTUuNDM1NDA4IDEzLjQwNjI0MSwtMTMuNTkzNzUgMi43OTkzMywtOC4zOTg4ODYgMi42ODAyMiwtMTYuNDc1Nzc2IDAsLTI3LjI1IC0xLjkyNTc4LC03Ljc1NzQ0MSAtNS42MDM4NywtMTMuNTkzNzUgLTEzLjQwNjI0MSwtMTMuNTkzNzUgeiBtIC0xNS4wNjI1LDY0LjY1NjI1IGMgMi43Nzk0NzgsM2UtNiA1LjAzMTI1LDIuMjc3NDE3IDUuMDMxMjUsNS4wOTM3NDcgLTJlLTYsMi44MjYzNTQgLTIuMjUxNzc1LDUuMTI1MDA0IC01LjAzMTI1LDUuMTI1MDA0IC0yLjc2OTU1LDAgLTUuMDMxMjUsLTIuMjk4NjUgLTUuMDMxMjUsLTUuMTI1MDA0IDJlLTYsLTIuODE2MzMgMi4yNjE2OTcsLTUuMDkzNzQ3IDUuMDMxMjUsLTUuMDkzNzQ3IHoiLz4KPC9zdmc+Cg==);
  --jp-icon-r-kernel: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8cGF0aCBjbGFzcz0ianAtaWNvbi1jb250cmFzdDMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjMjE5NkYzIiBkPSJNNC40IDIuNWMxLjItLjEgMi45LS4zIDQuOS0uMyAyLjUgMCA0LjEuNCA1LjIgMS4zIDEgLjcgMS41IDEuOSAxLjUgMy41IDAgMi0xLjQgMy41LTIuOSA0LjEgMS4yLjQgMS43IDEuNiAyLjIgMyAuNiAxLjkgMSAzLjkgMS4zIDQuNmgtMy44Yy0uMy0uNC0uOC0xLjctMS4yLTMuN3MtMS4yLTIuNi0yLjYtMi42aC0uOXY2LjRINC40VjIuNXptMy43IDYuOWgxLjRjMS45IDAgMi45LS45IDIuOS0yLjNzLTEtMi4zLTIuOC0yLjNjLS43IDAtMS4zIDAtMS42LjJ2NC41aC4xdi0uMXoiLz4KPC9zdmc+Cg==);
  --jp-icon-react: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMTUwIDE1MCA1NDEuOSAyOTUuMyI+CiAgPGcgY2xhc3M9ImpwLWljb24tYnJhbmQyIGpwLWljb24tc2VsZWN0YWJsZSIgZmlsbD0iIzYxREFGQiI+CiAgICA8cGF0aCBkPSJNNjY2LjMgMjk2LjVjMC0zMi41LTQwLjctNjMuMy0xMDMuMS04Mi40IDE0LjQtNjMuNiA4LTExNC4yLTIwLjItMTMwLjQtNi41LTMuOC0xNC4xLTUuNi0yMi40LTUuNnYyMi4zYzQuNiAwIDguMy45IDExLjQgMi42IDEzLjYgNy44IDE5LjUgMzcuNSAxNC45IDc1LjctMS4xIDkuNC0yLjkgMTkuMy01LjEgMjkuNC0xOS42LTQuOC00MS04LjUtNjMuNS0xMC45LTEzLjUtMTguNS0yNy41LTM1LjMtNDEuNi01MCAzMi42LTMwLjMgNjMuMi00Ni45IDg0LTQ2LjlWNzhjLTI3LjUgMC02My41IDE5LjYtOTkuOSA1My42LTM2LjQtMzMuOC03Mi40LTUzLjItOTkuOS01My4ydjIyLjNjMjAuNyAwIDUxLjQgMTYuNSA4NCA0Ni42LTE0IDE0LjctMjggMzEuNC00MS4zIDQ5LjktMjIuNiAyLjQtNDQgNi4xLTYzLjYgMTEtMi4zLTEwLTQtMTkuNy01LjItMjktNC43LTM4LjIgMS4xLTY3LjkgMTQuNi03NS44IDMtMS44IDYuOS0yLjYgMTEuNS0yLjZWNzguNWMtOC40IDAtMTYgMS44LTIyLjYgNS42LTI4LjEgMTYuMi0zNC40IDY2LjctMTkuOSAxMzAuMS02Mi4yIDE5LjItMTAyLjcgNDkuOS0xMDIuNyA4Mi4zIDAgMzIuNSA0MC43IDYzLjMgMTAzLjEgODIuNC0xNC40IDYzLjYtOCAxMTQuMiAyMC4yIDEzMC40IDYuNSAzLjggMTQuMSA1LjYgMjIuNSA1LjYgMjcuNSAwIDYzLjUtMTkuNiA5OS45LTUzLjYgMzYuNCAzMy44IDcyLjQgNTMuMiA5OS45IDUzLjIgOC40IDAgMTYtMS44IDIyLjYtNS42IDI4LjEtMTYuMiAzNC40LTY2LjcgMTkuOS0xMzAuMSA2Mi0xOS4xIDEwMi41LTQ5LjkgMTAyLjUtODIuM3ptLTEzMC4yLTY2LjdjLTMuNyAxMi45LTguMyAyNi4yLTEzLjUgMzkuNS00LjEtOC04LjQtMTYtMTMuMS0yNC00LjYtOC05LjUtMTUuOC0xNC40LTIzLjQgMTQuMiAyLjEgMjcuOSA0LjcgNDEgNy45em0tNDUuOCAxMDYuNWMtNy44IDEzLjUtMTUuOCAyNi4zLTI0LjEgMzguMi0xNC45IDEuMy0zMCAyLTQ1LjIgMi0xNS4xIDAtMzAuMi0uNy00NS0xLjktOC4zLTExLjktMTYuNC0yNC42LTI0LjItMzgtNy42LTEzLjEtMTQuNS0yNi40LTIwLjgtMzkuOCA2LjItMTMuNCAxMy4yLTI2LjggMjAuNy0zOS45IDcuOC0xMy41IDE1LjgtMjYuMyAyNC4xLTM4LjIgMTQuOS0xLjMgMzAtMiA0NS4yLTIgMTUuMSAwIDMwLjIuNyA0NSAxLjkgOC4zIDExLjkgMTYuNCAyNC42IDI0LjIgMzggNy42IDEzLjEgMTQuNSAyNi40IDIwLjggMzkuOC02LjMgMTMuNC0xMy4yIDI2LjgtMjAuNyAzOS45em0zMi4zLTEzYzUuNCAxMy40IDEwIDI2LjggMTMuOCAzOS44LTEzLjEgMy4yLTI2LjkgNS45LTQxLjIgOCA0LjktNy43IDkuOC0xNS42IDE0LjQtMjMuNyA0LjYtOCA4LjktMTYuMSAxMy0yNC4xek00MjEuMiA0MzBjLTkuMy05LjYtMTguNi0yMC4zLTI3LjgtMzIgOSAuNCAxOC4yLjcgMjcuNS43IDkuNCAwIDE4LjctLjIgMjcuOC0uNy05IDExLjctMTguMyAyMi40LTI3LjUgMzJ6bS03NC40LTU4LjljLTE0LjItMi4xLTI3LjktNC43LTQxLTcuOSAzLjctMTIuOSA4LjMtMjYuMiAxMy41LTM5LjUgNC4xIDggOC40IDE2IDEzLjEgMjQgNC43IDggOS41IDE1LjggMTQuNCAyMy40ek00MjAuNyAxNjNjOS4zIDkuNiAxOC42IDIwLjMgMjcuOCAzMi05LS40LTE4LjItLjctMjcuNS0uNy05LjQgMC0xOC43LjItMjcuOC43IDktMTEuNyAxOC4zLTIyLjQgMjcuNS0zMnptLTc0IDU4LjljLTQuOSA3LjctOS44IDE1LjYtMTQuNCAyMy43LTQuNiA4LTguOSAxNi0xMyAyNC01LjQtMTMuNC0xMC0yNi44LTEzLjgtMzkuOCAxMy4xLTMuMSAyNi45LTUuOCA0MS4yLTcuOXptLTkwLjUgMTI1LjJjLTM1LjQtMTUuMS01OC4zLTM0LjktNTguMy01MC42IDAtMTUuNyAyMi45LTM1LjYgNTguMy01MC42IDguNi0zLjcgMTgtNyAyNy43LTEwLjEgNS43IDE5LjYgMTMuMiA0MCAyMi41IDYwLjktOS4yIDIwLjgtMTYuNiA0MS4xLTIyLjIgNjAuNi05LjktMy4xLTE5LjMtNi41LTI4LTEwLjJ6TTMxMCA0OTBjLTEzLjYtNy44LTE5LjUtMzcuNS0xNC45LTc1LjcgMS4xLTkuNCAyLjktMTkuMyA1LjEtMjkuNCAxOS42IDQuOCA0MSA4LjUgNjMuNSAxMC45IDEzLjUgMTguNSAyNy41IDM1LjMgNDEuNiA1MC0zMi42IDMwLjMtNjMuMiA0Ni45LTg0IDQ2LjktNC41LS4xLTguMy0xLTExLjMtMi43em0yMzcuMi03Ni4yYzQuNyAzOC4yLTEuMSA2Ny45LTE0LjYgNzUuOC0zIDEuOC02LjkgMi42LTExLjUgMi42LTIwLjcgMC01MS40LTE2LjUtODQtNDYuNiAxNC0xNC43IDI4LTMxLjQgNDEuMy00OS45IDIyLjYtMi40IDQ0LTYuMSA2My42LTExIDIuMyAxMC4xIDQuMSAxOS44IDUuMiAyOS4xem0zOC41LTY2LjdjLTguNiAzLjctMTggNy0yNy43IDEwLjEtNS43LTE5LjYtMTMuMi00MC0yMi41LTYwLjkgOS4yLTIwLjggMTYuNi00MS4xIDIyLjItNjAuNiA5LjkgMy4xIDE5LjMgNi41IDI4LjEgMTAuMiAzNS40IDE1LjEgNTguMyAzNC45IDU4LjMgNTAuNi0uMSAxNS43LTIzIDM1LjYtNTguNCA1MC42ek0zMjAuOCA3OC40eiIvPgogICAgPGNpcmNsZSBjeD0iNDIwLjkiIGN5PSIyOTYuNSIgcj0iNDUuNyIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-redo: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjE2Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgICA8cGF0aCBkPSJNMCAwaDI0djI0SDB6IiBmaWxsPSJub25lIi8+PHBhdGggZD0iTTE4LjQgMTAuNkMxNi41NSA4Ljk5IDE0LjE1IDggMTEuNSA4Yy00LjY1IDAtOC41OCAzLjAzLTkuOTYgNy4yMkwzLjkgMTZjMS4wNS0zLjE5IDQuMDUtNS41IDcuNi01LjUgMS45NSAwIDMuNzMuNzIgNS4xMiAxLjg4TDEzIDE2aDlWN2wtMy42IDMuNnoiLz4KICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-refresh: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDE4IDE4Ij4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTkgMTMuNWMtMi40OSAwLTQuNS0yLjAxLTQuNS00LjVTNi41MSA0LjUgOSA0LjVjMS4yNCAwIDIuMzYuNTIgMy4xNyAxLjMzTDEwIDhoNVYzbC0xLjc2IDEuNzZDMTIuMTUgMy42OCAxMC42NiAzIDkgMyA1LjY5IDMgMy4wMSA1LjY5IDMuMDEgOVM1LjY5IDE1IDkgMTVjMi45NyAwIDUuNDMtMi4xNiA1LjktNWgtMS41MmMtLjQ2IDItMi4yNCAzLjUtNC4zOCAzLjV6Ii8+CiAgICA8L2c+Cjwvc3ZnPgo=);
  --jp-icon-regex: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIwIDIwIj4KICA8ZyBjbGFzcz0ianAtaWNvbjIiIGZpbGw9IiM0MTQxNDEiPgogICAgPHJlY3QgeD0iMiIgeT0iMiIgd2lkdGg9IjE2IiBoZWlnaHQ9IjE2Ii8+CiAgPC9nPgoKICA8ZyBjbGFzcz0ianAtaWNvbi1hY2NlbnQyIiBmaWxsPSIjRkZGIj4KICAgIDxjaXJjbGUgY2xhc3M9InN0MiIgY3g9IjUuNSIgY3k9IjE0LjUiIHI9IjEuNSIvPgogICAgPHJlY3QgeD0iMTIiIHk9IjQiIGNsYXNzPSJzdDIiIHdpZHRoPSIxIiBoZWlnaHQ9IjgiLz4KICAgIDxyZWN0IHg9IjguNSIgeT0iNy41IiB0cmFuc2Zvcm09Im1hdHJpeCgwLjg2NiAtMC41IDAuNSAwLjg2NiAtMi4zMjU1IDcuMzIxOSkiIGNsYXNzPSJzdDIiIHdpZHRoPSI4IiBoZWlnaHQ9IjEiLz4KICAgIDxyZWN0IHg9IjEyIiB5PSI0IiB0cmFuc2Zvcm09Im1hdHJpeCgwLjUgLTAuODY2IDAuODY2IDAuNSAtMC42Nzc5IDE0LjgyNTIpIiBjbGFzcz0ic3QyIiB3aWR0aD0iMSIgaGVpZ2h0PSI4Ii8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-run: url(data:image/svg+xml;base64,PHN2ZyBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTggNXYxNGwxMS03eiIvPgogICAgPC9nPgo8L3N2Zz4K);
  --jp-icon-running: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDUxMiA1MTIiPgogIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICA8cGF0aCBkPSJNMjU2IDhDMTE5IDggOCAxMTkgOCAyNTZzMTExIDI0OCAyNDggMjQ4IDI0OC0xMTEgMjQ4LTI0OFMzOTMgOCAyNTYgOHptOTYgMzI4YzAgOC44LTcuMiAxNi0xNiAxNkgxNzZjLTguOCAwLTE2LTcuMi0xNi0xNlYxNzZjMC04LjggNy4yLTE2IDE2LTE2aDE2MGM4LjggMCAxNiA3LjIgMTYgMTZ2MTYweiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-save: url(data:image/svg+xml;base64,PHN2ZyBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTE3IDNINWMtMS4xMSAwLTIgLjktMiAydjE0YzAgMS4xLjg5IDIgMiAyaDE0YzEuMSAwIDItLjkgMi0yVjdsLTQtNHptLTUgMTZjLTEuNjYgMC0zLTEuMzQtMy0zczEuMzQtMyAzLTMgMyAxLjM0IDMgMy0xLjM0IDMtMyAzem0zLTEwSDVWNWgxMHY0eiIvPgogICAgPC9nPgo8L3N2Zz4K);
  --jp-icon-search: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMTggMTgiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTEyLjEsMTAuOWgtMC43bC0wLjItMC4yYzAuOC0wLjksMS4zLTIuMiwxLjMtMy41YzAtMy0yLjQtNS40LTUuNC01LjRTMS44LDQuMiwxLjgsNy4xczIuNCw1LjQsNS40LDUuNCBjMS4zLDAsMi41LTAuNSwzLjUtMS4zbDAuMiwwLjJ2MC43bDQuMSw0LjFsMS4yLTEuMkwxMi4xLDEwLjl6IE03LjEsMTAuOWMtMi4xLDAtMy43LTEuNy0zLjctMy43czEuNy0zLjcsMy43LTMuN3MzLjcsMS43LDMuNywzLjcgUzkuMiwxMC45LDcuMSwxMC45eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-settings: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8cGF0aCBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIiBkPSJNMTkuNDMgMTIuOThjLjA0LS4zMi4wNy0uNjQuMDctLjk4cy0uMDMtLjY2LS4wNy0uOThsMi4xMS0xLjY1Yy4xOS0uMTUuMjQtLjQyLjEyLS42NGwtMi0zLjQ2Yy0uMTItLjIyLS4zOS0uMy0uNjEtLjIybC0yLjQ5IDFjLS41Mi0uNC0xLjA4LS43My0xLjY5LS45OGwtLjM4LTIuNjVBLjQ4OC40ODggMCAwMDE0IDJoLTRjLS4yNSAwLS40Ni4xOC0uNDkuNDJsLS4zOCAyLjY1Yy0uNjEuMjUtMS4xNy41OS0xLjY5Ljk4bC0yLjQ5LTFjLS4yMy0uMDktLjQ5IDAtLjYxLjIybC0yIDMuNDZjLS4xMy4yMi0uMDcuNDkuMTIuNjRsMi4xMSAxLjY1Yy0uMDQuMzItLjA3LjY1LS4wNy45OHMuMDMuNjYuMDcuOThsLTIuMTEgMS42NWMtLjE5LjE1LS4yNC40Mi0uMTIuNjRsMiAzLjQ2Yy4xMi4yMi4zOS4zLjYxLjIybDIuNDktMWMuNTIuNCAxLjA4LjczIDEuNjkuOThsLjM4IDIuNjVjLjAzLjI0LjI0LjQyLjQ5LjQyaDRjLjI1IDAgLjQ2LS4xOC40OS0uNDJsLjM4LTIuNjVjLjYxLS4yNSAxLjE3LS41OSAxLjY5LS45OGwyLjQ5IDFjLjIzLjA5LjQ5IDAgLjYxLS4yMmwyLTMuNDZjLjEyLS4yMi4wNy0uNDktLjEyLS42NGwtMi4xMS0xLjY1ek0xMiAxNS41Yy0xLjkzIDAtMy41LTEuNTctMy41LTMuNXMxLjU3LTMuNSAzLjUtMy41IDMuNSAxLjU3IDMuNSAzLjUtMS41NyAzLjUtMy41IDMuNXoiLz4KPC9zdmc+Cg==);
  --jp-icon-share: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIHZpZXdCb3g9IjAgMCAyNCAyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTSAxOCAyIEMgMTYuMzU0OTkgMiAxNSAzLjM1NDk5MDQgMTUgNSBDIDE1IDUuMTkwOTUyOSAxNS4wMjE3OTEgNS4zNzcxMjI0IDE1LjA1NjY0MSA1LjU1ODU5MzggTCA3LjkyMTg3NSA5LjcyMDcwMzEgQyA3LjM5ODUzOTkgOS4yNzc4NTM5IDYuNzMyMDc3MSA5IDYgOSBDIDQuMzU0OTkwNCA5IDMgMTAuMzU0OTkgMyAxMiBDIDMgMTMuNjQ1MDEgNC4zNTQ5OTA0IDE1IDYgMTUgQyA2LjczMjA3NzEgMTUgNy4zOTg1Mzk5IDE0LjcyMjE0NiA3LjkyMTg3NSAxNC4yNzkyOTcgTCAxNS4wNTY2NDEgMTguNDM5NDUzIEMgMTUuMDIxNTU1IDE4LjYyMTUxNCAxNSAxOC44MDgzODYgMTUgMTkgQyAxNSAyMC42NDUwMSAxNi4zNTQ5OSAyMiAxOCAyMiBDIDE5LjY0NTAxIDIyIDIxIDIwLjY0NTAxIDIxIDE5IEMgMjEgMTcuMzU0OTkgMTkuNjQ1MDEgMTYgMTggMTYgQyAxNy4yNjc0OCAxNiAxNi42MDE1OTMgMTYuMjc5MzI4IDE2LjA3ODEyNSAxNi43MjI2NTYgTCA4Ljk0MzM1OTQgMTIuNTU4NTk0IEMgOC45NzgyMDk1IDEyLjM3NzEyMiA5IDEyLjE5MDk1MyA5IDEyIEMgOSAxMS44MDkwNDcgOC45NzgyMDk1IDExLjYyMjg3OCA4Ljk0MzM1OTQgMTEuNDQxNDA2IEwgMTYuMDc4MTI1IDcuMjc5Mjk2OSBDIDE2LjYwMTQ2IDcuNzIyMTQ2MSAxNy4yNjc5MjMgOCAxOCA4IEMgMTkuNjQ1MDEgOCAyMSA2LjY0NTAwOTYgMjEgNSBDIDIxIDMuMzU0OTkwNCAxOS42NDUwMSAyIDE4IDIgeiBNIDE4IDQgQyAxOC41NjQxMjkgNCAxOSA0LjQzNTg3MDYgMTkgNSBDIDE5IDUuNTY0MTI5NCAxOC41NjQxMjkgNiAxOCA2IEMgMTcuNDM1ODcxIDYgMTcgNS41NjQxMjk0IDE3IDUgQyAxNyA0LjQzNTg3MDYgMTcuNDM1ODcxIDQgMTggNCB6IE0gNiAxMSBDIDYuNTY0MTI5NCAxMSA3IDExLjQzNTg3MSA3IDEyIEMgNyAxMi41NjQxMjkgNi41NjQxMjk0IDEzIDYgMTMgQyA1LjQzNTg3MDYgMTMgNSAxMi41NjQxMjkgNSAxMiBDIDUgMTEuNDM1ODcxIDUuNDM1ODcwNiAxMSA2IDExIHogTSAxOCAxOCBDIDE4LjU2NDEyOSAxOCAxOSAxOC40MzU4NzEgMTkgMTkgQyAxOSAxOS41NjQxMjkgMTguNTY0MTI5IDIwIDE4IDIwIEMgMTcuNDM1ODcxIDIwIDE3IDE5LjU2NDEyOSAxNyAxOSBDIDE3IDE4LjQzNTg3MSAxNy40MzU4NzEgMTggMTggMTggeiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-spreadsheet: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8cGF0aCBjbGFzcz0ianAtaWNvbi1jb250cmFzdDEganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNENBRjUwIiBkPSJNMi4yIDIuMnYxNy42aDE3LjZWMi4ySDIuMnptMTUuNCA3LjdoLTUuNVY0LjRoNS41djUuNXpNOS45IDQuNHY1LjVINC40VjQuNGg1LjV6bS01LjUgNy43aDUuNXY1LjVINC40di01LjV6bTcuNyA1LjV2LTUuNWg1LjV2NS41aC01LjV6Ii8+Cjwvc3ZnPgo=);
  --jp-icon-stop: url(data:image/svg+xml;base64,PHN2ZyBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTAgMGgyNHYyNEgweiIgZmlsbD0ibm9uZSIvPgogICAgICAgIDxwYXRoIGQ9Ik02IDZoMTJ2MTJINnoiLz4KICAgIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-tab: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTIxIDNIM2MtMS4xIDAtMiAuOS0yIDJ2MTRjMCAxLjEuOSAyIDIgMmgxOGMxLjEgMCAyLS45IDItMlY1YzAtMS4xLS45LTItMi0yem0wIDE2SDNWNWgxMHY0aDh2MTB6Ii8+CiAgPC9nPgo8L3N2Zz4K);
  --jp-icon-table-rows: url(data:image/svg+xml;base64,PHN2ZyBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTAgMGgyNHYyNEgweiIgZmlsbD0ibm9uZSIvPgogICAgICAgIDxwYXRoIGQ9Ik0yMSw4SDNWNGgxOFY4eiBNMjEsMTBIM3Y0aDE4VjEweiBNMjEsMTZIM3Y0aDE4VjE2eiIvPgogICAgPC9nPgo8L3N2Zz4K);
  --jp-icon-tag: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjgiIGhlaWdodD0iMjgiIHZpZXdCb3g9IjAgMCA0MyAyOCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KCTxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CgkJPHBhdGggZD0iTTI4LjgzMzIgMTIuMzM0TDMyLjk5OTggMTYuNTAwN0wzNy4xNjY1IDEyLjMzNEgyOC44MzMyWiIvPgoJCTxwYXRoIGQ9Ik0xNi4yMDk1IDIxLjYxMDRDMTUuNjg3MyAyMi4xMjk5IDE0Ljg0NDMgMjIuMTI5OSAxNC4zMjQ4IDIxLjYxMDRMNi45ODI5IDE0LjcyNDVDNi41NzI0IDE0LjMzOTQgNi4wODMxMyAxMy42MDk4IDYuMDQ3ODYgMTMuMDQ4MkM1Ljk1MzQ3IDExLjUyODggNi4wMjAwMiA4LjYxOTQ0IDYuMDY2MjEgNy4wNzY5NUM2LjA4MjgxIDYuNTE0NzcgNi41NTU0OCA2LjA0MzQ3IDcuMTE4MDQgNi4wMzA1NUM5LjA4ODYzIDUuOTg0NzMgMTMuMjYzOCA1LjkzNTc5IDEzLjY1MTggNi4zMjQyNUwyMS43MzY5IDEzLjYzOUMyMi4yNTYgMTQuMTU4NSAyMS43ODUxIDE1LjQ3MjQgMjEuMjYyIDE1Ljk5NDZMMTYuMjA5NSAyMS42MTA0Wk05Ljc3NTg1IDguMjY1QzkuMzM1NTEgNy44MjU2NiA4LjYyMzUxIDcuODI1NjYgOC4xODI4IDguMjY1QzcuNzQzNDYgOC43MDU3MSA3Ljc0MzQ2IDkuNDE3MzMgOC4xODI4IDkuODU2NjdDOC42MjM4MiAxMC4yOTY0IDkuMzM1ODIgMTAuMjk2NCA5Ljc3NTg1IDkuODU2NjdDMTAuMjE1NiA5LjQxNzMzIDEwLjIxNTYgOC43MDUzMyA5Ljc3NTg1IDguMjY1WiIvPgoJPC9nPgo8L3N2Zz4K);
  --jp-icon-terminal: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0IiA+CiAgICA8cmVjdCBjbGFzcz0ianAtdGVybWluYWwtaWNvbi1iYWNrZ3JvdW5kLWNvbG9yIGpwLWljb24tc2VsZWN0YWJsZSIgd2lkdGg9IjIwIiBoZWlnaHQ9IjIwIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgyIDIpIiBmaWxsPSIjMzMzMzMzIi8+CiAgICA8cGF0aCBjbGFzcz0ianAtdGVybWluYWwtaWNvbi1jb2xvciBqcC1pY29uLXNlbGVjdGFibGUtaW52ZXJzZSIgZD0iTTUuMDU2NjQgOC43NjE3MkM1LjA1NjY0IDguNTk3NjYgNS4wMzEyNSA4LjQ1MzEyIDQuOTgwNDcgOC4zMjgxMkM0LjkzMzU5IDguMTk5MjIgNC44NTU0NyA4LjA4MjAzIDQuNzQ2MDkgNy45NzY1NkM0LjY0MDYyIDcuODcxMDkgNC41IDcuNzc1MzkgNC4zMjQyMiA3LjY4OTQ1QzQuMTUyMzQgNy41OTk2MSAzLjk0MzM2IDcuNTExNzIgMy42OTcyNyA3LjQyNTc4QzMuMzAyNzMgNy4yODUxNiAyLjk0MzM2IDcuMTM2NzIgMi42MTkxNCA2Ljk4MDQ3QzIuMjk0OTIgNi44MjQyMiAyLjAxNzU4IDYuNjQyNTggMS43ODcxMSA2LjQzNTU1QzEuNTYwNTUgNi4yMjg1MiAxLjM4NDc3IDUuOTg4MjggMS4yNTk3NyA1LjcxNDg0QzEuMTM0NzcgNS40Mzc1IDEuMDcyMjcgNS4xMDkzOCAxLjA3MjI3IDQuNzMwNDdDMS4wNzIyNyA0LjM5ODQ0IDEuMTI4OTEgNC4wOTU3IDEuMjQyMTkgMy44MjIyN0MxLjM1NTQ3IDMuNTQ0OTIgMS41MTU2MiAzLjMwNDY5IDEuNzIyNjYgMy4xMDE1NkMxLjkyOTY5IDIuODk4NDQgMi4xNzk2OSAyLjczNDM3IDIuNDcyNjYgMi42MDkzOEMyLjc2NTYyIDIuNDg0MzggMy4wOTE4IDIuNDA0MyAzLjQ1MTE3IDIuMzY5MTRWMS4xMDkzOEg0LjM4ODY3VjIuMzgwODZDNC43NDAyMyAyLjQyNzczIDUuMDU2NjQgMi41MjM0NCA1LjMzNzg5IDIuNjY3OTdDNS42MTkxNCAyLjgxMjUgNS44NTc0MiAzLjAwMTk1IDYuMDUyNzMgMy4yMzYzM0M2LjI1MTk1IDMuNDY2OCA2LjQwNDMgMy43NDAyMyA2LjUwOTc3IDQuMDU2NjRDNi42MTkxNCA0LjM2OTE0IDYuNjczODMgNC43MjA3IDYuNjczODMgNS4xMTEzM0g1LjA0NDkyQzUuMDQ0OTIgNC42Mzg2NyA0LjkzNzUgNC4yODEyNSA0LjcyMjY2IDQuMDM5MDZDNC41MDc4MSAzLjc5Mjk3IDQuMjE2OCAzLjY2OTkyIDMuODQ5NjEgMy42Njk5MkMzLjY1MDM5IDMuNjY5OTIgMy40NzY1NiAzLjY5NzI3IDMuMzI4MTIgMy43NTE5NUMzLjE4MzU5IDMuODAyNzMgMy4wNjQ0NSAzLjg3Njk1IDIuOTcwNyAzLjk3NDYxQzIuODc2OTUgNC4wNjgzNiAyLjgwNjY0IDQuMTc5NjkgMi43NTk3NyA0LjMwODU5QzIuNzE2OCA0LjQzNzUgMi42OTUzMSA0LjU3ODEyIDIuNjk1MzEgNC43MzA0N0MyLjY5NTMxIDQuODgyODEgMi43MTY4IDUuMDE5NTMgMi43NTk3NyA1LjE0MDYyQzIuODA2NjQgNS4yNTc4MSAyLjg4MjgxIDUuMzY3MTkgMi45ODgyOCA1LjQ2ODc1QzMuMDk3NjYgNS41NzAzMSAzLjI0MDIzIDUuNjY3OTcgMy40MTYwMiA1Ljc2MTcyQzMuNTkxOCA1Ljg1MTU2IDMuODEwNTUgNS45NDMzNiA0LjA3MjI3IDYuMDM3MTFDNC40NjY4IDYuMTg1NTUgNC44MjQyMiA2LjMzOTg0IDUuMTQ0NTMgNi41QzUuNDY0ODQgNi42NTYyNSA1LjczODI4IDYuODM5ODQgNS45NjQ4NCA3LjA1MDc4QzYuMTk1MzEgNy4yNTc4MSA2LjM3MTA5IDcuNSA2LjQ5MjE5IDcuNzc3MzRDNi42MTcxOSA4LjA1MDc4IDYuNjc5NjkgOC4zNzUgNi42Nzk2OSA4Ljc1QzYuNjc5NjkgOS4wOTM3NSA2LjYyMzA1IDkuNDA0MyA2LjUwOTc3IDkuNjgxNjRDNi4zOTY0OCA5Ljk1NTA4IDYuMjM0MzggMTAuMTkxNCA2LjAyMzQ0IDEwLjM5MDZDNS44MTI1IDEwLjU4OTggNS41NTg1OSAxMC43NSA1LjI2MTcyIDEwLjg3MTFDNC45NjQ4NCAxMC45ODgzIDQuNjMyODEgMTEuMDY0NSA0LjI2NTYyIDExLjA5OTZWMTIuMjQ4SDMuMzMzOThWMTEuMDk5NkMzLjAwMTk1IDExLjA2ODQgMi42Nzk2OSAxMC45OTYxIDIuMzY3MTkgMTAuODgyOEMyLjA1NDY5IDEwLjc2NTYgMS43NzczNCAxMC41OTc3IDEuNTM1MTYgMTAuMzc4OUMxLjI5Njg4IDEwLjE2MDIgMS4xMDU0NyA5Ljg4NDc3IDAuOTYwOTM4IDkuNTUyNzNDMC44MTY0MDYgOS4yMTY4IDAuNzQ0MTQxIDguODE0NDUgMC43NDQxNDEgOC4zNDU3SDIuMzc4OTFDMi4zNzg5MSA4LjYyNjk1IDIuNDE5OTIgOC44NjMyOCAyLjUwMTk1IDkuMDU0NjlDMi41ODM5OCA5LjI0MjE5IDIuNjg5NDUgOS4zOTI1OCAyLjgxODM2IDkuNTA1ODZDMi45NTExNyA5LjYxNTIzIDMuMTAxNTYgOS42OTMzNiAzLjI2OTUzIDkuNzQwMjNDMy40Mzc1IDkuNzg3MTEgMy42MDkzOCA5LjgxMDU1IDMuNzg1MTYgOS44MTA1NUM0LjIwMzEyIDkuODEwNTUgNC41MTk1MyA5LjcxMjg5IDQuNzM0MzggOS41MTc1OEM0Ljk0OTIyIDkuMzIyMjcgNS4wNTY2NCA5LjA3MDMxIDUuMDU2NjQgOC43NjE3MlpNMTMuNDE4IDEyLjI3MTVIOC4wNzQyMlYxMUgxMy40MThWMTIuMjcxNVoiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDMuOTUyNjQgNikiIGZpbGw9IndoaXRlIi8+Cjwvc3ZnPgo=);
  --jp-icon-text-editor: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8cGF0aCBjbGFzcz0ianAtdGV4dC1lZGl0b3ItaWNvbi1jb2xvciBqcC1pY29uLXNlbGVjdGFibGUiIGZpbGw9IiM2MTYxNjEiIGQ9Ik0xNSAxNUgzdjJoMTJ2LTJ6bTAtOEgzdjJoMTJWN3pNMyAxM2gxOHYtMkgzdjJ6bTAgOGgxOHYtMkgzdjJ6TTMgM3YyaDE4VjNIM3oiLz4KPC9zdmc+Cg==);
  --jp-icon-toc: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0Ij4KICA8ZyBjbGFzcz0ianAtaWNvbjMganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjNjE2MTYxIj4KICAgIDxwYXRoIGQ9Ik03LDVIMjFWN0g3VjVNNywxM1YxMUgyMVYxM0g3TTQsNC41QTEuNSwxLjUgMCAwLDEgNS41LDZBMS41LDEuNSAwIDAsMSA0LDcuNUExLjUsMS41IDAgMCwxIDIuNSw2QTEuNSwxLjUgMCAwLDEgNCw0LjVNNCwxMC41QTEuNSwxLjUgMCAwLDEgNS41LDEyQTEuNSwxLjUgMCAwLDEgNCwxMy41QTEuNSwxLjUgMCAwLDEgMi41LDEyQTEuNSwxLjUgMCAwLDEgNCwxMC41TTcsMTlWMTdIMjFWMTlIN000LDE2LjVBMS41LDEuNSAwIDAsMSA1LjUsMThBMS41LDEuNSAwIDAsMSA0LDE5LjVBMS41LDEuNSAwIDAsMSAyLjUsMThBMS41LDEuNSAwIDAsMSA0LDE2LjVaIiAvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-tree-view: url(data:image/svg+xml;base64,PHN2ZyBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxnIGNsYXNzPSJqcC1pY29uMyIgZmlsbD0iIzYxNjE2MSI+CiAgICAgICAgPHBhdGggZD0iTTAgMGgyNHYyNEgweiIgZmlsbD0ibm9uZSIvPgogICAgICAgIDxwYXRoIGQ9Ik0yMiAxMVYzaC03djNIOVYzSDJ2OGg3VjhoMnYxMGg0djNoN3YtOGgtN3YzaC0yVjhoMnYzeiIvPgogICAgPC9nPgo8L3N2Zz4K);
  --jp-icon-trusted: url(data:image/svg+xml;base64,PHN2ZyBmaWxsPSJub25lIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDI0IDI1Ij4KICAgIDxwYXRoIGNsYXNzPSJqcC1pY29uMiIgc3Ryb2tlPSIjMzMzMzMzIiBzdHJva2Utd2lkdGg9IjIiIHRyYW5zZm9ybT0idHJhbnNsYXRlKDIgMykiIGQ9Ik0xLjg2MDk0IDExLjQ0MDlDMC44MjY0NDggOC43NzAyNyAwLjg2Mzc3OSA2LjA1NzY0IDEuMjQ5MDcgNC4xOTkzMkMyLjQ4MjA2IDMuOTMzNDcgNC4wODA2OCAzLjQwMzQ3IDUuNjAxMDIgMi44NDQ5QzcuMjM1NDkgMi4yNDQ0IDguODU2NjYgMS41ODE1IDkuOTg3NiAxLjA5NTM5QzExLjA1OTcgMS41ODM0MSAxMi42MDk0IDIuMjQ0NCAxNC4yMTggMi44NDMzOUMxNS43NTAzIDMuNDEzOTQgMTcuMzk5NSAzLjk1MjU4IDE4Ljc1MzkgNC4yMTM4NUMxOS4xMzY0IDYuMDcxNzcgMTkuMTcwOSA4Ljc3NzIyIDE4LjEzOSAxMS40NDA5QzE3LjAzMDMgMTQuMzAzMiAxNC42NjY4IDE3LjE4NDQgOS45OTk5OSAxOC45MzU0QzUuMzMzMiAxNy4xODQ0IDIuOTY5NjggMTQuMzAzMiAxLjg2MDk0IDExLjQ0MDlaIi8+CiAgICA8cGF0aCBjbGFzcz0ianAtaWNvbjIiIGZpbGw9IiMzMzMzMzMiIHN0cm9rZT0iIzMzMzMzMyIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoOCA5Ljg2NzE5KSIgZD0iTTIuODYwMTUgNC44NjUzNUwwLjcyNjU0OSAyLjk5OTU5TDAgMy42MzA0NUwyLjg2MDE1IDYuMTMxNTdMOCAwLjYzMDg3Mkw3LjI3ODU3IDBMMi44NjAxNSA0Ljg2NTM1WiIvPgo8L3N2Zz4K);
  --jp-icon-undo: url(data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHdpZHRoPSIxNiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTEyLjUgOGMtMi42NSAwLTUuMDUuOTktNi45IDIuNkwyIDd2OWg5bC0zLjYyLTMuNjJjMS4zOS0xLjE2IDMuMTYtMS44OCA1LjEyLTEuODggMy41NCAwIDYuNTUgMi4zMSA3LjYgNS41bDIuMzctLjc4QzIxLjA4IDExLjAzIDE3LjE1IDggMTIuNSA4eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-user: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIHZpZXdCb3g9IjAgMCAyNCAyNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8ZyBjbGFzcz0ianAtaWNvbjMiIGZpbGw9IiM2MTYxNjEiPgogICAgPHBhdGggZD0iTTE2IDdhNCA0IDAgMTEtOCAwIDQgNCAwIDAxOCAwek0xMiAxNGE3IDcgMCAwMC03IDdoMTRhNyA3IDAgMDAtNy03eiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-users: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHZlcnNpb249IjEuMSIgdmlld0JveD0iMCAwIDM2IDI0IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgogPGcgY2xhc3M9ImpwLWljb24zIiB0cmFuc2Zvcm09Im1hdHJpeCgxLjczMjcgMCAwIDEuNzMyNyAtMy42MjgyIC4wOTk1NzcpIiBmaWxsPSIjNjE2MTYxIj4KICA8cGF0aCB0cmFuc2Zvcm09Im1hdHJpeCgxLjUsMCwwLDEuNSwwLC02KSIgZD0ibTEyLjE4NiA3LjUwOThjLTEuMDUzNSAwLTEuOTc1NyAwLjU2NjUtMi40Nzg1IDEuNDEwMiAwLjc1MDYxIDAuMzEyNzcgMS4zOTc0IDAuODI2NDggMS44NzMgMS40NzI3aDMuNDg2M2MwLTEuNTkyLTEuMjg4OS0yLjg4MjgtMi44ODA5LTIuODgyOHoiLz4KICA8cGF0aCBkPSJtMjAuNDY1IDIuMzg5NWEyLjE4ODUgMi4xODg1IDAgMCAxLTIuMTg4NCAyLjE4ODUgMi4xODg1IDIuMTg4NSAwIDAgMS0yLjE4ODUtMi4xODg1IDIuMTg4NSAyLjE4ODUgMCAwIDEgMi4xODg1LTIuMTg4NSAyLjE4ODUgMi4xODg1IDAgMCAxIDIuMTg4NCAyLjE4ODV6Ii8+CiAgPHBhdGggdHJhbnNmb3JtPSJtYXRyaXgoMS41LDAsMCwxLjUsMCwtNikiIGQ9Im0zLjU4OTggOC40MjE5Yy0xLjExMjYgMC0yLjAxMzcgMC45MDExMS0yLjAxMzcgMi4wMTM3aDIuODE0NWMwLjI2Nzk3LTAuMzczMDkgMC41OTA3LTAuNzA0MzUgMC45NTg5OC0wLjk3ODUyLTAuMzQ0MzMtMC42MTY4OC0xLjAwMzEtMS4wMzUyLTEuNzU5OC0xLjAzNTJ6Ii8+CiAgPHBhdGggZD0ibTYuOTE1NCA0LjYyM2ExLjUyOTQgMS41Mjk0IDAgMCAxLTEuNTI5NCAxLjUyOTQgMS41Mjk0IDEuNTI5NCAwIDAgMS0xLjUyOTQtMS41Mjk0IDEuNTI5NCAxLjUyOTQgMCAwIDEgMS41Mjk0LTEuNTI5NCAxLjUyOTQgMS41Mjk0IDAgMCAxIDEuNTI5NCAxLjUyOTR6Ii8+CiAgPHBhdGggZD0ibTYuMTM1IDEzLjUzNWMwLTMuMjM5MiAyLjYyNTktNS44NjUgNS44NjUtNS44NjUgMy4yMzkyIDAgNS44NjUgMi42MjU5IDUuODY1IDUuODY1eiIvPgogIDxjaXJjbGUgY3g9IjEyIiBjeT0iMy43Njg1IiByPSIyLjk2ODUiLz4KIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-vega: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8ZyBjbGFzcz0ianAtaWNvbjEganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjMjEyMTIxIj4KICAgIDxwYXRoIGQ9Ik0xMC42IDUuNGwyLjItMy4ySDIuMnY3LjNsNC02LjZ6Ii8+CiAgICA8cGF0aCBkPSJNMTUuOCAyLjJsLTQuNCA2LjZMNyA2LjNsLTQuOCA4djUuNWgxNy42VjIuMmgtNHptLTcgMTUuNEg1LjV2LTQuNGgzLjN2NC40em00LjQgMEg5LjhWOS44aDMuNHY3Ljh6bTQuNCAwaC0zLjRWNi41aDMuNHYxMS4xeiIvPgogIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-word: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIwIDIwIj4KIDxnIGNsYXNzPSJqcC1pY29uMiIgZmlsbD0iIzQxNDE0MSI+CiAgPHJlY3QgeD0iMiIgeT0iMiIgd2lkdGg9IjE2IiBoZWlnaHQ9IjE2Ii8+CiA8L2c+CiA8ZyBjbGFzcz0ianAtaWNvbi1hY2NlbnQyIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSguNDMgLjA0MDEpIiBmaWxsPSIjZmZmIj4KICA8cGF0aCBkPSJtNC4xNCA4Ljc2cTAuMDY4Mi0xLjg5IDIuNDItMS44OSAxLjE2IDAgMS42OCAwLjQyIDAuNTY3IDAuNDEgMC41NjcgMS4xNnYzLjQ3cTAgMC40NjIgMC41MTQgMC40NjIgMC4xMDMgMCAwLjItMC4wMjMxdjAuNzE0cS0wLjM5OSAwLjEwMy0wLjY1MSAwLjEwMy0wLjQ1MiAwLTAuNjkzLTAuMjItMC4yMzEtMC4yLTAuMjg0LTAuNjYyLTAuOTU2IDAuODcyLTIgMC44NzItMC45MDMgMC0xLjQ3LTAuNDcyLTAuNTI1LTAuNDcyLTAuNTI1LTEuMjYgMC0wLjI2MiAwLjA0NTItMC40NzIgMC4wNTY3LTAuMjIgMC4xMTYtMC4zNzggMC4wNjgyLTAuMTY4IDAuMjMxLTAuMzA0IDAuMTU4LTAuMTQ3IDAuMjYyLTAuMjQyIDAuMTE2LTAuMDkxNCAwLjM2OC0wLjE2OCAwLjI2Mi0wLjA5MTQgMC4zOTktMC4xMjYgMC4xMzYtMC4wNDUyIDAuNDcyLTAuMTAzIDAuMzM2LTAuMDU3OCAwLjUwNC0wLjA3OTggMC4xNTgtMC4wMjMxIDAuNTY3LTAuMDc5OCAwLjU1Ni0wLjA2ODIgMC43NzctMC4yMjEgMC4yMi0wLjE1MiAwLjIyLTAuNDQxdi0wLjI1MnEwLTAuNDMtMC4zNTctMC42NjItMC4zMzYtMC4yMzEtMC45NzYtMC4yMzEtMC42NjIgMC0wLjk5OCAwLjI2Mi0wLjMzNiAwLjI1Mi0wLjM5OSAwLjc5OHptMS44OSAzLjY4cTAuNzg4IDAgMS4yNi0wLjQxIDAuNTA0LTAuNDIgMC41MDQtMC45MDN2LTEuMDVxLTAuMjg0IDAuMTM2LTAuODYxIDAuMjMxLTAuNTY3IDAuMDkxNC0wLjk4NyAwLjE1OC0wLjQyIDAuMDY4Mi0wLjc2NiAwLjMyNi0wLjMzNiAwLjI1Mi0wLjMzNiAwLjcwNHQwLjMwNCAwLjcwNCAwLjg2MSAwLjI1MnoiIHN0cm9rZS13aWR0aD0iMS4wNSIvPgogIDxwYXRoIGQ9Im0xMCA0LjU2aDAuOTQ1djMuMTVxMC42NTEtMC45NzYgMS44OS0wLjk3NiAxLjE2IDAgMS44OSAwLjg0IDAuNjgyIDAuODQgMC42ODIgMi4zMSAwIDEuNDctMC43MDQgMi40Mi0wLjcwNCAwLjg4Mi0xLjg5IDAuODgyLTEuMjYgMC0xLjg5LTEuMDJ2MC43NjZoLTAuODV6bTIuNjIgMy4wNHEtMC43NDYgMC0xLjE2IDAuNjQtMC40NTIgMC42My0wLjQ1MiAxLjY4IDAgMS4wNSAwLjQ1MiAxLjY4dDEuMTYgMC42M3EwLjc3NyAwIDEuMjYtMC42MyAwLjQ5NC0wLjY0IDAuNDk0LTEuNjggMC0xLjA1LTAuNDcyLTEuNjgtMC40NjItMC42NC0xLjI2LTAuNjR6IiBzdHJva2Utd2lkdGg9IjEuMDUiLz4KICA8cGF0aCBkPSJtMi43MyAxNS44IDEzLjYgMC4wMDgxYzAuMDA2OSAwIDAtMi42IDAtMi42IDAtMC4wMDc4LTEuMTUgMC0xLjE1IDAtMC4wMDY5IDAtMC4wMDgzIDEuNS0wLjAwODMgMS41LTJlLTMgLTAuMDAxNC0xMS4zLTAuMDAxNC0xMS4zLTAuMDAxNGwtMC4wMDU5Mi0xLjVjMC0wLjAwNzgtMS4xNyAwLjAwMTMtMS4xNyAwLjAwMTN6IiBzdHJva2Utd2lkdGg9Ii45NzUiLz4KIDwvZz4KPC9zdmc+Cg==);
  --jp-icon-yaml: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgdmlld0JveD0iMCAwIDIyIDIyIj4KICA8ZyBjbGFzcz0ianAtaWNvbi1jb250cmFzdDIganAtaWNvbi1zZWxlY3RhYmxlIiBmaWxsPSIjRDgxQjYwIj4KICAgIDxwYXRoIGQ9Ik03LjIgMTguNnYtNS40TDMgNS42aDMuM2wxLjQgMy4xYy4zLjkuNiAxLjYgMSAyLjUuMy0uOC42LTEuNiAxLTIuNWwxLjQtMy4xaDMuNGwtNC40IDcuNnY1LjVsLTIuOS0uMXoiLz4KICAgIDxjaXJjbGUgY2xhc3M9InN0MCIgY3g9IjE3LjYiIGN5PSIxNi41IiByPSIyLjEiLz4KICAgIDxjaXJjbGUgY2xhc3M9InN0MCIgY3g9IjE3LjYiIGN5PSIxMSIgcj0iMi4xIi8+CiAgPC9nPgo8L3N2Zz4K);
}

/* Icon CSS class declarations */

.jp-AddAboveIcon {
  background-image: var(--jp-icon-add-above);
}

.jp-AddBelowIcon {
  background-image: var(--jp-icon-add-below);
}

.jp-AddIcon {
  background-image: var(--jp-icon-add);
}

.jp-BellIcon {
  background-image: var(--jp-icon-bell);
}

.jp-BugDotIcon {
  background-image: var(--jp-icon-bug-dot);
}

.jp-BugIcon {
  background-image: var(--jp-icon-bug);
}

.jp-BuildIcon {
  background-image: var(--jp-icon-build);
}

.jp-CaretDownEmptyIcon {
  background-image: var(--jp-icon-caret-down-empty);
}

.jp-CaretDownEmptyThinIcon {
  background-image: var(--jp-icon-caret-down-empty-thin);
}

.jp-CaretDownIcon {
  background-image: var(--jp-icon-caret-down);
}

.jp-CaretLeftIcon {
  background-image: var(--jp-icon-caret-left);
}

.jp-CaretRightIcon {
  background-image: var(--jp-icon-caret-right);
}

.jp-CaretUpEmptyThinIcon {
  background-image: var(--jp-icon-caret-up-empty-thin);
}

.jp-CaretUpIcon {
  background-image: var(--jp-icon-caret-up);
}

.jp-CaseSensitiveIcon {
  background-image: var(--jp-icon-case-sensitive);
}

.jp-CheckIcon {
  background-image: var(--jp-icon-check);
}

.jp-CircleEmptyIcon {
  background-image: var(--jp-icon-circle-empty);
}

.jp-CircleIcon {
  background-image: var(--jp-icon-circle);
}

.jp-ClearIcon {
  background-image: var(--jp-icon-clear);
}

.jp-CloseIcon {
  background-image: var(--jp-icon-close);
}

.jp-CodeCheckIcon {
  background-image: var(--jp-icon-code-check);
}

.jp-CodeIcon {
  background-image: var(--jp-icon-code);
}

.jp-CollapseAllIcon {
  background-image: var(--jp-icon-collapse-all);
}

.jp-ConsoleIcon {
  background-image: var(--jp-icon-console);
}

.jp-CopyIcon {
  background-image: var(--jp-icon-copy);
}

.jp-CopyrightIcon {
  background-image: var(--jp-icon-copyright);
}

.jp-CutIcon {
  background-image: var(--jp-icon-cut);
}

.jp-DeleteIcon {
  background-image: var(--jp-icon-delete);
}

.jp-DownloadIcon {
  background-image: var(--jp-icon-download);
}

.jp-DuplicateIcon {
  background-image: var(--jp-icon-duplicate);
}

.jp-EditIcon {
  background-image: var(--jp-icon-edit);
}

.jp-EllipsesIcon {
  background-image: var(--jp-icon-ellipses);
}

.jp-ErrorIcon {
  background-image: var(--jp-icon-error);
}

.jp-ExpandAllIcon {
  background-image: var(--jp-icon-expand-all);
}

.jp-ExtensionIcon {
  background-image: var(--jp-icon-extension);
}

.jp-FastForwardIcon {
  background-image: var(--jp-icon-fast-forward);
}

.jp-FileIcon {
  background-image: var(--jp-icon-file);
}

.jp-FileUploadIcon {
  background-image: var(--jp-icon-file-upload);
}

.jp-FilterDotIcon {
  background-image: var(--jp-icon-filter-dot);
}

.jp-FilterIcon {
  background-image: var(--jp-icon-filter);
}

.jp-FilterListIcon {
  background-image: var(--jp-icon-filter-list);
}

.jp-FolderFavoriteIcon {
  background-image: var(--jp-icon-folder-favorite);
}

.jp-FolderIcon {
  background-image: var(--jp-icon-folder);
}

.jp-HomeIcon {
  background-image: var(--jp-icon-home);
}

.jp-Html5Icon {
  background-image: var(--jp-icon-html5);
}

.jp-ImageIcon {
  background-image: var(--jp-icon-image);
}

.jp-InfoIcon {
  background-image: var(--jp-icon-info);
}

.jp-InspectorIcon {
  background-image: var(--jp-icon-inspector);
}

.jp-JsonIcon {
  background-image: var(--jp-icon-json);
}

.jp-JuliaIcon {
  background-image: var(--jp-icon-julia);
}

.jp-JupyterFaviconIcon {
  background-image: var(--jp-icon-jupyter-favicon);
}

.jp-JupyterIcon {
  background-image: var(--jp-icon-jupyter);
}

.jp-JupyterlabWordmarkIcon {
  background-image: var(--jp-icon-jupyterlab-wordmark);
}

.jp-KernelIcon {
  background-image: var(--jp-icon-kernel);
}

.jp-KeyboardIcon {
  background-image: var(--jp-icon-keyboard);
}

.jp-LaunchIcon {
  background-image: var(--jp-icon-launch);
}

.jp-LauncherIcon {
  background-image: var(--jp-icon-launcher);
}

.jp-LineFormIcon {
  background-image: var(--jp-icon-line-form);
}

.jp-LinkIcon {
  background-image: var(--jp-icon-link);
}

.jp-ListIcon {
  background-image: var(--jp-icon-list);
}

.jp-MarkdownIcon {
  background-image: var(--jp-icon-markdown);
}

.jp-MoveDownIcon {
  background-image: var(--jp-icon-move-down);
}

.jp-MoveUpIcon {
  background-image: var(--jp-icon-move-up);
}

.jp-NewFolderIcon {
  background-image: var(--jp-icon-new-folder);
}

.jp-NotTrustedIcon {
  background-image: var(--jp-icon-not-trusted);
}

.jp-NotebookIcon {
  background-image: var(--jp-icon-notebook);
}

.jp-NumberingIcon {
  background-image: var(--jp-icon-numbering);
}

.jp-OfflineBoltIcon {
  background-image: var(--jp-icon-offline-bolt);
}

.jp-PaletteIcon {
  background-image: var(--jp-icon-palette);
}

.jp-PasteIcon {
  background-image: var(--jp-icon-paste);
}

.jp-PdfIcon {
  background-image: var(--jp-icon-pdf);
}

.jp-PythonIcon {
  background-image: var(--jp-icon-python);
}

.jp-RKernelIcon {
  background-image: var(--jp-icon-r-kernel);
}

.jp-ReactIcon {
  background-image: var(--jp-icon-react);
}

.jp-RedoIcon {
  background-image: var(--jp-icon-redo);
}

.jp-RefreshIcon {
  background-image: var(--jp-icon-refresh);
}

.jp-RegexIcon {
  background-image: var(--jp-icon-regex);
}

.jp-RunIcon {
  background-image: var(--jp-icon-run);
}

.jp-RunningIcon {
  background-image: var(--jp-icon-running);
}

.jp-SaveIcon {
  background-image: var(--jp-icon-save);
}

.jp-SearchIcon {
  background-image: var(--jp-icon-search);
}

.jp-SettingsIcon {
  background-image: var(--jp-icon-settings);
}

.jp-ShareIcon {
  background-image: var(--jp-icon-share);
}

.jp-SpreadsheetIcon {
  background-image: var(--jp-icon-spreadsheet);
}

.jp-StopIcon {
  background-image: var(--jp-icon-stop);
}

.jp-TabIcon {
  background-image: var(--jp-icon-tab);
}

.jp-TableRowsIcon {
  background-image: var(--jp-icon-table-rows);
}

.jp-TagIcon {
  background-image: var(--jp-icon-tag);
}

.jp-TerminalIcon {
  background-image: var(--jp-icon-terminal);
}

.jp-TextEditorIcon {
  background-image: var(--jp-icon-text-editor);
}

.jp-TocIcon {
  background-image: var(--jp-icon-toc);
}

.jp-TreeViewIcon {
  background-image: var(--jp-icon-tree-view);
}

.jp-TrustedIcon {
  background-image: var(--jp-icon-trusted);
}

.jp-UndoIcon {
  background-image: var(--jp-icon-undo);
}

.jp-UserIcon {
  background-image: var(--jp-icon-user);
}

.jp-UsersIcon {
  background-image: var(--jp-icon-users);
}

.jp-VegaIcon {
  background-image: var(--jp-icon-vega);
}

.jp-WordIcon {
  background-image: var(--jp-icon-word);
}

.jp-YamlIcon {
  background-image: var(--jp-icon-yaml);
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/**
 * (DEPRECATED) Support for consuming icons as CSS background images
 */

.jp-Icon,
.jp-MaterialIcon {
  background-position: center;
  background-repeat: no-repeat;
  background-size: 16px;
  min-width: 16px;
  min-height: 16px;
}

.jp-Icon-cover {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

/**
 * (DEPRECATED) Support for specific CSS icon sizes
 */

.jp-Icon-16 {
  background-size: 16px;
  min-width: 16px;
  min-height: 16px;
}

.jp-Icon-18 {
  background-size: 18px;
  min-width: 18px;
  min-height: 18px;
}

.jp-Icon-20 {
  background-size: 20px;
  min-width: 20px;
  min-height: 20px;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.lm-TabBar .lm-TabBar-addButton {
  align-items: center;
  display: flex;
  padding: 4px;
  padding-bottom: 5px;
  margin-right: 1px;
  background-color: var(--jp-layout-color2);
}

.lm-TabBar .lm-TabBar-addButton:hover {
  background-color: var(--jp-layout-color1);
}

.lm-DockPanel-tabBar .lm-TabBar-tab {
  width: var(--jp-private-horizontal-tab-width);
}

.lm-DockPanel-tabBar .lm-TabBar-content {
  flex: unset;
}

.lm-DockPanel-tabBar[data-orientation='horizontal'] {
  flex: 1 1 auto;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/**
 * Support for icons as inline SVG HTMLElements
 */

/* recolor the primary elements of an icon */
.jp-icon0[fill] {
  fill: var(--jp-inverse-layout-color0);
}

.jp-icon1[fill] {
  fill: var(--jp-inverse-layout-color1);
}

.jp-icon2[fill] {
  fill: var(--jp-inverse-layout-color2);
}

.jp-icon3[fill] {
  fill: var(--jp-inverse-layout-color3);
}

.jp-icon4[fill] {
  fill: var(--jp-inverse-layout-color4);
}

.jp-icon0[stroke] {
  stroke: var(--jp-inverse-layout-color0);
}

.jp-icon1[stroke] {
  stroke: var(--jp-inverse-layout-color1);
}

.jp-icon2[stroke] {
  stroke: var(--jp-inverse-layout-color2);
}

.jp-icon3[stroke] {
  stroke: var(--jp-inverse-layout-color3);
}

.jp-icon4[stroke] {
  stroke: var(--jp-inverse-layout-color4);
}

/* recolor the accent elements of an icon */
.jp-icon-accent0[fill] {
  fill: var(--jp-layout-color0);
}

.jp-icon-accent1[fill] {
  fill: var(--jp-layout-color1);
}

.jp-icon-accent2[fill] {
  fill: var(--jp-layout-color2);
}

.jp-icon-accent3[fill] {
  fill: var(--jp-layout-color3);
}

.jp-icon-accent4[fill] {
  fill: var(--jp-layout-color4);
}

.jp-icon-accent0[stroke] {
  stroke: var(--jp-layout-color0);
}

.jp-icon-accent1[stroke] {
  stroke: var(--jp-layout-color1);
}

.jp-icon-accent2[stroke] {
  stroke: var(--jp-layout-color2);
}

.jp-icon-accent3[stroke] {
  stroke: var(--jp-layout-color3);
}

.jp-icon-accent4[stroke] {
  stroke: var(--jp-layout-color4);
}

/* set the color of an icon to transparent */
.jp-icon-none[fill] {
  fill: none;
}

.jp-icon-none[stroke] {
  stroke: none;
}

/* brand icon colors. Same for light and dark */
.jp-icon-brand0[fill] {
  fill: var(--jp-brand-color0);
}

.jp-icon-brand1[fill] {
  fill: var(--jp-brand-color1);
}

.jp-icon-brand2[fill] {
  fill: var(--jp-brand-color2);
}

.jp-icon-brand3[fill] {
  fill: var(--jp-brand-color3);
}

.jp-icon-brand4[fill] {
  fill: var(--jp-brand-color4);
}

.jp-icon-brand0[stroke] {
  stroke: var(--jp-brand-color0);
}

.jp-icon-brand1[stroke] {
  stroke: var(--jp-brand-color1);
}

.jp-icon-brand2[stroke] {
  stroke: var(--jp-brand-color2);
}

.jp-icon-brand3[stroke] {
  stroke: var(--jp-brand-color3);
}

.jp-icon-brand4[stroke] {
  stroke: var(--jp-brand-color4);
}

/* warn icon colors. Same for light and dark */
.jp-icon-warn0[fill] {
  fill: var(--jp-warn-color0);
}

.jp-icon-warn1[fill] {
  fill: var(--jp-warn-color1);
}

.jp-icon-warn2[fill] {
  fill: var(--jp-warn-color2);
}

.jp-icon-warn3[fill] {
  fill: var(--jp-warn-color3);
}

.jp-icon-warn0[stroke] {
  stroke: var(--jp-warn-color0);
}

.jp-icon-warn1[stroke] {
  stroke: var(--jp-warn-color1);
}

.jp-icon-warn2[stroke] {
  stroke: var(--jp-warn-color2);
}

.jp-icon-warn3[stroke] {
  stroke: var(--jp-warn-color3);
}

/* icon colors that contrast well with each other and most backgrounds */
.jp-icon-contrast0[fill] {
  fill: var(--jp-icon-contrast-color0);
}

.jp-icon-contrast1[fill] {
  fill: var(--jp-icon-contrast-color1);
}

.jp-icon-contrast2[fill] {
  fill: var(--jp-icon-contrast-color2);
}

.jp-icon-contrast3[fill] {
  fill: var(--jp-icon-contrast-color3);
}

.jp-icon-contrast0[stroke] {
  stroke: var(--jp-icon-contrast-color0);
}

.jp-icon-contrast1[stroke] {
  stroke: var(--jp-icon-contrast-color1);
}

.jp-icon-contrast2[stroke] {
  stroke: var(--jp-icon-contrast-color2);
}

.jp-icon-contrast3[stroke] {
  stroke: var(--jp-icon-contrast-color3);
}

.jp-icon-dot[fill] {
  fill: var(--jp-warn-color0);
}

.jp-jupyter-icon-color[fill] {
  fill: var(--jp-jupyter-icon-color, var(--jp-warn-color0));
}

.jp-notebook-icon-color[fill] {
  fill: var(--jp-notebook-icon-color, var(--jp-warn-color0));
}

.jp-json-icon-color[fill] {
  fill: var(--jp-json-icon-color, var(--jp-warn-color1));
}

.jp-console-icon-color[fill] {
  fill: var(--jp-console-icon-color, white);
}

.jp-console-icon-background-color[fill] {
  fill: var(--jp-console-icon-background-color, var(--jp-brand-color1));
}

.jp-terminal-icon-color[fill] {
  fill: var(--jp-terminal-icon-color, var(--jp-layout-color2));
}

.jp-terminal-icon-background-color[fill] {
  fill: var(
    --jp-terminal-icon-background-color,
    var(--jp-inverse-layout-color2)
  );
}

.jp-text-editor-icon-color[fill] {
  fill: var(--jp-text-editor-icon-color, var(--jp-inverse-layout-color3));
}

.jp-inspector-icon-color[fill] {
  fill: var(--jp-inspector-icon-color, var(--jp-inverse-layout-color3));
}

/* CSS for icons in selected filebrowser listing items */
.jp-DirListing-item.jp-mod-selected .jp-icon-selectable[fill] {
  fill: #fff;
}

.jp-DirListing-item.jp-mod-selected .jp-icon-selectable-inverse[fill] {
  fill: var(--jp-brand-color1);
}

/* stylelint-disable selector-max-class, selector-max-compound-selectors */

/**
* TODO: come up with non css-hack solution for showing the busy icon on top
*  of the close icon
* CSS for complex behavior of close icon of tabs in the main area tabbar
*/
.lm-DockPanel-tabBar
  .lm-TabBar-tab.lm-mod-closable.jp-mod-dirty
  > .lm-TabBar-tabCloseIcon
  > :not(:hover)
  > .jp-icon3[fill] {
  fill: none;
}

.lm-DockPanel-tabBar
  .lm-TabBar-tab.lm-mod-closable.jp-mod-dirty
  > .lm-TabBar-tabCloseIcon
  > :not(:hover)
  > .jp-icon-busy[fill] {
  fill: var(--jp-inverse-layout-color3);
}

/* stylelint-enable selector-max-class, selector-max-compound-selectors */

/* CSS for icons in status bar */
#jp-main-statusbar .jp-mod-selected .jp-icon-selectable[fill] {
  fill: #fff;
}

#jp-main-statusbar .jp-mod-selected .jp-icon-selectable-inverse[fill] {
  fill: var(--jp-brand-color1);
}

/* special handling for splash icon CSS. While the theme CSS reloads during
   splash, the splash icon can loose theming. To prevent that, we set a
   default for its color variable */
:root {
  --jp-warn-color0: var(--md-orange-700);
}

/* not sure what to do with this one, used in filebrowser listing */
.jp-DragIcon {
  margin-right: 4px;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/**
 * Support for alt colors for icons as inline SVG HTMLElements
 */

/* alt recolor the primary elements of an icon */
.jp-icon-alt .jp-icon0[fill] {
  fill: var(--jp-layout-color0);
}

.jp-icon-alt .jp-icon1[fill] {
  fill: var(--jp-layout-color1);
}

.jp-icon-alt .jp-icon2[fill] {
  fill: var(--jp-layout-color2);
}

.jp-icon-alt .jp-icon3[fill] {
  fill: var(--jp-layout-color3);
}

.jp-icon-alt .jp-icon4[fill] {
  fill: var(--jp-layout-color4);
}

.jp-icon-alt .jp-icon0[stroke] {
  stroke: var(--jp-layout-color0);
}

.jp-icon-alt .jp-icon1[stroke] {
  stroke: var(--jp-layout-color1);
}

.jp-icon-alt .jp-icon2[stroke] {
  stroke: var(--jp-layout-color2);
}

.jp-icon-alt .jp-icon3[stroke] {
  stroke: var(--jp-layout-color3);
}

.jp-icon-alt .jp-icon4[stroke] {
  stroke: var(--jp-layout-color4);
}

/* alt recolor the accent elements of an icon */
.jp-icon-alt .jp-icon-accent0[fill] {
  fill: var(--jp-inverse-layout-color0);
}

.jp-icon-alt .jp-icon-accent1[fill] {
  fill: var(--jp-inverse-layout-color1);
}

.jp-icon-alt .jp-icon-accent2[fill] {
  fill: var(--jp-inverse-layout-color2);
}

.jp-icon-alt .jp-icon-accent3[fill] {
  fill: var(--jp-inverse-layout-color3);
}

.jp-icon-alt .jp-icon-accent4[fill] {
  fill: var(--jp-inverse-layout-color4);
}

.jp-icon-alt .jp-icon-accent0[stroke] {
  stroke: var(--jp-inverse-layout-color0);
}

.jp-icon-alt .jp-icon-accent1[stroke] {
  stroke: var(--jp-inverse-layout-color1);
}

.jp-icon-alt .jp-icon-accent2[stroke] {
  stroke: var(--jp-inverse-layout-color2);
}

.jp-icon-alt .jp-icon-accent3[stroke] {
  stroke: var(--jp-inverse-layout-color3);
}

.jp-icon-alt .jp-icon-accent4[stroke] {
  stroke: var(--jp-inverse-layout-color4);
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-icon-hoverShow:not(:hover) .jp-icon-hoverShow-content {
  display: none !important;
}

/**
 * Support for hover colors for icons as inline SVG HTMLElements
 */

/**
 * regular colors
 */

/* recolor the primary elements of an icon */
.jp-icon-hover :hover .jp-icon0-hover[fill] {
  fill: var(--jp-inverse-layout-color0);
}

.jp-icon-hover :hover .jp-icon1-hover[fill] {
  fill: var(--jp-inverse-layout-color1);
}

.jp-icon-hover :hover .jp-icon2-hover[fill] {
  fill: var(--jp-inverse-layout-color2);
}

.jp-icon-hover :hover .jp-icon3-hover[fill] {
  fill: var(--jp-inverse-layout-color3);
}

.jp-icon-hover :hover .jp-icon4-hover[fill] {
  fill: var(--jp-inverse-layout-color4);
}

.jp-icon-hover :hover .jp-icon0-hover[stroke] {
  stroke: var(--jp-inverse-layout-color0);
}

.jp-icon-hover :hover .jp-icon1-hover[stroke] {
  stroke: var(--jp-inverse-layout-color1);
}

.jp-icon-hover :hover .jp-icon2-hover[stroke] {
  stroke: var(--jp-inverse-layout-color2);
}

.jp-icon-hover :hover .jp-icon3-hover[stroke] {
  stroke: var(--jp-inverse-layout-color3);
}

.jp-icon-hover :hover .jp-icon4-hover[stroke] {
  stroke: var(--jp-inverse-layout-color4);
}

/* recolor the accent elements of an icon */
.jp-icon-hover :hover .jp-icon-accent0-hover[fill] {
  fill: var(--jp-layout-color0);
}

.jp-icon-hover :hover .jp-icon-accent1-hover[fill] {
  fill: var(--jp-layout-color1);
}

.jp-icon-hover :hover .jp-icon-accent2-hover[fill] {
  fill: var(--jp-layout-color2);
}

.jp-icon-hover :hover .jp-icon-accent3-hover[fill] {
  fill: var(--jp-layout-color3);
}

.jp-icon-hover :hover .jp-icon-accent4-hover[fill] {
  fill: var(--jp-layout-color4);
}

.jp-icon-hover :hover .jp-icon-accent0-hover[stroke] {
  stroke: var(--jp-layout-color0);
}

.jp-icon-hover :hover .jp-icon-accent1-hover[stroke] {
  stroke: var(--jp-layout-color1);
}

.jp-icon-hover :hover .jp-icon-accent2-hover[stroke] {
  stroke: var(--jp-layout-color2);
}

.jp-icon-hover :hover .jp-icon-accent3-hover[stroke] {
  stroke: var(--jp-layout-color3);
}

.jp-icon-hover :hover .jp-icon-accent4-hover[stroke] {
  stroke: var(--jp-layout-color4);
}

/* set the color of an icon to transparent */
.jp-icon-hover :hover .jp-icon-none-hover[fill] {
  fill: none;
}

.jp-icon-hover :hover .jp-icon-none-hover[stroke] {
  stroke: none;
}

/**
 * inverse colors
 */

/* inverse recolor the primary elements of an icon */
.jp-icon-hover.jp-icon-alt :hover .jp-icon0-hover[fill] {
  fill: var(--jp-layout-color0);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon1-hover[fill] {
  fill: var(--jp-layout-color1);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon2-hover[fill] {
  fill: var(--jp-layout-color2);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon3-hover[fill] {
  fill: var(--jp-layout-color3);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon4-hover[fill] {
  fill: var(--jp-layout-color4);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon0-hover[stroke] {
  stroke: var(--jp-layout-color0);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon1-hover[stroke] {
  stroke: var(--jp-layout-color1);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon2-hover[stroke] {
  stroke: var(--jp-layout-color2);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon3-hover[stroke] {
  stroke: var(--jp-layout-color3);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon4-hover[stroke] {
  stroke: var(--jp-layout-color4);
}

/* inverse recolor the accent elements of an icon */
.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent0-hover[fill] {
  fill: var(--jp-inverse-layout-color0);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent1-hover[fill] {
  fill: var(--jp-inverse-layout-color1);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent2-hover[fill] {
  fill: var(--jp-inverse-layout-color2);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent3-hover[fill] {
  fill: var(--jp-inverse-layout-color3);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent4-hover[fill] {
  fill: var(--jp-inverse-layout-color4);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent0-hover[stroke] {
  stroke: var(--jp-inverse-layout-color0);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent1-hover[stroke] {
  stroke: var(--jp-inverse-layout-color1);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent2-hover[stroke] {
  stroke: var(--jp-inverse-layout-color2);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent3-hover[stroke] {
  stroke: var(--jp-inverse-layout-color3);
}

.jp-icon-hover.jp-icon-alt :hover .jp-icon-accent4-hover[stroke] {
  stroke: var(--jp-inverse-layout-color4);
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-IFrame {
  width: 100%;
  height: 100%;
}

.jp-IFrame > iframe {
  border: none;
}

/*
When drag events occur, `lm-mod-override-cursor` is added to the body.
Because iframes steal all cursor events, the following two rules are necessary
to suppress pointer events while resize drags are occurring. There may be a
better solution to this problem.
*/
body.lm-mod-override-cursor .jp-IFrame {
  position: relative;
}

body.lm-mod-override-cursor .jp-IFrame::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: transparent;
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2014-2016, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-HoverBox {
  position: fixed;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-FormGroup-content fieldset {
  border: none;
  padding: 0;
  min-width: 0;
  width: 100%;
}

/* stylelint-disable selector-max-type */

.jp-FormGroup-content fieldset .jp-inputFieldWrapper input,
.jp-FormGroup-content fieldset .jp-inputFieldWrapper select,
.jp-FormGroup-content fieldset .jp-inputFieldWrapper textarea {
  font-size: var(--jp-content-font-size2);
  border-color: var(--jp-input-border-color);
  border-style: solid;
  border-radius: var(--jp-border-radius);
  border-width: 1px;
  padding: 6px 8px;
  background: none;
  color: var(--jp-ui-font-color0);
  height: inherit;
}

.jp-FormGroup-content fieldset input[type='checkbox'] {
  position: relative;
  top: 2px;
  margin-left: 0;
}

.jp-FormGroup-content button.jp-mod-styled {
  cursor: pointer;
}

.jp-FormGroup-content .checkbox label {
  cursor: pointer;
  font-size: var(--jp-content-font-size1);
}

.jp-FormGroup-content .jp-root > fieldset > legend {
  display: none;
}

.jp-FormGroup-content .jp-root > fieldset > p {
  display: none;
}

/** copy of `input.jp-mod-styled:focus` style */
.jp-FormGroup-content fieldset input:focus,
.jp-FormGroup-content fieldset select:focus {
  -moz-outline-radius: unset;
  outline: var(--jp-border-width) solid var(--md-blue-500);
  outline-offset: -1px;
  box-shadow: inset 0 0 4px var(--md-blue-300);
}

.jp-FormGroup-content fieldset input:hover:not(:focus),
.jp-FormGroup-content fieldset select:hover:not(:focus) {
  background-color: var(--jp-border-color2);
}

/* stylelint-enable selector-max-type */

.jp-FormGroup-content .checkbox .field-description {
  /* Disable default description field for checkbox:
   because other widgets do not have description fields,
   we add descriptions to each widget on the field level.
  */
  display: none;
}

.jp-FormGroup-content #root__description {
  display: none;
}

.jp-FormGroup-content .jp-modifiedIndicator {
  width: 5px;
  background-color: var(--jp-brand-color2);
  margin-top: 0;
  margin-left: calc(var(--jp-private-settingeditor-modifier-indent) * -1);
  flex-shrink: 0;
}

.jp-FormGroup-content .jp-modifiedIndicator.jp-errorIndicator {
  background-color: var(--jp-error-color0);
  margin-right: 0.5em;
}

/* RJSF ARRAY style */

.jp-arrayFieldWrapper legend {
  font-size: var(--jp-content-font-size2);
  color: var(--jp-ui-font-color0);
  flex-basis: 100%;
  padding: 4px 0;
  font-weight: var(--jp-content-heading-font-weight);
  border-bottom: 1px solid var(--jp-border-color2);
}

.jp-arrayFieldWrapper .field-description {
  padding: 4px 0;
  white-space: pre-wrap;
}

.jp-arrayFieldWrapper .array-item {
  width: 100%;
  border: 1px solid var(--jp-border-color2);
  border-radius: 4px;
  margin: 4px;
}

.jp-ArrayOperations {
  display: flex;
  margin-left: 8px;
}

.jp-ArrayOperationsButton {
  margin: 2px;
}

.jp-ArrayOperationsButton .jp-icon3[fill] {
  fill: var(--jp-ui-font-color0);
}

button.jp-ArrayOperationsButton.jp-mod-styled:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

/* RJSF form validation error */

.jp-FormGroup-content .validationErrors {
  color: var(--jp-error-color0);
}

/* Hide panel level error as duplicated the field level error */
.jp-FormGroup-content .panel.errors {
  display: none;
}

/* RJSF normal content (settings-editor) */

.jp-FormGroup-contentNormal {
  display: flex;
  align-items: center;
  flex-wrap: wrap;
}

.jp-FormGroup-contentNormal .jp-FormGroup-contentItem {
  margin-left: 7px;
  color: var(--jp-ui-font-color0);
}

.jp-FormGroup-contentNormal .jp-FormGroup-description {
  flex-basis: 100%;
  padding: 4px 7px;
}

.jp-FormGroup-contentNormal .jp-FormGroup-default {
  flex-basis: 100%;
  padding: 4px 7px;
}

.jp-FormGroup-contentNormal .jp-FormGroup-fieldLabel {
  font-size: var(--jp-content-font-size1);
  font-weight: normal;
  min-width: 120px;
}

.jp-FormGroup-contentNormal fieldset:not(:first-child) {
  margin-left: 7px;
}

.jp-FormGroup-contentNormal .field-array-of-string .array-item {
  /* Display `jp-ArrayOperations` buttons side-by-side with content except
    for small screens where flex-wrap will place them one below the other.
  */
  display: flex;
  align-items: center;
  flex-wrap: wrap;
}

.jp-FormGroup-contentNormal .jp-objectFieldWrapper .form-group {
  padding: 2px 8px 2px var(--jp-private-settingeditor-modifier-indent);
  margin-top: 2px;
}

/* RJSF compact content (metadata-form) */

.jp-FormGroup-content.jp-FormGroup-contentCompact {
  width: 100%;
}

.jp-FormGroup-contentCompact .form-group {
  display: flex;
  padding: 0.5em 0.2em 0.5em 0;
}

.jp-FormGroup-contentCompact
  .jp-FormGroup-compactTitle
  .jp-FormGroup-description {
  font-size: var(--jp-ui-font-size1);
  color: var(--jp-ui-font-color2);
}

.jp-FormGroup-contentCompact .jp-FormGroup-fieldLabel {
  padding-bottom: 0.3em;
}

.jp-FormGroup-contentCompact .jp-inputFieldWrapper .form-control {
  width: 100%;
  box-sizing: border-box;
}

.jp-FormGroup-contentCompact .jp-arrayFieldWrapper .jp-FormGroup-compactTitle {
  padding-bottom: 7px;
}

.jp-FormGroup-contentCompact
  .jp-objectFieldWrapper
  .jp-objectFieldWrapper
  .form-group {
  padding: 2px 8px 2px var(--jp-private-settingeditor-modifier-indent);
  margin-top: 2px;
}

.jp-FormGroup-contentCompact ul.error-detail {
  margin-block-start: 0.5em;
  margin-block-end: 0.5em;
  padding-inline-start: 1em;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

.jp-SidePanel {
  display: flex;
  flex-direction: column;
  min-width: var(--jp-sidebar-min-width);
  overflow-y: auto;
  color: var(--jp-ui-font-color1);
  background: var(--jp-layout-color1);
  font-size: var(--jp-ui-font-size1);
}

.jp-SidePanel-header {
  flex: 0 0 auto;
  display: flex;
  border-bottom: var(--jp-border-width) solid var(--jp-border-color2);
  font-size: var(--jp-ui-font-size0);
  font-weight: 600;
  letter-spacing: 1px;
  margin: 0;
  padding: 2px;
  text-transform: uppercase;
}

.jp-SidePanel-toolbar {
  flex: 0 0 auto;
}

.jp-SidePanel-content {
  flex: 1 1 auto;
}

.jp-SidePanel-toolbar,
.jp-AccordionPanel-toolbar {
  height: var(--jp-private-toolbar-height);
}

.jp-SidePanel-toolbar.jp-Toolbar-micro {
  display: none;
}

.lm-AccordionPanel .jp-AccordionPanel-title {
  box-sizing: border-box;
  line-height: 25px;
  margin: 0;
  display: flex;
  align-items: center;
  background: var(--jp-layout-color1);
  color: var(--jp-ui-font-color1);
  border-bottom: var(--jp-border-width) solid var(--jp-toolbar-border-color);
  box-shadow: var(--jp-toolbar-box-shadow);
  font-size: var(--jp-ui-font-size0);
}

.jp-AccordionPanel-title {
  cursor: pointer;
  user-select: none;
  -moz-user-select: none;
  -webkit-user-select: none;
  text-transform: uppercase;
}

.lm-AccordionPanel[data-orientation='horizontal'] > .jp-AccordionPanel-title {
  /* Title is rotated for horizontal accordion panel using CSS */
  display: block;
  transform-origin: top left;
  transform: rotate(-90deg) translate(-100%);
}

.jp-AccordionPanel-title .lm-AccordionPanel-titleLabel {
  user-select: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.jp-AccordionPanel-title .lm-AccordionPanel-titleCollapser {
  transform: rotate(-90deg);
  margin: auto 0;
  height: 16px;
}

.jp-AccordionPanel-title.lm-mod-expanded .lm-AccordionPanel-titleCollapser {
  transform: rotate(0deg);
}

.lm-AccordionPanel .jp-AccordionPanel-toolbar {
  background: none;
  box-shadow: none;
  border: none;
  margin-left: auto;
}

.lm-AccordionPanel .lm-SplitPanel-handle:hover {
  background: var(--jp-layout-color3);
}

.jp-text-truncated {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2017, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-Spinner {
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 10;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: var(--jp-layout-color0);
  outline: none;
}

.jp-SpinnerContent {
  font-size: 10px;
  margin: 50px auto;
  text-indent: -9999em;
  width: 3em;
  height: 3em;
  border-radius: 50%;
  background: var(--jp-brand-color3);
  background: linear-gradient(
    to right,
    #f37626 10%,
    rgba(255, 255, 255, 0) 42%
  );
  position: relative;
  animation: load3 1s infinite linear, fadeIn 1s;
}

.jp-SpinnerContent::before {
  width: 50%;
  height: 50%;
  background: #f37626;
  border-radius: 100% 0 0;
  position: absolute;
  top: 0;
  left: 0;
  content: '';
}

.jp-SpinnerContent::after {
  background: var(--jp-layout-color0);
  width: 75%;
  height: 75%;
  border-radius: 50%;
  content: '';
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@keyframes load3 {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2014-2017, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

button.jp-mod-styled {
  font-size: var(--jp-ui-font-size1);
  color: var(--jp-ui-font-color0);
  border: none;
  box-sizing: border-box;
  text-align: center;
  line-height: 32px;
  height: 32px;
  padding: 0 12px;
  letter-spacing: 0.8px;
  outline: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
}

input.jp-mod-styled {
  background: var(--jp-input-background);
  height: 28px;
  box-sizing: border-box;
  border: var(--jp-border-width) solid var(--jp-border-color1);
  padding-left: 7px;
  padding-right: 7px;
  font-size: var(--jp-ui-font-size2);
  color: var(--jp-ui-font-color0);
  outline: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
}

input[type='checkbox'].jp-mod-styled {
  appearance: checkbox;
  -webkit-appearance: checkbox;
  -moz-appearance: checkbox;
  height: auto;
}

input.jp-mod-styled:focus {
  border: var(--jp-border-width) solid var(--md-blue-500);
  box-shadow: inset 0 0 4px var(--md-blue-300);
}

.jp-select-wrapper {
  display: flex;
  position: relative;
  flex-direction: column;
  padding: 1px;
  background-color: var(--jp-layout-color1);
  box-sizing: border-box;
  margin-bottom: 12px;
}

.jp-select-wrapper:not(.multiple) {
  height: 28px;
}

.jp-select-wrapper.jp-mod-focused select.jp-mod-styled {
  border: var(--jp-border-width) solid var(--jp-input-active-border-color);
  box-shadow: var(--jp-input-box-shadow);
  background-color: var(--jp-input-active-background);
}

select.jp-mod-styled:hover {
  cursor: pointer;
  color: var(--jp-ui-font-color0);
  background-color: var(--jp-input-hover-background);
  box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.5);
}

select.jp-mod-styled {
  flex: 1 1 auto;
  width: 100%;
  font-size: var(--jp-ui-font-size2);
  background: var(--jp-input-background);
  color: var(--jp-ui-font-color0);
  padding: 0 25px 0 8px;
  border: var(--jp-border-width) solid var(--jp-input-border-color);
  border-radius: 0;
  outline: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
}

select.jp-mod-styled:not([multiple]) {
  height: 32px;
}

select.jp-mod-styled[multiple] {
  max-height: 200px;
  overflow-y: auto;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-switch {
  display: flex;
  align-items: center;
  padding-left: 4px;
  padding-right: 4px;
  font-size: var(--jp-ui-font-size1);
  background-color: transparent;
  color: var(--jp-ui-font-color1);
  border: none;
  height: 20px;
}

.jp-switch:hover {
  background-color: var(--jp-layout-color2);
}

.jp-switch-label {
  margin-right: 5px;
  font-family: var(--jp-ui-font-family);
}

.jp-switch-track {
  cursor: pointer;
  background-color: var(--jp-switch-color, var(--jp-border-color1));
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 34px;
  height: 16px;
  width: 35px;
  position: relative;
}

.jp-switch-track::before {
  content: '';
  position: absolute;
  height: 10px;
  width: 10px;
  margin: 3px;
  left: 0;
  background-color: var(--jp-ui-inverse-font-color1);
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 50%;
}

.jp-switch[aria-checked='true'] .jp-switch-track {
  background-color: var(--jp-switch-true-position-color, var(--jp-warn-color0));
}

.jp-switch[aria-checked='true'] .jp-switch-track::before {
  /* track width (35) - margins (3 + 3) - thumb width (10) */
  left: 19px;
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2014-2016, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

:root {
  --jp-private-toolbar-height: calc(
    28px + var(--jp-border-width)
  ); /* leave 28px for content */
}

.jp-Toolbar {
  color: var(--jp-ui-font-color1);
  flex: 0 0 auto;
  display: flex;
  flex-direction: row;
  border-bottom: var(--jp-border-width) solid var(--jp-toolbar-border-color);
  box-shadow: var(--jp-toolbar-box-shadow);
  background: var(--jp-toolbar-background);
  min-height: var(--jp-toolbar-micro-height);
  padding: 2px;
  z-index: 8;
  overflow-x: hidden;
}

/* Toolbar items */

.jp-Toolbar > .jp-Toolbar-item.jp-Toolbar-spacer {
  flex-grow: 1;
  flex-shrink: 1;
}

.jp-Toolbar-item.jp-Toolbar-kernelStatus {
  display: inline-block;
  width: 32px;
  background-repeat: no-repeat;
  background-position: center;
  background-size: 16px;
}

.jp-Toolbar > .jp-Toolbar-item {
  flex: 0 0 auto;
  display: flex;
  padding-left: 1px;
  padding-right: 1px;
  font-size: var(--jp-ui-font-size1);
  line-height: var(--jp-private-toolbar-height);
  height: 100%;
}

/* Toolbar buttons */

/* This is the div we use to wrap the react component into a Widget */
div.jp-ToolbarButton {
  color: transparent;
  border: none;
  box-sizing: border-box;
  outline: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  padding: 0;
  margin: 0;
}

button.jp-ToolbarButtonComponent {
  background: var(--jp-layout-color1);
  border: none;
  box-sizing: border-box;
  outline: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  padding: 0 6px;
  margin: 0;
  height: 24px;
  border-radius: var(--jp-border-radius);
  display: flex;
  align-items: center;
  text-align: center;
  font-size: 14px;
  min-width: unset;
  min-height: unset;
}

button.jp-ToolbarButtonComponent:disabled {
  opacity: 0.4;
}

button.jp-ToolbarButtonComponent > span {
  padding: 0;
  flex: 0 0 auto;
}

button.jp-ToolbarButtonComponent .jp-ToolbarButtonComponent-label {
  font-size: var(--jp-ui-font-size1);
  line-height: 100%;
  padding-left: 2px;
  color: var(--jp-ui-font-color1);
  font-family: var(--jp-ui-font-family);
}

#jp-main-dock-panel[data-mode='single-document']
  .jp-MainAreaWidget
  > .jp-Toolbar.jp-Toolbar-micro {
  padding: 0;
  min-height: 0;
}

#jp-main-dock-panel[data-mode='single-document']
  .jp-MainAreaWidget
  > .jp-Toolbar {
  border: none;
  box-shadow: none;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

.jp-WindowedPanel-outer {
  position: relative;
  overflow-y: auto;
}

.jp-WindowedPanel-inner {
  position: relative;
}

.jp-WindowedPanel-window {
  position: absolute;
  left: 0;
  right: 0;
  overflow: visible;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/* Sibling imports */

body {
  color: var(--jp-ui-font-color1);
  font-size: var(--jp-ui-font-size1);
}

/* Disable native link decoration styles everywhere outside of dialog boxes */
a {
  text-decoration: unset;
  color: unset;
}

a:hover {
  text-decoration: unset;
  color: unset;
}

/* Accessibility for links inside dialog box text */
.jp-Dialog-content a {
  text-decoration: revert;
  color: var(--jp-content-link-color);
}

.jp-Dialog-content a:hover {
  text-decoration: revert;
}

/* Styles for ui-components */
.jp-Button {
  color: var(--jp-ui-font-color2);
  border-radius: var(--jp-border-radius);
  padding: 0 12px;
  font-size: var(--jp-ui-font-size1);

  /* Copy from blueprint 3 */
  display: inline-flex;
  flex-direction: row;
  border: none;
  cursor: pointer;
  align-items: center;
  justify-content: center;
  text-align: left;
  vertical-align: middle;
  min-height: 30px;
  min-width: 30px;
}

.jp-Button:disabled {
  cursor: not-allowed;
}

.jp-Button:empty {
  padding: 0 !important;
}

.jp-Button.jp-mod-small {
  min-height: 24px;
  min-width: 24px;
  font-size: 12px;
  padding: 0 7px;
}

/* Use our own theme for hover styles */
.jp-Button.jp-mod-minimal:hover {
  background-color: var(--jp-layout-color2);
}

.jp-Button.jp-mod-minimal {
  background: none;
}

.jp-InputGroup {
  display: block;
  position: relative;
}

.jp-InputGroup input {
  box-sizing: border-box;
  border: none;
  border-radius: 0;
  background-color: transparent;
  color: var(--jp-ui-font-color0);
  box-shadow: inset 0 0 0 var(--jp-border-width) var(--jp-input-border-color);
  padding-bottom: 0;
  padding-top: 0;
  padding-left: 10px;
  padding-right: 28px;
  position: relative;
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: 14px;
  font-weight: 400;
  height: 30px;
  line-height: 30px;
  outline: none;
  vertical-align: middle;
}

.jp-InputGroup input:focus {
  box-shadow: inset 0 0 0 var(--jp-border-width)
      var(--jp-input-active-box-shadow-color),
    inset 0 0 0 3px var(--jp-input-active-box-shadow-color);
}

.jp-InputGroup input:disabled {
  cursor: not-allowed;
  resize: block;
  background-color: var(--jp-layout-color2);
  color: var(--jp-ui-font-color2);
}

.jp-InputGroup input:disabled ~ span {
  cursor: not-allowed;
  color: var(--jp-ui-font-color2);
}

.jp-InputGroup input::placeholder,
input::placeholder {
  color: var(--jp-ui-font-color2);
}

.jp-InputGroupAction {
  position: absolute;
  bottom: 1px;
  right: 0;
  padding: 6px;
}

.jp-HTMLSelect.jp-DefaultStyle select {
  background-color: initial;
  border: none;
  border-radius: 0;
  box-shadow: none;
  color: var(--jp-ui-font-color0);
  display: block;
  font-size: var(--jp-ui-font-size1);
  font-family: var(--jp-ui-font-family);
  height: 24px;
  line-height: 14px;
  padding: 0 25px 0 10px;
  text-align: left;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.jp-HTMLSelect.jp-DefaultStyle select:disabled {
  background-color: var(--jp-layout-color2);
  color: var(--jp-ui-font-color2);
  cursor: not-allowed;
  resize: block;
}

.jp-HTMLSelect.jp-DefaultStyle select:disabled ~ span {
  cursor: not-allowed;
}

/* Use our own theme for hover and option styles */
/* stylelint-disable-next-line selector-max-type */
.jp-HTMLSelect.jp-DefaultStyle select:hover,
.jp-HTMLSelect.jp-DefaultStyle select > option {
  background-color: var(--jp-layout-color2);
  color: var(--jp-ui-font-color0);
}

select {
  box-sizing: border-box;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Styles
|----------------------------------------------------------------------------*/

.jp-StatusBar-Widget {
  display: flex;
  align-items: center;
  background: var(--jp-layout-color2);
  min-height: var(--jp-statusbar-height);
  justify-content: space-between;
  padding: 0 10px;
}

.jp-StatusBar-Left {
  display: flex;
  align-items: center;
  flex-direction: row;
}

.jp-StatusBar-Middle {
  display: flex;
  align-items: center;
}

.jp-StatusBar-Right {
  display: flex;
  align-items: center;
  flex-direction: row-reverse;
}

.jp-StatusBar-Item {
  max-height: var(--jp-statusbar-height);
  margin: 0 2px;
  height: var(--jp-statusbar-height);
  white-space: nowrap;
  text-overflow: ellipsis;
  color: var(--jp-ui-font-color1);
  padding: 0 6px;
}

.jp-mod-highlighted:hover {
  background-color: var(--jp-layout-color3);
}

.jp-mod-clicked {
  background-color: var(--jp-brand-color1);
}

.jp-mod-clicked:hover {
  background-color: var(--jp-brand-color0);
}

.jp-mod-clicked .jp-StatusBar-TextItem {
  color: var(--jp-ui-inverse-font-color1);
}

.jp-StatusBar-HoverItem {
  box-shadow: '0px 4px 4px rgba(0, 0, 0, 0.25)';
}

.jp-StatusBar-TextItem {
  font-size: var(--jp-ui-font-size1);
  font-family: var(--jp-ui-font-family);
  line-height: 24px;
  color: var(--jp-ui-font-color1);
}

.jp-StatusBar-GroupItem {
  display: flex;
  align-items: center;
  flex-direction: row;
}

.jp-Statusbar-ProgressCircle svg {
  display: block;
  margin: 0 auto;
  width: 16px;
  height: 24px;
  align-self: normal;
}

.jp-Statusbar-ProgressCircle path {
  fill: var(--jp-inverse-layout-color3);
}

.jp-Statusbar-ProgressBar-progress-bar {
  height: 10px;
  width: 100px;
  border: solid 0.25px var(--jp-brand-color2);
  border-radius: 3px;
  overflow: hidden;
  align-self: center;
}

.jp-Statusbar-ProgressBar-progress-bar > div {
  background-color: var(--jp-brand-color2);
  background-image: linear-gradient(
    -45deg,
    rgba(255, 255, 255, 0.2) 25%,
    transparent 25%,
    transparent 50%,
    rgba(255, 255, 255, 0.2) 50%,
    rgba(255, 255, 255, 0.2) 75%,
    transparent 75%,
    transparent
  );
  background-size: 40px 40px;
  float: left;
  width: 0%;
  height: 100%;
  font-size: 12px;
  line-height: 14px;
  color: #fff;
  text-align: center;
  animation: jp-Statusbar-ExecutionTime-progress-bar 2s linear infinite;
}

.jp-Statusbar-ProgressBar-progress-bar p {
  color: var(--jp-ui-font-color1);
  font-family: var(--jp-ui-font-family);
  font-size: var(--jp-ui-font-size1);
  line-height: 10px;
  width: 100px;
}

@keyframes jp-Statusbar-ExecutionTime-progress-bar {
  0% {
    background-position: 0 0;
  }

  100% {
    background-position: 40px 40px;
  }
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Variables
|----------------------------------------------------------------------------*/

:root {
  --jp-private-commandpalette-search-height: 28px;
}

/*-----------------------------------------------------------------------------
| Overall styles
|----------------------------------------------------------------------------*/

.lm-CommandPalette {
  padding-bottom: 0;
  color: var(--jp-ui-font-color1);
  background: var(--jp-layout-color1);

  /* This is needed so that all font sizing of children done in ems is
   * relative to this base size */
  font-size: var(--jp-ui-font-size1);
}

/*-----------------------------------------------------------------------------
| Modal variant
|----------------------------------------------------------------------------*/

.jp-ModalCommandPalette {
  position: absolute;
  z-index: 10000;
  top: 38px;
  left: 30%;
  margin: 0;
  padding: 4px;
  width: 40%;
  box-shadow: var(--jp-elevation-z4);
  border-radius: 4px;
  background: var(--jp-layout-color0);
}

.jp-ModalCommandPalette .lm-CommandPalette {
  max-height: 40vh;
}

.jp-ModalCommandPalette .lm-CommandPalette .lm-close-icon::after {
  display: none;
}

.jp-ModalCommandPalette .lm-CommandPalette .lm-CommandPalette-header {
  display: none;
}

.jp-ModalCommandPalette .lm-CommandPalette .lm-CommandPalette-item {
  margin-left: 4px;
  margin-right: 4px;
}

.jp-ModalCommandPalette
  .lm-CommandPalette
  .lm-CommandPalette-item.lm-mod-disabled {
  display: none;
}

/*-----------------------------------------------------------------------------
| Search
|----------------------------------------------------------------------------*/

.lm-CommandPalette-search {
  padding: 4px;
  background-color: var(--jp-layout-color1);
  z-index: 2;
}

.lm-CommandPalette-wrapper {
  overflow: overlay;
  padding: 0 9px;
  background-color: var(--jp-input-active-background);
  height: 30px;
  box-shadow: inset 0 0 0 var(--jp-border-width) var(--jp-input-border-color);
}

.lm-CommandPalette.lm-mod-focused .lm-CommandPalette-wrapper {
  box-shadow: inset 0 0 0 1px var(--jp-input-active-box-shadow-color),
    inset 0 0 0 3px var(--jp-input-active-box-shadow-color);
}

.jp-SearchIconGroup {
  color: white;
  background-color: var(--jp-brand-color1);
  position: absolute;
  top: 4px;
  right: 4px;
  padding: 5px 5px 1px;
}

.jp-SearchIconGroup svg {
  height: 20px;
  width: 20px;
}

.jp-SearchIconGroup .jp-icon3[fill] {
  fill: var(--jp-layout-color0);
}

.lm-CommandPalette-input {
  background: transparent;
  width: calc(100% - 18px);
  float: left;
  border: none;
  outline: none;
  font-size: var(--jp-ui-font-size1);
  color: var(--jp-ui-font-color0);
  line-height: var(--jp-private-commandpalette-search-height);
}

.lm-CommandPalette-input::-webkit-input-placeholder,
.lm-CommandPalette-input::-moz-placeholder,
.lm-CommandPalette-input:-ms-input-placeholder {
  color: var(--jp-ui-font-color2);
  font-size: var(--jp-ui-font-size1);
}

/*-----------------------------------------------------------------------------
| Results
|----------------------------------------------------------------------------*/

.lm-CommandPalette-header:first-child {
  margin-top: 0;
}

.lm-CommandPalette-header {
  border-bottom: solid var(--jp-border-width) var(--jp-border-color2);
  color: var(--jp-ui-font-color1);
  cursor: pointer;
  display: flex;
  font-size: var(--jp-ui-font-size0);
  font-weight: 600;
  letter-spacing: 1px;
  margin-top: 8px;
  padding: 8px 0 8px 12px;
  text-transform: uppercase;
}

.lm-CommandPalette-header.lm-mod-active {
  background: var(--jp-layout-color2);
}

.lm-CommandPalette-header > mark {
  background-color: transparent;
  font-weight: bold;
  color: var(--jp-ui-font-color1);
}

.lm-CommandPalette-item {
  padding: 4px 12px 4px 4px;
  color: var(--jp-ui-font-color1);
  font-size: var(--jp-ui-font-size1);
  font-weight: 400;
  display: flex;
}

.lm-CommandPalette-item.lm-mod-disabled {
  color: var(--jp-ui-font-color2);
}

.lm-CommandPalette-item.lm-mod-active {
  color: var(--jp-ui-inverse-font-color1);
  background: var(--jp-brand-color1);
}

.lm-CommandPalette-item.lm-mod-active .lm-CommandPalette-itemLabel > mark {
  color: var(--jp-ui-inverse-font-color0);
}

.lm-CommandPalette-item.lm-mod-active .jp-icon-selectable[fill] {
  fill: var(--jp-layout-color0);
}

.lm-CommandPalette-item.lm-mod-active:hover:not(.lm-mod-disabled) {
  color: var(--jp-ui-inverse-font-color1);
  background: var(--jp-brand-color1);
}

.lm-CommandPalette-item:hover:not(.lm-mod-active):not(.lm-mod-disabled) {
  background: var(--jp-layout-color2);
}

.lm-CommandPalette-itemContent {
  overflow: hidden;
}

.lm-CommandPalette-itemLabel > mark {
  color: var(--jp-ui-font-color0);
  background-color: transparent;
  font-weight: bold;
}

.lm-CommandPalette-item.lm-mod-disabled mark {
  color: var(--jp-ui-font-color2);
}

.lm-CommandPalette-item .lm-CommandPalette-itemIcon {
  margin: 0 4px 0 0;
  position: relative;
  width: 16px;
  top: 2px;
  flex: 0 0 auto;
}

.lm-CommandPalette-item.lm-mod-disabled .lm-CommandPalette-itemIcon {
  opacity: 0.6;
}

.lm-CommandPalette-item .lm-CommandPalette-itemShortcut {
  flex: 0 0 auto;
}

.lm-CommandPalette-itemCaption {
  display: none;
}

.lm-CommandPalette-content {
  background-color: var(--jp-layout-color1);
}

.lm-CommandPalette-content:empty::after {
  content: 'No results';
  margin: auto;
  margin-top: 20px;
  width: 100px;
  display: block;
  font-size: var(--jp-ui-font-size2);
  font-family: var(--jp-ui-font-family);
  font-weight: lighter;
}

.lm-CommandPalette-emptyMessage {
  text-align: center;
  margin-top: 24px;
  line-height: 1.32;
  padding: 0 8px;
  color: var(--jp-content-font-color3);
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2014-2017, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-Dialog {
  position: absolute;
  z-index: 10000;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  width: 100%;
  height: 100%;
  background: var(--jp-dialog-background);
}

.jp-Dialog-content {
  display: flex;
  flex-direction: column;
  margin-left: auto;
  margin-right: auto;
  background: var(--jp-layout-color1);
  padding: 24px 24px 12px;
  min-width: 300px;
  min-height: 150px;
  max-width: 1000px;
  max-height: 500px;
  box-sizing: border-box;
  box-shadow: var(--jp-elevation-z20);
  word-wrap: break-word;
  border-radius: var(--jp-border-radius);

  /* This is needed so that all font sizing of children done in ems is
   * relative to this base size */
  font-size: var(--jp-ui-font-size1);
  color: var(--jp-ui-font-color1);
  resize: both;
}

.jp-Dialog-content.jp-Dialog-content-small {
  max-width: 500px;
}

.jp-Dialog-button {
  overflow: visible;
}

button.jp-Dialog-button:focus {
  outline: 1px solid var(--jp-brand-color1);
  outline-offset: 4px;
  -moz-outline-radius: 0;
}

button.jp-Dialog-button:focus::-moz-focus-inner {
  border: 0;
}

button.jp-Dialog-button.jp-mod-styled.jp-mod-accept:focus,
button.jp-Dialog-button.jp-mod-styled.jp-mod-warn:focus,
button.jp-Dialog-button.jp-mod-styled.jp-mod-reject:focus {
  outline-offset: 4px;
  -moz-outline-radius: 0;
}

button.jp-Dialog-button.jp-mod-styled.jp-mod-accept:focus {
  outline: 1px solid var(--jp-accept-color-normal, var(--jp-brand-color1));
}

button.jp-Dialog-button.jp-mod-styled.jp-mod-warn:focus {
  outline: 1px solid var(--jp-warn-color-normal, var(--jp-error-color1));
}

button.jp-Dialog-button.jp-mod-styled.jp-mod-reject:focus {
  outline: 1px solid var(--jp-reject-color-normal, var(--md-grey-600));
}

button.jp-Dialog-close-button {
  padding: 0;
  height: 100%;
  min-width: unset;
  min-height: unset;
}

.jp-Dialog-header {
  display: flex;
  justify-content: space-between;
  flex: 0 0 auto;
  padding-bottom: 12px;
  font-size: var(--jp-ui-font-size3);
  font-weight: 400;
  color: var(--jp-ui-font-color1);
}

.jp-Dialog-body {
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
  font-size: var(--jp-ui-font-size1);
  background: var(--jp-layout-color1);
  color: var(--jp-ui-font-color1);
  overflow: auto;
}

.jp-Dialog-footer {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  flex: 0 0 auto;
  margin-left: -12px;
  margin-right: -12px;
  padding: 12px;
}

.jp-Dialog-checkbox {
  padding-right: 5px;
}

.jp-Dialog-checkbox > input:focus-visible {
  outline: 1px solid var(--jp-input-active-border-color);
  outline-offset: 1px;
}

.jp-Dialog-spacer {
  flex: 1 1 auto;
}

.jp-Dialog-title {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.jp-Dialog-body > .jp-select-wrapper {
  width: 100%;
}

.jp-Dialog-body > button {
  padding: 0 16px;
}

.jp-Dialog-body > label {
  line-height: 1.4;
  color: var(--jp-ui-font-color0);
}

.jp-Dialog-button.jp-mod-styled:not(:last-child) {
  margin-right: 12px;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

.jp-Input-Boolean-Dialog {
  flex-direction: row-reverse;
  align-items: end;
  width: 100%;
}

.jp-Input-Boolean-Dialog > label {
  flex: 1 1 auto;
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2014-2016, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-MainAreaWidget > :focus {
  outline: none;
}

.jp-MainAreaWidget .jp-MainAreaWidget-error {
  padding: 6px;
}

.jp-MainAreaWidget .jp-MainAreaWidget-error > pre {
  width: auto;
  padding: 10px;
  background: var(--jp-error-color3);
  border: var(--jp-border-width) solid var(--jp-error-color1);
  border-radius: var(--jp-border-radius);
  color: var(--jp-ui-font-color1);
  font-size: var(--jp-ui-font-size1);
  white-space: pre-wrap;
  word-wrap: break-word;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/**
 * google-material-color v1.2.6
 * https://github.com/danlevan/google-material-color
 */
:root {
  --md-red-50: #ffebee;
  --md-red-100: #ffcdd2;
  --md-red-200: #ef9a9a;
  --md-red-300: #e57373;
  --md-red-400: #ef5350;
  --md-red-500: #f44336;
  --md-red-600: #e53935;
  --md-red-700: #d32f2f;
  --md-red-800: #c62828;
  --md-red-900: #b71c1c;
  --md-red-A100: #ff8a80;
  --md-red-A200: #ff5252;
  --md-red-A400: #ff1744;
  --md-red-A700: #d50000;
  --md-pink-50: #fce4ec;
  --md-pink-100: #f8bbd0;
  --md-pink-200: #f48fb1;
  --md-pink-300: #f06292;
  --md-pink-400: #ec407a;
  --md-pink-500: #e91e63;
  --md-pink-600: #d81b60;
  --md-pink-700: #c2185b;
  --md-pink-800: #ad1457;
  --md-pink-900: #880e4f;
  --md-pink-A100: #ff80ab;
  --md-pink-A200: #ff4081;
  --md-pink-A400: #f50057;
  --md-pink-A700: #c51162;
  --md-purple-50: #f3e5f5;
  --md-purple-100: #e1bee7;
  --md-purple-200: #ce93d8;
  --md-purple-300: #ba68c8;
  --md-purple-400: #ab47bc;
  --md-purple-500: #9c27b0;
  --md-purple-600: #8e24aa;
  --md-purple-700: #7b1fa2;
  --md-purple-800: #6a1b9a;
  --md-purple-900: #4a148c;
  --md-purple-A100: #ea80fc;
  --md-purple-A200: #e040fb;
  --md-purple-A400: #d500f9;
  --md-purple-A700: #a0f;
  --md-deep-purple-50: #ede7f6;
  --md-deep-purple-100: #d1c4e9;
  --md-deep-purple-200: #b39ddb;
  --md-deep-purple-300: #9575cd;
  --md-deep-purple-400: #7e57c2;
  --md-deep-purple-500: #673ab7;
  --md-deep-purple-600: #5e35b1;
  --md-deep-purple-700: #512da8;
  --md-deep-purple-800: #4527a0;
  --md-deep-purple-900: #311b92;
  --md-deep-purple-A100: #b388ff;
  --md-deep-purple-A200: #7c4dff;
  --md-deep-purple-A400: #651fff;
  --md-deep-purple-A700: #6200ea;
  --md-indigo-50: #e8eaf6;
  --md-indigo-100: #c5cae9;
  --md-indigo-200: #9fa8da;
  --md-indigo-300: #7986cb;
  --md-indigo-400: #5c6bc0;
  --md-indigo-500: #3f51b5;
  --md-indigo-600: #3949ab;
  --md-indigo-700: #303f9f;
  --md-indigo-800: #283593;
  --md-indigo-900: #1a237e;
  --md-indigo-A100: #8c9eff;
  --md-indigo-A200: #536dfe;
  --md-indigo-A400: #3d5afe;
  --md-indigo-A700: #304ffe;
  --md-blue-50: #e3f2fd;
  --md-blue-100: #bbdefb;
  --md-blue-200: #90caf9;
  --md-blue-300: #64b5f6;
  --md-blue-400: #42a5f5;
  --md-blue-500: #2196f3;
  --md-blue-600: #1e88e5;
  --md-blue-700: #1976d2;
  --md-blue-800: #1565c0;
  --md-blue-900: #0d47a1;
  --md-blue-A100: #82b1ff;
  --md-blue-A200: #448aff;
  --md-blue-A400: #2979ff;
  --md-blue-A700: #2962ff;
  --md-light-blue-50: #e1f5fe;
  --md-light-blue-100: #b3e5fc;
  --md-light-blue-200: #81d4fa;
  --md-light-blue-300: #4fc3f7;
  --md-light-blue-400: #29b6f6;
  --md-light-blue-500: #03a9f4;
  --md-light-blue-600: #039be5;
  --md-light-blue-700: #0288d1;
  --md-light-blue-800: #0277bd;
  --md-light-blue-900: #01579b;
  --md-light-blue-A100: #80d8ff;
  --md-light-blue-A200: #40c4ff;
  --md-light-blue-A400: #00b0ff;
  --md-light-blue-A700: #0091ea;
  --md-cyan-50: #e0f7fa;
  --md-cyan-100: #b2ebf2;
  --md-cyan-200: #80deea;
  --md-cyan-300: #4dd0e1;
  --md-cyan-400: #26c6da;
  --md-cyan-500: #00bcd4;
  --md-cyan-600: #00acc1;
  --md-cyan-700: #0097a7;
  --md-cyan-800: #00838f;
  --md-cyan-900: #006064;
  --md-cyan-A100: #84ffff;
  --md-cyan-A200: #18ffff;
  --md-cyan-A400: #00e5ff;
  --md-cyan-A700: #00b8d4;
  --md-teal-50: #e0f2f1;
  --md-teal-100: #b2dfdb;
  --md-teal-200: #80cbc4;
  --md-teal-300: #4db6ac;
  --md-teal-400: #26a69a;
  --md-teal-500: #009688;
  --md-teal-600: #00897b;
  --md-teal-700: #00796b;
  --md-teal-800: #00695c;
  --md-teal-900: #004d40;
  --md-teal-A100: #a7ffeb;
  --md-teal-A200: #64ffda;
  --md-teal-A400: #1de9b6;
  --md-teal-A700: #00bfa5;
  --md-green-50: #e8f5e9;
  --md-green-100: #c8e6c9;
  --md-green-200: #a5d6a7;
  --md-green-300: #81c784;
  --md-green-400: #66bb6a;
  --md-green-500: #4caf50;
  --md-green-600: #43a047;
  --md-green-700: #388e3c;
  --md-green-800: #2e7d32;
  --md-green-900: #1b5e20;
  --md-green-A100: #b9f6ca;
  --md-green-A200: #69f0ae;
  --md-green-A400: #00e676;
  --md-green-A700: #00c853;
  --md-light-green-50: #f1f8e9;
  --md-light-green-100: #dcedc8;
  --md-light-green-200: #c5e1a5;
  --md-light-green-300: #aed581;
  --md-light-green-400: #9ccc65;
  --md-light-green-500: #8bc34a;
  --md-light-green-600: #7cb342;
  --md-light-green-700: #689f38;
  --md-light-green-800: #558b2f;
  --md-light-green-900: #33691e;
  --md-light-green-A100: #ccff90;
  --md-light-green-A200: #b2ff59;
  --md-light-green-A400: #76ff03;
  --md-light-green-A700: #64dd17;
  --md-lime-50: #f9fbe7;
  --md-lime-100: #f0f4c3;
  --md-lime-200: #e6ee9c;
  --md-lime-300: #dce775;
  --md-lime-400: #d4e157;
  --md-lime-500: #cddc39;
  --md-lime-600: #c0ca33;
  --md-lime-700: #afb42b;
  --md-lime-800: #9e9d24;
  --md-lime-900: #827717;
  --md-lime-A100: #f4ff81;
  --md-lime-A200: #eeff41;
  --md-lime-A400: #c6ff00;
  --md-lime-A700: #aeea00;
  --md-yellow-50: #fffde7;
  --md-yellow-100: #fff9c4;
  --md-yellow-200: #fff59d;
  --md-yellow-300: #fff176;
  --md-yellow-400: #ffee58;
  --md-yellow-500: #ffeb3b;
  --md-yellow-600: #fdd835;
  --md-yellow-700: #fbc02d;
  --md-yellow-800: #f9a825;
  --md-yellow-900: #f57f17;
  --md-yellow-A100: #ffff8d;
  --md-yellow-A200: #ff0;
  --md-yellow-A400: #ffea00;
  --md-yellow-A700: #ffd600;
  --md-amber-50: #fff8e1;
  --md-amber-100: #ffecb3;
  --md-amber-200: #ffe082;
  --md-amber-300: #ffd54f;
  --md-amber-400: #ffca28;
  --md-amber-500: #ffc107;
  --md-amber-600: #ffb300;
  --md-amber-700: #ffa000;
  --md-amber-800: #ff8f00;
  --md-amber-900: #ff6f00;
  --md-amber-A100: #ffe57f;
  --md-amber-A200: #ffd740;
  --md-amber-A400: #ffc400;
  --md-amber-A700: #ffab00;
  --md-orange-50: #fff3e0;
  --md-orange-100: #ffe0b2;
  --md-orange-200: #ffcc80;
  --md-orange-300: #ffb74d;
  --md-orange-400: #ffa726;
  --md-orange-500: #ff9800;
  --md-orange-600: #fb8c00;
  --md-orange-700: #f57c00;
  --md-orange-800: #ef6c00;
  --md-orange-900: #e65100;
  --md-orange-A100: #ffd180;
  --md-orange-A200: #ffab40;
  --md-orange-A400: #ff9100;
  --md-orange-A700: #ff6d00;
  --md-deep-orange-50: #fbe9e7;
  --md-deep-orange-100: #ffccbc;
  --md-deep-orange-200: #ffab91;
  --md-deep-orange-300: #ff8a65;
  --md-deep-orange-400: #ff7043;
  --md-deep-orange-500: #ff5722;
  --md-deep-orange-600: #f4511e;
  --md-deep-orange-700: #e64a19;
  --md-deep-orange-800: #d84315;
  --md-deep-orange-900: #bf360c;
  --md-deep-orange-A100: #ff9e80;
  --md-deep-orange-A200: #ff6e40;
  --md-deep-orange-A400: #ff3d00;
  --md-deep-orange-A700: #dd2c00;
  --md-brown-50: #efebe9;
  --md-brown-100: #d7ccc8;
  --md-brown-200: #bcaaa4;
  --md-brown-300: #a1887f;
  --md-brown-400: #8d6e63;
  --md-brown-500: #795548;
  --md-brown-600: #6d4c41;
  --md-brown-700: #5d4037;
  --md-brown-800: #4e342e;
  --md-brown-900: #3e2723;
  --md-grey-50: #fafafa;
  --md-grey-100: #f5f5f5;
  --md-grey-200: #eee;
  --md-grey-300: #e0e0e0;
  --md-grey-400: #bdbdbd;
  --md-grey-500: #9e9e9e;
  --md-grey-600: #757575;
  --md-grey-700: #616161;
  --md-grey-800: #424242;
  --md-grey-900: #212121;
  --md-blue-grey-50: #eceff1;
  --md-blue-grey-100: #cfd8dc;
  --md-blue-grey-200: #b0bec5;
  --md-blue-grey-300: #90a4ae;
  --md-blue-grey-400: #78909c;
  --md-blue-grey-500: #607d8b;
  --md-blue-grey-600: #546e7a;
  --md-blue-grey-700: #455a64;
  --md-blue-grey-800: #37474f;
  --md-blue-grey-900: #263238;
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2014-2017, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| RenderedText
|----------------------------------------------------------------------------*/

:root {
  /* This is the padding value to fill the gaps between lines containing spans with background color. */
  --jp-private-code-span-padding: calc(
    (var(--jp-code-line-height) - 1) * var(--jp-code-font-size) / 2
  );
}

.jp-RenderedText {
  text-align: left;
  padding-left: var(--jp-code-padding);
  line-height: var(--jp-code-line-height);
  font-family: var(--jp-code-font-family);
}

.jp-RenderedText pre,
.jp-RenderedJavaScript pre,
.jp-RenderedHTMLCommon pre {
  color: var(--jp-content-font-color1);
  font-size: var(--jp-code-font-size);
  border: none;
  margin: 0;
  padding: 0;
}

.jp-RenderedText pre a:link {
  text-decoration: none;
  color: var(--jp-content-link-color);
}

.jp-RenderedText pre a:hover {
  text-decoration: underline;
  color: var(--jp-content-link-color);
}

.jp-RenderedText pre a:visited {
  text-decoration: none;
  color: var(--jp-content-link-color);
}

/* console foregrounds and backgrounds */
.jp-RenderedText pre .ansi-black-fg {
  color: #3e424d;
}

.jp-RenderedText pre .ansi-red-fg {
  color: #e75c58;
}

.jp-RenderedText pre .ansi-green-fg {
  color: #00a250;
}

.jp-RenderedText pre .ansi-yellow-fg {
  color: #ddb62b;
}

.jp-RenderedText pre .ansi-blue-fg {
  color: #208ffb;
}

.jp-RenderedText pre .ansi-magenta-fg {
  color: #d160c4;
}

.jp-RenderedText pre .ansi-cyan-fg {
  color: #60c6c8;
}

.jp-RenderedText pre .ansi-white-fg {
  color: #c5c1b4;
}

.jp-RenderedText pre .ansi-black-bg {
  background-color: #3e424d;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-red-bg {
  background-color: #e75c58;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-green-bg {
  background-color: #00a250;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-yellow-bg {
  background-color: #ddb62b;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-blue-bg {
  background-color: #208ffb;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-magenta-bg {
  background-color: #d160c4;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-cyan-bg {
  background-color: #60c6c8;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-white-bg {
  background-color: #c5c1b4;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-black-intense-fg {
  color: #282c36;
}

.jp-RenderedText pre .ansi-red-intense-fg {
  color: #b22b31;
}

.jp-RenderedText pre .ansi-green-intense-fg {
  color: #007427;
}

.jp-RenderedText pre .ansi-yellow-intense-fg {
  color: #b27d12;
}

.jp-RenderedText pre .ansi-blue-intense-fg {
  color: #0065ca;
}

.jp-RenderedText pre .ansi-magenta-intense-fg {
  color: #a03196;
}

.jp-RenderedText pre .ansi-cyan-intense-fg {
  color: #258f8f;
}

.jp-RenderedText pre .ansi-white-intense-fg {
  color: #a1a6b2;
}

.jp-RenderedText pre .ansi-black-intense-bg {
  background-color: #282c36;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-red-intense-bg {
  background-color: #b22b31;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-green-intense-bg {
  background-color: #007427;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-yellow-intense-bg {
  background-color: #b27d12;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-blue-intense-bg {
  background-color: #0065ca;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-magenta-intense-bg {
  background-color: #a03196;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-cyan-intense-bg {
  background-color: #258f8f;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-white-intense-bg {
  background-color: #a1a6b2;
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-default-inverse-fg {
  color: var(--jp-ui-inverse-font-color0);
}

.jp-RenderedText pre .ansi-default-inverse-bg {
  background-color: var(--jp-inverse-layout-color0);
  padding: var(--jp-private-code-span-padding) 0;
}

.jp-RenderedText pre .ansi-bold {
  font-weight: bold;
}

.jp-RenderedText pre .ansi-underline {
  text-decoration: underline;
}

.jp-RenderedText[data-mime-type='application/vnd.jupyter.stderr'] {
  background: var(--jp-rendermime-error-background);
  padding-top: var(--jp-code-padding);
}

/*-----------------------------------------------------------------------------
| RenderedLatex
|----------------------------------------------------------------------------*/

.jp-RenderedLatex {
  color: var(--jp-content-font-color1);
  font-size: var(--jp-content-font-size1);
  line-height: var(--jp-content-line-height);
}

/* Left-justify outputs.*/
.jp-OutputArea-output.jp-RenderedLatex {
  padding: var(--jp-code-padding);
  text-align: left;
}

/*-----------------------------------------------------------------------------
| RenderedHTML
|----------------------------------------------------------------------------*/

.jp-RenderedHTMLCommon {
  color: var(--jp-content-font-color1);
  font-family: var(--jp-content-font-family);
  font-size: var(--jp-content-font-size1);
  line-height: var(--jp-content-line-height);

  /* Give a bit more R padding on Markdown text to keep line lengths reasonable */
  padding-right: 20px;
}

.jp-RenderedHTMLCommon em {
  font-style: italic;
}

.jp-RenderedHTMLCommon strong {
  font-weight: bold;
}

.jp-RenderedHTMLCommon u {
  text-decoration: underline;
}

.jp-RenderedHTMLCommon a:link {
  text-decoration: none;
  color: var(--jp-content-link-color);
}

.jp-RenderedHTMLCommon a:hover {
  text-decoration: underline;
  color: var(--jp-content-link-color);
}

.jp-RenderedHTMLCommon a:visited {
  text-decoration: none;
  color: var(--jp-content-link-color);
}

/* Headings */

.jp-RenderedHTMLCommon h1,
.jp-RenderedHTMLCommon h2,
.jp-RenderedHTMLCommon h3,
.jp-RenderedHTMLCommon h4,
.jp-RenderedHTMLCommon h5,
.jp-RenderedHTMLCommon h6 {
  line-height: var(--jp-content-heading-line-height);
  font-weight: var(--jp-content-heading-font-weight);
  font-style: normal;
  margin: var(--jp-content-heading-margin-top) 0
    var(--jp-content-heading-margin-bottom) 0;
}

.jp-RenderedHTMLCommon h1:first-child,
.jp-RenderedHTMLCommon h2:first-child,
.jp-RenderedHTMLCommon h3:first-child,
.jp-RenderedHTMLCommon h4:first-child,
.jp-RenderedHTMLCommon h5:first-child,
.jp-RenderedHTMLCommon h6:first-child {
  margin-top: calc(0.5 * var(--jp-content-heading-margin-top));
}

.jp-RenderedHTMLCommon h1:last-child,
.jp-RenderedHTMLCommon h2:last-child,
.jp-RenderedHTMLCommon h3:last-child,
.jp-RenderedHTMLCommon h4:last-child,
.jp-RenderedHTMLCommon h5:last-child,
.jp-RenderedHTMLCommon h6:last-child {
  margin-bottom: calc(0.5 * var(--jp-content-heading-margin-bottom));
}

.jp-RenderedHTMLCommon h1 {
  font-size: var(--jp-content-font-size5);
}

.jp-RenderedHTMLCommon h2 {
  font-size: var(--jp-content-font-size4);
}

.jp-RenderedHTMLCommon h3 {
  font-size: var(--jp-content-font-size3);
}

.jp-RenderedHTMLCommon h4 {
  font-size: var(--jp-content-font-size2);
}

.jp-RenderedHTMLCommon h5 {
  font-size: var(--jp-content-font-size1);
}

.jp-RenderedHTMLCommon h6 {
  font-size: var(--jp-content-font-size0);
}

/* Lists */

/* stylelint-disable selector-max-type, selector-max-compound-selectors */

.jp-RenderedHTMLCommon ul:not(.list-inline),
.jp-RenderedHTMLCommon ol:not(.list-inline) {
  padding-left: 2em;
}

.jp-RenderedHTMLCommon ul {
  list-style: disc;
}

.jp-RenderedHTMLCommon ul ul {
  list-style: square;
}

.jp-RenderedHTMLCommon ul ul ul {
  list-style: circle;
}

.jp-RenderedHTMLCommon ol {
  list-style: decimal;
}

.jp-RenderedHTMLCommon ol ol {
  list-style: upper-alpha;
}

.jp-RenderedHTMLCommon ol ol ol {
  list-style: lower-alpha;
}

.jp-RenderedHTMLCommon ol ol ol ol {
  list-style: lower-roman;
}

.jp-RenderedHTMLCommon ol ol ol ol ol {
  list-style: decimal;
}

.jp-RenderedHTMLCommon ol,
.jp-RenderedHTMLCommon ul {
  margin-bottom: 1em;
}

.jp-RenderedHTMLCommon ul ul,
.jp-RenderedHTMLCommon ul ol,
.jp-RenderedHTMLCommon ol ul,
.jp-RenderedHTMLCommon ol ol {
  margin-bottom: 0;
}

/* stylelint-enable selector-max-type, selector-max-compound-selectors */

.jp-RenderedHTMLCommon hr {
  color: var(--jp-border-color2);
  background-color: var(--jp-border-color1);
  margin-top: 1em;
  margin-bottom: 1em;
}

.jp-RenderedHTMLCommon > pre {
  margin: 1.5em 2em;
}

.jp-RenderedHTMLCommon pre,
.jp-RenderedHTMLCommon code {
  border: 0;
  background-color: var(--jp-layout-color0);
  color: var(--jp-content-font-color1);
  font-family: var(--jp-code-font-family);
  font-size: inherit;
  line-height: var(--jp-code-line-height);
  padding: 0;
  white-space: pre-wrap;
}

.jp-RenderedHTMLCommon :not(pre) > code {
  background-color: var(--jp-layout-color2);
  padding: 1px 5px;
}

/* Tables */

.jp-RenderedHTMLCommon table {
  border-collapse: collapse;
  border-spacing: 0;
  border: none;
  color: var(--jp-ui-font-color1);
  font-size: var(--jp-ui-font-size1);
  table-layout: fixed;
  margin-left: auto;
  margin-bottom: 1em;
  margin-right: auto;
}

.jp-RenderedHTMLCommon thead {
  border-bottom: var(--jp-border-width) solid var(--jp-border-color1);
  vertical-align: bottom;
}

.jp-RenderedHTMLCommon td,
.jp-RenderedHTMLCommon th,
.jp-RenderedHTMLCommon tr {
  vertical-align: middle;
  padding: 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}

.jp-RenderedMarkdown.jp-RenderedHTMLCommon td,
.jp-RenderedMarkdown.jp-RenderedHTMLCommon th {
  max-width: none;
}

:not(.jp-RenderedMarkdown).jp-RenderedHTMLCommon td,
:not(.jp-RenderedMarkdown).jp-RenderedHTMLCommon th,
:not(.jp-RenderedMarkdown).jp-RenderedHTMLCommon tr {
  text-align: right;
}

.jp-RenderedHTMLCommon th {
  font-weight: bold;
}

.jp-RenderedHTMLCommon tbody tr:nth-child(odd) {
  background: var(--jp-layout-color0);
}

.jp-RenderedHTMLCommon tbody tr:nth-child(even) {
  background: var(--jp-rendermime-table-row-background);
}

.jp-RenderedHTMLCommon tbody tr:hover {
  background: var(--jp-rendermime-table-row-hover-background);
}

.jp-RenderedHTMLCommon p {
  text-align: left;
  margin: 0;
  margin-bottom: 1em;
}

.jp-RenderedHTMLCommon img {
  -moz-force-broken-image-icon: 1;
}

/* Restrict to direct children as other images could be nested in other content. */
.jp-RenderedHTMLCommon > img {
  display: block;
  margin-left: 0;
  margin-right: 0;
  margin-bottom: 1em;
}

/* Change color behind transparent images if they need it... */
[data-jp-theme-light='false'] .jp-RenderedImage img.jp-needs-light-background {
  background-color: var(--jp-inverse-layout-color1);
}

[data-jp-theme-light='true'] .jp-RenderedImage img.jp-needs-dark-background {
  background-color: var(--jp-inverse-layout-color1);
}

.jp-RenderedHTMLCommon img,
.jp-RenderedImage img,
.jp-RenderedHTMLCommon svg,
.jp-RenderedSVG svg {
  max-width: 100%;
  height: auto;
}

.jp-RenderedHTMLCommon img.jp-mod-unconfined,
.jp-RenderedImage img.jp-mod-unconfined,
.jp-RenderedHTMLCommon svg.jp-mod-unconfined,
.jp-RenderedSVG svg.jp-mod-unconfined {
  max-width: none;
}

.jp-RenderedHTMLCommon .alert {
  padding: var(--jp-notebook-padding);
  border: var(--jp-border-width) solid transparent;
  border-radius: var(--jp-border-radius);
  margin-bottom: 1em;
}

.jp-RenderedHTMLCommon .alert-info {
  color: var(--jp-info-color0);
  background-color: var(--jp-info-color3);
  border-color: var(--jp-info-color2);
}

.jp-RenderedHTMLCommon .alert-info hr {
  border-color: var(--jp-info-color3);
}

.jp-RenderedHTMLCommon .alert-info > p:last-child,
.jp-RenderedHTMLCommon .alert-info > ul:last-child {
  margin-bottom: 0;
}

.jp-RenderedHTMLCommon .alert-warning {
  color: var(--jp-warn-color0);
  background-color: var(--jp-warn-color3);
  border-color: var(--jp-warn-color2);
}

.jp-RenderedHTMLCommon .alert-warning hr {
  border-color: var(--jp-warn-color3);
}

.jp-RenderedHTMLCommon .alert-warning > p:last-child,
.jp-RenderedHTMLCommon .alert-warning > ul:last-child {
  margin-bottom: 0;
}

.jp-RenderedHTMLCommon .alert-success {
  color: var(--jp-success-color0);
  background-color: var(--jp-success-color3);
  border-color: var(--jp-success-color2);
}

.jp-RenderedHTMLCommon .alert-success hr {
  border-color: var(--jp-success-color3);
}

.jp-RenderedHTMLCommon .alert-success > p:last-child,
.jp-RenderedHTMLCommon .alert-success > ul:last-child {
  margin-bottom: 0;
}

.jp-RenderedHTMLCommon .alert-danger {
  color: var(--jp-error-color0);
  background-color: var(--jp-error-color3);
  border-color: var(--jp-error-color2);
}

.jp-RenderedHTMLCommon .alert-danger hr {
  border-color: var(--jp-error-color3);
}

.jp-RenderedHTMLCommon .alert-danger > p:last-child,
.jp-RenderedHTMLCommon .alert-danger > ul:last-child {
  margin-bottom: 0;
}

.jp-RenderedHTMLCommon blockquote {
  margin: 1em 2em;
  padding: 0 1em;
  border-left: 5px solid var(--jp-border-color2);
}

a.jp-InternalAnchorLink {
  visibility: hidden;
  margin-left: 8px;
  color: var(--md-blue-800);
}

h1:hover .jp-InternalAnchorLink,
h2:hover .jp-InternalAnchorLink,
h3:hover .jp-InternalAnchorLink,
h4:hover .jp-InternalAnchorLink,
h5:hover .jp-InternalAnchorLink,
h6:hover .jp-InternalAnchorLink {
  visibility: visible;
}

.jp-RenderedHTMLCommon kbd {
  background-color: var(--jp-rendermime-table-row-background);
  border: 1px solid var(--jp-border-color0);
  border-bottom-color: var(--jp-border-color2);
  border-radius: 3px;
  box-shadow: inset 0 -1px 0 rgba(0, 0, 0, 0.25);
  display: inline-block;
  font-size: var(--jp-ui-font-size0);
  line-height: 1em;
  padding: 0.2em 0.5em;
}

/* Most direct children of .jp-RenderedHTMLCommon have a margin-bottom of 1.0.
 * At the bottom of cells this is a bit too much as there is also spacing
 * between cells. Going all the way to 0 gets too tight between markdown and
 * code cells.
 */
.jp-RenderedHTMLCommon > *:last-child {
  margin-bottom: 0.5em;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Copyright (c) 2014-2017, PhosphorJS Contributors
|
| Distributed under the terms of the BSD 3-Clause License.
|
| The full license is in the file LICENSE, distributed with this software.
|----------------------------------------------------------------------------*/

.lm-cursor-backdrop {
  position: fixed;
  width: 200px;
  height: 200px;
  margin-top: -100px;
  margin-left: -100px;
  will-change: transform;
  z-index: 100;
}

.lm-mod-drag-image {
  will-change: transform;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

.jp-lineFormSearch {
  padding: 4px 12px;
  background-color: var(--jp-layout-color2);
  box-shadow: var(--jp-toolbar-box-shadow);
  z-index: 2;
  font-size: var(--jp-ui-font-size1);
}

.jp-lineFormCaption {
  font-size: var(--jp-ui-font-size0);
  line-height: var(--jp-ui-font-size1);
  margin-top: 4px;
  color: var(--jp-ui-font-color0);
}

.jp-baseLineForm {
  border: none;
  border-radius: 0;
  position: absolute;
  background-size: 16px;
  background-repeat: no-repeat;
  background-position: center;
  outline: none;
}

.jp-lineFormButtonContainer {
  top: 4px;
  right: 8px;
  height: 24px;
  padding: 0 12px;
  width: 12px;
}

.jp-lineFormButtonIcon {
  top: 0;
  right: 0;
  background-color: var(--jp-brand-color1);
  height: 100%;
  width: 100%;
  box-sizing: border-box;
  padding: 4px 6px;
}

.jp-lineFormButton {
  top: 0;
  right: 0;
  background-color: transparent;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

.jp-lineFormWrapper {
  overflow: hidden;
  padding: 0 8px;
  border: 1px solid var(--jp-border-color0);
  background-color: var(--jp-input-active-background);
  height: 22px;
}

.jp-lineFormWrapperFocusWithin {
  border: var(--jp-border-width) solid var(--md-blue-500);
  box-shadow: inset 0 0 4px var(--md-blue-300);
}

.jp-lineFormInput {
  background: transparent;
  width: 200px;
  height: 100%;
  border: none;
  outline: none;
  color: var(--jp-ui-font-color0);
  line-height: 28px;
}

/*-----------------------------------------------------------------------------
| Copyright (c) 2014-2016, Jupyter Development Team.
|
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-JSONEditor {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.jp-JSONEditor-host {
  flex: 1 1 auto;
  border: var(--jp-border-width) solid var(--jp-input-border-color);
  border-radius: 0;
  background: var(--jp-layout-color0);
  min-height: 50px;
  padding: 1px;
}

.jp-JSONEditor.jp-mod-error .jp-JSONEditor-host {
  border-color: red;
  outline-color: red;
}

.jp-JSONEditor-header {
  display: flex;
  flex: 1 0 auto;
  padding: 0 0 0 12px;
}

.jp-JSONEditor-header label {
  flex: 0 0 auto;
}

.jp-JSONEditor-commitButton {
  height: 16px;
  width: 16px;
  background-size: 18px;
  background-repeat: no-repeat;
  background-position: center;
}

.jp-JSONEditor-host.jp-mod-focused {
  background-color: var(--jp-input-active-background);
  border: 1px solid var(--jp-input-active-border-color);
  box-shadow: var(--jp-input-box-shadow);
}

.jp-Editor.jp-mod-dropTarget {
  border: var(--jp-border-width) solid var(--jp-input-active-border-color);
  box-shadow: var(--jp-input-box-shadow);
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/
.jp-DocumentSearch-input {
  border: none;
  outline: none;
  color: var(--jp-ui-font-color0);
  font-size: var(--jp-ui-font-size1);
  background-color: var(--jp-layout-color0);
  font-family: var(--jp-ui-font-family);
  padding: 2px 1px;
  resize: none;
}

.jp-DocumentSearch-overlay {
  position: absolute;
  background-color: var(--jp-toolbar-background);
  border-bottom: var(--jp-border-width) solid var(--jp-toolbar-border-color);
  border-left: var(--jp-border-width) solid var(--jp-toolbar-border-color);
  top: 0;
  right: 0;
  z-index: 7;
  min-width: 405px;
  padding: 2px;
  font-size: var(--jp-ui-font-size1);

  --jp-private-document-search-button-height: 20px;
}

.jp-DocumentSearch-overlay button {
  background-color: var(--jp-toolbar-background);
  outline: 0;
}

.jp-DocumentSearch-overlay button:hover {
  background-color: var(--jp-layout-color2);
}

.jp-DocumentSearch-overlay button:active {
  background-color: var(--jp-layout-color3);
}

.jp-DocumentSearch-overlay-row {
  display: flex;
  align-items: center;
  margin-bottom: 2px;
}

.jp-DocumentSearch-button-content {
  display: inline-block;
  cursor: pointer;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

.jp-DocumentSearch-button-content svg {
  width: 100%;
  height: 100%;
}

.jp-DocumentSearch-input-wrapper {
  border: var(--jp-border-width) solid var(--jp-border-color0);
  display: flex;
  background-color: var(--jp-layout-color0);
  margin: 2px;
}

.jp-DocumentSearch-input-wrapper:focus-within {
  border-color: var(--jp-cell-editor-active-border-color);
}

.jp-DocumentSearch-toggle-wrapper,
.jp-DocumentSearch-button-wrapper {
  all: initial;
  overflow: hidden;
  display: inline-block;
  border: none;
  box-sizing: border-box;
}

.jp-DocumentSearch-toggle-wrapper {
  width: 14px;
  height: 14px;
}

.jp-DocumentSearch-button-wrapper {
  width: var(--jp-private-document-search-button-height);
  height: var(--jp-private-document-search-button-height);
}

.jp-DocumentSearch-toggle-wrapper:focus,
.jp-DocumentSearch-button-wrapper:focus {
  outline: var(--jp-border-width) solid
    var(--jp-cell-editor-active-border-color);
  outline-offset: -1px;
}

.jp-DocumentSearch-toggle-wrapper,
.jp-DocumentSearch-button-wrapper,
.jp-DocumentSearch-button-content:focus {
  outline: none;
}

.jp-DocumentSearch-toggle-placeholder {
  width: 5px;
}

.jp-DocumentSearch-input-button::before {
  display: block;
  padding-top: 100%;
}

.jp-DocumentSearch-input-button-off {
  opacity: var(--jp-search-toggle-off-opacity);
}

.jp-DocumentSearch-input-button-off:hover {
  opacity: var(--jp-search-toggle-hover-opacity);
}

.jp-DocumentSearch-input-button-on {
  opacity: var(--jp-search-toggle-on-opacity);
}

.jp-DocumentSearch-index-counter {
  padding-left: 10px;
  padding-right: 10px;
  user-select: none;
  min-width: 35px;
  display: inline-block;
}

.jp-DocumentSearch-up-down-wrapper {
  display: inline-block;
  padding-right: 2px;
  margin-left: auto;
  white-space: nowrap;
}

.jp-DocumentSearch-spacer {
  margin-left: auto;
}

.jp-DocumentSearch-up-down-wrapper button {
  outline: 0;
  border: none;
  width: var(--jp-private-document-search-button-height);
  height: var(--jp-private-document-search-button-height);
  vertical-align: middle;
  margin: 1px 5px 2px;
}

.jp-DocumentSearch-up-down-button:hover {
  background-color: var(--jp-layout-color2);
}

.jp-DocumentSearch-up-down-button:active {
  background-color: var(--jp-layout-color3);
}

.jp-DocumentSearch-filter-button {
  border-radius: var(--jp-border-radius);
}

.jp-DocumentSearch-filter-button:hover {
  background-color: var(--jp-layout-color2);
}

.jp-DocumentSearch-filter-button-enabled {
  background-color: var(--jp-layout-color2);
}

.jp-DocumentSearch-filter-button-enabled:hover {
  background-color: var(--jp-layout-color3);
}

.jp-DocumentSearch-search-options {
  padding: 0 8px;
  margin-left: 3px;
  width: 100%;
  display: grid;
  justify-content: start;
  grid-template-columns: 1fr 1fr;
  align-items: center;
  justify-items: stretch;
}

.jp-DocumentSearch-search-filter-disabled {
  color: var(--jp-ui-font-color2);
}

.jp-DocumentSearch-search-filter {
  display: flex;
  align-items: center;
  user-select: none;
}

.jp-DocumentSearch-regex-error {
  color: var(--jp-error-color0);
}

.jp-DocumentSearch-replace-button-wrapper {
  overflow: hidden;
  display: inline-block;
  box-sizing: border-box;
  border: var(--jp-border-width) solid var(--jp-border-color0);
  margin: auto 2px;
  padding: 1px 4px;
  height: calc(var(--jp-private-document-search-button-height) + 2px);
}

.jp-DocumentSearch-replace-button-wrapper:focus {
  border: var(--jp-border-width) solid var(--jp-cell-editor-active-border-color);
}

.jp-DocumentSearch-replace-button {
  display: inline-block;
  text-align: center;
  cursor: pointer;
  box-sizing: border-box;
  color: var(--jp-ui-font-color1);

  /* height - 2 * (padding of wrapper) */
  line-height: calc(var(--jp-private-document-search-button-height) - 2px);
  width: 100%;
  height: 100%;
}

.jp-DocumentSearch-replace-button:focus {
  outline: none;
}

.jp-DocumentSearch-replace-wrapper-class {
  margin-left: 14px;
  display: flex;
}

.jp-DocumentSearch-replace-toggle {
  border: none;
  background-color: var(--jp-toolbar-background);
  border-radius: var(--jp-border-radius);
}

.jp-DocumentSearch-replace-toggle:hover {
  background-color: var(--jp-layout-color2);
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.cm-editor {
  line-height: var(--jp-code-line-height);
  font-size: var(--jp-code-font-size);
  font-family: var(--jp-code-font-family);
  border: 0;
  border-radius: 0;
  height: auto;

  /* Changed to auto to autogrow */
}

.cm-editor pre {
  padding: 0 var(--jp-code-padding);
}

.jp-CodeMirrorEditor[data-type='inline'] .cm-dialog {
  background-color: var(--jp-layout-color0);
  color: var(--jp-content-font-color1);
}

.jp-CodeMirrorEditor {
  cursor: text;
}

/* When zoomed out 67% and 33% on a screen of 1440 width x 900 height */
@media screen and (min-width: 2138px) and (max-width: 4319px) {
  .jp-CodeMirrorEditor[data-type='inline'] .cm-cursor {
    border-left: var(--jp-code-cursor-width1) solid
      var(--jp-editor-cursor-color);
  }
}

/* When zoomed out less than 33% */
@media screen and (min-width: 4320px) {
  .jp-CodeMirrorEditor[data-type='inline'] .cm-cursor {
    border-left: var(--jp-code-cursor-width2) solid
      var(--jp-editor-cursor-color);
  }
}

.cm-editor.jp-mod-readOnly .cm-cursor {
  display: none;
}

.jp-CollaboratorCursor {
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: none;
  border-bottom: 3px solid;
  background-clip: content-box;
  margin-left: -5px;
  margin-right: -5px;
}

.cm-searching,
.cm-searching span {
  /* `.cm-searching span`: we need to override syntax highlighting */
  background-color: var(--jp-search-unselected-match-background-color);
  color: var(--jp-search-unselected-match-color);
}

.cm-searching::selection,
.cm-searching span::selection {
  background-color: var(--jp-search-unselected-match-background-color);
  color: var(--jp-search-unselected-match-color);
}

.jp-current-match > .cm-searching,
.jp-current-match > .cm-searching span,
.cm-searching > .jp-current-match,
.cm-searching > .jp-current-match span {
  background-color: var(--jp-search-selected-match-background-color);
  color: var(--jp-search-selected-match-color);
}

.jp-current-match > .cm-searching::selection,
.cm-searching > .jp-current-match::selection,
.jp-current-match > .cm-searching span::selection {
  background-color: var(--jp-search-selected-match-background-color);
  color: var(--jp-search-selected-match-color);
}

.cm-trailingspace {
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAFCAYAAAB4ka1VAAAAsElEQVQIHQGlAFr/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA7+r3zKmT0/+pk9P/7+r3zAAAAAAAAAAABAAAAAAAAAAA6OPzM+/q9wAAAAAA6OPzMwAAAAAAAAAAAgAAAAAAAAAAGR8NiRQaCgAZIA0AGR8NiQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQyoYJ/SY80UAAAAASUVORK5CYII=);
  background-position: center left;
  background-repeat: repeat-x;
}

.jp-CollaboratorCursor-hover {
  position: absolute;
  z-index: 1;
  transform: translateX(-50%);
  color: white;
  border-radius: 3px;
  padding-left: 4px;
  padding-right: 4px;
  padding-top: 1px;
  padding-bottom: 1px;
  text-align: center;
  font-size: var(--jp-ui-font-size1);
  white-space: nowrap;
}

.jp-CodeMirror-ruler {
  border-left: 1px dashed var(--jp-border-color2);
}

/* Styles for shared cursors (remote cursor locations and selected ranges) */
.jp-CodeMirrorEditor .cm-ySelectionCaret {
  position: relative;
  border-left: 1px solid black;
  margin-left: -1px;
  margin-right: -1px;
  box-sizing: border-box;
}

.jp-CodeMirrorEditor .cm-ySelectionCaret > .cm-ySelectionInfo {
  white-space: nowrap;
  position: absolute;
  top: -1.15em;
  padding-bottom: 0.05em;
  left: -1px;
  font-size: 0.95em;
  font-family: var(--jp-ui-font-family);
  font-weight: bold;
  line-height: normal;
  user-select: none;
  color: white;
  padding-left: 2px;
  padding-right: 2px;
  z-index: 101;
  transition: opacity 0.3s ease-in-out;
}

.jp-CodeMirrorEditor .cm-ySelectionInfo {
  transition-delay: 0.7s;
  opacity: 0;
}

.jp-CodeMirrorEditor .cm-ySelectionCaret:hover > .cm-ySelectionInfo {
  opacity: 1;
  transition-delay: 0s;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-MimeDocument {
  outline: none;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Variables
|----------------------------------------------------------------------------*/

:root {
  --jp-private-filebrowser-button-height: 28px;
  --jp-private-filebrowser-button-width: 48px;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-FileBrowser .jp-SidePanel-content {
  display: flex;
  flex-direction: column;
}

.jp-FileBrowser-toolbar.jp-Toolbar {
  flex-wrap: wrap;
  row-gap: 12px;
  border-bottom: none;
  height: auto;
  margin: 8px 12px 0;
  box-shadow: none;
  padding: 0;
  justify-content: flex-start;
}

.jp-FileBrowser-Panel {
  flex: 1 1 auto;
  display: flex;
  flex-direction: column;
}

.jp-BreadCrumbs {
  flex: 0 0 auto;
  margin: 8px 12px;
}

.jp-BreadCrumbs-item {
  margin: 0 2px;
  padding: 0 2px;
  border-radius: var(--jp-border-radius);
  cursor: pointer;
}

.jp-BreadCrumbs-item:hover {
  background-color: var(--jp-layout-color2);
}

.jp-BreadCrumbs-item:first-child {
  margin-left: 0;
}

.jp-BreadCrumbs-item.jp-mod-dropTarget {
  background-color: var(--jp-brand-color2);
  opacity: 0.7;
}

/*-----------------------------------------------------------------------------
| Buttons
|----------------------------------------------------------------------------*/

.jp-FileBrowser-toolbar > .jp-Toolbar-item {
  flex: 0 0 auto;
  padding-left: 0;
  padding-right: 2px;
  align-items: center;
  height: unset;
}

.jp-FileBrowser-toolbar > .jp-Toolbar-item .jp-ToolbarButtonComponent {
  width: 40px;
}

/*-----------------------------------------------------------------------------
| Other styles
|----------------------------------------------------------------------------*/

.jp-FileDialog.jp-mod-conflict input {
  color: var(--jp-error-color1);
}

.jp-FileDialog .jp-new-name-title {
  margin-top: 12px;
}

.jp-LastModified-hidden {
  display: none;
}

.jp-FileSize-hidden {
  display: none;
}

.jp-FileBrowser .lm-AccordionPanel > h3:first-child {
  display: none;
}

/*-----------------------------------------------------------------------------
| DirListing
|----------------------------------------------------------------------------*/

.jp-DirListing {
  flex: 1 1 auto;
  display: flex;
  flex-direction: column;
  outline: 0;
}

.jp-DirListing-header {
  flex: 0 0 auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  overflow: hidden;
  border-top: var(--jp-border-width) solid var(--jp-border-color2);
  border-bottom: var(--jp-border-width) solid var(--jp-border-color1);
  box-shadow: var(--jp-toolbar-box-shadow);
  z-index: 2;
}

.jp-DirListing-headerItem {
  padding: 4px 12px 2px;
  font-weight: 500;
}

.jp-DirListing-headerItem:hover {
  background: var(--jp-layout-color2);
}

.jp-DirListing-headerItem.jp-id-name {
  flex: 1 0 84px;
}

.jp-DirListing-headerItem.jp-id-modified {
  flex: 0 0 112px;
  border-left: var(--jp-border-width) solid var(--jp-border-color2);
  text-align: right;
}

.jp-DirListing-headerItem.jp-id-filesize {
  flex: 0 0 75px;
  border-left: var(--jp-border-width) solid var(--jp-border-color2);
  text-align: right;
}

.jp-id-narrow {
  display: none;
  flex: 0 0 5px;
  padding: 4px;
  border-left: var(--jp-border-width) solid var(--jp-border-color2);
  text-align: right;
  color: var(--jp-border-color2);
}

.jp-DirListing-narrow .jp-id-narrow {
  display: block;
}

.jp-DirListing-narrow .jp-id-modified,
.jp-DirListing-narrow .jp-DirListing-itemModified {
  display: none;
}

.jp-DirListing-headerItem.jp-mod-selected {
  font-weight: 600;
}

/* increase specificity to override bundled default */
.jp-DirListing-content {
  flex: 1 1 auto;
  margin: 0;
  padding: 0;
  list-style-type: none;
  overflow: auto;
  background-color: var(--jp-layout-color1);
}

.jp-DirListing-content mark {
  color: var(--jp-ui-font-color0);
  background-color: transparent;
  font-weight: bold;
}

.jp-DirListing-content .jp-DirListing-item.jp-mod-selected mark {
  color: var(--jp-ui-inverse-font-color0);
}

/* Style the directory listing content when a user drops a file to upload */
.jp-DirListing.jp-mod-native-drop .jp-DirListing-content {
  outline: 5px dashed rgba(128, 128, 128, 0.5);
  outline-offset: -10px;
  cursor: copy;
}

.jp-DirListing-item {
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 4px 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.jp-DirListing-checkboxWrapper {
  /* Increases hit area of checkbox. */
  padding: 4px;
}

.jp-DirListing-header
  .jp-DirListing-checkboxWrapper
  + .jp-DirListing-headerItem {
  padding-left: 4px;
}

.jp-DirListing-content .jp-DirListing-checkboxWrapper {
  position: relative;
  left: -4px;
  margin: -4px 0 -4px -8px;
}

.jp-DirListing-checkboxWrapper.jp-mod-visible {
  visibility: visible;
}

/* For devices that support hovering, hide checkboxes until hovered, selected...
*/
@media (hover: hover) {
  .jp-DirListing-checkboxWrapper {
    visibility: hidden;
  }

  .jp-DirListing-item:hover .jp-DirListing-checkboxWrapper,
  .jp-DirListing-item.jp-mod-selected .jp-DirListing-checkboxWrapper {
    visibility: visible;
  }
}

.jp-DirListing-item[data-is-dot] {
  opacity: 75%;
}

.jp-DirListing-item.jp-mod-selected {
  color: var(--jp-ui-inverse-font-color1);
  background: var(--jp-brand-color1);
}

.jp-DirListing-item.jp-mod-dropTarget {
  background: var(--jp-brand-color3);
}

.jp-DirListing-item:hover:not(.jp-mod-selected) {
  background: var(--jp-layout-color2);
}

.jp-DirListing-itemIcon {
  flex: 0 0 20px;
  margin-right: 4px;
}

.jp-DirListing-itemText {
  flex: 1 0 64px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  user-select: none;
}

.jp-DirListing-itemText:focus {
  outline-width: 2px;
  outline-color: var(--jp-inverse-layout-color1);
  outline-style: solid;
  outline-offset: 1px;
}

.jp-DirListing-item.jp-mod-selected .jp-DirListing-itemText:focus {
  outline-color: var(--jp-layout-color1);
}

.jp-DirListing-itemModified {
  flex: 0 0 125px;
  text-align: right;
}

.jp-DirListing-itemFileSize {
  flex: 0 0 90px;
  text-align: right;
}

.jp-DirListing-editor {
  flex: 1 0 64px;
  outline: none;
  border: none;
  color: var(--jp-ui-font-color1);
  background-color: var(--jp-layout-color1);
}

.jp-DirListing-item.jp-mod-running .jp-DirListing-itemIcon::before {
  color: var(--jp-success-color1);
  content: '\25CF';
  font-size: 8px;
  position: absolute;
  left: -8px;
}

.jp-DirListing-item.jp-mod-running.jp-mod-selected
  .jp-DirListing-itemIcon::before {
  color: var(--jp-ui-inverse-font-color1);
}

.jp-DirListing-item.lm-mod-drag-image,
.jp-DirListing-item.jp-mod-selected.lm-mod-drag-image {
  font-size: var(--jp-ui-font-size1);
  padding-left: 4px;
  margin-left: 4px;
  width: 160px;
  background-color: var(--jp-ui-inverse-font-color2);
  box-shadow: var(--jp-elevation-z2);
  border-radius: 0;
  color: var(--jp-ui-font-color1);
  transform: translateX(-40%) translateY(-58%);
}

.jp-Document {
  min-width: 120px;
  min-height: 120px;
  outline: none;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Main OutputArea
| OutputArea has a list of Outputs
|----------------------------------------------------------------------------*/

.jp-OutputArea {
  overflow-y: auto;
}

.jp-OutputArea-child {
  display: table;
  table-layout: fixed;
  width: 100%;
  overflow: hidden;
}

.jp-OutputPrompt {
  width: var(--jp-cell-prompt-width);
  color: var(--jp-cell-outprompt-font-color);
  font-family: var(--jp-cell-prompt-font-family);
  padding: var(--jp-code-padding);
  letter-spacing: var(--jp-cell-prompt-letter-spacing);
  line-height: var(--jp-code-line-height);
  font-size: var(--jp-code-font-size);
  border: var(--jp-border-width) solid transparent;
  opacity: var(--jp-cell-prompt-opacity);

  /* Right align prompt text, don't wrap to handle large prompt numbers */
  text-align: right;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  /* Disable text selection */
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.jp-OutputArea-prompt {
  display: table-cell;
  vertical-align: top;
}

.jp-OutputArea-output {
  display: table-cell;
  width: 100%;
  height: auto;
  overflow: auto;
  user-select: text;
  -moz-user-select: text;
  -webkit-user-select: text;
  -ms-user-select: text;
}

.jp-OutputArea .jp-RenderedText {
  padding-left: 1ch;
}

/**
 * Prompt overlay.
 */

.jp-OutputArea-promptOverlay {
  position: absolute;
  top: 0;
  width: var(--jp-cell-prompt-width);
  height: 100%;
  opacity: 0.5;
}

.jp-OutputArea-promptOverlay:hover {
  background: var(--jp-layout-color2);
  box-shadow: inset 0 0 1px var(--jp-inverse-layout-color0);
  cursor: zoom-out;
}

.jp-mod-outputsScrolled .jp-OutputArea-promptOverlay:hover {
  cursor: zoom-in;
}

/**
 * Isolated output.
 */
.jp-OutputArea-output.jp-mod-isolated {
  width: 100%;
  display: block;
}

/*
When drag events occur, `lm-mod-override-cursor` is added to the body.
Because iframes steal all cursor events, the following two rules are necessary
to suppress pointer events while resize drags are occurring. There may be a
better solution to this problem.
*/
body.lm-mod-override-cursor .jp-OutputArea-output.jp-mod-isolated {
  position: relative;
}

body.lm-mod-override-cursor .jp-OutputArea-output.jp-mod-isolated::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: transparent;
}

/* pre */

.jp-OutputArea-output pre {
  border: none;
  margin: 0;
  padding: 0;
  overflow-x: auto;
  overflow-y: auto;
  word-break: break-all;
  word-wrap: break-word;
  white-space: pre-wrap;
}

/* tables */

.jp-OutputArea-output.jp-RenderedHTMLCommon table {
  margin-left: 0;
  margin-right: 0;
}

/* description lists */

.jp-OutputArea-output dl,
.jp-OutputArea-output dt,
.jp-OutputArea-output dd {
  display: block;
}

.jp-OutputArea-output dl {
  width: 100%;
  overflow: hidden;
  padding: 0;
  margin: 0;
}

.jp-OutputArea-output dt {
  font-weight: bold;
  float: left;
  width: 20%;
  padding: 0;
  margin: 0;
}

.jp-OutputArea-output dd {
  float: left;
  width: 80%;
  padding: 0;
  margin: 0;
}

.jp-TrimmedOutputs pre {
  background: var(--jp-layout-color3);
  font-size: calc(var(--jp-code-font-size) * 1.4);
  text-align: center;
  text-transform: uppercase;
}

/* Hide the gutter in case of
 *  - nested output areas (e.g. in the case of output widgets)
 *  - mirrored output areas
 */
.jp-OutputArea .jp-OutputArea .jp-OutputArea-prompt {
  display: none;
}

/* Hide empty lines in the output area, for instance due to cleared widgets */
.jp-OutputArea-prompt:empty {
  padding: 0;
  border: 0;
}

/*-----------------------------------------------------------------------------
| executeResult is added to any Output-result for the display of the object
| returned by a cell
|----------------------------------------------------------------------------*/

.jp-OutputArea-output.jp-OutputArea-executeResult {
  margin-left: 0;
  width: 100%;
}

/* Text output with the Out[] prompt needs a top padding to match the
 * alignment of the Out[] prompt itself.
 */
.jp-OutputArea-executeResult .jp-RenderedText.jp-OutputArea-output {
  padding-top: var(--jp-code-padding);
  border-top: var(--jp-border-width) solid transparent;
}

/*-----------------------------------------------------------------------------
| The Stdin output
|----------------------------------------------------------------------------*/

.jp-Stdin-prompt {
  color: var(--jp-content-font-color0);
  padding-right: var(--jp-code-padding);
  vertical-align: baseline;
  flex: 0 0 auto;
}

.jp-Stdin-input {
  font-family: var(--jp-code-font-family);
  font-size: inherit;
  color: inherit;
  background-color: inherit;
  width: 42%;
  min-width: 200px;

  /* make sure input baseline aligns with prompt */
  vertical-align: baseline;

  /* padding + margin = 0.5em between prompt and cursor */
  padding: 0 0.25em;
  margin: 0 0.25em;
  flex: 0 0 70%;
}

.jp-Stdin-input::placeholder {
  opacity: 0;
}

.jp-Stdin-input:focus {
  box-shadow: none;
}

.jp-Stdin-input:focus::placeholder {
  opacity: 1;
}

/*-----------------------------------------------------------------------------
| Output Area View
|----------------------------------------------------------------------------*/

.jp-LinkedOutputView .jp-OutputArea {
  height: 100%;
  display: block;
}

.jp-LinkedOutputView .jp-OutputArea-output:only-child {
  height: 100%;
}

/*-----------------------------------------------------------------------------
| Printing
|----------------------------------------------------------------------------*/

@media print {
  .jp-OutputArea-child {
    break-inside: avoid-page;
  }
}

/*-----------------------------------------------------------------------------
| Mobile
|----------------------------------------------------------------------------*/
@media only screen and (max-width: 760px) {
  .jp-OutputPrompt {
    display: table-row;
    text-align: left;
  }

  .jp-OutputArea-child .jp-OutputArea-output {
    display: table-row;
    margin-left: var(--jp-notebook-padding);
  }
}

/* Trimmed outputs warning */
.jp-TrimmedOutputs > a {
  margin: 10px;
  text-decoration: none;
  cursor: pointer;
}

.jp-TrimmedOutputs > a:hover {
  text-decoration: none;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Table of Contents
|----------------------------------------------------------------------------*/

:root {
  --jp-private-toc-active-width: 4px;
}

.jp-TableOfContents {
  display: flex;
  flex-direction: column;
  background: var(--jp-layout-color1);
  color: var(--jp-ui-font-color1);
  font-size: var(--jp-ui-font-size1);
  height: 100%;
}

.jp-TableOfContents-placeholder {
  text-align: center;
}

.jp-TableOfContents-placeholderContent {
  color: var(--jp-content-font-color2);
  padding: 8px;
}

.jp-TableOfContents-placeholderContent > h3 {
  margin-bottom: var(--jp-content-heading-margin-bottom);
}

.jp-TableOfContents .jp-SidePanel-content {
  overflow-y: auto;
}

.jp-TableOfContents-tree {
  margin: 4px;
}

.jp-TableOfContents ol {
  list-style-type: none;
}

/* stylelint-disable-next-line selector-max-type */
.jp-TableOfContents li > ol {
  /* Align left border with triangle icon center */
  padding-left: 11px;
}

.jp-TableOfContents-content {
  /* left margin for the active heading indicator */
  margin: 0 0 0 var(--jp-private-toc-active-width);
  padding: 0;
  background-color: var(--jp-layout-color1);
}

.jp-tocItem {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.jp-tocItem-heading {
  display: flex;
  cursor: pointer;
}

.jp-tocItem-heading:hover {
  background-color: var(--jp-layout-color2);
}

.jp-tocItem-content {
  display: block;
  padding: 4px 0;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow-x: hidden;
}

.jp-tocItem-collapser {
  height: 20px;
  margin: 2px 2px 0;
  padding: 0;
  background: none;
  border: none;
  cursor: pointer;
}

.jp-tocItem-collapser:hover {
  background-color: var(--jp-layout-color3);
}

/* Active heading indicator */

.jp-tocItem-heading::before {
  content: ' ';
  background: transparent;
  width: var(--jp-private-toc-active-width);
  height: 24px;
  position: absolute;
  left: 0;
  border-radius: var(--jp-border-radius);
}

.jp-tocItem-heading.jp-tocItem-active::before {
  background-color: var(--jp-brand-color1);
}

.jp-tocItem-heading:hover.jp-tocItem-active::before {
  background: var(--jp-brand-color0);
  opacity: 1;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

.jp-Collapser {
  flex: 0 0 var(--jp-cell-collapser-width);
  padding: 0;
  margin: 0;
  border: none;
  outline: none;
  background: transparent;
  border-radius: var(--jp-border-radius);
  opacity: 1;
}

.jp-Collapser-child {
  display: block;
  width: 100%;
  box-sizing: border-box;

  /* height: 100% doesn't work because the height of its parent is computed from content */
  position: absolute;
  top: 0;
  bottom: 0;
}

/*-----------------------------------------------------------------------------
| Printing
|----------------------------------------------------------------------------*/

/*
Hiding collapsers in print mode.

Note: input and output wrappers have "display: block" propery in print mode.
*/

@media print {
  .jp-Collapser {
    display: none;
  }
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Header/Footer
|----------------------------------------------------------------------------*/

/* Hidden by zero height by default */
.jp-CellHeader,
.jp-CellFooter {
  height: 0;
  width: 100%;
  padding: 0;
  margin: 0;
  border: none;
  outline: none;
  background: transparent;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Input
|----------------------------------------------------------------------------*/

/* All input areas */
.jp-InputArea {
  display: table;
  table-layout: fixed;
  width: 100%;
  overflow: hidden;
}

.jp-InputArea-editor {
  display: table-cell;
  overflow: hidden;
  vertical-align: top;

  /* This is the non-active, default styling */
  border: var(--jp-border-width) solid var(--jp-cell-editor-border-color);
  border-radius: 0;
  background: var(--jp-cell-editor-background);
}

.jp-InputPrompt {
  display: table-cell;
  vertical-align: top;
  width: var(--jp-cell-prompt-width);
  color: var(--jp-cell-inprompt-font-color);
  font-family: var(--jp-cell-prompt-font-family);
  padding: var(--jp-code-padding);
  letter-spacing: var(--jp-cell-prompt-letter-spacing);
  opacity: var(--jp-cell-prompt-opacity);
  line-height: var(--jp-code-line-height);
  font-size: var(--jp-code-font-size);
  border: var(--jp-border-width) solid transparent;

  /* Right align prompt text, don't wrap to handle large prompt numbers */
  text-align: right;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  /* Disable text selection */
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

/*-----------------------------------------------------------------------------
| Mobile
|----------------------------------------------------------------------------*/
@media only screen and (max-width: 760px) {
  .jp-InputArea-editor {
    display: table-row;
    margin-left: var(--jp-notebook-padding);
  }

  .jp-InputPrompt {
    display: table-row;
    text-align: left;
  }
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Placeholder
|----------------------------------------------------------------------------*/

.jp-Placeholder {
  display: table;
  table-layout: fixed;
  width: 100%;
}

.jp-Placeholder-prompt {
  display: table-cell;
  box-sizing: border-box;
}

.jp-Placeholder-content {
  display: table-cell;
  padding: 4px 6px;
  border: 1px solid transparent;
  border-radius: 0;
  background: none;
  box-sizing: border-box;
  cursor: pointer;
}

.jp-Placeholder-contentContainer {
  display: flex;
}

.jp-Placeholder-content:hover,
.jp-InputPlaceholder > .jp-Placeholder-content:hover {
  border-color: var(--jp-layout-color3);
}

.jp-Placeholder-content .jp-MoreHorizIcon {
  width: 32px;
  height: 16px;
  border: 1px solid transparent;
  border-radius: var(--jp-border-radius);
}

.jp-Placeholder-content .jp-MoreHorizIcon:hover {
  border: 1px solid var(--jp-border-color1);
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.25);
  background-color: var(--jp-layout-color0);
}

.jp-PlaceholderText {
  white-space: nowrap;
  overflow-x: hidden;
  color: var(--jp-inverse-layout-color3);
  font-family: var(--jp-code-font-family);
}

.jp-InputPlaceholder > .jp-Placeholder-content {
  border-color: var(--jp-cell-editor-border-color);
  background: var(--jp-cell-editor-background);
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Private CSS variables
|----------------------------------------------------------------------------*/

:root {
  --jp-private-cell-scrolling-output-offset: 5px;
}

/*-----------------------------------------------------------------------------
| Cell
|----------------------------------------------------------------------------*/

.jp-Cell {
  padding: var(--jp-cell-padding);
  margin: 0;
  border: none;
  outline: none;
  background: transparent;
}

/*-----------------------------------------------------------------------------
| Common input/output
|----------------------------------------------------------------------------*/

.jp-Cell-inputWrapper,
.jp-Cell-outputWrapper {
  display: flex;
  flex-direction: row;
  padding: 0;
  margin: 0;

  /* Added to reveal the box-shadow on the input and output collapsers. */
  overflow: visible;
}

/* Only input/output areas inside cells */
.jp-Cell-inputArea,
.jp-Cell-outputArea {
  flex: 1 1 auto;
}

/*-----------------------------------------------------------------------------
| Collapser
|----------------------------------------------------------------------------*/

/* Make the output collapser disappear when there is not output, but do so
 * in a manner that leaves it in the layout and preserves its width.
 */
.jp-Cell.jp-mod-noOutputs .jp-Cell-outputCollapser {
  border: none !important;
  background: transparent !important;
}

.jp-Cell:not(.jp-mod-noOutputs) .jp-Cell-outputCollapser {
  min-height: var(--jp-cell-collapser-min-height);
}

/*-----------------------------------------------------------------------------
| Output
|----------------------------------------------------------------------------*/

/* Put a space between input and output when there IS output */
.jp-Cell:not(.jp-mod-noOutputs) .jp-Cell-outputWrapper {
  margin-top: 5px;
}

.jp-CodeCell.jp-mod-outputsScrolled .jp-Cell-outputArea {
  overflow-y: auto;
  max-height: 24em;
  margin-left: var(--jp-private-cell-scrolling-output-offset);
  resize: vertical;
}

.jp-CodeCell.jp-mod-outputsScrolled .jp-Cell-outputArea[style*='height'] {
  max-height: unset;
}

.jp-CodeCell.jp-mod-outputsScrolled .jp-Cell-outputArea::after {
  content: ' ';
  box-shadow: inset 0 0 6px 2px rgb(0 0 0 / 30%);
  width: 100%;
  height: 100%;
  position: sticky;
  bottom: 0;
  top: 0;
  margin-top: -50%;
  float: left;
  display: block;
  pointer-events: none;
}

.jp-CodeCell.jp-mod-outputsScrolled .jp-OutputArea-child {
  padding-top: 6px;
}

.jp-CodeCell.jp-mod-outputsScrolled .jp-OutputArea-prompt {
  width: calc(
    var(--jp-cell-prompt-width) - var(--jp-private-cell-scrolling-output-offset)
  );
}

.jp-CodeCell.jp-mod-outputsScrolled .jp-OutputArea-promptOverlay {
  left: calc(-1 * var(--jp-private-cell-scrolling-output-offset));
}

/*-----------------------------------------------------------------------------
| CodeCell
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| MarkdownCell
|----------------------------------------------------------------------------*/

.jp-MarkdownOutput {
  display: table-cell;
  width: 100%;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: var(--jp-code-padding);
}

.jp-MarkdownOutput.jp-RenderedHTMLCommon {
  overflow: auto;
}

/* collapseHeadingButton (show always if hiddenCellsButton is _not_ shown) */
.jp-collapseHeadingButton {
  display: flex;
  min-height: var(--jp-cell-collapser-min-height);
  font-size: var(--jp-code-font-size);
  position: absolute;
  background-color: transparent;
  background-size: 25px;
  background-repeat: no-repeat;
  background-position-x: center;
  background-position-y: top;
  background-image: var(--jp-icon-caret-down);
  right: 0;
  top: 0;
  bottom: 0;
}

.jp-collapseHeadingButton.jp-mod-collapsed {
  background-image: var(--jp-icon-caret-right);
}

/*
 set the container font size to match that of content
 so that the nested collapse buttons have the right size
*/
.jp-MarkdownCell .jp-InputPrompt {
  font-size: var(--jp-content-font-size1);
}

/*
  Align collapseHeadingButton with cell top header
  The font sizes are identical to the ones in packages/rendermime/style/base.css
*/
.jp-mod-rendered .jp-collapseHeadingButton[data-heading-level='1'] {
  font-size: var(--jp-content-font-size5);
  background-position-y: calc(0.3 * var(--jp-content-font-size5));
}

.jp-mod-rendered .jp-collapseHeadingButton[data-heading-level='2'] {
  font-size: var(--jp-content-font-size4);
  background-position-y: calc(0.3 * var(--jp-content-font-size4));
}

.jp-mod-rendered .jp-collapseHeadingButton[data-heading-level='3'] {
  font-size: var(--jp-content-font-size3);
  background-position-y: calc(0.3 * var(--jp-content-font-size3));
}

.jp-mod-rendered .jp-collapseHeadingButton[data-heading-level='4'] {
  font-size: var(--jp-content-font-size2);
  background-position-y: calc(0.3 * var(--jp-content-font-size2));
}

.jp-mod-rendered .jp-collapseHeadingButton[data-heading-level='5'] {
  font-size: var(--jp-content-font-size1);
  background-position-y: top;
}

.jp-mod-rendered .jp-collapseHeadingButton[data-heading-level='6'] {
  font-size: var(--jp-content-font-size0);
  background-position-y: top;
}

/* collapseHeadingButton (show only on (hover,active) if hiddenCellsButton is shown) */
.jp-Notebook.jp-mod-showHiddenCellsButton .jp-collapseHeadingButton {
  display: none;
}

.jp-Notebook.jp-mod-showHiddenCellsButton
  :is(.jp-MarkdownCell:hover, .jp-mod-active)
  .jp-collapseHeadingButton {
  display: flex;
}

/* showHiddenCellsButton (only show if jp-mod-showHiddenCellsButton is set, which
is a consequence of the showHiddenCellsButton option in Notebook Settings)*/
.jp-Notebook.jp-mod-showHiddenCellsButton .jp-showHiddenCellsButton {
  margin-left: calc(var(--jp-cell-prompt-width) + 2 * var(--jp-code-padding));
  margin-top: var(--jp-code-padding);
  border: 1px solid var(--jp-border-color2);
  background-color: var(--jp-border-color3) !important;
  color: var(--jp-content-font-color0) !important;
  display: flex;
}

.jp-Notebook.jp-mod-showHiddenCellsButton .jp-showHiddenCellsButton:hover {
  background-color: var(--jp-border-color2) !important;
}

.jp-showHiddenCellsButton {
  display: none;
}

/*-----------------------------------------------------------------------------
| Printing
|----------------------------------------------------------------------------*/

/*
Using block instead of flex to allow the use of the break-inside CSS property for
cell outputs.
*/

@media print {
  .jp-Cell-inputWrapper,
  .jp-Cell-outputWrapper {
    display: block;
  }
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Variables
|----------------------------------------------------------------------------*/

:root {
  --jp-notebook-toolbar-padding: 2px 5px 2px 2px;
}

/*-----------------------------------------------------------------------------

/*-----------------------------------------------------------------------------
| Styles
|----------------------------------------------------------------------------*/

.jp-NotebookPanel-toolbar {
  padding: var(--jp-notebook-toolbar-padding);

  /* disable paint containment from lumino 2.0 default strict CSS containment */
  contain: style size !important;
}

.jp-Toolbar-item.jp-Notebook-toolbarCellType .jp-select-wrapper.jp-mod-focused {
  border: none;
  box-shadow: none;
}

.jp-Notebook-toolbarCellTypeDropdown select {
  height: 24px;
  font-size: var(--jp-ui-font-size1);
  line-height: 14px;
  border-radius: 0;
  display: block;
}

.jp-Notebook-toolbarCellTypeDropdown span {
  top: 5px !important;
}

.jp-Toolbar-responsive-popup {
  position: absolute;
  height: fit-content;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-end;
  border-bottom: var(--jp-border-width) solid var(--jp-toolbar-border-color);
  box-shadow: var(--jp-toolbar-box-shadow);
  background: var(--jp-toolbar-background);
  min-height: var(--jp-toolbar-micro-height);
  padding: var(--jp-notebook-toolbar-padding);
  z-index: 1;
  right: 0;
  top: 0;
}

.jp-Toolbar > .jp-Toolbar-responsive-opener {
  margin-left: auto;
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Variables
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------

/*-----------------------------------------------------------------------------
| Styles
|----------------------------------------------------------------------------*/

.jp-Notebook-ExecutionIndicator {
  position: relative;
  display: inline-block;
  height: 100%;
  z-index: 9997;
}

.jp-Notebook-ExecutionIndicator-tooltip {
  visibility: hidden;
  height: auto;
  width: max-content;
  width: -moz-max-content;
  background-color: var(--jp-layout-color2);
  color: var(--jp-ui-font-color1);
  text-align: justify;
  border-radius: 6px;
  padding: 0 5px;
  position: fixed;
  display: table;
}

.jp-Notebook-ExecutionIndicator-tooltip.up {
  transform: translateX(-50%) translateY(-100%) translateY(-32px);
}

.jp-Notebook-ExecutionIndicator-tooltip.down {
  transform: translateX(calc(-100% + 16px)) translateY(5px);
}

.jp-Notebook-ExecutionIndicator-tooltip.hidden {
  display: none;
}

.jp-Notebook-ExecutionIndicator:hover .jp-Notebook-ExecutionIndicator-tooltip {
  visibility: visible;
}

.jp-Notebook-ExecutionIndicator span {
  font-size: var(--jp-ui-font-size1);
  font-family: var(--jp-ui-font-family);
  color: var(--jp-ui-font-color1);
  line-height: 24px;
  display: block;
}

.jp-Notebook-ExecutionIndicator-progress-bar {
  display: flex;
  justify-content: center;
  height: 100%;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

/*
 * Execution indicator
 */
.jp-tocItem-content::after {
  content: '';

  /* Must be identical to form a circle */
  width: 12px;
  height: 12px;
  background: none;
  border: none;
  position: absolute;
  right: 0;
}

.jp-tocItem-content[data-running='0']::after {
  border-radius: 50%;
  border: var(--jp-border-width) solid var(--jp-inverse-layout-color3);
  background: none;
}

.jp-tocItem-content[data-running='1']::after {
  border-radius: 50%;
  border: var(--jp-border-width) solid var(--jp-inverse-layout-color3);
  background-color: var(--jp-inverse-layout-color3);
}

.jp-tocItem-content[data-running='0'],
.jp-tocItem-content[data-running='1'] {
  margin-right: 12px;
}

/*
 * Copyright (c) Jupyter Development Team.
 * Distributed under the terms of the Modified BSD License.
 */

.jp-Notebook-footer {
  height: 27px;
  margin-left: calc(
    var(--jp-cell-prompt-width) + var(--jp-cell-collapser-width) +
      var(--jp-cell-padding)
  );
  width: calc(
    100% -
      (
        var(--jp-cell-prompt-width) + var(--jp-cell-collapser-width) +
          var(--jp-cell-padding) + var(--jp-cell-padding)
      )
  );
  border: var(--jp-border-width) solid var(--jp-cell-editor-border-color);
  color: var(--jp-ui-font-color3);
  margin-top: 6px;
  background: none;
  cursor: pointer;
}

.jp-Notebook-footer:focus {
  border-color: var(--jp-cell-editor-active-border-color);
}

/* For devices that support hovering, hide footer until hover */
@media (hover: hover) {
  .jp-Notebook-footer {
    opacity: 0;
  }

  .jp-Notebook-footer:focus,
  .jp-Notebook-footer:hover {
    opacity: 1;
  }
}

/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| Imports
|----------------------------------------------------------------------------*/

/*-----------------------------------------------------------------------------
| CSS variables
|----------------------------------------------------------------------------*/

:root {
  --jp-side-by-side-output-size: 1fr;
  --jp-side-by-side-resized-cell: var(--jp-side-by-side-output-size);
  --jp-private-notebook-dragImage-width: 304px;
  --jp-private-notebook-dragImage-height: 36px;
  --jp-private-notebook-selected-color: var(--md-blue-400);
  --jp-private-notebook-active-color: var(--md-green-400);
}

/*-----------------------------------------------------------------------------
| Notebook
|----------------------------------------------------------------------------*/

/* stylelint-disable selector-max-class */

.jp-NotebookPanel {
  display: block;
  height: 100%;
}

.jp-NotebookPanel.jp-Document {
  min-width: 240px;
  min-height: 120px;
}

.jp-Notebook {
  padding: var(--jp-notebook-padding);
  outline: none;
  overflow: auto;
  background: var(--jp-layout-color0);
}

.jp-Notebook.jp-mod-scrollPastEnd::after {
  display: block;
  content: '';
  min-height: var(--jp-notebook-scroll-padding);
}

.jp-MainAreaWidget-ContainStrict .jp-Notebook * {
  contain: strict;
}

.jp-Notebook .jp-Cell {
  overflow: visible;
}

.jp-Notebook .jp-Cell .jp-InputPrompt {
  cursor: move;
}

/*-----------------------------------------------------------------------------
| Notebook state related styling
|
| The notebook and cells each have states, here are the possibilities:
|
| - Notebook
|   - Command
|   - Edit
| - Cell
|   - None
|   - Active (only one can be active)
|   - Selected (the cells actions are applied to)
|   - Multiselected (when multiple selected, the cursor)
|   - No outputs
|----------------------------------------------------------------------------*/

/* Command or edit modes */

.jp-Notebook .jp-Cell:not(.jp-mod-active) .jp-InputPrompt {
  opacity: var(--jp-cell-prompt-not-active-opacity);
  color: var(--jp-cell-prompt-not-active-font-color);
}

.jp-Notebook .jp-Cell:not(.jp-mod-active) .jp-OutputPrompt {
  opacity: var(--jp-cell-prompt-not-active-opacity);
  color: var(--jp-cell-prompt-not-active-font-color);
}

/* cell is active */
.jp-Notebook .jp-Cell.jp-mod-active .jp-Collapser {
  background: var(--jp-brand-color1);
}

/* cell is dirty */
.jp-Notebook .jp-Cell.jp-mod-dirty .jp-InputPrompt {
  color: var(--jp-warn-color1);
}

.jp-Notebook .jp-Cell.jp-mod-dirty .jp-InputPrompt::before {
  color: var(--jp-warn-color1);
  content: '';
}

.jp-Notebook .jp-Cell.jp-mod-active.jp-mod-dirty .jp-Collapser {
  background: var(--jp-warn-color1);
}

/* collapser is hovered */
.jp-Notebook .jp-Cell .jp-Collapser:hover {
  box-shadow: var(--jp-elevation-z2);
  background: var(--jp-brand-color1);
  opacity: var(--jp-cell-collapser-not-active-hover-opacity);
}

/* cell is active and collapser is hovered */
.jp-Notebook .jp-Cell.jp-mod-active .jp-Collapser:hover {
  background: var(--jp-brand-color0);
  opacity: 1;
}

/* Command mode */

.jp-Notebook.jp-mod-commandMode .jp-Cell.jp-mod-selected {
  background: var(--jp-notebook-multiselected-color);
}

.jp-Notebook.jp-mod-commandMode
  .jp-Cell.jp-mod-active.jp-mod-selected:not(.jp-mod-multiSelected) {
  background: transparent;
}

/* Edit mode */

.jp-Notebook.jp-mod-editMode .jp-Cell.jp-mod-active .jp-InputArea-editor {
  border: var(--jp-border-width) solid var(--jp-cell-editor-active-border-color);
  box-shadow: var(--jp-input-box-shadow);
  background-color: var(--jp-cell-editor-active-background);
}

/*-----------------------------------------------------------------------------
| Notebook drag and drop
|----------------------------------------------------------------------------*/

.jp-Notebook-cell.jp-mod-dropSource {
  opacity: 0.5;
}

.jp-Notebook-cell.jp-mod-dropTarget,
.jp-Notebook.jp-mod-commandMode
  .jp-Notebook-cell.jp-mod-active.jp-mod-selected.jp-mod-dropTarget {
  border-top-color: var(--jp-private-notebook-selected-color);
  border-top-style: solid;
  border-top-width: 2px;
}

.jp-dragImage {
  display: block;
  flex-direction: row;
  width: var(--jp-private-notebook-dragImage-width);
  height: var(--jp-private-notebook-dragImage-height);
  border: var(--jp-border-width) solid var(--jp-cell-editor-border-color);
  background: var(--jp-cell-editor-background);
  overflow: visible;
}

.jp-dragImage-singlePrompt {
  box-shadow: 2px 2px 4px 0 rgba(0, 0, 0, 0.12);
}

.jp-dragImage .jp-dragImage-content {
  flex: 1 1 auto;
  z-index: 2;
  font-size: var(--jp-code-font-size);
  font-family: var(--jp-code-font-family);
  line-height: var(--jp-code-line-height);
  padding: var(--jp-code-padding);
  border: var(--jp-border-width) solid var(--jp-cell-editor-border-color);
  background: var(--jp-cell-editor-background-color);
  color: var(--jp-content-font-color3);
  text-align: left;
  margin: 4px 4px 4px 0;
}

.jp-dragImage .jp-dragImage-prompt {
  flex: 0 0 auto;
  min-width: 36px;
  color: var(--jp-cell-inprompt-font-color);
  padding: var(--jp-code-padding);
  padding-left: 12px;
  font-family: var(--jp-cell-prompt-font-family);
  letter-spacing: var(--jp-cell-prompt-letter-spacing);
  line-height: 1.9;
  font-size: var(--jp-code-font-size);
  border: var(--jp-border-width) solid transparent;
}

.jp-dragImage-multipleBack {
  z-index: -1;
  position: absolute;
  height: 32px;
  width: 300px;
  top: 8px;
  left: 8px;
  background: var(--jp-layout-color2);
  border: var(--jp-border-width) solid var(--jp-input-border-color);
  box-shadow: 2px 2px 4px 0 rgba(0, 0, 0, 0.12);
}

/*-----------------------------------------------------------------------------
| Cell toolbar
|----------------------------------------------------------------------------*/

.jp-NotebookTools {
  display: block;
  min-width: var(--jp-sidebar-min-width);
  color: var(--jp-ui-font-color1);
  background: var(--jp-layout-color1);

  /* This is needed so that all font sizing of children done in ems is
    * relative to this base size */
  font-size: var(--jp-ui-font-size1);
  overflow: auto;
}

.jp-ActiveCellTool {
  padding: 12px 0;
  display: flex;
}

.jp-ActiveCellTool-Content {
  flex: 1 1 auto;
}

.jp-ActiveCellTool .jp-ActiveCellTool-CellContent {
  background: var(--jp-cell-editor-background);
  border: var(--jp-border-width) solid var(--jp-cell-editor-border-color);
  border-radius: 0;
  min-height: 29px;
}

.jp-ActiveCellTool .jp-InputPrompt {
  min-width: calc(var(--jp-cell-prompt-width) * 0.75);
}

.jp-ActiveCellTool-CellContent > pre {
  padding: 5px 4px;
  margin: 0;
  white-space: normal;
}

.jp-MetadataEditorTool {
  flex-direction: column;
  padding: 12px 0;
}

.jp-RankedPanel > :not(:first-child) {
  margin-top: 12px;
}

.jp-KeySelector select.jp-mod-styled {
  font-size: var(--jp-ui-font-size1);
  color: var(--jp-ui-font-color0);
  border: var(--jp-border-width) solid var(--jp-border-color1);
}

.jp-KeySelector label,
.jp-MetadataEditorTool label,
.jp-NumberSetter label {
  line-height: 1.4;
}

.jp-NotebookTools .jp-select-wrapper {
  margin-top: 4px;
  margin-bottom: 0;
}

.jp-NumberSetter input {
  width: 100%;
  margin-top: 4px;
}

.jp-NotebookTools .jp-Collapse {
  margin-top: 16px;
}

/*-----------------------------------------------------------------------------
| Presentation Mode (.jp-mod-presentationMode)
|----------------------------------------------------------------------------*/

.jp-mod-presentationMode .jp-Notebook {
  --jp-content-font-size1: var(--jp-content-presentation-font-size1);
  --jp-code-font-size: var(--jp-code-presentation-font-size);
}

.jp-mod-presentationMode .jp-Notebook .jp-Cell .jp-InputPrompt,
.jp-mod-presentationMode .jp-Notebook .jp-Cell .jp-OutputPrompt {
  flex: 0 0 110px;
}

/*-----------------------------------------------------------------------------
| Side-by-side Mode (.jp-mod-sideBySide)
|----------------------------------------------------------------------------*/
.jp-mod-sideBySide.jp-Notebook .jp-Notebook-cell {
  margin-top: 3em;
  margin-bottom: 3em;
  margin-left: 5%;
  margin-right: 5%;
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell {
  display: grid;
  grid-template-columns: minmax(0, 1fr) min-content minmax(
      0,
      var(--jp-side-by-side-output-size)
    );
  grid-template-rows: auto minmax(0, 1fr) auto;
  grid-template-areas:
    'header header header'
    'input handle output'
    'footer footer footer';
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell.jp-mod-resizedCell {
  grid-template-columns: minmax(0, 1fr) min-content minmax(
      0,
      var(--jp-side-by-side-resized-cell)
    );
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell .jp-CellHeader {
  grid-area: header;
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell .jp-Cell-inputWrapper {
  grid-area: input;
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell .jp-Cell-outputWrapper {
  /* overwrite the default margin (no vertical separation needed in side by side move */
  margin-top: 0;
  grid-area: output;
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell .jp-CellFooter {
  grid-area: footer;
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell .jp-CellResizeHandle {
  grid-area: handle;
  user-select: none;
  display: block;
  height: 100%;
  cursor: ew-resize;
  padding: 0 var(--jp-cell-padding);
}

.jp-mod-sideBySide.jp-Notebook .jp-CodeCell .jp-CellResizeHandle::after {
  content: '';
  display: block;
  background: var(--jp-border-color2);
  height: 100%;
  width: 5px;
}

.jp-mod-sideBySide.jp-Notebook
  .jp-CodeCell.jp-mod-resizedCell
  .jp-CellResizeHandle::after {
  background: var(--jp-border-color0);
}

.jp-CellResizeHandle {
  display: none;
}

/*-----------------------------------------------------------------------------
| Placeholder
|----------------------------------------------------------------------------*/

.jp-Cell-Placeholder {
  padding-left: 55px;
}

.jp-Cell-Placeholder-wrapper {
  background: #fff;
  border: 1px solid;
  border-color: #e5e6e9 #dfe0e4 #d0d1d5;
  border-radius: 4px;
  -webkit-border-radius: 4px;
  margin: 10px 15px;
}

.jp-Cell-Placeholder-wrapper-inner {
  padding: 15px;
  position: relative;
}

.jp-Cell-Placeholder-wrapper-body {
  background-repeat: repeat;
  background-size: 50% auto;
}

.jp-Cell-Placeholder-wrapper-body div {
  background: #f6f7f8;
  background-image: -webkit-linear-gradient(
    left,
    #f6f7f8 0%,
    #edeef1 20%,
    #f6f7f8 40%,
    #f6f7f8 100%
  );
  background-repeat: no-repeat;
  background-size: 800px 104px;
  height: 104px;
  position: absolute;
  right: 15px;
  left: 15px;
  top: 15px;
}

div.jp-Cell-Placeholder-h1 {
  top: 20px;
  height: 20px;
  left: 15px;
  width: 150px;
}

div.jp-Cell-Placeholder-h2 {
  left: 15px;
  top: 50px;
  height: 10px;
  width: 100px;
}

div.jp-Cell-Placeholder-content-1,
div.jp-Cell-Placeholder-content-2,
div.jp-Cell-Placeholder-content-3 {
  left: 15px;
  right: 15px;
  height: 10px;
}

div.jp-Cell-Placeholder-content-1 {
  top: 100px;
}

div.jp-Cell-Placeholder-content-2 {
  top: 120px;
}

div.jp-Cell-Placeholder-content-3 {
  top: 140px;
}

</style>
<style type="text/css">
/*-----------------------------------------------------------------------------
| Copyright (c) Jupyter Development Team.
| Distributed under the terms of the Modified BSD License.
|----------------------------------------------------------------------------*/

/*
The following CSS variables define the main, public API for styling JupyterLab.
These variables should be used by all plugins wherever possible. In other
words, plugins should not define custom colors, sizes, etc unless absolutely
necessary. This enables users to change the visual theme of JupyterLab
by changing these variables.

Many variables appear in an ordered sequence (0,1,2,3). These sequences
are designed to work well together, so for example, `--jp-border-color1` should
be used with `--jp-layout-color1`. The numbers have the following meanings:

* 0: super-primary, reserved for special emphasis
* 1: primary, most important under normal situations
* 2: secondary, next most important under normal situations
* 3: tertiary, next most important under normal situations

Throughout JupyterLab, we are mostly following principles from Google's
Material Design when selecting colors. We are not, however, following
all of MD as it is not optimized for dense, information rich UIs.
*/

:root {
  /* Elevation
   *
   * We style box-shadows using Material Design's idea of elevation. These particular numbers are taken from here:
   *
   * https://github.com/material-components/material-components-web
   * https://material-components-web.appspot.com/elevation.html
   */

  --jp-shadow-base-lightness: 0;
  --jp-shadow-umbra-color: rgba(
    var(--jp-shadow-base-lightness),
    var(--jp-shadow-base-lightness),
    var(--jp-shadow-base-lightness),
    0.2
  );
  --jp-shadow-penumbra-color: rgba(
    var(--jp-shadow-base-lightness),
    var(--jp-shadow-base-lightness),
    var(--jp-shadow-base-lightness),
    0.14
  );
  --jp-shadow-ambient-color: rgba(
    var(--jp-shadow-base-lightness),
    var(--jp-shadow-base-lightness),
    var(--jp-shadow-base-lightness),
    0.12
  );
  --jp-elevation-z0: none;
  --jp-elevation-z1: 0 2px 1px -1px var(--jp-shadow-umbra-color),
    0 1px 1px 0 var(--jp-shadow-penumbra-color),
    0 1px 3px 0 var(--jp-shadow-ambient-color);
  --jp-elevation-z2: 0 3px 1px -2px var(--jp-shadow-umbra-color),
    0 2px 2px 0 var(--jp-shadow-penumbra-color),
    0 1px 5px 0 var(--jp-shadow-ambient-color);
  --jp-elevation-z4: 0 2px 4px -1px var(--jp-shadow-umbra-color),
    0 4px 5px 0 var(--jp-shadow-penumbra-color),
    0 1px 10px 0 var(--jp-shadow-ambient-color);
  --jp-elevation-z6: 0 3px 5px -1px var(--jp-shadow-umbra-color),
    0 6px 10px 0 var(--jp-shadow-penumbra-color),
    0 1px 18px 0 var(--jp-shadow-ambient-color);
  --jp-elevation-z8: 0 5px 5px -3px var(--jp-shadow-umbra-color),
    0 8px 10px 1px var(--jp-shadow-penumbra-color),
    0 3px 14px 2px var(--jp-shadow-ambient-color);
  --jp-elevation-z12: 0 7px 8px -4px var(--jp-shadow-umbra-color),
    0 12px 17px 2px var(--jp-shadow-penumbra-color),
    0 5px 22px 4px var(--jp-shadow-ambient-color);
  --jp-elevation-z16: 0 8px 10px -5px var(--jp-shadow-umbra-color),
    0 16px 24px 2px var(--jp-shadow-penumbra-color),
    0 6px 30px 5px var(--jp-shadow-ambient-color);
  --jp-elevation-z20: 0 10px 13px -6px var(--jp-shadow-umbra-color),
    0 20px 31px 3px var(--jp-shadow-penumbra-color),
    0 8px 38px 7px var(--jp-shadow-ambient-color);
  --jp-elevation-z24: 0 11px 15px -7px var(--jp-shadow-umbra-color),
    0 24px 38px 3px var(--jp-shadow-penumbra-color),
    0 9px 46px 8px var(--jp-shadow-ambient-color);

  /* Borders
   *
   * The following variables, specify the visual styling of borders in JupyterLab.
   */

  --jp-border-width: 1px;
  --jp-border-color0: var(--md-grey-400);
  --jp-border-color1: var(--md-grey-400);
  --jp-border-color2: var(--md-grey-300);
  --jp-border-color3: var(--md-grey-200);
  --jp-inverse-border-color: var(--md-grey-600);
  --jp-border-radius: 2px;

  /* UI Fonts
   *
   * The UI font CSS variables are used for the typography all of the JupyterLab
   * user interface elements that are not directly user generated content.
   *
   * The font sizing here is done assuming that the body font size of --jp-ui-font-size1
   * is applied to a parent element. When children elements, such as headings, are sized
   * in em all things will be computed relative to that body size.
   */

  --jp-ui-font-scale-factor: 1.2;
  --jp-ui-font-size0: 0.83333em;
  --jp-ui-font-size1: 13px; /* Base font size */
  --jp-ui-font-size2: 1.2em;
  --jp-ui-font-size3: 1.44em;
  --jp-ui-font-family: system-ui, -apple-system, blinkmacsystemfont, 'Segoe UI',
    helvetica, arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
    'Segoe UI Symbol';

  /*
   * Use these font colors against the corresponding main layout colors.
   * In a light theme, these go from dark to light.
   */

  /* Defaults use Material Design specification */
  --jp-ui-font-color0: rgba(0, 0, 0, 1);
  --jp-ui-font-color1: rgba(0, 0, 0, 0.87);
  --jp-ui-font-color2: rgba(0, 0, 0, 0.54);
  --jp-ui-font-color3: rgba(0, 0, 0, 0.38);

  /*
   * Use these against the brand/accent/warn/error colors.
   * These will typically go from light to darker, in both a dark and light theme.
   */

  --jp-ui-inverse-font-color0: rgba(255, 255, 255, 1);
  --jp-ui-inverse-font-color1: rgba(255, 255, 255, 1);
  --jp-ui-inverse-font-color2: rgba(255, 255, 255, 0.7);
  --jp-ui-inverse-font-color3: rgba(255, 255, 255, 0.5);

  /* Content Fonts
   *
   * Content font variables are used for typography of user generated content.
   *
   * The font sizing here is done assuming that the body font size of --jp-content-font-size1
   * is applied to a parent element. When children elements, such as headings, are sized
   * in em all things will be computed relative to that body size.
   */

  --jp-content-line-height: 1.6;
  --jp-content-font-scale-factor: 1.2;
  --jp-content-font-size0: 0.83333em;
  --jp-content-font-size1: 14px; /* Base font size */
  --jp-content-font-size2: 1.2em;
  --jp-content-font-size3: 1.44em;
  --jp-content-font-size4: 1.728em;
  --jp-content-font-size5: 2.0736em;

  /* This gives a magnification of about 125% in presentation mode over normal. */
  --jp-content-presentation-font-size1: 17px;
  --jp-content-heading-line-height: 1;
  --jp-content-heading-margin-top: 1.2em;
  --jp-content-heading-margin-bottom: 0.8em;
  --jp-content-heading-font-weight: 500;

  /* Defaults use Material Design specification */
  --jp-content-font-color0: rgba(0, 0, 0, 1);
  --jp-content-font-color1: rgba(0, 0, 0, 0.87);
  --jp-content-font-color2: rgba(0, 0, 0, 0.54);
  --jp-content-font-color3: rgba(0, 0, 0, 0.38);
  --jp-content-link-color: var(--md-blue-900);
  --jp-content-font-family: system-ui, -apple-system, blinkmacsystemfont,
    'Segoe UI', helvetica, arial, sans-serif, 'Apple Color Emoji',
    'Segoe UI Emoji', 'Segoe UI Symbol';

  /*
   * Code Fonts
   *
   * Code font variables are used for typography of code and other monospaces content.
   */

  --jp-code-font-size: 13px;
  --jp-code-line-height: 1.3077; /* 17px for 13px base */
  --jp-code-padding: 5px; /* 5px for 13px base, codemirror highlighting needs integer px value */
  --jp-code-font-family-default: menlo, consolas, 'DejaVu Sans Mono', monospace;
  --jp-code-font-family: var(--jp-code-font-family-default);

  /* This gives a magnification of about 125% in presentation mode over normal. */
  --jp-code-presentation-font-size: 16px;

  /* may need to tweak cursor width if you change font size */
  --jp-code-cursor-width0: 1.4px;
  --jp-code-cursor-width1: 2px;
  --jp-code-cursor-width2: 4px;

  /* Layout
   *
   * The following are the main layout colors use in JupyterLab. In a light
   * theme these would go from light to dark.
   */

  --jp-layout-color0: white;
  --jp-layout-color1: white;
  --jp-layout-color2: var(--md-grey-200);
  --jp-layout-color3: var(--md-grey-400);
  --jp-layout-color4: var(--md-grey-600);

  /* Inverse Layout
   *
   * The following are the inverse layout colors use in JupyterLab. In a light
   * theme these would go from dark to light.
   */

  --jp-inverse-layout-color0: #111;
  --jp-inverse-layout-color1: var(--md-grey-900);
  --jp-inverse-layout-color2: var(--md-grey-800);
  --jp-inverse-layout-color3: var(--md-grey-700);
  --jp-inverse-layout-color4: var(--md-grey-600);

  /* Brand/accent */

  --jp-brand-color0: var(--md-blue-900);
  --jp-brand-color1: var(--md-blue-700);
  --jp-brand-color2: var(--md-blue-300);
  --jp-brand-color3: var(--md-blue-100);
  --jp-brand-color4: var(--md-blue-50);
  --jp-accent-color0: var(--md-green-900);
  --jp-accent-color1: var(--md-green-700);
  --jp-accent-color2: var(--md-green-300);
  --jp-accent-color3: var(--md-green-100);

  /* State colors (warn, error, success, info) */

  --jp-warn-color0: var(--md-orange-900);
  --jp-warn-color1: var(--md-orange-700);
  --jp-warn-color2: var(--md-orange-300);
  --jp-warn-color3: var(--md-orange-100);
  --jp-error-color0: var(--md-red-900);
  --jp-error-color1: var(--md-red-700);
  --jp-error-color2: var(--md-red-300);
  --jp-error-color3: var(--md-red-100);
  --jp-success-color0: var(--md-green-900);
  --jp-success-color1: var(--md-green-700);
  --jp-success-color2: var(--md-green-300);
  --jp-success-color3: var(--md-green-100);
  --jp-info-color0: var(--md-cyan-900);
  --jp-info-color1: var(--md-cyan-700);
  --jp-info-color2: var(--md-cyan-300);
  --jp-info-color3: var(--md-cyan-100);

  /* Cell specific styles */

  --jp-cell-padding: 5px;
  --jp-cell-collapser-width: 8px;
  --jp-cell-collapser-min-height: 20px;
  --jp-cell-collapser-not-active-hover-opacity: 0.6;
  --jp-cell-editor-background: var(--md-grey-100);
  --jp-cell-editor-border-color: var(--md-grey-300);
  --jp-cell-editor-box-shadow: inset 0 0 2px var(--md-blue-300);
  --jp-cell-editor-active-background: var(--jp-layout-color0);
  --jp-cell-editor-active-border-color: var(--jp-brand-color1);
  --jp-cell-prompt-width: 64px;
  --jp-cell-prompt-font-family: var(--jp-code-font-family-default);
  --jp-cell-prompt-letter-spacing: 0;
  --jp-cell-prompt-opacity: 1;
  --jp-cell-prompt-not-active-opacity: 0.5;
  --jp-cell-prompt-not-active-font-color: var(--md-grey-700);

  /* A custom blend of MD grey and blue 600
   * See https://meyerweb.com/eric/tools/color-blend/#546E7A:1E88E5:5:hex */
  --jp-cell-inprompt-font-color: #307fc1;

  /* A custom blend of MD grey and orange 600
   * https://meyerweb.com/eric/tools/color-blend/#546E7A:F4511E:5:hex */
  --jp-cell-outprompt-font-color: #bf5b3d;

  /* Notebook specific styles */

  --jp-notebook-padding: 10px;
  --jp-notebook-select-background: var(--jp-layout-color1);
  --jp-notebook-multiselected-color: var(--md-blue-50);

  /* The scroll padding is calculated to fill enough space at the bottom of the
  notebook to show one single-line cell (with appropriate padding) at the top
  when the notebook is scrolled all the way to the bottom. We also subtract one
  pixel so that no scrollbar appears if we have just one single-line cell in the
  notebook. This padding is to enable a 'scroll past end' feature in a notebook.
  */
  --jp-notebook-scroll-padding: calc(
    100% - var(--jp-code-font-size) * var(--jp-code-line-height) -
      var(--jp-code-padding) - var(--jp-cell-padding) - 1px
  );

  /* Rendermime styles */

  --jp-rendermime-error-background: #fdd;
  --jp-rendermime-table-row-background: var(--md-grey-100);
  --jp-rendermime-table-row-hover-background: var(--md-light-blue-50);

  /* Dialog specific styles */

  --jp-dialog-background: rgba(0, 0, 0, 0.25);

  /* Console specific styles */

  --jp-console-padding: 10px;

  /* Toolbar specific styles */

  --jp-toolbar-border-color: var(--jp-border-color1);
  --jp-toolbar-micro-height: 8px;
  --jp-toolbar-background: var(--jp-layout-color1);
  --jp-toolbar-box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.24);
  --jp-toolbar-header-margin: 4px 4px 0 4px;
  --jp-toolbar-active-background: var(--md-grey-300);

  /* Statusbar specific styles */

  --jp-statusbar-height: 24px;

  /* Input field styles */

  --jp-input-box-shadow: inset 0 0 2px var(--md-blue-300);
  --jp-input-active-background: var(--jp-layout-color1);
  --jp-input-hover-background: var(--jp-layout-color1);
  --jp-input-background: var(--md-grey-100);
  --jp-input-border-color: var(--jp-inverse-border-color);
  --jp-input-active-border-color: var(--jp-brand-color1);
  --jp-input-active-box-shadow-color: rgba(19, 124, 189, 0.3);

  /* General editor styles */

  --jp-editor-selected-background: #d9d9d9;
  --jp-editor-selected-focused-background: #d7d4f0;
  --jp-editor-cursor-color: var(--jp-ui-font-color0);

  /* Code mirror specific styles */

  --jp-mirror-editor-keyword-color: #008000;
  --jp-mirror-editor-atom-color: #88f;
  --jp-mirror-editor-number-color: #080;
  --jp-mirror-editor-def-color: #00f;
  --jp-mirror-editor-variable-color: var(--md-grey-900);
  --jp-mirror-editor-variable-2-color: rgb(0, 54, 109);
  --jp-mirror-editor-variable-3-color: #085;
  --jp-mirror-editor-punctuation-color: #05a;
  --jp-mirror-editor-property-color: #05a;
  --jp-mirror-editor-operator-color: #a2f;
  --jp-mirror-editor-comment-color: #408080;
  --jp-mirror-editor-string-color: #ba2121;
  --jp-mirror-editor-string-2-color: #708;
  --jp-mirror-editor-meta-color: #a2f;
  --jp-mirror-editor-qualifier-color: #555;
  --jp-mirror-editor-builtin-color: #008000;
  --jp-mirror-editor-bracket-color: #997;
  --jp-mirror-editor-tag-color: #170;
  --jp-mirror-editor-attribute-color: #00c;
  --jp-mirror-editor-header-color: blue;
  --jp-mirror-editor-quote-color: #090;
  --jp-mirror-editor-link-color: #00c;
  --jp-mirror-editor-error-color: #f00;
  --jp-mirror-editor-hr-color: #999;

  /*
    RTC user specific colors.
    These colors are used for the cursor, username in the editor,
    and the icon of the user.
  */

  --jp-collaborator-color1: #ffad8e;
  --jp-collaborator-color2: #dac83d;
  --jp-collaborator-color3: #72dd76;
  --jp-collaborator-color4: #00e4d0;
  --jp-collaborator-color5: #45d4ff;
  --jp-collaborator-color6: #e2b1ff;
  --jp-collaborator-color7: #ff9de6;

  /* Vega extension styles */

  --jp-vega-background: white;

  /* Sidebar-related styles */

  --jp-sidebar-min-width: 250px;

  /* Search-related styles */

  --jp-search-toggle-off-opacity: 0.5;
  --jp-search-toggle-hover-opacity: 0.8;
  --jp-search-toggle-on-opacity: 1;
  --jp-search-selected-match-background-color: rgb(245, 200, 0);
  --jp-search-selected-match-color: black;
  --jp-search-unselected-match-background-color: var(
    --jp-inverse-layout-color0
  );
  --jp-search-unselected-match-color: var(--jp-ui-inverse-font-color0);

  /* Icon colors that work well with light or dark backgrounds */
  --jp-icon-contrast-color0: var(--md-purple-600);
  --jp-icon-contrast-color1: var(--md-green-600);
  --jp-icon-contrast-color2: var(--md-pink-600);
  --jp-icon-contrast-color3: var(--md-blue-600);

  /* Button colors */
  --jp-accept-color-normal: var(--md-blue-700);
  --jp-accept-color-hover: var(--md-blue-800);
  --jp-accept-color-active: var(--md-blue-900);
  --jp-warn-color-normal: var(--md-red-700);
  --jp-warn-color-hover: var(--md-red-800);
  --jp-warn-color-active: var(--md-red-900);
  --jp-reject-color-normal: var(--md-grey-600);
  --jp-reject-color-hover: var(--md-grey-700);
  --jp-reject-color-active: var(--md-grey-800);

  /* File or activity icons and switch semantic variables */
  --jp-jupyter-icon-color: #f37626;
  --jp-notebook-icon-color: #f37626;
  --jp-json-icon-color: var(--md-orange-700);
  --jp-console-icon-background-color: var(--md-blue-700);
  --jp-console-icon-color: white;
  --jp-terminal-icon-background-color: var(--md-grey-800);
  --jp-terminal-icon-color: var(--md-grey-200);
  --jp-text-editor-icon-color: var(--md-grey-700);
  --jp-inspector-icon-color: var(--md-grey-700);
  --jp-switch-color: var(--md-grey-400);
  --jp-switch-true-position-color: var(--md-orange-900);
}
</style>
<style type="text/css">
/* Force rendering true colors when outputing to pdf */
* {
  -webkit-print-color-adjust: exact;
}

/* Misc */
a.anchor-link {
  display: none;
}

/* Input area styling */
.jp-InputArea {
  overflow: hidden;
}

.jp-InputArea-editor {
  overflow: hidden;
}

.cm-editor.cm-s-jupyter .highlight pre {
/* weird, but --jp-code-padding defined to be 5px but 4px horizontal padding is hardcoded for pre.cm-line */
  padding: var(--jp-code-padding) 4px;
  margin: 0;

  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;

}

.jp-OutputArea-output pre {
  line-height: inherit;
  font-family: inherit;
}

.jp-RenderedText pre {
  color: var(--jp-content-font-color1);
  font-size: var(--jp-code-font-size);
}

/* Hiding the collapser by default */
.jp-Collapser {
  display: none;
}

@page {
    margin: 0.5in; /* Margin for each printed piece of paper */
}

@media print {
  .jp-Cell-inputWrapper,
  .jp-Cell-outputWrapper {
    display: block;
  }
}
</style>
<!-- Load mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS_CHTML-full,Safe"> </script>
<!-- MathJax configuration -->
<script type="text/x-mathjax-config">
    init_mathjax = function() {
        if (window.MathJax) {
        // MathJax loaded
            MathJax.Hub.Config({
                TeX: {
                    equationNumbers: {
                    autoNumber: "AMS",
                    useLabelIds: true
                    }
                },
                tex2jax: {
                    inlineMath: [ ['$','$'], ["\\(","\\)"] ],
                    displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
                    processEscapes: true,
                    processEnvironments: true
                },
                displayAlign: 'center',
                CommonHTML: {
                    linebreaks: {
                    automatic: true
                    }
                }
            });

            MathJax.Hub.Queue(["Typeset", MathJax.Hub]);
        }
    }
    init_mathjax();
    </script>
<!-- End of mathjax configuration --><script type="module">
  document.addEventListener("DOMContentLoaded", async () => {
    const diagrams = document.querySelectorAll(".jp-Mermaid > pre.mermaid");
    // do not load mermaidjs if not needed
    if (!diagrams.length) {
      return;
    }
    const mermaid = (await import("https://cdnjs.cloudflare.com/ajax/libs/mermaid/10.7.0/mermaid.esm.min.mjs")).default;
    const parser = new DOMParser();

    mermaid.initialize({
      maxTextSize: 100000,
      maxEdges: 100000,
      startOnLoad: false,
      fontFamily: window
        .getComputedStyle(document.body)
        .getPropertyValue("--jp-ui-font-family"),
      theme: document.querySelector("body[data-jp-theme-light='true']")
        ? "default"
        : "dark",
    });

    let _nextMermaidId = 0;

    function makeMermaidImage(svg) {
      const img = document.createElement("img");
      const doc = parser.parseFromString(svg, "image/svg+xml");
      const svgEl = doc.querySelector("svg");
      const { maxWidth } = svgEl?.style || {};
      const firstTitle = doc.querySelector("title");
      const firstDesc = doc.querySelector("desc");

      img.setAttribute("src", `data:image/svg+xml,${encodeURIComponent(svg)}`);
      if (maxWidth) {
        img.width = parseInt(maxWidth);
      }
      if (firstTitle) {
        img.setAttribute("alt", firstTitle.textContent);
      }
      if (firstDesc) {
        const caption = document.createElement("figcaption");
        caption.className = "sr-only";
        caption.textContent = firstDesc.textContent;
        return [img, caption];
      }
      return [img];
    }

    async function makeMermaidError(text) {
      let errorMessage = "";
      try {
        await mermaid.parse(text);
      } catch (err) {
        errorMessage = `${err}`;
      }

      const result = document.createElement("details");
      result.className = 'jp-RenderedMermaid-Details';
      const summary = document.createElement("summary");
      summary.className = 'jp-RenderedMermaid-Summary';
      const pre = document.createElement("pre");
      const code = document.createElement("code");
      code.innerText = text;
      pre.appendChild(code);
      summary.appendChild(pre);
      result.appendChild(summary);

      const warning = document.createElement("pre");
      warning.innerText = errorMessage;
      result.appendChild(warning);
      return [result];
    }

    async function renderOneMarmaid(src) {
      const id = `jp-mermaid-${_nextMermaidId++}`;
      const parent = src.parentNode;
      let raw = src.textContent.trim();
      const el = document.createElement("div");
      el.style.visibility = "hidden";
      document.body.appendChild(el);
      let results = null;
      let output = null;
      try {
        let { svg } = await mermaid.render(id, raw, el);
        svg = cleanMermaidSvg(svg);
        results = makeMermaidImage(svg);
        output = document.createElement("figure");
        results.map(output.appendChild, output);
      } catch (err) {
        parent.classList.add("jp-mod-warning");
        results = await makeMermaidError(raw);
        output = results[0];
      } finally {
        el.remove();
      }
      parent.classList.add("jp-RenderedMermaid");
      parent.appendChild(output);
    }


    /**
     * Post-process to ensure mermaid diagrams contain only valid SVG and XHTML.
     */
    function cleanMermaidSvg(svg) {
      return svg.replace(RE_VOID_ELEMENT, replaceVoidElement);
    }


    /**
     * A regular expression for all void elements, which may include attributes and
     * a slash.
     *
     * @see https://developer.mozilla.org/en-US/docs/Glossary/Void_element
     *
     * Of these, only `<br>` is generated by Mermaid in place of `\n`,
     * but _any_ "malformed" tag will break the SVG rendering entirely.
     */
    const RE_VOID_ELEMENT =
      /<\s*(area|base|br|col|embed|hr|img|input|link|meta|param|source|track|wbr)\s*([^>]*?)\s*>/gi;

    /**
     * Ensure a void element is closed with a slash, preserving any attributes.
     */
    function replaceVoidElement(match, tag, rest) {
      rest = rest.trim();
      if (!rest.endsWith('/')) {
        rest = `${rest} /`;
      }
      return `<${tag} ${rest}>`;
    }

    void Promise.all([...diagrams].map(renderOneMarmaid));
  });
</script>
<style>
  .jp-Mermaid:not(.jp-RenderedMermaid) {
    display: none;
  }

  .jp-RenderedMermaid {
    overflow: auto;
    display: flex;
  }

  .jp-RenderedMermaid.jp-mod-warning {
    width: auto;
    padding: 0.5em;
    margin-top: 0.5em;
    border: var(--jp-border-width) solid var(--jp-warn-color2);
    border-radius: var(--jp-border-radius);
    color: var(--jp-ui-font-color1);
    font-size: var(--jp-ui-font-size1);
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  .jp-RenderedMermaid figure {
    margin: 0;
    overflow: auto;
    max-width: 100%;
  }

  .jp-RenderedMermaid img {
    max-width: 100%;
  }

  .jp-RenderedMermaid-Details > pre {
    margin-top: 1em;
  }

  .jp-RenderedMermaid-Summary {
    color: var(--jp-warn-color2);
  }

  .jp-RenderedMermaid:not(.jp-mod-warning) pre {
    display: none;
  }

  .jp-RenderedMermaid-Summary > pre {
    display: inline-block;
    white-space: normal;
  }
</style>
<!-- End of mermaid configuration --></head>
<body class="jp-Notebook" data-jp-theme-light="true" data-jp-theme-name="JupyterLab Light">
<main>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=db70b75b-56b6-4611-9e5d-63f35668d365">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Loading Python libraries</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=6ead6872-5637-4e31-a6e3-543238a8e5e2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[1]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Use pathlib for file path manipulation</span>
<span class="kn">import</span> <span class="nn">pathlib</span>

<span class="c1"># Install numpy</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>

<span class="c1"># Install Pandas for tabular data</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="c1"># Install plotnine: a ggplot2-compatible Python plotting package</span>
<span class="kn">from</span> <span class="nn">plotnine</span> <span class="kn">import</span> <span class="o">*</span>

<span class="c1"># Always show all columns in a Pandas DataFrame</span>
<span class="n">pd</span><span class="o">.</span><span class="n">set_option</span><span class="p">(</span><span class="s1">'display.max_columns'</span><span class="p">,</span> <span class="kc">None</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=ef9c9cce-81a5-4c44-8dbf-cc91cc67ef9b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Set paths</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=854afae5-dc23-4c88-96fb-89ff5175aeee">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[2]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">REL7_PATH</span> <span class="o">=</span> <span class="n">pathlib</span><span class="o">.</span><span class="n">Path</span><span class="p">(</span><span class="n">pathlib</span><span class="o">.</span><span class="n">Path</span><span class="o">.</span><span class="n">home</span><span class="p">(),</span> <span class="s1">'workspace/gp2_tier2_eu_release7_30042024'</span><span class="p">)</span>
<span class="o">!</span>ls<span class="w"> </span>-hal<span class="w"> </span><span class="o">{</span>REL7_PATH<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>total 61K
-r--r--r-- 1 jupyter jupyter 61K Jul 23  2024 README_release7_30042024.txt
dr-xr-xr-x 1 jupyter jupyter   0 Mar  5 10:22 clinical_data
dr-xr-xr-x 1 jupyter jupyter   0 Mar  5 10:22 imputed_genotypes
dr-xr-xr-x 1 jupyter jupyter   0 Mar  5 10:22 meta_data
dr-xr-xr-x 1 jupyter jupyter   0 Mar  5 10:22 raw_genotypes
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=351766ac-da37-42a1-aed8-38e0d044c429">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Make working directory</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=d2df0a09-9eed-421f-815e-47fe9ea36a6a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[3]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=28d5c734-cb5f-498c-bb37-b9e6d2872fdb">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[4]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># make sure all tools installed</span>
<span class="o">!</span><span class="w"> </span>ls<span class="w"> </span>/home/jupyter/tools
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>LICENSE		       plink2				rvtests
annovar		       plink2_linux_x86_64_latest.zip	toy.map
annovar.latest.tar.gz  plink_linux_x86_64_20190304.zip	toy.ped
plink		       prettify
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=9150c8dc-16b1-44ad-9d51-7c3c9708990a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[5]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># give permission</span>

<span class="c1"># chmod to make sure you have permission to run the program</span>
<span class="o">!</span><span class="w"> </span>chmod<span class="w"> </span>u+x<span class="w"> </span>/home/jupyter/tools/plink
<span class="o">!</span><span class="w"> </span>chmod<span class="w"> </span>u+x<span class="w"> </span>/home/jupyter/tools/plink2
<span class="o">!</span><span class="w"> </span>chmod<span class="w"> </span><span class="m">777</span><span class="w"> </span>/home/jupyter/tools/rvtests/executable/rvtest
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=20f842ba-c03c-4604-972f-3e9b5b37d530">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[6]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span>%%bash
<span class="c1"># making working directory</span>
<span class="c1">#Loop over all the ancestries</span>
<span class="k">for</span><span class="w"> </span>ancestry<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="o">{</span><span class="s1">'AAC'</span>,<span class="s1">'AFR'</span>,<span class="s1">'AJ'</span>,<span class="s1">'AMR'</span>,<span class="s1">'CAS'</span>,<span class="s1">'EAS'</span>,<span class="s1">'EUR'</span>,<span class="s1">'FIN'</span>,<span class="s1">'MDE'</span>,<span class="s1">'SAS'</span>,<span class="s1">'CAH'</span><span class="o">}</span><span class="w"> </span><span class="p">;</span>
<span class="k">do</span>

<span class="c1">#Make a folder for each ancestry</span>
mkdir<span class="w"> </span>~/workspace/ws_files/TMEM175/TMEM175_<span class="s2">"</span><span class="nv">$ancestry</span><span class="s2">"</span>

<span class="k">done</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=a4b8934f-368c-4876-b461-0b20e7814b15">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Create a covariate file with GP2 data</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=805da8a5-55d5-45b0-a15e-a97bb8b60049">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[4]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">CLINICAL_DATA_PATH</span> <span class="o">=</span> <span class="n">pathlib</span><span class="o">.</span><span class="n">Path</span><span class="p">(</span><span class="n">REL7_PATH</span><span class="p">,</span> <span class="s1">'clinical_data/master_key_release7_final_vwb.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=6f7224e2-9a40-4b68-a648-a78ce85ecbc3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Let's load the master key</span>
<span class="n">key</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">CLINICAL_DATA_PATH</span><span class="p">,</span> <span class="n">low_memory</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">key</span><span class="o">.</span><span class="n">shape</span><span class="p">)</span>
<span class="n">key</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>(58446, 26)
</pre>
</div>
</div>
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[8]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>GP2ID</th>
<th>GP2sampleID</th>
<th>study</th>
<th>study_arm</th>
<th>study_type</th>
<th>diagnosis</th>
<th>baseline_GP2_phenotype_for_qc</th>
<th>baseline_GP2_phenotype</th>
<th>biological_sex_for_qc</th>
<th>age_at_sample_collection</th>
<th>age_of_onset</th>
<th>age_at_diagnosis</th>
<th>age_at_death</th>
<th>age_at_last_follow_up</th>
<th>race_for_qc</th>
<th>family_history_for_qc</th>
<th>region_for_qc</th>
<th>manifest_id</th>
<th>Genotyping_site</th>
<th>sample_type</th>
<th>GDPR</th>
<th>monogenic</th>
<th>label</th>
<th>pruned</th>
<th>pruned_reason</th>
<th>related</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000193</td>
<td>APGS_000193_s1</td>
<td>APGS</td>
<td>PD</td>
<td>Case(/Control)</td>
<td>PD</td>
<td>PD</td>
<td>PD</td>
<td>Male</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>Not Reported</td>
<td>Not Reported</td>
<td>AUS</td>
<td>m1</td>
<td>NIH</td>
<td>Saliva</td>
<td>0</td>
<td>0</td>
<td>EUR</td>
<td>0</td>
<td>NaN</td>
<td>0</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000194</td>
<td>APGS_000194_s1</td>
<td>APGS</td>
<td>PD</td>
<td>Case(/Control)</td>
<td>PD</td>
<td>PD</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>Not Reported</td>
<td>Not Reported</td>
<td>AUS</td>
<td>m1</td>
<td>NIH</td>
<td>Saliva</td>
<td>0</td>
<td>0</td>
<td>EUR</td>
<td>0</td>
<td>NaN</td>
<td>0</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000195</td>
<td>APGS_000195_s1</td>
<td>APGS</td>
<td>PD</td>
<td>Case(/Control)</td>
<td>PD</td>
<td>PD</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>Not Reported</td>
<td>Not Reported</td>
<td>AUS</td>
<td>m1</td>
<td>NIH</td>
<td>Saliva</td>
<td>0</td>
<td>0</td>
<td>EUR</td>
<td>0</td>
<td>NaN</td>
<td>0</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000196</td>
<td>APGS_000196_s1</td>
<td>APGS</td>
<td>PD</td>
<td>Case(/Control)</td>
<td>PD</td>
<td>PD</td>
<td>PD</td>
<td>Male</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>Not Reported</td>
<td>Not Reported</td>
<td>AUS</td>
<td>m1</td>
<td>NIH</td>
<td>Saliva</td>
<td>0</td>
<td>0</td>
<td>EUR</td>
<td>0</td>
<td>NaN</td>
<td>0</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000197</td>
<td>APGS_000197_s1</td>
<td>APGS</td>
<td>PD</td>
<td>Case(/Control)</td>
<td>PD</td>
<td>PD</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>Not Reported</td>
<td>Not Reported</td>
<td>AUS</td>
<td>m1</td>
<td>NIH</td>
<td>Saliva</td>
<td>0</td>
<td>0</td>
<td>EUR</td>
<td>0</td>
<td>NaN</td>
<td>0</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e81204c4-0636-4cb9-b2d4-417d8c040263">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Subsetting to keep only a few columns </span>
<span class="n">key</span> <span class="o">=</span> <span class="n">key</span><span class="p">[[</span><span class="s1">'GP2sampleID'</span><span class="p">,</span> <span class="s1">'baseline_GP2_phenotype_for_qc'</span><span class="p">,</span> <span class="s1">'biological_sex_for_qc'</span><span class="p">,</span> <span class="s1">'age_at_sample_collection'</span><span class="p">,</span> <span class="s1">'age_of_onset'</span><span class="p">,</span> <span class="s1">'label'</span><span class="p">]]</span>
<span class="c1"># Renaming the columns</span>
<span class="n">key</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'GP2sampleID'</span><span class="p">:</span><span class="s1">'IID'</span><span class="p">,</span>
                                     <span class="s1">'baseline_GP2_phenotype_for_qc'</span><span class="p">:</span><span class="s1">'phenotype'</span><span class="p">,</span>
                                     <span class="s1">'biological_sex_for_qc'</span><span class="p">:</span><span class="s1">'SEX'</span><span class="p">,</span> 
                                     <span class="s1">'age_at_sample_collection'</span><span class="p">:</span><span class="s1">'AGE'</span><span class="p">,</span> 
                                     <span class="s1">'age_of_onset'</span><span class="p">:</span><span class="s1">'AAO'</span><span class="p">},</span> <span class="n">inplace</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span>
<span class="n">key</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[9]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>phenotype</th>
<th>SEX</th>
<th>AGE</th>
<th>AAO</th>
<th>label</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000193_s1</td>
<td>PD</td>
<td>Male</td>
<td>NaN</td>
<td>NaN</td>
<td>EUR</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000194_s1</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>NaN</td>
<td>EUR</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000195_s1</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>NaN</td>
<td>EUR</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000196_s1</td>
<td>PD</td>
<td>Male</td>
<td>NaN</td>
<td>NaN</td>
<td>EUR</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000197_s1</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>NaN</td>
<td>EUR</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58441</th>
<td>m-UROMETV_000031_s1</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>46.0</td>
<td>EUR</td>
</tr>
<tr>
<th>58442</th>
<td>m-UROMETV_000032_s1</td>
<td>PD</td>
<td>Male</td>
<td>NaN</td>
<td>41.0</td>
<td>EUR</td>
</tr>
<tr>
<th>58443</th>
<td>m-UROMETV_000033_s1</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>38.0</td>
<td>EUR</td>
</tr>
<tr>
<th>58444</th>
<td>m-UROMETV_000034_s1</td>
<td>PD</td>
<td>Male</td>
<td>NaN</td>
<td>50.0</td>
<td>EUR</td>
</tr>
<tr>
<th>58445</th>
<td>m-UROMETV_000035_s1</td>
<td>PD</td>
<td>Female</td>
<td>NaN</td>
<td>46.0</td>
<td>EUR</td>
</tr>
</tbody>
</table>
<p>58446 rows  6 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a515ecb9-cb3d-4904-a581-b9e0e1857d36">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>pwd
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>/home/jupyter/workspace/ws_files
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=d9fc4b4b-3517-43a9-b806-2193a5d2b082">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">ancestries</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'AAC'</span><span class="p">,</span><span class="s1">'AFR'</span><span class="p">,</span><span class="s1">'AJ'</span><span class="p">,</span><span class="s1">'AMR'</span><span class="p">,</span><span class="s1">'CAS'</span><span class="p">,</span><span class="s1">'EAS'</span><span class="p">,</span><span class="s1">'EUR'</span><span class="p">,</span><span class="s1">'FIN'</span><span class="p">,</span><span class="s1">'MDE'</span><span class="p">,</span><span class="s1">'SAS'</span><span class="p">,</span><span class="s1">'CAH'</span><span class="p">}</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'WORKING ON: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">## Subset to keep ancestry of interest </span>
    <span class="n">ancestry_key</span> <span class="o">=</span> <span class="n">key</span><span class="p">[</span><span class="n">key</span><span class="p">[</span><span class="s1">'label'</span><span class="p">]</span><span class="o">==</span><span class="n">ancestry</span><span class="p">]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="n">ancestry_key</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
     <span class="c1"># Load information about related individuals in the ancestry analyzed</span>
    <span class="n">related_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">REL7_PATH</span><span class="si">}</span><span class="s1">/meta_data/related_samples/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">_release7_vwb.related'</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Related individuals: </span><span class="si">{</span><span class="n">related_df</span><span class="o">.</span><span class="n">shape</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1"># Make a list of just one set of related people</span>
    <span class="n">related_list</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">related_df</span><span class="p">[</span><span class="s1">'IID1'</span><span class="p">])</span>
    
    <span class="c1"># Check value counts of related and remove only one related individual</span>
    <span class="n">ancestry_key</span> <span class="o">=</span> <span class="n">ancestry_key</span><span class="p">[</span><span class="o">~</span><span class="n">ancestry_key</span><span class="p">[</span><span class="s2">"IID"</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">related_list</span><span class="p">)]</span>
    
    <span class="c1"># Check size</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Unrelated individuals: </span><span class="si">{</span><span class="n">ancestry_key</span><span class="o">.</span><span class="n">shape</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1"># Convert phenotype to binary (1/2)</span>
    <span class="c1">## Assign conditions so case=2 and controls=1, and -9 otherwise (matching PLINK convention)</span>
    <span class="c1"># PD = 2; control = 1</span>
    <span class="n">pheno_mapping</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"PD"</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span> <span class="s2">"Control"</span><span class="p">:</span> <span class="mi">1</span><span class="p">}</span>
    <span class="n">ancestry_key</span><span class="p">[</span><span class="s1">'PHENO'</span><span class="p">]</span> <span class="o">=</span> <span class="n">ancestry_key</span><span class="p">[</span><span class="s1">'phenotype'</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">pheno_mapping</span><span class="p">)</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="s1">'Int64'</span><span class="p">)</span>
    
    <span class="c1"># Check value counts of pheno</span>
    <span class="n">ancestry_key</span><span class="p">[</span><span class="s1">'PHENO'</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    
    <span class="c1">## Get the PCs</span>
    <span class="n">pcs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">REL7_PATH</span><span class="si">}</span><span class="s1">/meta_data/qc_metrics/projected_pcs_vwb.csv'</span><span class="p">)</span>
    
     <span class="c1">#Select just first 5 PCs</span>
    <span class="n">selected_columns</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'IID'</span><span class="p">,</span> <span class="s1">'PC1'</span><span class="p">,</span> <span class="s1">'PC2'</span><span class="p">,</span> <span class="s1">'PC3'</span><span class="p">,</span> <span class="s1">'PC4'</span><span class="p">,</span> <span class="s1">'PC5'</span><span class="p">]</span>
    <span class="n">pcs</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">pcs</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">1</span><span class="p">:</span><span class="mi">7</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="n">selected_columns</span><span class="p">)</span>
    
     <span class="c1"># Drop the first row (since it's now the column names)</span>
    <span class="n">pcs</span> <span class="o">=</span> <span class="n">pcs</span><span class="o">.</span><span class="n">drop</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>
    
    <span class="c1"># Reset the index to remove any potential issues</span>
    <span class="n">pcs</span> <span class="o">=</span> <span class="n">pcs</span><span class="o">.</span><span class="n">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="c1"># Check size</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'PCs: </span><span class="si">{</span><span class="n">pcs</span><span class="o">.</span><span class="n">shape</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1"># Check value counts of SEX</span>
    <span class="n">sex_og_values</span> <span class="o">=</span> <span class="n">ancestry_key</span><span class="p">[</span><span class="s1">'SEX'</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Sex value counts - original:</span><span class="se">\n</span><span class="s1"> </span><span class="si">{</span><span class="n">sex_og_values</span><span class="o">.</span><span class="n">to_string</span><span class="p">()</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
     <span class="c1"># Convert sex to binary (1/2)</span>
    <span class="c1">## Assign conditions so female=2 and men=1, and -9 otherwise (matching PLINK convention)</span>
    <span class="c1"># Female = 2; Male = 1</span>
    <span class="n">sex_mapping</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"Female"</span><span class="p">:</span> <span class="mi">2</span><span class="p">,</span> <span class="s2">"Male"</span><span class="p">:</span> <span class="mi">1</span><span class="p">}</span>
    <span class="n">ancestry_key</span><span class="p">[</span><span class="s1">'SEX'</span><span class="p">]</span> <span class="o">=</span> <span class="n">ancestry_key</span><span class="p">[</span><span class="s1">'SEX'</span><span class="p">]</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="n">sex_mapping</span><span class="p">)</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="s1">'Int64'</span><span class="p">)</span>
    
    <span class="c1"># Check value counts of SEX after recoding</span>
    <span class="n">sex_recode_values</span> <span class="o">=</span> <span class="n">ancestry_key</span><span class="p">[</span><span class="s1">'SEX'</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">(</span><span class="n">dropna</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Sex value counts - recoded:</span><span class="se">\n</span><span class="si">{</span><span class="n">sex_recode_values</span><span class="o">.</span><span class="n">to_string</span><span class="p">()</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">## Make covariate file</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">pcs</span><span class="p">,</span> <span class="n">ancestry_key</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s1">'IID'</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="s1">'left'</span><span class="p">)</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Check columns for covariate file: </span><span class="si">{</span><span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">#Make additional columns - FID, fatid and matid - these are needed for RVtests!!</span>
    <span class="c1">#RVtests needs the first 5 columns to be fid, iid, fatid, matid and sex otherwise it does not run correctly</span>
    <span class="c1">#Uppercase column name is ok</span>
    <span class="c1">#See https://zhanxw.github.io/rvtests/#phenotype-file</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">'FID'</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">'FATID'</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">df</span><span class="p">[</span><span class="s1">'MATID'</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>
    
    <span class="c1">## Clean up and keep columns we need </span>
    <span class="n">final_df</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">'FID'</span><span class="p">,</span><span class="s1">'IID'</span><span class="p">,</span> <span class="s1">'FATID'</span><span class="p">,</span> <span class="s1">'MATID'</span><span class="p">,</span> <span class="s1">'SEX'</span><span class="p">,</span> <span class="s1">'AGE'</span><span class="p">,</span> <span class="s1">'PHENO'</span><span class="p">,</span><span class="s1">'PC1'</span><span class="p">,</span> <span class="s1">'PC2'</span><span class="p">,</span> <span class="s1">'PC3'</span><span class="p">,</span> <span class="s1">'PC4'</span><span class="p">,</span> <span class="s1">'PC5'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    
    <span class="c1">##DO NOT replace missing values with -9 as this is misinterpreted by RVtests - needs to be nonnumeric</span>
    <span class="c1">#Leave missing values as NA</span>
    
    <span class="c1">#Check number of PD cases missing age</span>
    <span class="n">pd_missAge</span> <span class="o">=</span> <span class="n">final_df</span><span class="p">[(</span><span class="n">final_df</span><span class="p">[</span><span class="s1">'PHENO'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">final_df</span><span class="p">[</span><span class="s1">'AGE'</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">())]</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Number of PD cases missing age: </span><span class="si">{</span><span class="n">pd_missAge</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">#Check number of controls missing age</span>
    <span class="n">control_missAge</span> <span class="o">=</span> <span class="n">final_df</span><span class="p">[(</span><span class="n">final_df</span><span class="p">[</span><span class="s1">'PHENO'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span><span class="o">&amp;</span><span class="p">(</span><span class="n">final_df</span><span class="p">[</span><span class="s1">'AGE'</span><span class="p">]</span><span class="o">.</span><span class="n">isna</span><span class="p">())]</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Number of controls missing age: </span><span class="si">{</span><span class="n">control_missAge</span><span class="o">.</span><span class="n">shape</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">## Make file of sample IDs to keep </span>
    <span class="n">samples_toKeep</span> <span class="o">=</span> <span class="n">final_df</span><span class="p">[[</span><span class="s1">'FID'</span><span class="p">,</span> <span class="s1">'IID'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="n">samples_toKeep</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">.samplestoKeep'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span>
    
    <span class="c1">## Make your covariate file</span>
    <span class="c1">#Included na_rep to write out missing/NA values explicitly as string/text, not as blank otherwise they are misread in RVtests</span>
    <span class="n">final_df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">,</span> <span class="n">na_rep</span><span class="o">=</span><span class="s1">'NA'</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: AFR
Related individuals: (228, 9)
Unrelated individuals: (2752, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          1553
Female                        1197
Other/Unknown/Not Reported       2
Sex value counts - recoded:
SEX
1       1553
2       1197
&lt;NA&gt;       2
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 835
Number of controls missing age: 809
WORKING ON: AAC
Related individuals: (28, 9)
Unrelated individuals: (1181, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Female                        690
Male                          489
Other/Unknown/Not Reported      2
Sex value counts - recoded:
SEX
2       690
1       489
&lt;NA&gt;      2
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 91
Number of controls missing age: 96
WORKING ON: FIN
Related individuals: (7, 9)
Unrelated individuals: (117, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Female    63
Male      54
Sex value counts - recoded:
SEX
2    63
1    54
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 62
Number of controls missing age: 3
WORKING ON: AMR
Related individuals: (51, 9)
Unrelated individuals: (672, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          374
Female                        296
Other/Unknown/Not Reported      2
Sex value counts - recoded:
SEX
1       374
2       296
&lt;NA&gt;      2
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 106
Number of controls missing age: 55
WORKING ON: EAS
Related individuals: (350, 9)
Unrelated individuals: (5303, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          3398
Female                        1899
Other/Unknown/Not Reported       6
Sex value counts - recoded:
SEX
1       3398
2       1899
&lt;NA&gt;       6
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 1349
Number of controls missing age: 1268
WORKING ON: CAH
Related individuals: (65, 9)
Unrelated individuals: (1108, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          585
Female                        498
Other/Unknown/Not Reported     25
Sex value counts - recoded:
SEX
1       585
2       498
&lt;NA&gt;     25
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 136
Number of controls missing age: 52
WORKING ON: CAS
Related individuals: (137, 9)
Unrelated individuals: (933, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Female    483
Male      450
Sex value counts - recoded:
SEX
2    483
1    450
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 111
Number of controls missing age: 88
WORKING ON: SAS
Related individuals: (55, 9)
Unrelated individuals: (624, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          397
Female                        226
Other/Unknown/Not Reported      1
Sex value counts - recoded:
SEX
1       397
2       226
&lt;NA&gt;      1
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 203
Number of controls missing age: 83
WORKING ON: EUR
Related individuals: (3662, 9)
Unrelated individuals: (39710, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          23733
Female                        15784
Other/Unknown/Not Reported      193
Sex value counts - recoded:
SEX
1       23733
2       15784
&lt;NA&gt;      193
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 5947
Number of controls missing age: 3347
WORKING ON: AJ
Related individuals: (245, 9)
Unrelated individuals: (2675, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          1655
Female                        1009
Other/Unknown/Not Reported      11
Sex value counts - recoded:
SEX
1       1655
2       1009
&lt;NA&gt;      11
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 95
Number of controls missing age: 195
WORKING ON: MDE
Related individuals: (19, 9)
Unrelated individuals: (589, 6)
PCs: (58209, 6)
Sex value counts - original:
 SEX
Male                          368
Female                        219
Other/Unknown/Not Reported      2
Sex value counts - recoded:
SEX
1       368
2       219
&lt;NA&gt;      2
Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',
       'AAO', 'label', 'PHENO'],
      dtype='object')
Number of PD cases missing age: 140
Number of controls missing age: 12
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=e23131f5-8bd6-482d-85c9-008c2aff08a7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Annotation of the gene</p>
<p>Extract the region using PLINK</p>
<p>Extract TMEM175 gene</p>
<p>TMEM175 coordinates: Chromosome 4:932387-4:958656(GRCh38/hg38)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=45bdfb86-1535-40b2-8d8d-2255c06a3d10">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[14]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">## extract region using plink</span>
<span class="n">ancestries</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'AAC'</span><span class="p">,</span><span class="s1">'AFR'</span><span class="p">,</span><span class="s1">'AJ'</span><span class="p">,</span><span class="s1">'AMR'</span><span class="p">,</span><span class="s1">'CAS'</span><span class="p">,</span><span class="s1">'EAS'</span><span class="p">,</span><span class="s1">'EUR'</span><span class="p">,</span><span class="s1">'FIN'</span><span class="p">,</span><span class="s1">'MDE'</span><span class="p">,</span><span class="s1">'SAS'</span><span class="p">,</span><span class="s1">'CAH'</span><span class="p">}</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>

    <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
    <span class="o">--</span><span class="n">pfile</span> <span class="p">{</span><span class="n">REL7_PATH</span><span class="p">}</span><span class="o">/</span><span class="n">imputed_genotypes</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">/</span><span class="n">chr4_</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_release7_vwb</span> \
    <span class="o">--</span><span class="nb">chr</span> <span class="mi">4</span> \
    <span class="o">--</span><span class="n">from</span><span class="o">-</span><span class="n">bp</span> <span class="mi">882387</span> \
    <span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">bp</span> <span class="mi">1008656</span> \
    <span class="o">--</span><span class="n">make</span><span class="o">-</span><span class="n">bed</span> \
    <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:54:42 2024
52223 MiB RAM detected, ~50711 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2643 samples (1163 females, 1480 males; 2643 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.psam.
4555051 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.pvar.
1 binary phenotype loaded (942 cases, 1679 controls).
3853 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed
... done.
End time: Wed Jul 31 09:54:46 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:54:46 2024
52223 MiB RAM detected, ~50654 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1111 samples (656 females, 455 males; 1111 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.psam.
3986699 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.pvar.
1 binary phenotype loaded (285 cases, 801 controls).
3449 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed
... done.
End time: Wed Jul 31 09:54:49 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:54:50 2024
52223 MiB RAM detected, ~50648 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
114 samples (61 females, 53 males; 114 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.psam.
891614 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.pvar.
1 binary phenotype loaded (98 cases, 8 controls).
605 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed
... done.
End time: Wed Jul 31 09:54:51 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:54:51 2024
52223 MiB RAM detected, ~50640 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
646 samples (283 females, 363 males; 646 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.psam.
2407746 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.pvar.
1 binary phenotype loaded (458 cases, 155 controls).
1987 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed
... done.
End time: Wed Jul 31 09:54:53 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:54:54 2024
52223 MiB RAM detected, ~50633 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
5167 samples (1889 females, 3278 males; 5167 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.psam.
3235052 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.pvar.
1 binary phenotype loaded (2662 cases, 2461 controls).
3171 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed
... done.
End time: Wed Jul 31 09:54:57 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:54:57 2024
52223 MiB RAM detected, ~50610 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
851 samples (405 females, 446 males; 851 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.psam.
3678288 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.pvar.
1 binary phenotype loaded (525 cases, 302 controls).
3090 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed
... done.
End time: Wed Jul 31 09:55:00 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:55:00 2024
52223 MiB RAM detected, ~50610 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
903 samples (464 females, 439 males; 903 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.psam.
2200655 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.pvar.
1 binary phenotype loaded (552 cases, 343 controls).
1955 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed
... done.
End time: Wed Jul 31 09:55:03 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:55:03 2024
52223 MiB RAM detected, ~50566 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
635 samples (227 females, 408 males; 635 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.psam.
1933318 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.pvar.
1 binary phenotype loaded (387 cases, 222 controls).
1728 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed
... done.
End time: Wed Jul 31 09:55:05 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:55:05 2024
52223 MiB RAM detected, ~50591 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
38839 samples (15543 females, 23296 males; 38839 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.psam.
10419161 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.pvar.
1 binary phenotype loaded (21198 cases, 9214 controls).
9504 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed
... done.
End time: Wed Jul 31 09:55:14 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:55:14 2024
52223 MiB RAM detected, ~50585 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2655 samples (1007 females, 1648 males; 2655 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.psam.
1669760 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.pvar.
1 binary phenotype loaded (1292 cases, 411 controls).
1285 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam ...
done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim ...
done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed ...
done.
End time: Wed Jul 31 09:55:16 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb
  --to-bp 1008656

Start time: Wed Jul 31 09:55:17 2024
52223 MiB RAM detected, ~50594 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
581 samples (212 females, 369 males; 581 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.psam.
2161336 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.pvar.
1 binary phenotype loaded (311 cases, 225 controls).
1945 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed
... done.
End time: Wed Jul 31 09:55:19 2024
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=65d2818f-be1b-4fbd-993f-1741cfe3a5b9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[17]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Visualize bim file</span>
<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s1">'/home/jupyter/workspace/ws_files/TMEM175/'</span>
<span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.bim
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>4	chr4:882390:C:A	0	882390	A	C
4	chr4:882397:C:T	0	882397	T	C
4	chr4:882422:C:T	0	882422	T	C
4	chr4:882432:G:A	0	882432	A	G
4	chr4:882454:C:T	0	882454	T	C
4	chr4:882471:G:A	0	882471	A	G
4	chr4:882483:C:G	0	882483	G	C
4	chr4:882498:C:G	0	882498	G	C
4	chr4:882507:G:A	0	882507	A	G
4	chr4:882517:C:A	0	882517	A	C
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a75b27d6-b85d-4678-aa3b-336aad9b9f88">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[18]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Visualize bim file</span>
<span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.fam
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>0	APGS_000002_s1	0	0	1	2
0	APGS_000003_s1	0	0	1	2
0	APGS_000004_s1	0	0	2	2
0	APGS_000008_s1	0	0	1	2
0	APGS_000011_s1	0	0	1	2
0	APGS_000016_s1	0	0	1	2
0	APGS_000018_s1	0	0	1	2
0	APGS_000023_s1	0	0	1	2
0	APGS_000026_s1	0	0	2	2
0	APGS_000027_s1	0	0	1	2
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=c5e0bdd8-d997-4932-8c48-e65a34338186">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[21]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">/'</span>
    
    <span class="o">!</span><span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/<span class="o">{</span>ancestry<span class="o">}</span>_TMEM175.fam<span class="w"> </span>&gt;<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/<span class="o">{</span>ancestry<span class="o">}</span>_s1.txt
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=02fda682-a3db-41ee-a76b-e7eb853ce4b5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[22]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>head<span class="w"> </span>/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_s1.txt
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>0	APGS_000002_s1	0	0	1	2
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=305bd2da-933b-4b45-862c-8997ee82ffb2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Turn binary files into VCF</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=03654fe1-3d77-4a0e-8ddf-f8012d0f664c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[23]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    
    <span class="c1">## Turn binary files into VCF</span>
    <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
    <span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span> \
    <span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_s1</span><span class="o">.</span><span class="n">txt</span> \
    <span class="o">--</span><span class="n">make</span><span class="o">-</span><span class="n">bed</span> \
    <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175_v1</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1

Start time: Wed Jul 31 10:13:40 2024
52223 MiB RAM detected, ~50482 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2643 samples (1163 females, 1480 males; 2643 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam.
3853 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim.
1 binary phenotype loaded (942 cases, 1679 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:40 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1

Start time: Wed Jul 31 10:13:40 2024
52223 MiB RAM detected, ~50465 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1111 samples (656 females, 455 males; 1111 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam.
3449 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim.
1 binary phenotype loaded (285 cases, 801 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:41 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1

Start time: Wed Jul 31 10:13:41 2024
52223 MiB RAM detected, ~50489 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
114 samples (61 females, 53 males; 114 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam.
605 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim.
1 binary phenotype loaded (98 cases, 8 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:42 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1

Start time: Wed Jul 31 10:13:42 2024
52223 MiB RAM detected, ~50502 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
646 samples (283 females, 363 males; 646 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam.
1987 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim.
1 binary phenotype loaded (458 cases, 155 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:42 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1

Start time: Wed Jul 31 10:13:43 2024
52223 MiB RAM detected, ~50501 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
5167 samples (1889 females, 3278 males; 5167 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam.
3171 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim.
1 binary phenotype loaded (2662 cases, 2461 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:43 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1

Start time: Wed Jul 31 10:13:43 2024
52223 MiB RAM detected, ~50501 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
851 samples (405 females, 446 males; 851 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam.
3090 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim.
1 binary phenotype loaded (525 cases, 302 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:44 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1

Start time: Wed Jul 31 10:13:44 2024
52223 MiB RAM detected, ~50501 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
903 samples (464 females, 439 males; 903 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam.
1955 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim.
1 binary phenotype loaded (552 cases, 343 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:45 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1

Start time: Wed Jul 31 10:13:45 2024
52223 MiB RAM detected, ~50496 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
635 samples (227 females, 408 males; 635 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam.
1728 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim.
1 binary phenotype loaded (387 cases, 222 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:45 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1

Start time: Wed Jul 31 10:13:45 2024
52223 MiB RAM detected, ~50502 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
38839 samples (15543 females, 23296 males; 38839 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam.
9504 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim.
1 binary phenotype loaded (21198 cases, 9214 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:46 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1

Start time: Wed Jul 31 10:13:46 2024
52223 MiB RAM detected, ~50502 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2655 samples (1007 females, 1648 males; 2655 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam.
1285 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim.
1 binary phenotype loaded (1292 cases, 411 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:47 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_s1.txt
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1

Start time: Wed Jul 31 10:13:47 2024
52223 MiB RAM detected, ~50497 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
581 samples (212 females, 369 males; 581 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam.
1945 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim.
1 binary phenotype loaded (311 cases, 225 controls).
--keep: 1 sample remaining.
1 sample (0 females, 1 male; 1 founder) remaining after main filters.
1 case and 0 controls remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.bed
... done.
End time: Wed Jul 31 10:13:48 2024
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a988e523-8aad-44dd-b516-1803ba26c42b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[24]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    
    <span class="c1">## Turn binary files into VCF</span>
    <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
    <span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175_v1</span> \
    <span class="o">--</span><span class="n">recode</span> <span class="n">vcf</span><span class="o">-</span><span class="n">fid</span> \
    <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175_v1</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1

Start time: Wed Jul 31 10:15:08 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50568 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.fam.
3853 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.vcf ...
1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:09 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1

Start time: Wed Jul 31 10:15:09 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50561 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.fam.
3449 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.vcf ...
1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:09 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1

Start time: Wed Jul 31 10:15:10 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50553 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.fam.
605 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.vcf ...
10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:10 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1

Start time: Wed Jul 31 10:15:10 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50571 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.fam.
1987 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.vcf ...
1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:10 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1

Start time: Wed Jul 31 10:15:11 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50572 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.fam.
3171 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.vcf ...
1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:11 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1

Start time: Wed Jul 31 10:15:11 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50555 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.fam.
3090 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.vcf ...
101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:12 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1

Start time: Wed Jul 31 10:15:12 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50555 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.fam.
1955 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.vcf ...
10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:12 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1

Start time: Wed Jul 31 10:15:13 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50555 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.fam.
1728 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.vcf ...
1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:13 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1

Start time: Wed Jul 31 10:15:13 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50556 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.fam.
9504 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.vcf ...
1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:13 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1

Start time: Wed Jul 31 10:15:14 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50557 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.fam.
1285 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.vcf ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:14 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1
  --export vcf-fid
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1

Start time: Wed Jul 31 10:15:14 2024
Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.
52223 MiB RAM detected, ~50557 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1 sample (0 females, 1 male; 1 founder) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.fam.
1945 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.bim.
1 binary phenotype loaded (1 case, 0 controls).
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.vcf ...
10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 10:15:14 2024
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=1d0f3130-fefc-40d0-b3ea-ca92fccd12a9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[25]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">### Bgzip and Tabix (zip and index the file)</span>
<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    <span class="o">!</span><span class="w"> </span>bgzip<span class="w"> </span>-f<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/<span class="o">{</span>ancestry<span class="o">}</span>_TMEM175_v1.vcf
    <span class="o">!</span><span class="w"> </span>tabix<span class="w"> </span>-f<span class="w"> </span>-p<span class="w"> </span>vcf<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/<span class="o">{</span>ancestry<span class="o">}</span>_TMEM175_v1.vcf.gz<span class="w"> </span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=5b70ddab-a941-4024-a640-37d6bc978640">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Annotate using ANNOVAR</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=95cc7254-f11d-4798-aab0-e1c72cc8ce35">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[26]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">## annotate using ANNOVAR</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    
    <span class="o">!</span><span class="w"> </span>perl<span class="w"> </span>/home/jupyter/tools/annovar/table_annovar.pl<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/<span class="o">{</span>ancestry<span class="o">}</span>_TMEM175_v1.vcf.gz<span class="w"> </span>/home/jupyter/tools/annovar/humandb/<span class="w"> </span>-buildver<span class="w"> </span>hg38<span class="w"> </span><span class="err">\</span>
    <span class="o">-</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span><span class="o">.</span><span class="n">annovar</span> \
    <span class="o">-</span><span class="n">remove</span> <span class="o">-</span><span class="n">protocol</span> <span class="n">refGene</span><span class="p">,</span><span class="n">clinvar_20140902</span> \
    <span class="o">-</span><span class="n">operation</span> <span class="n">g</span><span class="p">,</span><span class="n">f</span> \
    <span class="o">--</span><span class="n">nopolish</span> \
    <span class="o">-</span><span class="n">nastring</span> <span class="o">.</span> \
    <span class="o">-</span><span class="n">vcfinput</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 3860 lines from VCF file
NOTICE: A total of 3853 locus in VCF file passed QC threshold, representing 3628 SNPs (2760 transitions and 868 transversions) and 225 indels/substitutions
NOTICE: Finished writing allele frequencies based on 3628 SNP genotypes (2760 transitions and 868 transversions) and 225 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 3853 unique variants in 3853 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 3853 unique variants in 3853 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 3456 lines from VCF file
NOTICE: A total of 3449 locus in VCF file passed QC threshold, representing 3247 SNPs (2444 transitions and 803 transversions) and 202 indels/substitutions
NOTICE: Finished writing allele frequencies based on 3247 SNP genotypes (2444 transitions and 803 transversions) and 202 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 3449 unique variants in 3449 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 3449 unique variants in 3449 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 612 lines from VCF file
NOTICE: A total of 605 locus in VCF file passed QC threshold, representing 573 SNPs (427 transitions and 146 transversions) and 32 indels/substitutions
NOTICE: Finished writing allele frequencies based on 573 SNP genotypes (427 transitions and 146 transversions) and 32 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 605 unique variants in 605 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 605 unique variants in 605 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.vcf

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 1994 lines from VCF file
NOTICE: A total of 1987 locus in VCF file passed QC threshold, representing 1870 SNPs (1403 transitions and 467 transversions) and 117 indels/substitutions
NOTICE: Finished writing allele frequencies based on 1870 SNP genotypes (1403 transitions and 467 transversions) and 117 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 1987 unique variants in 1987 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 1987 unique variants in 1987 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 3178 lines from VCF file
NOTICE: A total of 3171 locus in VCF file passed QC threshold, representing 3001 SNPs (2330 transitions and 671 transversions) and 170 indels/substitutions
NOTICE: Finished writing allele frequencies based on 3001 SNP genotypes (2330 transitions and 671 transversions) and 170 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 3171 unique variants in 3171 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 3171 unique variants in 3171 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 3097 lines from VCF file
NOTICE: A total of 3090 locus in VCF file passed QC threshold, representing 2917 SNPs (2209 transitions and 708 transversions) and 173 indels/substitutions
NOTICE: Finished writing allele frequencies based on 2917 SNP genotypes (2209 transitions and 708 transversions) and 173 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 3090 unique variants in 3090 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 3090 unique variants in 3090 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 1962 lines from VCF file
NOTICE: A total of 1955 locus in VCF file passed QC threshold, representing 1848 SNPs (1430 transitions and 418 transversions) and 107 indels/substitutions
NOTICE: Finished writing allele frequencies based on 1848 SNP genotypes (1430 transitions and 418 transversions) and 107 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 1955 unique variants in 1955 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 1955 unique variants in 1955 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 1735 lines from VCF file
NOTICE: A total of 1728 locus in VCF file passed QC threshold, representing 1628 SNPs (1231 transitions and 397 transversions) and 100 indels/substitutions
NOTICE: Finished writing allele frequencies based on 1628 SNP genotypes (1231 transitions and 397 transversions) and 100 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 1728 unique variants in 1728 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 1728 unique variants in 1728 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 9511 lines from VCF file
NOTICE: A total of 9504 locus in VCF file passed QC threshold, representing 9001 SNPs (6707 transitions and 2294 transversions) and 503 indels/substitutions
NOTICE: Finished writing allele frequencies based on 9001 SNP genotypes (6707 transitions and 2294 transversions) and 503 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 9504 unique variants in 9504 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 9504 unique variants in 9504 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 1292 lines from VCF file
NOTICE: A total of 1285 locus in VCF file passed QC threshold, representing 1212 SNPs (912 transitions and 300 transversions) and 73 indels/substitutions
NOTICE: Finished writing allele frequencies based on 1212 SNP genotypes (912 transitions and 300 transversions) and 73 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 1285 unique variants in 1285 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 1285 unique variants in 1285 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.vcf

NOTICE: Running with system command &lt;convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.vcf.gz &gt; /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput&gt;
NOTICE: Finished reading 1952 lines from VCF file
NOTICE: A total of 1945 locus in VCF file passed QC threshold, representing 1842 SNPs (1404 transitions and 438 transversions) and 103 indels/substitutions
NOTICE: Finished writing allele frequencies based on 1842 SNP genotypes (1404 transitions and 438 transversions) and 103 indels/substitutions for 1 samples

NOTICE: Running with system command &lt;/home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo&gt;
-----------------------------------------------------------------
NOTICE: Processing operation=g protocol=refGene

NOTICE: Running with system command &lt;annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene.exonic_variant_function
NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes
NOTICE: Processing next batch with 1945 unique variants in 1945 input lines
NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences
WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation
-----------------------------------------------------------------
NOTICE: Processing operation=f protocol=clinvar_20140902

NOTICE: Running system command &lt;annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/&gt;
NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format
NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_clinvar_20140902_filtered
NOTICE: Processing next batch with 1945 unique variants in 1945 input lines
NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10
NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done
-----------------------------------------------------------------
NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt
NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt
-----------------------------------------------------------------
NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.vcf

gzip: stdout: Broken pipe
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=3b20d62a-e603-4be0-9a46-a3b5091a1aa0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[29]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882400</td>
<td>882400</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882400</td>
<td>chr4:882400:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882455</td>
<td>882455</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882455</td>
<td>chr4:882455:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882471</td>
<td>882471</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882471</td>
<td>chr4:882471:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882498</td>
<td>882498</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882498</td>
<td>chr4:882498:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>3444</th>
<td>4</td>
<td>1008499</td>
<td>1008499</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3935;dist=3127</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008499</td>
<td>chr4:1008499:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3445</th>
<td>4</td>
<td>1008515</td>
<td>1008515</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3951;dist=3111</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008515</td>
<td>chr4:1008515:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3446</th>
<td>4</td>
<td>1008535</td>
<td>1008535</td>
<td>G</td>
<td>C</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3971;dist=3091</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008535</td>
<td>chr4:1008535:G:C</td>
<td>G</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3447</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3448</th>
<td>4</td>
<td>1008654</td>
<td>1008654</td>
<td>G</td>
<td>C</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4090;dist=2972</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008654</td>
<td>chr4:1008654:G:C</td>
<td>G</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>3449 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=183f61f5-40f9-48fd-ad39-f655d33915c3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[30]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[30]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           2746
exonic              287
intergenic          175
UTR3                102
downstream           58
UTR5                 42
upstream             36
splicing              2
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=022b5a71-dc71-4585-a9df-f198923ad6c3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[31]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[31]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   287
Start                 287
End                   287
Ref                   287
Alt                   287
Func.refGene          287
Gene.refGene          287
GeneDetail.refGene    287
ExonicFunc.refGene    287
AAChange.refGene      287
clinvar_20140902      287
Otherinfo1            287
Otherinfo2            287
Otherinfo3            287
Otherinfo4            287
Otherinfo5            287
Otherinfo6            287
Otherinfo7            287
Otherinfo8            287
Otherinfo9            287
Otherinfo10           287
Otherinfo11           287
Otherinfo12           287
Otherinfo13           287
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ff40e234-de59-492d-9323-c6a91f7b5d0e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[32]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[32]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV         155
synonymous SNV            123
stopgain                    4
nonframeshift deletion      3
frameshift deletion         2
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5bfb8ae0-3f37-430f-8db4-b8f092555733">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[33]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882400</td>
<td>882400</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882400</td>
<td>chr4:882400:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882454</td>
<td>882454</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882454</td>
<td>chr4:882454:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882471</td>
<td>882471</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882471</td>
<td>chr4:882471:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882498</td>
<td>882498</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882498</td>
<td>chr4:882498:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>3848</th>
<td>4</td>
<td>1008499</td>
<td>1008499</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3935;dist=3127</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008499</td>
<td>chr4:1008499:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3849</th>
<td>4</td>
<td>1008524</td>
<td>1008524</td>
<td>G</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3960;dist=3102</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008524</td>
<td>chr4:1008524:G:T</td>
<td>G</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3850</th>
<td>4</td>
<td>1008620</td>
<td>1008620</td>
<td>G</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4056;dist=3006</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008620</td>
<td>chr4:1008620:G:T</td>
<td>G</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3851</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3852</th>
<td>4</td>
<td>1008654</td>
<td>1008654</td>
<td>G</td>
<td>C</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4090;dist=2972</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008654</td>
<td>chr4:1008654:G:C</td>
<td>G</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>3853 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e60f2eb3-68c6-42c9-bc59-8d472fcd9653">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[34]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[34]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           3069
exonic              314
intergenic          203
UTR3                118
downstream           63
UTR5                 45
upstream             40
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=32cdb2bc-755b-4c3c-9ec3-3d4e22bc00bf">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[35]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[35]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   314
Start                 314
End                   314
Ref                   314
Alt                   314
Func.refGene          314
Gene.refGene          314
GeneDetail.refGene    314
ExonicFunc.refGene    314
AAChange.refGene      314
clinvar_20140902      314
Otherinfo1            314
Otherinfo2            314
Otherinfo3            314
Otherinfo4            314
Otherinfo5            314
Otherinfo6            314
Otherinfo7            314
Otherinfo8            314
Otherinfo9            314
Otherinfo10           314
Otherinfo11           314
Otherinfo12           314
Otherinfo13           314
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ad2afccb-7b7a-4e27-98a4-5610366fa5f6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[36]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[36]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV          161
synonymous SNV             143
frameshift deletion          4
nonframeshift deletion       3
stopgain                     2
nonframeshift insertion      1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c96f016a-dcbe-4666-af1c-b9d1fb2c4b9c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[37]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882498</td>
<td>882498</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882498</td>
<td>chr4:882498:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882530</td>
<td>882530</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882611</td>
<td>882611</td>
<td>T</td>
<td>C</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882611</td>
<td>chr4:882611:T:C</td>
<td>T</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882745</td>
<td>882745</td>
<td>G</td>
<td>A</td>
<td>exonic</td>
<td>GAK</td>
<td>.</td>
<td>synonymous SNV</td>
<td>GAK:NM_001318134:exon11:c.C1242T:p.H414H,GAK:N...</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882745</td>
<td>chr4:882745:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882801</td>
<td>882801</td>
<td>C</td>
<td>T</td>
<td>exonic</td>
<td>GAK</td>
<td>.</td>
<td>nonsynonymous SNV</td>
<td>GAK:NM_001318134:exon11:c.G1186A:p.A396T,GAK:N...</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882801</td>
<td>chr4:882801:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1280</th>
<td>4</td>
<td>1008337</td>
<td>1008337</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3773;dist=3289</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>1.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>1/1</td>
</tr>
<tr>
<th>1281</th>
<td>4</td>
<td>1008371</td>
<td>1008371</td>
<td>C</td>
<td>-</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3807;dist=3255</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008370</td>
<td>chr4:1008370:TC:T</td>
<td>TC</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1282</th>
<td>4</td>
<td>1008480</td>
<td>1008480</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3916;dist=3146</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1283</th>
<td>4</td>
<td>1008538</td>
<td>1008538</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3974;dist=3088</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008538</td>
<td>chr4:1008538:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1284</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>1285 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=2b731629-eeb1-4241-b58e-17a6e4e57702">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[38]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[38]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           1024
exonic              125
intergenic           60
UTR3                 43
downstream           20
upstream              6
UTR5                  6
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=d8a69325-86ae-49ef-8bfe-f6d786b23a19">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[39]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[39]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   125
Start                 125
End                   125
Ref                   125
Alt                   125
Func.refGene          125
Gene.refGene          125
GeneDetail.refGene    125
ExonicFunc.refGene    125
AAChange.refGene      125
clinvar_20140902      125
Otherinfo1            125
Otherinfo2            125
Otherinfo3            125
Otherinfo4            125
Otherinfo5            125
Otherinfo6            125
Otherinfo7            125
Otherinfo8            125
Otherinfo9            125
Otherinfo10           125
Otherinfo11           125
Otherinfo12           125
Otherinfo13           125
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1e87e13a-ef0d-491d-a277-85c7ea9d53ef">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[40]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[40]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV         66
synonymous SNV            55
frameshift deletion        2
stopgain                   1
nonframeshift deletion     1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c178e42f-9588-4327-8319-5db30d88fea0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[41]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882397</td>
<td>882397</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882397</td>
<td>chr4:882397:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882400</td>
<td>882400</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882400</td>
<td>chr4:882400:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882454</td>
<td>882454</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882454</td>
<td>chr4:882454:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882498</td>
<td>882498</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882498</td>
<td>chr4:882498:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1982</th>
<td>4</td>
<td>1008395</td>
<td>1008395</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3831;dist=3231</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008395</td>
<td>chr4:1008395:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1983</th>
<td>4</td>
<td>1008480</td>
<td>1008480</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3916;dist=3146</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1984</th>
<td>4</td>
<td>1008515</td>
<td>1008515</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3951;dist=3111</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008515</td>
<td>chr4:1008515:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1985</th>
<td>4</td>
<td>1008571</td>
<td>1008571</td>
<td>G</td>
<td>A</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4007;dist=3055</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008571</td>
<td>chr4:1008571:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1986</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>1987 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=9ae7b44d-44b3-4373-a79c-8ea15c484de3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[42]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[42]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           1596
exonic              178
intergenic          104
UTR3                 56
downstream           23
UTR5                 17
upstream             12
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=3b76abdc-89ed-453d-a65b-e243be9c0a02">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[43]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[43]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   178
Start                 178
End                   178
Ref                   178
Alt                   178
Func.refGene          178
Gene.refGene          178
GeneDetail.refGene    178
ExonicFunc.refGene    178
AAChange.refGene      178
clinvar_20140902      178
Otherinfo1            178
Otherinfo2            178
Otherinfo3            178
Otherinfo4            178
Otherinfo5            178
Otherinfo6            178
Otherinfo7            178
Otherinfo8            178
Otherinfo9            178
Otherinfo10           178
Otherinfo11           178
Otherinfo12           178
Otherinfo13           178
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=62bb1733-90be-48db-9383-33868e742cc3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[44]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[44]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV          89
synonymous SNV             85
frameshift deletion         2
stopgain                    1
nonframeshift insertion     1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=6f2f6d87-5e50-49b8-beb5-c4fcea98b4be">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[45]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882400</td>
<td>882400</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882400</td>
<td>chr4:882400:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882454</td>
<td>882454</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882454</td>
<td>chr4:882454:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882498</td>
<td>882498</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882498</td>
<td>chr4:882498:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882524</td>
<td>882524</td>
<td>A</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882524</td>
<td>chr4:882524:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>3085</th>
<td>4</td>
<td>1008480</td>
<td>1008480</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3916;dist=3146</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3086</th>
<td>4</td>
<td>1008515</td>
<td>1008515</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3951;dist=3111</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008515</td>
<td>chr4:1008515:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3087</th>
<td>4</td>
<td>1008570</td>
<td>1008570</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4006;dist=3056</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008570</td>
<td>chr4:1008570:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3088</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3089</th>
<td>4</td>
<td>1008654</td>
<td>1008654</td>
<td>G</td>
<td>C</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4090;dist=2972</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008654</td>
<td>chr4:1008654:G:C</td>
<td>G</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>3090 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=99ef454c-3694-4aca-826c-6d1554b4b0d9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[46]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[46]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           2467
exonic              260
intergenic          168
UTR3                 87
downstream           46
UTR5                 31
upstream             30
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=202ab877-11f8-4d19-9939-30e0ca0ac0d1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[47]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[47]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   260
Start                 260
End                   260
Ref                   260
Alt                   260
Func.refGene          260
Gene.refGene          260
GeneDetail.refGene    260
ExonicFunc.refGene    260
AAChange.refGene      260
clinvar_20140902      260
Otherinfo1            260
Otherinfo2            260
Otherinfo3            260
Otherinfo4            260
Otherinfo5            260
Otherinfo6            260
Otherinfo7            260
Otherinfo8            260
Otherinfo9            260
Otherinfo10           260
Otherinfo11           260
Otherinfo12           260
Otherinfo13           260
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=bf5af1bd-c7a5-4e09-b7c7-b77ba7154a05">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[48]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[48]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV         139
synonymous SNV            113
stopgain                    4
frameshift deletion         2
nonframeshift deletion      1
frameshift insertion        1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ebcfd55c-1755-474d-920d-60c3b00de695">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[49]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882429</td>
<td>882429</td>
<td>G</td>
<td>C</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882429</td>
<td>chr4:882429:G:C</td>
<td>G</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882468</td>
<td>882468</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882468</td>
<td>chr4:882468:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882498</td>
<td>882498</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882498</td>
<td>chr4:882498:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882501</td>
<td>882501</td>
<td>C</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882501</td>
<td>chr4:882501:C:A</td>
<td>C</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1950</th>
<td>4</td>
<td>1008371</td>
<td>1008371</td>
<td>C</td>
<td>-</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3807;dist=3255</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008370</td>
<td>chr4:1008370:TC:T</td>
<td>TC</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1951</th>
<td>4</td>
<td>1008480</td>
<td>1008480</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3916;dist=3146</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1952</th>
<td>4</td>
<td>1008538</td>
<td>1008538</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3974;dist=3088</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008538</td>
<td>chr4:1008538:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1953</th>
<td>4</td>
<td>1008570</td>
<td>1008570</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4006;dist=3056</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008570</td>
<td>chr4:1008570:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1954</th>
<td>4</td>
<td>1008600</td>
<td>1008600</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4036;dist=3026</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008600</td>
<td>chr4:1008600:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>1955 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=2e7db3c5-aa82-41c0-9dce-969fbf842974">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[50]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[50]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           1568
exonic              167
intergenic           96
UTR3                 62
UTR5                 26
downstream           22
upstream             13
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=742ef450-0bf8-4c77-8bcd-1a4bd7810dab">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[51]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[51]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   167
Start                 167
End                   167
Ref                   167
Alt                   167
Func.refGene          167
Gene.refGene          167
GeneDetail.refGene    167
ExonicFunc.refGene    167
AAChange.refGene      167
clinvar_20140902      167
Otherinfo1            167
Otherinfo2            167
Otherinfo3            167
Otherinfo4            167
Otherinfo5            167
Otherinfo6            167
Otherinfo7            167
Otherinfo8            167
Otherinfo9            167
Otherinfo10           167
Otherinfo11           167
Otherinfo12           167
Otherinfo13           167
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a6b69f47-a29c-49a0-836e-503d9fe25949">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[52]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[52]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV      89
synonymous SNV         73
stopgain                4
frameshift deletion     1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=cfdcbdd5-b1a2-4960-a114-a83ea9006fa1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[53]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882390</td>
<td>882390</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882390</td>
<td>chr4:882390:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882501</td>
<td>882501</td>
<td>C</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882501</td>
<td>chr4:882501:C:A</td>
<td>C</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882507</td>
<td>882507</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882507</td>
<td>chr4:882507:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882530</td>
<td>882530</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>1</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>1/1</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>3166</th>
<td>4</td>
<td>1008570</td>
<td>1008570</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4006;dist=3056</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008570</td>
<td>chr4:1008570:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3167</th>
<td>4</td>
<td>1008634</td>
<td>1008634</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4070;dist=2992</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008634</td>
<td>chr4:1008634:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3168</th>
<td>4</td>
<td>1008636</td>
<td>1008636</td>
<td>C</td>
<td>A</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4072;dist=2990</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008636</td>
<td>chr4:1008636:C:A</td>
<td>C</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3169</th>
<td>4</td>
<td>1008644</td>
<td>1008644</td>
<td>T</td>
<td>C</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4080;dist=2982</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008644</td>
<td>chr4:1008644:T:C</td>
<td>T</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3170</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>3171 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=152d60cd-05b5-48b6-8204-743f58f429cb">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[54]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[54]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           2518
exonic              285
intergenic          174
UTR3                 99
UTR5                 43
downstream           33
upstream             17
exonic;splicing       1
splicing              1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=467c7db6-6f1e-46c1-a29b-f8cb99319b20">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[55]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[55]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   285
Start                 285
End                   285
Ref                   285
Alt                   285
Func.refGene          285
Gene.refGene          285
GeneDetail.refGene    285
ExonicFunc.refGene    285
AAChange.refGene      285
clinvar_20140902      285
Otherinfo1            285
Otherinfo2            285
Otherinfo3            285
Otherinfo4            285
Otherinfo5            285
Otherinfo6            285
Otherinfo7            285
Otherinfo8            285
Otherinfo9            285
Otherinfo10           285
Otherinfo11           285
Otherinfo12           285
Otherinfo13           285
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e1a6f2c1-b2a5-4869-9df1-56cd26d3e73e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[56]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[56]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV         164
synonymous SNV            109
stopgain                    6
frameshift deletion         4
nonframeshift deletion      2
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=dd0eec6d-9c07-404f-a661-51410ef13813">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[57]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882471</td>
<td>882471</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882471</td>
<td>chr4:882471:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882498</td>
<td>882498</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882498</td>
<td>chr4:882498:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882517</td>
<td>882517</td>
<td>C</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882517</td>
<td>chr4:882517:C:A</td>
<td>C</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882530</td>
<td>882530</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.5</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/1</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1940</th>
<td>4</td>
<td>1008371</td>
<td>1008371</td>
<td>C</td>
<td>-</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3807;dist=3255</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008370</td>
<td>chr4:1008370:TC:T</td>
<td>TC</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1941</th>
<td>4</td>
<td>1008480</td>
<td>1008480</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3916;dist=3146</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.5</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/1</td>
</tr>
<tr>
<th>1942</th>
<td>4</td>
<td>1008515</td>
<td>1008515</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3951;dist=3111</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008515</td>
<td>chr4:1008515:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1943</th>
<td>4</td>
<td>1008521</td>
<td>1008521</td>
<td>T</td>
<td>C</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3957;dist=3105</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008521</td>
<td>chr4:1008521:T:C</td>
<td>T</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1944</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>1945 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=dba5c22f-2ef2-43e0-8c28-3a69f75d4e16">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[58]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[58]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           1547
exonic              177
intergenic           94
UTR3                 59
downstream           34
UTR5                 22
upstream             11
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c3038cd0-f63b-46b7-ae85-042f100efc39">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[59]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[59]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   177
Start                 177
End                   177
Ref                   177
Alt                   177
Func.refGene          177
Gene.refGene          177
GeneDetail.refGene    177
ExonicFunc.refGene    177
AAChange.refGene      177
clinvar_20140902      177
Otherinfo1            177
Otherinfo2            177
Otherinfo3            177
Otherinfo4            177
Otherinfo5            177
Otherinfo6            177
Otherinfo7            177
Otherinfo8            177
Otherinfo9            177
Otherinfo10           177
Otherinfo11           177
Otherinfo12           177
Otherinfo13           177
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5681f86b-7d10-4321-ad82-6af8fd5d8db1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[60]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[60]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
synonymous SNV         91
nonsynonymous SNV      84
frameshift deletion     1
stopgain                1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ed235bcd-9dab-483a-86eb-c1aa46becf06">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[61]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882397</td>
<td>882397</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882397</td>
<td>chr4:882397:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882530</td>
<td>882530</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882566</td>
<td>882566</td>
<td>G</td>
<td>-</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882565</td>
<td>chr4:882565:AG:A</td>
<td>AG</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882611</td>
<td>882611</td>
<td>T</td>
<td>C</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882611</td>
<td>chr4:882611:T:C</td>
<td>T</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882678</td>
<td>882678</td>
<td>C</td>
<td>G</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882678</td>
<td>chr4:882678:C:G</td>
<td>C</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1723</th>
<td>4</td>
<td>1008246</td>
<td>1008246</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3682;dist=3380</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008246</td>
<td>chr4:1008246:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1724</th>
<td>4</td>
<td>1008337</td>
<td>1008337</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3773;dist=3289</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.5</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/1</td>
</tr>
<tr>
<th>1725</th>
<td>4</td>
<td>1008371</td>
<td>1008371</td>
<td>C</td>
<td>-</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3807;dist=3255</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008370</td>
<td>chr4:1008370:TC:T</td>
<td>TC</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1726</th>
<td>4</td>
<td>1008480</td>
<td>1008480</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3916;dist=3146</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1727</th>
<td>4</td>
<td>1008538</td>
<td>1008538</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3974;dist=3088</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008538</td>
<td>chr4:1008538:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>1728 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e258c45e-a6ef-4a0a-bca1-6c1671be0e7b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[62]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[62]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           1412
exonic              143
intergenic           72
UTR3                 50
UTR5                 24
downstream           18
upstream              8
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=14f514fe-fccd-416d-9ff7-94546b7e1546">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[63]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[63]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   143
Start                 143
End                   143
Ref                   143
Alt                   143
Func.refGene          143
Gene.refGene          143
GeneDetail.refGene    143
ExonicFunc.refGene    143
AAChange.refGene      143
clinvar_20140902      143
Otherinfo1            143
Otherinfo2            143
Otherinfo3            143
Otherinfo4            143
Otherinfo5            143
Otherinfo6            143
Otherinfo7            143
Otherinfo8            143
Otherinfo9            143
Otherinfo10           143
Otherinfo11           143
Otherinfo12           143
Otherinfo13           143
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=bd7cfd6a-f766-4eee-8bb9-d7093d5bc574">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[64]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[64]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV      75
synonymous SNV         64
stopgain                3
frameshift deletion     1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1d24c787-a2f9-4dcc-9283-35fd498f0279">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[65]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882530</td>
<td>882530</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.5</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/1</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882611</td>
<td>882611</td>
<td>T</td>
<td>C</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882611</td>
<td>chr4:882611:T:C</td>
<td>T</td>
<td>C</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882849</td>
<td>882849</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882849</td>
<td>chr4:882849:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>883018</td>
<td>883018</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>883018</td>
<td>chr4:883018:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>883133</td>
<td>883133</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>883133</td>
<td>chr4:883133:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>600</th>
<td>4</td>
<td>1008009</td>
<td>1008009</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3445;dist=3617</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0.5</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008009</td>
<td>chr4:1008009:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/1</td>
</tr>
<tr>
<th>601</th>
<td>4</td>
<td>1008209</td>
<td>1008209</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3645;dist=3417</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008209</td>
<td>chr4:1008209:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>602</th>
<td>4</td>
<td>1008337</td>
<td>1008337</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3773;dist=3289</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>1</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>1/1</td>
</tr>
<tr>
<th>603</th>
<td>4</td>
<td>1008371</td>
<td>1008371</td>
<td>C</td>
<td>-</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3807;dist=3255</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008370</td>
<td>chr4:1008370:TC:T</td>
<td>TC</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>604</th>
<td>4</td>
<td>1008480</td>
<td>1008480</td>
<td>A</td>
<td>G</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=3916;dist=3146</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>605 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0950d33b-6128-42a7-acfb-787928219a5c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[66]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[66]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           491
exonic              50
intergenic          29
UTR3                20
downstream           6
upstream             5
UTR5                 3
exonic;splicing      1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=367cb26b-7ac4-4c74-8b4e-3e778db2aa6a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[67]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[67]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   50
Start                 50
End                   50
Ref                   50
Alt                   50
Func.refGene          50
Gene.refGene          50
GeneDetail.refGene    50
ExonicFunc.refGene    50
AAChange.refGene      50
clinvar_20140902      50
Otherinfo1            50
Otherinfo2            50
Otherinfo3            50
Otherinfo4            50
Otherinfo5            50
Otherinfo6            50
Otherinfo7            50
Otherinfo8            50
Otherinfo9            50
Otherinfo10           50
Otherinfo11           50
Otherinfo12           50
Otherinfo13           50
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=2ef51c7e-e50e-4179-a8ea-fed89c7002f5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[68]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[68]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
synonymous SNV       27
nonsynonymous SNV    23
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=636c9cb5-45be-4134-b557-00c2756945ea">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[74]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Read in ANNOVAR multianno file</span>
<span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">display</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Chr</th>
<th>Start</th>
<th>End</th>
<th>Ref</th>
<th>Alt</th>
<th>Func.refGene</th>
<th>Gene.refGene</th>
<th>GeneDetail.refGene</th>
<th>ExonicFunc.refGene</th>
<th>AAChange.refGene</th>
<th>clinvar_20140902</th>
<th>Otherinfo1</th>
<th>Otherinfo2</th>
<th>Otherinfo3</th>
<th>Otherinfo4</th>
<th>Otherinfo5</th>
<th>Otherinfo6</th>
<th>Otherinfo7</th>
<th>Otherinfo8</th>
<th>Otherinfo9</th>
<th>Otherinfo10</th>
<th>Otherinfo11</th>
<th>Otherinfo12</th>
<th>Otherinfo13</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882390</td>
<td>882390</td>
<td>C</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882390</td>
<td>chr4:882390:C:A</td>
<td>C</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882397</td>
<td>882397</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882397</td>
<td>chr4:882397:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882422</td>
<td>882422</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882422</td>
<td>chr4:882422:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882432</td>
<td>882432</td>
<td>G</td>
<td>A</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882432</td>
<td>chr4:882432:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882454</td>
<td>882454</td>
<td>C</td>
<td>T</td>
<td>intronic</td>
<td>GAK</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>882454</td>
<td>chr4:882454:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>9499</th>
<td>4</td>
<td>1008570</td>
<td>1008570</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4006;dist=3056</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008570</td>
<td>chr4:1008570:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>9500</th>
<td>4</td>
<td>1008571</td>
<td>1008571</td>
<td>G</td>
<td>A</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4007;dist=3055</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008571</td>
<td>chr4:1008571:G:A</td>
<td>G</td>
<td>A</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>9501</th>
<td>4</td>
<td>1008596</td>
<td>1008596</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4032;dist=3030</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008596</td>
<td>chr4:1008596:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>./.</td>
</tr>
<tr>
<th>9502</th>
<td>4</td>
<td>1008600</td>
<td>1008600</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4036;dist=3026</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008600</td>
<td>chr4:1008600:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
<tr>
<th>9503</th>
<td>4</td>
<td>1008649</td>
<td>1008649</td>
<td>C</td>
<td>T</td>
<td>intergenic</td>
<td>IDUA;FGFRL1</td>
<td>dist=4085;dist=2977</td>
<td>.</td>
<td>.</td>
<td>.</td>
<td>0</td>
<td>.</td>
<td>.</td>
<td>4</td>
<td>1008649</td>
<td>chr4:1008649:C:T</td>
<td>C</td>
<td>T</td>
<td>.</td>
<td>.</td>
<td>PR</td>
<td>GT</td>
<td>0/0</td>
</tr>
</tbody>
</table>
<p>9504 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ba43863e-9f25-45a1-be06-f9f18d7cf601">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[75]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">gene</span><span class="p">[</span><span class="s2">"Func.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[75]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Func.refGene
intronic           7451
exonic              951
intergenic          474
UTR3                266
downstream          152
UTR5                114
upstream             90
splicing              5
exonic;splicing       1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=05abffe5-765f-4334-9595-283cf048038b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[76]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># Filter exonic variants</span>
<span class="n">coding</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
<span class="n">coding</span><span class="o">.</span><span class="n">count</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[76]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>Chr                   951
Start                 951
End                   951
Ref                   951
Alt                   951
Func.refGene          951
Gene.refGene          951
GeneDetail.refGene    951
ExonicFunc.refGene    951
AAChange.refGene      951
clinvar_20140902      951
Otherinfo1            951
Otherinfo2            951
Otherinfo3            951
Otherinfo4            951
Otherinfo5            951
Otherinfo6            951
Otherinfo7            951
Otherinfo8            951
Otherinfo9            951
Otherinfo10           951
Otherinfo11           951
Otherinfo12           951
Otherinfo13           951
dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=85f7ace5-2ae5-4147-b33a-d6f43df01af6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[77]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">coding</span><span class="p">[</span><span class="s2">"ExonicFunc.refGene"</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[77]:</div>
<div class="jp-RenderedText jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/plain" tabindex="0">
<pre>ExonicFunc.refGene
nonsynonymous SNV          548
synonymous SNV             367
nonframeshift deletion      12
stopgain                    10
frameshift deletion          9
frameshift insertion         4
nonframeshift insertion      1
Name: count, dtype: int64</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=47b09ef1-37c1-48f7-9ab1-4ae876af99c8">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[69]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Make lists of variants to keep - all coding, coding nonsynonymous (missense - as they are coded in ANNOVAR), deleterious (CADD_phred &gt; 20)</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'WORKING ON: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1"># Read in ANNOVAR multianno file</span>
    <span class="n">gene</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">_TMEM175.annovar.hg38_multianno.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">#Print number of variants in the different categories</span>
    <span class="n">results</span> <span class="o">=</span> <span class="p">[]</span> 
    
    <span class="n">utr5</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span><span class="o">==</span> <span class="s1">'UTR5'</span><span class="p">]</span>
    <span class="n">intronic</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span><span class="o">==</span> <span class="s1">'intronic'</span><span class="p">]</span>
    <span class="n">exonic</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span><span class="o">==</span> <span class="s1">'exonic'</span><span class="p">]</span>
    <span class="n">utr3</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span><span class="o">==</span> <span class="s1">'UTR3'</span><span class="p">]</span>
    <span class="n">coding_nonsynonymous</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[(</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">gene</span><span class="p">[</span><span class="s1">'ExonicFunc.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'nonsynonymous SNV'</span><span class="p">)]</span>
    <span class="n">coding_synonymous</span> <span class="o">=</span> <span class="n">gene</span><span class="p">[(</span><span class="n">gene</span><span class="p">[</span><span class="s1">'Func.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'exonic'</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">gene</span><span class="p">[</span><span class="s1">'ExonicFunc.refGene'</span><span class="p">]</span> <span class="o">!=</span> <span class="s1">'nonsynonymous SNV'</span><span class="p">)]</span>
    <span class="n">lof</span> <span class="o">=</span> <span class="n">exonic</span><span class="p">[(</span><span class="n">exonic</span><span class="p">[</span><span class="s1">'ExonicFunc.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'stopgain'</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">exonic</span><span class="p">[</span><span class="s1">'ExonicFunc.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'stoploss'</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">exonic</span><span class="p">[</span><span class="s1">'ExonicFunc.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'frameshift deletion'</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">exonic</span><span class="p">[</span><span class="s1">'ExonicFunc.refGene'</span><span class="p">]</span> <span class="o">==</span> <span class="s1">'frameshift insertion'</span><span class="p">)]</span>
    <span class="n">nonsynonymous_lof</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">coding_nonsynonymous</span><span class="p">,</span> <span class="n">lof</span><span class="p">])</span>
    
    <span class="nb">print</span><span class="p">({</span><span class="n">ancestry</span><span class="p">})</span>
    <span class="nb">print</span><span class="p">(</span><span class="s1">'Total variants: '</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">gene</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s2">"Intronic: "</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">intronic</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s1">'UTR3: '</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">utr3</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s1">'UTR5: '</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">utr5</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s2">"Total exonic: "</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">exonic</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s1">'  Synonymous: '</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">coding_synonymous</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s2">"  Nonsynonymous: "</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">coding_nonsynonymous</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s2">"nonsynonymous_lof: "</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">nonsynonymous_lof</span><span class="p">))</span>
    <span class="n">results</span><span class="o">.</span><span class="n">append</span><span class="p">((</span><span class="n">gene</span><span class="p">,</span> <span class="n">intronic</span><span class="p">,</span> <span class="n">utr3</span><span class="p">,</span> <span class="n">utr5</span><span class="p">,</span> <span class="n">exonic</span><span class="p">,</span> <span class="n">coding_synonymous</span><span class="p">,</span> <span class="n">coding_nonsynonymous</span><span class="p">,</span> <span class="n">nonsynonymous_lof</span><span class="p">))</span>
    <span class="nb">print</span><span class="p">(</span><span class="s1">'</span><span class="se">\n</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1"># Save in PLINK format - coding nonsynonymous </span>
    <span class="c1"># These are missense variants - other types of nonsynonymous variants (e.g stopgain/loss, or frameshift variants are coded differently in the ExonicFunc.refGene </span>
    <span class="n">variants_toKeep</span> <span class="o">=</span> <span class="n">nonsynonymous_lof</span><span class="p">[[</span><span class="s1">'Chr'</span><span class="p">,</span> <span class="s1">'Start'</span><span class="p">,</span> <span class="s1">'End'</span><span class="p">,</span> <span class="s1">'Gene.refGene'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="n">variants_toKeep</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">_TMEM175.nonsynonymous_lof.variantstoKeep.txt'</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    
    <span class="c1"># Save in PLINK format - all coding variants</span>
    <span class="n">variants_toKeep2</span> <span class="o">=</span> <span class="n">exonic</span><span class="p">[[</span><span class="s1">'Chr'</span><span class="p">,</span> <span class="s1">'Start'</span><span class="p">,</span> <span class="s1">'End'</span><span class="p">,</span> <span class="s1">'Gene.refGene'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="n">variants_toKeep2</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">_TMEM175.exonic.variantstoKeep.txt'</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: AFR
{'AFR'}
Total variants:  3853
Intronic:  3069
UTR3:  118
UTR5:  45
Total exonic:  314
  Synonymous:  153
  Nonsynonymous:  161
nonsynonymous_lof:  167


WORKING ON: AAC
{'AAC'}
Total variants:  3449
Intronic:  2746
UTR3:  102
UTR5:  42
Total exonic:  287
  Synonymous:  132
  Nonsynonymous:  155
nonsynonymous_lof:  161


WORKING ON: FIN
{'FIN'}
Total variants:  605
Intronic:  491
UTR3:  20
UTR5:  3
Total exonic:  50
  Synonymous:  27
  Nonsynonymous:  23
nonsynonymous_lof:  23


WORKING ON: AMR
{'AMR'}
Total variants:  1987
Intronic:  1596
UTR3:  56
UTR5:  17
Total exonic:  178
  Synonymous:  89
  Nonsynonymous:  89
nonsynonymous_lof:  92


WORKING ON: EAS
{'EAS'}
Total variants:  3171
Intronic:  2518
UTR3:  99
UTR5:  43
Total exonic:  285
  Synonymous:  121
  Nonsynonymous:  164
nonsynonymous_lof:  174


WORKING ON: CAH
{'CAH'}
Total variants:  3090
Intronic:  2467
UTR3:  87
UTR5:  31
Total exonic:  260
  Synonymous:  121
  Nonsynonymous:  139
nonsynonymous_lof:  146


WORKING ON: CAS
{'CAS'}
Total variants:  1955
Intronic:  1568
UTR3:  62
UTR5:  26
Total exonic:  167
  Synonymous:  78
  Nonsynonymous:  89
nonsynonymous_lof:  94


WORKING ON: SAS
{'SAS'}
Total variants:  1728
Intronic:  1412
UTR3:  50
UTR5:  24
Total exonic:  143
  Synonymous:  68
  Nonsynonymous:  75
nonsynonymous_lof:  79


WORKING ON: EUR
{'EUR'}
Total variants:  9504
Intronic:  7451
UTR3:  266
UTR5:  114
Total exonic:  951
  Synonymous:  403
  Nonsynonymous:  548
nonsynonymous_lof:  571


WORKING ON: AJ
{'AJ'}
Total variants:  1285
Intronic:  1024
UTR3:  43
UTR5:  6
Total exonic:  125
  Synonymous:  59
  Nonsynonymous:  66
nonsynonymous_lof:  69


WORKING ON: MDE
{'MDE'}
Total variants:  1945
Intronic:  1547
UTR3:  59
UTR5:  22
Total exonic:  177
  Synonymous:  93
  Nonsynonymous:  84
nonsynonymous_lof:  86


</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=a117d5ea-00f5-49aa-9199-b775529c28d9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>ALL variants</p>
<p>assoc</p>
<p>glm</p>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=7af229a1-2872-4b47-8957-8dff7d966fab">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>ASSOC</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=14c14ecc-1aa1-47d4-bd32-c2d2592db62e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[70]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Run case-control analysis using plink assoc for all variants, not adjusting for any covariates</span>
<span class="n">ancestries</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'AAC'</span><span class="p">,</span> <span class="s1">'AFR'</span><span class="p">,</span> <span class="s1">'AJ'</span><span class="p">,</span> <span class="s1">'AMR'</span><span class="p">,</span> <span class="s1">'CAS'</span><span class="p">,</span> <span class="s1">'EAS'</span><span class="p">,</span> <span class="s1">'EUR'</span><span class="p">,</span> <span class="s1">'FIN'</span><span class="p">,</span> <span class="s1">'MDE'</span><span class="p">,</span> <span class="s1">'SAS'</span><span class="p">,</span> <span class="s1">'CAH'</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>

    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    
    <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink<span class="w"> </span><span class="err">\</span>
    <span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span> \
    <span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">.</span><span class="n">samplestoKeep</span> \
    <span class="o">--</span><span class="n">assoc</span> \
    <span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">no</span><span class="o">-</span><span class="n">sex</span> \
    <span class="o">--</span><span class="n">ci</span> <span class="mf">0.95</span> \
    <span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
    <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span><span class="o">.</span><span class="n">all</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
3449 variants loaded from .bim file.
1111 people (455 males, 656 females) loaded from .fam.
1086 phenotype values loaded from .fam.
--keep: 1111 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 1111 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.998498.
2861 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
588 variants and 1111 people pass filters and QC.
Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes
are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
3853 variants loaded from .bim file.
2643 people (1480 males, 1163 females) loaded from .fam.
2621 phenotype values loaded from .fam.
--keep: 2643 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 2643 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.996619.
3262 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
591 variants and 2643 people pass filters and QC.
Among remaining phenotypes, 942 are cases and 1679 are controls.  (22
phenotypes are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
1285 variants loaded from .bim file.
2655 people (1648 males, 1007 females) loaded from .fam.
1703 phenotype values loaded from .fam.
--keep: 2655 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 2655 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.998935.
869 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
416 variants and 2655 people pass filters and QC.
Among remaining phenotypes, 1292 are cases and 411 are controls.  (952
phenotypes are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
1987 variants loaded from .bim file.
646 people (363 males, 283 females) loaded from .fam.
613 phenotype values loaded from .fam.
--keep: 646 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 646 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.997492.
1582 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
405 variants and 646 people pass filters and QC.
Among remaining phenotypes, 458 are cases and 155 are controls.  (33 phenotypes
are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
1955 variants loaded from .bim file.
903 people (439 males, 464 females) loaded from .fam.
895 phenotype values loaded from .fam.
--keep: 903 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 903 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.995381.
1526 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
429 variants and 903 people pass filters and QC.
Among remaining phenotypes, 552 are cases and 343 are controls.  (8 phenotypes
are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
3171 variants loaded from .bim file.
5167 people (3278 males, 1889 females) loaded from .fam.
5123 phenotype values loaded from .fam.
--keep: 5167 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 5167 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.994312.
2794 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
377 variants and 5167 people pass filters and QC.
Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44
phenotypes are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
9504 variants loaded from .bim file.
38839 people (23296 males, 15543 females) loaded from .fam.
30412 phenotype values loaded from .fam.
--keep: 38838 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 38838 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate in remaining samples is 0.760113.
9107 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
397 variants and 38838 people pass filters and QC.
Among remaining phenotypes, 21197 are cases and 9214 are controls.  (8427
phenotypes are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
605 variants loaded from .bim file.
114 people (53 males, 61 females) loaded from .fam.
106 phenotype values loaded from .fam.
--keep: 114 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 114 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.991605.
200 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
405 variants and 114 people pass filters and QC.
Among remaining phenotypes, 98 are cases and 8 are controls.  (8 phenotypes are
missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
1945 variants loaded from .bim file.
581 people (369 males, 212 females) loaded from .fam.
536 phenotype values loaded from .fam.
--keep: 581 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 581 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.993732.
1547 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
398 variants and 581 people pass filters and QC.
Among remaining phenotypes, 311 are cases and 225 are controls.  (45 phenotypes
are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
1728 variants loaded from .bim file.
635 people (408 males, 227 females) loaded from .fam.
609 phenotype values loaded from .fam.
--keep: 635 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 635 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.994328.
1251 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
477 variants and 635 people pass filters and QC.
Among remaining phenotypes, 387 are cases and 222 are controls.  (26 phenotypes
are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all.assoc ...
done.
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
3090 variants loaded from .bim file.
851 people (446 males, 405 females) loaded from .fam.
827 phenotype values loaded from .fam.
--keep: 851 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 851 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.997283.
2542 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
548 variants and 851 people pass filters and QC.
Among remaining phenotypes, 525 are cases and 302 are controls.  (24 phenotypes
are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all.assoc ...
done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4092d28d-e7ad-495d-9cc3-cf53afa17488">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># test if we can see TMEM175 p.M393T when set maf&lt;0.01 in AFR</span>
<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_AFR'</span>

<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">AFR_TMEM175</span> \
<span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">AFR</span><span class="o">.</span><span class="n">samplestoKeep</span> \
<span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">no</span><span class="o">-</span><span class="n">sex</span> \
<span class="o">--</span><span class="nb">max</span><span class="o">-</span><span class="n">maf</span> <span class="mf">0.01</span> \
<span class="o">--</span><span class="n">ci</span> <span class="mf">0.95</span> \
<span class="o">--</span><span class="n">assoc</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">AFR_TMEM175</span><span class="o">.</span><span class="n">all</span><span class="o">.</span><span class="n">test</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.test.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep
  --max-maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.test

12984 MB RAM detected; reserving 6492 MB for main workspace.
3853 variants loaded from .bim file.
2643 people (1480 males, 1163 females) loaded from .fam.
2621 phenotype values loaded from .fam.
--keep: 2643 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 2643 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.996619.
591 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
3262 variants and 2643 people pass filters and QC.
Among remaining phenotypes, 942 are cases and 1679 are controls.  (22
phenotypes are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.test.assoc
... 316294done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=ce9fa5a0-790b-48f0-b4c5-bf3693c9054f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_AFR'</span>
<span class="n">freq</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/AFR_TMEM175.all.test.assoc'</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">'\s+'</span><span class="p">)</span>
<span class="n">freq</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/AFR.all_nonadj.test.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=02ec7380-f232-46a4-bb9a-78600c917316">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[81]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">ancestries</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'AAC'</span><span class="p">,</span> <span class="s1">'AFR'</span><span class="p">,</span> <span class="s1">'AJ'</span><span class="p">,</span> <span class="s1">'AMR'</span><span class="p">,</span> <span class="s1">'CAS'</span><span class="p">,</span> <span class="s1">'EAS'</span><span class="p">,</span> <span class="s1">'EUR'</span><span class="p">,</span> <span class="s1">'FIN'</span><span class="p">,</span> <span class="s1">'MDE'</span><span class="p">,</span> <span class="s1">'SAS'</span><span class="p">,</span> <span class="s1">'CAH'</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'WORKING ON: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">#Look at assoc results, only variants with nominal p-value &lt; 0.05</span>
    <span class="n">freq</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">_TMEM175.all.assoc'</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">'\s+'</span><span class="p">)</span>
    <span class="n">sig_all_nonadj</span> <span class="o">=</span> <span class="n">freq</span><span class="p">[</span><span class="n">freq</span><span class="p">[</span><span class="s1">'P'</span><span class="p">]</span><span class="o">&lt;</span><span class="mf">0.05</span><span class="p">]</span>
    
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Variants with p-value &lt; 0.05: </span><span class="si">{</span><span class="n">sig_all_nonadj</span><span class="o">.</span><span class="n">shape</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">#Save FREQ to csv</span>
    <span class="n">freq</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">.all_nonadj.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: AAC
Variants with p-value &lt; 0.05: (26, 13)
WORKING ON: AFR
Variants with p-value &lt; 0.05: (90, 13)
WORKING ON: AJ
Variants with p-value &lt; 0.05: (115, 13)
WORKING ON: AMR
Variants with p-value &lt; 0.05: (7, 13)
WORKING ON: CAS
Variants with p-value &lt; 0.05: (38, 13)
WORKING ON: EAS
Variants with p-value &lt; 0.05: (200, 13)
WORKING ON: EUR
Variants with p-value &lt; 0.05: (202, 13)
WORKING ON: FIN
Variants with p-value &lt; 0.05: (9, 13)
WORKING ON: MDE
Variants with p-value &lt; 0.05: (84, 13)
WORKING ON: SAS
Variants with p-value &lt; 0.05: (24, 13)
WORKING ON: CAH
Variants with p-value &lt; 0.05: (54, 13)
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=fb196a2e-2c9c-4154-8732-036513c49883">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[82]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Run case-control analysis with covariates</span>
<span class="n">ancestries</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'AAC'</span><span class="p">,</span> <span class="s1">'AFR'</span><span class="p">,</span> <span class="s1">'AJ'</span><span class="p">,</span> <span class="s1">'AMR'</span><span class="p">,</span> <span class="s1">'CAS'</span><span class="p">,</span> <span class="s1">'EAS'</span><span class="p">,</span> <span class="s1">'EUR'</span><span class="p">,</span> <span class="s1">'FIN'</span><span class="p">,</span> <span class="s1">'MDE'</span><span class="p">,</span> <span class="s1">'SAS'</span><span class="p">,</span> <span class="s1">'CAH'</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>

    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>

    <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
    <span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span> \
    <span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">.</span><span class="n">samplestoKeep</span> \
    <span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">no</span><span class="o">-</span><span class="n">sex</span> \
    <span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
    <span class="o">--</span><span class="n">ci</span> <span class="mf">0.95</span> \
    <span class="o">--</span><span class="n">glm</span> \
    <span class="o">--</span><span class="n">covar</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_covariate_file</span><span class="o">.</span><span class="n">txt</span> \
    <span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">name</span> <span class="n">SEX</span><span class="p">,</span><span class="n">AGE</span><span class="p">,</span><span class="n">PC1</span><span class="p">,</span><span class="n">PC2</span><span class="p">,</span><span class="n">PC3</span><span class="p">,</span><span class="n">PC4</span><span class="p">,</span><span class="n">PC5</span> \
    <span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">variance</span><span class="o">-</span><span class="n">standardize</span> \
    <span class="o">--</span><span class="n">neg9</span><span class="o">-</span><span class="n">pheno</span><span class="o">-</span><span class="n">really</span><span class="o">-</span><span class="n">missing</span> \
    <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span><span class="o">.</span><span class="n">all_adj</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj

Start time: Wed Jul 31 11:04:25 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50540 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1111 samples (656 females, 455 males; 1111 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam.
3449 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim.
1 binary phenotype loaded (285 cases, 801 controls).
--keep: 1111 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt.
1111 samples (656 females, 455 males; 1111 founders) remaining after main
filters.
285 cases and 801 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
2861 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
588 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:25 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj

Start time: Wed Jul 31 11:04:26 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50537 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2643 samples (1163 females, 1480 males; 2643 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam.
3853 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim.
1 binary phenotype loaded (942 cases, 1679 controls).
--keep: 2643 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt.
2643 samples (1163 females, 1480 males; 2643 founders) remaining after main
filters.
942 cases and 1679 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
3262 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
591 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:26 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj

Start time: Wed Jul 31 11:04:26 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50516 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2655 samples (1007 females, 1648 males; 2655 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam.
1285 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim.
1 binary phenotype loaded (1292 cases, 411 controls).
--keep: 2655 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt.
2655 samples (1007 females, 1648 males; 2655 founders) remaining after main
filters.
1292 cases and 411 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
869 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
416 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:26 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj

Start time: Wed Jul 31 11:04:27 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
646 samples (283 females, 363 males; 646 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam.
1987 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim.
1 binary phenotype loaded (458 cases, 155 controls).
--keep: 646 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt.
646 samples (283 females, 363 males; 646 founders) remaining after main
filters.
458 cases and 155 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
1582 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
405 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:27 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj

Start time: Wed Jul 31 11:04:27 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50516 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
903 samples (464 females, 439 males; 903 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam.
1955 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim.
1 binary phenotype loaded (552 cases, 343 controls).
--keep: 903 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt.
903 samples (464 females, 439 males; 903 founders) remaining after main
filters.
552 cases and 343 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
1526 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
429 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:27 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj

Start time: Wed Jul 31 11:04:28 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50529 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
5167 samples (1889 females, 3278 males; 5167 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam.
3171 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim.
1 binary phenotype loaded (2662 cases, 2461 controls).
--keep: 5167 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt.
5167 samples (1889 females, 3278 males; 5167 founders) remaining after main
filters.
2662 cases and 2461 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
2794 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
377 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:28 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj

Start time: Wed Jul 31 11:04:28 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50516 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
38839 samples (15543 females, 23296 males; 38839 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam.
9504 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim.
1 binary phenotype loaded (21198 cases, 9214 controls).
--keep: 38838 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt.
38838 samples (15543 females, 23295 males; 38838 founders) remaining after main
filters.
21197 cases and 9214 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
9107 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
397 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:29 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj

Start time: Wed Jul 31 11:04:29 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
114 samples (61 females, 53 males; 114 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam.
605 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim.
1 binary phenotype loaded (98 cases, 8 controls).
--keep: 114 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt.
114 samples (61 females, 53 males; 114 founders) remaining after main filters.
98 cases and 8 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
200 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
405 variants remaining after main filters.
Warning: --glm remaining control count is less than 10x predictor count for
phenotype 'PHENO1'.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:30 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj

Start time: Wed Jul 31 11:04:30 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
581 samples (212 females, 369 males; 581 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam.
1945 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim.
1 binary phenotype loaded (311 cases, 225 controls).
--keep: 581 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt.
581 samples (212 females, 369 males; 581 founders) remaining after main
filters.
311 cases and 225 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
1547 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
398 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:30 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj

Start time: Wed Jul 31 11:04:30 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50517 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
635 samples (227 females, 408 males; 635 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam.
1728 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim.
1 binary phenotype loaded (387 cases, 222 controls).
--keep: 635 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt.
635 samples (227 females, 408 males; 635 founders) remaining after main
filters.
387 cases and 222 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
1251 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
477 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:31 2024
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj

Start time: Wed Jul 31 11:04:31 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50518 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
851 samples (405 females, 446 males; 851 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam.
3090 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim.
1 binary phenotype loaded (525 cases, 302 controls).
--keep: 851 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt.
851 samples (405 females, 446 males; 851 founders) remaining after main
filters.
525 cases and 302 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
2542 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
548 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Wed Jul 31 11:04:31 2024
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=957dbb8c-a86b-49b7-943d-d239e70bd49d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[6]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># test if we can see TMEM175 p.M393T when set maf&lt;0.01 in AFR</span>
<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_AFR'</span>

<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">AFR_TMEM175</span> \
<span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">AFR</span><span class="o">.</span><span class="n">samplestoKeep</span> \
<span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">no</span><span class="o">-</span><span class="n">sex</span> \
<span class="o">--</span><span class="nb">max</span><span class="o">-</span><span class="n">maf</span> <span class="mf">0.01</span> \
<span class="o">--</span><span class="n">ci</span> <span class="mf">0.95</span> \
<span class="o">--</span><span class="n">glm</span> \
<span class="o">--</span><span class="n">covar</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">AFR_covariate_file</span><span class="o">.</span><span class="n">txt</span> \
<span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">name</span> <span class="n">SEX</span><span class="p">,</span><span class="n">AGE</span><span class="p">,</span><span class="n">PC1</span><span class="p">,</span><span class="n">PC2</span><span class="p">,</span><span class="n">PC3</span><span class="p">,</span><span class="n">PC4</span><span class="p">,</span><span class="n">PC5</span> \
<span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">variance</span><span class="o">-</span><span class="n">standardize</span> \
<span class="o">--</span><span class="n">neg9</span><span class="o">-</span><span class="n">pheno</span><span class="o">-</span><span class="n">really</span><span class="o">-</span><span class="n">missing</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">AFR_TMEM175</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">test</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.test.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep
  --max-maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.test

Start time: Wed Mar  5 10:28:20 2025
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
12984 MiB RAM detected, ~11798 available; reserving 6492 MiB for main
workspace.
Using up to 2 compute threads.
2643 samples (1163 females, 1480 males; 2643 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam.
3853 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim.
1 binary phenotype loaded (942 cases, 1679 controls).
--keep: 2643 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt.
2643 samples (1163 females, 1480 males; 2643 founders) remaining after main
filters.
942 cases and 1679 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
591 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
3262 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.test.PHENO1.glm.logistic.hybrid .
End time: Wed Mar  5 10:28:23 2025
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e0b25479-357c-45dc-9680-ba7cbb96cc7b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Read in plink glm results</span>
<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_AFR'</span>
<span class="n">assoc</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/AFR_TMEM175.all_adj.test.PHENO1.glm.logistic.hybrid'</span><span class="p">,</span> <span class="n">delim_whitespace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">assoc_add</span> <span class="o">=</span> <span class="n">assoc</span><span class="p">[</span><span class="n">assoc</span><span class="p">[</span><span class="s1">'TEST'</span><span class="p">]</span><span class="o">==</span><span class="s2">"ADD"</span><span class="p">]</span>
<span class="n">assoc_add</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/AFR.all_adj.test.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_2920/502712416.py:3: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e4e16bfa-a27e-4e6d-a941-31724185d0a5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[83]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Process results from plink glm analysis including covariates</span>
<span class="n">ancestries</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'AAC'</span><span class="p">,</span> <span class="s1">'AFR'</span><span class="p">,</span> <span class="s1">'AJ'</span><span class="p">,</span> <span class="s1">'AMR'</span><span class="p">,</span> <span class="s1">'CAS'</span><span class="p">,</span> <span class="s1">'EAS'</span><span class="p">,</span> <span class="s1">'EUR'</span><span class="p">,</span> <span class="s1">'FIN'</span><span class="p">,</span> <span class="s1">'MDE'</span><span class="p">,</span> <span class="s1">'SAS'</span><span class="p">,</span> <span class="s1">'CAH'</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'WORKING ON: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="c1">#Read in plink glm results</span>
    <span class="n">assoc</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">_TMEM175.all_adj.PHENO1.glm.logistic.hybrid'</span><span class="p">,</span> <span class="n">delim_whitespace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
    
    <span class="c1">#Filter for additive test only - this is the variant results</span>
    <span class="n">assoc_add</span> <span class="o">=</span> <span class="n">assoc</span><span class="p">[</span><span class="n">assoc</span><span class="p">[</span><span class="s1">'TEST'</span><span class="p">]</span><span class="o">==</span><span class="s2">"ADD"</span><span class="p">]</span>
    
    <span class="c1">#Check if there are any significant (p &lt; 0.05) variants</span>
    <span class="n">significant</span> <span class="o">=</span> <span class="n">assoc_add</span><span class="p">[</span><span class="n">assoc_add</span><span class="p">[</span><span class="s1">'P'</span><span class="p">]</span><span class="o">&lt;</span><span class="mf">0.05</span><span class="p">]</span>

    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'There are </span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">significant</span><span class="p">)</span><span class="si">}</span><span class="s1"> variants with p-value &lt; 0.05 in glm'</span><span class="p">)</span>
    
    <span class="c1">#Check if there are any significant (p &lt; 0.05) variants</span>
    <span class="n">GWsignificant</span> <span class="o">=</span> <span class="n">assoc_add</span><span class="p">[</span><span class="n">assoc_add</span><span class="p">[</span><span class="s1">'P'</span><span class="p">]</span><span class="o">&lt;</span><span class="mf">5e-8</span><span class="p">]</span>

    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'There are </span><span class="si">{</span><span class="nb">len</span><span class="p">(</span><span class="n">GWsignificant</span><span class="p">)</span><span class="si">}</span><span class="s1"> variants with p-value &lt; 5e-8 in glm'</span><span class="p">)</span>
    
    <span class="c1">#Save assoc_add to csv</span>
    <span class="n">assoc_add</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">.all_adj.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: AAC
There are 24 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: AFR
There are 46 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: AJ
There are 121 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
WORKING ON: AMR
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>There are 7 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
WORKING ON: CAS
There are 26 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: EAS
There are 44 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
WORKING ON: EUR
There are 148 variants with p-value &lt; 0.05 in glm
There are 40 variants with p-value &lt; 5e-8 in glm
WORKING ON: FIN
There are 15 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: MDE
There are 124 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
WORKING ON: SAS
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>There are 42 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
WORKING ON: CAH
There are 27 variants with p-value &lt; 0.05 in glm
There are 0 variants with p-value &lt; 5e-8 in glm
</pre>
</div>
</div>
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_7808/2601943929.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=d96aca21-208e-4d4b-a1a1-7e7510886504">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Burden Analyses using RVTests</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=acb6ba3c-8ab5-4cd8-8599-74853bb7ede2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[84]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">ancestries</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'AAC'</span><span class="p">,</span> <span class="s1">'AFR'</span><span class="p">,</span> <span class="s1">'AJ'</span><span class="p">,</span> <span class="s1">'AMR'</span><span class="p">,</span> <span class="s1">'CAS'</span><span class="p">,</span> <span class="s1">'EAS'</span><span class="p">,</span> <span class="s1">'EUR'</span><span class="p">,</span> <span class="s1">'FIN'</span><span class="p">,</span> <span class="s1">'MDE'</span><span class="p">,</span> <span class="s1">'SAS'</span><span class="p">,</span> <span class="s1">'CAH'</span><span class="p">]</span>
<span class="n">variant_classes</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'exonic'</span><span class="p">,</span> <span class="s1">'nonsynonymous_lof'</span><span class="p">]</span>

<span class="c1">#Loop over all the ancestries and the 2 variant classes - run rvtests for all coding and missense variants</span>
<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">variant_class</span> <span class="ow">in</span> <span class="n">variant_classes</span><span class="p">:</span>
        
        <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
        
        <span class="c1"># Print the command to be executed (for debugging purposes)</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Running plink to extract </span><span class="si">{</span><span class="n">variant_class</span><span class="si">}</span><span class="s1"> variants for ancestry: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
        
        <span class="c1">#Extract relevant variants</span>
        <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
        <span class="o">--</span><span class="n">pfile</span> <span class="p">{</span><span class="n">REL7_PATH</span><span class="p">}</span><span class="o">/</span><span class="n">imputed_genotypes</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">/</span><span class="n">chr4_</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_release7_vwb</span> \
        <span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">.</span><span class="n">samplestoKeep</span> \
        <span class="o">--</span><span class="n">extract</span> <span class="nb">range</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span><span class="o">.</span><span class="p">{</span><span class="n">variant_class</span><span class="p">}</span><span class="o">.</span><span class="n">variantstoKeep</span><span class="o">.</span><span class="n">txt</span> \
        <span class="o">--</span><span class="n">recode</span> <span class="n">vcf</span><span class="o">-</span><span class="n">iid</span> \
        <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span><span class="o">.</span><span class="p">{</span><span class="n">variant_class</span><span class="p">}</span>
        
        <span class="c1"># Print the command to be executed (for debugging purposes)</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Running bgzip and tabix for </span><span class="si">{</span><span class="n">variant_class</span><span class="si">}</span><span class="s1"> variants for ancestry: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
        
        <span class="c1">## Bgzip and Tabix (zip and index the file)</span>
        <span class="o">!</span><span class="w"> </span>bgzip<span class="w"> </span>-f<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/<span class="o">{</span>ancestry<span class="o">}</span>_TMEM175.<span class="o">{</span>variant_class<span class="o">}</span>.vcf
        <span class="o">!</span><span class="w"> </span>tabix<span class="w"> </span>-f<span class="w"> </span>-p<span class="w"> </span>vcf<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/<span class="o">{</span>ancestry<span class="o">}</span>_TMEM175.<span class="o">{</span>variant_class<span class="o">}</span>.vcf.gz
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Running plink to extract exonic variants for ancestry: AAC
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb

Start time: Wed Jul 31 11:09:28 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50560 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1111 samples (656 females, 455 males; 1111 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.psam.
3986699 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.pvar.
1 binary phenotype loaded (285 cases, 801 controls).
--extract bed1: 3986417 variants excluded.
--keep: 1111 samples remaining.
1111 samples (656 females, 455 males; 1111 founders) remaining after main
filters.
285 cases and 801 controls remaining after main filters.
282 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.vcf ...
10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 11:09:29 2024
Running bgzip and tabix for exonic variants for ancestry: AAC
Running plink to extract nonsynonymous_lof variants for ancestry: AAC
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb

Start time: Wed Jul 31 11:09:30 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50514 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
1111 samples (656 females, 455 males; 1111 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.psam.
3986699 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr4_AAC_release7_vwb.pvar.
1 binary phenotype loaded (285 cases, 801 controls).
--extract bed1: 3986540 variants excluded.
--keep: 1111 samples remaining.
1111 samples (656 females, 455 males; 1111 founders) remaining after main
filters.
285 cases and 801 controls remaining after main filters.
159 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.vcf
... 1010111112131314151516161718181920202122222323242525262727282829303031323233333435353637373838394040414242434444454546474748494950505152525354545555565757585959606161626263646465666667676869697071717272737474757676777778797980818182838384848586868788888989909191929393949495969697989899done.
End time: Wed Jul 31 11:09:31 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AAC
Running plink to extract exonic variants for ancestry: AFR
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb

Start time: Wed Jul 31 11:09:31 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50513 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2643 samples (1163 females, 1480 males; 2643 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.psam.
4555051 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.pvar.
1 binary phenotype loaded (942 cases, 1679 controls).
--extract bed1: 4554744 variants excluded.
--keep: 2643 samples remaining.
2643 samples (1163 females, 1480 males; 2643 founders) remaining after main
filters.
942 cases and 1679 controls remaining after main filters.
307 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.vcf ...
1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 11:09:33 2024
Running bgzip and tabix for exonic variants for ancestry: AFR
Running plink to extract nonsynonymous_lof variants for ancestry: AFR
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb

Start time: Wed Jul 31 11:09:33 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50512 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2643 samples (1163 females, 1480 males; 2643 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.psam.
4555051 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr4_AFR_release7_vwb.pvar.
1 binary phenotype loaded (942 cases, 1679 controls).
--extract bed1: 4554888 variants excluded.
--keep: 2643 samples remaining.
2643 samples (1163 females, 1480 males; 2643 founders) remaining after main
filters.
942 cases and 1679 controls remaining after main filters.
163 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.vcf
... 1011111212131414151516171718191920202122222323242525262627282829303031313233333434353636373838393940414142424344444546464747484949505051525253535455555657575858596060616162636364656566666768686969707171727373747475767677777879798080818282838484858586878788888990909192929393949595969697989899done.
End time: Wed Jul 31 11:09:34 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AFR
Running plink to extract exonic variants for ancestry: AJ
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb

Start time: Wed Jul 31 11:09:35 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50544 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2655 samples (1007 females, 1648 males; 2655 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.psam.
1669760 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.pvar.
1 binary phenotype loaded (1292 cases, 411 controls).
--extract bed1: 1669638 variants excluded.
--keep: 2655 samples remaining.
2655 samples (1007 females, 1648 males; 2655 founders) remaining after main
filters.
1292 cases and 411 controls remaining after main filters.
122 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.vcf ...
10111213131415161718181920212222232425262727282930313132333435363637383940404142434445454647484950505152535454555657585959606162636364656667686869707172727374757677777879808181828384858686878889909091929394959596979899done.
End time: Wed Jul 31 11:09:36 2024
Running bgzip and tabix for exonic variants for ancestry: AJ
Running plink to extract nonsynonymous_lof variants for ancestry: AJ
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb

Start time: Wed Jul 31 11:09:36 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50512 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
2655 samples (1007 females, 1648 males; 2655 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.psam.
1669760 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr4_AJ_release7_vwb.pvar.
1 binary phenotype loaded (1292 cases, 411 controls).
--extract bed1: 1669693 variants excluded.
--keep: 2655 samples remaining.
2655 samples (1007 females, 1648 males; 2655 founders) remaining after main
filters.
1292 cases and 411 controls remaining after main filters.
67 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.vcf
... 101113141617192022232526282931323435373840414344464749505253555658596162646567687071737476777980828385868889919294959798done.
End time: Wed Jul 31 11:09:37 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AJ
Running plink to extract exonic variants for ancestry: AMR
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb

Start time: Wed Jul 31 11:09:38 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50511 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
646 samples (283 females, 363 males; 646 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.psam.
2407746 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.pvar.
1 binary phenotype loaded (458 cases, 155 controls).
--extract bed1: 2407570 variants excluded.
--keep: 646 samples remaining.
646 samples (283 females, 363 males; 646 founders) remaining after main
filters.
458 cases and 155 controls remaining after main filters.
176 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.vcf ...
1010111112131314141515161717181819192021212222232324252526262727282829303031313232333434353536363738383939404041424243434444454646474748484950505151525253535455555656575758595960606161626363646465656667676868696970717172727373747575767677777878798080818182828384848585868687888889899090919292939394949596969797989899done.
End time: Wed Jul 31 11:09:38 2024
Running bgzip and tabix for exonic variants for ancestry: AMR
Running plink to extract nonsynonymous_lof variants for ancestry: AMR
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb

Start time: Wed Jul 31 11:09:39 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50513 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
646 samples (283 females, 363 males; 646 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.psam.
2407746 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr4_AMR_release7_vwb.pvar.
1 binary phenotype loaded (458 cases, 155 controls).
--extract bed1: 2407656 variants excluded.
--keep: 646 samples remaining.
646 samples (283 females, 363 males; 646 founders) remaining after main
filters.
458 cases and 155 controls remaining after main filters.
90 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.vcf
... 101112131415161718202122232425262728303132333435363738404142434445464748505152535455565758606162636465666768707172737475767778808182838485868788909192939495969798done.
End time: Wed Jul 31 11:09:40 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AMR
Running plink to extract exonic variants for ancestry: CAS
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb

Start time: Wed Jul 31 11:09:40 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50514 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
903 samples (464 females, 439 males; 903 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.psam.
2200655 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.pvar.
1 binary phenotype loaded (552 cases, 343 controls).
--extract bed1: 2200489 variants excluded.
--keep: 903 samples remaining.
903 samples (464 females, 439 males; 903 founders) remaining after main
filters.
552 cases and 343 controls remaining after main filters.
166 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.vcf ...
1010111212131314151516161718181919202121222223242425252627272828293030313132333334343536363737383939404041424243434445454646474848495050515152535354545556565757585959606061626263636465656666676868696970717172727374747575767777787879808081818283838484858686878788898990909192929393949595969697989899done.
End time: Wed Jul 31 11:09:41 2024
Running bgzip and tabix for exonic variants for ancestry: CAS
Running plink to extract nonsynonymous_lof variants for ancestry: CAS
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb

Start time: Wed Jul 31 11:09:42 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50512 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
903 samples (464 females, 439 males; 903 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.psam.
2200655 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr4_CAS_release7_vwb.pvar.
1 binary phenotype loaded (552 cases, 343 controls).
--extract bed1: 2200562 variants excluded.
--keep: 903 samples remaining.
903 samples (464 females, 439 males; 903 founders) remaining after main
filters.
552 cases and 343 controls remaining after main filters.
93 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.vcf
... 1011121315161718192021222324252627293031323334353637383940414344454647484950515253545556585960616263646566676869707273747576777879808182838486878889909192939495969798done.
End time: Wed Jul 31 11:09:42 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAS
Running plink to extract exonic variants for ancestry: EAS
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb

Start time: Wed Jul 31 11:09:43 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50513 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
5167 samples (1889 females, 3278 males; 5167 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.psam.
3235052 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.pvar.
1 binary phenotype loaded (2662 cases, 2461 controls).
--extract bed1: 3234774 variants excluded.
--keep: 5167 samples remaining.
5167 samples (1889 females, 3278 males; 5167 founders) remaining after main
filters.
2662 cases and 2461 controls remaining after main filters.
278 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.vcf ...
10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 11:09:44 2024
Running bgzip and tabix for exonic variants for ancestry: EAS
Running plink to extract nonsynonymous_lof variants for ancestry: EAS
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb

Start time: Wed Jul 31 11:09:45 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50510 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
5167 samples (1889 females, 3278 males; 5167 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.psam.
3235052 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.pvar.
1 binary phenotype loaded (2662 cases, 2461 controls).
--extract bed1: 3234883 variants excluded.
--keep: 5167 samples remaining.
5167 samples (1889 females, 3278 males; 5167 founders) remaining after main
filters.
2662 cases and 2461 controls remaining after main filters.
169 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.vcf
... 1010111112131314141515161717181819202021212223232424252626272728282930303131323333343435363637373839394040414242434344444546464747484949505051525253535455555656575758595960606162626363646565666667686869697071717272737374757576767778787979808181828283848485858686878888898990919192929394949595969797989899done.
End time: Wed Jul 31 11:09:46 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EAS
Running plink to extract exonic variants for ancestry: EUR
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb

Start time: Wed Jul 31 11:09:47 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50515 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
38839 samples (15543 females, 23296 males; 38839 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.psam.
10419161 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.pvar.
1 binary phenotype loaded (21198 cases, 9214 controls).
--extract bed1: 10418228 variants excluded.
--keep: 38838 samples remaining.
38838 samples (15543 females, 23295 males; 38838 founders) remaining after main
filters.
21197 cases and 9214 controls remaining after main filters.
933 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf ...
1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 11:09:50 2024
Running bgzip and tabix for exonic variants for ancestry: EUR
Running plink to extract nonsynonymous_lof variants for ancestry: EUR
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb

Start time: Wed Jul 31 11:09:52 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50511 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
38839 samples (15543 females, 23296 males; 38839 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.psam.
10419161 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr4_EUR_release7_vwb.pvar.
1 binary phenotype loaded (21198 cases, 9214 controls).
--extract bed1: 10418592 variants excluded.
--keep: 38838 samples remaining.
38838 samples (15543 females, 23295 males; 38838 founders) remaining after main
filters.
21197 cases and 9214 controls remaining after main filters.
569 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.vcf
... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 11:09:54 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EUR
Running plink to extract exonic variants for ancestry: FIN
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb

Start time: Wed Jul 31 11:09:56 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50524 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
114 samples (61 females, 53 males; 114 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.psam.
891614 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.pvar.
1 binary phenotype loaded (98 cases, 8 controls).
--extract bed1: 891564 variants excluded.
--keep: 114 samples remaining.
114 samples (61 females, 53 males; 114 founders) remaining after main filters.
98 cases and 8 controls remaining after main filters.
50 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.vcf ...
101214161820222426283032343638404244464850525456586062646668707274767880828486889092949698done.
End time: Wed Jul 31 11:09:56 2024
Running bgzip and tabix for exonic variants for ancestry: FIN
Running plink to extract nonsynonymous_lof variants for ancestry: FIN
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb

Start time: Wed Jul 31 11:09:57 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50514 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
114 samples (61 females, 53 males; 114 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.psam.
891614 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr4_FIN_release7_vwb.pvar.
1 binary phenotype loaded (98 cases, 8 controls).
--extract bed1: 891591 variants excluded.
--keep: 114 samples remaining.
114 samples (61 females, 53 males; 114 founders) remaining after main filters.
98 cases and 8 controls remaining after main filters.
23 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.vcf
... 1317212630343943475256606569737882869195done.
End time: Wed Jul 31 11:09:57 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: FIN
Running plink to extract exonic variants for ancestry: MDE
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb

Start time: Wed Jul 31 11:09:58 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50514 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
581 samples (212 females, 369 males; 581 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.psam.
2161336 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.pvar.
1 binary phenotype loaded (311 cases, 225 controls).
--extract bed1: 2161160 variants excluded.
--keep: 581 samples remaining.
581 samples (212 females, 369 males; 581 founders) remaining after main
filters.
311 cases and 225 controls remaining after main filters.
176 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.vcf ...
1010111112131314141515161717181819192021212222232324252526262727282829303031313232333434353536363738383939404041424243434444454646474748484950505151525253535455555656575758595960606161626363646465656667676868696970717172727373747575767677777878798080818182828384848585868687888889899090919292939394949596969797989899done.
End time: Wed Jul 31 11:09:59 2024
Running bgzip and tabix for exonic variants for ancestry: MDE
Running plink to extract nonsynonymous_lof variants for ancestry: MDE
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb

Start time: Wed Jul 31 11:10:00 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50513 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
581 samples (212 females, 369 males; 581 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.psam.
2161336 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr4_MDE_release7_vwb.pvar.
1 binary phenotype loaded (311 cases, 225 controls).
--extract bed1: 2161251 variants excluded.
--keep: 581 samples remaining.
581 samples (212 females, 369 males; 581 founders) remaining after main
filters.
311 cases and 225 controls remaining after main filters.
85 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.vcf
... 10111214151617182021222324252728293031323435363738404142434445474849505152545556575860616263646567686970717274757677788081828384858788899091929495969798done.
End time: Wed Jul 31 11:10:00 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: MDE
Running plink to extract exonic variants for ancestry: SAS
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb

Start time: Wed Jul 31 11:10:01 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50501 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
635 samples (227 females, 408 males; 635 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.psam.
1933318 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.pvar.
1 binary phenotype loaded (387 cases, 222 controls).
--extract bed1: 1933176 variants excluded.
--keep: 635 samples remaining.
635 samples (227 females, 408 males; 635 founders) remaining after main
filters.
387 cases and 222 controls remaining after main filters.
142 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.vcf ...
10111112131414151616171819192021212223232425262627282829303031323333343535363738383940404142424344454546474748495050515252535454555657575859596061616263646465666667686969707171727373747576767778787980808182838384858586878888899090919292939495959697979899done.
End time: Wed Jul 31 11:10:02 2024
Running bgzip and tabix for exonic variants for ancestry: SAS
Running plink to extract nonsynonymous_lof variants for ancestry: SAS
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb

Start time: Wed Jul 31 11:10:02 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50510 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
635 samples (227 females, 408 males; 635 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.psam.
1933318 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr4_SAS_release7_vwb.pvar.
1 binary phenotype loaded (387 cases, 222 controls).
--extract bed1: 1933239 variants excluded.
--keep: 635 samples remaining.
635 samples (227 females, 408 males; 635 founders) remaining after main
filters.
387 cases and 222 controls remaining after main filters.
79 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.vcf
... 1011121315161718202122242526272930313234353637394041434445464849505153545556585960626364656768697072737475777879818283848687888991929394969798done.
End time: Wed Jul 31 11:10:03 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: SAS
Running plink to extract exonic variants for ancestry: CAH
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb

Start time: Wed Jul 31 11:10:04 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50507 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
851 samples (405 females, 446 males; 851 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.psam.
3678288 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.pvar.
1 binary phenotype loaded (525 cases, 302 controls).
--extract bed1: 3678031 variants excluded.
--keep: 851 samples remaining.
851 samples (405 females, 446 males; 851 founders) remaining after main
filters.
525 cases and 302 controls remaining after main filters.
257 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.vcf ...
1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.
End time: Wed Jul 31 11:10:05 2024
Running bgzip and tabix for exonic variants for ancestry: CAH
Running plink to extract nonsynonymous_lof variants for ancestry: CAH
PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.log.
Options in effect:
  --export vcf-iid
  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.variantstoKeep.txt
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.samplestoKeep
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb

Start time: Wed Jul 31 11:10:05 2024
Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.
52223 MiB RAM detected, ~50512 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
851 samples (405 females, 446 males; 851 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.psam.
3678288 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr4_CAH_release7_vwb.pvar.
1 binary phenotype loaded (525 cases, 302 controls).
--extract bed1: 3678144 variants excluded.
--keep: 851 samples remaining.
851 samples (405 females, 446 males; 851 founders) remaining after main
filters.
525 cases and 302 controls remaining after main filters.
144 variants remaining after main filters.
--export vcf to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.vcf
... 101111121313141515161718181920202122222324252526272728292930313132333434353636373838394040414243434445454647474849505051525253545455565657585959606161626363646565666768686970707172727374757576777778797980818182838484858686878888899090919293939495959697979899done.
End time: Wed Jul 31 11:10:06 2024
Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAH
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1df08dc0-1dc4-4d35-af7f-4230011700bc">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[4]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Run RVtests</span>
<span class="n">ancestries</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'AAC'</span><span class="p">,</span> <span class="s1">'AFR'</span><span class="p">,</span> <span class="s1">'AJ'</span><span class="p">,</span> <span class="s1">'AMR'</span><span class="p">,</span> <span class="s1">'CAS'</span><span class="p">,</span> <span class="s1">'EAS'</span><span class="p">,</span> <span class="s1">'EUR'</span><span class="p">,</span> <span class="s1">'FIN'</span><span class="p">,</span> <span class="s1">'MDE'</span><span class="p">,</span> <span class="s1">'SAS'</span><span class="p">,</span> <span class="s1">'CAH'</span><span class="p">]</span>
<span class="n">variant_classes</span> <span class="o">=</span> <span class="p">[</span><span class="s1">'exonic'</span><span class="p">,</span> <span class="s1">'nonsynonymous_lof'</span><span class="p">]</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    <span class="k">for</span> <span class="n">variant_class</span> <span class="ow">in</span> <span class="n">variant_classes</span><span class="p">:</span>
        
        <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
        
        <span class="c1"># Print the command to be executed (for debugging purposes)</span>
        <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'Running RVtests for </span><span class="si">{</span><span class="n">variant_class</span><span class="si">}</span><span class="s1"> variants for ancestry: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
        
        <span class="c1">## RVtests with covariates </span>
        <span class="c1">#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex</span>
        <span class="c1">#The pheno-name flag only works when the pheno/covar file is structured properly</span>
        <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/rvtests/executable/rvtest<span class="w"> </span>--noweb<span class="w"> </span>--hide-covar<span class="w"> </span><span class="err">\</span>
        <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span><span class="o">.</span><span class="n">burden</span><span class="o">.</span><span class="p">{</span><span class="n">variant_class</span><span class="p">}</span> \
        <span class="o">--</span><span class="n">kernel</span> <span class="n">skato</span> \
        <span class="o">--</span><span class="n">inVcf</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span><span class="o">.</span><span class="p">{</span><span class="n">variant_class</span><span class="p">}</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
        <span class="o">--</span><span class="n">pheno</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_covariate_file</span><span class="o">.</span><span class="n">txt</span> \
        <span class="o">--</span><span class="n">pheno</span><span class="o">-</span><span class="n">name</span> <span class="n">PHENO</span> \
        <span class="o">--</span><span class="n">gene</span> <span class="n">TMEM175</span> \
        <span class="o">--</span><span class="n">geneFile</span> <span class="o">~/</span><span class="n">workspace</span><span class="o">/</span><span class="n">ws_files</span><span class="o">/</span><span class="n">TMEM175</span><span class="o">/</span><span class="n">refFlat</span><span class="o">.</span><span class="n">txt</span> \
        <span class="o">--</span><span class="n">covar</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_covariate_file</span><span class="o">.</span><span class="n">txt</span> \
        <span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">name</span> <span class="n">SEX</span><span class="p">,</span><span class="n">AGE</span><span class="p">,</span><span class="n">PC1</span><span class="p">,</span><span class="n">PC2</span><span class="p">,</span><span class="n">PC3</span><span class="p">,</span><span class="n">PC4</span><span class="p">,</span><span class="n">PC5</span> \
        <span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span>
<span class="c1"># --burden cmc,zeggini,mb,fp,cmcWald --kernel skat,skato \</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Running RVtests for exonic variants for ancestry: AAC
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:45 2024
[INFO]	Loaded [ 1111 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57045 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 1154 ] sample phenotypes
[INFO]	Discard [ 76 ] samples as they do not have genotypes
[WARN]	Total [ 33 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BLAACPD-UC_000007_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004696_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004698_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007392_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007401_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007710_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008604_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008724_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008765_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008863_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 23 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 2425 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2433 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2487 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2498 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2553 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2560 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2570 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2658 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2701 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2753 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 153 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57131 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57121 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 153 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ BCM_000142_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000150_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000204_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000215_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000270_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000277_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000287_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000375_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000418_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000470_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 143 ] samples due to missing covariate.
[INFO]	Loaded 378 male, 547 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt
[INFO]	Loaded 214 cases, 711 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 925 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 57 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:44:47 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: AAC
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:48 2024
[INFO]	Loaded [ 1111 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57045 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 1154 ] sample phenotypes
[INFO]	Discard [ 76 ] samples as they do not have genotypes
[WARN]	Total [ 33 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BLAACPD-UC_000007_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004696_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004698_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007392_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007401_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007710_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008604_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008724_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008765_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008863_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 23 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 2425 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2433 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2487 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2498 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2553 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2560 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2570 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2658 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2701 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2753 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 153 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57131 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57121 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 153 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ BCM_000142_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000150_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000204_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000215_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000270_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000277_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000287_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000375_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000418_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000470_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 143 ] samples due to missing covariate.
[INFO]	Loaded 378 male, 547 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_covariate_file.txt
[INFO]	Loaded 214 cases, 711 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 925 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 38 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:44:49 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: AFR
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:49 2024
[INFO]	Loaded [ 2643 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 55477 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 2722 ] sample phenotypes
[INFO]	Discard [ 165 ] samples as they do not have genotypes
[WARN]	Total [ 86 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BLAACPD-LSU_000020_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BLAACPD-LSU_000021_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006426_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006574_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006582_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006611_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007385_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007386_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007387_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007389_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 76 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 2379 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2456 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2507 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2569 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2598 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2641 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2685 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2698 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2707 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2896 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 1495 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 55652 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 55642 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 1495 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ BCM_000096_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000173_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000224_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000286_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000315_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000358_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000402_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000415_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000424_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000613_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 1485 ] samples due to missing covariate.
[INFO]	Loaded 444 male, 618 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
[INFO]	Loaded 188 cases, 874 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 1062 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 60 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:44:51 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: AFR
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:51 2024
[INFO]	Loaded [ 2643 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 55477 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 2722 ] sample phenotypes
[INFO]	Discard [ 165 ] samples as they do not have genotypes
[WARN]	Total [ 86 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BLAACPD-LSU_000020_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BLAACPD-LSU_000021_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006426_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006574_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006582_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006611_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007385_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007386_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007387_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_007389_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 76 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 2379 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2456 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2507 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2569 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2598 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2641 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2685 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2698 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2707 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2896 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 1495 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 55652 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 55642 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 1495 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ BCM_000096_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000173_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000224_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000286_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000315_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000358_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000402_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000415_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000424_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000613_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 1485 ] samples due to missing covariate.
[INFO]	Loaded 444 male, 618 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_covariate_file.txt
[INFO]	Loaded 188 cases, 874 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 1062 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 38 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:44:53 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: AJ
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:53 2024
[INFO]	Loaded [ 2655 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 56397 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 1802 ] sample phenotypes
[INFO]	Discard [ 114 ] samples as they do not have genotypes
[WARN]	Total [ 967 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000024_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000027_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000103_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000168_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000474_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000478_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000548_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000567_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000651_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000688_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 957 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 21 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 475 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 486 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 646 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 805 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1056 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1063 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1423 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1470 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1473 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 278 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 56521 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 56511 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 278 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000021_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000475_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000486_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000646_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000808_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001059_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001066_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001426_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001474_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001477_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 268 ] samples due to missing covariate.
[INFO]	Loaded 940 male, 470 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt
[INFO]	Loaded 1187 cases, 223 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 1410 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 20 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:44:55 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: AJ
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:55 2024
[INFO]	Loaded [ 2655 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 56397 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 1802 ] sample phenotypes
[INFO]	Discard [ 114 ] samples as they do not have genotypes
[WARN]	Total [ 967 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000024_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000027_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000103_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000168_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000474_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000478_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000548_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000567_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000651_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000688_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 957 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 21 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 475 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 486 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 646 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 805 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1056 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1063 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1423 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1470 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1473 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 278 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 56521 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 56511 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 278 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000021_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000475_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000486_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000646_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000808_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001059_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001066_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001426_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001474_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001477_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 268 ] samples due to missing covariate.
[INFO]	Loaded 940 male, 470 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_covariate_file.txt
[INFO]	Loaded 1187 cases, 223 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 1410 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 13 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:44:56 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: AMR
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:57 2024
[INFO]	Loaded [ 646 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57562 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 637 ] sample phenotypes
[INFO]	Discard [ 36 ] samples as they do not have genotypes
[WARN]	Total [ 45 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000070_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000452_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_000771_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_000892_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004251_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008658_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008712_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008747_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008788_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008819_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 35 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 157 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 551 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 609 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 978 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1000 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1554 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1730 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1782 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2008 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2157 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 154 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57608 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57598 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 154 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000157_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000551_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000609_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000981_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001003_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ AUTH_000081_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000167_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000219_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000446_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000595_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 144 ] samples due to missing covariate.
[INFO]	Loaded 249 male, 198 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt
[INFO]	Loaded 344 cases, 103 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 447 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 36 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:44:58 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: AMR
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:44:59 2024
[INFO]	Loaded [ 646 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57562 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 637 ] sample phenotypes
[INFO]	Discard [ 36 ] samples as they do not have genotypes
[WARN]	Total [ 45 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000070_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000452_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_000771_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_000892_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004251_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008658_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008712_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008747_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008788_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008819_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 35 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 157 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 551 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 609 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 978 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1000 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1554 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1730 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1782 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2008 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2157 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 154 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57608 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57598 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 154 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000157_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000551_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000609_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000981_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001003_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ AUTH_000081_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000167_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000219_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000446_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BBDP_000595_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 144 ] samples due to missing covariate.
[INFO]	Loaded 249 male, 198 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_covariate_file.txt
[INFO]	Loaded 344 cases, 103 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 447 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 18 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:00 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: CAS
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:00 2024
[INFO]	Loaded [ 903 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57275 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 924 ] sample phenotypes
[INFO]	Discard [ 86 ] samples as they do not have genotypes
[WARN]	Total [ 65 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ CATPD_000009_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000022_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000054_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000103_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000216_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000302_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000303_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000349_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000364_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000365_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 55 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 443 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1132 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1256 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2362 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2424 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2554 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2766 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2788 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 3629 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 19004 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 180 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57371 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57361 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 180 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000443_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001135_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001259_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000079_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000141_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000271_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000483_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000505_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ CALYPSO_000091_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ JGUM_000144_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 170 ] samples due to missing covariate.
[INFO]	Loaded 273 male, 385 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt
[INFO]	Loaded 425 cases, 233 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 658 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 30 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:02 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: CAS
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:02 2024
[INFO]	Loaded [ 903 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57275 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 924 ] sample phenotypes
[INFO]	Discard [ 86 ] samples as they do not have genotypes
[WARN]	Total [ 65 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ CATPD_000009_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000022_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000054_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000103_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000216_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000302_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000303_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000349_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000364_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000365_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 55 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 443 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1132 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1256 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2362 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2424 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2554 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2766 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2788 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 3629 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 19004 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 180 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57371 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57361 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 180 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000443_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001135_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001259_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000079_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000141_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000271_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000483_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000505_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ CALYPSO_000091_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ JGUM_000144_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 170 ] samples due to missing covariate.
[INFO]	Loaded 273 male, 385 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_covariate_file.txt
[INFO]	Loaded 425 cases, 233 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 658 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 22 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:04 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: EAS
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:05 2024
[INFO]	Loaded [ 5167 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 52944 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 5255 ] sample phenotypes
[INFO]	Discard [ 337 ] samples as they do not have genotypes
[WARN]	Total [ 249 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ CORIELL_008702_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008775_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008810_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008867_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008922_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008940_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008942_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008999_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009025_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ DODONA_000655_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 239 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 51 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 111 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 113 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 135 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 186 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 201 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 204 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 263 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 314 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 415 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 2395 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 53291 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 53281 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 2395 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000051_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000111_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000113_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000135_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000186_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000201_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000204_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000263_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000314_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000415_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 2385 ] samples due to missing covariate.
[INFO]	Loaded 1313 male, 1210 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
[INFO]	Loaded 1253 cases, 1270 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 2523 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 45 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:06 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: EAS
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:07 2024
[INFO]	Loaded [ 5167 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 52944 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 5255 ] sample phenotypes
[INFO]	Discard [ 337 ] samples as they do not have genotypes
[WARN]	Total [ 249 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ CORIELL_008702_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008775_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008810_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008867_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008922_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008940_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008942_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008999_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009025_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ DODONA_000655_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 239 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 51 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 111 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 113 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 135 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 186 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 201 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 204 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 263 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 314 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 415 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 2395 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 53291 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 53281 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 2395 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000051_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000111_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000113_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000135_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000186_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000201_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000204_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000263_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000314_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000415_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 2385 ] samples due to missing covariate.
[INFO]	Loaded 1313 male, 1210 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
[INFO]	Loaded 1253 cases, 1270 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 2523 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 31 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:08 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: EUR
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:09 2024
[INFO]	Loaded [ 38838 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 16 [ 0	APGS_000016_s1	0	0	NA	NA	NA	-30.73116188775354	-35.70956669045879	-8.981065917918896	-8.204220280633406	21.16466107403938 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 19 [ 0	APGS_000019_s1	0	0	NA	NA	NA	-32.85276254867941	-34.97747925899874	-12.46936779285856	-7.942340016898765	22.92381260558889 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 21 [ 0	APGS_000021_s1	0	0	NA	NA	NA	-28.028682074030776	-35.73536300551804	-6.580636620691621	-17.501411400398084	-23.026102052263894 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 51 [ 0	APGS_000051_s1	0	0	NA	NA	NA	-11.099685189731892	83.97372193838893	-10.188699145918108	-28.239732867127746	1.5121308708705503 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 62 [ 0	APGS_000062_s1	0	0	NA	NA	NA	-18.64307786414933	10.44804060538148	56.45591899943591	19.558079798702437	2.613308352009664 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 81 [ 0	APGS_000081_s1	0	0	NA	NA	NA	-28.50152850143756	-32.79522188762641	-11.344796408436856	-9.137528317277589	19.6391880615385 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 111 [ 0	APGS_000111_s1	0	0	NA	NA	NA	-11.020377609784008	69.20196087521234	2.908328982593293	-15.95903093667702	-1.92510776452849 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 113 [ 0	APGS_000113_s1	0	0	NA	NA	NA	-12.69193179633094	85.44070750810215	-10.451702135643112	-23.738136439277177	0.2653173025631583 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 114 [ 0	APGS_000114_s1	0	0	NA	NA	NA	-8.644799444122128	-0.4546406081980039	5.388368663597835	-2.62342306022476	12.250260673192534 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 119 [ 0	APGS_000119_s1	0	0	NA	NA	NA	-17.346120495361607	11.654071761099145	54.21353563180626	17.49025406831134	-0.8995341127177866 ] ... 
[WARN]	Skip: Additional [ 27087 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 31112 ] sample phenotypes
[INFO]	Discard [ 1356 ] samples as they do not have genotypes
[WARN]	Total [ 9082 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ APGS_000016_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ APGS_001460_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ AUTH_000082_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000009_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000013_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000015_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000019_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000020_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000023_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000026_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 9072 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 2 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 3 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 4 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 5 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 6 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 7 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 8 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 9 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 10 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 11 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 8838 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 28453 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000016_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000019_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000021_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000035_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000051_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000062_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000081_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000086_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000111_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000113_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000114_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 28443 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 8838 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000002_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000003_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000004_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000008_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000011_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000018_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000023_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000026_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000027_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000028_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 8828 ] samples due to missing covariate.
[INFO]	Loaded 12421 male, 8497 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
[INFO]	Loaded 15111 cases, 5807 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 20918 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 174 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:13 2024
[INFO]	Analysis took 4 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: EUR
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:14 2024
[INFO]	Loaded [ 38838 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 16 [ 0	APGS_000016_s1	0	0	NA	NA	NA	-30.73116188775354	-35.70956669045879	-8.981065917918896	-8.204220280633406	21.16466107403938 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 19 [ 0	APGS_000019_s1	0	0	NA	NA	NA	-32.85276254867941	-34.97747925899874	-12.46936779285856	-7.942340016898765	22.92381260558889 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 21 [ 0	APGS_000021_s1	0	0	NA	NA	NA	-28.028682074030776	-35.73536300551804	-6.580636620691621	-17.501411400398084	-23.026102052263894 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 51 [ 0	APGS_000051_s1	0	0	NA	NA	NA	-11.099685189731892	83.97372193838893	-10.188699145918108	-28.239732867127746	1.5121308708705503 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 62 [ 0	APGS_000062_s1	0	0	NA	NA	NA	-18.64307786414933	10.44804060538148	56.45591899943591	19.558079798702437	2.613308352009664 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 81 [ 0	APGS_000081_s1	0	0	NA	NA	NA	-28.50152850143756	-32.79522188762641	-11.344796408436856	-9.137528317277589	19.6391880615385 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 111 [ 0	APGS_000111_s1	0	0	NA	NA	NA	-11.020377609784008	69.20196087521234	2.908328982593293	-15.95903093667702	-1.92510776452849 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 113 [ 0	APGS_000113_s1	0	0	NA	NA	NA	-12.69193179633094	85.44070750810215	-10.451702135643112	-23.738136439277177	0.2653173025631583 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 114 [ 0	APGS_000114_s1	0	0	NA	NA	NA	-8.644799444122128	-0.4546406081980039	5.388368663597835	-2.62342306022476	12.250260673192534 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 119 [ 0	APGS_000119_s1	0	0	NA	NA	NA	-17.346120495361607	11.654071761099145	54.21353563180626	17.49025406831134	-0.8995341127177866 ] ... 
[WARN]	Skip: Additional [ 27087 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 31112 ] sample phenotypes
[INFO]	Discard [ 1356 ] samples as they do not have genotypes
[WARN]	Total [ 9082 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ APGS_000016_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ APGS_001460_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ AUTH_000082_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000009_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000013_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000015_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000019_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000020_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000023_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000026_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 9072 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 2 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 3 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 4 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 5 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 6 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 7 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 8 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 9 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 10 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 11 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 8838 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 28453 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000016_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000019_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000021_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000035_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000051_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000062_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000081_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000086_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000111_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000113_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000114_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 28443 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 8838 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000002_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000003_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000004_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000008_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000011_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000018_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000023_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000026_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000027_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000028_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 8828 ] samples due to missing covariate.
[INFO]	Loaded 12421 male, 8497 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
[INFO]	Loaded 15111 cases, 5807 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 20918 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 105 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:18 2024
[INFO]	Analysis took 4 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: FIN
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:18 2024
[INFO]	Loaded [ 114 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 58091 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 108 ] sample phenotypes
[INFO]	Discard [ 5 ] samples as they do not have genotypes
[WARN]	Total [ 11 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000507_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_000183_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004505_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008587_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008885_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008956_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ REGARDS_000882_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ REGARDS_002338_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ REGARDS_005749_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ SYNAPS-KZ_000878_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 1 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 140 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 28739 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 40171 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48859 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48861 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48863 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48871 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48877 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48878 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48880 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 63 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 58106 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 58096 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 63 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000140_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ PAGE_002458_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ PROSPOS_000093_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000438_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000440_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000442_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000450_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000456_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000457_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000459_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 53 ] samples due to missing covariate.
[INFO]	Loaded 21 male, 19 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt
[INFO]	Loaded 36 cases, 4 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 40 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 2 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:19 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: FIN
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:20 2024
[INFO]	Loaded [ 114 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 58091 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 108 ] sample phenotypes
[INFO]	Discard [ 5 ] samples as they do not have genotypes
[WARN]	Total [ 11 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000507_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_000183_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004505_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008587_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008885_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008956_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ REGARDS_000882_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ REGARDS_002338_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ REGARDS_005749_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ SYNAPS-KZ_000878_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 1 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 140 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 28739 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 40171 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48859 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48861 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48863 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48871 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48877 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48878 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 48880 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 63 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 58106 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 58096 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 63 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000140_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ PAGE_002458_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ PROSPOS_000093_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000438_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000440_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000442_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000450_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000456_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000457_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ SYNAPS-KZ_000459_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 53 ] samples due to missing covariate.
[INFO]	Loaded 21 male, 19 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_covariate_file.txt
[INFO]	Loaded 36 cases, 4 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 40 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 2 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:21 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: MDE
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:21 2024
[INFO]	Loaded [ 581 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57656 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 543 ] sample phenotypes
[INFO]	Discard [ 20 ] samples as they do not have genotypes
[WARN]	Total [ 58 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ CATPD_000026_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000027_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000046_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000426_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000427_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000435_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000440_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_002630_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004549_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008609_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 48 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 259 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 450 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 533 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1029 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1049 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1082 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1152 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1372 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1504 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1520 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 144 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57686 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57676 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 144 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000259_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000450_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000533_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001032_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001052_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001085_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001155_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001375_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ AUTH_000029_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ AUTH_000045_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 134 ] samples due to missing covariate.
[INFO]	Loaded 240 male, 139 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt
[INFO]	Loaded 174 cases, 205 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 379 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 30 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:24 2024
[INFO]	Analysis took 3 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: MDE
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:25 2024
[INFO]	Loaded [ 581 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57656 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 543 ] sample phenotypes
[INFO]	Discard [ 20 ] samples as they do not have genotypes
[WARN]	Total [ 58 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ CATPD_000026_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000027_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000046_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000426_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000427_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000435_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CATPD_000440_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_002630_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_004549_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008609_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 48 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 259 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 450 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 533 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1029 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1049 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1082 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1152 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1372 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1504 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1520 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 144 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57686 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57676 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 144 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000259_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000450_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000533_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001032_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001052_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001085_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001155_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001375_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ AUTH_000029_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ AUTH_000045_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 134 ] samples due to missing covariate.
[INFO]	Loaded 240 male, 139 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_covariate_file.txt
[INFO]	Loaded 174 cases, 205 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 379 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 15 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:27 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: SAS
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:27 2024
[INFO]	Loaded [ 635 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57604 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 595 ] sample phenotypes
[INFO]	Discard [ 24 ] samples as they do not have genotypes
[WARN]	Total [ 64 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BCM_000801_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_003435_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008898_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008971_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008985_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009004_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009035_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009060_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ m-KUL_000003_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ m-KUL_000023_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 54 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 62 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 119 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 331 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 378 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 454 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 502 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 526 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 604 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 717 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 748 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 275 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57638 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57628 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 275 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000062_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000119_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000331_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000378_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000454_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000502_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000526_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000604_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000719_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000751_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 265 ] samples due to missing covariate.
[INFO]	Loaded 176 male, 120 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt
[INFO]	Loaded 170 cases, 126 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 296 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 17 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:29 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: SAS
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:29 2024
[INFO]	Loaded [ 635 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57604 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 595 ] sample phenotypes
[INFO]	Discard [ 24 ] samples as they do not have genotypes
[WARN]	Total [ 64 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BCM_000801_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_003435_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008898_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008971_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_008985_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009004_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009035_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_009060_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ m-KUL_000003_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ m-KUL_000023_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 54 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 62 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 119 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 331 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 378 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 454 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 502 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 526 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 604 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 717 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 748 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 275 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57638 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57628 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 275 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000062_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000119_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000331_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000378_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000454_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000502_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000526_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000604_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000719_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000751_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 265 ] samples due to missing covariate.
[INFO]	Loaded 176 male, 120 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_covariate_file.txt
[INFO]	Loaded 170 cases, 126 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 296 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 12 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:31 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for exonic variants for ancestry: CAH
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.burden.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.burden.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:31 2024
[INFO]	Loaded [ 851 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57146 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 1053 ] sample phenotypes
[INFO]	Discard [ 249 ] samples as they do not have genotypes
[WARN]	Total [ 47 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000585_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BIOFIND_000151_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_001486_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_002530_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006661_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006663_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006666_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006670_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006705_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006708_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 37 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 114 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 299 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 386 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 416 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 682 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 722 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 848 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1045 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1287 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2410 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 111 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57405 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57395 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 111 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000114_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000299_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000386_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000416_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000684_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000724_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000851_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001048_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001290_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000127_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 101 ] samples due to missing covariate.
[INFO]	Loaded 364 male, 329 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt
[INFO]	Loaded 422 cases, 271 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 693 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 43 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:33 2024
[INFO]	Analysis took 2 seconds
RVTESTS finished successfully
Running RVtests for nonsynonymous_lof variants for ancestry: CAH
Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.burden.nonsynonymous_lof]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [], --vt [], --kernel [skato]
                          --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.burden.nonsynonymous_lof
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt
    --pheno-name PHENO
    --kernel skato
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 12:45:33 2024
[INFO]	Loaded [ 851 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 2 [ 0	APGS_000002_s1	0	0	NA	NA	NA	-29.483298155783107	-38.47300456340984	-9.39890353917677	-10.84286522751987	25.744648136070445 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 3 [ 0	APGS_000003_s1	0	0	NA	NA	NA	-33.0689577343486	-34.28886163706171	-11.116950676466551	-8.056146283429992	22.924043088964268 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 4 [ 0	APGS_000004_s1	0	0	NA	NA	NA	-28.204440958420165	-30.969480686953425	-15.883093043436062	-7.004351236034862	18.64700675659124 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 5 [ 0	APGS_000005_s1	0	0	NA	NA	NA	-35.039753178343084	-39.56582208982667	-12.522752975129649	-9.339178860899043	21.95653300811111 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 6 [ 0	APGS_000006_s1	0	0	NA	NA	NA	-30.27990655722104	-34.988904787169055	-10.664628191675195	-7.407177267281592	26.357547504611546 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 7 [ 0	APGS_000007_s1	0	0	NA	NA	NA	-32.72563807076602	-36.35915760895503	-11.727021032740998	-3.411258951209475	24.64853681070508 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 8 [ 0	APGS_000008_s1	0	0	NA	NA	NA	-33.30320473306731	-35.52791248433792	-10.818936542850327	-12.897919754817568	6.851688108549393 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 9 [ 0	APGS_000009_s1	0	0	NA	NA	NA	-33.7692580908843	-35.779988070994435	-9.513244142204664	-10.921645737258968	12.102471391324084 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 10 [ 0	APGS_000010_s1	0	0	NA	NA	NA	-34.14916338338358	-37.06667840865392	-12.846488455854802	-6.953801969476714	22.46577687418317 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 11 [ 0	APGS_000011_s1	0	0	NA	NA	NA	-35.69827789703175	-34.94087795568433	-11.376420692830346	-5.590933798882794	24.57894017349767 ] ... 
[WARN]	Skip: Additional [ 57146 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 1053 ] sample phenotypes
[INFO]	Discard [ 249 ] samples as they do not have genotypes
[WARN]	Total [ 47 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ BBDP_000585_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BIOFIND_000151_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_001486_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_002530_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006661_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006663_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006666_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006670_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006705_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ CORIELL_006708_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 37 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 114 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 299 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 386 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 416 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 682 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 722 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 848 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1045 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 1287 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 2410 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 111 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 57405 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000002_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000003_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000004_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000005_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000006_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000007_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000008_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000009_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000010_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000011_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000012_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 57395 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 111 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000114_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000299_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000386_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000416_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000684_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000724_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000851_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001048_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_001290_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ BCM_000127_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 101 ] samples due to missing covariate.
[INFO]	Loaded 364 male, 329 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_covariate_file.txt
[INFO]	Loaded 422 cases, 271 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 693 ] samples...
[INFO]	SKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 26 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 12:45:34 2024
[INFO]	Analysis took 1 seconds
RVTESTS finished successfully
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=d9690b20-309e-48b8-99fe-59e28abacb2e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>EUR</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=840337d9-30fa-4123-aef4-c003496b088d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[5]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175"</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ce2ab430-4375-40ed-a375-7d9e97d39449">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[6]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EUR all_coding variant results</span>
<span class="c1">## next step--do burden test</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	323773	0.9	0.0304464
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=7a2fac39-db8d-42ee-9137-ec6ff5c630a7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex</span>
        <span class="c1">#The pheno-name flag only works when the pheno/covar file is structured properly</span>
<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/rvtests/executable/rvtest<span class="w"> </span>--noweb<span class="w"> </span>--hide-covar<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR_TMEM175</span><span class="o">.</span><span class="n">burdenTEST</span><span class="o">.</span><span class="n">exonic</span> \
<span class="o">--</span><span class="n">burden</span> <span class="n">cmc</span><span class="p">,</span><span class="n">zeggini</span><span class="p">,</span><span class="n">mb</span><span class="p">,</span><span class="n">fp</span><span class="p">,</span><span class="n">cmcWald</span> \
<span class="o">--</span><span class="n">inVcf</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR_TMEM175</span><span class="o">.</span><span class="n">exonic</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> \
<span class="o">--</span><span class="n">pheno</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR_covariate_file</span><span class="o">.</span><span class="n">txt</span> \
<span class="o">--</span><span class="n">pheno</span><span class="o">-</span><span class="n">name</span> <span class="n">PHENO</span> \
<span class="o">--</span><span class="n">gene</span> <span class="n">TMEM175</span> \
<span class="o">--</span><span class="n">geneFile</span> <span class="o">~/</span><span class="n">workspace</span><span class="o">/</span><span class="n">ws_files</span><span class="o">/</span><span class="n">TMEM175</span><span class="o">/</span><span class="n">refFlat</span><span class="o">.</span><span class="n">txt</span> \
<span class="o">--</span><span class="n">covar</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR_covariate_file</span><span class="o">.</span><span class="n">txt</span> \
<span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">name</span> <span class="n">SEX</span><span class="p">,</span><span class="n">AGE</span><span class="p">,</span><span class="n">PC1</span><span class="p">,</span><span class="n">PC2</span><span class="p">,</span><span class="n">PC3</span><span class="p">,</span><span class="n">PC4</span><span class="p">,</span><span class="n">PC5</span> \
<span class="o">--</span><span class="n">freqUpper</span> <span class="mf">0.01</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)
  For documentations, refer to http://zhanxw.github.io/rvtests/
  For questions and comments, plase send to Xiaowei Zhan &lt;zhanxw@umich.edu&gt;
  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf.gz]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic]
                          --outputRaw
       Specify Covariate:
                          --covar [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt]
                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5], --sex
       Specify Phenotype:
                          --pheno [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt]
                          --inverseNormal, --useResidualAsPhenotype, --mpheno []
                          --pheno-name [PHENO], --qtl, --multiplePheno []
        Specify Genotype: --dosage [], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [], --burden [cmc,zeggini,mb,fp,cmcWald]
                          --vt [], --kernel [], --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit :
                          --geneFile [/home/jupyter/workspace/ws_files/TMEM175/refFlat.txt]
                          --gene [TMEM175], --setList [], --setFile [], --set []
        Frequency Cutoff: --freqUpper [0.01], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition []
     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []
                          --outputID, --help

Effective Options
    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf.gz
    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic
    --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
    --pheno /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
    --pheno-name PHENO
    --burden cmc,zeggini,mb,fp,cmcWald
    --geneFile /home/jupyter/workspace/ws_files/TMEM175/refFlat.txt
    --gene TMEM175
    --freqUpper 0.01
    --noweb
    --hide-covar

[INFO]	Program version: 20190205
[INFO]	Analysis started at: Wed Aug 28 13:00:39 2024
[INFO]	Loaded [ 38838 ] samples from genotype files
[WARN]	Skip: Missing or invalid phenotype type, skipping line 16 [ 0	APGS_000016_s1	0	0	NA	NA	NA	-30.73116188775354	-35.70956669045879	-8.981065917918896	-8.204220280633406	21.16466107403938 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 19 [ 0	APGS_000019_s1	0	0	NA	NA	NA	-32.85276254867941	-34.97747925899874	-12.46936779285856	-7.942340016898765	22.92381260558889 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 21 [ 0	APGS_000021_s1	0	0	NA	NA	NA	-28.028682074030776	-35.73536300551804	-6.580636620691621	-17.501411400398084	-23.026102052263894 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 51 [ 0	APGS_000051_s1	0	0	NA	NA	NA	-11.099685189731892	83.97372193838893	-10.188699145918108	-28.239732867127746	1.5121308708705503 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 62 [ 0	APGS_000062_s1	0	0	NA	NA	NA	-18.64307786414933	10.44804060538148	56.45591899943591	19.558079798702437	2.613308352009664 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 81 [ 0	APGS_000081_s1	0	0	NA	NA	NA	-28.50152850143756	-32.79522188762641	-11.344796408436856	-9.137528317277589	19.6391880615385 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 111 [ 0	APGS_000111_s1	0	0	NA	NA	NA	-11.020377609784008	69.20196087521234	2.908328982593293	-15.95903093667702	-1.92510776452849 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 113 [ 0	APGS_000113_s1	0	0	NA	NA	NA	-12.69193179633094	85.44070750810215	-10.451702135643112	-23.738136439277177	0.2653173025631583 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 114 [ 0	APGS_000114_s1	0	0	NA	NA	NA	-8.644799444122128	-0.4546406081980039	5.388368663597835	-2.62342306022476	12.250260673192534 ] ... 
[WARN]	Skip: Missing or invalid phenotype type, skipping line 119 [ 0	APGS_000119_s1	0	0	NA	NA	NA	-17.346120495361607	11.654071761099145	54.21353563180626	17.49025406831134	-0.8995341127177866 ] ... 
[WARN]	Skip: Additional [ 27087 ] lines have missing or invalid phenotype type
[INFO]	Loaded [ 31112 ] sample phenotypes
[INFO]	Discard [ 1356 ] samples as they do not have genotypes
[WARN]	Total [ 9082 ] samples are dropped from VCF file due to missing phenotype.
[WARN]	Drop sample [ APGS_000016_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ APGS_001460_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ AUTH_000082_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000009_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000013_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000015_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000019_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000020_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000023_s1 ] from VCF file due to missing phenotype
[WARN]	Drop sample [ BBDP_000026_s1 ] from VCF file due to missing phenotype
[WARN]	Skip outputting additional [ 9072 ] samples due to missing phenotype.
[INFO]	Begin to read covariate file
[WARN]	Covariate file line [ 2 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 3 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 4 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 5 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 6 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 7 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 8 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 9 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 10 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Covariate file line [ 11 ] has non-numerical value [ NA ], we will skip this sample
[WARN]	Total [ 8838 ] lines in covariate file contain non-numerical values, we will skip these lines
[WARN]	Total [ 28453 ] samples are skipped from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000016_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000019_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000021_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000035_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000051_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000062_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000081_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000086_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000111_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000113_s1 ] from covariate file due to missing phenotype
[WARN]	Skip sample [ APGS_000114_s1 ] from covariate file due to missing phenotype
[WARN]	Skip outputting additional [ 28443 ] samples from covariate file with missing phenotypes
[WARN]	Total [ 8838 ] samples are dropped from VCF file due to missing covariate.
[WARN]	Drop sample [ APGS_000002_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000003_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000004_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000008_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000011_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000018_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000023_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000026_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000027_s1 ] from VCF file due to missing covariate
[WARN]	Drop sample [ APGS_000028_s1 ] from VCF file due to missing covariate
[WARN]	Skip outputting additional [ 8828 ] samples due to missing covariate.
[INFO]	Loaded 12421 male, 8497 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_covariate_file.txt
[INFO]	Loaded 15111 cases, 5807 controls, and 0 missing phenotypes
[WARN]	-- Enabling binary phenotype mode -- 
[INFO]	Analysis begins with [ 20918 ] samples...
[INFO]	MadsonBrowning test significance will be evaluated using 10000 permutations at alpha = 0.05
[INFO]	Loaded [ 1 ] genes.
[INFO]	Impute missing genotype to mean (by default)
[INFO]	Set upper minor allele frequency limit to 0.01
[INFO]	Analysis started
[INFO]	Analyzed [ 174 ] variants from [ 1 ] genes/regions
[INFO]	Analysis ends at: Wed Aug 28 13:02:06 2024
[INFO]	Analysis took 87 seconds
RVTESTS finished successfully
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=f39966b1-733f-4ab7-8121-ab54deda9595">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># see bureden test in exonic in EUR</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.CMCWald.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	NonRefSite	Beta	SE	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	0.0708554	0.0620621	0.253585
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	-0.576489	0.0316899	6.02168e-74
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	0.0309305	0.00133732	2.37789e-118
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	-0.0224515	0.0086344	0.00931601
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	0.0140758	0.00750917	0.0608638
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	-0.0182897	0.00785175	0.019839
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	0.0119784	0.00817279	0.142746
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	-0.0137788	0.00332666	3.44415e-05
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=09f35168-655d-4578-9781-31392635b4c9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[12]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.CMC.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	NonRefSite	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	1500	0.253509
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c04bab3a-965e-4a40-a628-a6a95bfce513">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.Fp.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	0.0365734
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=eae9f761-d531-4751-b4e5-23897b50ac8a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.MadsonBrowning.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	10000	6504	6.10068	1000	0	0.153752
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=669f74ea-0f4c-406e-93c5-4e59ff95eced">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burdenTEST.exonic.Zeggini.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	174	109	0.227047
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=73b80527-c554-440c-98f8-2912523823a6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[5]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EUR nonsynonymous_lof variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	105	65	107233	0.0560423	10000	10000	107233	537	0	0.0537
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=eb15e4af-aa73-4093-a242-82034b94d596">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EUR nonsynonymous_lof variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	20918	105	65	53616.4	0	0.101898
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=58037516-2b15-462d-97de-2e55f60e048f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>AAC</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e418c688-d599-45d1-82d8-2ebf530c3f4d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AAC all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AAC/AAC_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	925	57	45	5938.41	0.692352	10000	1323	5938.41	1000	0	0.755858
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=bde40041-a3f2-4b81-a2b2-535674408424">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AAC all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AAC/AAC_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	925	57	45	2977.49	1	0.5655
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=94df761e-db76-4dff-907a-8eb1ed55d4d9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AAC all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AAC/AAC_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	925	38	31	3451.67	0.742505	10000	1275	3451.67	1000	0	0.784314
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8907bd9e-60b8-4007-9c88-48cb85fe7ea8">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AAC all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AAC/AAC_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	925	38	31	1792.92	1	0.607447
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=a308e74f-7bd9-46e5-b0a8-b624954d2f3c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>AFR</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=695a19a1-ab3f-400e-a349-9615998a5272">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AFR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AFR/AFR_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1062	60	43	7846.37	0.311869	10000	2587	7846.37	1000	0	0.386548
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=be66e3ac-1bec-4246-a2d7-257b772cbb8b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[12]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AFR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AFR/AFR_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1062	60	43	3923.18	0	0.495811
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=77c06bcc-9d2d-403c-988c-0c4fe3be46c3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AFR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AFR/AFR_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1062	38	27	6145.09	0.371636	10000	2581	6145.09	1000	0	0.387447
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a5d0d75b-0de2-4692-81b3-617b53700861">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[14]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AFR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AFR/AFR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1062	38	27	3072.54	0	0.56138
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=17bd6c61-04e4-4583-8c65-edef1c0c9e3e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>AJ</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5156f450-8eae-43dc-839c-1fb9a10ceaf1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[15]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AJ all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AJ/AJ_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1410	20	15	2660.25	0.548165	10000	1693	2660.25	1000	0	0.590667
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=aa3b6f44-4458-420d-a005-e1675d1d261d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[16]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AJ all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AJ/AJ_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1410	20	15	594.277	1	0.740559
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=df29028e-5840-4153-b257-195b5c98e2af">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[17]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AJ all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AJ/AJ_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1410	13	10	2189.49	0.52697	10000	1745	2189.49	1000	0	0.573066
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4529c89f-c2c7-4852-be83-b5d5d049ef7e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[18]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AJ all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AJ/AJ_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	1410	13	10	1094.74	0	0.708875
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=3c707c32-e6bb-4b34-a5d3-88efc7549532">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>AMR</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=10133437-0ec7-4259-a009-99ca7c3c9e93">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[19]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AMR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AMR/AMR_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	447	36	26	4759.54	0.274054	10000	4048	4759.54	1000	0	0.247036
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=35000fab-b264-4606-8036-a58096047ccd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[20]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AMR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AMR/AMR_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	447	36	26	2379.77	0	0.433785
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=875e300f-827c-407d-bac5-7e0bf851142b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[21]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AMR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AMR/AMR_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	447	18	14	3656.15	0.188299	10000	6542	3656.15	1000	0	0.152858
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=11e5bf7f-4893-4523-8597-17958132073d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[22]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check AMR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_AMR/AMR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	447	18	14	1828.07	0	0.311882
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=92ac6585-9a94-48d5-9abd-583955d33981">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>CAH</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a1a97589-691f-4b68-ad26-ab250818a07e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[23]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAH all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAH/CAH_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	693	43	34	4011.03	0.742795	10000	1290	4011.03	1000	0	0.775194
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e66124f6-8a3d-4266-be86-74f1aa587803">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[24]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAH all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAH/CAH_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	693	43	34	14226.6	1	0.0312151
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0427bf5a-edca-4bfc-8e71-5e095cc02db7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[25]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAH all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAH/CAH_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	693	26	22	3545.62	0.531319	10000	1628	3545.62	1000	0	0.614251
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=9921df41-6a24-46dd-b22b-08852e1fc73e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[26]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAH all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAH/CAH_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	693	26	22	10667.7	1	0.0308928
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=79e545a3-e1ca-403c-aea5-cde44c7adf5d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>CAS</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=f47c3562-960b-4ecd-89b2-80c452c27d5e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[27]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAS/CAS_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	658	30	19	589.951	0.984523	10000	1000	589.951	1000	0	1
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=82279004-c38b-4dfc-b741-573b038eb4b3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[28]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAS/CAS_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	658	30	19	447.27	1	0.767453
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=f05b96cd-2f32-4f05-838e-e91e111b6d7e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[29]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAS/CAS_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	658	22	11	445.823	0.237621	10000	4284	445.823	1000	0	0.233427
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=d3b936c8-fcaf-43db-b556-6ca05dc0b40f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[30]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check CAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_CAS/CAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	658	22	11	222.912	0	0.351341
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=46739ace-d30a-45cb-8297-69b651306b5e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>EAS</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=dae22fc7-d2fe-479a-a2e4-d95137a4ff72">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[31]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EAS/EAS_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	2523	45	24	6681.36	0.298387	10000	3613	6681.36	1000	0	0.276778
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1f9d7fac-4a28-4889-a62c-f7e8deecca9f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[32]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EAS/EAS_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	2523	45	24	3340.68	0	0.45024
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=563ae9e5-893b-47ac-a330-7391fbc43311">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[33]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EAS/EAS_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	2523	31	15	7571.83	0.0863703	10000	10000	7571.83	648	0	0.0648
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8cb94d61-f323-479a-a044-5798dbb22f62">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[34]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EAS/EAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	2523	31	15	3785.91	0	0.141074
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=7d1309d4-476a-468e-9049-8d309a4572c2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>FIN</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0ad2bbb1-20ca-4558-ae60-323a070657a1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[35]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check FIN all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_FIN/FIN_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	40	2	0	NA	NA	NA	NA	NA	NA	NA	NA
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=3c122967-946b-40c5-8384-3b44eae1e8e6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[36]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check FIN all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_FIN/FIN_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	40	2	0	NA	NA	NA
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=21fad881-5588-4641-8b87-ba18504048a1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[37]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check FIN all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_FIN/FIN_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	40	2	0	NA	NA	NA	NA	NA	NA	NA	NA
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=7f5a4d36-a104-4811-88f1-12d60568e576">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[38]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check FIN all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_FIN/FIN_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	40	2	0	NA	NA	NA
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=09845364-a730-4d2f-b11e-d1f1fe20be17">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>MDE</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=00afe19c-a5b1-4d97-b132-9d4df43d20c4">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[39]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check MDE all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_MDE/MDE_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	379	30	13	234.147	0.708651	10000	1101	234.147	1000	0	0.908265
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e58ca032-c70b-494b-b745-937682ecafd8">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[40]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check MDE all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_MDE/MDE_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	379	30	13	149.403	1	0.681695
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=3d0f1ad4-b3e5-43dd-a8af-73248852600c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[41]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check MDE all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_MDE/MDE_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	379	15	6	58.3182	1	10000	1013	58.3182	1000	0	0.987167
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=64ff3d48-8086-4c26-bfdf-03dd8b7eeac3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[42]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check MDE all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_MDE/MDE_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	379	15	6	29.0677	1	0.839732
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=23649329-4f8c-498b-a89b-68a6de9a8794">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>SAS</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=7dcf1de6-60d9-41ba-84bc-3526dec00edd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[43]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check SAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_SAS/SAS_TMEM175.burden.exonic.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	296	17	10	1810.68	0.10187	10000	10000	1810.68	430	0	0.043
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=f175eda8-19b7-48af-b2bf-8e344347e422">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[44]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check SAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_SAS/SAS_TMEM175.burden.exonic.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	296	17	10	2890.96	1	0.0183843
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=39cd9af1-4adc-479e-8ed9-fa0a00d17d24">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[45]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check SAS all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_SAS/SAS_TMEM175.burden.nonsynonymous_lof.Skat.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	Pvalue	NumPerm	ActualPerm	Stat	NumGreater	NumEqual	PermPvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	296	12	8	1859.12	0.0974177	10000	10000	1859.12	397	0	0.0397
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=f173e038-1b62-4d61-ae3e-2c94be0ce806">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[46]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Check EUR all_coding variant results</span>
<span class="o">!</span><span class="w"> </span>cat<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_SAS/SAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Gene	RANGE	N_INFORMATIVE	NumVar	NumPolyVar	Q	rho	Pvalue
TMEM175	4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932459-958655,4:932386-958655,4:932459-958655	296	12	8	2650.87	1	0.0230414
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=bcc9c529-229b-4638-abcc-0916246180c5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>change format of association file in EUR</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=22c1f5b9-f9c8-4dc1-a815-9c3041e7c543">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#If you need to read in the sumstats into python - you can skip this is if your data is already in python</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/EUR.all_adj.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=12a52854-3f18-4e6c-92b4-f241364c8c54">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">df</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[8]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Unnamed: 0</th>
<th>#CHROM</th>
<th>POS</th>
<th>ID</th>
<th>REF</th>
<th>ALT</th>
<th>PROVISIONAL_REF?</th>
<th>A1</th>
<th>OMITTED</th>
<th>A1_FREQ</th>
<th>FIRTH?</th>
<th>TEST</th>
<th>OBS_CT</th>
<th>OR</th>
<th>LOG(OR)_SE</th>
<th>L95</th>
<th>U95</th>
<th>Z_STAT</th>
<th>P</th>
<th>ERRCODE</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.164467</td>
<td>N</td>
<td>ADD</td>
<td>20837</td>
<td>1.034410</td>
<td>0.030398</td>
<td>0.974580</td>
<td>1.097910</td>
<td>1.112890</td>
<td>0.265757</td>
<td>.</td>
</tr>
<tr>
<th>1</th>
<td>8</td>
<td>4</td>
<td>882611</td>
<td>chr4:882611:T:C</td>
<td>T</td>
<td>C</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.051185</td>
<td>N</td>
<td>ADD</td>
<td>20885</td>
<td>0.931946</td>
<td>0.050102</td>
<td>0.844780</td>
<td>1.028110</td>
<td>-1.406740</td>
<td>0.159504</td>
<td>.</td>
</tr>
<tr>
<th>2</th>
<td>16</td>
<td>4</td>
<td>883018</td>
<td>chr4:883018:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.050599</td>
<td>N</td>
<td>ADD</td>
<td>20880</td>
<td>0.928749</td>
<td>0.050369</td>
<td>0.841441</td>
<td>1.025120</td>
<td>-1.467500</td>
<td>0.142240</td>
<td>.</td>
</tr>
<tr>
<th>3</th>
<td>24</td>
<td>4</td>
<td>883133</td>
<td>chr4:883133:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.012734</td>
<td>N</td>
<td>ADD</td>
<td>20811</td>
<td>0.809012</td>
<td>0.096557</td>
<td>0.669523</td>
<td>0.977562</td>
<td>-2.194990</td>
<td>0.028164</td>
<td>.</td>
</tr>
<tr>
<th>4</th>
<td>32</td>
<td>4</td>
<td>883483</td>
<td>chr4:883483:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.035357</td>
<td>N</td>
<td>ADD</td>
<td>20901</td>
<td>0.883241</td>
<td>0.058911</td>
<td>0.786926</td>
<td>0.991344</td>
<td>-2.107530</td>
<td>0.035071</td>
<td>.</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>392</th>
<td>3136</td>
<td>4</td>
<td>1007954</td>
<td>chr4:1007954:T:C</td>
<td>T</td>
<td>C</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.279743</td>
<td>N</td>
<td>ADD</td>
<td>18971</td>
<td>0.913461</td>
<td>0.025919</td>
<td>0.868217</td>
<td>0.961063</td>
<td>-3.492280</td>
<td>0.000479</td>
<td>.</td>
</tr>
<tr>
<th>393</th>
<td>3144</td>
<td>4</td>
<td>1008009</td>
<td>chr4:1008009:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.444961</td>
<td>N</td>
<td>ADD</td>
<td>18814</td>
<td>0.949197</td>
<td>0.023666</td>
<td>0.906174</td>
<td>0.994262</td>
<td>-2.203110</td>
<td>0.027587</td>
<td>.</td>
</tr>
<tr>
<th>394</th>
<td>3152</td>
<td>4</td>
<td>1008209</td>
<td>chr4:1008209:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.036468</td>
<td>N</td>
<td>ADD</td>
<td>20717</td>
<td>1.002060</td>
<td>0.059756</td>
<td>0.891315</td>
<td>1.126570</td>
<td>0.034499</td>
<td>0.972480</td>
<td>.</td>
</tr>
<tr>
<th>395</th>
<td>3160</td>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.279831</td>
<td>N</td>
<td>ADD</td>
<td>18965</td>
<td>0.913397</td>
<td>0.025912</td>
<td>0.868167</td>
<td>0.960983</td>
<td>-3.495880</td>
<td>0.000473</td>
<td>.</td>
</tr>
<tr>
<th>396</th>
<td>3168</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>Y</td>
<td>G</td>
<td>A</td>
<td>0.053420</td>
<td>N</td>
<td>ADD</td>
<td>20264</td>
<td>0.903891</td>
<td>0.049512</td>
<td>0.820297</td>
<td>0.996003</td>
<td>-2.040850</td>
<td>0.041266</td>
<td>.</td>
</tr>
</tbody>
</table>
<p>397 rows  20 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=90b7a98c-2f17-48e5-9d12-acaec4a6c8ca">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#First just select relevant columns: chromosome, bp position and p-value</span>
<span class="n">export_ldassoc</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">'#CHROM'</span><span class="p">,</span> <span class="s1">'POS'</span><span class="p">,</span> <span class="s1">'P'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=6ebfb5a7-78d0-4526-8165-77d1ecb82f82">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">export_ldassoc</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[10]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>#CHROM</th>
<th>POS</th>
<th>P</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882530</td>
<td>0.265757</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882611</td>
<td>0.159504</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>883018</td>
<td>0.142240</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>883133</td>
<td>0.028164</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>883483</td>
<td>0.035071</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>392</th>
<td>4</td>
<td>1007954</td>
<td>0.000479</td>
</tr>
<tr>
<th>393</th>
<td>4</td>
<td>1008009</td>
<td>0.027587</td>
</tr>
<tr>
<th>394</th>
<td>4</td>
<td>1008209</td>
<td>0.972480</td>
</tr>
<tr>
<th>395</th>
<td>4</td>
<td>1008337</td>
<td>0.000473</td>
</tr>
<tr>
<th>396</th>
<td>4</td>
<td>1008480</td>
<td>0.041266</td>
</tr>
</tbody>
</table>
<p>397 rows  3 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=73408b68-2c18-40a2-982c-2ae3b8ece7e1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Rename the #CHROM column to remove the hashtag as I think this might be confusing LDassoc</span>
<span class="n">export_ldassoc</span> <span class="o">=</span> <span class="n">export_ldassoc</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">'#CHROM'</span><span class="p">:</span> <span class="s1">'CHROM'</span><span class="p">})</span> 
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=11fef93b-f702-4da5-b66d-b81e5d0448e3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[12]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">export_ldassoc</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[12]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>CHROM</th>
<th>POS</th>
<th>P</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882530</td>
<td>0.265757</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882611</td>
<td>0.159504</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>883018</td>
<td>0.142240</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>883133</td>
<td>0.028164</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>883483</td>
<td>0.035071</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>392</th>
<td>4</td>
<td>1007954</td>
<td>0.000479</td>
</tr>
<tr>
<th>393</th>
<td>4</td>
<td>1008009</td>
<td>0.027587</td>
</tr>
<tr>
<th>394</th>
<td>4</td>
<td>1008209</td>
<td>0.972480</td>
</tr>
<tr>
<th>395</th>
<td>4</td>
<td>1008337</td>
<td>0.000473</td>
</tr>
<tr>
<th>396</th>
<td>4</td>
<td>1008480</td>
<td>0.041266</td>
</tr>
</tbody>
</table>
<p>397 rows  3 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=c548f5fd-9b1a-47eb-b429-c7400cf02a10">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Then export as a tab-separated, not comma-separated file</span>

<span class="n">export_ldassoc</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">'EUR.all_adj.formatted.tab'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=62779cf6-c244-4e8f-99a1-f44dbd017352">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs3755958)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=c136956e-9463-4ddd-94b7-5f7cadf5accc">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[15]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5b7cb319-bee4-41f0-b4c8-36a45fc43ad3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/EUR_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 958159 </span><span class="se">\</span>
<span class="s2">--to-bp 958159 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/TMEM175_rs34311866_nalls </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs34311866_nalls.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175
  --chr 4
  --double-id
  --from-bp 958159
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs34311866_nalls
  --recode A
  --to-bp 958159

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 9504 variants loaded from .bim file.
38839 people (23296 males, 15543 females) loaded from .fam.
30412 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 38839 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.975643.
1 variant and 38839 people pass filters and QC.
Among remaining phenotypes, 21198 are cases and 9214 are controls.  (8427
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs34311866_nalls.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c951c26f-be59-4a2f-a588-81c5842049f9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/EUR_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 969254 </span><span class="se">\</span>
<span class="s2">--to-bp 969254 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/TMEM175_rs3755958 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs3755958.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175
  --chr 4
  --double-id
  --from-bp 969254
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs3755958
  --recode A
  --to-bp 969254

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 9504 variants loaded from .bim file.
38839 people (23296 males, 15543 females) loaded from .fam.
30412 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 38839 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.999743.
1 variant and 38839 people pass filters and QC.
Among remaining phenotypes, 21198 are cases and 9214 are controls.  (8427
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs3755958.raw ...
101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=1e9a5f71-723a-4a1d-bc2b-b459c8974e93">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs3755958</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs3755958.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=6e77f463-05c5-46dd-b467-71fed9d2a81c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs3755958</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[10]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:969254:C:A_A</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
</tr>
<tr>
<th>38835</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>38836</th>
<td>0</td>
<td>YMS_000048_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
</tr>
<tr>
<th>38837</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
</tr>
<tr>
<th>38838</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<p>38839 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=2c1b2651-435d-4de7-972f-87512c38ff19">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/EUR_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[13]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.0</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>67.0</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>69.0</td>
<td>2.0</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>70.0</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e66491a8-f042-4f61-a666-adaab61ecfca">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[14]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[14]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:969254:C:A_A</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38833</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>70.0</td>
<td>1.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
<tr>
<th>38835</th>
<td>0</td>
<td>YMS_000048_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>76.0</td>
<td>NaN</td>
<td>-34.569818</td>
<td>-33.847077</td>
<td>-6.792226</td>
<td>-3.816187</td>
<td>21.473458</td>
</tr>
<tr>
<th>38836</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>67.0</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>38837</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>70.0</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>38838 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=17872250-4ce5-43be-a14b-77da660f657f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[15]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[15]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:969254:C:A_A</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38830</th>
<td>0</td>
<td>YMS_000034_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>58.0</td>
<td>2.0</td>
<td>-34.720268</td>
<td>-33.255119</td>
<td>-10.532694</td>
<td>-6.846457</td>
<td>23.835050</td>
</tr>
<tr>
<th>38831</th>
<td>0</td>
<td>YMS_000037_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>54.0</td>
<td>1.0</td>
<td>-33.437058</td>
<td>-36.858277</td>
<td>-11.974232</td>
<td>-9.557805</td>
<td>22.390045</td>
</tr>
<tr>
<th>38832</th>
<td>0</td>
<td>YMS_000039_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>62.0</td>
<td>2.0</td>
<td>-33.216739</td>
<td>-40.302510</td>
<td>-11.191126</td>
<td>-9.715245</td>
<td>22.341078</td>
</tr>
<tr>
<th>38833</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>70.0</td>
<td>1.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
</tbody>
</table>
<p>30411 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=b346497a-bf72-4f75-a848-374781c275f2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[16]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:969254:C:A_A"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=6b7de9dd-a975-4075-aea2-2bc8a6b17161">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[18]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs3755958"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[18]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs3755958</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000002_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000003_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000004_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000008_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000011_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38830</th>
<td>YMS_000034_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>58.0</td>
<td>-34.720268</td>
<td>-33.255119</td>
<td>-10.532694</td>
<td>-6.846457</td>
<td>23.835050</td>
</tr>
<tr>
<th>38831</th>
<td>YMS_000037_s1</td>
<td>1</td>
<td>1.0</td>
<td>1.0</td>
<td>2.0</td>
<td>54.0</td>
<td>-33.437058</td>
<td>-36.858277</td>
<td>-11.974232</td>
<td>-9.557805</td>
<td>22.390045</td>
</tr>
<tr>
<th>38832</th>
<td>YMS_000039_s1</td>
<td>2</td>
<td>2.0</td>
<td>1.0</td>
<td>1.0</td>
<td>62.0</td>
<td>-33.216739</td>
<td>-40.302510</td>
<td>-11.191126</td>
<td>-9.715245</td>
<td>22.341078</td>
</tr>
<tr>
<th>38833</th>
<td>YMS_000043_s1</td>
<td>1</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>70.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>YMS_000044_s1</td>
<td>2</td>
<td>NaN</td>
<td>0.0</td>
<td>1.0</td>
<td>73.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
</tbody>
</table>
<p>30411 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=d6391783-b675-4306-b865-4f01609f1d84">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[19]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs34311866_nalls_rs3755958.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=a1b4f7ef-5686-4748-8fed-ef89fb0b02ff">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[20]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_EUR/TMEM175_rs34311866_nalls_rs3755958.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a234687f-04a0-41a0-a05e-e2e8ef658e51">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[21]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs3755958 SEX AGE
1 0 APGS_000002_s1         2                        0                 0   1  NA
2 1 APGS_000003_s1         2                        0                 0   1  NA
3 2 APGS_000004_s1         2                        1                 0   2  NA
4 3 APGS_000008_s1         2                        1                 1   1  NA
5 4 APGS_000011_s1         2                        1                 0   1  NA
6 5 APGS_000016_s1         2                        2                 2  NA  NA
        PC1       PC2        PC3        PC4       PC5
1 -29.48330 -38.47300  -9.398904 -10.842865 25.744648
2 -33.06896 -34.28886 -11.116951  -8.056146 22.924043
3 -28.20444 -30.96948 -15.883093  -7.004351 18.647007
4 -33.30320 -35.52791 -10.818937 -12.897920  6.851688
5 -35.69828 -34.94088 -11.376421  -5.590934 24.578940
6 -30.73116 -35.70957  -8.981066  -8.204220 21.164661
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=f1728fa4-34aa-45f2-9184-8a24447e419a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[22]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs3755958</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs3755958, data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.0843 -0.1797 -0.1797 -0.0843  1.8202 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)       0.179750   0.002513   71.52   &lt;2e-16 ***
TMEM175_rs3755958 0.904552   0.004751  190.38   &lt;2e-16 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.3814 on 29621 degrees of freedom
  (788 observations deleted due to missingness)
Multiple R-squared:  0.5503,	Adjusted R-squared:  0.5503 
F-statistic: 3.624e+04 on 1 and 29621 DF,  p-value: &lt; 2.2e-16

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=15cc5e75-7c1e-4b81-8ea9-47fd22364403">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[24]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs3755958</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs3755958 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               1.4640748  0.0670080  21.849  &lt; 2e-16 ***
TMEM175_rs34311866_nalls  0.0444886  0.0080195   5.548 2.93e-08 ***
TMEM175_rs3755958         0.0062393  0.0097750   0.638 0.523293    
SEX                      -0.1143468  0.0062396 -18.326  &lt; 2e-16 ***
AGE                       0.0062552  0.0002632  23.769  &lt; 2e-16 ***
PC1                      -0.0033226  0.0016881  -1.968 0.049046 *  
PC2                       0.0024482  0.0014562   1.681 0.092733 .  
PC3                      -0.0032285  0.0015289  -2.112 0.034733 *  
PC4                       0.0023981  0.0015756   1.522 0.128014    
PC5                      -0.0023372  0.0006426  -3.637 0.000276 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1911056)

    Null deviance: 4098.0  on 20410  degrees of freedom
Residual deviance: 3898.7  on 20401  degrees of freedom
  (10000 observations deleted due to missingness)
AIC: 24157

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=3a0d5caa-af3e-4533-a729-613ca8ef9298">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs11731377)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=2f25904f-1429-442c-9c7b-4317ce595e89">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[5]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/EUR_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 977478 </span><span class="se">\</span>
<span class="s2">--to-bp 977478 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/TMEM175_rs11731377 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs11731377.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175
  --chr 4
  --double-id
  --from-bp 977478
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs11731377
  --recode A
  --to-bp 977478

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 9504 variants loaded from .bim file.
38839 people (23296 males, 15543 females) loaded from .fam.
30412 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 38839 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.996756.
1 variant and 38839 people pass filters and QC.
Among remaining phenotypes, 21198 are cases and 9214 are controls.  (8427
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs11731377.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=3bc9a3ba-d801-4b14-b1fa-3820a093cfb4">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[6]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs11731377</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs11731377.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=613cb7a5-2bbc-42a9-aa2f-d920e59de3b6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs11731377</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[7]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:977478:T:G_G</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
</tr>
<tr>
<th>38835</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>38836</th>
<td>0</td>
<td>YMS_000048_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
</tr>
<tr>
<th>38837</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
</tr>
<tr>
<th>38838</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<p>38839 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=cd25d7f2-305d-46f7-9a21-f872edf0ebee">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/EUR_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[8]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.0</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>67.0</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>69.0</td>
<td>2.0</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>70.0</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=eae03130-267f-4568-a69b-a0eebd23dfbd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[9]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:977478:T:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38833</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>70.0</td>
<td>1.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
<tr>
<th>38835</th>
<td>0</td>
<td>YMS_000048_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>76.0</td>
<td>NaN</td>
<td>-34.569818</td>
<td>-33.847077</td>
<td>-6.792226</td>
<td>-3.816187</td>
<td>21.473458</td>
</tr>
<tr>
<th>38836</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>67.0</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>38837</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>70.0</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>38838 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=59d537d0-c289-4c81-a58b-ae4ed57947ab">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[10]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:977478:T:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38830</th>
<td>0</td>
<td>YMS_000034_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>58.0</td>
<td>2.0</td>
<td>-34.720268</td>
<td>-33.255119</td>
<td>-10.532694</td>
<td>-6.846457</td>
<td>23.835050</td>
</tr>
<tr>
<th>38831</th>
<td>0</td>
<td>YMS_000037_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>54.0</td>
<td>1.0</td>
<td>-33.437058</td>
<td>-36.858277</td>
<td>-11.974232</td>
<td>-9.557805</td>
<td>22.390045</td>
</tr>
<tr>
<th>38832</th>
<td>0</td>
<td>YMS_000039_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>62.0</td>
<td>2.0</td>
<td>-33.216739</td>
<td>-40.302510</td>
<td>-11.191126</td>
<td>-9.715245</td>
<td>22.341078</td>
</tr>
<tr>
<th>38833</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>70.0</td>
<td>1.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
</tbody>
</table>
<p>30411 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=4fefdbf1-92af-420c-be2e-8d8195c2fe7c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:977478:T:G_G"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ace0d973-792b-4638-b5e9-fac8ec0e45e7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[12]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs11731377"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[12]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs11731377</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000002_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000003_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000004_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000008_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000011_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38830</th>
<td>YMS_000034_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>58.0</td>
<td>-34.720268</td>
<td>-33.255119</td>
<td>-10.532694</td>
<td>-6.846457</td>
<td>23.835050</td>
</tr>
<tr>
<th>38831</th>
<td>YMS_000037_s1</td>
<td>1</td>
<td>1.0</td>
<td>1.0</td>
<td>2.0</td>
<td>54.0</td>
<td>-33.437058</td>
<td>-36.858277</td>
<td>-11.974232</td>
<td>-9.557805</td>
<td>22.390045</td>
</tr>
<tr>
<th>38832</th>
<td>YMS_000039_s1</td>
<td>2</td>
<td>2.0</td>
<td>1.0</td>
<td>1.0</td>
<td>62.0</td>
<td>-33.216739</td>
<td>-40.302510</td>
<td>-11.191126</td>
<td>-9.715245</td>
<td>22.341078</td>
</tr>
<tr>
<th>38833</th>
<td>YMS_000043_s1</td>
<td>1</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>70.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>YMS_000044_s1</td>
<td>2</td>
<td>NaN</td>
<td>0.0</td>
<td>1.0</td>
<td>73.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
</tbody>
</table>
<p>30411 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=4498e079-b853-4bf0-89e2-837fa0b48c97">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs34311866_nalls_rs11731377.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=696bad3d-bd08-4660-895d-cde2a92811aa">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[16]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_EUR/TMEM175_rs34311866_nalls_rs11731377.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a3c08e2f-ce77-4407-93da-bfa0379736cd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[17]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs11731377 SEX
1 0 APGS_000002_s1         2                        0                  0   1
2 1 APGS_000003_s1         2                        0                  0   1
3 2 APGS_000004_s1         2                        1                  0   2
4 3 APGS_000008_s1         2                        1                  1   1
5 4 APGS_000011_s1         2                        1                  0   1
6 5 APGS_000016_s1         2                        2                  2  NA
  AGE       PC1       PC2        PC3        PC4       PC5
1  NA -29.48330 -38.47300  -9.398904 -10.842865 25.744648
2  NA -33.06896 -34.28886 -11.116951  -8.056146 22.924043
3  NA -28.20444 -30.96948 -15.883093  -7.004351 18.647007
4  NA -33.30320 -35.52791 -10.818937 -12.897920  6.851688
5  NA -35.69828 -34.94088 -11.376421  -5.590934 24.578940
6  NA -30.73116 -35.70957  -8.981066  -8.204220 21.164661
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1c9f51ab-5fc7-4045-a9ad-19e6302b9a75">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[18]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs11731377</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs11731377, data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.5365 -0.1908 -0.1908  0.1363  1.8092 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        0.190852   0.003092   61.73   &lt;2e-16 ***
TMEM175_rs11731377 0.672804   0.005106  131.78   &lt;2e-16 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.4513 on 29531 degrees of freedom
  (878 observations deleted due to missingness)
Multiple R-squared:  0.3703,	Adjusted R-squared:  0.3703 
F-statistic: 1.737e+04 on 1 and 29531 DF,  p-value: &lt; 2.2e-16

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=92f394b3-c3fc-42a1-9fcb-3dd06e0c3add">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[19]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs11731377</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs11731377 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               1.4639329  0.0671373  21.805  &lt; 2e-16 ***
TMEM175_rs34311866_nalls  0.0478139  0.0067679   7.065 1.66e-12 ***
TMEM175_rs11731377        0.0004801  0.0074602   0.064 0.948688    
SEX                      -0.1141855  0.0062491 -18.272  &lt; 2e-16 ***
AGE                       0.0062403  0.0002636  23.671  &lt; 2e-16 ***
PC1                      -0.0034063  0.0016915  -2.014 0.044049 *  
PC2                       0.0025310  0.0014578   1.736 0.082553 .  
PC3                      -0.0031779  0.0015307  -2.076 0.037903 *  
PC4                       0.0022741  0.0015782   1.441 0.149605    
PC5                      -0.0023100  0.0006439  -3.588 0.000335 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1911794)

    Null deviance: 4087.8  on 20358  degrees of freedom
Residual deviance: 3890.3  on 20349  degrees of freedom
  (10052 observations deleted due to missingness)
AIC: 24103

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=10937270-fc81-45d9-8934-198deaec105a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm(rs6599391)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1ff72018-be01-4d63-8da4-d5c051369c02">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[20]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/EUR_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 985391 </span><span class="se">\</span>
<span class="s2">--to-bp 985391 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EUR/TMEM175_rs6599391 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs6599391.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175
  --chr 4
  --double-id
  --from-bp 985391
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs6599391
  --recode A
  --to-bp 985391

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 9504 variants loaded from .bim file.
38839 people (23296 males, 15543 females) loaded from .fam.
30412 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 38839 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.97451.
1 variant and 38839 people pass filters and QC.
Among remaining phenotypes, 21198 are cases and 9214 are controls.  (8427
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/TMEM175_rs6599391.raw ...
101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=3951fb9f-bcee-4e6a-a8b4-2b15d947f43e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[21]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs6599391</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs6599391.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=d86385a8-6b77-458c-82ce-6a6a40a5cd6a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[22]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs6599391</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[22]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:985391:T:G_G</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
</tr>
<tr>
<th>38835</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>38836</th>
<td>0</td>
<td>YMS_000048_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>1.0</td>
</tr>
<tr>
<th>38837</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
</tr>
<tr>
<th>38838</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<p>38839 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=fc518c9c-2dd0-4e75-ad52-582489b5c88f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[23]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[23]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:985391:T:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38833</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>70.0</td>
<td>1.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
<tr>
<th>38835</th>
<td>0</td>
<td>YMS_000048_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>76.0</td>
<td>NaN</td>
<td>-34.569818</td>
<td>-33.847077</td>
<td>-6.792226</td>
<td>-3.816187</td>
<td>21.473458</td>
</tr>
<tr>
<th>38836</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>67.0</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>38837</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>70.0</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>38838 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0fe4d2d1-146e-4722-8a60-2ad57c638326">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[24]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[24]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:985391:T:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000008_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000011_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38830</th>
<td>0</td>
<td>YMS_000034_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>58.0</td>
<td>2.0</td>
<td>-34.720268</td>
<td>-33.255119</td>
<td>-10.532694</td>
<td>-6.846457</td>
<td>23.835050</td>
</tr>
<tr>
<th>38831</th>
<td>0</td>
<td>YMS_000037_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>54.0</td>
<td>1.0</td>
<td>-33.437058</td>
<td>-36.858277</td>
<td>-11.974232</td>
<td>-9.557805</td>
<td>22.390045</td>
</tr>
<tr>
<th>38832</th>
<td>0</td>
<td>YMS_000039_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>62.0</td>
<td>2.0</td>
<td>-33.216739</td>
<td>-40.302510</td>
<td>-11.191126</td>
<td>-9.715245</td>
<td>22.341078</td>
</tr>
<tr>
<th>38833</th>
<td>0</td>
<td>YMS_000043_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>70.0</td>
<td>1.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>0</td>
<td>YMS_000044_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
</tbody>
</table>
<p>30411 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=fe628838-1f3f-4504-8834-3b3fa9999a9c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[25]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:985391:T:G_G"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=b67bc6fc-fbe4-4b47-89c6-4c3eee3d771f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[26]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs6599391"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[26]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs6599391</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000002_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000003_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000004_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000008_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-33.303205</td>
<td>-35.527912</td>
<td>-10.818937</td>
<td>-12.897920</td>
<td>6.851688</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000011_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-35.698278</td>
<td>-34.940878</td>
<td>-11.376421</td>
<td>-5.590934</td>
<td>24.578940</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>38830</th>
<td>YMS_000034_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>58.0</td>
<td>-34.720268</td>
<td>-33.255119</td>
<td>-10.532694</td>
<td>-6.846457</td>
<td>23.835050</td>
</tr>
<tr>
<th>38831</th>
<td>YMS_000037_s1</td>
<td>1</td>
<td>1.0</td>
<td>1.0</td>
<td>2.0</td>
<td>54.0</td>
<td>-33.437058</td>
<td>-36.858277</td>
<td>-11.974232</td>
<td>-9.557805</td>
<td>22.390045</td>
</tr>
<tr>
<th>38832</th>
<td>YMS_000039_s1</td>
<td>2</td>
<td>2.0</td>
<td>1.0</td>
<td>1.0</td>
<td>62.0</td>
<td>-33.216739</td>
<td>-40.302510</td>
<td>-11.191126</td>
<td>-9.715245</td>
<td>22.341078</td>
</tr>
<tr>
<th>38833</th>
<td>YMS_000043_s1</td>
<td>1</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>70.0</td>
<td>-33.608572</td>
<td>-35.293481</td>
<td>-11.033493</td>
<td>-9.916257</td>
<td>24.680699</td>
</tr>
<tr>
<th>38834</th>
<td>YMS_000044_s1</td>
<td>2</td>
<td>NaN</td>
<td>0.0</td>
<td>1.0</td>
<td>73.0</td>
<td>-32.287712</td>
<td>-35.705842</td>
<td>-11.819036</td>
<td>-6.511252</td>
<td>18.970359</td>
</tr>
</tbody>
</table>
<p>30411 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=a57200e8-a676-45f1-9191-b0521248ce71">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[27]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/TMEM175_rs34311866_nalls_rs6599391.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=ce00414e-a1f6-4e8f-9f01-8d8d386977ac">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[28]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_EUR/TMEM175_rs34311866_nalls_rs6599391.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=dca35806-238f-4f41-ba5d-98432cffa813">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[29]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs6599391 SEX AGE
1 0 APGS_000002_s1         2                        0                 0   1  NA
2 1 APGS_000003_s1         2                        0                 0   1  NA
3 2 APGS_000004_s1         2                        1                 0   2  NA
4 3 APGS_000008_s1         2                        1                 1   1  NA
5 4 APGS_000011_s1         2                        1                 0   1  NA
6 5 APGS_000016_s1         2                        2                 2  NA  NA
        PC1       PC2        PC3        PC4       PC5
1 -29.48330 -38.47300  -9.398904 -10.842865 25.744648
2 -33.06896 -34.28886 -11.116951  -8.056146 22.924043
3 -28.20444 -30.96948 -15.883093  -7.004351 18.647007
4 -33.30320 -35.52791 -10.818937 -12.897920  6.851688
5 -35.69828 -34.94088 -11.376421  -5.590934 24.578940
6 -30.73116 -35.70957  -8.981066  -8.204220 21.164661
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=adaa9ec7-3345-419e-a53d-190f34f856b1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[30]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs6599391</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs6599391, data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.5355 -0.1963 -0.1963  0.1341  1.8037 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)       0.196263   0.003114   63.02   &lt;2e-16 ***
TMEM175_rs6599391 0.669607   0.005276  126.92   &lt;2e-16 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.4534 on 28870 degrees of freedom
  (1539 observations deleted due to missingness)
Multiple R-squared:  0.3581,	Adjusted R-squared:  0.3581 
F-statistic: 1.611e+04 on 1 and 28870 DF,  p-value: &lt; 2.2e-16

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=582117a0-70d5-4ad9-8689-161d8b70d26f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[31]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs6599391</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs6599391 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               1.4611815  0.0679926  21.490  &lt; 2e-16 ***
TMEM175_rs34311866_nalls  0.0492999  0.0068278   7.220 5.37e-13 ***
TMEM175_rs6599391        -0.0003852  0.0076512  -0.050  0.95985    
SEX                      -0.1157631  0.0063212 -18.314  &lt; 2e-16 ***
AGE                       0.0062624  0.0002664  23.510  &lt; 2e-16 ***
PC1                      -0.0035487  0.0017098  -2.075  0.03796 *  
PC2                       0.0026485  0.0014743   1.796  0.07245 .  
PC3                      -0.0030284  0.0015488  -1.955  0.05055 .  
PC4                       0.0023599  0.0015956   1.479  0.13914    
PC5                      -0.0020974  0.0006556  -3.199  0.00138 ** 
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1915455)

    Null deviance: 4014.2  on 19940  degrees of freedom
Residual deviance: 3817.7  on 19931  degrees of freedom
  (10470 observations deleted due to missingness)
AIC: 23647

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=0bff69f2-c135-4aa0-a747-d20cbb856704">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs3822023)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=ca69ba56-d8d8-49e9-86db-679ce342019e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[6]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%</span><span class="k">load_ext</span> rpy2.ipython
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1a737825-c4bd-4a56-bc96-1e130a813fea">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AAC/AAC_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 958159 </span><span class="se">\</span>
<span class="s2">--to-bp 958159 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AAC/TMEM175_rs34311866_nalls </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs34311866_nalls.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/AAC_TMEM175
  --chr 4
  --double-id
  --from-bp 958159
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs34311866_nalls
  --recode A
  --to-bp 958159

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 3449 variants loaded from .bim file.
1111 people (455 males, 656 females) loaded from .fam.
1086 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 1111 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.992799.
1 variant and 1111 people pass filters and QC.
Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs34311866_nalls.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=728230eb-10ee-4514-976a-0abe863fce23">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AAC/AAC_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 902698 </span><span class="se">\</span>
<span class="s2">--to-bp 902698 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AAC/TMEM175_rs3822023 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs3822023.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/AAC_TMEM175
  --chr 4
  --double-id
  --from-bp 902698
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs3822023
  --recode A
  --to-bp 902698

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 3449 variants loaded from .bim file.
1111 people (455 males, 656 females) loaded from .fam.
1086 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 1111 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.978398.
1 variant and 1111 people pass filters and QC.
Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs3822023.raw ...
101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=206fe64d-cff5-4405-b5b9-a64a7cf04751">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs3822023</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/TMEM175_rs3822023.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=842410f1-8fc2-4314-89d2-5ad4a725df0b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs3822023</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[10]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:902698:T:C_C</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>BCM_000142_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>BCM_000150_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>BCM_000204_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>BCM_000215_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>BCM_000270_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>0</td>
<td>UPENNU19_000409_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
</tr>
<tr>
<th>1107</th>
<td>0</td>
<td>UPENNU19_000432_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>1108</th>
<td>0</td>
<td>VIPD_000151_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1109</th>
<td>0</td>
<td>VIPD_000679_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>1110</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>2.0</td>
</tr>
</tbody>
</table>
<p>1111 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=b4132f51-1803-41b3-be5b-3bcbfca32889">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/AAC_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[11]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>53.0</td>
<td>1.0</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=209de9df-b5c1-4950-8084-795239bca52a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[12]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[12]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:902698:T:C_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>BCM_000142_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>42.344694</td>
<td>-18.080400</td>
<td>-4.379556</td>
<td>-4.231518</td>
<td>11.945426</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>BCM_000150_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>46.211997</td>
<td>-16.728937</td>
<td>-2.939694</td>
<td>-4.306565</td>
<td>5.530103</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>BCM_000204_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>53.796533</td>
<td>-13.345479</td>
<td>-2.312777</td>
<td>-4.141204</td>
<td>3.542342</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>BCM_000215_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>33.149467</td>
<td>-17.891235</td>
<td>-5.542113</td>
<td>-2.706507</td>
<td>6.964059</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>BCM_000270_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>1.0</td>
<td>44.397319</td>
<td>-15.104927</td>
<td>-4.648018</td>
<td>-3.126988</td>
<td>7.345741</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>0</td>
<td>UPENNU19_000409_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>79.386111</td>
<td>2.0</td>
<td>59.974838</td>
<td>-11.604280</td>
<td>-4.482499</td>
<td>-2.758223</td>
<td>3.541477</td>
</tr>
<tr>
<th>1107</th>
<td>0</td>
<td>UPENNU19_000432_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>77.661111</td>
<td>2.0</td>
<td>55.937250</td>
<td>-11.575462</td>
<td>-5.033753</td>
<td>-2.485275</td>
<td>6.130544</td>
</tr>
<tr>
<th>1108</th>
<td>0</td>
<td>VIPD_000151_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.000000</td>
<td>2.0</td>
<td>45.339968</td>
<td>-18.329921</td>
<td>-5.496370</td>
<td>-0.623469</td>
<td>2.827065</td>
</tr>
<tr>
<th>1109</th>
<td>0</td>
<td>VIPD_000679_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.000000</td>
<td>2.0</td>
<td>24.426378</td>
<td>-24.074607</td>
<td>-5.665107</td>
<td>-6.488002</td>
<td>12.017162</td>
</tr>
<tr>
<th>1110</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>53.000000</td>
<td>1.0</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
</tbody>
</table>
<p>1111 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4eee6bcc-43df-4d53-bdfe-9041c8cf9bd8">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[13]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:902698:T:C_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>BCM_000142_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>42.344694</td>
<td>-18.080400</td>
<td>-4.379556</td>
<td>-4.231518</td>
<td>11.945426</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>BCM_000150_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>46.211997</td>
<td>-16.728937</td>
<td>-2.939694</td>
<td>-4.306565</td>
<td>5.530103</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>BCM_000204_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>53.796533</td>
<td>-13.345479</td>
<td>-2.312777</td>
<td>-4.141204</td>
<td>3.542342</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>BCM_000215_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>33.149467</td>
<td>-17.891235</td>
<td>-5.542113</td>
<td>-2.706507</td>
<td>6.964059</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>BCM_000270_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>1.0</td>
<td>44.397319</td>
<td>-15.104927</td>
<td>-4.648018</td>
<td>-3.126988</td>
<td>7.345741</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>0</td>
<td>UPENNU19_000409_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>79.386111</td>
<td>2.0</td>
<td>59.974838</td>
<td>-11.604280</td>
<td>-4.482499</td>
<td>-2.758223</td>
<td>3.541477</td>
</tr>
<tr>
<th>1107</th>
<td>0</td>
<td>UPENNU19_000432_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>77.661111</td>
<td>2.0</td>
<td>55.937250</td>
<td>-11.575462</td>
<td>-5.033753</td>
<td>-2.485275</td>
<td>6.130544</td>
</tr>
<tr>
<th>1108</th>
<td>0</td>
<td>VIPD_000151_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.000000</td>
<td>2.0</td>
<td>45.339968</td>
<td>-18.329921</td>
<td>-5.496370</td>
<td>-0.623469</td>
<td>2.827065</td>
</tr>
<tr>
<th>1109</th>
<td>0</td>
<td>VIPD_000679_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.000000</td>
<td>2.0</td>
<td>24.426378</td>
<td>-24.074607</td>
<td>-5.665107</td>
<td>-6.488002</td>
<td>12.017162</td>
</tr>
<tr>
<th>1110</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>53.000000</td>
<td>1.0</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
</tbody>
</table>
<p>1086 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=7c024bcb-f044-4a44-bae5-2dda01462ac5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[14]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:902698:T:C_C"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=77f12059-2f09-4777-8036-9e1a0bee7202">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[15]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs3822023"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[15]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs3822023</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>BCM_000142_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>42.344694</td>
<td>-18.080400</td>
<td>-4.379556</td>
<td>-4.231518</td>
<td>11.945426</td>
</tr>
<tr>
<th>1</th>
<td>BCM_000150_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>46.211997</td>
<td>-16.728937</td>
<td>-2.939694</td>
<td>-4.306565</td>
<td>5.530103</td>
</tr>
<tr>
<th>2</th>
<td>BCM_000204_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>53.796533</td>
<td>-13.345479</td>
<td>-2.312777</td>
<td>-4.141204</td>
<td>3.542342</td>
</tr>
<tr>
<th>3</th>
<td>BCM_000215_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>33.149467</td>
<td>-17.891235</td>
<td>-5.542113</td>
<td>-2.706507</td>
<td>6.964059</td>
</tr>
<tr>
<th>4</th>
<td>BCM_000270_s1</td>
<td>1</td>
<td>0.0</td>
<td>1.0</td>
<td>2.0</td>
<td>NaN</td>
<td>44.397319</td>
<td>-15.104927</td>
<td>-4.648018</td>
<td>-3.126988</td>
<td>7.345741</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>UPENNU19_000409_s1</td>
<td>2</td>
<td>0.0</td>
<td>2.0</td>
<td>1.0</td>
<td>79.386111</td>
<td>59.974838</td>
<td>-11.604280</td>
<td>-4.482499</td>
<td>-2.758223</td>
<td>3.541477</td>
</tr>
<tr>
<th>1107</th>
<td>UPENNU19_000432_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>2.0</td>
<td>77.661111</td>
<td>55.937250</td>
<td>-11.575462</td>
<td>-5.033753</td>
<td>-2.485275</td>
<td>6.130544</td>
</tr>
<tr>
<th>1108</th>
<td>VIPD_000151_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>63.000000</td>
<td>45.339968</td>
<td>-18.329921</td>
<td>-5.496370</td>
<td>-0.623469</td>
<td>2.827065</td>
</tr>
<tr>
<th>1109</th>
<td>VIPD_000679_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>2.0</td>
<td>60.000000</td>
<td>24.426378</td>
<td>-24.074607</td>
<td>-5.665107</td>
<td>-6.488002</td>
<td>12.017162</td>
</tr>
<tr>
<th>1110</th>
<td>YMS_000052_s1</td>
<td>1</td>
<td>1.0</td>
<td>2.0</td>
<td>1.0</td>
<td>53.000000</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
</tbody>
</table>
<p>1086 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=be867ec3-b511-4d32-a24e-fb9239268bef">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[16]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/TMEM175_rs34311866_nalls_rs3822023.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=dab0161d-7d39-42c3-9436-f67a4a453519">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[17]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_AAC/TMEM175_rs34311866_nalls_rs3822023.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=31afafc5-04ee-440d-ad9f-897440c20df6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[18]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X           IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs3822023 SEX AGE
1 0 BCM_000142_s1         2                        0                 1   1  NA
2 1 BCM_000150_s1         2                        0                 0   2  NA
3 2 BCM_000204_s1         2                        0                 0   2  NA
4 3 BCM_000215_s1         2                        0                 0   1  NA
5 4 BCM_000270_s1         1                        0                 1   2  NA
6 5 BCM_000277_s1         1                        1                 1   2  NA
       PC1       PC2       PC3       PC4        PC5
1 42.34469 -18.08040 -4.379556 -4.231518 11.9454258
2 46.21200 -16.72894 -2.939694 -4.306565  5.5301026
3 53.79653 -13.34548 -2.312777 -4.141204  3.5423418
4 33.14947 -17.89123 -5.542113 -2.706507  6.9640586
5 44.39732 -15.10493 -4.648018 -3.126988  7.3457406
6 63.11613 -11.49276 -2.482421 -1.453234  0.2000133
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=2882baf9-4b18-4bd6-afe4-55d7b689049a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[19]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs3822023</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs3822023, data = total)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.10548 -0.10548 -0.09122 -0.09122  1.90878 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        0.10548    0.01440   7.325 4.72e-13 ***
TMEM175_rs3822023 -0.01427    0.01340  -1.064    0.287    
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.2979 on 1055 degrees of freedom
  (29 observations deleted due to missingness)
Multiple R-squared:  0.001073,	Adjusted R-squared:  0.0001258 
F-statistic: 1.133 on 1 and 1055 DF,  p-value: 0.2874

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=da68c439-36a3-4bf9-91af-188f8c24de21">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[20]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs3822023</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs3822023 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                          Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               2.363025   0.119358  19.798  &lt; 2e-16 ***
TMEM175_rs34311866_nalls -0.025002   0.041030  -0.609 0.542434    
TMEM175_rs3822023         0.060082   0.018168   3.307 0.000981 ***
SEX                      -0.097427   0.025418  -3.833 0.000135 ***
AGE                       0.003558   0.001217   2.924 0.003544 ** 
PC1                      -0.016870   0.001332 -12.663  &lt; 2e-16 ***
PC2                       0.014707   0.004384   3.355 0.000827 ***
PC3                      -0.009898   0.004942  -2.003 0.045501 *  
PC4                       0.005723   0.006161   0.929 0.353165    
PC5                      -0.044458   0.005513  -8.065 2.36e-15 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1381122)

    Null deviance: 158.90  on 900  degrees of freedom
Residual deviance: 123.06  on 891  degrees of freedom
  (185 observations deleted due to missingness)
AIC: 785.17

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=825b7067-81b7-43df-91c6-28cc64b5337d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs188897299)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=98e71482-db9d-419f-b34c-b30e249dd520">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[21]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_CAH/CAH_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 896281 </span><span class="se">\</span>
<span class="s2">--to-bp 896281 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_CAH/TMEM175_rs188897299 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/TMEM175_rs188897299.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/CAH_TMEM175
  --chr 4
  --double-id
  --from-bp 896281
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/TMEM175_rs188897299
  --recode A
  --to-bp 896281

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 3090 variants loaded from .bim file.
851 people (446 males, 405 females) loaded from .fam.
827 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 851 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.994125.
1 variant and 851 people pass filters and QC.
Among remaining phenotypes, 525 are cases and 302 are controls.  (24 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/TMEM175_rs188897299.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8b501335-0d33-4185-9071-493a7034f2fe">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[22]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_CAH/CAH_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 958159 </span><span class="se">\</span>
<span class="s2">--to-bp 958159 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_CAH/TMEM175_rs34311866_nalls </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/TMEM175_rs34311866_nalls.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/CAH_TMEM175
  --chr 4
  --double-id
  --from-bp 958159
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/TMEM175_rs34311866_nalls
  --recode A
  --to-bp 958159

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 3090 variants loaded from .bim file.
851 people (446 males, 405 females) loaded from .fam.
827 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 851 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.975323.
1 variant and 851 people pass filters and QC.
Among remaining phenotypes, 525 are cases and 302 are controls.  (24 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_CAH/TMEM175_rs34311866_nalls.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=bdbfccca-9c3d-4f65-8fbb-f0c1a878dcf7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[23]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_CAH/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs188897299</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_CAH/TMEM175_rs188897299.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=24d4bf12-aca0-4e9c-aa8d-fc942462b756">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[24]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs188897299</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[24]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:896281:A:C_C</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000114_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000299_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000386_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000416_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000684_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>846</th>
<td>0</td>
<td>VIPD_000068_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>847</th>
<td>0</td>
<td>VIPD_000375_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>848</th>
<td>0</td>
<td>VIPD_000647_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
</tr>
<tr>
<th>849</th>
<td>0</td>
<td>VIPD_000742_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>850</th>
<td>0</td>
<td>YMS_000013_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<p>851 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0b505a53-4f16-4e7e-a1f4-16769f8eaa96">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[25]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_CAH/CAH_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[25]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ba161265-8c7b-4a14-9851-cb1dd090c290">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[26]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[26]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:896281:A:C_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000114_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-8.644799</td>
<td>-0.454641</td>
<td>5.388369</td>
<td>-2.623423</td>
<td>12.250261</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000299_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.117332</td>
<td>-6.534283</td>
<td>18.966258</td>
<td>4.719556</td>
<td>10.820025</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000386_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.918558</td>
<td>3.239078</td>
<td>-8.978111</td>
<td>-17.386542</td>
<td>16.116924</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000416_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-19.850066</td>
<td>-15.093607</td>
<td>11.557804</td>
<td>-2.183026</td>
<td>7.027059</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000684_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-26.649221</td>
<td>-7.140707</td>
<td>-4.581517</td>
<td>-7.305596</td>
<td>13.061927</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>846</th>
<td>0</td>
<td>VIPD_000068_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>58.0</td>
<td>2.0</td>
<td>-25.086819</td>
<td>-15.832343</td>
<td>17.822988</td>
<td>4.304263</td>
<td>12.155667</td>
</tr>
<tr>
<th>847</th>
<td>0</td>
<td>VIPD_000375_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>56.0</td>
<td>2.0</td>
<td>16.583566</td>
<td>-22.589907</td>
<td>-4.884012</td>
<td>-5.291012</td>
<td>11.423993</td>
</tr>
<tr>
<th>848</th>
<td>0</td>
<td>VIPD_000647_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>56.0</td>
<td>1.0</td>
<td>-2.504160</td>
<td>-11.905903</td>
<td>-7.795025</td>
<td>-8.134426</td>
<td>14.603985</td>
</tr>
<tr>
<th>849</th>
<td>0</td>
<td>VIPD_000742_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>81.0</td>
<td>2.0</td>
<td>-27.020436</td>
<td>-7.366797</td>
<td>-10.300603</td>
<td>-15.562663</td>
<td>-7.277205</td>
</tr>
<tr>
<th>850</th>
<td>0</td>
<td>YMS_000013_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>-9</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>63.0</td>
<td>NaN</td>
<td>-16.576064</td>
<td>-17.109194</td>
<td>-21.058167</td>
<td>8.182851</td>
<td>4.101956</td>
</tr>
</tbody>
</table>
<p>851 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1ae2d28d-5d91-40f2-9d5d-ba2d06858283">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[27]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[27]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:896281:A:C_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000114_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-8.644799</td>
<td>-0.454641</td>
<td>5.388369</td>
<td>-2.623423</td>
<td>12.250261</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000299_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.117332</td>
<td>-6.534283</td>
<td>18.966258</td>
<td>4.719556</td>
<td>10.820025</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000386_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.918558</td>
<td>3.239078</td>
<td>-8.978111</td>
<td>-17.386542</td>
<td>16.116924</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000416_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-19.850066</td>
<td>-15.093607</td>
<td>11.557804</td>
<td>-2.183026</td>
<td>7.027059</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000684_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-26.649221</td>
<td>-7.140707</td>
<td>-4.581517</td>
<td>-7.305596</td>
<td>13.061927</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>845</th>
<td>0</td>
<td>UPENNU19_000406_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>67.405556</td>
<td>2.0</td>
<td>-10.061335</td>
<td>-19.094153</td>
<td>-22.136097</td>
<td>10.401768</td>
<td>4.482694</td>
</tr>
<tr>
<th>846</th>
<td>0</td>
<td>VIPD_000068_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>58.000000</td>
<td>2.0</td>
<td>-25.086819</td>
<td>-15.832343</td>
<td>17.822988</td>
<td>4.304263</td>
<td>12.155667</td>
</tr>
<tr>
<th>847</th>
<td>0</td>
<td>VIPD_000375_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>56.000000</td>
<td>2.0</td>
<td>16.583566</td>
<td>-22.589907</td>
<td>-4.884012</td>
<td>-5.291012</td>
<td>11.423993</td>
</tr>
<tr>
<th>848</th>
<td>0</td>
<td>VIPD_000647_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>56.000000</td>
<td>1.0</td>
<td>-2.504160</td>
<td>-11.905903</td>
<td>-7.795025</td>
<td>-8.134426</td>
<td>14.603985</td>
</tr>
<tr>
<th>849</th>
<td>0</td>
<td>VIPD_000742_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>81.000000</td>
<td>2.0</td>
<td>-27.020436</td>
<td>-7.366797</td>
<td>-10.300603</td>
<td>-15.562663</td>
<td>-7.277205</td>
</tr>
</tbody>
</table>
<p>827 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=1395388d-cb06-42dd-a6a4-516f31db33c5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[28]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:896281:A:C_C"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=60832f2a-199f-4aba-a35b-2c34889b800a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[29]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs188897299"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[29]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs188897299</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000114_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-8.644799</td>
<td>-0.454641</td>
<td>5.388369</td>
<td>-2.623423</td>
<td>12.250261</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000299_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-27.117332</td>
<td>-6.534283</td>
<td>18.966258</td>
<td>4.719556</td>
<td>10.820025</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000386_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-27.918558</td>
<td>3.239078</td>
<td>-8.978111</td>
<td>-17.386542</td>
<td>16.116924</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000416_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-19.850066</td>
<td>-15.093607</td>
<td>11.557804</td>
<td>-2.183026</td>
<td>7.027059</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000684_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-26.649221</td>
<td>-7.140707</td>
<td>-4.581517</td>
<td>-7.305596</td>
<td>13.061927</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>845</th>
<td>UPENNU19_000406_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>67.405556</td>
<td>-10.061335</td>
<td>-19.094153</td>
<td>-22.136097</td>
<td>10.401768</td>
<td>4.482694</td>
</tr>
<tr>
<th>846</th>
<td>VIPD_000068_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>58.000000</td>
<td>-25.086819</td>
<td>-15.832343</td>
<td>17.822988</td>
<td>4.304263</td>
<td>12.155667</td>
</tr>
<tr>
<th>847</th>
<td>VIPD_000375_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>56.000000</td>
<td>16.583566</td>
<td>-22.589907</td>
<td>-4.884012</td>
<td>-5.291012</td>
<td>11.423993</td>
</tr>
<tr>
<th>848</th>
<td>VIPD_000647_s1</td>
<td>1</td>
<td>NaN</td>
<td>0.0</td>
<td>2.0</td>
<td>56.000000</td>
<td>-2.504160</td>
<td>-11.905903</td>
<td>-7.795025</td>
<td>-8.134426</td>
<td>14.603985</td>
</tr>
<tr>
<th>849</th>
<td>VIPD_000742_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>81.000000</td>
<td>-27.020436</td>
<td>-7.366797</td>
<td>-10.300603</td>
<td>-15.562663</td>
<td>-7.277205</td>
</tr>
</tbody>
</table>
<p>827 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=847397c9-f8e6-40a6-9bcc-7c100f0cc693">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[30]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_CAH/TMEM175_rs34311866_nalls_rs188897299.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=ade8043c-0a2f-47d0-94c4-d6b6cfcddc26">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[31]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_CAH/TMEM175_rs34311866_nalls_rs188897299.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a43b89c3-d13d-4f26-ac7a-972ae6d3bbeb">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[32]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs188897299 SEX
1 0 APGS_000114_s1         2                        0                   0   1
2 1 APGS_000299_s1         2                        0                   0   1
3 2 APGS_000386_s1         2                        1                   0   2
4 3 APGS_000416_s1         2                        0                   0   1
5 4 APGS_000684_s1         2                        0                   0   1
6 5 APGS_000724_s1         2                        0                   0   1
  AGE        PC1         PC2       PC3        PC4       PC5
1  NA  -8.644799  -0.4546406  5.388369  -2.623423 12.250261
2  NA -27.117332  -6.5342828 18.966258   4.719556 10.820025
3  NA -27.918558   3.2390784 -8.978111 -17.386542 16.116924
4  NA -19.850066 -15.0936070 11.557804  -2.183026  7.027059
5  NA -26.649221  -7.1407070 -4.581517  -7.305596 13.061927
6  NA  -3.051137  -9.3939669  4.065017  -4.493076  6.053414
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=07416a1b-646a-4ac9-a126-941ccf7590e1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[33]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs188897299</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs188897299, 
    data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.3123 -0.3123 -0.3123  0.6877  1.6877 

Coefficients:
                    Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          0.31227    0.01789  17.452   &lt;2e-16 ***
TMEM175_rs188897299 -0.18282    0.09582  -1.908   0.0568 .  
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.4994 on 801 degrees of freedom
  (24 observations deleted due to missingness)
Multiple R-squared:  0.004524,	Adjusted R-squared:  0.003281 
F-statistic:  3.64 on 1 and 801 DF,  p-value: 0.05675

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0e397e08-fe7a-41af-966c-a0f488565352">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[34]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs188897299</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs188897299 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                          Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               1.246928   0.082737  15.071  &lt; 2e-16 ***
TMEM175_rs34311866_nalls -0.010578   0.034906  -0.303 0.761959    
TMEM175_rs188897299      -0.354698   0.095306  -3.722 0.000215 ***
SEX                      -0.105538   0.033602  -3.141 0.001760 ** 
AGE                       0.010337   0.001039   9.954  &lt; 2e-16 ***
PC1                       0.001416   0.001143   1.239 0.215828    
PC2                      -0.001086   0.002031  -0.535 0.592971    
PC3                      -0.004465   0.002309  -1.934 0.053560 .  
PC4                       0.003527   0.003393   1.040 0.298937    
PC5                      -0.012965   0.003909  -3.317 0.000960 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1865895)

    Null deviance: 160.81  on 673  degrees of freedom
Residual deviance: 123.90  on 664  degrees of freedom
  (153 observations deleted due to missingness)
AIC: 793.11

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=b558a55a-4bfd-429a-af41-10b7c6ba1ca9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs3733344)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a461c85e-ae39-47e1-b3dc-3e58d4c63f4f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[36]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EAS/EAS_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 958159 </span><span class="se">\</span>
<span class="s2">--to-bp 958159 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EAS/TMEM175_rs34311866_nalls </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/TMEM175_rs34311866_nalls.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/EAS_TMEM175
  --chr 4
  --double-id
  --from-bp 958159
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/TMEM175_rs34311866_nalls
  --recode A
  --to-bp 958159

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 3171 variants loaded from .bim file.
5167 people (3278 males, 1889 females) loaded from .fam.
5123 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 5167 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.899168.
1 variant and 5167 people pass filters and QC.
Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/TMEM175_rs34311866_nalls.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=743fc99c-a416-495e-9a06-c0146add645a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[37]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EAS/EAS_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 957252 </span><span class="se">\</span>
<span class="s2">--to-bp 957252 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_EAS/TMEM175_rs3733344 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/TMEM175_rs3733344.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/EAS_TMEM175
  --chr 4
  --double-id
  --from-bp 957252
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/TMEM175_rs3733344
  --recode A
  --to-bp 957252

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 3171 variants loaded from .bim file.
5167 people (3278 males, 1889 females) loaded from .fam.
5123 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 5167 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.915425.
1 variant and 5167 people pass filters and QC.
Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EAS/TMEM175_rs3733344.raw ...
101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=5532f40f-4d4f-475c-95a3-36e456239469">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[38]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EAS/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs3733344</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EAS/TMEM175_rs3733344.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4ba005dd-8507-4352-b66d-0c28cff7569f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[39]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs3733344</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[39]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:957252:T:G_G</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000111_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000113_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000135_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000186_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>5162</th>
<td>0</td>
<td>VIPD_000429_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
</tr>
<tr>
<th>5163</th>
<td>0</td>
<td>VIPD_000459_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
</tr>
<tr>
<th>5164</th>
<td>0</td>
<td>VIPD_000488_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
</tr>
<tr>
<th>5165</th>
<td>0</td>
<td>VIPD_000518_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>2.0</td>
</tr>
<tr>
<th>5166</th>
<td>0</td>
<td>VIPD_000542_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
</tbody>
</table>
<p>5167 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c7cdaccd-a6f9-43b4-9aa4-573049cd22f6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[40]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EAS/EAS_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[40]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4cd29a2f-b0a3-4a43-955c-6c6ba4816ba0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[41]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[41]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:957252:T:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.099685</td>
<td>83.973722</td>
<td>-10.188699</td>
<td>-28.239733</td>
<td>1.512131</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000111_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.020378</td>
<td>69.201961</td>
<td>2.908329</td>
<td>-15.959031</td>
<td>-1.925108</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000113_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-12.691932</td>
<td>85.440708</td>
<td>-10.451702</td>
<td>-23.738136</td>
<td>0.265317</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000135_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.713387</td>
<td>85.918556</td>
<td>-11.571290</td>
<td>-21.040961</td>
<td>0.389155</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000186_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.635789</td>
<td>82.221344</td>
<td>-10.439033</td>
<td>-23.815033</td>
<td>-4.063397</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>5162</th>
<td>0</td>
<td>VIPD_000429_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.0</td>
<td>1.0</td>
<td>-13.523491</td>
<td>87.708089</td>
<td>-12.864001</td>
<td>-21.945369</td>
<td>-1.873662</td>
</tr>
<tr>
<th>5163</th>
<td>0</td>
<td>VIPD_000459_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>1.0</td>
<td>-14.882064</td>
<td>85.648896</td>
<td>-10.945596</td>
<td>-20.418271</td>
<td>0.562908</td>
</tr>
<tr>
<th>5164</th>
<td>0</td>
<td>VIPD_000488_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>82.0</td>
<td>1.0</td>
<td>-10.597269</td>
<td>84.769771</td>
<td>-11.325140</td>
<td>-24.225874</td>
<td>-1.196641</td>
</tr>
<tr>
<th>5165</th>
<td>0</td>
<td>VIPD_000518_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>55.0</td>
<td>1.0</td>
<td>-13.600388</td>
<td>84.885882</td>
<td>-8.698538</td>
<td>-26.835134</td>
<td>-0.078363</td>
</tr>
<tr>
<th>5166</th>
<td>0</td>
<td>VIPD_000542_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>67.0</td>
<td>2.0</td>
<td>-16.256692</td>
<td>82.899155</td>
<td>-10.399602</td>
<td>-26.915106</td>
<td>-0.286868</td>
</tr>
</tbody>
</table>
<p>5167 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=bacbaf4f-b3da-49d5-a5a0-a353cfa90a45">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[42]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[42]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:957252:T:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.099685</td>
<td>83.973722</td>
<td>-10.188699</td>
<td>-28.239733</td>
<td>1.512131</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000111_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>NaN</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.020378</td>
<td>69.201961</td>
<td>2.908329</td>
<td>-15.959031</td>
<td>-1.925108</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000113_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-12.691932</td>
<td>85.440708</td>
<td>-10.451702</td>
<td>-23.738136</td>
<td>0.265317</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000135_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.713387</td>
<td>85.918556</td>
<td>-11.571290</td>
<td>-21.040961</td>
<td>0.389155</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000186_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-11.635789</td>
<td>82.221344</td>
<td>-10.439033</td>
<td>-23.815033</td>
<td>-4.063397</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>5162</th>
<td>0</td>
<td>VIPD_000429_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.0</td>
<td>1.0</td>
<td>-13.523491</td>
<td>87.708089</td>
<td>-12.864001</td>
<td>-21.945369</td>
<td>-1.873662</td>
</tr>
<tr>
<th>5163</th>
<td>0</td>
<td>VIPD_000459_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>1.0</td>
<td>-14.882064</td>
<td>85.648896</td>
<td>-10.945596</td>
<td>-20.418271</td>
<td>0.562908</td>
</tr>
<tr>
<th>5164</th>
<td>0</td>
<td>VIPD_000488_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>82.0</td>
<td>1.0</td>
<td>-10.597269</td>
<td>84.769771</td>
<td>-11.325140</td>
<td>-24.225874</td>
<td>-1.196641</td>
</tr>
<tr>
<th>5165</th>
<td>0</td>
<td>VIPD_000518_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>55.0</td>
<td>1.0</td>
<td>-13.600388</td>
<td>84.885882</td>
<td>-8.698538</td>
<td>-26.835134</td>
<td>-0.078363</td>
</tr>
<tr>
<th>5166</th>
<td>0</td>
<td>VIPD_000542_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>67.0</td>
<td>2.0</td>
<td>-16.256692</td>
<td>82.899155</td>
<td>-10.399602</td>
<td>-26.915106</td>
<td>-0.286868</td>
</tr>
</tbody>
</table>
<p>5123 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=b1a799fa-a96c-4e68-969f-87e6cc12bb80">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[43]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:957252:T:G_G"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=80404511-a80d-4fb5-b931-404cee091df7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[44]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs3733344"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[44]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs3733344</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000051_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-11.099685</td>
<td>83.973722</td>
<td>-10.188699</td>
<td>-28.239733</td>
<td>1.512131</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000111_s1</td>
<td>2</td>
<td>NaN</td>
<td>1.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-11.020378</td>
<td>69.201961</td>
<td>2.908329</td>
<td>-15.959031</td>
<td>-1.925108</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000113_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-12.691932</td>
<td>85.440708</td>
<td>-10.451702</td>
<td>-23.738136</td>
<td>0.265317</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000135_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-11.713387</td>
<td>85.918556</td>
<td>-11.571290</td>
<td>-21.040961</td>
<td>0.389155</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000186_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-11.635789</td>
<td>82.221344</td>
<td>-10.439033</td>
<td>-23.815033</td>
<td>-4.063397</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>5162</th>
<td>VIPD_000429_s1</td>
<td>1</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>63.0</td>
<td>-13.523491</td>
<td>87.708089</td>
<td>-12.864001</td>
<td>-21.945369</td>
<td>-1.873662</td>
</tr>
<tr>
<th>5163</th>
<td>VIPD_000459_s1</td>
<td>1</td>
<td>0.0</td>
<td>1.0</td>
<td>1.0</td>
<td>73.0</td>
<td>-14.882064</td>
<td>85.648896</td>
<td>-10.945596</td>
<td>-20.418271</td>
<td>0.562908</td>
</tr>
<tr>
<th>5164</th>
<td>VIPD_000488_s1</td>
<td>1</td>
<td>1.0</td>
<td>1.0</td>
<td>2.0</td>
<td>82.0</td>
<td>-10.597269</td>
<td>84.769771</td>
<td>-11.325140</td>
<td>-24.225874</td>
<td>-1.196641</td>
</tr>
<tr>
<th>5165</th>
<td>VIPD_000518_s1</td>
<td>1</td>
<td>0.0</td>
<td>2.0</td>
<td>2.0</td>
<td>55.0</td>
<td>-13.600388</td>
<td>84.885882</td>
<td>-8.698538</td>
<td>-26.835134</td>
<td>-0.078363</td>
</tr>
<tr>
<th>5166</th>
<td>VIPD_000542_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>2.0</td>
<td>67.0</td>
<td>-16.256692</td>
<td>82.899155</td>
<td>-10.399602</td>
<td>-26.915106</td>
<td>-0.286868</td>
</tr>
</tbody>
</table>
<p>5123 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=1fcc7d50-5d50-45ae-8e8d-9037a07a6081">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[45]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EAS/TMEM175_rs34311866_nalls_rs3733344.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=0f3bedfb-4f62-4463-ab2b-f35ea2c3efb0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[46]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_EAS/TMEM175_rs34311866_nalls_rs3733344.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=61492f80-805a-4918-9f4b-06e498fad628">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[47]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs3733344 SEX AGE
1 0 APGS_000051_s1         2                        0                 0   1  NA
2 1 APGS_000111_s1         2                       NA                 1   2  NA
3 2 APGS_000113_s1         2                        0                 0   1  NA
4 3 APGS_000135_s1         2                        1                 0   1  NA
5 4 APGS_000186_s1         2                        0                 0   1  NA
6 5 APGS_000201_s1         2                        0                 1   1  NA
        PC1      PC2        PC3       PC4        PC5
1 -11.09969 83.97372 -10.188699 -28.23973  1.5121309
2 -11.02038 69.20196   2.908329 -15.95903 -1.9251078
3 -12.69193 85.44071 -10.451702 -23.73814  0.2653173
4 -11.71339 85.91856 -11.571290 -21.04096  0.3891547
5 -11.63579 82.22134 -10.439033 -23.81503 -4.0633972
6 -12.55287 83.85939 -10.935292 -23.49726 -1.3028277
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=719f17d7-e17e-4c64-94b8-269b1ac307fa">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[48]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs3733344</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs3733344, data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.4003 -0.2807 -0.2807  0.5997  1.8389 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        0.40032    0.01225   32.67   &lt;2e-16 ***
TMEM175_rs3733344 -0.11959    0.01058  -11.31   &lt;2e-16 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.4965 on 4279 degrees of freedom
  (842 observations deleted due to missingness)
Multiple R-squared:  0.02901,	Adjusted R-squared:  0.02878 
F-statistic: 127.8 on 1 and 4279 DF,  p-value: &lt; 2.2e-16

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8dd2ea36-38f1-4128-ae44-4984b2b52a1e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[49]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs3733344</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs3733344 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               0.6578537  0.2514478   2.616  0.00895 ** 
TMEM175_rs34311866_nalls  0.0263518  0.0196011   1.344  0.17896    
TMEM175_rs3733344        -0.0333941  0.0142200  -2.348  0.01895 *  
SEX                      -0.0780377  0.0193676  -4.029 5.79e-05 ***
AGE                       0.0074881  0.0009197   8.141 6.62e-16 ***
PC1                      -0.0052974  0.0058178  -0.911  0.36264    
PC2                      -0.0031903  0.0035305  -0.904  0.36628    
PC3                      -0.0136178  0.0033581  -4.055 5.19e-05 ***
PC4                      -0.0291088  0.0032354  -8.997  &lt; 2e-16 ***
PC5                       0.0033633  0.0058578   0.574  0.56592    
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1963404)

    Null deviance: 526.75  on 2106  degrees of freedom
Residual deviance: 411.73  on 2097  degrees of freedom
  (3016 observations deleted due to missingness)
AIC: 2561.4

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=8af20ee9-ac94-4619-8260-0f860738ffcd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs11736451)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=88b8acbe-0b67-40b9-8b49-cf2133c97dcb">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[50]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_MDE/MDE_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 958159 </span><span class="se">\</span>
<span class="s2">--to-bp 958159 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_MDE/TMEM175_rs34311866_nalls </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/TMEM175_rs34311866_nalls.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/MDE_TMEM175
  --chr 4
  --double-id
  --from-bp 958159
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/TMEM175_rs34311866_nalls
  --recode A
  --to-bp 958159

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 1945 variants loaded from .bim file.
581 people (369 males, 212 females) loaded from .fam.
536 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 581 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.927711.
1 variant and 581 people pass filters and QC.
Among remaining phenotypes, 311 are cases and 225 are controls.  (45 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/TMEM175_rs34311866_nalls.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=57aac5be-425c-474f-9315-0d66918c9901">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[51]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_MDE/MDE_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 983234 </span><span class="se">\</span>
<span class="s2">--to-bp 983234 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_MDE/TMEM175_rs11736451 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/TMEM175_rs11736451.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/MDE_TMEM175
  --chr 4
  --double-id
  --from-bp 983234
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/TMEM175_rs11736451
  --recode A
  --to-bp 983234

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 1945 variants loaded from .bim file.
581 people (369 males, 212 females) loaded from .fam.
536 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 581 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.996558.
1 variant and 581 people pass filters and QC.
Among remaining phenotypes, 311 are cases and 225 are controls.  (45 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_MDE/TMEM175_rs11736451.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=f4583cda-7659-4db9-96b8-6fde910d7f44">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[52]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_MDE/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs11736451</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_MDE/TMEM175_rs11736451.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0d805355-278f-46a8-bece-c318b6dce2f3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[53]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs11736451</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[53]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:983234:A:G_G</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000259_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000450_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000533_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_001032_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_001052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>576</th>
<td>0</td>
<td>UPENNU19_000213_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
</tr>
<tr>
<th>577</th>
<td>0</td>
<td>VIPD_000392_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>578</th>
<td>0</td>
<td>VIPD_000513_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>579</th>
<td>0</td>
<td>VIPD_000686_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>580</th>
<td>0</td>
<td>YMS_000025_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<p>581 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=893b1a0f-7c03-43fd-ab69-8394339243be">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[54]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_MDE/MDE_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[54]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=efa4fc11-b6ed-4851-b2dd-1e3867c8331f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[55]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[55]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:983234:A:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000259_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.473365</td>
<td>-35.756649</td>
<td>-2.366968</td>
<td>-14.380825</td>
<td>-18.192957</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000450_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.280911</td>
<td>-26.394449</td>
<td>3.902227</td>
<td>-9.422375</td>
<td>-7.879637</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000533_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-26.885886</td>
<td>-36.017059</td>
<td>-1.258755</td>
<td>-16.204448</td>
<td>-17.018512</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_001032_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-25.519682</td>
<td>-34.281074</td>
<td>3.505687</td>
<td>-7.390488</td>
<td>-14.634728</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_001052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-23.508154</td>
<td>-35.614394</td>
<td>-2.816420</td>
<td>-15.496602</td>
<td>-15.969493</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>576</th>
<td>0</td>
<td>UPENNU19_000213_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>60.952778</td>
<td>2.0</td>
<td>-31.049844</td>
<td>-37.984529</td>
<td>-4.238697</td>
<td>-12.146182</td>
<td>-13.399832</td>
</tr>
<tr>
<th>577</th>
<td>0</td>
<td>VIPD_000392_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>82.000000</td>
<td>2.0</td>
<td>-19.256276</td>
<td>-35.006982</td>
<td>-0.734883</td>
<td>-14.500160</td>
<td>-17.138191</td>
</tr>
<tr>
<th>578</th>
<td>0</td>
<td>VIPD_000513_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>70.000000</td>
<td>2.0</td>
<td>-29.819688</td>
<td>-33.986530</td>
<td>-0.535363</td>
<td>-12.587826</td>
<td>-20.197365</td>
</tr>
<tr>
<th>579</th>
<td>0</td>
<td>VIPD_000686_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-21.248868</td>
<td>-30.072230</td>
<td>-2.408483</td>
<td>-14.194412</td>
<td>-11.072102</td>
</tr>
<tr>
<th>580</th>
<td>0</td>
<td>YMS_000025_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>84.000000</td>
<td>2.0</td>
<td>-29.200935</td>
<td>-33.914850</td>
<td>-3.377396</td>
<td>-10.866576</td>
<td>-11.084782</td>
</tr>
</tbody>
</table>
<p>581 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=7e6c6220-3633-4938-9d0c-b7d8556b4383">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[56]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[56]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:983234:A:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000259_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.473365</td>
<td>-35.756649</td>
<td>-2.366968</td>
<td>-14.380825</td>
<td>-18.192957</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000450_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-27.280911</td>
<td>-26.394449</td>
<td>3.902227</td>
<td>-9.422375</td>
<td>-7.879637</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000533_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-26.885886</td>
<td>-36.017059</td>
<td>-1.258755</td>
<td>-16.204448</td>
<td>-17.018512</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_001032_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-25.519682</td>
<td>-34.281074</td>
<td>3.505687</td>
<td>-7.390488</td>
<td>-14.634728</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_001052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-23.508154</td>
<td>-35.614394</td>
<td>-2.816420</td>
<td>-15.496602</td>
<td>-15.969493</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>576</th>
<td>0</td>
<td>UPENNU19_000213_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>60.952778</td>
<td>2.0</td>
<td>-31.049844</td>
<td>-37.984529</td>
<td>-4.238697</td>
<td>-12.146182</td>
<td>-13.399832</td>
</tr>
<tr>
<th>577</th>
<td>0</td>
<td>VIPD_000392_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>82.000000</td>
<td>2.0</td>
<td>-19.256276</td>
<td>-35.006982</td>
<td>-0.734883</td>
<td>-14.500160</td>
<td>-17.138191</td>
</tr>
<tr>
<th>578</th>
<td>0</td>
<td>VIPD_000513_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>70.000000</td>
<td>2.0</td>
<td>-29.819688</td>
<td>-33.986530</td>
<td>-0.535363</td>
<td>-12.587826</td>
<td>-20.197365</td>
</tr>
<tr>
<th>579</th>
<td>0</td>
<td>VIPD_000686_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-21.248868</td>
<td>-30.072230</td>
<td>-2.408483</td>
<td>-14.194412</td>
<td>-11.072102</td>
</tr>
<tr>
<th>580</th>
<td>0</td>
<td>YMS_000025_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>84.000000</td>
<td>2.0</td>
<td>-29.200935</td>
<td>-33.914850</td>
<td>-3.377396</td>
<td>-10.866576</td>
<td>-11.084782</td>
</tr>
</tbody>
</table>
<p>536 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=d8848784-b92a-42ea-aa82-740f0b6d925e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[57]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:983234:A:G_G"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=eee31d9d-0c06-4643-8606-a4468e44421c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[58]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs11736451"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[58]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs11736451</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000259_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-27.473365</td>
<td>-35.756649</td>
<td>-2.366968</td>
<td>-14.380825</td>
<td>-18.192957</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000450_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-27.280911</td>
<td>-26.394449</td>
<td>3.902227</td>
<td>-9.422375</td>
<td>-7.879637</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000533_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-26.885886</td>
<td>-36.017059</td>
<td>-1.258755</td>
<td>-16.204448</td>
<td>-17.018512</td>
</tr>
<tr>
<th>3</th>
<td>APGS_001032_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-25.519682</td>
<td>-34.281074</td>
<td>3.505687</td>
<td>-7.390488</td>
<td>-14.634728</td>
</tr>
<tr>
<th>4</th>
<td>APGS_001052_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-23.508154</td>
<td>-35.614394</td>
<td>-2.816420</td>
<td>-15.496602</td>
<td>-15.969493</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>576</th>
<td>UPENNU19_000213_s1</td>
<td>2</td>
<td>1.0</td>
<td>2.0</td>
<td>1.0</td>
<td>60.952778</td>
<td>-31.049844</td>
<td>-37.984529</td>
<td>-4.238697</td>
<td>-12.146182</td>
<td>-13.399832</td>
</tr>
<tr>
<th>577</th>
<td>VIPD_000392_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>82.000000</td>
<td>-19.256276</td>
<td>-35.006982</td>
<td>-0.734883</td>
<td>-14.500160</td>
<td>-17.138191</td>
</tr>
<tr>
<th>578</th>
<td>VIPD_000513_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>2.0</td>
<td>70.000000</td>
<td>-29.819688</td>
<td>-33.986530</td>
<td>-0.535363</td>
<td>-12.587826</td>
<td>-20.197365</td>
</tr>
<tr>
<th>579</th>
<td>VIPD_000686_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-21.248868</td>
<td>-30.072230</td>
<td>-2.408483</td>
<td>-14.194412</td>
<td>-11.072102</td>
</tr>
<tr>
<th>580</th>
<td>YMS_000025_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>84.000000</td>
<td>-29.200935</td>
<td>-33.914850</td>
<td>-3.377396</td>
<td>-10.866576</td>
<td>-11.084782</td>
</tr>
</tbody>
</table>
<p>536 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=42605b52-3183-40a0-beeb-f7b730208285">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[59]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_MDE/TMEM175_rs34311866_nalls_rs11736451.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=73cc81b9-eab7-4e00-b31b-0c2f94acb9ea">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[60]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_MDE/TMEM175_rs34311866_nalls_rs11736451.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=6e9830a0-0749-451d-8f99-eb86ba2bbc76">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[61]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs11736451 SEX
1 0 APGS_000259_s1         2                        0                  0   1
2 1 APGS_000450_s1         2                        0                  1   1
3 2 APGS_000533_s1         2                        0                  0   1
4 3 APGS_001032_s1         2                        0                  0   1
5 4 APGS_001052_s1         2                        0                  0   1
6 5 APGS_001085_s1         2                        0                  0   2
  AGE       PC1       PC2       PC3        PC4        PC5
1  NA -27.47337 -35.75665 -2.366968 -14.380825 -18.192957
2  NA -27.28091 -26.39445  3.902227  -9.422375  -7.879637
3  NA -26.88589 -36.01706 -1.258755 -16.204448 -17.018512
4  NA -25.51968 -34.28107  3.505687  -7.390488 -14.634728
5  NA -23.50815 -35.61439 -2.816420 -15.496602 -15.969493
6  NA -27.14900 -34.28756 -2.630555 -15.052248 -16.566321
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=79ca4cc1-9a85-4202-85db-5bb7866675f2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[62]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs11736451</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs11736451, data = total)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.64573 -0.21563 -0.21563  0.06932  1.78437 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)         0.21563    0.02574   8.376 5.77e-16 ***
TMEM175_rs11736451  0.71505    0.04180  17.107  &lt; 2e-16 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.487 on 491 degrees of freedom
  (43 observations deleted due to missingness)
Multiple R-squared:  0.3734,	Adjusted R-squared:  0.3722 
F-statistic: 292.6 on 1 and 491 DF,  p-value: &lt; 2.2e-16

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=fba4f97f-7541-4d76-95f4-0be0cdd597c0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[63]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs11736451</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs11736451 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                          Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               1.186923   0.214160   5.542 6.05e-08 ***
TMEM175_rs34311866_nalls  0.047466   0.050130   0.947   0.3444    
TMEM175_rs11736451        0.104750   0.058520   1.790   0.0744 .  
SEX                      -0.087899   0.050259  -1.749   0.0812 .  
AGE                       0.013538   0.002143   6.318 8.42e-10 ***
PC1                       0.009936   0.004588   2.166   0.0311 *  
PC2                       0.010594   0.007843   1.351   0.1777    
PC3                      -0.013670   0.009357  -1.461   0.1449    
PC4                       0.003441   0.012337   0.279   0.7805    
PC5                      -0.015837   0.009017  -1.756   0.0799 .  
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.2002292)

    Null deviance: 86.211  on 345  degrees of freedom
Residual deviance: 67.277  on 336  degrees of freedom
  (190 observations deleted due to missingness)
AIC: 437.29

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=bf232fe5-1201-408d-b4dd-200caee8b17f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs41286661)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=ef0912bb-a18c-453a-9c73-7558f7517dd4">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[64]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_SAS/SAS_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 958159 </span><span class="se">\</span>
<span class="s2">--to-bp 958159 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_SAS/TMEM175_rs34311866_nalls </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/TMEM175_rs34311866_nalls.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/SAS_TMEM175
  --chr 4
  --double-id
  --from-bp 958159
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/TMEM175_rs34311866_nalls
  --recode A
  --to-bp 958159

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 1728 variants loaded from .bim file.
635 people (408 males, 227 females) loaded from .fam.
609 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 635 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.91811.
1 variant and 635 people pass filters and QC.
Among remaining phenotypes, 387 are cases and 222 are controls.  (26 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/TMEM175_rs34311866_nalls.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=516438af-a883-4180-868a-8747c73b1487">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[65]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_SAS/SAS_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 963496 </span><span class="se">\</span>
<span class="s2">--to-bp 963496 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_SAS/TMEM175_rs41286661 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/TMEM175_rs41286661.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/SAS_TMEM175
  --chr 4
  --double-id
  --from-bp 963496
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/TMEM175_rs41286661
  --recode A
  --to-bp 963496

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 1728 variants loaded from .bim file.
635 people (408 males, 227 females) loaded from .fam.
609 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 635 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.982677.
1 variant and 635 people pass filters and QC.
Among remaining phenotypes, 387 are cases and 222 are controls.  (26 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_SAS/TMEM175_rs41286661.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=906e1de5-7f91-4101-a21d-e06a4328e865">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[119]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_SAS/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs41286661</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_SAS/TMEM175_rs41286661.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=10d92ef4-0af6-4634-ac04-1e4734767926">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[120]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs41286661</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[120]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:963496:C:T_T</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000062_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000119_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000331_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000378_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>2.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000454_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>630</th>
<td>0</td>
<td>VIPD_000468_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>631</th>
<td>0</td>
<td>VIPD_000528_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
</tr>
<tr>
<th>632</th>
<td>0</td>
<td>VIPD_000621_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>633</th>
<td>0</td>
<td>VIPD_000685_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>634</th>
<td>0</td>
<td>VIPD_000749_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
</tbody>
</table>
<p>635 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=16e35989-fce9-428c-90f9-e7166703fa47">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[121]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_SAS/SAS_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[121]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=58394dfc-442d-42f3-9b06-1f67d7397c3a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[122]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[122]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:963496:C:T_T</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000062_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-18.643078</td>
<td>10.448041</td>
<td>56.455919</td>
<td>19.558080</td>
<td>2.613308</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000119_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-17.346120</td>
<td>11.654072</td>
<td>54.213536</td>
<td>17.490254</td>
<td>-0.899534</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000331_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-17.542875</td>
<td>9.571579</td>
<td>54.761073</td>
<td>20.988884</td>
<td>-3.120079</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000378_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-18.133737</td>
<td>10.699043</td>
<td>54.508907</td>
<td>21.545525</td>
<td>-0.074081</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000454_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-18.943754</td>
<td>11.428258</td>
<td>55.862024</td>
<td>17.373778</td>
<td>2.199425</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>630</th>
<td>0</td>
<td>VIPD_000468_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>77.0</td>
<td>2.0</td>
<td>-23.470497</td>
<td>-4.351335</td>
<td>34.866203</td>
<td>13.610903</td>
<td>-2.024323</td>
</tr>
<tr>
<th>631</th>
<td>0</td>
<td>VIPD_000528_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>56.0</td>
<td>1.0</td>
<td>-19.677553</td>
<td>7.093584</td>
<td>45.937983</td>
<td>17.255940</td>
<td>0.348782</td>
</tr>
<tr>
<th>632</th>
<td>0</td>
<td>VIPD_000621_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.0</td>
<td>2.0</td>
<td>-19.140984</td>
<td>8.864642</td>
<td>56.528287</td>
<td>21.219827</td>
<td>1.010630</td>
</tr>
<tr>
<th>633</th>
<td>0</td>
<td>VIPD_000685_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-20.743509</td>
<td>-5.905185</td>
<td>37.703391</td>
<td>11.985115</td>
<td>-1.618144</td>
</tr>
<tr>
<th>634</th>
<td>0</td>
<td>VIPD_000749_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.0</td>
<td>2.0</td>
<td>-19.553184</td>
<td>11.077353</td>
<td>54.546702</td>
<td>20.627734</td>
<td>-0.300472</td>
</tr>
</tbody>
</table>
<p>635 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a0f8b7a2-2cb6-4f39-9ba5-a4ad4f2cb35d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[123]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[123]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:963496:C:T_T</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000062_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-18.643078</td>
<td>10.448041</td>
<td>56.455919</td>
<td>19.558080</td>
<td>2.613308</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000119_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-17.346120</td>
<td>11.654072</td>
<td>54.213536</td>
<td>17.490254</td>
<td>-0.899534</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000331_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-17.542875</td>
<td>9.571579</td>
<td>54.761073</td>
<td>20.988884</td>
<td>-3.120079</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000378_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-18.133737</td>
<td>10.699043</td>
<td>54.508907</td>
<td>21.545525</td>
<td>-0.074081</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000454_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-18.943754</td>
<td>11.428258</td>
<td>55.862024</td>
<td>17.373778</td>
<td>2.199425</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>630</th>
<td>0</td>
<td>VIPD_000468_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>77.0</td>
<td>2.0</td>
<td>-23.470497</td>
<td>-4.351335</td>
<td>34.866203</td>
<td>13.610903</td>
<td>-2.024323</td>
</tr>
<tr>
<th>631</th>
<td>0</td>
<td>VIPD_000528_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>56.0</td>
<td>1.0</td>
<td>-19.677553</td>
<td>7.093584</td>
<td>45.937983</td>
<td>17.255940</td>
<td>0.348782</td>
</tr>
<tr>
<th>632</th>
<td>0</td>
<td>VIPD_000621_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.0</td>
<td>2.0</td>
<td>-19.140984</td>
<td>8.864642</td>
<td>56.528287</td>
<td>21.219827</td>
<td>1.010630</td>
</tr>
<tr>
<th>633</th>
<td>0</td>
<td>VIPD_000685_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-20.743509</td>
<td>-5.905185</td>
<td>37.703391</td>
<td>11.985115</td>
<td>-1.618144</td>
</tr>
<tr>
<th>634</th>
<td>0</td>
<td>VIPD_000749_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.0</td>
<td>2.0</td>
<td>-19.553184</td>
<td>11.077353</td>
<td>54.546702</td>
<td>20.627734</td>
<td>-0.300472</td>
</tr>
</tbody>
</table>
<p>609 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=41ba355c-ea7d-4607-847d-193374db7031">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[124]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:963496:C:T_T"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=7c49e728-f629-430c-966a-55a85630ed37">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[125]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs41286661"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[125]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs41286661</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000062_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-18.643078</td>
<td>10.448041</td>
<td>56.455919</td>
<td>19.558080</td>
<td>2.613308</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000119_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-17.346120</td>
<td>11.654072</td>
<td>54.213536</td>
<td>17.490254</td>
<td>-0.899534</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000331_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-17.542875</td>
<td>9.571579</td>
<td>54.761073</td>
<td>20.988884</td>
<td>-3.120079</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000378_s1</td>
<td>2</td>
<td>0.0</td>
<td>2.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-18.133737</td>
<td>10.699043</td>
<td>54.508907</td>
<td>21.545525</td>
<td>-0.074081</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000454_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-18.943754</td>
<td>11.428258</td>
<td>55.862024</td>
<td>17.373778</td>
<td>2.199425</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>630</th>
<td>VIPD_000468_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>1.0</td>
<td>77.0</td>
<td>-23.470497</td>
<td>-4.351335</td>
<td>34.866203</td>
<td>13.610903</td>
<td>-2.024323</td>
</tr>
<tr>
<th>631</th>
<td>VIPD_000528_s1</td>
<td>1</td>
<td>2.0</td>
<td>1.0</td>
<td>2.0</td>
<td>56.0</td>
<td>-19.677553</td>
<td>7.093584</td>
<td>45.937983</td>
<td>17.255940</td>
<td>0.348782</td>
</tr>
<tr>
<th>632</th>
<td>VIPD_000621_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>60.0</td>
<td>-19.140984</td>
<td>8.864642</td>
<td>56.528287</td>
<td>21.219827</td>
<td>1.010630</td>
</tr>
<tr>
<th>633</th>
<td>VIPD_000685_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-20.743509</td>
<td>-5.905185</td>
<td>37.703391</td>
<td>11.985115</td>
<td>-1.618144</td>
</tr>
<tr>
<th>634</th>
<td>VIPD_000749_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>2.0</td>
<td>63.0</td>
<td>-19.553184</td>
<td>11.077353</td>
<td>54.546702</td>
<td>20.627734</td>
<td>-0.300472</td>
</tr>
</tbody>
</table>
<p>609 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=e349fa63-45d9-4c9d-b3b9-0992323149c1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[126]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_SAS/TMEM175_rs34311866_nalls_rs41286661.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=ab81ea01-3c2e-4f3d-ab22-6dd6d3130742">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[127]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_SAS/TMEM175_rs34311866_nalls_rs41286661.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4b606074-72e1-48fa-a374-1410de6afea4">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[128]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs41286661 SEX
1 0 APGS_000062_s1         2                        1                  0   1
2 1 APGS_000119_s1         2                        0                  0   1
3 2 APGS_000331_s1         2                        1                  0   1
4 3 APGS_000378_s1         2                        0                  2   2
5 4 APGS_000454_s1         2                        0                  1   2
6 5 APGS_000502_s1         2                        1                  0   2
  AGE       PC1       PC2      PC3      PC4         PC5
1  NA -18.64308 10.448041 56.45592 19.55808  2.61330835
2  NA -17.34612 11.654072 54.21354 17.49025 -0.89953411
3  NA -17.54287  9.571579 54.76107 20.98888 -3.12007878
4  NA -18.13374 10.699043 54.50891 21.54553 -0.07408062
5  NA -18.94375 11.428258 55.86202 17.37378  2.19942509
6  NA -21.84607 -6.697138 32.28942 14.85375  4.39039445
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1ec4ba92-627e-4322-9406-d962116e715c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[129]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs41286661</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs41286661, data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.8047 -0.6437  0.2758  0.3563  1.3563 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)         0.64372    0.03587  17.947   &lt;2e-16 ***
TMEM175_rs41286661  0.08051    0.05049   1.594    0.111    
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.6771 on 547 degrees of freedom
  (60 observations deleted due to missingness)
Multiple R-squared:  0.004626,	Adjusted R-squared:  0.002806 
F-statistic: 2.542 on 1 and 547 DF,  p-value: 0.1114

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=a4b80060-de82-4b27-a7e8-d55ab14d47c3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[130]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs41286661</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs41286661 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                          Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               1.509685   0.330146   4.573 7.46e-06 ***
TMEM175_rs34311866_nalls  0.110888   0.040011   2.771  0.00599 ** 
TMEM175_rs41286661        0.078018   0.045723   1.706  0.08915 .  
SEX                      -0.118111   0.055894  -2.113  0.03555 *  
AGE                       0.008347   0.001917   4.354 1.93e-05 ***
PC1                      -0.010355   0.011495  -0.901  0.36851    
PC2                      -0.005929   0.004349  -1.363  0.17393    
PC3                      -0.017942   0.007805  -2.299  0.02231 *  
PC4                       0.018871   0.013792   1.368  0.17241    
PC5                       0.015171   0.011597   1.308  0.19196    
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1921166)

    Null deviance: 65.368  on 268  degrees of freedom
Residual deviance: 49.758  on 259  degrees of freedom
  (340 observations deleted due to missingness)
AIC: 331.44

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=3a60afaa-2363-4ac7-9a7a-5e01868a4127">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs6599388)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=02a9d6a3-7ab1-4d30-a448-64aadf12d2f5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[78]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AJ/AJ_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 958159 </span><span class="se">\</span>
<span class="s2">--to-bp 958159 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AJ/TMEM175_rs34311866_nalls </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs34311866_nalls.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/AJ_TMEM175
  --chr 4
  --double-id
  --from-bp 958159
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs34311866_nalls
  --recode A
  --to-bp 958159

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 1285 variants loaded from .bim file.
2655 people (1648 males, 1007 females) loaded from .fam.
1703 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 2655 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.995104.
1 variant and 2655 people pass filters and QC.
Among remaining phenotypes, 1292 are cases and 411 are controls.  (952
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs34311866_nalls.raw
... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=64d8b199-e991-407b-adfe-7f042bde8ad1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[79]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AJ/AJ_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 945299 </span><span class="se">\</span>
<span class="s2">--to-bp 945299 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AJ/TMEM175_rs6599388 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs6599388.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/AJ_TMEM175
  --chr 4
  --double-id
  --from-bp 945299
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs6599388
  --recode A
  --to-bp 945299

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 1285 variants loaded from .bim file.
2655 people (1648 males, 1007 females) loaded from .fam.
1703 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 2655 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.999623.
1 variant and 2655 people pass filters and QC.
Among remaining phenotypes, 1292 are cases and 411 are controls.  (952
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs6599388.raw ...
101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=c1f4cd3e-3034-4ec8-8976-5a0c960400cd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[80]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs6599388</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/TMEM175_rs6599388.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=140c831a-ce96-4851-b324-8317a63b11e2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[81]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs6599388</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[81]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:945299:C:T_C</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000021_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000475_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000486_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000646_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000808_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>0</td>
<td>VIPD_000746_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>2651</th>
<td>0</td>
<td>VIPD_000748_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>2652</th>
<td>0</td>
<td>YMS_000030_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2653</th>
<td>0</td>
<td>YMS_000031_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
</tr>
<tr>
<th>2654</th>
<td>0</td>
<td>YMS_000038_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
</tr>
</tbody>
</table>
<p>2655 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=47406e86-e20b-4969-b1ee-9063625cdf42">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[82]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/AJ_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[82]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=0cf51672-bdc4-4985-93be-d1772157672f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[83]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[83]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:945299:C:T_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000021_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.028682</td>
<td>-35.735363</td>
<td>-6.580637</td>
<td>-17.501411</td>
<td>-23.026102</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000475_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.154728</td>
<td>-36.708128</td>
<td>-5.730325</td>
<td>-20.750956</td>
<td>-21.729629</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000486_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.810620</td>
<td>-37.561248</td>
<td>-9.373941</td>
<td>-16.773748</td>
<td>-27.482223</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000646_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-31.967732</td>
<td>-34.309571</td>
<td>-8.603258</td>
<td>-17.143810</td>
<td>-21.027547</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000808_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.929420</td>
<td>-33.911640</td>
<td>-9.610071</td>
<td>-18.087631</td>
<td>-24.559511</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>0</td>
<td>VIPD_000746_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>74.0</td>
<td>2.0</td>
<td>-31.455498</td>
<td>-35.058387</td>
<td>-9.777408</td>
<td>-18.399794</td>
<td>-27.254806</td>
</tr>
<tr>
<th>2651</th>
<td>0</td>
<td>VIPD_000748_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-31.319358</td>
<td>-39.960596</td>
<td>-10.932457</td>
<td>-17.389794</td>
<td>-1.963005</td>
</tr>
<tr>
<th>2652</th>
<td>0</td>
<td>YMS_000030_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>72.0</td>
<td>2.0</td>
<td>-29.635489</td>
<td>-34.956373</td>
<td>-6.466043</td>
<td>-14.761441</td>
<td>-20.758634</td>
</tr>
<tr>
<th>2653</th>
<td>0</td>
<td>YMS_000031_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>69.0</td>
<td>1.0</td>
<td>-32.862131</td>
<td>-35.306282</td>
<td>-9.166195</td>
<td>-15.693077</td>
<td>-21.088431</td>
</tr>
<tr>
<th>2654</th>
<td>0</td>
<td>YMS_000038_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>65.0</td>
<td>2.0</td>
<td>-31.121242</td>
<td>-36.512965</td>
<td>-5.037368</td>
<td>-16.566811</td>
<td>-24.178578</td>
</tr>
</tbody>
</table>
<p>2655 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=daa27db4-b5c4-4dc8-b9d7-6308970cbd8f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[84]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[84]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:945299:C:T_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000021_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.028682</td>
<td>-35.735363</td>
<td>-6.580637</td>
<td>-17.501411</td>
<td>-23.026102</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000475_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.154728</td>
<td>-36.708128</td>
<td>-5.730325</td>
<td>-20.750956</td>
<td>-21.729629</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000486_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.810620</td>
<td>-37.561248</td>
<td>-9.373941</td>
<td>-16.773748</td>
<td>-27.482223</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000646_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-31.967732</td>
<td>-34.309571</td>
<td>-8.603258</td>
<td>-17.143810</td>
<td>-21.027547</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000808_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.929420</td>
<td>-33.911640</td>
<td>-9.610071</td>
<td>-18.087631</td>
<td>-24.559511</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>0</td>
<td>VIPD_000746_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>74.0</td>
<td>2.0</td>
<td>-31.455498</td>
<td>-35.058387</td>
<td>-9.777408</td>
<td>-18.399794</td>
<td>-27.254806</td>
</tr>
<tr>
<th>2651</th>
<td>0</td>
<td>VIPD_000748_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-31.319358</td>
<td>-39.960596</td>
<td>-10.932457</td>
<td>-17.389794</td>
<td>-1.963005</td>
</tr>
<tr>
<th>2652</th>
<td>0</td>
<td>YMS_000030_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>72.0</td>
<td>2.0</td>
<td>-29.635489</td>
<td>-34.956373</td>
<td>-6.466043</td>
<td>-14.761441</td>
<td>-20.758634</td>
</tr>
<tr>
<th>2653</th>
<td>0</td>
<td>YMS_000031_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>69.0</td>
<td>1.0</td>
<td>-32.862131</td>
<td>-35.306282</td>
<td>-9.166195</td>
<td>-15.693077</td>
<td>-21.088431</td>
</tr>
<tr>
<th>2654</th>
<td>0</td>
<td>YMS_000038_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>65.0</td>
<td>2.0</td>
<td>-31.121242</td>
<td>-36.512965</td>
<td>-5.037368</td>
<td>-16.566811</td>
<td>-24.178578</td>
</tr>
</tbody>
</table>
<p>1703 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=ec340652-fd64-40c4-ba45-49433747fa8b">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[85]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:945299:C:T_C"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=23a212ba-4744-49ac-a6a1-ab40c4ff59a0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[86]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs6599388"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[86]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs6599388</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000021_s1</td>
<td>2</td>
<td>2.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-28.028682</td>
<td>-35.735363</td>
<td>-6.580637</td>
<td>-17.501411</td>
<td>-23.026102</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000475_s1</td>
<td>2</td>
<td>0.0</td>
<td>2.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-28.154728</td>
<td>-36.708128</td>
<td>-5.730325</td>
<td>-20.750956</td>
<td>-21.729629</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000486_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-29.810620</td>
<td>-37.561248</td>
<td>-9.373941</td>
<td>-16.773748</td>
<td>-27.482223</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000646_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-31.967732</td>
<td>-34.309571</td>
<td>-8.603258</td>
<td>-17.143810</td>
<td>-21.027547</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000808_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-29.929420</td>
<td>-33.911640</td>
<td>-9.610071</td>
<td>-18.087631</td>
<td>-24.559511</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>VIPD_000746_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>74.0</td>
<td>-31.455498</td>
<td>-35.058387</td>
<td>-9.777408</td>
<td>-18.399794</td>
<td>-27.254806</td>
</tr>
<tr>
<th>2651</th>
<td>VIPD_000748_s1</td>
<td>2</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>73.0</td>
<td>-31.319358</td>
<td>-39.960596</td>
<td>-10.932457</td>
<td>-17.389794</td>
<td>-1.963005</td>
</tr>
<tr>
<th>2652</th>
<td>YMS_000030_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>72.0</td>
<td>-29.635489</td>
<td>-34.956373</td>
<td>-6.466043</td>
<td>-14.761441</td>
<td>-20.758634</td>
</tr>
<tr>
<th>2653</th>
<td>YMS_000031_s1</td>
<td>1</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>69.0</td>
<td>-32.862131</td>
<td>-35.306282</td>
<td>-9.166195</td>
<td>-15.693077</td>
<td>-21.088431</td>
</tr>
<tr>
<th>2654</th>
<td>YMS_000038_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>1.0</td>
<td>65.0</td>
<td>-31.121242</td>
<td>-36.512965</td>
<td>-5.037368</td>
<td>-16.566811</td>
<td>-24.178578</td>
</tr>
</tbody>
</table>
<p>1703 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=9ce13a40-5d26-45fa-b05c-b80c6830bed0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[87]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/TMEM175_rs34311866_nalls_rs6599388.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=f679dc05-e5c9-4935-9f59-0c29662ab9c7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[88]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_AJ/TMEM175_rs34311866_nalls_rs6599388.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=92db3a5c-b737-4d41-8da6-fd1563cadee5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[89]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs6599388 SEX AGE
1 0 APGS_000021_s1         2                        2                 0   1  NA
2 1 APGS_000475_s1         2                        0                 2   1  NA
3 2 APGS_000486_s1         2                        0                 1   1  NA
4 3 APGS_000646_s1         2                        1                 0   2  NA
5 4 APGS_000808_s1         2                        0                 1   1  NA
6 5 APGS_001059_s1         2                        2                 0   1  NA
        PC1       PC2       PC3       PC4       PC5
1 -28.02868 -35.73536 -6.580637 -17.50141 -23.02610
2 -28.15473 -36.70813 -5.730325 -20.75096 -21.72963
3 -29.81062 -37.56125 -9.373941 -16.77375 -27.48222
4 -31.96773 -34.30957 -8.603258 -17.14381 -21.02755
5 -29.92942 -33.91164 -9.610071 -18.08763 -24.55951
6 -28.19315 -35.13333 -4.791519 -16.36048 -22.63633
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c275ce87-4f08-42f7-9874-2a72d5a1050a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[90]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs6599388</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs6599388, data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.2440 -0.2440 -0.0387  0.3586  1.3586 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        1.24402    0.02051   60.67   &lt;2e-16 ***
TMEM175_rs6599388 -0.60266    0.01739  -34.66   &lt;2e-16 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.5062 on 1694 degrees of freedom
  (7 observations deleted due to missingness)
Multiple R-squared:  0.4149,	Adjusted R-squared:  0.4146 
F-statistic:  1201 on 1 and 1694 DF,  p-value: &lt; 2.2e-16

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=194e559c-d4a5-4198-8f3f-049a7af832dc">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[91]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs6599388</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs6599388 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               2.1758085  0.2664038   8.167 6.99e-16 ***
TMEM175_rs34311866_nalls -0.0155229  0.0190741  -0.814  0.41588    
TMEM175_rs6599388        -0.0716701  0.0180552  -3.970 7.57e-05 ***
SEX                      -0.1095720  0.0203843  -5.375 8.95e-08 ***
AGE                       0.0030017  0.0009700   3.094  0.00201 ** 
PC1                       0.0044928  0.0061006   0.736  0.46157    
PC2                       0.0073501  0.0052384   1.403  0.16080    
PC3                      -0.0004471  0.0052311  -0.085  0.93191    
PC4                      -0.0069490  0.0053371  -1.302  0.19313    
PC5                       0.0018644  0.0025604   0.728  0.46664    
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1284528)

    Null deviance: 187.61  on 1404  degrees of freedom
Residual deviance: 179.19  on 1395  degrees of freedom
  (298 observations deleted due to missingness)
AIC: 1115.8

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=e007e71b-9bad-4d37-9e27-a6c8b18c0684">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs74391911)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=3795916b-97ff-4efe-a365-f322d309c84f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[92]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AJ/AJ_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 956095 </span><span class="se">\</span>
<span class="s2">--to-bp 956095 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AJ/TMEM175_rs74391911 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs74391911.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/AJ_TMEM175
  --chr 4
  --double-id
  --from-bp 956095
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs74391911
  --recode A
  --to-bp 956095

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 1285 variants loaded from .bim file.
2655 people (1648 males, 1007 females) loaded from .fam.
1703 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 2655 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.999247.
1 variant and 2655 people pass filters and QC.
Among remaining phenotypes, 1292 are cases and 411 are controls.  (952
phenotypes are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AJ/TMEM175_rs74391911.raw ...
101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=ab5ff0a8-32e5-4596-89e8-1a16c7eaa14e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[93]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs74391911</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/TMEM175_rs74391911.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5ed27a03-24c7-4bcd-b052-2468859c4b50">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[94]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs74391911</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[94]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:956095:C:G_G</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000021_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000475_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000486_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000646_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000808_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>0</td>
<td>VIPD_000746_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2651</th>
<td>0</td>
<td>VIPD_000748_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2652</th>
<td>0</td>
<td>YMS_000030_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2653</th>
<td>0</td>
<td>YMS_000031_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
</tr>
<tr>
<th>2654</th>
<td>0</td>
<td>YMS_000038_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
</tbody>
</table>
<p>2655 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e6166899-9a25-497d-bed6-2ca3ea0484bd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[95]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/AJ_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[95]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=808533ee-d070-4922-82e6-3723db7f9d30">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[96]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[96]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:956095:C:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000021_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.028682</td>
<td>-35.735363</td>
<td>-6.580637</td>
<td>-17.501411</td>
<td>-23.026102</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000475_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.154728</td>
<td>-36.708128</td>
<td>-5.730325</td>
<td>-20.750956</td>
<td>-21.729629</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000486_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.810620</td>
<td>-37.561248</td>
<td>-9.373941</td>
<td>-16.773748</td>
<td>-27.482223</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000646_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-31.967732</td>
<td>-34.309571</td>
<td>-8.603258</td>
<td>-17.143810</td>
<td>-21.027547</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000808_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.929420</td>
<td>-33.911640</td>
<td>-9.610071</td>
<td>-18.087631</td>
<td>-24.559511</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>0</td>
<td>VIPD_000746_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>74.0</td>
<td>2.0</td>
<td>-31.455498</td>
<td>-35.058387</td>
<td>-9.777408</td>
<td>-18.399794</td>
<td>-27.254806</td>
</tr>
<tr>
<th>2651</th>
<td>0</td>
<td>VIPD_000748_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-31.319358</td>
<td>-39.960596</td>
<td>-10.932457</td>
<td>-17.389794</td>
<td>-1.963005</td>
</tr>
<tr>
<th>2652</th>
<td>0</td>
<td>YMS_000030_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>72.0</td>
<td>2.0</td>
<td>-29.635489</td>
<td>-34.956373</td>
<td>-6.466043</td>
<td>-14.761441</td>
<td>-20.758634</td>
</tr>
<tr>
<th>2653</th>
<td>0</td>
<td>YMS_000031_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>69.0</td>
<td>1.0</td>
<td>-32.862131</td>
<td>-35.306282</td>
<td>-9.166195</td>
<td>-15.693077</td>
<td>-21.088431</td>
</tr>
<tr>
<th>2654</th>
<td>0</td>
<td>YMS_000038_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>65.0</td>
<td>2.0</td>
<td>-31.121242</td>
<td>-36.512965</td>
<td>-5.037368</td>
<td>-16.566811</td>
<td>-24.178578</td>
</tr>
</tbody>
</table>
<p>2655 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=79333ff4-21da-437e-8e7a-146d2dce9979">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[97]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[97]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:956095:C:G_G</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000021_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>2.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.028682</td>
<td>-35.735363</td>
<td>-6.580637</td>
<td>-17.501411</td>
<td>-23.026102</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000475_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-28.154728</td>
<td>-36.708128</td>
<td>-5.730325</td>
<td>-20.750956</td>
<td>-21.729629</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000486_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.810620</td>
<td>-37.561248</td>
<td>-9.373941</td>
<td>-16.773748</td>
<td>-27.482223</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000646_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-31.967732</td>
<td>-34.309571</td>
<td>-8.603258</td>
<td>-17.143810</td>
<td>-21.027547</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000808_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>-29.929420</td>
<td>-33.911640</td>
<td>-9.610071</td>
<td>-18.087631</td>
<td>-24.559511</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>0</td>
<td>VIPD_000746_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>74.0</td>
<td>2.0</td>
<td>-31.455498</td>
<td>-35.058387</td>
<td>-9.777408</td>
<td>-18.399794</td>
<td>-27.254806</td>
</tr>
<tr>
<th>2651</th>
<td>0</td>
<td>VIPD_000748_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>73.0</td>
<td>2.0</td>
<td>-31.319358</td>
<td>-39.960596</td>
<td>-10.932457</td>
<td>-17.389794</td>
<td>-1.963005</td>
</tr>
<tr>
<th>2652</th>
<td>0</td>
<td>YMS_000030_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>72.0</td>
<td>2.0</td>
<td>-29.635489</td>
<td>-34.956373</td>
<td>-6.466043</td>
<td>-14.761441</td>
<td>-20.758634</td>
</tr>
<tr>
<th>2653</th>
<td>0</td>
<td>YMS_000031_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>69.0</td>
<td>1.0</td>
<td>-32.862131</td>
<td>-35.306282</td>
<td>-9.166195</td>
<td>-15.693077</td>
<td>-21.088431</td>
</tr>
<tr>
<th>2654</th>
<td>0</td>
<td>YMS_000038_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>65.0</td>
<td>2.0</td>
<td>-31.121242</td>
<td>-36.512965</td>
<td>-5.037368</td>
<td>-16.566811</td>
<td>-24.178578</td>
</tr>
</tbody>
</table>
<p>1703 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=7f407949-666d-4b46-ba62-58ed341bc746">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[98]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:956095:C:G_G"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5eb8b198-a4f7-4e18-b5c0-7f29e3473167">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[99]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs74391911"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[99]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs74391911</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>APGS_000021_s1</td>
<td>2</td>
<td>2.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-28.028682</td>
<td>-35.735363</td>
<td>-6.580637</td>
<td>-17.501411</td>
<td>-23.026102</td>
</tr>
<tr>
<th>1</th>
<td>APGS_000475_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-28.154728</td>
<td>-36.708128</td>
<td>-5.730325</td>
<td>-20.750956</td>
<td>-21.729629</td>
</tr>
<tr>
<th>2</th>
<td>APGS_000486_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-29.810620</td>
<td>-37.561248</td>
<td>-9.373941</td>
<td>-16.773748</td>
<td>-27.482223</td>
</tr>
<tr>
<th>3</th>
<td>APGS_000646_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>-31.967732</td>
<td>-34.309571</td>
<td>-8.603258</td>
<td>-17.143810</td>
<td>-21.027547</td>
</tr>
<tr>
<th>4</th>
<td>APGS_000808_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>-29.929420</td>
<td>-33.911640</td>
<td>-9.610071</td>
<td>-18.087631</td>
<td>-24.559511</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>2650</th>
<td>VIPD_000746_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>74.0</td>
<td>-31.455498</td>
<td>-35.058387</td>
<td>-9.777408</td>
<td>-18.399794</td>
<td>-27.254806</td>
</tr>
<tr>
<th>2651</th>
<td>VIPD_000748_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>73.0</td>
<td>-31.319358</td>
<td>-39.960596</td>
<td>-10.932457</td>
<td>-17.389794</td>
<td>-1.963005</td>
</tr>
<tr>
<th>2652</th>
<td>YMS_000030_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>72.0</td>
<td>-29.635489</td>
<td>-34.956373</td>
<td>-6.466043</td>
<td>-14.761441</td>
<td>-20.758634</td>
</tr>
<tr>
<th>2653</th>
<td>YMS_000031_s1</td>
<td>1</td>
<td>1.0</td>
<td>0.0</td>
<td>1.0</td>
<td>69.0</td>
<td>-32.862131</td>
<td>-35.306282</td>
<td>-9.166195</td>
<td>-15.693077</td>
<td>-21.088431</td>
</tr>
<tr>
<th>2654</th>
<td>YMS_000038_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>65.0</td>
<td>-31.121242</td>
<td>-36.512965</td>
<td>-5.037368</td>
<td>-16.566811</td>
<td>-24.178578</td>
</tr>
</tbody>
</table>
<p>1703 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=7fd14645-f65b-42f6-b429-bc9a8db4d88d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[100]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AJ/TMEM175_rs34311866_nalls_rs74391911.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=778cb1de-4801-4c78-b732-931963267efd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[102]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_AJ/TMEM175_rs34311866_nalls_rs74391911.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=c7ad5ef2-144e-4a8e-915f-5de4c90474a9">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[103]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X            IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs74391911 SEX
1 0 APGS_000021_s1         2                        2                  0   1
2 1 APGS_000475_s1         2                        0                  0   1
3 2 APGS_000486_s1         2                        0                  0   1
4 3 APGS_000646_s1         2                        1                  0   2
5 4 APGS_000808_s1         2                        0                  0   1
6 5 APGS_001059_s1         2                        2                  0   1
  AGE       PC1       PC2       PC3       PC4       PC5
1  NA -28.02868 -35.73536 -6.580637 -17.50141 -23.02610
2  NA -28.15473 -36.70813 -5.730325 -20.75096 -21.72963
3  NA -29.81062 -37.56125 -9.373941 -16.77375 -27.48222
4  NA -31.96773 -34.30957 -8.603258 -17.14381 -21.02755
5  NA -29.92942 -33.91164 -9.610071 -18.08763 -24.55951
6  NA -28.19315 -35.13333 -4.791519 -16.36048 -22.63633
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=972744b7-13d1-4824-a790-0c3ae6360628">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[104]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs74391911</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs74391911, data = total)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.6832 -0.6832  0.3168  0.3168  1.3168 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)         0.68316    0.01621  42.142  &lt; 2e-16 ***
TMEM175_rs74391911 -0.35884    0.10969  -3.271  0.00109 ** 
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.6599 on 1692 degrees of freedom
  (9 observations deleted due to missingness)
Multiple R-squared:  0.006285,	Adjusted R-squared:  0.005698 
F-statistic:  10.7 on 1 and 1692 DF,  p-value: 0.001092

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8f1f0454-eda8-4c7e-acc4-1f87703b1e76">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[105]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs74391911</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs74391911 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               2.0268287  0.2643574   7.667 3.28e-14 ***
TMEM175_rs34311866_nalls  0.0290173  0.0145414   1.995  0.04618 *  
TMEM175_rs74391911       -0.3015576  0.0687406  -4.387 1.24e-05 ***
SEX                      -0.1092647  0.0203588  -5.367 9.37e-08 ***
AGE                       0.0030782  0.0009690   3.177  0.00152 ** 
PC1                       0.0026219  0.0060937   0.430  0.66707    
PC2                       0.0076134  0.0052298   1.456  0.14568    
PC3                       0.0006995  0.0052268   0.134  0.89357    
PC4                      -0.0068997  0.0053302  -1.294  0.19573    
PC5                       0.0011696  0.0025393   0.461  0.64515    
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.128136)

    Null deviance: 187.61  on 1404  degrees of freedom
Residual deviance: 178.75  on 1395  degrees of freedom
  (298 observations deleted due to missingness)
AIC: 1112.4

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=2e83040c-5097-42cc-8e04-3c03059b976c">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>Person and conditional analysis with glm (rs6856425)</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1ef38076-0378-4a7f-ad0a-5bf393628525">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[106]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>

<span class="n">command</span> <span class="o">=</span> <span class="sa">f</span><span class="s2">"""</span>
<span class="s2">/home/jupyter/tools/plink </span><span class="se">\</span>
<span class="s2">--bfile </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AAC/AAC_TMEM175 </span><span class="se">\</span>
<span class="s2">--chr 4 </span><span class="se">\</span>
<span class="s2">--from-bp 983130 </span><span class="se">\</span>
<span class="s2">--to-bp 983130 </span><span class="se">\</span>
<span class="s2">--recode A </span><span class="se">\</span>
<span class="s2">--out </span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s2">/TMEM175_AAC/TMEM175_rs6856425 </span><span class="se">\</span>
<span class="s2">--double-id</span>
<span class="s2">"""</span>

<span class="o">!{</span>command<span class="o">}</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs6856425.log.
Options in effect:
  --bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/AAC_TMEM175
  --chr 4
  --double-id
  --from-bp 983130
  --out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs6856425
  --recode A
  --to-bp 983130

52223 MB RAM detected; reserving 26111 MB for main workspace.
1 out of 3449 variants loaded from .bim file.
1111 people (455 males, 656 females) loaded from .fam.
1086 phenotype values loaded from .fam.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 1111 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.976598.
1 variant and 1111 people pass filters and QC.
Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes
are missing.)
--recode A to
/home/jupyter/workspace/ws_files/TMEM175//TMEM175_AAC/TMEM175_rs6856425.raw ...
101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=9be0a7bd-0aee-47f5-95d9-64f8bce3999f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[107]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">rs34311866</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/TMEM175_rs34311866_nalls.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
<span class="n">rs6856425</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/TMEM175_rs6856425.raw'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s2">" "</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e4ee5197-a429-457e-94dd-b09155429423">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[108]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">rs34311866</span><span class="p">,</span> <span class="n">rs6856425</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[108]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:983130:T:C_C</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>BCM_000142_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>BCM_000150_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>BCM_000204_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>BCM_000215_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>BCM_000270_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>0</td>
<td>UPENNU19_000409_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1107</th>
<td>0</td>
<td>UPENNU19_000432_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1108</th>
<td>0</td>
<td>VIPD_000151_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
</tr>
<tr>
<th>1109</th>
<td>0</td>
<td>VIPD_000679_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
</tr>
<tr>
<th>1110</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
</tr>
</tbody>
</table>
<p>1111 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8d802be3-b0de-4650-8be7-6a964caa9f25">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[109]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">cov</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/AAC_covariate_file.txt'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">)</span>
<span class="n">cov</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[109]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID</th>
<th>IID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>APGS_000002_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-29.483298</td>
<td>-38.473005</td>
<td>-9.398904</td>
<td>-10.842865</td>
<td>25.744648</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>APGS_000003_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.068958</td>
<td>-34.288862</td>
<td>-11.116951</td>
<td>-8.056146</td>
<td>22.924043</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>APGS_000004_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-28.204441</td>
<td>-30.969481</td>
<td>-15.883093</td>
<td>-7.004351</td>
<td>18.647007</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>APGS_000005_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-35.039753</td>
<td>-39.565822</td>
<td>-12.522753</td>
<td>-9.339179</td>
<td>21.956533</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>APGS_000006_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-30.279907</td>
<td>-34.988905</td>
<td>-10.664628</td>
<td>-7.407177</td>
<td>26.357548</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>58204</th>
<td>0</td>
<td>YMS_000049_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.320502</td>
<td>-37.170809</td>
<td>-9.867466</td>
<td>-6.818637</td>
<td>18.344976</td>
</tr>
<tr>
<th>58205</th>
<td>0</td>
<td>YMS_000050_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-33.550740</td>
<td>-38.934634</td>
<td>-13.550840</td>
<td>-11.161991</td>
<td>26.601555</td>
</tr>
<tr>
<th>58206</th>
<td>0</td>
<td>YMS_000051_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-36.415950</td>
<td>-36.634710</td>
<td>-9.623981</td>
<td>-8.333470</td>
<td>19.165537</td>
</tr>
<tr>
<th>58207</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>53.0</td>
<td>1.0</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
<tr>
<th>58208</th>
<td>0</td>
<td>YMS_000053_s1</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>-32.401921</td>
<td>-39.458156</td>
<td>-7.987121</td>
<td>-9.101960</td>
<td>19.051343</td>
</tr>
</tbody>
</table>
<p>58209 rows  12 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=bb5be62c-bdd1-4555-9b25-eeffffe21ec2">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[110]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">merge</span><span class="p">(</span><span class="n">total</span><span class="p">,</span> <span class="n">cov</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="s2">"IID"</span><span class="p">)</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[110]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:983130:T:C_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>BCM_000142_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>42.344694</td>
<td>-18.080400</td>
<td>-4.379556</td>
<td>-4.231518</td>
<td>11.945426</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>BCM_000150_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>46.211997</td>
<td>-16.728937</td>
<td>-2.939694</td>
<td>-4.306565</td>
<td>5.530103</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>BCM_000204_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>53.796533</td>
<td>-13.345479</td>
<td>-2.312777</td>
<td>-4.141204</td>
<td>3.542342</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>BCM_000215_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>33.149467</td>
<td>-17.891235</td>
<td>-5.542113</td>
<td>-2.706507</td>
<td>6.964059</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>BCM_000270_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>1.0</td>
<td>44.397319</td>
<td>-15.104927</td>
<td>-4.648018</td>
<td>-3.126988</td>
<td>7.345741</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>0</td>
<td>UPENNU19_000409_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>79.386111</td>
<td>2.0</td>
<td>59.974838</td>
<td>-11.604280</td>
<td>-4.482499</td>
<td>-2.758223</td>
<td>3.541477</td>
</tr>
<tr>
<th>1107</th>
<td>0</td>
<td>UPENNU19_000432_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>77.661111</td>
<td>2.0</td>
<td>55.937250</td>
<td>-11.575462</td>
<td>-5.033753</td>
<td>-2.485275</td>
<td>6.130544</td>
</tr>
<tr>
<th>1108</th>
<td>0</td>
<td>VIPD_000151_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.000000</td>
<td>2.0</td>
<td>45.339968</td>
<td>-18.329921</td>
<td>-5.496370</td>
<td>-0.623469</td>
<td>2.827065</td>
</tr>
<tr>
<th>1109</th>
<td>0</td>
<td>VIPD_000679_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.000000</td>
<td>2.0</td>
<td>24.426378</td>
<td>-24.074607</td>
<td>-5.665107</td>
<td>-6.488002</td>
<td>12.017162</td>
</tr>
<tr>
<th>1110</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>53.000000</td>
<td>1.0</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
</tbody>
</table>
<p>1111 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=fe0357dd-f04d-409b-ae53-92c8e585f602">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[111]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">1</span><span class="p">)</span> <span class="o">|</span> <span class="p">(</span><span class="n">total</span><span class="p">[</span><span class="s1">'PHENOTYPE_x'</span><span class="p">]</span><span class="o">==</span><span class="mi">2</span><span class="p">)]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[111]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>FID_x</th>
<th>IID</th>
<th>PAT_x</th>
<th>MAT_x</th>
<th>SEX_x</th>
<th>PHENOTYPE_x</th>
<th>chr4:958159:T:C_C</th>
<th>FID_y</th>
<th>PAT_y</th>
<th>MAT_y</th>
<th>SEX_y</th>
<th>PHENOTYPE_y</th>
<th>chr4:983130:T:C_C</th>
<th>FID</th>
<th>FATID</th>
<th>MATID</th>
<th>SEX</th>
<th>AGE</th>
<th>PHENO</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>BCM_000142_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>42.344694</td>
<td>-18.080400</td>
<td>-4.379556</td>
<td>-4.231518</td>
<td>11.945426</td>
</tr>
<tr>
<th>1</th>
<td>0</td>
<td>BCM_000150_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>46.211997</td>
<td>-16.728937</td>
<td>-2.939694</td>
<td>-4.306565</td>
<td>5.530103</td>
</tr>
<tr>
<th>2</th>
<td>0</td>
<td>BCM_000204_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>2.0</td>
<td>53.796533</td>
<td>-13.345479</td>
<td>-2.312777</td>
<td>-4.141204</td>
<td>3.542342</td>
</tr>
<tr>
<th>3</th>
<td>0</td>
<td>BCM_000215_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>NaN</td>
<td>2.0</td>
<td>33.149467</td>
<td>-17.891235</td>
<td>-5.542113</td>
<td>-2.706507</td>
<td>6.964059</td>
</tr>
<tr>
<th>4</th>
<td>0</td>
<td>BCM_000270_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>1</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>NaN</td>
<td>1.0</td>
<td>44.397319</td>
<td>-15.104927</td>
<td>-4.648018</td>
<td>-3.126988</td>
<td>7.345741</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>0</td>
<td>UPENNU19_000409_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>79.386111</td>
<td>2.0</td>
<td>59.974838</td>
<td>-11.604280</td>
<td>-4.482499</td>
<td>-2.758223</td>
<td>3.541477</td>
</tr>
<tr>
<th>1107</th>
<td>0</td>
<td>UPENNU19_000432_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>77.661111</td>
<td>2.0</td>
<td>55.937250</td>
<td>-11.575462</td>
<td>-5.033753</td>
<td>-2.485275</td>
<td>6.130544</td>
</tr>
<tr>
<th>1108</th>
<td>0</td>
<td>VIPD_000151_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>63.000000</td>
<td>2.0</td>
<td>45.339968</td>
<td>-18.329921</td>
<td>-5.496370</td>
<td>-0.623469</td>
<td>2.827065</td>
</tr>
<tr>
<th>1109</th>
<td>0</td>
<td>VIPD_000679_s1</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>0.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>2.0</td>
<td>60.000000</td>
<td>2.0</td>
<td>24.426378</td>
<td>-24.074607</td>
<td>-5.665107</td>
<td>-6.488002</td>
<td>12.017162</td>
</tr>
<tr>
<th>1110</th>
<td>0</td>
<td>YMS_000052_s1</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1.0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1.0</td>
<td>53.000000</td>
<td>1.0</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
</tbody>
</table>
<p>1086 rows  24 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=01e5a69a-fe9c-41d6-8a05-58ba3e8109ab">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[112]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span> <span class="o">=</span> <span class="n">total</span><span class="p">[[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE_x'</span><span class="p">,</span><span class="s2">"chr4:958159:T:C_C"</span><span class="p">,</span><span class="s2">"chr4:983130:T:C_C"</span><span class="p">,</span> <span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=2aca3912-fa25-47f3-82bc-f1e54985eb95">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[113]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">columns</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"IID"</span><span class="p">,</span><span class="s1">'PHENOTYPE'</span><span class="p">,</span><span class="s2">"TMEM175_rs34311866_nalls"</span><span class="p">,</span><span class="s2">"TMEM175_rs6856425"</span><span class="p">,</span><span class="s2">"SEX"</span><span class="p">,</span> <span class="s2">"AGE"</span><span class="p">,</span> <span class="s2">"PC1"</span><span class="p">,</span><span class="s2">"PC2"</span><span class="p">,</span><span class="s2">"PC3"</span><span class="p">,</span><span class="s2">"PC4"</span><span class="p">,</span><span class="s2">"PC5"</span><span class="p">]</span>
<span class="n">total</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[113]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>IID</th>
<th>PHENOTYPE</th>
<th>TMEM175_rs34311866_nalls</th>
<th>TMEM175_rs6856425</th>
<th>SEX</th>
<th>AGE</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
<th>PC4</th>
<th>PC5</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>BCM_000142_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>42.344694</td>
<td>-18.080400</td>
<td>-4.379556</td>
<td>-4.231518</td>
<td>11.945426</td>
</tr>
<tr>
<th>1</th>
<td>BCM_000150_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>46.211997</td>
<td>-16.728937</td>
<td>-2.939694</td>
<td>-4.306565</td>
<td>5.530103</td>
</tr>
<tr>
<th>2</th>
<td>BCM_000204_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>53.796533</td>
<td>-13.345479</td>
<td>-2.312777</td>
<td>-4.141204</td>
<td>3.542342</td>
</tr>
<tr>
<th>3</th>
<td>BCM_000215_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>NaN</td>
<td>33.149467</td>
<td>-17.891235</td>
<td>-5.542113</td>
<td>-2.706507</td>
<td>6.964059</td>
</tr>
<tr>
<th>4</th>
<td>BCM_000270_s1</td>
<td>1</td>
<td>0.0</td>
<td>0.0</td>
<td>2.0</td>
<td>NaN</td>
<td>44.397319</td>
<td>-15.104927</td>
<td>-4.648018</td>
<td>-3.126988</td>
<td>7.345741</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>1106</th>
<td>UPENNU19_000409_s1</td>
<td>2</td>
<td>0.0</td>
<td>0.0</td>
<td>1.0</td>
<td>79.386111</td>
<td>59.974838</td>
<td>-11.604280</td>
<td>-4.482499</td>
<td>-2.758223</td>
<td>3.541477</td>
</tr>
<tr>
<th>1107</th>
<td>UPENNU19_000432_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>77.661111</td>
<td>55.937250</td>
<td>-11.575462</td>
<td>-5.033753</td>
<td>-2.485275</td>
<td>6.130544</td>
</tr>
<tr>
<th>1108</th>
<td>VIPD_000151_s1</td>
<td>2</td>
<td>0.0</td>
<td>1.0</td>
<td>2.0</td>
<td>63.000000</td>
<td>45.339968</td>
<td>-18.329921</td>
<td>-5.496370</td>
<td>-0.623469</td>
<td>2.827065</td>
</tr>
<tr>
<th>1109</th>
<td>VIPD_000679_s1</td>
<td>2</td>
<td>1.0</td>
<td>0.0</td>
<td>2.0</td>
<td>60.000000</td>
<td>24.426378</td>
<td>-24.074607</td>
<td>-5.665107</td>
<td>-6.488002</td>
<td>12.017162</td>
</tr>
<tr>
<th>1110</th>
<td>YMS_000052_s1</td>
<td>1</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>53.000000</td>
<td>55.200734</td>
<td>-12.367759</td>
<td>-2.246785</td>
<td>-0.681236</td>
<td>5.281365</td>
</tr>
</tbody>
</table>
<p>1086 rows  11 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=1ea7a7fb-5aa2-4810-8e67-b81b2669d531">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[114]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">total</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_AAC/TMEM175_rs34311866_nalls_rs6856425.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=c1d1cf3f-be57-4ce4-8873-f74832ef52af">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[115]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">total</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">"~/workspace/ws_files/TMEM175/TMEM175_AAC/TMEM175_rs34311866_nalls_rs6856425.csv"</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=36090ff1-5a4e-457e-97d4-cf84b2220594">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[116]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="nf">head</span><span class="p">(</span><span class="n">total</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>  X           IID PHENOTYPE TMEM175_rs34311866_nalls TMEM175_rs6856425 SEX AGE
1 0 BCM_000142_s1         2                        0                 0   1  NA
2 1 BCM_000150_s1         2                        0                 0   2  NA
3 2 BCM_000204_s1         2                        0                 0   2  NA
4 3 BCM_000215_s1         2                        0                 0   1  NA
5 4 BCM_000270_s1         1                        0                 0   2  NA
6 5 BCM_000277_s1         1                        1                 0   2  NA
       PC1       PC2       PC3       PC4        PC5
1 42.34469 -18.08040 -4.379556 -4.231518 11.9454258
2 46.21200 -16.72894 -2.939694 -4.306565  5.5301026
3 53.79653 -13.34548 -2.312777 -4.141204  3.5423418
4 33.14947 -17.89123 -5.542113 -2.706507  6.9640586
5 44.39732 -15.10493 -4.648018 -3.126988  7.3457406
6 63.11613 -11.49276 -2.482421 -1.453234  0.2000133
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4f711007-e48d-4ca1-8133-2a9b8cbdc51a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[117]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>

<span class="n">correlation.model</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lm</span><span class="p">(</span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs6856425</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">correlation.model</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
lm(formula = TMEM175_rs34311866_nalls ~ TMEM175_rs6856425, data = total)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.11449 -0.11449 -0.11449 -0.04595  1.88551 

Coefficients:
                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        0.11449    0.01059  10.812  &lt; 2e-16 ***
TMEM175_rs6856425 -0.06854    0.01842  -3.721 0.000209 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

Residual standard error: 0.2975 on 1054 degrees of freedom
  (30 observations deleted due to missingness)
Multiple R-squared:  0.01297,	Adjusted R-squared:  0.01203 
F-statistic: 13.85 on 1 and 1054 DF,  p-value: 0.0002089

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=b22592b5-f712-440b-a520-8ff7c30934b3">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[118]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">%%</span><span class="n">R</span>
<span class="n">dep_corr</span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">glm</span><span class="p">(</span><span class="n">PHENOTYPE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">TMEM175_rs34311866_nalls</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">TMEM175_rs6856425</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SEX</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">AGE</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC1</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC2</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC4</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">PC5</span><span class="p">,</span><span class="w"> </span><span class="n">data</span><span class="o">=</span><span class="n">total</span><span class="p">)</span>
<span class="nf">summary</span><span class="p">(</span><span class="n">dep_corr</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>
Call:
glm(formula = PHENOTYPE ~ TMEM175_rs34311866_nalls + TMEM175_rs6856425 + 
    SEX + AGE + PC1 + PC2 + PC3 + PC4 + PC5, data = total)

Coefficients:
                          Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               2.410913   0.116470  20.700  &lt; 2e-16 ***
TMEM175_rs34311866_nalls -0.017839   0.041021  -0.435 0.663766    
TMEM175_rs6856425         0.073564   0.025318   2.906 0.003756 ** 
SEX                      -0.101059   0.025386  -3.981 7.42e-05 ***
AGE                       0.003901   0.001211   3.220 0.001327 ** 
PC1                      -0.017289   0.001316 -13.133  &lt; 2e-16 ***
PC2                       0.015487   0.004307   3.595 0.000342 ***
PC3                      -0.009166   0.004893  -1.873 0.061341 .  
PC4                       0.006983   0.006148   1.136 0.256380    
PC5                      -0.044488   0.005483  -8.113 1.63e-15 ***
---
Signif. codes:  0 *** 0.001 ** 0.01 * 0.05 . 0.1   1

(Dispersion parameter for gaussian family taken to be 0.1370694)

    Null deviance: 158.85  on 899  degrees of freedom
Residual deviance: 121.99  on 890  degrees of freedom
  (186 observations deleted due to missingness)
AIC: 777.49

Number of Fisher Scoring iterations: 2

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=7328c621-5cbc-4dda-951c-a8d7800d5403">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span> 
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=c2d418ff-f73f-43a2-8da8-822a8e79dcf5">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span> 
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=ec5c5126-f45a-4380-87d5-72ba1e5f9b40">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>GCTA conditional analysis</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=25c84835-2b0b-4a18-95b6-a7c02cfeb266">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[6]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Read in summary statistics</span>
<span class="n">sumstats</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/EUR.all_adj.csv'</span><span class="p">)</span>
<span class="n">sumstats</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[6]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Unnamed: 0</th>
<th>#CHROM</th>
<th>POS</th>
<th>ID</th>
<th>REF</th>
<th>ALT</th>
<th>PROVISIONAL_REF?</th>
<th>A1</th>
<th>OMITTED</th>
<th>A1_FREQ</th>
<th>FIRTH?</th>
<th>TEST</th>
<th>OBS_CT</th>
<th>OR</th>
<th>LOG(OR)_SE</th>
<th>L95</th>
<th>U95</th>
<th>Z_STAT</th>
<th>P</th>
<th>ERRCODE</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.164467</td>
<td>N</td>
<td>ADD</td>
<td>20837</td>
<td>1.034410</td>
<td>0.030398</td>
<td>0.974580</td>
<td>1.097910</td>
<td>1.112890</td>
<td>0.265757</td>
<td>.</td>
</tr>
<tr>
<th>1</th>
<td>8</td>
<td>4</td>
<td>882611</td>
<td>chr4:882611:T:C</td>
<td>T</td>
<td>C</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.051185</td>
<td>N</td>
<td>ADD</td>
<td>20885</td>
<td>0.931946</td>
<td>0.050102</td>
<td>0.844780</td>
<td>1.028110</td>
<td>-1.406740</td>
<td>0.159504</td>
<td>.</td>
</tr>
<tr>
<th>2</th>
<td>16</td>
<td>4</td>
<td>883018</td>
<td>chr4:883018:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.050599</td>
<td>N</td>
<td>ADD</td>
<td>20880</td>
<td>0.928749</td>
<td>0.050369</td>
<td>0.841441</td>
<td>1.025120</td>
<td>-1.467500</td>
<td>0.142240</td>
<td>.</td>
</tr>
<tr>
<th>3</th>
<td>24</td>
<td>4</td>
<td>883133</td>
<td>chr4:883133:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.012734</td>
<td>N</td>
<td>ADD</td>
<td>20811</td>
<td>0.809012</td>
<td>0.096557</td>
<td>0.669523</td>
<td>0.977562</td>
<td>-2.194990</td>
<td>0.028164</td>
<td>.</td>
</tr>
<tr>
<th>4</th>
<td>32</td>
<td>4</td>
<td>883483</td>
<td>chr4:883483:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.035357</td>
<td>N</td>
<td>ADD</td>
<td>20901</td>
<td>0.883241</td>
<td>0.058911</td>
<td>0.786926</td>
<td>0.991344</td>
<td>-2.107530</td>
<td>0.035071</td>
<td>.</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>392</th>
<td>3136</td>
<td>4</td>
<td>1007954</td>
<td>chr4:1007954:T:C</td>
<td>T</td>
<td>C</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.279743</td>
<td>N</td>
<td>ADD</td>
<td>18971</td>
<td>0.913461</td>
<td>0.025919</td>
<td>0.868217</td>
<td>0.961063</td>
<td>-3.492280</td>
<td>0.000479</td>
<td>.</td>
</tr>
<tr>
<th>393</th>
<td>3144</td>
<td>4</td>
<td>1008009</td>
<td>chr4:1008009:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.444961</td>
<td>N</td>
<td>ADD</td>
<td>18814</td>
<td>0.949197</td>
<td>0.023666</td>
<td>0.906174</td>
<td>0.994262</td>
<td>-2.203110</td>
<td>0.027587</td>
<td>.</td>
</tr>
<tr>
<th>394</th>
<td>3152</td>
<td>4</td>
<td>1008209</td>
<td>chr4:1008209:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.036468</td>
<td>N</td>
<td>ADD</td>
<td>20717</td>
<td>1.002060</td>
<td>0.059756</td>
<td>0.891315</td>
<td>1.126570</td>
<td>0.034499</td>
<td>0.972480</td>
<td>.</td>
</tr>
<tr>
<th>395</th>
<td>3160</td>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.279831</td>
<td>N</td>
<td>ADD</td>
<td>18965</td>
<td>0.913397</td>
<td>0.025912</td>
<td>0.868167</td>
<td>0.960983</td>
<td>-3.495880</td>
<td>0.000473</td>
<td>.</td>
</tr>
<tr>
<th>396</th>
<td>3168</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>Y</td>
<td>G</td>
<td>A</td>
<td>0.053420</td>
<td>N</td>
<td>ADD</td>
<td>20264</td>
<td>0.903891</td>
<td>0.049512</td>
<td>0.820297</td>
<td>0.996003</td>
<td>-2.040850</td>
<td>0.041266</td>
<td>.</td>
</tr>
</tbody>
</table>
<p>397 rows  20 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=f61dba53-18e0-41b3-9245-6b8769328c4a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Format summary statistics for GCTA-COJO</span>
<span class="c1">#First get the log odds ratio - this is required for COJO</span>
<span class="c1">#1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.</span>
<span class="n">sumstats_formatted</span> <span class="o">=</span> <span class="n">sumstats</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
<span class="n">sumstats_formatted</span><span class="p">[</span><span class="s1">'b'</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log</span><span class="p">(</span><span class="n">sumstats_formatted</span><span class="p">[</span><span class="s1">'OR'</span><span class="p">])</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=63194ead-379c-4203-8599-4d8a819f63f6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">sumstats_formatted</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[9]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>Unnamed: 0</th>
<th>#CHROM</th>
<th>POS</th>
<th>ID</th>
<th>REF</th>
<th>ALT</th>
<th>PROVISIONAL_REF?</th>
<th>A1</th>
<th>OMITTED</th>
<th>A1_FREQ</th>
<th>FIRTH?</th>
<th>TEST</th>
<th>OBS_CT</th>
<th>OR</th>
<th>LOG(OR)_SE</th>
<th>L95</th>
<th>U95</th>
<th>Z_STAT</th>
<th>P</th>
<th>ERRCODE</th>
<th>b</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>0</td>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.164467</td>
<td>N</td>
<td>ADD</td>
<td>20837</td>
<td>1.034410</td>
<td>0.030398</td>
<td>0.974580</td>
<td>1.097910</td>
<td>1.112890</td>
<td>0.265757</td>
<td>.</td>
<td>0.033831</td>
</tr>
<tr>
<th>1</th>
<td>8</td>
<td>4</td>
<td>882611</td>
<td>chr4:882611:T:C</td>
<td>T</td>
<td>C</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.051185</td>
<td>N</td>
<td>ADD</td>
<td>20885</td>
<td>0.931946</td>
<td>0.050102</td>
<td>0.844780</td>
<td>1.028110</td>
<td>-1.406740</td>
<td>0.159504</td>
<td>.</td>
<td>-0.070480</td>
</tr>
<tr>
<th>2</th>
<td>16</td>
<td>4</td>
<td>883018</td>
<td>chr4:883018:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.050599</td>
<td>N</td>
<td>ADD</td>
<td>20880</td>
<td>0.928749</td>
<td>0.050369</td>
<td>0.841441</td>
<td>1.025120</td>
<td>-1.467500</td>
<td>0.142240</td>
<td>.</td>
<td>-0.073917</td>
</tr>
<tr>
<th>3</th>
<td>24</td>
<td>4</td>
<td>883133</td>
<td>chr4:883133:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.012734</td>
<td>N</td>
<td>ADD</td>
<td>20811</td>
<td>0.809012</td>
<td>0.096557</td>
<td>0.669523</td>
<td>0.977562</td>
<td>-2.194990</td>
<td>0.028164</td>
<td>.</td>
<td>-0.211942</td>
</tr>
<tr>
<th>4</th>
<td>32</td>
<td>4</td>
<td>883483</td>
<td>chr4:883483:G:A</td>
<td>G</td>
<td>A</td>
<td>Y</td>
<td>A</td>
<td>G</td>
<td>0.035357</td>
<td>N</td>
<td>ADD</td>
<td>20901</td>
<td>0.883241</td>
<td>0.058911</td>
<td>0.786926</td>
<td>0.991344</td>
<td>-2.107530</td>
<td>0.035071</td>
<td>.</td>
<td>-0.124157</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>392</th>
<td>3136</td>
<td>4</td>
<td>1007954</td>
<td>chr4:1007954:T:C</td>
<td>T</td>
<td>C</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.279743</td>
<td>N</td>
<td>ADD</td>
<td>18971</td>
<td>0.913461</td>
<td>0.025919</td>
<td>0.868217</td>
<td>0.961063</td>
<td>-3.492280</td>
<td>0.000479</td>
<td>.</td>
<td>-0.090515</td>
</tr>
<tr>
<th>393</th>
<td>3144</td>
<td>4</td>
<td>1008009</td>
<td>chr4:1008009:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.444961</td>
<td>N</td>
<td>ADD</td>
<td>18814</td>
<td>0.949197</td>
<td>0.023666</td>
<td>0.906174</td>
<td>0.994262</td>
<td>-2.203110</td>
<td>0.027587</td>
<td>.</td>
<td>-0.052139</td>
</tr>
<tr>
<th>394</th>
<td>3152</td>
<td>4</td>
<td>1008209</td>
<td>chr4:1008209:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.036468</td>
<td>N</td>
<td>ADD</td>
<td>20717</td>
<td>1.002060</td>
<td>0.059756</td>
<td>0.891315</td>
<td>1.126570</td>
<td>0.034499</td>
<td>0.972480</td>
<td>.</td>
<td>0.002058</td>
</tr>
<tr>
<th>395</th>
<td>3160</td>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.279831</td>
<td>N</td>
<td>ADD</td>
<td>18965</td>
<td>0.913397</td>
<td>0.025912</td>
<td>0.868167</td>
<td>0.960983</td>
<td>-3.495880</td>
<td>0.000473</td>
<td>.</td>
<td>-0.090585</td>
</tr>
<tr>
<th>396</th>
<td>3168</td>
<td>4</td>
<td>1008480</td>
<td>chr4:1008480:A:G</td>
<td>A</td>
<td>G</td>
<td>Y</td>
<td>G</td>
<td>A</td>
<td>0.053420</td>
<td>N</td>
<td>ADD</td>
<td>20264</td>
<td>0.903891</td>
<td>0.049512</td>
<td>0.820297</td>
<td>0.996003</td>
<td>-2.040850</td>
<td>0.041266</td>
<td>.</td>
<td>-0.101047</td>
</tr>
</tbody>
</table>
<p>397 rows  21 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=854e5738-f644-4e64-8be4-2bdc9315ddd7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Now select just the necessary columns for COJO</span>
<span class="n">sumstats_export</span> <span class="o">=</span> <span class="n">sumstats_formatted</span><span class="p">[[</span><span class="s1">'ID'</span><span class="p">,</span> <span class="s1">'A1'</span><span class="p">,</span> <span class="s1">'OMITTED'</span><span class="p">,</span> <span class="s1">'A1_FREQ'</span><span class="p">,</span> <span class="s1">'b'</span><span class="p">,</span> <span class="s1">'LOG(OR)_SE'</span><span class="p">,</span> <span class="s1">'P'</span><span class="p">,</span> <span class="s1">'OBS_CT'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e375d6e1-fd3e-46b2-b27b-938d0a760502">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">sumstats_export</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[11]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>ID</th>
<th>A1</th>
<th>OMITTED</th>
<th>A1_FREQ</th>
<th>b</th>
<th>LOG(OR)_SE</th>
<th>P</th>
<th>OBS_CT</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>chr4:882530:C:T</td>
<td>T</td>
<td>C</td>
<td>0.164467</td>
<td>0.033831</td>
<td>0.030398</td>
<td>0.265757</td>
<td>20837</td>
</tr>
<tr>
<th>1</th>
<td>chr4:882611:T:C</td>
<td>C</td>
<td>T</td>
<td>0.051185</td>
<td>-0.070480</td>
<td>0.050102</td>
<td>0.159504</td>
<td>20885</td>
</tr>
<tr>
<th>2</th>
<td>chr4:883018:G:A</td>
<td>A</td>
<td>G</td>
<td>0.050599</td>
<td>-0.073917</td>
<td>0.050369</td>
<td>0.142240</td>
<td>20880</td>
</tr>
<tr>
<th>3</th>
<td>chr4:883133:G:A</td>
<td>A</td>
<td>G</td>
<td>0.012734</td>
<td>-0.211942</td>
<td>0.096557</td>
<td>0.028164</td>
<td>20811</td>
</tr>
<tr>
<th>4</th>
<td>chr4:883483:G:A</td>
<td>A</td>
<td>G</td>
<td>0.035357</td>
<td>-0.124157</td>
<td>0.058911</td>
<td>0.035071</td>
<td>20901</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>392</th>
<td>chr4:1007954:T:C</td>
<td>T</td>
<td>C</td>
<td>0.279743</td>
<td>-0.090515</td>
<td>0.025919</td>
<td>0.000479</td>
<td>18971</td>
</tr>
<tr>
<th>393</th>
<td>chr4:1008009:C:T</td>
<td>C</td>
<td>T</td>
<td>0.444961</td>
<td>-0.052139</td>
<td>0.023666</td>
<td>0.027587</td>
<td>18814</td>
</tr>
<tr>
<th>394</th>
<td>chr4:1008209:C:T</td>
<td>T</td>
<td>C</td>
<td>0.036468</td>
<td>0.002058</td>
<td>0.059756</td>
<td>0.972480</td>
<td>20717</td>
</tr>
<tr>
<th>395</th>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>0.279831</td>
<td>-0.090585</td>
<td>0.025912</td>
<td>0.000473</td>
<td>18965</td>
</tr>
<tr>
<th>396</th>
<td>chr4:1008480:A:G</td>
<td>G</td>
<td>A</td>
<td>0.053420</td>
<td>-0.101047</td>
<td>0.049512</td>
<td>0.041266</td>
<td>20264</td>
</tr>
</tbody>
</table>
<p>397 rows  8 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=03438e16-0c14-44e8-94f8-cc2fdcd8a693">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[12]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Rename columns following COJO format</span>
<span class="n">sumstats_export</span> <span class="o">=</span> <span class="n">sumstats_export</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'ID'</span><span class="p">:</span><span class="s1">'SNP'</span><span class="p">,</span> <span class="s1">'OMITTED'</span><span class="p">:</span><span class="s1">'A2'</span><span class="p">,</span> <span class="s1">'A1_FREQ'</span><span class="p">:</span><span class="s1">'freq'</span><span class="p">,</span> <span class="s1">'LOG(OR)_SE'</span><span class="p">:</span><span class="s1">'se'</span><span class="p">,</span> <span class="s1">'P'</span><span class="p">:</span><span class="s1">'p'</span><span class="p">,</span> <span class="s1">'OBS_CT'</span><span class="p">:</span><span class="s1">'N'</span><span class="p">})</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=46643f0a-4350-43ec-bb08-8fc162186075">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">sumstats_export</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[13]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>SNP</th>
<th>A1</th>
<th>A2</th>
<th>freq</th>
<th>b</th>
<th>se</th>
<th>p</th>
<th>N</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>chr4:882530:C:T</td>
<td>T</td>
<td>C</td>
<td>0.164467</td>
<td>0.033831</td>
<td>0.030398</td>
<td>0.265757</td>
<td>20837</td>
</tr>
<tr>
<th>1</th>
<td>chr4:882611:T:C</td>
<td>C</td>
<td>T</td>
<td>0.051185</td>
<td>-0.070480</td>
<td>0.050102</td>
<td>0.159504</td>
<td>20885</td>
</tr>
<tr>
<th>2</th>
<td>chr4:883018:G:A</td>
<td>A</td>
<td>G</td>
<td>0.050599</td>
<td>-0.073917</td>
<td>0.050369</td>
<td>0.142240</td>
<td>20880</td>
</tr>
<tr>
<th>3</th>
<td>chr4:883133:G:A</td>
<td>A</td>
<td>G</td>
<td>0.012734</td>
<td>-0.211942</td>
<td>0.096557</td>
<td>0.028164</td>
<td>20811</td>
</tr>
<tr>
<th>4</th>
<td>chr4:883483:G:A</td>
<td>A</td>
<td>G</td>
<td>0.035357</td>
<td>-0.124157</td>
<td>0.058911</td>
<td>0.035071</td>
<td>20901</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>392</th>
<td>chr4:1007954:T:C</td>
<td>T</td>
<td>C</td>
<td>0.279743</td>
<td>-0.090515</td>
<td>0.025919</td>
<td>0.000479</td>
<td>18971</td>
</tr>
<tr>
<th>393</th>
<td>chr4:1008009:C:T</td>
<td>C</td>
<td>T</td>
<td>0.444961</td>
<td>-0.052139</td>
<td>0.023666</td>
<td>0.027587</td>
<td>18814</td>
</tr>
<tr>
<th>394</th>
<td>chr4:1008209:C:T</td>
<td>T</td>
<td>C</td>
<td>0.036468</td>
<td>0.002058</td>
<td>0.059756</td>
<td>0.972480</td>
<td>20717</td>
</tr>
<tr>
<th>395</th>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>0.279831</td>
<td>-0.090585</td>
<td>0.025912</td>
<td>0.000473</td>
<td>18965</td>
</tr>
<tr>
<th>396</th>
<td>chr4:1008480:A:G</td>
<td>G</td>
<td>A</td>
<td>0.053420</td>
<td>-0.101047</td>
<td>0.049512</td>
<td>0.041266</td>
<td>20264</td>
</tr>
</tbody>
</table>
<p>397 rows  8 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=5661a3d3-fc9d-454a-8966-28413c9e8327">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[14]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Export</span>
<span class="n">sumstats_export</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_EUR/EUR.all_adj.sumstats.ma'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=00c788a1-627d-4944-ab1c-e33a4c3eae30">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[15]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>wget<span class="w"> </span>https://yanglab.westlake.edu.cn/software/gcta/bin/gcta-1.94.1-linux-kernel-3-x86_64.zip
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>--2024-08-07 11:14:11--  https://yanglab.westlake.edu.cn/software/gcta/bin/gcta-1.94.1-linux-kernel-3-x86_64.zip
Resolving yanglab.westlake.edu.cn (yanglab.westlake.edu.cn)... 124.160.108.195, 42.247.30.189, 2001:250:6413:1002:250:56ff:10:195
Connecting to yanglab.westlake.edu.cn (yanglab.westlake.edu.cn)|124.160.108.195|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14333133 (14M) [application/zip]
Saving to: gcta-1.94.1-linux-kernel-3-x86_64.zip

gcta-1.94.1-linux-k 100%[===================&gt;]  13.67M   844KB/s    in 17s     

2024-08-07 11:14:32 (824 KB/s) - gcta-1.94.1-linux-kernel-3-x86_64.zip saved [14333133/14333133]

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=043b2e49-cc83-46ba-81a8-056813ce5eda">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[16]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>unzip<span class="w"> </span>gcta-1.94.1-linux-kernel-3-x86_64.zip
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Archive:  gcta-1.94.1-linux-kernel-3-x86_64.zip
   creating: gcta-1.94.1-linux-kernel-3-x86_64/
  inflating: __MACOSX/._gcta-1.94.1-linux-kernel-3-x86_64  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/gcta64  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._gcta64  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/.DS_Store  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._.DS_Store  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.bim  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.bim  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/MIT_License.txt  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._MIT_License.txt  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.fam  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.fam  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/README.txt  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._README.txt  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.bed  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.bed  
  inflating: gcta-1.94.1-linux-kernel-3-x86_64/test.phen  
  inflating: __MACOSX/gcta-1.94.1-linux-kernel-3-x86_64/._test.phen  
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=cbac1270-5ff7-4db2-9bf2-ad7e397b271d">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[17]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>mv<span class="w"> </span>~/workspace/ws_files/gcta-1.94.1-linux-kernel-3-x86_64<span class="w"> </span>/home/jupyter/tools/
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=db0e57b4-9d75-48c7-b2f5-a73cad87dc68">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[18]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>ls<span class="w"> </span>/home/jupyter/tools/
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>LICENSE				   plink			    prettify
annovar				   plink2			    rvtests
annovar.latest.tar.gz		   plink2_linux_x86_64_latest.zip   toy.map
gcta-1.94.1-linux-kernel-3-x86_64  plink_linux_x86_64_20190304.zip  toy.ped
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=4246a7e0-b0f5-43a7-b95f-3a9b08a9fef8">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[20]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64<span class="w"> </span>--version
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>/bin/bash: line 1: /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64: Permission denied
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=22598760-e0a7-4e3b-bcfe-64fad5c2d1f0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[21]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>chmod<span class="w"> </span>u+x<span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1110e4f9-7bab-4bd2-9441-4a94a15ccb62">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[24]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64<span class="w"> </span>--version
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 11:22:48 UTC on Wed Aug 07 2024.
Hostname: jupyterlabvertexai20240708

<span class="ansi-red-fg">Error: </span>the --out option is missing.
An error occurs, please check the options or data
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=390d7f06-93d0-41e6-b6a4-b35b0d674abb">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[25]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Select multiple associated SNPs based on significance p-value 5e-08</span>
<span class="c1">#Can change the p-value for significance if needed</span>
<span class="c1">#bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS</span>
<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64<span class="w"> </span>--bfile<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">file</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">sumstats</span><span class="o">.</span><span class="n">ma</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">p</span> <span class="mf">5e-8</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">slct</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">COJO</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 11:25:36 UTC on Wed Aug 07 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma
--cojo-p 5e-08
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].
38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].
9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...
Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 397 SNPs (9107 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma] ...
GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
397 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Finally, 1 associated SNPs are selected.
Performing joint analysis on all the 1 selected signals ...
Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO.jma.cojo] ...
Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO.ldr.cojo] ...
Saving the conditional analysis results of 396 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.COJO.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 11:25:42 UTC on Wed Aug 07 2024
Overall computational time: 6.04 sec.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=84db68f5-0f9e-476c-8880-ac714ef0fb79">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[28]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR.all_adj.COJO.cma.cojo
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Chr	SNP	bp	refA	freq	b	se	p	n	freq_geno	bC	bC_se	pC
4	chr4:882530:C:T	882530	T	0.164467	0.0338312	0.0303975	0.265725	20489.9	0.164537	-0.0279995	0.0303977	0.356994
4	chr4:882611:T:C	882611	C	0.0511851	-0.0704804	0.0501022	0.159507	21340.5	0.0504798	-0.0330724	0.0501033	0.5092
4	chr4:883018:G:A	883018	A	0.0505987	-0.0739168	0.0503694	0.142242	21346	0.0499974	-0.0364219	0.0503708	0.469634
4	chr4:883133:G:A	883133	A	0.0127337	-0.211942	0.0965569	0.0281642	22193.8	0.0132536	-0.170603	0.0965652	0.0772761
4	chr4:883483:G:A	883483	A	0.0353572	-0.124157	0.0589113	0.0350719	21976.2	0.0366916	-0.0786346	0.0589159	0.181977
4	chr4:883624:C:T	883624	T	0.181816	0.0947191	0.0295052	0.00132619	20080.8	0.179007	-0.0154726	0.029512	0.600084
4	chr4:883837:G:A	883837	A	0.400635	-0.031951	0.022935	0.163586	20597	0.404776	-0.028599	0.0229355	0.212423
4	chr4:883921:T:C	883921	C	0.448883	-0.0438296	0.02274	0.0539267	20335.1	0.452722	-0.0401386	0.0227415	0.0775649
4	chr4:884131:C:T	884131	T	0.0117174	0.0247414	0.104187	0.812291	20698.7	0.0122093	0.0706551	0.104185	0.497663
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=cb6ea87d-dbd9-4e25-8091-c2dca0a9cf91">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[29]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR.all_adj.COJO.jma.cojo
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Chr	SNP	bp	refA	freq	b	se	p	n	freq_geno	bJ	bJ_se	pJ	LD_r
4	chr4:958159:T:C	958159	C	0.203811	0.258912	0.0290424	4.87943e-19	18930.3	0.198229	0.258912	0.0291025	5.76097e-19	0
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8ed8ea88-eabb-461f-89b5-57e1cc4fa4f8">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[30]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR.all_adj.COJO.ldr.cojo
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>SNP	chr4:958159:T:C	
chr4:958159:T:C	1	
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=144e2490-d240-492d-9d1a-f3824696e2f1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[31]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR.all_adj.COJO.log
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
Analysis started at 11:25:36 UTC on Wed Aug 07 2024.
Hostname: jupyterlabvertexai20240708

</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=259ef615-8e97-4538-accc-04b835dfbba8">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[5]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Select multiple associated SNPs based on significance p-value 5e-08</span>
<span class="c1">#Can change the p-value for significance if needed</span>
<span class="c1">#bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS</span>
<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64<span class="w"> </span>--bfile<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">file</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">sumstats</span><span class="o">.</span><span class="n">ma</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">p</span> <span class="mf">4.59e-4</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">slct</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">LDprune</span><span class="o">.</span><span class="n">COJO</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 12:48:26 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].
38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].
9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...
Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 397 SNPs (9107 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma] ...
GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
397 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Finally, 1 associated SNPs are selected.
Performing joint analysis on all the 1 selected signals ...
Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO.jma.cojo] ...
Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO.ldr.cojo] ...
Saving the conditional analysis results of 396 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.LDprune.COJO.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 12:48:32 UTC on Tue Aug 20 2024
Overall computational time: 6.05 sec.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=e2bdfb2f-7e69-4222-b616-98e23aad57d1">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[32]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Or you can run COJO and select just the top 10 independent SNPs (can be below GWAS significance p-value)</span>
<span class="c1">#Select top 10 SNPs</span>
<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64<span class="w"> </span>--bfile<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR_TMEM175<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">file</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">sumstats</span><span class="o">.</span><span class="n">ma</span> \
<span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">top</span><span class="o">-</span><span class="n">SNPs</span> <span class="mi">10</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">TMEM175_EUR</span><span class="o">/</span><span class="n">EUR</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">top10</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 11:33:10 UTC on Wed Aug 07 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].
38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].
9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...
Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 397 SNPs (9107 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma] ...
GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
397 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 387 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175//TMEM175_EUR/EUR.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 11:33:17 UTC on Wed Aug 07 2024
Overall computational time: 7.85 sec.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=90f9be7c-83a3-4993-a219-014162e0db6e">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[33]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR.all_adj.top10.cma.cojo
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Chr	SNP	bp	refA	freq	b	se	p	n	freq_geno	bC	bC_se	pC
4	chr4:882530:C:T	882530	T	0.164467	0.0338312	0.0303975	0.265725	20489.9	0.164537	-0.277937	0.105335	0.00832474
4	chr4:882611:T:C	882611	C	0.0511851	-0.0704804	0.0501022	0.159507	21340.5	0.0504798	-0.122516	0.0814926	0.132737
4	chr4:883018:G:A	883018	A	0.0505987	-0.0739168	0.0503694	0.142242	21346	0.0499974	-0.114817	0.0418044	0.00602305
4	chr4:883483:G:A	883483	A	0.0353572	-0.124157	0.0589113	0.0350719	21976.2	0.0366916	0.177534	0.0999408	0.075669
4	chr4:883624:C:T	883624	T	0.181816	0.0947191	0.0295052	0.00132619	20080.8	0.179007	-0.0724453	0.0497149	0.145057
4	chr4:883837:G:A	883837	A	0.400635	-0.031951	0.022935	0.163586	20597	0.404776	0.282062	0.0325328	4.31799e-18
4	chr4:883921:T:C	883921	C	0.448883	-0.0438296	0.02274	0.0539267	20335.1	0.452722	0.110611	0.0509097	0.0298032
4	chr4:884131:C:T	884131	T	0.0117174	0.0247414	0.104187	0.812291	20698.7	0.0122093	-0.136355	0.0806028	0.0907057
4	chr4:884260:C:T	884260	T	0.335987	0.0328447	0.024008	0.17129	20231.7	0.337507	0.151407	0.0726496	0.0371541
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5e490c7c-cd01-488d-8ee5-11c1967c2b8a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[34]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="o">!</span><span class="w"> </span>head<span class="w"> </span><span class="o">{</span>WORK_DIR<span class="o">}</span>/TMEM175_EUR/EUR.all_adj.top10.jma.cojo
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>Chr	SNP	bp	refA	freq	b	se	p	n	freq_geno	bJ	bJ_se	pJ	LD_r
4	chr4:883133:G:A	883133	A	0.0127337	-0.211942	0.0965569	0.0281642	22193.8	0.0132536	-0.277937	0.105335	0.00832474	-0.0174097
4	chr4:905058:T:C	905058	C	0.0220098	0.103396	0.0776335	0.182912	20053.8	0.0211291	-0.122516	0.0814926	0.132737	-0.0460043
4	chr4:931588:C:T	931588	T	0.098972	-0.14224	0.0364747	9.63155e-05	21915.3	0.0994334	-0.114817	0.0418044	0.00602305	0.372451
4	chr4:933370:G:A	933370	A	0.0153177	0.0255118	0.0925403	0.782792	20143.3	0.0149901	0.177534	0.0999408	0.075669	-0.0298242
4	chr4:939809:A:G	939809	G	0.0536428	-0.111124	0.048938	0.0231648	21395.3	0.0540681	-0.0724453	0.0497149	0.145057	-0.117727
4	chr4:958159:T:C	958159	C	0.203811	0.258912	0.0290424	4.87943e-19	18930.3	0.198229	0.282062	0.0325328	4.31799e-18	-0.121267
4	chr4:983473:C:T	983473	T	0.0623297	0.02634	0.0460307	0.567168	21010.5	0.063255	0.110611	0.0509097	0.0298032	-0.0387902
4	chr4:988557:C:T	988557	T	0.022547	0.0935541	0.076603	0.221978	20117.6	0.0216671	-0.136355	0.0806028	0.0907057	-0.0231418
4	chr4:1001112:T:G	1001112	G	0.0262346	0.119018	0.0721329	0.0989471	19571.7	0.0264005	0.151407	0.0726496	0.0371541	-0.0354378
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=ada1374a-d186-40a4-9e8b-24df94291a8f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>use p threshould 4.59e-4</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=60408d55-c614-4be0-8891-63902efe79ee">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[12]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">ancestries</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'AAC'</span><span class="p">,</span><span class="s1">'AFR'</span><span class="p">,</span><span class="s1">'AJ'</span><span class="p">,</span><span class="s1">'AMR'</span><span class="p">,</span><span class="s1">'CAS'</span><span class="p">,</span><span class="s1">'EAS'</span><span class="p">,</span><span class="s1">'EUR'</span><span class="p">,</span><span class="s1">'FIN'</span><span class="p">,</span><span class="s1">'MDE'</span><span class="p">,</span><span class="s1">'SAS'</span><span class="p">,</span><span class="s1">'CAH'</span><span class="p">}</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    
    <span class="c1">#Select multiple associated SNPs based on LD pruning p-value 0.00238</span>
    <span class="c1">#Can change the p-value for significance if needed</span>
    <span class="c1">#bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS</span>
    <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64<span class="w"> </span><span class="err">\</span>
    <span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span> \
    <span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
    <span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">file</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">sumstats</span><span class="o">.</span><span class="n">ma</span> \
    <span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">p</span> <span class="mf">4.59e-4</span> \
    <span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">slct</span> \
    <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">ldprune</span><span class="o">.</span><span class="n">COJO</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:01 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].
646 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].
1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed] in SNP-major format ...
Genotype data for 646 individuals and 1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 405 SNPs (1582 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma] ...
GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 405 SNPs: 6.34329 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
405 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:02 UTC on Tue Aug 20 2024
Overall computational time: 0.29 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:02 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].
903 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].
1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed] in SNP-major format ...
Genotype data for 903 individuals and 1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 429 SNPs (1526 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma] ...
GWAS summary statistics of 429 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 429 SNPs: 4.98286 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
429 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 429 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:02 UTC on Tue Aug 20 2024
Overall computational time: 0.37 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:02 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].
581 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].
1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed] in SNP-major format ...
Genotype data for 581 individuals and 1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 398 SNPs (1547 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma] ...
GWAS summary statistics of 398 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 398 SNPs: 5.25039 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
398 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 398 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:03 UTC on Tue Aug 20 2024
Overall computational time: 0.30 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:03 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].
2655 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].
1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed] in SNP-major format ...
Genotype data for 2655 individuals and 1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 416 SNPs (869 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma] ...
GWAS summary statistics of 416 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 416 SNPs: 7.98751 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
416 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 416 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Finally, 2 associated SNPs are selected.
Performing joint analysis on all the 2 selected signals ...
Saving the 2 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.jma.cojo] ...
Saving the LD structure of 2 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.ldr.cojo] ...
Saving the conditional analysis results of 414 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:32:04 UTC on Tue Aug 20 2024
Overall computational time: 0.74 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:04 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].
2643 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].
3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed] in SNP-major format ...
Genotype data for 2643 individuals and 3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 591 SNPs (3262 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma] ...
GWAS summary statistics of 591 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 591 SNPs: 7.37522 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
591 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 591 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:04 UTC on Tue Aug 20 2024
Overall computational time: 0.53 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:05 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].
5167 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].
3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed] in SNP-major format ...
Genotype data for 5167 individuals and 3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 377 SNPs (2794 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma] ...
GWAS summary statistics of 377 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 377 SNPs: 5.163 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
377 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 377 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:05 UTC on Tue Aug 20 2024
Overall computational time: 0.54 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:05 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].
114 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].
605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed] in SNP-major format ...
Genotype data for 114 individuals and 605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 405 SNPs (200 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma] ...
GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 405 SNPs: 7.93715 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
405 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:06 UTC on Tue Aug 20 2024
Overall computational time: 0.23 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:06 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].
1111 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].
3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed] in SNP-major format ...
Genotype data for 1111 individuals and 3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 588 SNPs (2861 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma] ...
GWAS summary statistics of 588 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 588 SNPs: 7.51272 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
588 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 588 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:06 UTC on Tue Aug 20 2024
Overall computational time: 0.18 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:06 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].
851 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].
3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed] in SNP-major format ...
Genotype data for 851 individuals and 3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 547 SNPs (2543 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma] ...
GWAS summary statistics of 548 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 548 SNPs: 5.36028 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
547 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 547 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:06 UTC on Tue Aug 20 2024
Overall computational time: 0.32 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:07 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].
635 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].
1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed] in SNP-major format ...
Genotype data for 635 individuals and 1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 477 SNPs (1251 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma] ...
GWAS summary statistics of 477 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 477 SNPs: 5.50087 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
477 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 477 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
No SNPs have been selected.

<span class="ansi-green-fg">Analysis finished </span>at 13:32:07 UTC on Tue Aug 20 2024
Overall computational time: 0.27 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:32:07 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma
--cojo-p 0.000459
--cojo-slct
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].
38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].
9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...
Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 397 SNPs (9107 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma] ...
GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
397 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...

Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.000459; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Finally, 1 associated SNPs are selected.
Performing joint analysis on all the 1 selected signals ...
Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.jma.cojo] ...
Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.ldr.cojo] ...
Saving the conditional analysis results of 396 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:32:13 UTC on Tue Aug 20 2024
Overall computational time: 6.33 sec.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=1d8d8d4c-42e6-4a3e-b8b4-7c20800c094f">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=5a2ae318-13ad-4017-84a9-52b0341a16c6">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># prepare file for COJO</span>
<span class="n">ancestries</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'AAC'</span><span class="p">,</span><span class="s1">'AFR'</span><span class="p">,</span><span class="s1">'AJ'</span><span class="p">,</span><span class="s1">'AMR'</span><span class="p">,</span><span class="s1">'CAS'</span><span class="p">,</span><span class="s1">'EAS'</span><span class="p">,</span><span class="s1">'EUR'</span><span class="p">,</span><span class="s1">'FIN'</span><span class="p">,</span><span class="s1">'MDE'</span><span class="p">,</span><span class="s1">'SAS'</span><span class="p">,</span><span class="s1">'CAH'</span><span class="p">}</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    <span class="c1">#Read in summary statistics</span>
    <span class="n">sumstats</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">.all_adj.csv'</span><span class="p">)</span>
    
    <span class="c1">#Format summary statistics for GCTA-COJO</span>
    <span class="c1">#First get the log odds ratio - this is required for COJO</span>
    <span class="c1">#1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.</span>
    <span class="n">sumstats_formatted</span> <span class="o">=</span> <span class="n">sumstats</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    <span class="n">sumstats_formatted</span><span class="p">[</span><span class="s1">'b'</span><span class="p">]</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">log</span><span class="p">(</span><span class="n">sumstats_formatted</span><span class="p">[</span><span class="s1">'OR'</span><span class="p">])</span>
    
    <span class="c1">#Now select just the necessary columns for COJO</span>
    <span class="n">sumstats_export</span> <span class="o">=</span> <span class="n">sumstats_formatted</span><span class="p">[[</span><span class="s1">'ID'</span><span class="p">,</span> <span class="s1">'A1'</span><span class="p">,</span> <span class="s1">'OMITTED'</span><span class="p">,</span> <span class="s1">'A1_FREQ'</span><span class="p">,</span> <span class="s1">'b'</span><span class="p">,</span> <span class="s1">'LOG(OR)_SE'</span><span class="p">,</span> <span class="s1">'P'</span><span class="p">,</span> <span class="s1">'OBS_CT'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    
    <span class="c1">#Rename columns following COJO format</span>
    <span class="n">sumstats_export</span> <span class="o">=</span> <span class="n">sumstats_export</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'ID'</span><span class="p">:</span><span class="s1">'SNP'</span><span class="p">,</span> <span class="s1">'OMITTED'</span><span class="p">:</span><span class="s1">'A2'</span><span class="p">,</span> <span class="s1">'A1_FREQ'</span><span class="p">:</span><span class="s1">'freq'</span><span class="p">,</span> <span class="s1">'LOG(OR)_SE'</span><span class="p">:</span><span class="s1">'se'</span><span class="p">,</span> <span class="s1">'P'</span><span class="p">:</span><span class="s1">'p'</span><span class="p">,</span> <span class="s1">'OBS_CT'</span><span class="p">:</span><span class="s1">'N'</span><span class="p">})</span>
    
    <span class="c1">#Export</span>
    <span class="n">sumstats_export</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">.all_adj.sumstats.ma'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=db3be3bf-a31c-4e49-b358-4f68e0440015">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">ancestries</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'AAC'</span><span class="p">,</span><span class="s1">'AFR'</span><span class="p">,</span><span class="s1">'AJ'</span><span class="p">,</span><span class="s1">'AMR'</span><span class="p">,</span><span class="s1">'CAS'</span><span class="p">,</span><span class="s1">'EAS'</span><span class="p">,</span><span class="s1">'EUR'</span><span class="p">,</span><span class="s1">'FIN'</span><span class="p">,</span><span class="s1">'MDE'</span><span class="p">,</span><span class="s1">'SAS'</span><span class="p">,</span><span class="s1">'CAH'</span><span class="p">}</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span>
    
    <span class="c1">#Select multiple associated SNPs based on significance p-value 5e-08</span>
    <span class="c1">#Can change the p-value for significance if needed</span>
    <span class="c1">#bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS</span>
    <span class="o">!</span><span class="w"> </span>/home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64<span class="w"> </span><span class="err">\</span>
    <span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="n">_TMEM175</span> \
    <span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
    <span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">file</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">sumstats</span><span class="o">.</span><span class="n">ma</span> \
    <span class="o">--</span><span class="n">cojo</span><span class="o">-</span><span class="n">top</span><span class="o">-</span><span class="n">SNPs</span> <span class="mi">10</span> \
    <span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="p">{</span><span class="n">ancestry</span><span class="p">}</span><span class="o">.</span><span class="n">all_adj</span><span class="o">.</span><span class="n">top10</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:39 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].
646 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].
1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed] in SNP-major format ...
Genotype data for 646 individuals and 1987 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 405 SNPs (1582 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma] ...
GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 405 SNPs: 6.34329 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
405 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 395 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:40 UTC on Tue Aug 20 2024
Overall computational time: 0.84 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:40 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].
903 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].
1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed] in SNP-major format ...
Genotype data for 903 individuals and 1955 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 429 SNPs (1526 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma] ...
GWAS summary statistics of 429 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 429 SNPs: 4.98286 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
429 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 429 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 419 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:41 UTC on Tue Aug 20 2024
Overall computational time: 0.92 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:41 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].
581 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].
1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed] in SNP-major format ...
Genotype data for 581 individuals and 1945 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 398 SNPs (1547 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma] ...
GWAS summary statistics of 398 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 398 SNPs: 5.25039 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
398 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 398 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 388 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:42 UTC on Tue Aug 20 2024
Overall computational time: 0.96 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:43 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].
2655 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].
1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed] in SNP-major format ...
Genotype data for 2655 individuals and 1285 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 416 SNPs (869 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma] ...
GWAS summary statistics of 416 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 416 SNPs: 7.98751 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
416 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 416 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 406 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:44 UTC on Tue Aug 20 2024
Overall computational time: 1.06 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:44 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].
2643 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].
3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed] in SNP-major format ...
Genotype data for 2643 individuals and 3853 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 591 SNPs (3262 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma] ...
GWAS summary statistics of 591 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 591 SNPs: 7.37522 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
591 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 591 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 581 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:45 UTC on Tue Aug 20 2024
Overall computational time: 1.20 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:45 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].
5167 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].
3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed] in SNP-major format ...
Genotype data for 5167 individuals and 3171 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 377 SNPs (2794 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma] ...
GWAS summary statistics of 377 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 377 SNPs: 5.163 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
377 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 377 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 367 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:47 UTC on Tue Aug 20 2024
Overall computational time: 1.43 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:47 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].
114 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].
605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed] in SNP-major format ...
Genotype data for 114 individuals and 605 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 405 SNPs (200 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma] ...
GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 405 SNPs: 7.93715 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
405 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 395 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:48 UTC on Tue Aug 20 2024
Overall computational time: 0.74 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:48 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].
1111 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].
3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed] in SNP-major format ...
Genotype data for 1111 individuals and 3449 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 588 SNPs (2861 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma] ...
GWAS summary statistics of 588 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 588 SNPs: 7.51272 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
588 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 588 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 578 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:49 UTC on Tue Aug 20 2024
Overall computational time: 0.93 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:49 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].
851 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].
3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed] in SNP-major format ...
Genotype data for 851 individuals and 3090 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 547 SNPs (2543 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma] ...
GWAS summary statistics of 548 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 548 SNPs: 5.36028 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
547 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 547 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 537 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:50 UTC on Tue Aug 20 2024
Overall computational time: 0.86 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:50 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].
635 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].
1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed] in SNP-major format ...
Genotype data for 635 individuals and 1728 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 477 SNPs (1251 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma] ...
GWAS summary statistics of 477 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 477 SNPs: 5.50087 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
477 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 477 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 467 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:04:51 UTC on Tue Aug 20 2024
Overall computational time: 0.68 sec.
*******************************************************************
* Genome-wide Complex Trait Analysis (GCTA)
* version v1.94.1 Linux
* Built at Nov 15 2022 21:14:25, by GCC 8.5
* (C) 2010-present, Yang Lab, Westlake University
* Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;
*******************************************************************
<span class="ansi-green-fg">Analysis started </span>at 13:04:51 UTC on Tue Aug 20 2024.
Hostname: jupyterlabvertexai20240708

Accepted options:
--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175
--maf 0.01
--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma
--cojo-top-SNPs 10
--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10


Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].
38839 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].
Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].
9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].
Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...
Genotype data for 38839 individuals and 9504 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed].
Calculating allele frequencies ...
Filtering SNPs with MAF &gt; 0.01 ...
After filtering SNPs with MAF &gt; 0.01, there are 397 SNPs (9107 SNPs with MAF &lt; 0.01).

Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma] ...
GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma].
Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.20346 (variance of logit for case-control studies).
Matching the GWAS meta-analysis results to the genotype data ...
397 SNPs are matched to the genotype data.
Calculating the variance of SNP genotypes ...
The threshold p-value has been set to 0.5 because of the --cojo-top-SNPs option.

Performing stepwise model selection on 397 SNPs to select association signals ... (p cutoff = 0.5; collinearity cutoff = 0.9)
(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)
Switched to perform a forward model selection because the significance level is too low...
5 associated SNPs have been selected.
10 associated SNPs have been selected.
Performing backward elimination...
Finally, 10 associated SNPs are selected.
Performing joint analysis on all the 10 selected signals ...
Saving the 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10.jma.cojo] ...
Saving the LD structure of 10 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10.ldr.cojo] ...
Saving the conditional analysis results of 387 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.top10.cma.cojo] ...
(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)

<span class="ansi-green-fg">Analysis finished </span>at 13:05:00 UTC on Tue Aug 20 2024
Overall computational time: 8.86 sec.
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=0b081052-1e4b-4a64-8193-9ad339846750">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>change format of association file</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=5dc5cebc-5b19-44d6-9e10-12a2e57110dd">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[13]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">ancestries</span> <span class="o">=</span> <span class="p">{</span><span class="s1">'AAC'</span><span class="p">,</span><span class="s1">'AFR'</span><span class="p">,</span><span class="s1">'AJ'</span><span class="p">,</span><span class="s1">'AMR'</span><span class="p">,</span><span class="s1">'CAS'</span><span class="p">,</span><span class="s1">'EAS'</span><span class="p">,</span><span class="s1">'EUR'</span><span class="p">,</span><span class="s1">'FIN'</span><span class="p">,</span><span class="s1">'MDE'</span><span class="p">,</span><span class="s1">'SAS'</span><span class="p">,</span><span class="s1">'CAH'</span><span class="p">}</span>

<span class="k">for</span> <span class="n">ancestry</span> <span class="ow">in</span> <span class="n">ancestries</span><span class="p">:</span>
    <span class="n">WORK_DIR</span> <span class="o">=</span> <span class="s2">"~/workspace/ws_files/TMEM175/"</span>
    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">'WORKING ON: </span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">'</span><span class="p">)</span>
    
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">.all_adj.csv'</span><span class="p">)</span>
    
    <span class="c1">#First just select relevant columns: chromosome, bp position and p-value</span>
    <span class="n">export_ldassoc</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s1">'#CHROM'</span><span class="p">,</span> <span class="s1">'POS'</span><span class="p">,</span> <span class="s1">'P'</span><span class="p">]]</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>
    
    <span class="c1">#Rename the #CHROM column to remove the hashtag as I think this might be confusing LDassoc</span>
    <span class="n">export_ldassoc</span> <span class="o">=</span> <span class="n">export_ldassoc</span><span class="o">.</span><span class="n">rename</span><span class="p">(</span><span class="n">columns</span><span class="o">=</span><span class="p">{</span><span class="s1">'#CHROM'</span><span class="p">:</span> <span class="s1">'CHROM'</span><span class="p">})</span> 
    
    <span class="c1">#Then export as a tab-separated, not comma-separated file</span>

    <span class="n">export_ldassoc</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/TMEM175_</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">ancestry</span><span class="si">}</span><span class="s1">.all_adj.formatted.tab'</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">'</span><span class="se">\t</span><span class="s1">'</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>WORKING ON: AMR
WORKING ON: CAS
WORKING ON: MDE
WORKING ON: AJ
WORKING ON: AFR
WORKING ON: EAS
WORKING ON: FIN
WORKING ON: AAC
WORKING ON: CAH
WORKING ON: SAS
WORKING ON: EUR
</pre>
</div>
</div>
</div>
</div>
</div>
<div class="jp-Cell jp-MarkdownCell jp-Notebook-cell" id="cell-id=bd3f5bd2-61b7-4506-a86b-660b89c43cf7">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea"><div class="jp-InputPrompt jp-InputArea-prompt">
</div><div class="jp-RenderedHTMLCommon jp-RenderedMarkdown jp-MarkdownOutput" data-mime-type="text/markdown">
<p>validate GLM output file in EAS ancestry again</p>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=27d15018-1c04-43ec-9d14-4dd87f789bdc">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[4]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_EAS'</span>

<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">pfile</span> <span class="p">{</span><span class="n">REL7_PATH</span><span class="p">}</span><span class="o">/</span><span class="n">imputed_genotypes</span><span class="o">/</span><span class="n">EAS</span><span class="o">/</span><span class="n">chr4_EAS_release7_vwb</span> \
<span class="o">--</span><span class="nb">chr</span> <span class="mi">4</span> \
<span class="o">--</span><span class="n">from</span><span class="o">-</span><span class="n">bp</span> <span class="mi">882387</span> \
<span class="o">--</span><span class="n">to</span><span class="o">-</span><span class="n">bp</span> <span class="mi">1008656</span> \
<span class="o">--</span><span class="n">make</span><span class="o">-</span><span class="n">bed</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS_TMEM175</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.log.
Options in effect:
  --chr 4
  --from-bp 882387
  --make-bed
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb
  --to-bp 1008656

Start time: Thu Aug 29 12:41:04 2024
52223 MiB RAM detected, ~50548 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
5167 samples (1889 females, 3278 males; 5167 founders) loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.psam.
3235052 variants loaded from
/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr4_EAS_release7_vwb.pvar.
1 binary phenotype loaded (2662 cases, 2461 controls).
3171 variants remaining after main filters.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim
... done.
Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed
... done.
End time: Thu Aug 29 12:41:07 2024
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=3d202a6b-4743-4181-aafa-ae44334e73d0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[5]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># ASSOC analysis</span>

<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_EAS'</span>
    
<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS_TMEM175</span> \
<span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS</span><span class="o">.</span><span class="n">samplestoKeep</span> \
<span class="o">--</span><span class="n">assoc</span> \
<span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">no</span><span class="o">-</span><span class="n">sex</span> \
<span class="o">--</span><span class="n">ci</span> <span class="mf">0.95</span> \
<span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS_TMEM175</span><span class="o">.</span><span class="n">all</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.log.
Options in effect:
  --allow-no-sex
  --assoc
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
  --ci 0.95
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep
  --maf 0.01
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all

52223 MB RAM detected; reserving 26111 MB for main workspace.
3171 variants loaded from .bim file.
5167 people (3278 males, 1889 females) loaded from .fam.
5123 phenotype values loaded from .fam.
--keep: 5167 people remaining.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 5167 founders and 0 nonfounders present.
Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.
Total genotyping rate is 0.994312.
2794 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
377 variants and 5167 people pass filters and QC.
Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44
phenotypes are missing.)
Writing C/C --assoc report to
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.assoc ...
done.
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=25ef9e7c-cf2c-4736-b9fa-39c9e9bcd612">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[6]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Look at assoc results</span>
<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_EAS'</span>
<span class="n">freq</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/EAS_TMEM175.all.assoc'</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">'\s+'</span><span class="p">)</span>
<span class="n">freq</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[6]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>CHR</th>
<th>SNP</th>
<th>BP</th>
<th>A1</th>
<th>F_A</th>
<th>F_U</th>
<th>A2</th>
<th>CHISQ</th>
<th>P</th>
<th>OR</th>
<th>SE</th>
<th>L95</th>
<th>U95</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>chr4:882530:C:T</td>
<td>882530</td>
<td>T</td>
<td>0.29880</td>
<td>0.28300</td>
<td>C</td>
<td>3.0680</td>
<td>0.07984</td>
<td>1.0800</td>
<td>0.04382</td>
<td>0.9909</td>
<td>1.1770</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>chr4:882611:T:C</td>
<td>882611</td>
<td>C</td>
<td>0.06649</td>
<td>0.07835</td>
<td>T</td>
<td>5.3640</td>
<td>0.02056</td>
<td>0.8379</td>
<td>0.07645</td>
<td>0.7213</td>
<td>0.9733</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>chr4:883018:G:A</td>
<td>883018</td>
<td>A</td>
<td>0.06649</td>
<td>0.07814</td>
<td>G</td>
<td>5.1880</td>
<td>0.02275</td>
<td>0.8403</td>
<td>0.07649</td>
<td>0.7233</td>
<td>0.9762</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>chr4:883483:G:A</td>
<td>883483</td>
<td>A</td>
<td>0.07547</td>
<td>0.06904</td>
<td>G</td>
<td>1.5690</td>
<td>0.21040</td>
<td>1.1010</td>
<td>0.07663</td>
<td>0.9472</td>
<td>1.2790</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>chr4:883624:C:T</td>
<td>883624</td>
<td>T</td>
<td>0.40600</td>
<td>0.41950</td>
<td>C</td>
<td>1.8970</td>
<td>0.16840</td>
<td>0.9456</td>
<td>0.04061</td>
<td>0.8733</td>
<td>1.0240</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>372</th>
<td>4</td>
<td>chr4:1007954:T:C</td>
<td>1007954</td>
<td>T</td>
<td>0.42810</td>
<td>0.42160</td>
<td>C</td>
<td>0.4384</td>
<td>0.50790</td>
<td>1.0270</td>
<td>0.04030</td>
<td>0.9490</td>
<td>1.1110</td>
</tr>
<tr>
<th>373</th>
<td>4</td>
<td>chr4:1008009:C:T</td>
<td>1008009</td>
<td>T</td>
<td>0.37580</td>
<td>0.37980</td>
<td>C</td>
<td>0.1604</td>
<td>0.68880</td>
<td>0.9831</td>
<td>0.04266</td>
<td>0.9042</td>
<td>1.0690</td>
</tr>
<tr>
<th>374</th>
<td>4</td>
<td>chr4:1008064:G:C</td>
<td>1008064</td>
<td>C</td>
<td>0.02647</td>
<td>0.02219</td>
<td>G</td>
<td>1.9000</td>
<td>0.16810</td>
<td>1.1980</td>
<td>0.13130</td>
<td>0.9263</td>
<td>1.5500</td>
</tr>
<tr>
<th>375</th>
<td>4</td>
<td>chr4:1008209:C:T</td>
<td>1008209</td>
<td>T</td>
<td>0.10880</td>
<td>0.11340</td>
<td>C</td>
<td>0.5450</td>
<td>0.46040</td>
<td>0.9539</td>
<td>0.06397</td>
<td>0.8415</td>
<td>1.0810</td>
</tr>
<tr>
<th>376</th>
<td>4</td>
<td>chr4:1008337:C:T</td>
<td>1008337</td>
<td>C</td>
<td>0.42970</td>
<td>0.42260</td>
<td>T</td>
<td>0.5250</td>
<td>0.46870</td>
<td>1.0290</td>
<td>0.04012</td>
<td>0.9516</td>
<td>1.1140</td>
</tr>
</tbody>
</table>
<p>377 rows  13 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=b207f773-84b8-41cc-8fc0-5050029d4e54">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[7]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">#Save FREQ to csv</span>
<span class="n">freq</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/EAS.all_nonadj.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=1ccfeb4f-aa13-4b44-87cf-21727d6d1b54">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[8]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># GLM analysis</span>

<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_EAS'</span>

<span class="o">!</span><span class="w"> </span>/home/jupyter/tools/plink2<span class="w"> </span><span class="err">\</span>
<span class="o">--</span><span class="n">bfile</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS_TMEM175</span> \
<span class="o">--</span><span class="n">keep</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS</span><span class="o">.</span><span class="n">samplestoKeep</span> \
<span class="o">--</span><span class="n">allow</span><span class="o">-</span><span class="n">no</span><span class="o">-</span><span class="n">sex</span> \
<span class="o">--</span><span class="n">maf</span> <span class="mf">0.01</span> \
<span class="o">--</span><span class="n">ci</span> <span class="mf">0.95</span> \
<span class="o">--</span><span class="n">glm</span> \
<span class="o">--</span><span class="n">covar</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS_covariate_file</span><span class="o">.</span><span class="n">txt</span> \
<span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">name</span> <span class="n">SEX</span><span class="p">,</span><span class="n">AGE</span><span class="p">,</span><span class="n">PC1</span><span class="p">,</span><span class="n">PC2</span><span class="p">,</span><span class="n">PC3</span><span class="p">,</span><span class="n">PC4</span><span class="p">,</span><span class="n">PC5</span> \
<span class="o">--</span><span class="n">covar</span><span class="o">-</span><span class="n">variance</span><span class="o">-</span><span class="n">standardize</span> \
<span class="o">--</span><span class="n">neg9</span><span class="o">-</span><span class="n">pheno</span><span class="o">-</span><span class="n">really</span><span class="o">-</span><span class="n">missing</span> \
<span class="o">--</span><span class="n">out</span> <span class="p">{</span><span class="n">WORK_DIR</span><span class="p">}</span><span class="o">/</span><span class="n">EAS_TMEM175</span><span class="o">.</span><span class="n">all_adj</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="text/plain" tabindex="0">
<pre>PLINK v2.00a6LM 64-bit Intel (4 Jul 2024)      www.cog-genomics.org/plink/2.0/
(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.log.
Options in effect:
  --allow-no-sex
  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175
  --ci 0.95
  --covar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt
  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5
  --covar-variance-standardize
  --glm
  --keep /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.samplestoKeep
  --maf 0.01
  --neg9-pheno-really-missing
  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj

Start time: Thu Aug 29 12:42:49 2024
Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are
automatically excluded from association analysis when sex is a covariate, and
treated normally otherwise.)
52223 MiB RAM detected, ~50486 available; reserving 26111 MiB for main
workspace.
Using up to 8 compute threads.
5167 samples (1889 females, 3278 males; 5167 founders) loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam.
3171 variants loaded from
/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim.
1 binary phenotype loaded (2662 cases, 2461 controls).
--keep: 5167 samples remaining.
7 covariates loaded from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_covariate_file.txt.
5167 samples (1889 females, 3278 males; 5167 founders) remaining after main
filters.
2662 cases and 2461 controls remaining after main filters.
--covar-variance-standardize: 7 covariates transformed.
Calculating allele frequencies... done.
2794 variants removed due to allele frequency threshold(s)
(--maf/--max-maf/--mac/--max-mac).
377 variants remaining after main filters.
--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.
Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .
End time: Thu Aug 29 12:42:49 2024
</pre>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell" id="cell-id=8a3e6154-6528-4a51-92c0-4b24983a2431">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[9]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1"># read output file of glm</span>

<span class="c1">## Read in plink glm results</span>
<span class="n">WORK_DIR</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">'~/workspace/ws_files/TMEM175/TMEM175_EAS'</span>
<span class="n">assoc</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/EAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid'</span><span class="p">,</span> <span class="n">delim_whitespace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">assoc</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="jp-Cell-outputWrapper">
<div class="jp-Collapser jp-OutputCollapser jp-Cell-outputCollapser">
</div>
<div class="jp-OutputArea jp-Cell-outputArea">
<div class="jp-OutputArea-child">
<div class="jp-OutputPrompt jp-OutputArea-prompt"></div>
<div class="jp-RenderedText jp-OutputArea-output" data-mime-type="application/vnd.jupyter.stderr" tabindex="0">
<pre>/var/tmp/ipykernel_3382/2767392481.py:5: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
</pre>
</div>
</div>
<div class="jp-OutputArea-child jp-OutputArea-executeResult">
<div class="jp-OutputPrompt jp-OutputArea-prompt">Out[9]:</div>
<div class="jp-RenderedHTMLCommon jp-RenderedHTML jp-OutputArea-output jp-OutputArea-executeResult" data-mime-type="text/html" tabindex="0">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>#CHROM</th>
<th>POS</th>
<th>ID</th>
<th>REF</th>
<th>ALT</th>
<th>PROVISIONAL_REF?</th>
<th>A1</th>
<th>OMITTED</th>
<th>A1_FREQ</th>
<th>FIRTH?</th>
<th>TEST</th>
<th>OBS_CT</th>
<th>OR</th>
<th>LOG(OR)_SE</th>
<th>L95</th>
<th>U95</th>
<th>Z_STAT</th>
<th>P</th>
<th>ERRCODE</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.278626</td>
<td>N</td>
<td>ADD</td>
<td>2489</td>
<td>1.079170</td>
<td>0.071422</td>
<td>0.938201</td>
<td>1.241320</td>
<td>1.066800</td>
<td>2.860620e-01</td>
<td>.</td>
</tr>
<tr>
<th>1</th>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.278626</td>
<td>N</td>
<td>SEX</td>
<td>2489</td>
<td>0.804336</td>
<td>0.044091</td>
<td>0.737746</td>
<td>0.876936</td>
<td>-4.938340</td>
<td>7.878890e-07</td>
<td>.</td>
</tr>
<tr>
<th>2</th>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.278626</td>
<td>N</td>
<td>AGE</td>
<td>2489</td>
<td>1.510690</td>
<td>0.047695</td>
<td>1.375870</td>
<td>1.658720</td>
<td>8.650160</td>
<td>5.142550e-18</td>
<td>.</td>
</tr>
<tr>
<th>3</th>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.278626</td>
<td>N</td>
<td>PC1</td>
<td>2489</td>
<td>0.978250</td>
<td>0.044623</td>
<td>0.896327</td>
<td>1.067660</td>
<td>-0.492802</td>
<td>6.221530e-01</td>
<td>.</td>
</tr>
<tr>
<th>4</th>
<td>4</td>
<td>882530</td>
<td>chr4:882530:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>T</td>
<td>C</td>
<td>0.278626</td>
<td>N</td>
<td>PC2</td>
<td>2489</td>
<td>0.997227</td>
<td>0.116869</td>
<td>0.793077</td>
<td>1.253930</td>
<td>-0.023759</td>
<td>9.810450e-01</td>
<td>.</td>
</tr>
<tr>
<th>...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<th>3011</th>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.419561</td>
<td>N</td>
<td>PC1</td>
<td>2505</td>
<td>0.974019</td>
<td>0.044508</td>
<td>0.892651</td>
<td>1.062800</td>
<td>-0.591460</td>
<td>5.542120e-01</td>
<td>.</td>
</tr>
<tr>
<th>3012</th>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.419561</td>
<td>N</td>
<td>PC2</td>
<td>2505</td>
<td>0.980706</td>
<td>0.117453</td>
<td>0.779045</td>
<td>1.234570</td>
<td>-0.165874</td>
<td>8.682560e-01</td>
<td>.</td>
</tr>
<tr>
<th>3013</th>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.419561</td>
<td>N</td>
<td>PC3</td>
<td>2505</td>
<td>0.679168</td>
<td>0.102293</td>
<td>0.555782</td>
<td>0.829945</td>
<td>-3.782130</td>
<td>1.554910e-04</td>
<td>.</td>
</tr>
<tr>
<th>3014</th>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.419561</td>
<td>N</td>
<td>PC4</td>
<td>2505</td>
<td>0.562607</td>
<td>0.066139</td>
<td>0.494206</td>
<td>0.640475</td>
<td>-8.696510</td>
<td>3.422460e-18</td>
<td>.</td>
</tr>
<tr>
<th>3015</th>
<td>4</td>
<td>1008337</td>
<td>chr4:1008337:C:T</td>
<td>C</td>
<td>T</td>
<td>Y</td>
<td>C</td>
<td>T</td>
<td>0.419561</td>
<td>N</td>
<td>PC5</td>
<td>2505</td>
<td>1.009840</td>
<td>0.045912</td>
<td>0.922934</td>
<td>1.104920</td>
<td>0.213190</td>
<td>8.311790e-01</td>
<td>.</td>
</tr>
</tbody>
</table>
<p>3016 rows  19 columns</p>
</div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=5db2b5e3-f4ef-4c47-873c-5624ee91bba0">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[10]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">##Filter for additive test only - this is the variant results</span>
<span class="n">assoc_add</span> <span class="o">=</span> <span class="n">assoc</span><span class="p">[</span><span class="n">assoc</span><span class="p">[</span><span class="s1">'TEST'</span><span class="p">]</span><span class="o">==</span><span class="s2">"ADD"</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=aff36f76-a89a-40c2-9f1a-3de16a25ba2a">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[11]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span><span class="c1">##Save assoc_add to csv</span>
<span class="n">assoc_add</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="sa">f</span><span class="s1">'</span><span class="si">{</span><span class="n">WORK_DIR</span><span class="si">}</span><span class="s1">/EAS.all_adj.csv'</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
</div><div class="jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs" id="cell-id=f2f32bc6-94c4-4528-86b5-a04e5e6a4dcc">
<div class="jp-Cell-inputWrapper" tabindex="0">
<div class="jp-Collapser jp-InputCollapser jp-Cell-inputCollapser">
</div>
<div class="jp-InputArea jp-Cell-inputArea">
<div class="jp-InputPrompt jp-InputArea-prompt">In[]:</div>
<div class="jp-CodeMirrorEditor jp-Editor jp-InputArea-editor" data-type="inline">
<div class="cm-editor cm-s-jupyter">
<div class="highlight hl-ipython3"><pre><span></span> 
</pre></div>
</div>
</div>
</div>
</div>
</div>
</main>
</body>
</html>
