{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c29f044a-c4d3-4731-833f-f35d052d0dc3",
   "metadata": {
    "tags": []
   },
   "source": [
    "CTSB - Single gene analysis in GP2 Neurobooster genotyping data (all ancestries)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cab9ed3-dda0-49d8-ba0a-d5a635c498cd",
   "metadata": {},
   "source": [
    "Project: GP2 CTSB\n",
    "\n",
    "Version: Python/3.10.15,  R/4.3.3"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4af1c540-efad-4924-aa46-bc180b5c6686",
   "metadata": {},
   "source": [
    "Notebook Overview\n",
    "\n",
    "1. Description\n",
    "Loading Python libraries\n",
    "Set paths\n",
    "Make working directory\n",
    "\n",
    "2. Installing packages\n",
    "\n",
    "3. Create a covariate file with GP2 data\n",
    "\n",
    "4. Annotation of the gene CTSB\n",
    "\n",
    "5. Association analysis to compare allele frequencies between cases and controls\n",
    "\n",
    "6. GLM analysis adjusting for gender, age, PC1-5\n",
    "\n",
    "7. Burden test(SkatO, Skat, cmc,zeggini,mb,fp,cmcWald)\n",
    "\n",
    "8. Conditional analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3648ed7-6549-417d-87fd-704e02e19a2d",
   "metadata": {},
   "source": [
    "Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "80fda3ab-dfd6-4e59-87ab-7a92b0723b0e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Use pathlib for file path manipulation\n",
    "import pathlib\n",
    "\n",
    "# Install numpy\n",
    "import numpy as np\n",
    "\n",
    "# Install Pandas for tabular data\n",
    "import pandas as pd\n",
    "\n",
    "# Install plotnine: a ggplot2-compatible Python plotting package\n",
    "from plotnine import *\n",
    "\n",
    "# Always show all columns in a Pandas DataFrame\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0fe63059-59cf-4678-84be-b1c3fe3c2657",
   "metadata": {},
   "source": [
    "Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7a89ca4f-1c05-4f1a-999e-9a83f5bd0a96",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 61K\n",
      "-r--r--r-- 1 jupyter jupyter 61K Jul 23  2024 README_release7_30042024.txt\n",
      "dr-xr-xr-x 1 jupyter jupyter   0 Jun 12 11:04 clinical_data\n",
      "dr-xr-xr-x 1 jupyter jupyter   0 Jun 12 11:04 imputed_genotypes\n",
      "dr-xr-xr-x 1 jupyter jupyter   0 Jun 12 11:04 meta_data\n",
      "dr-xr-xr-x 1 jupyter jupyter   0 Jun 12 11:04 raw_genotypes\n"
     ]
    }
   ],
   "source": [
    "REL7_PATH = pathlib.Path(pathlib.Path.home(), 'workspace/gp2_tier2_eu_release7_30042024')\n",
    "!ls -hal {REL7_PATH}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9145494c-fb2d-4a15-aaca-1320dcbf84d3",
   "metadata": {},
   "source": [
    "Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6f473989-7152-479c-997e-3ddcc10fd56e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/CTSB/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b93660cc-b22d-41d4-b384-47e5c2da35ba",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LICENSE\t\t       plink2\t\t\t\trvtests\n",
      "annovar\t\t       plink2_linux_x86_64_latest.zip\ttoy.map\n",
      "annovar.latest.tar.gz  plink_linux_x86_64_20190304.zip\ttoy.ped\n",
      "plink\t\t       prettify\n"
     ]
    }
   ],
   "source": [
    "# make sure all tools installed\n",
    "! ls /home/jupyter/tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7eea076c-f7ed-4a8c-afb3-cbe7d75bc849",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# give permission\n",
    "\n",
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf9d4887-3db4-4432-b38c-71fa5d596bff",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "# making working directory\n",
    "#Loop over all the ancestries\n",
    "for ancestry in {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'} ;\n",
    "do\n",
    "\n",
    "#Make a folder for each ancestry\n",
    "mkdir ~/workspace/ws_files/CTSB/CTSB_\"$ancestry\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b8b47cb-cff6-41ae-a0e4-62807a600441",
   "metadata": {},
   "source": [
    "Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d6f87cfd-8b58-4c6d-9146-ad017b19e8a0",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "CLINICAL_DATA_PATH = pathlib.Path(REL7_PATH, 'clinical_data/master_key_release7_final_vwb.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffc23ba0-fb42-4008-b64d-7a65439b6b44",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Let's load the master key\n",
    "key = pd.read_csv(CLINICAL_DATA_PATH, low_memory=False)\n",
    "print(key.shape)\n",
    "key.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6aeb83f1-c1f5-436a-a4dc-9aa4b0fed78a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Subsetting to keep only a few columns \n",
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype_for_qc', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'label']]\n",
    "# Renaming the columns\n",
    "key.rename(columns = {'GP2sampleID':'IID',\n",
    "                                     'baseline_GP2_phenotype_for_qc':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n",
    "key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ac16e738-02b0-4810-ad75-033545fecff9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "Related individuals: (245, 9)\n",
      "Unrelated individuals: (2675, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          1655\n",
      "Female                        1009\n",
      "Other/Unknown/Not Reported      11\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       1655\n",
      "2       1009\n",
      "<NA>      11\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 95\n",
      "Number of controls missing age: 195\n",
      "WORKING ON: MDE\n",
      "Related individuals: (19, 9)\n",
      "Unrelated individuals: (589, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          368\n",
      "Female                        219\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       368\n",
      "2       219\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 140\n",
      "Number of controls missing age: 12\n",
      "WORKING ON: EAS\n",
      "Related individuals: (350, 9)\n",
      "Unrelated individuals: (5303, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          3398\n",
      "Female                        1899\n",
      "Other/Unknown/Not Reported       6\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       3398\n",
      "2       1899\n",
      "<NA>       6\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 1349\n",
      "Number of controls missing age: 1268\n",
      "WORKING ON: AMR\n",
      "Related individuals: (51, 9)\n",
      "Unrelated individuals: (672, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          374\n",
      "Female                        296\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       374\n",
      "2       296\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 106\n",
      "Number of controls missing age: 55\n",
      "WORKING ON: CAS\n",
      "Related individuals: (137, 9)\n",
      "Unrelated individuals: (933, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female    483\n",
      "Male      450\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2    483\n",
      "1    450\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 111\n",
      "Number of controls missing age: 88\n",
      "WORKING ON: FIN\n",
      "Related individuals: (7, 9)\n",
      "Unrelated individuals: (117, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female    63\n",
      "Male      54\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2    63\n",
      "1    54\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 62\n",
      "Number of controls missing age: 3\n",
      "WORKING ON: CAH\n",
      "Related individuals: (65, 9)\n",
      "Unrelated individuals: (1108, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          585\n",
      "Female                        498\n",
      "Other/Unknown/Not Reported     25\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       585\n",
      "2       498\n",
      "<NA>     25\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 136\n",
      "Number of controls missing age: 52\n",
      "WORKING ON: SAS\n",
      "Related individuals: (55, 9)\n",
      "Unrelated individuals: (624, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          397\n",
      "Female                        226\n",
      "Other/Unknown/Not Reported      1\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       397\n",
      "2       226\n",
      "<NA>      1\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 203\n",
      "Number of controls missing age: 83\n",
      "WORKING ON: AFR\n",
      "Related individuals: (228, 9)\n",
      "Unrelated individuals: (2752, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          1553\n",
      "Female                        1197\n",
      "Other/Unknown/Not Reported       2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       1553\n",
      "2       1197\n",
      "<NA>       2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 835\n",
      "Number of controls missing age: 809\n",
      "WORKING ON: EUR\n",
      "Related individuals: (3662, 9)\n",
      "Unrelated individuals: (39710, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Male                          23733\n",
      "Female                        15784\n",
      "Other/Unknown/Not Reported      193\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "1       23733\n",
      "2       15784\n",
      "<NA>      193\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 5947\n",
      "Number of controls missing age: 3347\n",
      "WORKING ON: AAC\n",
      "Related individuals: (28, 9)\n",
      "Unrelated individuals: (1181, 6)\n",
      "PCs: (58209, 6)\n",
      "Sex value counts - original:\n",
      " SEX\n",
      "Female                        690\n",
      "Male                          489\n",
      "Other/Unknown/Not Reported      2\n",
      "Sex value counts - recoded:\n",
      "SEX\n",
      "2       690\n",
      "1       489\n",
      "<NA>      2\n",
      "Check columns for covariate file: Index(['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'phenotype', 'SEX', 'AGE',\n",
      "       'AAO', 'label', 'PHENO'],\n",
      "      dtype='object')\n",
      "Number of PD cases missing age: 91\n",
      "Number of controls missing age: 96\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    ## Subset to keep ancestry of interest \n",
    "    ancestry_key = key[key['label']==ancestry].copy()\n",
    "    ancestry_key.reset_index(drop=True)\n",
    "    \n",
    "    # Load information about related individuals in the ancestry analyzed\n",
    "    related_df = pd.read_csv(f'{REL7_PATH}/meta_data/related_samples/{ancestry}_release7_vwb.related')\n",
    "    print(f'Related individuals: {related_df.shape}')\n",
    "    \n",
    "    # Make a list of just one set of related people\n",
    "    related_list = list(related_df['IID1'])\n",
    "    \n",
    "    # Check value counts of related and remove only one related individual\n",
    "    ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "    \n",
    "    # Check size\n",
    "    print(f'Unrelated individuals: {ancestry_key.shape}')\n",
    "    \n",
    "    # Convert phenotype to binary (1/2)\n",
    "    ## Assign conditions so case=2 and controls=1, and -9 otherwise (matching PLINK convention)\n",
    "    # PD = 2; control = 1\n",
    "    pheno_mapping = {\"PD\": 2, \"Control\": 1}\n",
    "    ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')\n",
    "    \n",
    "    # Check value counts of pheno\n",
    "    ancestry_key['PHENO'].value_counts(dropna=False)\n",
    "    \n",
    "    ## Get the PCs\n",
    "    pcs = pd.read_csv(f'{REL7_PATH}/meta_data/qc_metrics/projected_pcs_vwb.csv')\n",
    "    \n",
    "    #Select just first 5 PCs\n",
    "    selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "    pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "    \n",
    "    # Drop the first row (since it's now the column names)\n",
    "    pcs = pcs.drop(0)\n",
    "    \n",
    "    # Reset the index to remove any potential issues\n",
    "    pcs = pcs.reset_index(drop=True)\n",
    "    \n",
    "    # Check size\n",
    "    print(f'PCs: {pcs.shape}')\n",
    "    \n",
    "    # Check value counts of SEX\n",
    "    sex_og_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - original:\\n {sex_og_values.to_string()}')\n",
    "    \n",
    "    # Convert sex to binary (1/2)\n",
    "    ## Assign conditions so female=2 and men=1, and -9 otherwise (matching PLINK convention)\n",
    "    # Female = 2; Male = 1\n",
    "    sex_mapping = {\"Female\": 2, \"Male\": 1}\n",
    "    ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')\n",
    "    \n",
    "    # Check value counts of SEX after recoding\n",
    "    sex_recode_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - recoded:\\n{sex_recode_values.to_string()}')\n",
    "    \n",
    "    ## Make covariate file\n",
    "    df = pd.merge(pcs, ancestry_key, on='IID', how='left')\n",
    "    print(f'Check columns for covariate file: {df.columns}')\n",
    "    \n",
    "    #Make additional columns - FID, fatid and matid - these are needed for RVtests!!\n",
    "    #RVtests needs the first 5 columns to be fid, iid, fatid, matid and sex otherwise it does not run correctly\n",
    "    #Uppercase column name is ok\n",
    "    #See https://zhanxw.github.io/rvtests/#phenotype-file\n",
    "    df['FID'] = 0\n",
    "    df['FATID'] = 0\n",
    "    df['MATID'] = 0\n",
    "    \n",
    "    ## Clean up and keep columns we need \n",
    "    final_df = df[['FID','IID', 'FATID', 'MATID', 'SEX', 'AGE', 'PHENO','PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "    \n",
    "    ##DO NOT replace missing values with -9 as this is misinterpreted by RVtests - needs to be nonnumeric\n",
    "    #Leave missing values as NA\n",
    "    \n",
    "    #Check number of PD cases missing age\n",
    "    pd_missAge = final_df[(final_df['PHENO']==2)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of PD cases missing age: {pd_missAge.shape[0]}')\n",
    "    \n",
    "    #Check number of controls missing age\n",
    "    control_missAge = final_df[(final_df['PHENO']==1)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of controls missing age: {control_missAge.shape[0]}')\n",
    "    \n",
    "    ## Make file of sample IDs to keep \n",
    "    samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "    samples_toKeep.to_csv(f'~/workspace/ws_files/CTSB/CTSB_{ancestry}/{ancestry}.samplestoKeep', sep = '\\t', index=False, header=None)\n",
    "    \n",
    "    ## Make your covariate file\n",
    "    #Included na_rep to write out missing/NA values explicitly as string/text, not as blank otherwise they are misread in RVtests\n",
    "    final_df.to_csv(f'~/workspace/ws_files/CTSB/CTSB_{ancestry}/{ancestry}_covariate_file.txt', sep = '\\t', na_rep='NA', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74c1def6-fba4-4de5-94e9-b588be378ecf",
   "metadata": {},
   "source": [
    "Annotation of the gene\n",
    "\n",
    "Extract the region using PLINK\n",
    "\n",
    "Extract CTSB gene\n",
    "\n",
    "CTSB coordinates: Chromosome 8: 11,842,524-11,869,533(GRCh38/hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "53709e58-4dc5-4961-b6b2-8755fb0d7498",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr8_AJ_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:05:56 2025\n",
      "12984 MiB RAM detected, ~11272 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr8_AJ_release7_vwb.psam.\n",
      "1308065 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AJ/chr8_AJ_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "2136 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:05:58 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr8_MDE_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:05:58 2025\n",
      "12984 MiB RAM detected, ~11265 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr8_MDE_release7_vwb.psam.\n",
      "1708855 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/MDE/chr8_MDE_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "2545 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:06:01 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr8_EAS_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:01 2025\n",
      "12984 MiB RAM detected, ~11243 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr8_EAS_release7_vwb.psam.\n",
      "2581801 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EAS/chr8_EAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "4081 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bed ... 91done.\n",
      "End time: Thu Jun 12 11:06:05 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr8_AMR_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:05 2025\n",
      "12984 MiB RAM detected, ~11252 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr8_AMR_release7_vwb.psam.\n",
      "1937056 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AMR/chr8_AMR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "3038 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:06:09 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr8_CAS_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:09 2025\n",
      "12984 MiB RAM detected, ~11255 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr8_CAS_release7_vwb.psam.\n",
      "1742361 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAS/chr8_CAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "2607 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:06:12 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr8_FIN_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:12 2025\n",
      "12984 MiB RAM detected, ~11257 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr8_FIN_release7_vwb.psam.\n",
      "678945 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/FIN/chr8_FIN_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "1007 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:06:13 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr8_CAH_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:14 2025\n",
      "12984 MiB RAM detected, ~11252 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr8_CAH_release7_vwb.psam.\n",
      "2931233 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/CAH/chr8_CAH_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "4946 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:06:17 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr8_SAS_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:18 2025\n",
      "12984 MiB RAM detected, ~11253 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr8_SAS_release7_vwb.psam.\n",
      "1515719 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/SAS/chr8_SAS_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "2446 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:06:20 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr8_AFR_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:20 2025\n",
      "12984 MiB RAM detected, ~11254 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr8_AFR_release7_vwb.psam.\n",
      "3638791 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AFR/chr8_AFR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "6381 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bed ... 63done.\n",
      "End time: Thu Jun 12 11:06:24 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr8_EUR_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:24 2025\n",
      "12984 MiB RAM detected, ~11255 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr8_EUR_release7_vwb.psam.\n",
      "8334722 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/EUR/chr8_EUR_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "13481 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bed ... done.\n",
      "End time: Thu Jun 12 11:06:34 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr8_AAC_release7_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Thu Jun 12 11:06:35 2025\n",
      "12984 MiB RAM detected, ~11250 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr8_AAC_release7_vwb.psam.\n",
      "3182900 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release7_30042024/imputed_genotypes/AAC/chr8_AAC_release7_vwb.pvar.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "5351 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bed ... 65done.\n",
      "End time: Thu Jun 12 11:06:38 2025\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL7_PATH}/imputed_genotypes/{ancestry}/chr8_{ancestry}_release7_vwb \\\n",
    "    --chr 8 \\\n",
    "    --from-bp 11792524 \\\n",
    "    --to-bp 11919533 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "37034ce7-40bb-4790-91a6-6caf95f4913b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8\tchr8:11792554:C:T\t0\t11792554\tT\tC\n",
      "8\tchr8:11792575:T:C\t0\t11792575\tC\tT\n",
      "8\tchr8:11792593:T:G\t0\t11792593\tG\tT\n",
      "8\tchr8:11792594:C:G\t0\t11792594\tG\tC\n",
      "8\tchr8:11792658:C:G\t0\t11792658\tG\tC\n",
      "8\tchr8:11792660:C:G\t0\t11792660\tG\tC\n",
      "8\tchr8:11792663:T:C\t0\t11792663\tC\tT\n",
      "8\tchr8:11792668:T:G\t0\t11792668\tG\tT\n",
      "8\tchr8:11792679:G:C\t0\t11792679\tC\tG\n",
      "8\tchr8:11792680:C:T\t0\t11792680\tT\tC\n"
     ]
    }
   ],
   "source": [
    "# Visualize bim file\n",
    "WORK_DIR = '/home/jupyter/workspace/ws_files/CTSB/'\n",
    "! head {WORK_DIR}/CTSB_EUR/EUR_CTSB.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b236156-2221-4618-9ad3-249e390a978f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Visualize bim file\n",
    "! head {WORK_DIR}/CTSB_EUR/EUR_CTSB.fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3620d63b-a4fc-4651-97c1-d2a1630eaad6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}/'\n",
    "    \n",
    "    ! head -n 1 {WORK_DIR}/{ancestry}_CTSB.fam > {WORK_DIR}/{ancestry}_s1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "095977c9-1a7c-421f-8895-9247d03fb89c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! head /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_s1.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16c34dc6-c097-466c-bce6-0c9ea6ca1acf",
   "metadata": {},
   "source": [
    "Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "7bf683eb-d6f3-49a0-8309-4ca5522217f8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:41 2025\n",
      "12984 MiB RAM detected, ~11262 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam.\n",
      "2136 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.fam ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.bim ... done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.bed ... done.\n",
      "End time: Thu Jun 12 11:13:41 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:42 2025\n",
      "12984 MiB RAM detected, ~11244 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam.\n",
      "2545 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:42 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:43 2025\n",
      "12984 MiB RAM detected, ~11243 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam.\n",
      "4081 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:43 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:44 2025\n",
      "12984 MiB RAM detected, ~11243 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam.\n",
      "3038 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:45 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:45 2025\n",
      "12984 MiB RAM detected, ~11244 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam.\n",
      "2607 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:46 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:46 2025\n",
      "12984 MiB RAM detected, ~11243 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam.\n",
      "1007 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:47 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:47 2025\n",
      "12984 MiB RAM detected, ~11243 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam.\n",
      "4946 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:48 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:48 2025\n",
      "12984 MiB RAM detected, ~11245 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam.\n",
      "2446 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:49 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:49 2025\n",
      "12984 MiB RAM detected, ~11243 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam.\n",
      "6381 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:50 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:50 2025\n",
      "12984 MiB RAM detected, ~11243 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam.\n",
      "13481 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:51 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:13:51 2025\n",
      "12984 MiB RAM detected, ~11244 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam.\n",
      "5351 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Thu Jun 12 11:13:52 2025\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --keep {WORK_DIR}/{ancestry}_s1.txt \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e4afbf98-6634-4fd2-a626-462b2d87aeba",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:34 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11295 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.fam.\n",
      "2136 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324252526262727282829293030313132323333343435353636373738383939404041414242434344444545464647474848495050515152525353545455555656575758585959606061616262636364646565666667676868696970707171727273737475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:34 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:34 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11264 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.fam.\n",
      "2545 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.vcf\n",
      "... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:34 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:35 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11250 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.fam.\n",
      "4081 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:35 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:35 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11248 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.fam.\n",
      "3038 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.vcf\n",
      "... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:36 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:36 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11248 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.fam.\n",
      "2607 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:36 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:37 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11249 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.fam.\n",
      "1007 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:37 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:37 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11248 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.fam.\n",
      "4946 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.vcf\n",
      "... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:37 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:38 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11248 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.fam.\n",
      "2446 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.vcf\n",
      "... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:38 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:38 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11250 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.fam.\n",
      "6381 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:38 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:39 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11248 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.fam.\n",
      "13481 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:39 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1\n",
      "\n",
      "Start time: Thu Jun 12 11:18:39 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "12984 MiB RAM detected, ~11248 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.fam.\n",
      "5351 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.vcf\n",
      "... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Thu Jun 12 11:18:40 2025\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB_v1 \\\n",
    "    --recode vcf-fid \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "38d9a4f3-2a27-46ff-bd55-c6b440e7dd42",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    ! bgzip -f {WORK_DIR}/{ancestry}_CTSB_v1.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/{ancestry}_CTSB_v1.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c152d2d-e669-4d2d-b0fc-52dd16593239",
   "metadata": {},
   "source": [
    "Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7996d11d-9764-4979-a486-42bddf98f3d0",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 2143 lines from VCF file\n",
      "NOTICE: A total of 2136 locus in VCF file passed QC threshold, representing 2024 SNPs (1187 transitions and 837 transversions) and 112 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2024 SNP genotypes (1187 transitions and 837 transversions) and 112 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2136 unique variants in 2136 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2136 unique variants in 2136 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 2552 lines from VCF file\n",
      "NOTICE: A total of 2545 locus in VCF file passed QC threshold, representing 2405 SNPs (1425 transitions and 980 transversions) and 140 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2405 SNP genotypes (1425 transitions and 980 transversions) and 140 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2545 unique variants in 2545 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2545 unique variants in 2545 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 4088 lines from VCF file\n",
      "NOTICE: A total of 4081 locus in VCF file passed QC threshold, representing 3845 SNPs (2286 transitions and 1559 transversions) and 236 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3845 SNP genotypes (2286 transitions and 1559 transversions) and 236 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 4081 unique variants in 4081 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 4081 unique variants in 4081 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 3045 lines from VCF file\n",
      "NOTICE: A total of 3038 locus in VCF file passed QC threshold, representing 2881 SNPs (1734 transitions and 1147 transversions) and 157 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2881 SNP genotypes (1734 transitions and 1147 transversions) and 157 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3038 unique variants in 3038 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3038 unique variants in 3038 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 2614 lines from VCF file\n",
      "NOTICE: A total of 2607 locus in VCF file passed QC threshold, representing 2466 SNPs (1460 transitions and 1006 transversions) and 141 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2466 SNP genotypes (1460 transitions and 1006 transversions) and 141 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2607 unique variants in 2607 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2607 unique variants in 2607 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 1014 lines from VCF file\n",
      "NOTICE: A total of 1007 locus in VCF file passed QC threshold, representing 958 SNPs (590 transitions and 368 transversions) and 49 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 958 SNP genotypes (590 transitions and 368 transversions) and 49 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1007 unique variants in 1007 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1007 unique variants in 1007 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 4953 lines from VCF file\n",
      "NOTICE: A total of 4946 locus in VCF file passed QC threshold, representing 4675 SNPs (2717 transitions and 1958 transversions) and 271 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 4675 SNP genotypes (2717 transitions and 1958 transversions) and 271 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 4946 unique variants in 4946 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 4946 unique variants in 4946 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 2453 lines from VCF file\n",
      "NOTICE: A total of 2446 locus in VCF file passed QC threshold, representing 2306 SNPs (1402 transitions and 904 transversions) and 140 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2306 SNP genotypes (1402 transitions and 904 transversions) and 140 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2446 unique variants in 2446 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2446 unique variants in 2446 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 6388 lines from VCF file\n",
      "NOTICE: A total of 6381 locus in VCF file passed QC threshold, representing 6026 SNPs (3477 transitions and 2549 transversions) and 355 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 6026 SNP genotypes (3477 transitions and 2549 transversions) and 355 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 6381 unique variants in 6381 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 6381 unique variants in 6381 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 13488 lines from VCF file\n",
      "NOTICE: A total of 13481 locus in VCF file passed QC threshold, representing 12749 SNPs (7496 transitions and 5253 transversions) and 732 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 12749 SNP genotypes (7496 transitions and 5253 transversions) and 732 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 13481 unique variants in 13481 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 13481 unique variants in 13481 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 5358 lines from VCF file\n",
      "NOTICE: A total of 5351 locus in VCF file passed QC threshold, representing 5053 SNPs (2874 transitions and 2179 transversions) and 298 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 5053 SNP genotypes (2874 transitions and 2179 transversions) and 298 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 5351 unique variants in 5351 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 5351 unique variants in 5351 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_CTSB_v1.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "    -out {WORK_DIR}/{ancestry}_CTSB.annovar \\\n",
    "    -remove -protocol refGene,clinvar_20140902 \\\n",
    "    -operation g,f \\\n",
    "    --nopolish \\\n",
    "    -nastring . \\\n",
    "    -vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "d5cf484f-034d-44b4-91d2-f72b5ce9181c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"/home/jupyter/workspace/ws_files/CTSB\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "ee0b6c7d-2251-4e6e-98b3-d05c48db0884",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792650</td>\n",
       "      <td>11792650</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5305;dist=2923</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792650</td>\n",
       "      <td>chr8:11792650:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792688</td>\n",
       "      <td>11792688</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5343;dist=2885</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792688</td>\n",
       "      <td>chr8:11792688:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>11792691</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5346;dist=2882</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>chr8:11792691:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>11792730</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5385;dist=2843</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>chr8:11792730:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5346</th>\n",
       "      <td>8</td>\n",
       "      <td>11919417</td>\n",
       "      <td>11919417</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51330;dist=54520</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919417</td>\n",
       "      <td>chr8:11919417:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5347</th>\n",
       "      <td>8</td>\n",
       "      <td>11919431</td>\n",
       "      <td>11919431</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51344;dist=54506</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919431</td>\n",
       "      <td>chr8:11919431:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5348</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5349</th>\n",
       "      <td>8</td>\n",
       "      <td>11919498</td>\n",
       "      <td>11919498</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51411;dist=54439</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919498</td>\n",
       "      <td>chr8:11919498:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5350</th>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>11919512</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51425;dist=54425</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>chr8:11919512:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5351 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792650  11792650   G   T   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792688  11792688   G   C   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792691  11792691   C   T   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792730  11792730   C   A   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "5346    8  11919417  11919417   T   C   intergenic  CTSB;DEFB136   \n",
       "5347    8  11919431  11919431   T   C   intergenic  CTSB;DEFB136   \n",
       "5348    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "5349    8  11919498  11919498   C   G   intergenic  CTSB;DEFB136   \n",
       "5350    8  11919512  11919512   T   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5305;dist=2923                  .                .   \n",
       "2       dist=5343;dist=2885                  .                .   \n",
       "3       dist=5346;dist=2882                  .                .   \n",
       "4       dist=5385;dist=2843                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "5346  dist=51330;dist=54520                  .                .   \n",
       "5347  dist=51344;dist=54506                  .                .   \n",
       "5348  dist=51363;dist=54487                  .                .   \n",
       "5349  dist=51411;dist=54439                  .                .   \n",
       "5350  dist=51425;dist=54425                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "5346                .          0          .          .           8   \n",
       "5347                .          0          .          .           8   \n",
       "5348                .          0          .          .           8   \n",
       "5349                .          0          .          .           8   \n",
       "5350                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792650  chr8:11792650:G:T          G          T          .   \n",
       "2       11792688  chr8:11792688:G:C          G          C          .   \n",
       "3       11792691  chr8:11792691:C:T          C          T          .   \n",
       "4       11792730  chr8:11792730:C:A          C          A          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "5346    11919417  chr8:11919417:T:C          T          C          .   \n",
       "5347    11919431  chr8:11919431:T:C          T          C          .   \n",
       "5348    11919450  chr8:11919450:G:C          G          C          .   \n",
       "5349    11919498  chr8:11919498:C:G          C          G          .   \n",
       "5350    11919512  chr8:11919512:T:C          T          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "5346           .          PR          GT         0/0  \n",
       "5347           .          PR          GT         0/0  \n",
       "5348           .          PR          GT         0/0  \n",
       "5349           .          PR          GT         0/0  \n",
       "5350           .          PR          GT         0/0  \n",
       "\n",
       "[5351 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_AAC/AAC_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "13cc4a12-b7c6-480a-90a2-b25313eb704e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      2872\n",
       "intergenic    1816\n",
       "UTR5           193\n",
       "UTR3           178\n",
       "upstream       108\n",
       "downstream     104\n",
       "exonic          78\n",
       "splicing         2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "9bc946c2-aa92-42f5-ac0b-227fd00b1a62",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "ca3e6de5-cb55-48b1-830f-86be3c2deb53",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "synonymous SNV       39\n",
       "nonsynonymous SNV    38\n",
       "stopgain              1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "75dc52b6-3def-471a-ae81-97ee72978cd8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>11792594</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5249;dist=2979</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>chr8:11792594:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>11792594</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5249;dist=2979</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>chr8:11792594:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>11792691</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5346;dist=2882</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>chr8:11792691:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>11792730</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5385;dist=2843</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>chr8:11792730:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6376</th>\n",
       "      <td>8</td>\n",
       "      <td>11919417</td>\n",
       "      <td>11919417</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51330;dist=54520</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919417</td>\n",
       "      <td>chr8:11919417:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6377</th>\n",
       "      <td>8</td>\n",
       "      <td>11919431</td>\n",
       "      <td>11919431</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51344;dist=54506</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919431</td>\n",
       "      <td>chr8:11919431:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6378</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6379</th>\n",
       "      <td>8</td>\n",
       "      <td>11919454</td>\n",
       "      <td>11919454</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51367;dist=54483</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919454</td>\n",
       "      <td>chr8:11919454:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6380</th>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>11919512</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51425;dist=54425</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>chr8:11919512:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6381 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792594  11792594   C   A   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792594  11792594   C   G   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792691  11792691   C   T   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792730  11792730   C   A   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "6376    8  11919417  11919417   T   C   intergenic  CTSB;DEFB136   \n",
       "6377    8  11919431  11919431   T   C   intergenic  CTSB;DEFB136   \n",
       "6378    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "6379    8  11919454  11919454   T   C   intergenic  CTSB;DEFB136   \n",
       "6380    8  11919512  11919512   T   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5249;dist=2979                  .                .   \n",
       "2       dist=5249;dist=2979                  .                .   \n",
       "3       dist=5346;dist=2882                  .                .   \n",
       "4       dist=5385;dist=2843                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "6376  dist=51330;dist=54520                  .                .   \n",
       "6377  dist=51344;dist=54506                  .                .   \n",
       "6378  dist=51363;dist=54487                  .                .   \n",
       "6379  dist=51367;dist=54483                  .                .   \n",
       "6380  dist=51425;dist=54425                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "6376                .          0          .          .           8   \n",
       "6377                .          0          .          .           8   \n",
       "6378                .          0          .          .           8   \n",
       "6379                .          0          .          .           8   \n",
       "6380                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792594  chr8:11792594:C:A          C          A          .   \n",
       "2       11792594  chr8:11792594:C:G          C          G          .   \n",
       "3       11792691  chr8:11792691:C:T          C          T          .   \n",
       "4       11792730  chr8:11792730:C:A          C          A          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "6376    11919417  chr8:11919417:T:C          T          C          .   \n",
       "6377    11919431  chr8:11919431:T:C          T          C          .   \n",
       "6378    11919450  chr8:11919450:G:C          G          C          .   \n",
       "6379    11919454  chr8:11919454:T:C          T          C          .   \n",
       "6380    11919512  chr8:11919512:T:C          T          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "6376           .          PR          GT         0/0  \n",
       "6377           .          PR          GT         0/0  \n",
       "6378           .          PR          GT         0/0  \n",
       "6379           .          PR          GT         0/0  \n",
       "6380           .          PR          GT         0/0  \n",
       "\n",
       "[6381 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_AFR/AFR_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "51628a99-c812-4e52-95e5-78572f8886c9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      3425\n",
       "intergenic    2182\n",
       "UTR5           240\n",
       "UTR3           203\n",
       "upstream       119\n",
       "downstream     119\n",
       "exonic          92\n",
       "splicing         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "51cc9a8d-8f94-4b81-9bf6-a069f27cec7c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4bf1ec5f-afda-405e-b917-be5a50d05669",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    51\n",
       "synonymous SNV       41\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "9bf56571-e1d5-44a3-ab88-5643bdc90eee",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792661</td>\n",
       "      <td>11792662</td>\n",
       "      <td>TT</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5316;dist=2911</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792660</td>\n",
       "      <td>chr8:11792660:CTT:C</td>\n",
       "      <td>CTT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>11792812</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5467;dist=2761</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>chr8:11792812:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792935</td>\n",
       "      <td>11792941</td>\n",
       "      <td>ATTAAGT</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5590;dist=2632</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792934</td>\n",
       "      <td>chr8:11792934:AATTAAGT:A</td>\n",
       "      <td>AATTAAGT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792966</td>\n",
       "      <td>11792966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5621;dist=2607</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792966</td>\n",
       "      <td>chr8:11792966:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2131</th>\n",
       "      <td>8</td>\n",
       "      <td>11918972</td>\n",
       "      <td>11918972</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=50885;dist=54965</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11918972</td>\n",
       "      <td>chr8:11918972:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2132</th>\n",
       "      <td>8</td>\n",
       "      <td>11919062</td>\n",
       "      <td>11919062</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=50975;dist=54875</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919062</td>\n",
       "      <td>chr8:11919062:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2133</th>\n",
       "      <td>8</td>\n",
       "      <td>11919200</td>\n",
       "      <td>11919200</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51113;dist=54737</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919200</td>\n",
       "      <td>chr8:11919200:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2134</th>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>11919223</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51136;dist=54714</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>chr8:11919223:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2135</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2136 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End      Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575        T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792661  11792662       TT   -   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792812  11792812        C   T   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792935  11792941  ATTAAGT   -   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792966  11792966        T   C   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...      ...  ..          ...           ...   \n",
       "2131    8  11918972  11918972        G   A   intergenic  CTSB;DEFB136   \n",
       "2132    8  11919062  11919062        A   G   intergenic  CTSB;DEFB136   \n",
       "2133    8  11919200  11919200        G   A   intergenic  CTSB;DEFB136   \n",
       "2134    8  11919223  11919223        G   A   intergenic  CTSB;DEFB136   \n",
       "2135    8  11919450  11919450        G   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5316;dist=2911                  .                .   \n",
       "2       dist=5467;dist=2761                  .                .   \n",
       "3       dist=5590;dist=2632                  .                .   \n",
       "4       dist=5621;dist=2607                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "2131  dist=50885;dist=54965                  .                .   \n",
       "2132  dist=50975;dist=54875                  .                .   \n",
       "2133  dist=51113;dist=54737                  .                .   \n",
       "2134  dist=51136;dist=54714                  .                .   \n",
       "2135  dist=51363;dist=54487                  .                .   \n",
       "\n",
       "     clinvar_20140902  Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .         0.0          .          .           8   \n",
       "1                   .         0.0          .          .           8   \n",
       "2                   .         0.0          .          .           8   \n",
       "3                   .         0.0          .          .           8   \n",
       "4                   .         0.0          .          .           8   \n",
       "...               ...         ...        ...        ...         ...   \n",
       "2131                .         0.0          .          .           8   \n",
       "2132                .         0.0          .          .           8   \n",
       "2133                .         0.0          .          .           8   \n",
       "2134                .         0.0          .          .           8   \n",
       "2135                .         0.0          .          .           8   \n",
       "\n",
       "      Otherinfo5                Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575         chr8:11792575:T:C          T          C          .   \n",
       "1       11792660       chr8:11792660:CTT:C        CTT          C          .   \n",
       "2       11792812         chr8:11792812:C:T          C          T          .   \n",
       "3       11792934  chr8:11792934:AATTAAGT:A   AATTAAGT          A          .   \n",
       "4       11792966         chr8:11792966:T:C          T          C          .   \n",
       "...          ...                       ...        ...        ...        ...   \n",
       "2131    11918972         chr8:11918972:G:A          G          A          .   \n",
       "2132    11919062         chr8:11919062:A:G          A          G          .   \n",
       "2133    11919200         chr8:11919200:G:A          G          A          .   \n",
       "2134    11919223         chr8:11919223:G:A          G          A          .   \n",
       "2135    11919450         chr8:11919450:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "2131           .          PR          GT         0/0  \n",
       "2132           .          PR          GT         0/0  \n",
       "2133           .          PR          GT         0/0  \n",
       "2134           .          PR          GT         0/0  \n",
       "2135           .          PR          GT         0/0  \n",
       "\n",
       "[2136 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_AJ/AJ_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "3aa044be-3783-47a5-a264-5f89c912c506",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1131\n",
       "intergenic     720\n",
       "UTR5            94\n",
       "UTR3            65\n",
       "downstream      49\n",
       "upstream        46\n",
       "exonic          31\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "714765c6-1f64-415e-b452-ebfa5cd1feeb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "81884d39-f363-4136-beed-b3341d2651be",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    22\n",
       "synonymous SNV        9\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "628400ed-3ea7-4eff-a3db-a82a22f97d1f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792619</td>\n",
       "      <td>11792619</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5274;dist=2954</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792619</td>\n",
       "      <td>chr8:11792619:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>11792691</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5346;dist=2882</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>chr8:11792691:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>11792730</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5385;dist=2843</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>chr8:11792730:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792798</td>\n",
       "      <td>11792798</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5453;dist=2775</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792798</td>\n",
       "      <td>chr8:11792798:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3033</th>\n",
       "      <td>8</td>\n",
       "      <td>11919279</td>\n",
       "      <td>11919279</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51192;dist=54658</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919279</td>\n",
       "      <td>chr8:11919279:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3034</th>\n",
       "      <td>8</td>\n",
       "      <td>11919300</td>\n",
       "      <td>11919300</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51213;dist=54637</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919300</td>\n",
       "      <td>chr8:11919300:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3035</th>\n",
       "      <td>8</td>\n",
       "      <td>11919311</td>\n",
       "      <td>11919311</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51224;dist=54626</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919311</td>\n",
       "      <td>chr8:11919311:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3036</th>\n",
       "      <td>8</td>\n",
       "      <td>11919332</td>\n",
       "      <td>11919332</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51245;dist=54605</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919332</td>\n",
       "      <td>chr8:11919332:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3037</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3038 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792619  11792619   C   T   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792691  11792691   C   T   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792730  11792730   C   A   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792798  11792798   G   A   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "3033    8  11919279  11919279   A   G   intergenic  CTSB;DEFB136   \n",
       "3034    8  11919300  11919300   G   T   intergenic  CTSB;DEFB136   \n",
       "3035    8  11919311  11919311   C   T   intergenic  CTSB;DEFB136   \n",
       "3036    8  11919332  11919332   C   T   intergenic  CTSB;DEFB136   \n",
       "3037    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5274;dist=2954                  .                .   \n",
       "2       dist=5346;dist=2882                  .                .   \n",
       "3       dist=5385;dist=2843                  .                .   \n",
       "4       dist=5453;dist=2775                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "3033  dist=51192;dist=54658                  .                .   \n",
       "3034  dist=51213;dist=54637                  .                .   \n",
       "3035  dist=51224;dist=54626                  .                .   \n",
       "3036  dist=51245;dist=54605                  .                .   \n",
       "3037  dist=51363;dist=54487                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "3033                .          0          .          .           8   \n",
       "3034                .          0          .          .           8   \n",
       "3035                .          0          .          .           8   \n",
       "3036                .          0          .          .           8   \n",
       "3037                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792619  chr8:11792619:C:T          C          T          .   \n",
       "2       11792691  chr8:11792691:C:T          C          T          .   \n",
       "3       11792730  chr8:11792730:C:A          C          A          .   \n",
       "4       11792798  chr8:11792798:G:A          G          A          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "3033    11919279  chr8:11919279:A:G          A          G          .   \n",
       "3034    11919300  chr8:11919300:G:T          G          T          .   \n",
       "3035    11919311  chr8:11919311:C:T          C          T          .   \n",
       "3036    11919332  chr8:11919332:C:T          C          T          .   \n",
       "3037    11919450  chr8:11919450:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "3033           .          PR          GT         0/0  \n",
       "3034           .          PR          GT         0/0  \n",
       "3035           .          PR          GT         0/0  \n",
       "3036           .          PR          GT         0/0  \n",
       "3037           .          PR          GT         0/0  \n",
       "\n",
       "[3038 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_AMR/AMR_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "24737065-4eb4-40a6-89a9-976f7422b9b9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1621\n",
       "intergenic    1016\n",
       "UTR5           118\n",
       "UTR3            97\n",
       "downstream      78\n",
       "exonic          55\n",
       "upstream        53\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "715450f5-e014-4a08-8502-1485628297be",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "0f0277e9-cb5a-4a10-ba9c-80129f59084c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    35\n",
       "synonymous SNV       20\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "fb5087a5-4ede-4c35-9d48-e94c6544b851",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>11792594</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5249;dist=2979</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>chr8:11792594:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>11792691</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5346;dist=2882</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>chr8:11792691:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792713</td>\n",
       "      <td>11792713</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5368;dist=2860</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792713</td>\n",
       "      <td>chr8:11792713:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>11792730</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5385;dist=2843</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792730</td>\n",
       "      <td>chr8:11792730:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4941</th>\n",
       "      <td>8</td>\n",
       "      <td>11919376</td>\n",
       "      <td>11919376</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51289;dist=54561</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919376</td>\n",
       "      <td>chr8:11919376:A:T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4942</th>\n",
       "      <td>8</td>\n",
       "      <td>11919387</td>\n",
       "      <td>11919387</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51300;dist=54550</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919387</td>\n",
       "      <td>chr8:11919387:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4943</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4944</th>\n",
       "      <td>8</td>\n",
       "      <td>11919508</td>\n",
       "      <td>11919508</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51421;dist=54429</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919508</td>\n",
       "      <td>chr8:11919508:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4945</th>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>11919512</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51425;dist=54425</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>chr8:11919512:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4946 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792594  11792594   C   G   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792691  11792691   C   T   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792713  11792713   T   C   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792730  11792730   C   A   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "4941    8  11919376  11919376   A   T   intergenic  CTSB;DEFB136   \n",
       "4942    8  11919387  11919387   G   A   intergenic  CTSB;DEFB136   \n",
       "4943    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "4944    8  11919508  11919508   C   G   intergenic  CTSB;DEFB136   \n",
       "4945    8  11919512  11919512   T   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5249;dist=2979                  .                .   \n",
       "2       dist=5346;dist=2882                  .                .   \n",
       "3       dist=5368;dist=2860                  .                .   \n",
       "4       dist=5385;dist=2843                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "4941  dist=51289;dist=54561                  .                .   \n",
       "4942  dist=51300;dist=54550                  .                .   \n",
       "4943  dist=51363;dist=54487                  .                .   \n",
       "4944  dist=51421;dist=54429                  .                .   \n",
       "4945  dist=51425;dist=54425                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "4941                .          0          .          .           8   \n",
       "4942                .          0          .          .           8   \n",
       "4943                .          0          .          .           8   \n",
       "4944                .          0          .          .           8   \n",
       "4945                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792594  chr8:11792594:C:G          C          G          .   \n",
       "2       11792691  chr8:11792691:C:T          C          T          .   \n",
       "3       11792713  chr8:11792713:T:C          T          C          .   \n",
       "4       11792730  chr8:11792730:C:A          C          A          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "4941    11919376  chr8:11919376:A:T          A          T          .   \n",
       "4942    11919387  chr8:11919387:G:A          G          A          .   \n",
       "4943    11919450  chr8:11919450:G:C          G          C          .   \n",
       "4944    11919508  chr8:11919508:C:G          C          G          .   \n",
       "4945    11919512  chr8:11919512:T:C          T          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "4941           .          PR          GT         0/0  \n",
       "4942           .          PR          GT         0/0  \n",
       "4943           .          PR          GT         0/0  \n",
       "4944           .          PR          GT         0/0  \n",
       "4945           .          PR          GT         0/0  \n",
       "\n",
       "[4946 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_CAH/CAH_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "07f15c79-94bd-4bae-b741-b4f48cc52c0b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      2646\n",
       "intergenic    1677\n",
       "UTR5           192\n",
       "UTR3           154\n",
       "downstream     106\n",
       "upstream        92\n",
       "exonic          78\n",
       "splicing         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "755c6b25-8a4a-49f6-9aed-a5d920ec756f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "6b110045-ec47-4e56-9cb0-c9ac64a1dff4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV      40\n",
       "synonymous SNV         37\n",
       "frameshift deletion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "d3167670-ff93-4f98-bfcf-1b65329c1b61",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792660</td>\n",
       "      <td>11792660</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5315;dist=2913</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792660</td>\n",
       "      <td>chr8:11792660:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792688</td>\n",
       "      <td>11792688</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5343;dist=2885</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792688</td>\n",
       "      <td>chr8:11792688:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792784</td>\n",
       "      <td>11792784</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5439;dist=2789</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792784</td>\n",
       "      <td>chr8:11792784:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792794</td>\n",
       "      <td>11792794</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5449;dist=2779</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792794</td>\n",
       "      <td>chr8:11792794:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2602</th>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>11919223</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51136;dist=54714</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>chr8:11919223:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2603</th>\n",
       "      <td>8</td>\n",
       "      <td>11919310</td>\n",
       "      <td>11919310</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51223;dist=54627</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919310</td>\n",
       "      <td>chr8:11919310:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2604</th>\n",
       "      <td>8</td>\n",
       "      <td>11919373</td>\n",
       "      <td>11919373</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51286;dist=54564</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919373</td>\n",
       "      <td>chr8:11919373:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2605</th>\n",
       "      <td>8</td>\n",
       "      <td>11919399</td>\n",
       "      <td>11919399</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51312;dist=54538</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919399</td>\n",
       "      <td>chr8:11919399:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2606</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2607 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792660  11792660   C   G   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792688  11792688   G   C   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792784  11792784   G   A   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792794  11792794   T   C   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "2602    8  11919223  11919223   G   A   intergenic  CTSB;DEFB136   \n",
       "2603    8  11919310  11919310   C   G   intergenic  CTSB;DEFB136   \n",
       "2604    8  11919373  11919373   C   T   intergenic  CTSB;DEFB136   \n",
       "2605    8  11919399  11919399   C   A   intergenic  CTSB;DEFB136   \n",
       "2606    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5315;dist=2913                  .                .   \n",
       "2       dist=5343;dist=2885                  .                .   \n",
       "3       dist=5439;dist=2789                  .                .   \n",
       "4       dist=5449;dist=2779                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "2602  dist=51136;dist=54714                  .                .   \n",
       "2603  dist=51223;dist=54627                  .                .   \n",
       "2604  dist=51286;dist=54564                  .                .   \n",
       "2605  dist=51312;dist=54538                  .                .   \n",
       "2606  dist=51363;dist=54487                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "2602                .          0          .          .           8   \n",
       "2603                .          0          .          .           8   \n",
       "2604                .          0          .          .           8   \n",
       "2605                .          0          .          .           8   \n",
       "2606                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792660  chr8:11792660:C:G          C          G          .   \n",
       "2       11792688  chr8:11792688:G:C          G          C          .   \n",
       "3       11792784  chr8:11792784:G:A          G          A          .   \n",
       "4       11792794  chr8:11792794:T:C          T          C          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "2602    11919223  chr8:11919223:G:A          G          A          .   \n",
       "2603    11919310  chr8:11919310:C:G          C          G          .   \n",
       "2604    11919373  chr8:11919373:C:T          C          T          .   \n",
       "2605    11919399  chr8:11919399:C:A          C          A          .   \n",
       "2606    11919450  chr8:11919450:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "2602           .          PR          GT         0/0  \n",
       "2603           .          PR          GT         0/0  \n",
       "2604           .          PR          GT         0/0  \n",
       "2605           .          PR          GT         0/0  \n",
       "2606           .          PR          GT         0/0  \n",
       "\n",
       "[2607 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_CAS/CAS_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "271148ab-e6dd-4d24-a696-81173f1d4e94",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1373\n",
       "intergenic     873\n",
       "UTR5           113\n",
       "UTR3            93\n",
       "downstream      54\n",
       "upstream        53\n",
       "exonic          47\n",
       "splicing         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "b96f752a-342d-4690-ac42-5b9020f08a6e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "71fc9634-a05c-4241-975c-1101347028ec",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    31\n",
       "synonymous SNV       16\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "cfa2b379-8dd6-4a27-92ef-8eff2084e332",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792657</td>\n",
       "      <td>11792657</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5312;dist=2916</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792657</td>\n",
       "      <td>chr8:11792657:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792660</td>\n",
       "      <td>11792660</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5315;dist=2913</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792660</td>\n",
       "      <td>chr8:11792660:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792690</td>\n",
       "      <td>11792690</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5345;dist=2883</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792690</td>\n",
       "      <td>chr8:11792690:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792717</td>\n",
       "      <td>11792717</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5372;dist=2856</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792717</td>\n",
       "      <td>chr8:11792717:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4076</th>\n",
       "      <td>8</td>\n",
       "      <td>11919319</td>\n",
       "      <td>11919319</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51232;dist=54618</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919319</td>\n",
       "      <td>chr8:11919319:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4077</th>\n",
       "      <td>8</td>\n",
       "      <td>11919359</td>\n",
       "      <td>11919359</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51272;dist=54578</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919359</td>\n",
       "      <td>chr8:11919359:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4078</th>\n",
       "      <td>8</td>\n",
       "      <td>11919433</td>\n",
       "      <td>11919433</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51346;dist=54504</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919433</td>\n",
       "      <td>chr8:11919433:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4079</th>\n",
       "      <td>8</td>\n",
       "      <td>11919510</td>\n",
       "      <td>11919510</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51423;dist=54427</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919510</td>\n",
       "      <td>chr8:11919510:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4080</th>\n",
       "      <td>8</td>\n",
       "      <td>11919532</td>\n",
       "      <td>11919532</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51445;dist=54405</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919532</td>\n",
       "      <td>chr8:11919532:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4081 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792657  11792657   C   T   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792660  11792660   C   G   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792690  11792690   C   G   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792717  11792717   C   T   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "4076    8  11919319  11919319   C   A   intergenic  CTSB;DEFB136   \n",
       "4077    8  11919359  11919359   G   C   intergenic  CTSB;DEFB136   \n",
       "4078    8  11919433  11919433   C   T   intergenic  CTSB;DEFB136   \n",
       "4079    8  11919510  11919510   C   T   intergenic  CTSB;DEFB136   \n",
       "4080    8  11919532  11919532   T   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5312;dist=2916                  .                .   \n",
       "2       dist=5315;dist=2913                  .                .   \n",
       "3       dist=5345;dist=2883                  .                .   \n",
       "4       dist=5372;dist=2856                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "4076  dist=51232;dist=54618                  .                .   \n",
       "4077  dist=51272;dist=54578                  .                .   \n",
       "4078  dist=51346;dist=54504                  .                .   \n",
       "4079  dist=51423;dist=54427                  .                .   \n",
       "4080  dist=51445;dist=54405                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "4076                .          0          .          .           8   \n",
       "4077                .          0          .          .           8   \n",
       "4078                .          0          .          .           8   \n",
       "4079                .          0          .          .           8   \n",
       "4080                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792657  chr8:11792657:C:T          C          T          .   \n",
       "2       11792660  chr8:11792660:C:G          C          G          .   \n",
       "3       11792690  chr8:11792690:C:G          C          G          .   \n",
       "4       11792717  chr8:11792717:C:T          C          T          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "4076    11919319  chr8:11919319:C:A          C          A          .   \n",
       "4077    11919359  chr8:11919359:G:C          G          C          .   \n",
       "4078    11919433  chr8:11919433:C:T          C          T          .   \n",
       "4079    11919510  chr8:11919510:C:T          C          T          .   \n",
       "4080    11919532  chr8:11919532:T:C          T          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "4076           .          PR          GT         0/0  \n",
       "4077           .          PR          GT         0/0  \n",
       "4078           .          PR          GT         0/0  \n",
       "4079           .          PR          GT         0/0  \n",
       "4080           .          PR          GT         0/0  \n",
       "\n",
       "[4081 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_EAS/EAS_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "5bba47cd-9ba5-4c09-b676-15b5572f2fba",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      2164\n",
       "intergenic    1351\n",
       "UTR5           170\n",
       "UTR3           137\n",
       "downstream      93\n",
       "exonic          84\n",
       "upstream        80\n",
       "splicing         2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "714a67a0-3b02-4c63-aaff-c0b30078be89",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "9ab029ee-4344-4bd9-83cc-3e8206ced61e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV      48\n",
       "synonymous SNV         34\n",
       "frameshift deletion     2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "8cb66d83-36d1-4069-8e64-d6950604a152",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792593</td>\n",
       "      <td>11792593</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5248;dist=2980</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792593</td>\n",
       "      <td>chr8:11792593:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792688</td>\n",
       "      <td>11792688</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5343;dist=2885</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792688</td>\n",
       "      <td>chr8:11792688:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>11792691</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5346;dist=2882</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792691</td>\n",
       "      <td>chr8:11792691:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>11792812</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5467;dist=2761</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>chr8:11792812:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2540</th>\n",
       "      <td>8</td>\n",
       "      <td>11919062</td>\n",
       "      <td>11919062</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=50975;dist=54875</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919062</td>\n",
       "      <td>chr8:11919062:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2541</th>\n",
       "      <td>8</td>\n",
       "      <td>11919200</td>\n",
       "      <td>11919200</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51113;dist=54737</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919200</td>\n",
       "      <td>chr8:11919200:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2542</th>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>11919223</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51136;dist=54714</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>chr8:11919223:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2543</th>\n",
       "      <td>8</td>\n",
       "      <td>11919263</td>\n",
       "      <td>11919263</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51176;dist=54674</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919263</td>\n",
       "      <td>chr8:11919263:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2544</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2545 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792593  11792593   T   G   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792688  11792688   G   C   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792691  11792691   C   T   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792812  11792812   C   T   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "2540    8  11919062  11919062   A   G   intergenic  CTSB;DEFB136   \n",
       "2541    8  11919200  11919200   G   A   intergenic  CTSB;DEFB136   \n",
       "2542    8  11919223  11919223   G   A   intergenic  CTSB;DEFB136   \n",
       "2543    8  11919263  11919263   G   T   intergenic  CTSB;DEFB136   \n",
       "2544    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5248;dist=2980                  .                .   \n",
       "2       dist=5343;dist=2885                  .                .   \n",
       "3       dist=5346;dist=2882                  .                .   \n",
       "4       dist=5467;dist=2761                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "2540  dist=50975;dist=54875                  .                .   \n",
       "2541  dist=51113;dist=54737                  .                .   \n",
       "2542  dist=51136;dist=54714                  .                .   \n",
       "2543  dist=51176;dist=54674                  .                .   \n",
       "2544  dist=51363;dist=54487                  .                .   \n",
       "\n",
       "     clinvar_20140902  Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .         0.0          .          .           8   \n",
       "1                   .         0.0          .          .           8   \n",
       "2                   .         0.0          .          .           8   \n",
       "3                   .         0.0          .          .           8   \n",
       "4                   .         0.0          .          .           8   \n",
       "...               ...         ...        ...        ...         ...   \n",
       "2540                .         0.0          .          .           8   \n",
       "2541                .         0.0          .          .           8   \n",
       "2542                .         0.0          .          .           8   \n",
       "2543                .         0.0          .          .           8   \n",
       "2544                .         0.0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792593  chr8:11792593:T:G          T          G          .   \n",
       "2       11792688  chr8:11792688:G:C          G          C          .   \n",
       "3       11792691  chr8:11792691:C:T          C          T          .   \n",
       "4       11792812  chr8:11792812:C:T          C          T          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "2540    11919062  chr8:11919062:A:G          A          G          .   \n",
       "2541    11919200  chr8:11919200:G:A          G          A          .   \n",
       "2542    11919223  chr8:11919223:G:A          G          A          .   \n",
       "2543    11919263  chr8:11919263:G:T          G          T          .   \n",
       "2544    11919450  chr8:11919450:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "2540           .          PR          GT         0/0  \n",
       "2541           .          PR          GT         0/0  \n",
       "2542           .          PR          GT         0/0  \n",
       "2543           .          PR          GT         0/0  \n",
       "2544           .          PR          GT         0/0  \n",
       "\n",
       "[2545 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_MDE/MDE_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "4d707bc7-7191-46d9-a8e2-1f3e4e1b762a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1379\n",
       "intergenic     857\n",
       "UTR5            90\n",
       "UTR3            82\n",
       "downstream      48\n",
       "upstream        46\n",
       "exonic          42\n",
       "splicing         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "c8d785be-b714-49a3-9074-45f82755ce9a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "1320f64a-60ea-4cd3-89e8-b76b3c408abd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    26\n",
       "synonymous SNV       16\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "eba12e2a-bdeb-47b8-afe6-709adac3daac",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792683</td>\n",
       "      <td>11792683</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5338;dist=2890</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792683</td>\n",
       "      <td>chr8:11792683:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792784</td>\n",
       "      <td>11792784</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5439;dist=2789</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792784</td>\n",
       "      <td>chr8:11792784:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792794</td>\n",
       "      <td>11792794</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5449;dist=2779</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792794</td>\n",
       "      <td>chr8:11792794:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>11792812</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5467;dist=2761</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>chr8:11792812:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2441</th>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>11919223</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51136;dist=54714</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919223</td>\n",
       "      <td>chr8:11919223:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2442</th>\n",
       "      <td>8</td>\n",
       "      <td>11919373</td>\n",
       "      <td>11919373</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51286;dist=54564</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919373</td>\n",
       "      <td>chr8:11919373:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2443</th>\n",
       "      <td>8</td>\n",
       "      <td>11919376</td>\n",
       "      <td>11919376</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51289;dist=54561</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919376</td>\n",
       "      <td>chr8:11919376:A:T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2444</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2445</th>\n",
       "      <td>8</td>\n",
       "      <td>11919469</td>\n",
       "      <td>11919469</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51382;dist=54468</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919469</td>\n",
       "      <td>chr8:11919469:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2446 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792683  11792683   C   G   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792784  11792784   G   A   intergenic   NEIL2;FDFT1   \n",
       "3       8  11792794  11792794   T   C   intergenic   NEIL2;FDFT1   \n",
       "4       8  11792812  11792812   C   T   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "2441    8  11919223  11919223   G   A   intergenic  CTSB;DEFB136   \n",
       "2442    8  11919373  11919373   C   T   intergenic  CTSB;DEFB136   \n",
       "2443    8  11919376  11919376   A   T   intergenic  CTSB;DEFB136   \n",
       "2444    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "2445    8  11919469  11919469   T   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5338;dist=2890                  .                .   \n",
       "2       dist=5439;dist=2789                  .                .   \n",
       "3       dist=5449;dist=2779                  .                .   \n",
       "4       dist=5467;dist=2761                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "2441  dist=51136;dist=54714                  .                .   \n",
       "2442  dist=51286;dist=54564                  .                .   \n",
       "2443  dist=51289;dist=54561                  .                .   \n",
       "2444  dist=51363;dist=54487                  .                .   \n",
       "2445  dist=51382;dist=54468                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .        0.5          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "2441                .          0          .          .           8   \n",
       "2442                .          0          .          .           8   \n",
       "2443                .          0          .          .           8   \n",
       "2444                .          0          .          .           8   \n",
       "2445                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792683  chr8:11792683:C:G          C          G          .   \n",
       "2       11792784  chr8:11792784:G:A          G          A          .   \n",
       "3       11792794  chr8:11792794:T:C          T          C          .   \n",
       "4       11792812  chr8:11792812:C:T          C          T          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "2441    11919223  chr8:11919223:G:A          G          A          .   \n",
       "2442    11919373  chr8:11919373:C:T          C          T          .   \n",
       "2443    11919376  chr8:11919376:A:T          A          T          .   \n",
       "2444    11919450  chr8:11919450:G:C          G          C          .   \n",
       "2445    11919469  chr8:11919469:T:C          T          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/1  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "2441           .          PR          GT         0/0  \n",
       "2442           .          PR          GT         0/0  \n",
       "2443           .          PR          GT         0/0  \n",
       "2444           .          PR          GT         0/0  \n",
       "2445           .          PR          GT         0/0  \n",
       "\n",
       "[2446 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "94b1c19e-d264-43ed-9924-83fbc24eb841",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1254\n",
       "intergenic     871\n",
       "UTR5            94\n",
       "UTR3            86\n",
       "upstream        55\n",
       "exonic          47\n",
       "downstream      39\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "7af2484a-b5f9-4f4a-ac42-a8c305d9564d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "9cd41f64-d5e3-4a8a-b8b7-0bdf1ca0498a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         31\n",
       "synonymous SNV            15\n",
       "nonframeshift deletion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "2f4c72e5-3db5-4b1d-b528-0d938a04962a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>11792812</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5467;dist=2761</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>chr8:11792812:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792966</td>\n",
       "      <td>11792966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5621;dist=2607</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792966</td>\n",
       "      <td>chr8:11792966:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11793099</td>\n",
       "      <td>11793099</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5754;dist=2474</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11793099</td>\n",
       "      <td>chr8:11793099:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11793200</td>\n",
       "      <td>11793200</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5855;dist=2373</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11793200</td>\n",
       "      <td>chr8:11793200:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1002</th>\n",
       "      <td>8</td>\n",
       "      <td>11918551</td>\n",
       "      <td>11918551</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=50464;dist=55386</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11918551</td>\n",
       "      <td>chr8:11918551:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>8</td>\n",
       "      <td>11918618</td>\n",
       "      <td>11918618</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=50531;dist=55319</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11918618</td>\n",
       "      <td>chr8:11918618:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1004</th>\n",
       "      <td>8</td>\n",
       "      <td>11919200</td>\n",
       "      <td>11919200</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51113;dist=54737</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919200</td>\n",
       "      <td>chr8:11919200:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1005</th>\n",
       "      <td>8</td>\n",
       "      <td>11919263</td>\n",
       "      <td>11919263</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51176;dist=54674</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919263</td>\n",
       "      <td>chr8:11919263:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1006</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1007 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "1       8  11792812  11792812   C   T   intergenic   NEIL2;FDFT1   \n",
       "2       8  11792966  11792966   T   C   intergenic   NEIL2;FDFT1   \n",
       "3       8  11793099  11793099   T   C   intergenic   NEIL2;FDFT1   \n",
       "4       8  11793200  11793200   G   C   intergenic   NEIL2;FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "1002    8  11918551  11918551   G   T   intergenic  CTSB;DEFB136   \n",
       "1003    8  11918618  11918618   T   C   intergenic  CTSB;DEFB136   \n",
       "1004    8  11919200  11919200   G   A   intergenic  CTSB;DEFB136   \n",
       "1005    8  11919263  11919263   G   T   intergenic  CTSB;DEFB136   \n",
       "1006    8  11919450  11919450   G   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5467;dist=2761                  .                .   \n",
       "2       dist=5621;dist=2607                  .                .   \n",
       "3       dist=5754;dist=2474                  .                .   \n",
       "4       dist=5855;dist=2373                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "1002  dist=50464;dist=55386                  .                .   \n",
       "1003  dist=50531;dist=55319                  .                .   \n",
       "1004  dist=51113;dist=54737                  .                .   \n",
       "1005  dist=51176;dist=54674                  .                .   \n",
       "1006  dist=51363;dist=54487                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .          0          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "1002                .          1          .          .           8   \n",
       "1003                .          1          .          .           8   \n",
       "1004                .          0          .          .           8   \n",
       "1005                .          0          .          .           8   \n",
       "1006                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792812  chr8:11792812:C:T          C          T          .   \n",
       "2       11792966  chr8:11792966:T:C          T          C          .   \n",
       "3       11793099  chr8:11793099:T:C          T          C          .   \n",
       "4       11793200  chr8:11793200:G:C          G          C          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "1002    11918551  chr8:11918551:G:T          G          T          .   \n",
       "1003    11918618  chr8:11918618:T:C          T          C          .   \n",
       "1004    11919200  chr8:11919200:G:A          G          A          .   \n",
       "1005    11919263  chr8:11919263:G:T          G          T          .   \n",
       "1006    11919450  chr8:11919450:G:C          G          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "1002           .          PR          GT         1/1  \n",
       "1003           .          PR          GT         1/1  \n",
       "1004           .          PR          GT         0/0  \n",
       "1005           .          PR          GT         0/0  \n",
       "1006           .          PR          GT         0/0  \n",
       "\n",
       "[1007 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_FIN/FIN_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "01c0b445-942a-40bd-873b-69b1a1923067",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      524\n",
       "intergenic    358\n",
       "UTR5           38\n",
       "UTR3           37\n",
       "upstream       18\n",
       "downstream     18\n",
       "exonic         14\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "08a4298d-70af-4dde-96a1-8f7add25df9f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "d832fddc-2eed-4425-87e6-5b334f98eba6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    8\n",
       "synonymous SNV       6\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "7f9bdc77-c409-44b5-8180-b90a89e3381c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792554</td>\n",
       "      <td>11792554</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5209;dist=3019</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792554</td>\n",
       "      <td>chr8:11792554:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792593</td>\n",
       "      <td>11792593</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5248;dist=2980</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792593</td>\n",
       "      <td>chr8:11792593:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>./.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>11792594</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5249;dist=2979</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792594</td>\n",
       "      <td>chr8:11792594:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>./.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792658</td>\n",
       "      <td>11792658</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2;FDFT1</td>\n",
       "      <td>dist=5313;dist=2915</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792658</td>\n",
       "      <td>chr8:11792658:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13476</th>\n",
       "      <td>8</td>\n",
       "      <td>11919475</td>\n",
       "      <td>11919475</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51388;dist=54462</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919475</td>\n",
       "      <td>chr8:11919475:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>./.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13477</th>\n",
       "      <td>8</td>\n",
       "      <td>11919503</td>\n",
       "      <td>11919503</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51416;dist=54434</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919503</td>\n",
       "      <td>chr8:11919503:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13478</th>\n",
       "      <td>8</td>\n",
       "      <td>11919510</td>\n",
       "      <td>11919510</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51423;dist=54427</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919510</td>\n",
       "      <td>chr8:11919510:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13479</th>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>11919512</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51425;dist=54425</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919512</td>\n",
       "      <td>chr8:11919512:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13480</th>\n",
       "      <td>8</td>\n",
       "      <td>11919513</td>\n",
       "      <td>11919513</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB;DEFB136</td>\n",
       "      <td>dist=51426;dist=54424</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919513</td>\n",
       "      <td>chr8:11919513:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13481 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0        8  11792554  11792554   C   T   intergenic   NEIL2;FDFT1   \n",
       "1        8  11792575  11792575   T   C   intergenic   NEIL2;FDFT1   \n",
       "2        8  11792593  11792593   T   G   intergenic   NEIL2;FDFT1   \n",
       "3        8  11792594  11792594   C   G   intergenic   NEIL2;FDFT1   \n",
       "4        8  11792658  11792658   C   G   intergenic   NEIL2;FDFT1   \n",
       "...    ...       ...       ...  ..  ..          ...           ...   \n",
       "13476    8  11919475  11919475   T   C   intergenic  CTSB;DEFB136   \n",
       "13477    8  11919503  11919503   T   C   intergenic  CTSB;DEFB136   \n",
       "13478    8  11919510  11919510   C   T   intergenic  CTSB;DEFB136   \n",
       "13479    8  11919512  11919512   T   C   intergenic  CTSB;DEFB136   \n",
       "13480    8  11919513  11919513   G   C   intergenic  CTSB;DEFB136   \n",
       "\n",
       "          GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0        dist=5209;dist=3019                  .                .   \n",
       "1        dist=5230;dist=2998                  .                .   \n",
       "2        dist=5248;dist=2980                  .                .   \n",
       "3        dist=5249;dist=2979                  .                .   \n",
       "4        dist=5313;dist=2915                  .                .   \n",
       "...                      ...                ...              ...   \n",
       "13476  dist=51388;dist=54462                  .                .   \n",
       "13477  dist=51416;dist=54434                  .                .   \n",
       "13478  dist=51423;dist=54427                  .                .   \n",
       "13479  dist=51425;dist=54425                  .                .   \n",
       "13480  dist=51426;dist=54424                  .                .   \n",
       "\n",
       "      clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                    .          0          .          .           8   \n",
       "1                    .          0          .          .           8   \n",
       "2                    .          .          .          .           8   \n",
       "3                    .          .          .          .           8   \n",
       "4                    .          0          .          .           8   \n",
       "...                ...        ...        ...        ...         ...   \n",
       "13476                .          .          .          .           8   \n",
       "13477                .          0          .          .           8   \n",
       "13478                .          0          .          .           8   \n",
       "13479                .          0          .          .           8   \n",
       "13480                .          0          .          .           8   \n",
       "\n",
       "       Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0        11792554  chr8:11792554:C:T          C          T          .   \n",
       "1        11792575  chr8:11792575:T:C          T          C          .   \n",
       "2        11792593  chr8:11792593:T:G          T          G          .   \n",
       "3        11792594  chr8:11792594:C:G          C          G          .   \n",
       "4        11792658  chr8:11792658:C:G          C          G          .   \n",
       "...           ...                ...        ...        ...        ...   \n",
       "13476    11919475  chr8:11919475:T:C          T          C          .   \n",
       "13477    11919503  chr8:11919503:T:C          T          C          .   \n",
       "13478    11919510  chr8:11919510:C:T          C          T          .   \n",
       "13479    11919512  chr8:11919512:T:C          T          C          .   \n",
       "13480    11919513  chr8:11919513:G:C          G          C          .   \n",
       "\n",
       "      Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0               .          PR          GT         0/0  \n",
       "1               .          PR          GT         0/0  \n",
       "2               .          PR          GT         ./.  \n",
       "3               .          PR          GT         ./.  \n",
       "4               .          PR          GT         0/0  \n",
       "...           ...         ...         ...         ...  \n",
       "13476           .          PR          GT         ./.  \n",
       "13477           .          PR          GT         0/0  \n",
       "13478           .          PR          GT         0/0  \n",
       "13479           .          PR          GT         0/0  \n",
       "13480           .          PR          GT         0/0  \n",
       "\n",
       "[13481 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_EUR/EUR_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "f420b64a-2c2a-475e-8978-766d0264c946",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      7065\n",
       "intergenic    4618\n",
       "UTR5           537\n",
       "UTR3           455\n",
       "upstream       270\n",
       "downstream     269\n",
       "exonic         264\n",
       "splicing         3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "93ec3ae9-7d1b-4073-a180-08aacab6327b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "0b8ffa97-18ba-4f66-a34c-c1aea4a85309",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         164\n",
       "synonymous SNV             91\n",
       "nonframeshift deletion      6\n",
       "frameshift deletion         2\n",
       "frameshift insertion        1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b34e1a8-85d2-4ad8-a781-281b1d7ead4e",
   "metadata": {},
   "source": [
    "ASSOC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "8dc7b177-c7e1-4039-8862-0de250a9b019",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "5351 variants loaded from .bim file.\n",
      "1111 people (455 males, 656 females) loaded from .fam.\n",
      "1086 phenotype values loaded from .fam.\n",
      "--keep: 1111 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997295.\n",
      "4134 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1217 variants and 1111 people pass filters and QC.\n",
      "Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all.assoc ...\n",
      "84done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "6381 variants loaded from .bim file.\n",
      "2643 people (1480 males, 1163 females) loaded from .fam.\n",
      "2621 phenotype values loaded from .fam.\n",
      "--keep: 2643 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2643 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.994838.\n",
      "5149 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1232 variants and 2643 people pass filters and QC.\n",
      "Among remaining phenotypes, 942 are cases and 1679 are controls.  (22\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all.assoc ...\n",
      "83done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "2136 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "--keep: 2655 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.994732.\n",
      "1393 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "743 variants and 2655 people pass filters and QC.\n",
      "Among remaining phenotypes, 1292 are cases and 411 are controls.  (952\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "3038 variants loaded from .bim file.\n",
      "646 people (363 males, 283 females) loaded from .fam.\n",
      "613 phenotype values loaded from .fam.\n",
      "--keep: 646 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.993416.\n",
      "2319 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "719 variants and 646 people pass filters and QC.\n",
      "Among remaining phenotypes, 458 are cases and 155 are controls.  (33 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "2607 variants loaded from .bim file.\n",
      "903 people (439 males, 464 females) loaded from .fam.\n",
      "895 phenotype values loaded from .fam.\n",
      "--keep: 903 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 903 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.987931.\n",
      "1904 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "703 variants and 903 people pass filters and QC.\n",
      "Among remaining phenotypes, 552 are cases and 343 are controls.  (8 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "4081 variants loaded from .bim file.\n",
      "5167 people (3278 males, 1889 females) loaded from .fam.\n",
      "5123 phenotype values loaded from .fam.\n",
      "--keep: 5167 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5167 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.986103.\n",
      "3593 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "488 variants and 5167 people pass filters and QC.\n",
      "Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "13481 variants loaded from .bim file.\n",
      "38839 people (23296 males, 15543 females) loaded from .fam.\n",
      "30412 phenotype values loaded from .fam.\n",
      "--keep: 38838 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 38838 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.753414.\n",
      "12797 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "684 variants and 38838 people pass filters and QC.\n",
      "Among remaining phenotypes, 21197 are cases and 9214 are controls.  (8427\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "1007 variants loaded from .bim file.\n",
      "114 people (53 males, 61 females) loaded from .fam.\n",
      "106 phenotype values loaded from .fam.\n",
      "--keep: 114 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 114 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.991768.\n",
      "305 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "702 variants and 114 people pass filters and QC.\n",
      "Among remaining phenotypes, 98 are cases and 8 are controls.  (8 phenotypes are\n",
      "missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "2545 variants loaded from .bim file.\n",
      "581 people (369 males, 212 females) loaded from .fam.\n",
      "536 phenotype values loaded from .fam.\n",
      "--keep: 581 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.990923.\n",
      "1856 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "689 variants and 581 people pass filters and QC.\n",
      "Among remaining phenotypes, 311 are cases and 225 are controls.  (45 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "2446 variants loaded from .bim file.\n",
      "635 people (408 males, 227 females) loaded from .fam.\n",
      "609 phenotype values loaded from .fam.\n",
      "--keep: 635 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 635 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.992566.\n",
      "1730 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "716 variants and 635 people pass filters and QC.\n",
      "Among remaining phenotypes, 387 are cases and 222 are controls.  (26 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all\n",
      "\n",
      "12984 MB RAM detected; reserving 6492 MB for main workspace.\n",
      "4946 variants loaded from .bim file.\n",
      "851 people (446 males, 405 females) loaded from .fam.\n",
      "827 phenotype values loaded from .fam.\n",
      "--keep: 851 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 851 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.995169.\n",
      "3966 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "980 variants and 851 people pass filters and QC.\n",
      "Among remaining phenotypes, 525 are cases and 302 are controls.  (24 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis using plink assoc for all variants, not adjusting for any covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ! /home/jupyter/tools/plink \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "    --assoc \\\n",
    "    --allow-no-sex \\\n",
    "    --ci 0.95 \\\n",
    "    --maf 0.01 \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "dc946e1d-d59c-4b9e-a5b3-f412940e6ac4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "Variants with p-value < 0.05: (142, 13)\n",
      "WORKING ON: AFR\n",
      "Variants with p-value < 0.05: (101, 13)\n",
      "WORKING ON: AJ\n",
      "Variants with p-value < 0.05: (6, 13)\n",
      "WORKING ON: AMR\n",
      "Variants with p-value < 0.05: (200, 13)\n",
      "WORKING ON: CAS\n",
      "Variants with p-value < 0.05: (167, 13)\n",
      "WORKING ON: EAS\n",
      "Variants with p-value < 0.05: (167, 13)\n",
      "WORKING ON: EUR\n",
      "Variants with p-value < 0.05: (208, 13)\n",
      "WORKING ON: FIN\n",
      "Variants with p-value < 0.05: (25, 13)\n",
      "WORKING ON: MDE\n",
      "Variants with p-value < 0.05: (80, 13)\n",
      "WORKING ON: SAS\n",
      "Variants with p-value < 0.05: (26, 13)\n",
      "WORKING ON: CAH\n",
      "Variants with p-value < 0.05: (53, 13)\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Look at assoc results, only variants with nominal p-value < 0.05\n",
    "    freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_CTSB.all.assoc', sep='\\s+')\n",
    "    sig_all_nonadj = freq[freq['P']<0.05]\n",
    "    \n",
    "    print(f'Variants with p-value < 0.05: {sig_all_nonadj.shape}')\n",
    "    \n",
    "    #Save FREQ to csv\n",
    "    freq.to_csv(f'{WORK_DIR}/{ancestry}.all_nonadj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c75b6185-12c0-42b8-a73f-884bf6a53195",
   "metadata": {},
   "source": [
    "GLM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "b938baa2-7bc3-4d8f-9fe5-c6c672693732",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:30:51 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11226 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "1111 samples (656 females, 455 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam.\n",
      "5351 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim.\n",
      "1 binary phenotype loaded (285 cases, 801 controls).\n",
      "--keep: 1111 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_covariate_file.txt.\n",
      "1111 samples (656 females, 455 males; 1111 founders) remaining after main\n",
      "filters.\n",
      "285 cases and 801 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "4134 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1217 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:30:52 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:30:53 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11236 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam.\n",
      "6381 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim.\n",
      "1 binary phenotype loaded (942 cases, 1679 controls).\n",
      "--keep: 2643 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_covariate_file.txt.\n",
      "2643 samples (1163 females, 1480 males; 2643 founders) remaining after main\n",
      "filters.\n",
      "942 cases and 1679 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "5149 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1232 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:30:53 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:30:54 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11226 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam.\n",
      "2136 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim.\n",
      "1 binary phenotype loaded (1292 cases, 411 controls).\n",
      "--keep: 2655 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_covariate_file.txt.\n",
      "2655 samples (1007 females, 1648 males; 2655 founders) remaining after main\n",
      "filters.\n",
      "1292 cases and 411 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1393 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "743 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:30:54 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:30:55 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11225 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "646 samples (283 females, 363 males; 646 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam.\n",
      "3038 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim.\n",
      "1 binary phenotype loaded (458 cases, 155 controls).\n",
      "--keep: 646 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_covariate_file.txt.\n",
      "646 samples (283 females, 363 males; 646 founders) remaining after main\n",
      "filters.\n",
      "458 cases and 155 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2319 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "719 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:30:55 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:30:56 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11224 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "903 samples (464 females, 439 males; 903 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam.\n",
      "2607 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim.\n",
      "1 binary phenotype loaded (552 cases, 343 controls).\n",
      "--keep: 903 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_covariate_file.txt.\n",
      "903 samples (464 females, 439 males; 903 founders) remaining after main\n",
      "filters.\n",
      "552 cases and 343 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1904 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "703 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:30:56 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:30:56 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11225 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam.\n",
      "4081 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim.\n",
      "1 binary phenotype loaded (2662 cases, 2461 controls).\n",
      "--keep: 5167 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_covariate_file.txt.\n",
      "5167 samples (1889 females, 3278 males; 5167 founders) remaining after main\n",
      "filters.\n",
      "2662 cases and 2461 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3593 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "488 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:30:57 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:30:57 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11223 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "38839 samples (15543 females, 23296 males; 38839 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam.\n",
      "13481 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim.\n",
      "1 binary phenotype loaded (21198 cases, 9214 controls).\n",
      "--keep: 38838 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_covariate_file.txt.\n",
      "38838 samples (15543 females, 23295 males; 38838 founders) remaining after main\n",
      "filters.\n",
      "21197 cases and 9214 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "12797 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "684 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:31:02 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:31:02 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11210 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "114 samples (61 females, 53 males; 114 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam.\n",
      "1007 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim.\n",
      "1 binary phenotype loaded (98 cases, 8 controls).\n",
      "--keep: 114 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_covariate_file.txt.\n",
      "114 samples (61 females, 53 males; 114 founders) remaining after main filters.\n",
      "98 cases and 8 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "305 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "702 variants remaining after main filters.\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\n",
      "phenotype 'PHENO1'.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:31:03 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:31:03 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11218 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "581 samples (212 females, 369 males; 581 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam.\n",
      "2545 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim.\n",
      "1 binary phenotype loaded (311 cases, 225 controls).\n",
      "--keep: 581 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_covariate_file.txt.\n",
      "581 samples (212 females, 369 males; 581 founders) remaining after main\n",
      "filters.\n",
      "311 cases and 225 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1856 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "689 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:31:04 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:31:04 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11195 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "635 samples (227 females, 408 males; 635 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam.\n",
      "2446 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim.\n",
      "1 binary phenotype loaded (387 cases, 222 controls).\n",
      "--keep: 635 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_covariate_file.txt.\n",
      "635 samples (227 females, 408 males; 635 founders) remaining after main\n",
      "filters.\n",
      "387 cases and 222 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1730 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "716 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:31:05 2025\n",
      "PLINK v2.0.0-a.6.0LM 64-bit Intel (11 Nov 2024)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all_adj\n",
      "\n",
      "Start time: Thu Jun 12 11:31:05 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "12984 MiB RAM detected, ~11203 available; reserving 6492 MiB for main\n",
      "workspace.\n",
      "Using up to 2 compute threads.\n",
      "851 samples (405 females, 446 males; 851 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam.\n",
      "4946 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim.\n",
      "1 binary phenotype loaded (525 cases, 302 controls).\n",
      "--keep: 851 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_covariate_file.txt.\n",
      "851 samples (405 females, 446 males; 851 founders) remaining after main\n",
      "filters.\n",
      "525 cases and 302 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3966 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "980 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Thu Jun 12 11:31:06 2025\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis with covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --keep {WORK_DIR}/{ancestry}.samplestoKeep \\\n",
    "    --allow-no-sex \\\n",
    "    --maf 0.01 \\\n",
    "    --ci 0.95 \\\n",
    "    --glm \\\n",
    "    --covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --covar-variance-standardize \\\n",
    "    --neg9-pheno-really-missing \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "94b1aba0-e776-45e7-b220-b424fa338692",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "There are 42 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AFR\n",
      "There are 126 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "There are 21 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AMR\n",
      "There are 160 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: CAS\n",
      "There are 44 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EAS\n",
      "There are 9 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EUR\n",
      "There are 189 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: FIN\n",
      "There are 6 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: MDE\n",
      "There are 32 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: SAS\n",
      "There are 26 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: CAH\n",
      "There are 75 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/tmp/ipykernel_40616/973086827.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    }
   ],
   "source": [
    "#Process results from plink glm analysis including covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Read in plink glm results\n",
    "    assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_CTSB.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "    \n",
    "    #Filter for additive test only - this is the variant results\n",
    "    assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    significant = assoc_add[assoc_add['P']<0.05]\n",
    "\n",
    "    print(f'There are {len(significant)} variants with p-value < 0.05 in glm')\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    GWsignificant = assoc_add[assoc_add['P']<5e-8]\n",
    "\n",
    "    print(f'There are {len(GWsignificant)} variants with p-value < 5e-8 in glm')\n",
    "    \n",
    "    #Save assoc_add to csv\n",
    "    assoc_add.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a7391e4-3100-4c01-84ee-f49f2d24fd93",
   "metadata": {},
   "source": [
    "change format of association file in EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "7db1998d-5a9c-4bd6-a62a-dbbdd8710f16",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/CTSB/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "2ebf89e6-ff5a-4402-8151-41ad823aca0a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "WORKING ON: MDE\n",
      "WORKING ON: EAS\n",
      "WORKING ON: AMR\n",
      "WORKING ON: CAS\n",
      "WORKING ON: FIN\n",
      "WORKING ON: CAH\n",
      "WORKING ON: SAS\n",
      "WORKING ON: AFR\n",
      "WORKING ON: EUR\n",
      "WORKING ON: AAC\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = \"~/workspace/ws_files/CTSB/\"\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    df = pd.read_csv(f'{WORK_DIR}/CTSB_{ancestry}/{ancestry}.all_adj.csv')\n",
    "    \n",
    "    #First just select relevant columns: chromosome, bp position and p-value\n",
    "    export_ldassoc = df[['#CHROM', 'POS', 'P']].copy()\n",
    "    \n",
    "    #Rename the #CHROM column to remove the hashtag as I think this might be confusing LDassoc\n",
    "    export_ldassoc = export_ldassoc.rename(columns={'#CHROM': 'CHROM'}) \n",
    "    \n",
    "    #Then export as a tab-separated, not comma-separated file\n",
    "\n",
    "    export_ldassoc.to_csv(f'{WORK_DIR}/CTSB_{ancestry}/{ancestry}.all_adj.formatted.tab', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb0ce11d-4f7d-4a07-82dd-ad2218e581d3",
   "metadata": {},
   "source": [
    "GCTA conditional analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "53e06cd4-cafe-4039-8222-be04265a8c1b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# prepare file for COJO\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    #Read in summary statistics\n",
    "    sumstats = pd.read_csv(f'{WORK_DIR}/CTSB_{ancestry}/{ancestry}.all_adj.csv')\n",
    "    \n",
    "    #Format summary statistics for GCTA-COJO\n",
    "    #First get the log odds ratio - this is required for COJO\n",
    "    #1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.\n",
    "    sumstats_formatted = sumstats.copy()\n",
    "    sumstats_formatted['b'] = np.log(sumstats_formatted['OR'])\n",
    "    \n",
    "    #Now select just the necessary columns for COJO\n",
    "    sumstats_export = sumstats_formatted[['ID', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P', 'OBS_CT']].copy()\n",
    "    \n",
    "    #Rename columns following COJO format\n",
    "    sumstats_export = sumstats_export.rename(columns = {'ID':'SNP', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p', 'OBS_CT':'N'})\n",
    "    \n",
    "    #Export\n",
    "    sumstats_export.to_csv(f'{WORK_DIR}/CTSB_{ancestry}/{ancestry}.all_adj.sumstats.ma', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95968404-5d95-4eee-b2fb-2c9c5a3db062",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on significance p-value 5e-08\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 5e-8 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4becfdec-9490-4d34-a5af-193d17ff5a26",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on significance p-value 5e-08\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 4.27e-4 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.LDprune.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c05a8022-cdbd-4351-881c-26f98d6df7f9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on significance p-value 5e-08\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.94.1-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-top-SNPs 10 \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.top10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9c7b607-880e-4b95-b646-e4f4066f18fd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "python3",
   "name": ".m125",
   "type": "gcloud",
   "uri": "us-docker.pkg.dev/deeplearning-platform-release/gcr.io/:m125"
  },
  "kernelspec": {
   "display_name": "Python 3 (Local)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
