{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b4eea433-e830-4bfb-9e01-c7e0052bed5a",
   "metadata": {},
   "source": [
    "CTSB - Single gene analysis in GP2 Neurobooster genotyping data (all ancestries)\n",
    "\n",
    "Project: GP2 CTSB\n",
    "\n",
    "Version: Python/3.10.15, R/4.3.3\n",
    "\n",
    "Notebook Overview\n",
    "\n",
    "Description Loading Python libraries Set paths Make working directory\n",
    "\n",
    "Installing packages\n",
    "\n",
    "Create a covariate file with GP2 data\n",
    "\n",
    "Annotation of the gene CTSB\n",
    "\n",
    "Association analysis to compare allele frequencies between cases and controls\n",
    "\n",
    "GLM analysis adjusting for gender, age, PC1-5\n",
    "\n",
    "Burden test(SkatO, Skat, cmc,zeggini,mb,fp,cmcWald)\n",
    "\n",
    "Conditional analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "820c976e-11e1-409c-97b0-59170e933e5a",
   "metadata": {},
   "source": [
    "Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "916aeaad-1752-4784-b28b-2372bae0485b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use pathlib for file path manipulation\n",
    "import pathlib\n",
    "\n",
    "# Install numpy\n",
    "import numpy as np\n",
    "\n",
    "# Install Pandas for tabular data\n",
    "import pandas as pd\n",
    "\n",
    "# Install plotnine: a ggplot2-compatible Python plotting package\n",
    "from plotnine import *\n",
    "\n",
    "# Always show all columns in a Pandas DataFrame\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18da7de1-c922-4dfb-8dac-f30151e6f405",
   "metadata": {},
   "source": [
    "Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0c580c59-580a-4176-96e3-c82d4c938fbe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 100K\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 clinical_data\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 imputed_genotypes\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 meta_data\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 raw_genotypes\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 raw_genotypes_flipped\n",
      "-r--r--r--. 1 jupyter users 100K Jun 30 20:07 README_release10_01072025.txt\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 wgs\n"
     ]
    }
   ],
   "source": [
    "REL10_PATH = pathlib.Path(pathlib.Path.home(), 'workspace/gp2_tier2_eu_release10')\n",
    "!ls -hal {REL10_PATH}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "625f6ba3-e662-473c-be53-0a466c94e189",
   "metadata": {},
   "source": [
    "Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f814e125-ecc2-41ca-83a5-5265a6e012ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir ~/workspace/ws_files/CTSB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "dcf5b9ac-5c93-4071-9912-7f902eb84daa",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/CTSB/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f0c74b02-8841-4a5c-b727-d93cb47288f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\t\t\t\t       plink2\n",
      "annovar.latest.tar.gz\t\t       plink2_linux_x86_64_latest.zip\n",
      "gcta-1.95.0-linux-kernel-3-x86_64      plink_linux_x86_64_20190304.zip\n",
      "gcta-1.95.0-linux-kernel-3-x86_64.zip  prettify\n",
      "intel-simplified-software-license.txt  rvtests\n",
      "LICENSE\t\t\t\t       toy.map\n",
      "__MACOSX\t\t\t       toy.ped\n",
      "plink\t\t\t\t       vcf_subset\n"
     ]
    }
   ],
   "source": [
    "# make sure all tools installed\n",
    "! ls /home/jupyter/tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "306ff595-aef3-41d6-8e84-a429dcae6d5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# give permission\n",
    "\n",
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5b12f67a-4bb6-4e15-b866-81bea0daf14b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# making working directory\n",
    "#Loop over all the ancestries\n",
    "for ancestry in {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'} ;\n",
    "do\n",
    "\n",
    "#Make a folder for each ancestry\n",
    "mkdir ~/workspace/ws_files/CTSB/CTSB_\"$ancestry\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "22efab4b-efe9-4a4f-894b-7ad39688c8cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# covariate file has been created in another notebook without GBA carriers"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b7daa32-9143-421b-ac12-72891359f4b5",
   "metadata": {},
   "source": [
    "Annotation of the gene\n",
    "\n",
    "Extract the region using PLINK\n",
    "\n",
    "Extract CTSB gene\n",
    "\n",
    "CTSB coordinates: Chromosome 8: 11,842,524-11,869,533(GRCh38/hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a970c52d-2700-46e4-b1d0-c40d89c87b7d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:49:32 2025\n",
      "26046 MiB RAM detected, ~24671 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb.psam.\n",
      "715411 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "1057 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim ... \n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bed ... done.\n",
      "End time: Sat Jul 26 12:49:33 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:49:34 2025\n",
      "26046 MiB RAM detected, ~24665 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb.psam.\n",
      "3550811 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "5507 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bed ... 0%\n",
      "End time: Sat Jul 26 12:49:38 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:49:39 2025\n",
      "26046 MiB RAM detected, ~24635 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb.psam.\n",
      "2403439 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "3639 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bed ... 0%\n",
      "End time: Sat Jul 26 12:49:42 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:49:43 2025\n",
      "26046 MiB RAM detected, ~24633 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb.psam.\n",
      "3293679 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "5526 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim ... \n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bed ... done.\n",
      "End time: Sat Jul 26 12:49:47 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:49:48 2025\n",
      "26046 MiB RAM detected, ~24626 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb.psam.\n",
      "6200464 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "10388 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bed ... 0%\n",
      "End time: Sat Jul 26 12:49:54 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:49:55 2025\n",
      "26046 MiB RAM detected, ~24617 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb.psam.\n",
      "3231362 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "5508 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bed ... 0%\n",
      "End time: Sat Jul 26 12:49:58 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:49:59 2025\n",
      "26046 MiB RAM detected, ~24626 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb.psam.\n",
      "2184117 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "3249 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim ... \n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bed ... done.\n",
      "End time: Sat Jul 26 12:50:01 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:50:02 2025\n",
      "26046 MiB RAM detected, ~24629 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb.psam.\n",
      "1376363 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "2256 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim ... \n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bed ... done.\n",
      "End time: Sat Jul 26 12:50:04 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:50:05 2025\n",
      "26046 MiB RAM detected, ~24630 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb.psam.\n",
      "1753806 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "2801 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim ... \n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bed ... done.\n",
      "End time: Sat Jul 26 12:50:07 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:50:08 2025\n",
      "26046 MiB RAM detected, ~24629 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb.psam.\n",
      "2703578 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "4213 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim ... \n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bed ... done.\n",
      "End time: Sat Jul 26 12:50:11 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.log.\n",
      "Options in effect:\n",
      "  --chr 8\n",
      "  --from-bp 11792524\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb\n",
      "  --to-bp 11919533\n",
      "\n",
      "Start time: Sat Jul 26 12:50:12 2025\n",
      "26046 MiB RAM detected, ~24614 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb.psam.\n",
      "4081203 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "7164 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bed ... 0%\n",
      "End time: Sat Jul 26 12:50:15 2025\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr8_{ancestry}_release10_vwb \\\n",
    "    --chr 8 \\\n",
    "    --from-bp 11792524 \\\n",
    "    --to-bp 11919533 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "55a8afb5-f986-4901-8ae3-37f9b83d326d",
   "metadata": {},
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}/'\n",
    "    \n",
    "    ! head -n 1 {WORK_DIR}/{ancestry}_CTSB.fam > {WORK_DIR}/{ancestry}_s1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70461073-1497-469f-b079-1365cbec6797",
   "metadata": {},
   "outputs": [],
   "source": [
    "! head /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_s1.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00775fb2-fead-4c16-b53e-7dbb24d3dccb",
   "metadata": {},
   "source": [
    "Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0b93fc9b-860b-438d-8d75-fd901597673e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:51:59 2025\n",
      "26046 MiB RAM detected, ~24625 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam.\n",
      "1057 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:51:59 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:00 2025\n",
      "26046 MiB RAM detected, ~24609 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam.\n",
      "5507 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:01 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:02 2025\n",
      "26046 MiB RAM detected, ~24623 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam.\n",
      "3639 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (1 female, 0 males; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:03 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:04 2025\n",
      "26046 MiB RAM detected, ~24614 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam.\n",
      "5526 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (1 female, 0 males; 1 founder) remaining after main filters.\n",
      "0 cases and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:05 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:06 2025\n",
      "26046 MiB RAM detected, ~24631 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam.\n",
      "10388 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "0 cases and 1 control remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:06 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:07 2025\n",
      "26046 MiB RAM detected, ~24639 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam.\n",
      "5508 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:08 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:09 2025\n",
      "26046 MiB RAM detected, ~24625 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam.\n",
      "3249 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:10 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:11 2025\n",
      "26046 MiB RAM detected, ~24622 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam.\n",
      "2256 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.bed ... 0%\n",
      "End time: Sat Jul 26 12:52:12 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:13 2025\n",
      "26046 MiB RAM detected, ~24614 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam.\n",
      "2801 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:13 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:14 2025\n",
      "26046 MiB RAM detected, ~24609 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam.\n",
      "4213 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:15 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --keep /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:16 2025\n",
      "26046 MiB RAM detected, ~24633 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam.\n",
      "7164 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:17 2025\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --keep {WORK_DIR}/{ancestry}_s1.txt \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3c499c2e-2a2d-481b-a696-2ea22ca03831",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:42 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24629 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.fam.\n",
      "1057 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:43 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:44 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24618 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.fam.\n",
      "5507 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:44 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:45 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24635 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (1 female, 0 males; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.fam.\n",
      "3639 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:46 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:47 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24623 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (1 female, 0 males; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.fam.\n",
      "5526 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.bim.\n",
      "Note: No phenotype data present.\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:47 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:48 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24617 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.fam.\n",
      "10388 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (0 cases, 1 control).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 12:52:48 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:50 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24640 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.fam.\n",
      "5508 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:50 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:51 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24632 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.fam.\n",
      "3249 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:51 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:52 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24627 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.fam.\n",
      "2256 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:53 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:54 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24626 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.fam.\n",
      "2801 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:54 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:55 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24625 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.fam.\n",
      "4213 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:56 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1\n",
      "\n",
      "Start time: Sat Jul 26 12:52:57 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24621 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.fam.\n",
      "7164 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 12:52:57 2025\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB_v1 \\\n",
    "    --recode vcf-fid \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "40aeaf37-cc5a-49dd-afb4-7cc9010c75aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    ! bgzip -f {WORK_DIR}/{ancestry}_CTSB_v1.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/{ancestry}_CTSB_v1.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "075d2ccd-5dd5-4d7f-a3ff-6fccd755da01",
   "metadata": {},
   "source": [
    "Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "83955ecc-9a3e-42e0-9c29-420f2a8aacc2",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 1064 lines from VCF file\n",
      "NOTICE: A total of 1057 locus in VCF file passed QC threshold, representing 1008 SNPs (625 transitions and 383 transversions) and 49 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1008 SNP genotypes (625 transitions and 383 transversions) and 49 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1057 unique variants in 1057 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1057 unique variants in 1057 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 5514 lines from VCF file\n",
      "NOTICE: A total of 5507 locus in VCF file passed QC threshold, representing 5214 SNPs (3092 transitions and 2122 transversions) and 293 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 5214 SNP genotypes (3092 transitions and 2122 transversions) and 293 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 5507 unique variants in 5507 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 5507 unique variants in 5507 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 3646 lines from VCF file\n",
      "NOTICE: A total of 3639 locus in VCF file passed QC threshold, representing 3446 SNPs (2017 transitions and 1429 transversions) and 193 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3446 SNP genotypes (2017 transitions and 1429 transversions) and 193 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.refGene.exonic_variant_function\n",
      "Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3639 unique variants in 3639 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3639 unique variants in 3639 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 5533 lines from VCF file\n",
      "NOTICE: A total of 5526 locus in VCF file passed QC threshold, representing 5217 SNPs (2981 transitions and 2236 transversions) and 309 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 5217 SNP genotypes (2981 transitions and 2236 transversions) and 309 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 5526 unique variants in 5526 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 5526 unique variants in 5526 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 10395 lines from VCF file\n",
      "NOTICE: A total of 10388 locus in VCF file passed QC threshold, representing 9808 SNPs (5901 transitions and 3907 transversions) and 580 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 9808 SNP genotypes (5901 transitions and 3907 transversions) and 580 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 10388 unique variants in 10388 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 10388 unique variants in 10388 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 5515 lines from VCF file\n",
      "NOTICE: A total of 5508 locus in VCF file passed QC threshold, representing 5198 SNPs (3012 transitions and 2186 transversions) and 310 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 5198 SNP genotypes (3012 transitions and 2186 transversions) and 310 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 5508 unique variants in 5508 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 5508 unique variants in 5508 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 3256 lines from VCF file\n",
      "NOTICE: A total of 3249 locus in VCF file passed QC threshold, representing 3069 SNPs (1814 transitions and 1255 transversions) and 180 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3069 SNP genotypes (1814 transitions and 1255 transversions) and 180 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3249 unique variants in 3249 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3249 unique variants in 3249 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 2263 lines from VCF file\n",
      "NOTICE: A total of 2256 locus in VCF file passed QC threshold, representing 2141 SNPs (1254 transitions and 887 transversions) and 115 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2141 SNP genotypes (1254 transitions and 887 transversions) and 115 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2256 unique variants in 2256 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2256 unique variants in 2256 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 2808 lines from VCF file\n",
      "NOTICE: A total of 2801 locus in VCF file passed QC threshold, representing 2644 SNPs (1602 transitions and 1042 transversions) and 157 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2644 SNP genotypes (1602 transitions and 1042 transversions) and 157 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2801 unique variants in 2801 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2801 unique variants in 2801 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 4220 lines from VCF file\n",
      "NOTICE: A total of 4213 locus in VCF file passed QC threshold, representing 3976 SNPs (2382 transitions and 1594 transversions) and 237 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3976 SNP genotypes (2382 transitions and 1594 transversions) and 237 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 4213 unique variants in 4213 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 4213 unique variants in 4213 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB_v1.vcf.gz > /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput>\n",
      "NOTICE: Finished reading 7171 lines from VCF file\n",
      "NOTICE: A total of 7164 locus in VCF file passed QC threshold, representing 6769 SNPs (3901 transitions and 2868 transversions) and 395 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 6769 SNP genotypes (3901 transitions and 2868 transversions) and 395 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 7164 unique variants in 7164 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 24 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 7164 unique variants in 7164 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 0\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_CTSB_v1.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "    -out {WORK_DIR}/{ancestry}_CTSB.annovar \\\n",
    "    -remove -protocol refGene,clinvar_20140902 \\\n",
    "    -operation g,f \\\n",
    "    --nopolish \\\n",
    "    -nastring . \\\n",
    "    -vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "41996592-8388-476d-b199-35b3f2b46ee7",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = f'~/workspace/ws_files/CTSB/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "736a2e93-fb7f-4c08-a4bc-de053e4eeef6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2,FDFT1</td>\n",
       "      <td>dist=5230;dist=2998</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>chr8:11792575:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>11792593</td>\n",
       "      <td>11792593</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2,FDFT1</td>\n",
       "      <td>dist=5248;dist=2980</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792593</td>\n",
       "      <td>chr8:11792593:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>11792765</td>\n",
       "      <td>11792765</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2,FDFT1</td>\n",
       "      <td>dist=5420;dist=2808</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792765</td>\n",
       "      <td>chr8:11792765:T:A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>11792784</td>\n",
       "      <td>11792784</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2,FDFT1</td>\n",
       "      <td>dist=5439;dist=2789</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792784</td>\n",
       "      <td>chr8:11792784:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8</td>\n",
       "      <td>11792794</td>\n",
       "      <td>11792794</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>NEIL2,FDFT1</td>\n",
       "      <td>dist=5449;dist=2779</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11792794</td>\n",
       "      <td>chr8:11792794:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2796</th>\n",
       "      <td>8</td>\n",
       "      <td>11919376</td>\n",
       "      <td>11919376</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB,DEFB136</td>\n",
       "      <td>dist=51289;dist=54561</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919376</td>\n",
       "      <td>chr8:11919376:A:T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2797</th>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB,DEFB136</td>\n",
       "      <td>dist=51363;dist=54487</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>chr8:11919450:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2798</th>\n",
       "      <td>8</td>\n",
       "      <td>11919469</td>\n",
       "      <td>11919469</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB,DEFB136</td>\n",
       "      <td>dist=51382;dist=54468</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919469</td>\n",
       "      <td>chr8:11919469:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2799</th>\n",
       "      <td>8</td>\n",
       "      <td>11919475</td>\n",
       "      <td>11919475</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB,DEFB136</td>\n",
       "      <td>dist=51388;dist=54462</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919475</td>\n",
       "      <td>chr8:11919475:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2800</th>\n",
       "      <td>8</td>\n",
       "      <td>11919532</td>\n",
       "      <td>11919532</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>CTSB,DEFB136</td>\n",
       "      <td>dist=51445;dist=54405</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>8</td>\n",
       "      <td>11919532</td>\n",
       "      <td>chr8:11919532:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2801 rows  24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       8  11792575  11792575   T   C   intergenic   NEIL2,FDFT1   \n",
       "1       8  11792593  11792593   T   G   intergenic   NEIL2,FDFT1   \n",
       "2       8  11792765  11792765   T   A   intergenic   NEIL2,FDFT1   \n",
       "3       8  11792784  11792784   G   A   intergenic   NEIL2,FDFT1   \n",
       "4       8  11792794  11792794   T   C   intergenic   NEIL2,FDFT1   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "2796    8  11919376  11919376   A   T   intergenic  CTSB,DEFB136   \n",
       "2797    8  11919450  11919450   G   C   intergenic  CTSB,DEFB136   \n",
       "2798    8  11919469  11919469   T   C   intergenic  CTSB,DEFB136   \n",
       "2799    8  11919475  11919475   T   C   intergenic  CTSB,DEFB136   \n",
       "2800    8  11919532  11919532   T   C   intergenic  CTSB,DEFB136   \n",
       "\n",
       "         GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0       dist=5230;dist=2998                  .                .   \n",
       "1       dist=5248;dist=2980                  .                .   \n",
       "2       dist=5420;dist=2808                  .                .   \n",
       "3       dist=5439;dist=2789                  .                .   \n",
       "4       dist=5449;dist=2779                  .                .   \n",
       "...                     ...                ...              ...   \n",
       "2796  dist=51289;dist=54561                  .                .   \n",
       "2797  dist=51363;dist=54487                  .                .   \n",
       "2798  dist=51382;dist=54468                  .                .   \n",
       "2799  dist=51388;dist=54462                  .                .   \n",
       "2800  dist=51445;dist=54405                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           8   \n",
       "1                   .          0          .          .           8   \n",
       "2                   .          0          .          .           8   \n",
       "3                   .        0.5          .          .           8   \n",
       "4                   .          0          .          .           8   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "2796                .          0          .          .           8   \n",
       "2797                .          0          .          .           8   \n",
       "2798                .          0          .          .           8   \n",
       "2799                .          0          .          .           8   \n",
       "2800                .          0          .          .           8   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0       11792575  chr8:11792575:T:C          T          C          .   \n",
       "1       11792593  chr8:11792593:T:G          T          G          .   \n",
       "2       11792765  chr8:11792765:T:A          T          A          .   \n",
       "3       11792784  chr8:11792784:G:A          G          A          .   \n",
       "4       11792794  chr8:11792794:T:C          T          C          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "2796    11919376  chr8:11919376:A:T          A          T          .   \n",
       "2797    11919450  chr8:11919450:G:C          G          C          .   \n",
       "2798    11919469  chr8:11919469:T:C          T          C          .   \n",
       "2799    11919475  chr8:11919475:T:C          T          C          .   \n",
       "2800    11919532  chr8:11919532:T:C          T          C          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/1  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "2796           .          PR          GT         0/0  \n",
       "2797           .          PR          GT         0/0  \n",
       "2798           .          PR          GT         0/0  \n",
       "2799           .          PR          GT         0/0  \n",
       "2800           .          PR          GT         0/0  \n",
       "\n",
       "[2801 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/CTSB_SAS/SAS_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "0297f00f-6693-40e5-a624-a99dc43e096e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      1436\n",
       "intergenic    1002\n",
       "UTR5           109\n",
       "UTR3            97\n",
       "upstream        58\n",
       "exonic          56\n",
       "downstream      43\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "a419b123-3573-4d1b-bd22-63cf42c9719e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         38\n",
       "synonymous SNV            17\n",
       "nonframeshift deletion     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "d6a784a6-c431-4ab9-85b1-46becf4b2607",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: FIN\n",
      "{'FIN'}\n",
      "Total variants:  1057\n",
      "Intronic:  556\n",
      "UTR3:  37\n",
      "UTR5:  43\n",
      "Total exonic:  16\n",
      "  Synonymous:  7\n",
      "  Nonsynonymous:  9\n",
      "nonsynonymous_lof:  9\n",
      "\n",
      "\n",
      "WORKING ON: AMR\n",
      "{'AMR'}\n",
      "Total variants:  5507\n",
      "Intronic:  2903\n",
      "UTR3:  184\n",
      "UTR5:  224\n",
      "Total exonic:  90\n",
      "  Synonymous:  38\n",
      "  Nonsynonymous:  52\n",
      "nonsynonymous_lof:  53\n",
      "\n",
      "\n",
      "WORKING ON: MDE\n",
      "{'MDE'}\n",
      "Total variants:  3639\n",
      "Intronic:  1928\n",
      "UTR3:  107\n",
      "UTR5:  129\n",
      "Total exonic:  58\n",
      "  Synonymous:  22\n",
      "  Nonsynonymous:  36\n",
      "nonsynonymous_lof:  36\n",
      "\n",
      "\n",
      "WORKING ON: AAC\n",
      "{'AAC'}\n",
      "Total variants:  5526\n",
      "Intronic:  2972\n",
      "UTR3:  182\n",
      "UTR5:  197\n",
      "Total exonic:  75\n",
      "  Synonymous:  36\n",
      "  Nonsynonymous:  39\n",
      "nonsynonymous_lof:  39\n",
      "\n",
      "\n",
      "WORKING ON: EUR\n",
      "{'EUR'}\n",
      "Total variants:  10388\n",
      "Intronic:  5374\n",
      "UTR3:  352\n",
      "UTR5:  415\n",
      "Total exonic:  211\n",
      "  Synonymous:  78\n",
      "  Nonsynonymous:  133\n",
      "nonsynonymous_lof:  137\n",
      "\n",
      "\n",
      "WORKING ON: CAH\n",
      "{'CAH'}\n",
      "Total variants:  5508\n",
      "Intronic:  2957\n",
      "UTR3:  177\n",
      "UTR5:  203\n",
      "Total exonic:  83\n",
      "  Synonymous:  39\n",
      "  Nonsynonymous:  44\n",
      "nonsynonymous_lof:  45\n",
      "\n",
      "\n",
      "WORKING ON: CAS\n",
      "{'CAS'}\n",
      "Total variants:  3249\n",
      "Intronic:  1722\n",
      "UTR3:  115\n",
      "UTR5:  141\n",
      "Total exonic:  61\n",
      "  Synonymous:  23\n",
      "  Nonsynonymous:  38\n",
      "nonsynonymous_lof:  38\n",
      "\n",
      "\n",
      "WORKING ON: AJ\n",
      "{'AJ'}\n",
      "Total variants:  2256\n",
      "Intronic:  1191\n",
      "UTR3:  68\n",
      "UTR5:  96\n",
      "Total exonic:  33\n",
      "  Synonymous:  10\n",
      "  Nonsynonymous:  23\n",
      "nonsynonymous_lof:  23\n",
      "\n",
      "\n",
      "WORKING ON: SAS\n",
      "{'SAS'}\n",
      "Total variants:  2801\n",
      "Intronic:  1436\n",
      "UTR3:  97\n",
      "UTR5:  109\n",
      "Total exonic:  56\n",
      "  Synonymous:  18\n",
      "  Nonsynonymous:  38\n",
      "nonsynonymous_lof:  38\n",
      "\n",
      "\n",
      "WORKING ON: EAS\n",
      "{'EAS'}\n",
      "Total variants:  4213\n",
      "Intronic:  2219\n",
      "UTR3:  150\n",
      "UTR5:  178\n",
      "Total exonic:  89\n",
      "  Synonymous:  33\n",
      "  Nonsynonymous:  56\n",
      "nonsynonymous_lof:  58\n",
      "\n",
      "\n",
      "WORKING ON: AFR\n",
      "{'AFR'}\n",
      "Total variants:  7164\n",
      "Intronic:  3823\n",
      "UTR3:  233\n",
      "UTR5:  276\n",
      "Total exonic:  113\n",
      "  Synonymous:  51\n",
      "  Nonsynonymous:  62\n",
      "nonsynonymous_lof:  63\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Make lists of variants to keep - all coding, coding nonsynonymous (missense - as they are coded in ANNOVAR), deleterious (CADD_phred > 20)\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "\n",
    "    # Read in ANNOVAR multianno file\n",
    "    gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_CTSB.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "    \n",
    "    #Print number of variants in the different categories\n",
    "    results = [] \n",
    "\n",
    "    utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "    intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "    exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "    utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "    coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "    coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'nonsynonymous SNV')]\n",
    "    lof = exonic[(exonic['ExonicFunc.refGene'] == 'stopgain') | (exonic['ExonicFunc.refGene'] == 'stoploss') | (exonic['ExonicFunc.refGene'] == 'frameshift deletion') | (exonic['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "    nonsynonymous_lof = pd.concat([coding_nonsynonymous, lof])\n",
    "\n",
    "    print({ancestry})\n",
    "    print('Total variants: ', len(gene))\n",
    "    print(\"Intronic: \", len(intronic))\n",
    "    print('UTR3: ', len(utr3))\n",
    "    print('UTR5: ', len(utr5))\n",
    "    print(\"Total exonic: \", len(exonic))\n",
    "    print('  Synonymous: ', len(coding_synonymous))\n",
    "    print(\"  Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "    print(\"nonsynonymous_lof: \", len(nonsynonymous_lof))\n",
    "    results.append((gene, intronic, utr3, utr5, exonic, coding_synonymous, coding_nonsynonymous, nonsynonymous_lof))\n",
    "    print('\\n')\n",
    "\n",
    "    # Save in PLINK format - coding nonsynonymous \n",
    "    # These are missense variants - other types of nonsynonymous variants (e.g stopgain/loss, or frameshift variants are coded differently in the ExonicFunc.refGene \n",
    "    variants_toKeep = nonsynonymous_lof[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_CTSB.nonsynonymous_lof.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "\n",
    "\n",
    "    # Save in PLINK format - all coding variants\n",
    "    variants_toKeep2 = exonic[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry}_CTSB.exonic.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62b1c5df-4712-426f-8d03-3be417fd0fa7",
   "metadata": {},
   "source": [
    "ALL variants\n",
    "\n",
    "assoc\n",
    "\n",
    "glm\n",
    "\n",
    "ASSOC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "dc1b2cd2-5394-486b-a512-d6ee294df110",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "5526 variants loaded from .bim file.\n",
      "1215 people (532 males, 683 females) loaded from .fam.\n",
      "1133 phenotype values loaded from .fam.\n",
      "--keep: 1207 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1207 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.997149.\n",
      "4315 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1211 variants and 1207 people pass filters and QC.\n",
      "Among remaining phenotypes, 369 are cases and 756 are controls.  (82 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "7164 variants loaded from .bim file.\n",
      "3780 people (2087 males, 1693 females) loaded from .fam.\n",
      "3498 phenotype values loaded from .fam.\n",
      "--keep: 3581 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.995253.\n",
      "5946 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1218 variants and 3581 people pass filters and QC.\n",
      "Among remaining phenotypes, 1147 are cases and 2169 are controls.  (265\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "2256 variants loaded from .bim file.\n",
      "3472 people (2146 males, 1326 females) loaded from .fam.\n",
      "1966 phenotype values loaded from .fam.\n",
      "--keep: 2716 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2716 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.994882.\n",
      "1514 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "742 variants and 2716 people pass filters and QC.\n",
      "Among remaining phenotypes, 1285 are cases and 408 are controls.  (1023\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "5507 variants loaded from .bim file.\n",
      "3608 people (1735 males, 1873 females) loaded from .fam.\n",
      "3434 phenotype values loaded from .fam.\n",
      "--keep: 3471 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3471 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.990937.\n",
      "4827 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "680 variants and 3471 people pass filters and QC.\n",
      "Among remaining phenotypes, 1917 are cases and 1402 are controls.  (152\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3249 variants loaded from .bim file.\n",
      "1691 people (737 males, 954 females) loaded from .fam.\n",
      "1408 phenotype values loaded from .fam.\n",
      "--keep: 1518 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1518 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.989357.\n",
      "2564 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "685 variants and 1518 people pass filters and QC.\n",
      "Among remaining phenotypes, 734 are cases and 578 are controls.  (206\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "4213 variants loaded from .bim file.\n",
      "6662 people (4001 males, 2661 females) loaded from .fam.\n",
      "5116 phenotype values loaded from .fam.\n",
      "--keep: 6414 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 6414 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.983892.\n",
      "3736 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "477 variants and 6414 people pass filters and QC.\n",
      "Among remaining phenotypes, 2377 are cases and 2598 are controls.  (1439\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "10388 variants loaded from .bim file.\n",
      "58823 people (33074 males, 25749 females) loaded from .fam.\n",
      "37150 phenotype values loaded from .fam.\n",
      "--keep: 53641 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 53641 founders and 0 nonfounders present.\n",
      "Calculating allele frequenc done.\n",
      "Total genotyping rate in remaining samples is 0.738162.\n",
      "9703 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "685 variants and 53641 people pass filters and QC.\n",
      "Among remaining phenotypes, 24208 are cases and 9662 are controls.  (19771\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "1057 variants loaded from .bim file.\n",
      "144 people (62 males, 82 females) loaded from .fam.\n",
      "118 phenotype values loaded from .fam.\n",
      "--keep: 111 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.980397.\n",
      "375 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "682 variants and 111 people pass filters and QC.\n",
      "Among remaining phenotypes, 80 are cases and 11 are controls.  (20 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3639 variants loaded from .bim file.\n",
      "1493 people (815 males, 678 females) loaded from .fam.\n",
      "1311 phenotype values loaded from .fam.\n",
      "--keep: 1448 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1448 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.993178.\n",
      "2949 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "690 variants and 1448 people pass filters and QC.\n",
      "Among remaining phenotypes, 724 are cases and 551 are controls.  (173\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "2801 variants loaded from .bim file.\n",
      "945 people (606 males, 339 females) loaded from .fam.\n",
      "586 phenotype values loaded from .fam.\n",
      "--keep: 870 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 870 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.992861.\n",
      "2093 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "708 variants and 870 people pass filters and QC.\n",
      "Among remaining phenotypes, 309 are cases and 261 are controls.  (300\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "5508 variants loaded from .bim file.\n",
      "1111 people (589 males, 522 females) loaded from .fam.\n",
      "1040 phenotype values loaded from .fam.\n",
      "--keep: 1039 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1039 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.995908.\n",
      "4534 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "974 variants and 1039 people pass filters and QC.\n",
      "Among remaining phenotypes, 663 are cases and 318 are controls.  (58 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis using plink assoc for all variants, not adjusting for any covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    ! /home/jupyter/tools/plink \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "    --assoc \\\n",
    "    --allow-no-sex \\\n",
    "    --ci 0.95 \\\n",
    "    --maf 0.01 \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "807ea5d5-902e-48b6-aed1-1cb0ace9f76d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "Variants with p-value < 0.05: (112, 13)\n",
      "WORKING ON: AFR\n",
      "Variants with p-value < 0.05: (81, 13)\n",
      "WORKING ON: AJ\n",
      "Variants with p-value < 0.05: (23, 13)\n",
      "WORKING ON: AMR\n",
      "Variants with p-value < 0.05: (433, 13)\n",
      "WORKING ON: CAS\n",
      "Variants with p-value < 0.05: (86, 13)\n",
      "WORKING ON: EAS\n",
      "Variants with p-value < 0.05: (129, 13)\n",
      "WORKING ON: EUR\n",
      "Variants with p-value < 0.05: (272, 13)\n",
      "WORKING ON: FIN\n",
      "Variants with p-value < 0.05: (6, 13)\n",
      "WORKING ON: MDE\n",
      "Variants with p-value < 0.05: (46, 13)\n",
      "WORKING ON: SAS\n",
      "Variants with p-value < 0.05: (7, 13)\n",
      "WORKING ON: CAH\n",
      "Variants with p-value < 0.05: (31, 13)\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Look at assoc results, only variants with nominal p-value < 0.05\n",
    "    freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_CTSB.all.assoc', sep='\\s+')\n",
    "    sig_all_nonadj = freq[freq['P']<0.05]\n",
    "    \n",
    "    print(f'Variants with p-value < 0.05: {sig_all_nonadj.shape}')\n",
    "    \n",
    "    #Save FREQ to csv\n",
    "    freq.to_csv(f'{WORK_DIR}/{ancestry}.all_nonadj.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0ea69507-4978-46d7-b500-b7b5cfdb95fe",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:18 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24620 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam.\n",
      "5526 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--keep: 1207 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.noNA.txt.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "4315 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1211 variants remaining after main filters.\n",
      "done. logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:18 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:19 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24612 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam.\n",
      "7164 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--keep: 3581 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.noNA.txt.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "done.lating allele frequencies... 0%\n",
      "5946 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1218 variants remaining after main filters.\n",
      "done. logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:20 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:21 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24605 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam.\n",
      "2256 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--keep: 2716 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.noNA.txt.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1514 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "742 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:21 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:22 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24616 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam.\n",
      "5507 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.noNA.txt.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "4827 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "680 variants remaining after main filters.\n",
      "Error: Cannot proceed with --glm regression on phenotype 'PHENO1', since\n",
      "variance inflation factor for covariate 'PC2' is too high (VIF_TOO_HIGH). You\n",
      "may want to remove redundant covariates and try again.\n",
      "End time: Sat Jul 26 13:03:22 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:24 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24626 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam.\n",
      "3249 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--keep: 1518 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.noNA.txt.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2564 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "685 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:24 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:25 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24604 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam.\n",
      "4213 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--keep: 6414 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.noNA.txt.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3736 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "477 variants remaining after main filters.\n",
      "done. logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:25 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:27 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24603 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam.\n",
      "10388 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--keep: 53641 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.noNA.txt.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "done.lating allele frequencies... 0%\n",
      "9703 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "685 variants remaining after main filters.\n",
      "done. logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:29 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:30 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24617 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam.\n",
      "1057 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--keep: 111 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.noNA.txt.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "375 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "682 variants remaining after main filters.\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\n",
      "phenotype 'PHENO1'.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:30 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:31 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24614 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam.\n",
      "3639 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--keep: 1448 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.noNA.txt.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2949 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "690 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:31 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:33 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24624 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam.\n",
      "2801 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--keep: 870 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.noNA.txt.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2093 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "708 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:33 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:03:34 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24610 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam.\n",
      "5508 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--keep: 1039 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.noNA.txt.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "4534 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "974 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:03:34 2025\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis with covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "    --allow-no-sex \\\n",
    "    --maf 0.01 \\\n",
    "    --ci 0.95 \\\n",
    "    --glm \\\n",
    "    --covar /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.noNA.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --covar-variance-standardize \\\n",
    "    --neg9-pheno-really-missing \\\n",
    "    --out {WORK_DIR}/{ancestry}_CTSB.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "88c89bbd-a750-458e-a1e9-76323f50595c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 13:04:36 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24638 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam.\n",
      "5507 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "4827 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "680 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 13:04:37 2025\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_AMR'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/AMR_CTSB \\\n",
    "--keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep \\\n",
    "--allow-no-sex \\\n",
    "--maf 0.01 \\\n",
    "--ci 0.95 \\\n",
    "--glm \\\n",
    "--covar /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--neg9-pheno-really-missing \\\n",
    "--out {WORK_DIR}/AMR_CTSB.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "a3a3903b-442d-421d-b51f-615bb8cff310",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "There are 51 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AFR\n",
      "There are 94 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "There are 34 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: AMR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 96 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: CAS\n",
      "There are 19 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EAS\n",
      "There are 12 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: EUR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 198 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: FIN\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 0 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: MDE\n",
      "There are 49 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: SAS\n",
      "There are 3 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: CAH\n",
      "There are 33 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1691/829170199.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    }
   ],
   "source": [
    "#Process results from plink glm analysis including covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Read in plink glm results\n",
    "    assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_CTSB.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "\n",
    "    #Filter for additive test only - this is the variant results\n",
    "    assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    significant = assoc_add[assoc_add['P']<0.05]\n",
    "\n",
    "    print(f'There are {len(significant)} variants with p-value < 0.05 in glm')\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    GWsignificant = assoc_add[assoc_add['P']<5e-8]\n",
    "\n",
    "    print(f'There are {len(GWsignificant)} variants with p-value < 5e-8 in glm')\n",
    "    \n",
    "    #Save assoc_add to csv\n",
    "    assoc_add.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14f35cf1-a823-4820-9b7f-ad9cfd19a80f",
   "metadata": {},
   "source": [
    "format assoc file to do meta analysis(AFR/AJ/AMR/EAS/EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "d2113164-da74-4cf0-ad2a-6b634ae52a03",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define ancestry groups and their sample sizes\n",
    "ancestry_sample_sizes = {\n",
    "    'AFR': 3316,\n",
    "    'AJ': 1693,\n",
    "    'AMR': 3319,\n",
    "    'EAS': 4975,\n",
    "    'EUR': 33870\n",
    "}\n",
    "\n",
    "# Output container (optional)\n",
    "cleaned_results = {}\n",
    "\n",
    "# Loop over each ancestry\n",
    "for ancestry, sample_size in ancestry_sample_sizes.items():\n",
    "    # Load PLINK association result file\n",
    "    file_path = f'/home/jupyter/workspace/ws_files/CTSB/CTSB_{ancestry}/{ancestry}.all_adj.csv'\n",
    "    df = pd.read_csv(file_path)\n",
    "\n",
    "    # Build MARKERNAME\n",
    "    df['MARKERNAME'] = 'chr' + df['#CHROM'].astype(str) + ':' + df['POS'].astype(str)\n",
    "\n",
    "    # Rename columns\n",
    "    df.rename(columns={\n",
    "        '#CHROM': 'CHROMOSOME',\n",
    "        'POS': 'POSITION',\n",
    "        'REF': 'NEA',\n",
    "        'ALT': 'EA',\n",
    "        'A1_FREQ': 'EAF',\n",
    "        'LOG(OR)_SE': 'SE',\n",
    "        'L95': 'OR_95L',\n",
    "        'U95': 'OR_95U'\n",
    "    }, inplace=True)\n",
    "\n",
    "    # Add BETA = log(OR)\n",
    "    df['BETA'] = np.log(df['OR'])\n",
    "\n",
    "    # Add sample size\n",
    "    df['N'] = sample_size\n",
    "\n",
    "    # Keep desired columns\n",
    "    final_cols = [\n",
    "        'MARKERNAME', 'CHROMOSOME', 'POSITION', 'NEA', 'EA', 'EAF',\n",
    "        'BETA', 'OR', 'SE', 'OR_95L', 'OR_95U', 'P', 'N'\n",
    "    ]\n",
    "    df_cleaned = df[final_cols]\n",
    "\n",
    "    # Store or save\n",
    "    cleaned_results[ancestry] = df_cleaned\n",
    "\n",
    "    # Optional: save to file\n",
    "    output_path = f'/home/jupyter/workspace/ws_files/CTSB/CTSB_{ancestry}/{ancestry}.cleaned_assoc.txt.gz'\n",
    "    df_cleaned.to_csv(output_path, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "62d3c52e-7a50-41e0-becc-8cd38cb2efb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MARKERNAME</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>OR_95L</th>\n",
       "      <th>OR_95U</th>\n",
       "      <th>P</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8:11792575</td>\n",
       "      <td>8</td>\n",
       "      <td>11792575</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.019353</td>\n",
       "      <td>0.071269</td>\n",
       "      <td>1.073870</td>\n",
       "      <td>0.370647</td>\n",
       "      <td>0.519347</td>\n",
       "      <td>2.220490</td>\n",
       "      <td>0.847515</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr8:11792812</td>\n",
       "      <td>8</td>\n",
       "      <td>11792812</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.114708</td>\n",
       "      <td>-0.096086</td>\n",
       "      <td>0.908386</td>\n",
       "      <td>0.159408</td>\n",
       "      <td>0.664633</td>\n",
       "      <td>1.241540</td>\n",
       "      <td>0.546664</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr8:11792966</td>\n",
       "      <td>8</td>\n",
       "      <td>11792966</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.398739</td>\n",
       "      <td>-0.033624</td>\n",
       "      <td>0.966935</td>\n",
       "      <td>0.107804</td>\n",
       "      <td>0.782771</td>\n",
       "      <td>1.194430</td>\n",
       "      <td>0.755114</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8:11793099</td>\n",
       "      <td>8</td>\n",
       "      <td>11793099</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.154628</td>\n",
       "      <td>0.170721</td>\n",
       "      <td>1.186160</td>\n",
       "      <td>0.154521</td>\n",
       "      <td>0.876222</td>\n",
       "      <td>1.605730</td>\n",
       "      <td>0.269231</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8:11793200</td>\n",
       "      <td>8</td>\n",
       "      <td>11793200</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.129552</td>\n",
       "      <td>0.097363</td>\n",
       "      <td>1.102260</td>\n",
       "      <td>0.159246</td>\n",
       "      <td>0.806741</td>\n",
       "      <td>1.506030</td>\n",
       "      <td>0.540933</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>737</th>\n",
       "      <td>chr8:11918618</td>\n",
       "      <td>8</td>\n",
       "      <td>11918618</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.064459</td>\n",
       "      <td>-0.002321</td>\n",
       "      <td>0.997682</td>\n",
       "      <td>0.218229</td>\n",
       "      <td>0.650482</td>\n",
       "      <td>1.530200</td>\n",
       "      <td>0.991517</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>chr8:11918643</td>\n",
       "      <td>8</td>\n",
       "      <td>11918643</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.015526</td>\n",
       "      <td>-0.228613</td>\n",
       "      <td>0.795636</td>\n",
       "      <td>0.410404</td>\n",
       "      <td>0.355941</td>\n",
       "      <td>1.778490</td>\n",
       "      <td>0.577496</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>739</th>\n",
       "      <td>chr8:11919062</td>\n",
       "      <td>8</td>\n",
       "      <td>11919062</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.019649</td>\n",
       "      <td>-0.845029</td>\n",
       "      <td>0.429545</td>\n",
       "      <td>0.306669</td>\n",
       "      <td>0.235490</td>\n",
       "      <td>0.783512</td>\n",
       "      <td>0.005860</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>740</th>\n",
       "      <td>chr8:11919200</td>\n",
       "      <td>8</td>\n",
       "      <td>11919200</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.083690</td>\n",
       "      <td>-0.260764</td>\n",
       "      <td>0.770463</td>\n",
       "      <td>0.177479</td>\n",
       "      <td>0.544103</td>\n",
       "      <td>1.090990</td>\n",
       "      <td>0.141761</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>741</th>\n",
       "      <td>chr8:11919450</td>\n",
       "      <td>8</td>\n",
       "      <td>11919450</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.063611</td>\n",
       "      <td>-0.018521</td>\n",
       "      <td>0.981649</td>\n",
       "      <td>0.218267</td>\n",
       "      <td>0.639981</td>\n",
       "      <td>1.505730</td>\n",
       "      <td>0.932376</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>742 rows  13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        MARKERNAME  CHROMOSOME  POSITION NEA EA       EAF      BETA        OR  \\\n",
       "0    chr8:11792575           8  11792575   T  C  0.019353  0.071269  1.073870   \n",
       "1    chr8:11792812           8  11792812   C  T  0.114708 -0.096086  0.908386   \n",
       "2    chr8:11792966           8  11792966   T  C  0.398739 -0.033624  0.966935   \n",
       "3    chr8:11793099           8  11793099   T  C  0.154628  0.170721  1.186160   \n",
       "4    chr8:11793200           8  11793200   G  C  0.129552  0.097363  1.102260   \n",
       "..             ...         ...       ...  .. ..       ...       ...       ...   \n",
       "737  chr8:11918618           8  11918618   T  C  0.064459 -0.002321  0.997682   \n",
       "738  chr8:11918643           8  11918643   T  G  0.015526 -0.228613  0.795636   \n",
       "739  chr8:11919062           8  11919062   A  G  0.019649 -0.845029  0.429545   \n",
       "740  chr8:11919200           8  11919200   G  A  0.083690 -0.260764  0.770463   \n",
       "741  chr8:11919450           8  11919450   G  C  0.063611 -0.018521  0.981649   \n",
       "\n",
       "           SE    OR_95L    OR_95U         P     N  \n",
       "0    0.370647  0.519347  2.220490  0.847515  1693  \n",
       "1    0.159408  0.664633  1.241540  0.546664  1693  \n",
       "2    0.107804  0.782771  1.194430  0.755114  1693  \n",
       "3    0.154521  0.876222  1.605730  0.269231  1693  \n",
       "4    0.159246  0.806741  1.506030  0.540933  1693  \n",
       "..        ...       ...       ...       ...   ...  \n",
       "737  0.218229  0.650482  1.530200  0.991517  1693  \n",
       "738  0.410404  0.355941  1.778490  0.577496  1693  \n",
       "739  0.306669  0.235490  0.783512  0.005860  1693  \n",
       "740  0.177479  0.544103  1.090990  0.141761  1693  \n",
       "741  0.218267  0.639981  1.505730  0.932376  1693  \n",
       "\n",
       "[742 rows x 13 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ.cleaned_assoc.txt.gz', sep='\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee8794a3-74b9-467d-93dd-60e2316959bd",
   "metadata": {},
   "source": [
    "Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "cd9d3134-31d8-48e1-bbdb-ac63bd9f2498",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running plink to extract exonic variants for ancestry: AAC\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:08:44 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24623 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb.psam.\n",
      "3293679 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--extract bed1: 3293604 variants excluded.\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "75 variants remaining after main filters.\n",
      "--export vcf to\n",
      "done.\n",
      "End time: Sat Jul 26 13:08:45 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AAC\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AAC\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:08:48 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24606 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb.psam.\n",
      "3293679 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr8_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--extract bed1: 3293640 variants excluded.\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "39 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:08:49 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AAC\n",
      "Running plink to extract exonic variants for ancestry: AFR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:08:52 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24605 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb.psam.\n",
      "4081203 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--extract bed1: 4081091 variants excluded.\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "112 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:08:53 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AFR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AFR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:08:56 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24610 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb.psam.\n",
      "4081203 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr8_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--extract bed1: 4081141 variants excluded.\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "62 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 13:08:57 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AFR\n",
      "Running plink to extract exonic variants for ancestry: AJ\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:01 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24618 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb.psam.\n",
      "1376363 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--extract bed1: 1376330 variants excluded.\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "33 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:09:01 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AJ\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AJ\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:04 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24624 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb.psam.\n",
      "1376363 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr8_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--extract bed1: 1376340 variants excluded.\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "23 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.nonsynonymous_lof.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 13:09:05 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AJ\n",
      "Running plink to extract exonic variants for ancestry: AMR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:08 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24626 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb.psam.\n",
      "3550811 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--extract bed1: 3550723 variants excluded.\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "88 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:09:09 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AMR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AMR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:12 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24634 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb.psam.\n",
      "3550811 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr8_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--extract bed1: 3550759 variants excluded.\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "52 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:09:13 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AMR\n",
      "Running plink to extract exonic variants for ancestry: CAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:17 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24643 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb.psam.\n",
      "2184117 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--extract bed1: 2184057 variants excluded.\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "60 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.exonic.vcdone.\n",
      "End time: Sat Jul 26 13:09:17 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:21 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24619 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb.psam.\n",
      "2184117 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr8_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--extract bed1: 2184079 variants excluded.\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "38 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:09:21 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAS\n",
      "Running plink to extract exonic variants for ancestry: EAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:24 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24619 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb.psam.\n",
      "2703578 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--extract bed1: 2703491 variants excluded.\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "87 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:09:25 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: EAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:29 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24610 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb.psam.\n",
      "2703578 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr8_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--extract bed1: 2703521 variants excluded.\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "57 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:09:29 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EAS\n",
      "Running plink to extract exonic variants for ancestry: EUR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:33 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24609 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb.psam.\n",
      "6200464 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--extract bed1: 6200257 variants excluded.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "207 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.exondone.\n",
      "End time: Sat Jul 26 13:09:34 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: EUR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EUR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:38 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24640 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb.psam.\n",
      "6200464 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr8_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--extract bed1: 6200327 variants excluded.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "137 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 13:09:39 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EUR\n",
      "Running plink to extract exonic variants for ancestry: FIN\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:43 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24638 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb.psam.\n",
      "715411 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--extract bed1: 715395 variants excluded.\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "16 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:09:44 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: FIN\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: FIN\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:47 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24623 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb.psam.\n",
      "715411 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr8_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--extract bed1: 715402 variants excluded.\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "9 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:09:47 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: FIN\n",
      "Running plink to extract exonic variants for ancestry: MDE\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:51 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24623 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb.psam.\n",
      "2403439 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--extract bed1: 2403382 variants excluded.\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "57 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:09:52 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: MDE\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: MDE\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:55 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24612 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb.psam.\n",
      "2403439 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr8_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--extract bed1: 2403403 variants excluded.\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "36 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:09:56 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: MDE\n",
      "Running plink to extract exonic variants for ancestry: SAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:09:59 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24607 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb.psam.\n",
      "1753806 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--extract bed1: 1753751 variants excluded.\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "55 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:10:00 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: SAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: SAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:10:03 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24612 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb.psam.\n",
      "1753806 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr8_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--extract bed1: 1753768 variants excluded.\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "38 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:10:04 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: SAS\n",
      "Running plink to extract exonic variants for ancestry: CAH\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:10:08 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24612 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb.psam.\n",
      "3231362 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--extract bed1: 3231280 variants excluded.\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "82 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.exonic.vcf .done.\n",
      "End time: Sat Jul 26 13:10:08 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAH\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAH\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 13:10:12 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24617 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb.psam.\n",
      "3231362 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr8_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--extract bed1: 3231318 variants excluded.\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "44 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 13:10:12 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAH\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "#Loop over all the ancestries and the 2 variant classes - run rvtests for all coding and missense variants\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "\n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running plink to extract {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        #Extract relevant variants\n",
    "        ! /home/jupyter/tools/plink2 \\\n",
    "        --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr8_{ancestry}_release10_vwb \\\n",
    "        --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "        --extract range {WORK_DIR}/{ancestry}_CTSB.{variant_class}.variantstoKeep.txt \\\n",
    "        --recode vcf-iid \\\n",
    "        --out {WORK_DIR}/{ancestry}_CTSB.{variant_class}\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running bgzip and tabix for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## Bgzip and Tabix (zip and index the file)\n",
    "        ! bgzip -f {WORK_DIR}/{ancestry}_CTSB.{variant_class}.vcf\n",
    "        ! tabix -f -p vcf {WORK_DIR}/{ancestry}_CTSB.{variant_class}.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7dff37ef-7204-42ae-a544-73d195f63a17",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Run RVtests\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "\n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running RVtests for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## RVtests with covariates \n",
    "        #Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "        ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "        --out {WORK_DIR}/{ancestry}_CTSB.burden.{variant_class} \\\n",
    "        --kernel skato \\\n",
    "        --inVcf {WORK_DIR}/{ancestry}_CTSB.{variant_class}.vcf.gz \\\n",
    "        --pheno /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt \\\n",
    "        --pheno-name PHENO \\\n",
    "        --gene CTSB \\\n",
    "        --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "        --covar /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt \\\n",
    "        --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "        --freqUpper 0.01\n",
    "# --burden cmc,zeggini,mb,fp,cmcWald --kernel skat,skato \\"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23b201d4-ed1b-4998-a757-d271a13c0be7",
   "metadata": {},
   "source": [
    "Look at RVtest results SKAT-O"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "99fb6c0a-3a7d-4882-b9d2-b09c268b96b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/rpy2/rinterface/__init__.py:1211: UserWarning: Environment variable \"R_SESSION_TMPDIR\" redefined by R and overriding existing variable. Current: \"/tmp/RtmpZB9niY\", R: \"/tmp/RtmpWw6ESi\"\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "492fc356-dcc9-4013-ba76-24c5c4ca4e6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/CTSB\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "3d202a2e-0a50-44a8-a911-7d5f51fbff2c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t40128\t93\t81\t230895\t0\t0.57025\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_EUR/EUR_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "93003fa2-7850-4b47-a3dc-b939b1dbeb96",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t40128\t59\t55\t145610\t1\t0.558984\n"
     ]
    }
   ],
   "source": [
    "#Check EUR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_EUR/EUR_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "f3e5bbda-51a9-409e-b49c-f790af13da94",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t1061\t28\t26\t17177.6\t0\t0.747416\n"
     ]
    }
   ],
   "source": [
    "#Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_AAC/AAC_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "dcb7d409-bf45-4352-9919-7600e3ef3932",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t1061\t15\t15\t21269.3\t0\t0.27747\n"
     ]
    }
   ],
   "source": [
    "#Check AAC nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_AAC/AAC_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "b798e69c-64cb-4a18-a28e-bcadd1a87384",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t2738\t46\t29\t15284.2\t0\t0.83222\n"
     ]
    }
   ],
   "source": [
    "#Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_AFR/AFR_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "6e5718f5-8c11-4aed-a285-5c42e646323a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t2738\t25\t18\t44255\t0.3\t0.145056\n"
     ]
    }
   ],
   "source": [
    "#Check AFR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_AFR/AFR_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "b5298bd5-9d66-429e-ab56-e7485d2acad1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [0.83222, 0.145056]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "deb7de2d-1b58-4a7a-b0ac-59db83162190",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.832220 0.290112\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "3232ab11-c06c-441a-89b4-f11e8fa8869a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t2174\t13\t11\t4309.19\t1\t0.728951\n"
     ]
    }
   ],
   "source": [
    "#Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_AJ/AJ_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "5c401079-f179-4df7-8bb5-3927372d32c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t2174\t11\t9\t1176.1\t1\t0.881028\n"
     ]
    }
   ],
   "source": [
    "#Check AJ nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_AJ/AJ_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "af30a1ba-239d-42fa-b307-f6627e4d9efa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [0.728951, 0.881028]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "bbbfe2a0-df8b-4f0c-8dae-ccc1e071ba34",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.881028 0.881028\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "2e7db757-a4b1-46e9-b5f7-8ea0664d185c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t3265\t39\t26\t17400.8\t0\t0.745868\n"
     ]
    }
   ],
   "source": [
    "#Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_AMR/AMR_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "1aa0103e-c423-4178-b09d-a9289ec56ee1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t3265\t27\t19\t17122.4\t0\t0.471545\n"
     ]
    }
   ],
   "source": [
    "#Check AMR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_AMR/AMR_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "7661942b-aa06-4cad-8c47-886605b5bd55",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [0.745868, 0.471545]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "eedfa248-1f57-4613-8fd5-516156a7f809",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.745868 0.745868\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "e7b916b3-ca67-4f17-bf38-fb72bb372993",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t1104\t18\t11\t814.855\t1\t0.857092\n"
     ]
    }
   ],
   "source": [
    "#Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_CAS/CAS_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "9cf628ef-9232-4416-b207-b59669a231e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t1104\t12\t9\t1996.05\t1\t0.756806\n"
     ]
    }
   ],
   "source": [
    "#Check CAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_CAS/CAS_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "44574da8-e94c-40ba-b679-32a95c094d46",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [0.857092, 0.756806]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "7eca9c79-9589-46cd-99c2-4bdce04c989b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.857092 0.857092\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "787e3bde-01a7-4c68-9d2a-9ea13c1944b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t4670\t24\t14\t704.747\t1\t1\n"
     ]
    }
   ],
   "source": [
    "#Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_EAS/EAS_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "6c09f3df-8220-4fd2-8e8a-c808b853fd93",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t4670\t17\t10\t11348.2\t1\t0.539251\n"
     ]
    }
   ],
   "source": [
    "#Check EAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_EAS/EAS_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "638499a3-7c07-4b89-9ffe-5402fe822c46",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [1, 0.539251]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "c0e10936-e01e-4ad8-9342-22f1803903fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1 1\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "b56d3feb-4ba6-40d5-a6a5-54c8bd87e74d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t45\t3\t0\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check FIN all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_FIN/FIN_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "836bb378-8eb8-4d18-9820-0f36a6275a80",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t45\t2\t0\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check FIN nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_FIN/FIN_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "ff4cbfd2-4663-4d92-9a95-39e9b703dedd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t1230\t24\t14\t11656.6\t1\t0.400443\n"
     ]
    }
   ],
   "source": [
    "#Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_MDE/MDE_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "58c1991c-635e-4e28-b31e-a48fbf98655d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t1230\t15\t7\t2204.48\t1\t0.565363\n"
     ]
    }
   ],
   "source": [
    "#Check MDE nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_MDE/MDE_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "d7a1a8cb-cb8c-482f-bd09-b02541d88bf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [0.400443, 0.565363]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "b881547e-8632-4513-bed7-17597fb40e04",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.565363 0.565363\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "d65f4c45-2ffe-437e-bb79-ef2331a4ff69",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t689\t19\t8\t2239.37\t1\t0.419244\n"
     ]
    }
   ],
   "source": [
    "#Check SAS all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_SAS/SAS_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "f6918b02-57c2-4bbe-bcda-3278eaae185a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t689\t12\t5\t774.291\t0\t0.639828\n"
     ]
    }
   ],
   "source": [
    "#Check SAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_SAS/SAS_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "1d4a7226-259b-4117-843c-68147ac08894",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [0.419244, 0.639828]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "0e366063-0988-43e7-853c-1d68e12db204",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.639828 0.639828\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "4ef670d9-b944-4331-91ea-b0b2c607a16b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t900\t28\t21\t18155.3\t0\t0.730282\n"
     ]
    }
   ],
   "source": [
    "#Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/CTSB_CAH/CAH_CTSB.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "bea112f3-8766-458a-8ea7-1212d33fff91",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "CTSB\t8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868150,8:11842524-11868087,8:11842524-11868150,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479971-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568,8_KZ208915v1_fix:1479908-1505568\t900\t14\t12\t3469.22\t1\t0.817762\n"
     ]
    }
   ],
   "source": [
    "#Check CAH nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/CTSB_CAH/CAH_CTSB.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "03fbeec9-2136-4cd5-83db-aebf2d3ce54f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, define your p-values in Python\n",
    "pvalues = [0.730282, 0.817762]\n",
    "# Send to R\n",
    "import rpy2.robjects as ro\n",
    "ro.globalenv['pvalues'] = ro.FloatVector(pvalues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "52d27b88-2669-4c2a-98dd-944fb9b9ca37",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.817762 0.817762\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# Adjust using BH method\n",
    "fdrs <- p.adjust(pvalues, method = \"BH\")\n",
    "print(fdrs)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b991fea-52c1-4e13-b381-cfb59ca64bac",
   "metadata": {},
   "source": [
    "LD pruning(in EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "59a2b571-6d8f-40e9-8294-ce162bd19a7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "626d4507-d043-41a0-a501-74391e8e6c51",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//CTSB/CTSB_EUR/EUR_CTSB\n",
      "  --exclude /home/jupyter/workspace/ws_files//exclusion_regions_hg38.txt\n",
      "  --geno 0.05\n",
      "  --hwe 1e-5 0.001\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED\n",
      "\n",
      "Start time: Sat Jul 26 13:26:22 2025\n",
      "26046 MiB RAM detected, ~24630 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//CTSB/CTSB_EUR/EUR_CTSB.fam.\n",
      "10388 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//CTSB/CTSB_EUR/EUR_CTSB.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--exclude: 10388 variants remaining.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "done.lating allele frequencies... 0%\n",
      "--geno: 5516 variants removed due to missing genotype data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "4273 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "599 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED.bed ... 0%\n",
      "End time: Sat Jul 26 13:26:23 2025\n"
     ]
    }
   ],
   "source": [
    "# Make sure to use high-quality SNPs\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/CTSB/CTSB_EUR/EUR_CTSB \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.05 \\\n",
    "--hwe 1e-5 0.001 \\\n",
    "--make-bed \\\n",
    "--keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep \\\n",
    "--exclude {WORK_DIR}/exclusion_regions_hg38.txt \\\n",
    "--out {WORK_DIR}/CTSB/CTSB_UNIMPUTED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "059a4dba-f030-49f4-b27e-e0326ae34179",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//CTSB/prune_CTSB.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out /home/jupyter/workspace/ws_files//CTSB/prune_CTSB\n",
      "\n",
      "Start time: Sat Jul 26 13:26:47 2025\n",
      "26046 MiB RAM detected, ~24571 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED.fam.\n",
      "599 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//CTSB/CTSB_UNIMPUTED.bim.\n",
      "1 binary phenotype loaded (24208 cases, 9662 controls).\n",
      "Calculating allele frequencies... done.\n",
      "483/599 variants removed.ead): 0%\n",
      "Variant lists written to\n",
      "/home/jupyter/workspace/ws_files//CTSB/prune_CTSB.prune.in and\n",
      "/home/jupyter/workspace/ws_files//CTSB/prune_CTSB.prune.out .\n",
      "End time: Sat Jul 26 13:26:47 2025\n"
     ]
    }
   ],
   "source": [
    "# Prune out unnecessary SNPs (only need to do this to generate PCs)\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/CTSB/CTSB_UNIMPUTED \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out {WORK_DIR}/CTSB/prune_CTSB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "9f09a466-976b-4746-8f17-c299e0e47156",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "116 /home/jupyter/workspace/ws_files//CTSB/prune_CTSB.prune.in\n"
     ]
    }
   ],
   "source": [
    "!wc -l {WORK_DIR}/CTSB/prune_CTSB.prune.in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "110af728-fb5f-4db4-8090-88fc1819dc44",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 4.31E-4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd52f203-3e15-4b64-83d4-fbdb11134ea5",
   "metadata": {},
   "source": [
    "zoom locus plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "266d2aaf-1ffe-4b83-9bab-81caf65327e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "## locus zoom format\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "\n",
    "    df = pd.read_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')\n",
    "    df['b'] = np.log(df['OR'])\n",
    "    export_ldassoc = df[['#CHROM', 'POS', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P']].copy()\n",
    "    export_ldassoc = export_ldassoc.rename(columns = {'#CHROM': 'CHROM', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p'})\n",
    "    # save to files\n",
    "    export_ldassoc.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.formatted.tab', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81b83d70-c528-4ab2-bb8e-a914adee41ee",
   "metadata": {},
   "source": [
    "GCTA cojo analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "908115ee-aac6-4d29-80a1-219f95bb1614",
   "metadata": {},
   "outputs": [],
   "source": [
    "## cojo format\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "\n",
    "    sumstats = pd.read_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')\n",
    "    #Format summary statistics for GCTA-COJO\n",
    "    #First get the log odds ratio - this is required for COJO\n",
    "    #1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.\n",
    "    sumstats_formatted = sumstats.copy()\n",
    "    sumstats_formatted['b'] = np.log(sumstats_formatted['OR'])\n",
    "\n",
    "    #Now select just the necessary columns for COJO\n",
    "    sumstats_export = sumstats_formatted[['ID', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P', 'OBS_CT']].copy()\n",
    "\n",
    "    #Rename columns following COJO format\n",
    "    sumstats_export = sumstats_export.rename(columns = {'ID':'SNP', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p', 'OBS_CT':'N'})\n",
    "\n",
    "    #Export\n",
    "    sumstats_export.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.sumstats.ma', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "172681eb-a1e0-40cf-8c63-b37c1b3a6a52",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:35 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam].\n",
      "144 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim].\n",
      "1057 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 144 individuals and 1057 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 714 SNPs (343 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 682 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 682 SNPs: 20.8706 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "682 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 682 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:35 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.18 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:36 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam].\n",
      "3608 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim].\n",
      "5507 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 3608 individuals and 5507 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 680 SNPs (4827 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 680 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 680 SNPs: 5.63884 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "680 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 680 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 679 remaining SNPs to [/home/jupyter/workspace/ws_files/CTSB/CTSB_AMR/AMR.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:37 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.86 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:38 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam].\n",
      "1493 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim].\n",
      "3639 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 1493 individuals and 3639 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 687 SNPs (2952 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 690 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 690 SNPs: 5.08782 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "686 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 686 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:38 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.31 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:39 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam].\n",
      "1215 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim].\n",
      "5526 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 1215 individuals and 5526 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1212 SNPs (4314 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1211 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1211 SNPs: 6.13224 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1211 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1211 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:40 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.33 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:41 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam].\n",
      "58823 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim].\n",
      "10388 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 58823 individuals and 10388 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 685 SNPs (9703 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 685 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 685 SNPs: 6.03122 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "685 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 685 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 684 remaining SNPs to [/home/jupyter/workspace/ws_files/CTSB/CTSB_EUR/EUR.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:51 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 9.95 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:52 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim].\n",
      "5508 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 5508 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 964 SNPs (4544 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 974 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 974 SNPs: 5.5938 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "956 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 956 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:52 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.37 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:53 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam].\n",
      "1691 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim].\n",
      "3249 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 1691 individuals and 3249 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 686 SNPs (2563 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 685 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 685 SNPs: 4.43461 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "684 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 684 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:54 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.32 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:55 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam].\n",
      "3472 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim].\n",
      "2256 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 3472 individuals and 2256 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 743 SNPs (1513 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 742 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 742 SNPs: 7.9694 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "740 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 740 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:55 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.32 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:56 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam].\n",
      "945 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim].\n",
      "2801 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 945 individuals and 2801 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 709 SNPs (2092 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 708 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 708 SNPs: 5.6986 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "707 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 707 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:57 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.27 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:29:58 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam].\n",
      "6662 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim].\n",
      "4213 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 6662 individuals and 4213 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 477 SNPs (3736 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 477 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 477 SNPs: 5.30119 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "477 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 477 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:29:59 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.67 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 13:30:00 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000431\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam].\n",
      "3780 individuals to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim].\n",
      "7164 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bed] in SNP-major format ...\n",
      "Genotype data for 3780 individuals and 7164 SNPs to be included from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR_CTSB.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1216 SNPs (5948 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1218 SNPs read from [/home/jupyter/workspace/ws_files/CTSB/CTSB_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1218 SNPs: 4.96202 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1212 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1212 SNPs to select association signals ... (p cutoff = 0.000431; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 13:30:00 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.68 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/CTSB/CTSB_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on LD pruning p-value 4.67e-4\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.95.0-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_CTSB \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 4.31e-4 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.ldprune.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5569d4f-82c5-47ef-bd9c-d6dfa975fba1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
