{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "556590b5-12dc-47f2-b06a-38bc8c822c41",
   "metadata": {},
   "source": [
    "SCARB2 - Single gene analysis in GP2 Neurobooster genotyping data (all ancestries)\n",
    "\n",
    "Project: GP2 SCARB2\n",
    "\n",
    "Version: Python/3.10.15, R/4.3.3\n",
    "\n",
    "Notebook Overview\n",
    "\n",
    "Description Loading Python libraries Set paths Make working directory\n",
    "\n",
    "Installing packages\n",
    "\n",
    "LD pruning(TMEM175/SCARB2/CTSB)\n",
    "\n",
    "Create a covariate file with GP2 data\n",
    "\n",
    "Annotation of the gene SCARB2\n",
    "\n",
    "Association analysis to compare allele frequencies between cases and controls\n",
    "\n",
    "GLM analysis adjusting for gender, age, PC1-5\n",
    "\n",
    "Burden test(SkatO, Skat, cmc,zeggini,mb,fp,cmcWald)\n",
    "\n",
    "Conditional analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edcc13c2-14ba-4b6e-be5d-661a8f76535f",
   "metadata": {},
   "source": [
    "Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d0494f73-f8d8-4c57-8c36-c8805c829f93",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use pathlib for file path manipulation\n",
    "import pathlib\n",
    "\n",
    "# Install numpy\n",
    "import numpy as np\n",
    "\n",
    "# Install Pandas for tabular data\n",
    "import pandas as pd\n",
    "\n",
    "# Install plotnine: a ggplot2-compatible Python plotting package\n",
    "from plotnine import *\n",
    "\n",
    "# Always show all columns in a Pandas DataFrame\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c02d8370-fe89-4d56-a7d6-5775464c402a",
   "metadata": {},
   "source": [
    "Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dfc5d82c-522a-4e40-a1e6-9866ccb9a4a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 100K\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 clinical_data\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 imputed_genotypes\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 meta_data\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 raw_genotypes\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 raw_genotypes_flipped\n",
      "-r--r--r--. 1 jupyter users 100K Jun 30 20:07 README_release10_01072025.txt\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 26 10:56 wgs\n"
     ]
    }
   ],
   "source": [
    "REL10_PATH = pathlib.Path(pathlib.Path.home(), 'workspace/gp2_tier2_eu_release10')\n",
    "!ls -hal {REL10_PATH}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37f07df7-f673-4722-8689-f10508f73245",
   "metadata": {},
   "source": [
    "Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e17775fa-a578-43a3-9810-3a60125a0d95",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir ~/workspace/ws_files/SCARB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d1a4a42e-76e2-41d9-9689-7fcfecf76656",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/SCARB2/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8d877cda-c316-4864-a802-f60fc01bd785",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\t\t\t\t       plink2\n",
      "annovar.latest.tar.gz\t\t       plink2_linux_x86_64_latest.zip\n",
      "gcta-1.95.0-linux-kernel-3-x86_64      plink_linux_x86_64_20190304.zip\n",
      "gcta-1.95.0-linux-kernel-3-x86_64.zip  prettify\n",
      "intel-simplified-software-license.txt  rvtests\n",
      "LICENSE\t\t\t\t       toy.map\n",
      "__MACOSX\t\t\t       toy.ped\n",
      "plink\t\t\t\t       vcf_subset\n"
     ]
    }
   ],
   "source": [
    "# make sure all tools installed\n",
    "! ls /home/jupyter/tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4fafdad0-295c-480b-be96-f94cd8fed64f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# give permission\n",
    "\n",
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "020262a9-5c5c-493a-a10f-bb4e50b94107",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# making working directory\n",
    "#Loop over all the ancestries\n",
    "for ancestry in {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'} ;\n",
    "do\n",
    "\n",
    "#Make a folder for each ancestry\n",
    "mkdir ~/workspace/ws_files/SCARB2/SCARB2_\"$ancestry\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0aa46ee9-e44e-4e12-ab92-2580c643a345",
   "metadata": {},
   "outputs": [],
   "source": [
    "# covariate file has been created in another notebook without GBA carriers"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "13aaaaad-8d58-42db-b4fd-1bd067093d73",
   "metadata": {},
   "source": [
    "Annotation of the gene\n",
    "\n",
    "Extract the region using PLINK\n",
    "\n",
    "Extract SCARB2 gene\n",
    "\n",
    "SCARB2 coordinates: Chromosome 4: 76,158,737-76,234,536(GRCh38/hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4d5ad608-2415-4c31-8ea8-9556b2eec618",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:12 2025\n",
      "26046 MiB RAM detected, ~24954 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "2268 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:14 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:15 2025\n",
      "26046 MiB RAM detected, ~24885 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "1200 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:17 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:18 2025\n",
      "26046 MiB RAM detected, ~24834 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "3010 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:21 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:22 2025\n",
      "26046 MiB RAM detected, ~24824 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "3967 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:24 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:26 2025\n",
      "26046 MiB RAM detected, ~24825 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "4696 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:29 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:31 2025\n",
      "26046 MiB RAM detected, ~24806 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "2681 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:33 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:34 2025\n",
      "26046 MiB RAM detected, ~24806 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "3920 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:37 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:38 2025\n",
      "26046 MiB RAM detected, ~24796 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "1942 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:41 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:42 2025\n",
      "26046 MiB RAM detected, ~24807 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "4269 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:45 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:46 2025\n",
      "26046 MiB RAM detected, ~24790 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "7102 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:52 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 76108737\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "  --to-bp 76284536\n",
      "\n",
      "Start time: Sat Jul 26 11:01:54 2025\n",
      "26046 MiB RAM detected, ~24796 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "3094 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:01:57 2025\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release10_vwb \\\n",
    "    --chr 4 \\\n",
    "    --from-bp 76108737 \\\n",
    "    --to-bp 76284536 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_SCARB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "95832091-f7a1-4f78-97d3-b379bf4512b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}/'\n",
    "    \n",
    "    ! head -n 1 {WORK_DIR}/{ancestry}_SCARB2.fam > {WORK_DIR}/{ancestry}_s1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0168db9-6a6b-4622-bdde-d0436d2af95d",
   "metadata": {},
   "outputs": [],
   "source": [
    "! head /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_s1.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b091d320-cf1c-40b8-a462-b31fb9300d40",
   "metadata": {},
   "source": [
    "Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0feb4774-ce2f-4eef-be6e-cf9c66935a29",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:12 2025\n",
      "26046 MiB RAM detected, ~24848 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam.\n",
      "2268 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.bim\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:14 2025\n",
      "26046 MiB RAM detected, ~24829 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam.\n",
      "1200 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:15 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:16 2025\n",
      "26046 MiB RAM detected, ~24843 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam.\n",
      "3010 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (1 female, 0 males; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:16 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:17 2025\n",
      "26046 MiB RAM detected, ~24835 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam.\n",
      "3967 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (1 female, 0 males; 1 founder) remaining after main filters.\n",
      "0 cases and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:18 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:19 2025\n",
      "26046 MiB RAM detected, ~24822 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam.\n",
      "4696 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:20 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:21 2025\n",
      "26046 MiB RAM detected, ~24831 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam.\n",
      "2681 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:21 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:23 2025\n",
      "26046 MiB RAM detected, ~24854 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam.\n",
      "3920 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:23 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:24 2025\n",
      "26046 MiB RAM detected, ~24843 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam.\n",
      "1942 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.bed ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:04:25 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:26 2025\n",
      "26046 MiB RAM detected, ~24830 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam.\n",
      "4269 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:27 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:28 2025\n",
      "26046 MiB RAM detected, ~24848 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam.\n",
      "7102 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "0 cases and 1 control remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:28 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "  --keep /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_s1.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:04:30 2025\n",
      "26046 MiB RAM detected, ~24860 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam.\n",
      "3094 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--keep: 1 sample remaining.\n",
      "1 sample (0 females, 1 male; 1 founder) remaining after main filters.\n",
      "1 case and 0 controls remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.bed\n",
      "done.%\n",
      "End time: Sat Jul 26 11:04:30 2025\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --keep {WORK_DIR}/{ancestry}_s1.txt \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_SCARB2_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "11a35218-9a9d-4b85-9a77-89190f6d94de",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:15 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24890 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.fam.\n",
      "2268 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:15 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:16 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24851 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.fam.\n",
      "1200 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:17 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:18 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24835 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (1 female, 0 males; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.fam.\n",
      "3010 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:18 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:20 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24827 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (1 female, 0 males; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.fam.\n",
      "3967 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.bim.\n",
      "Note: No phenotype data present.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:20 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:21 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24843 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.fam.\n",
      "4696 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:22 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:23 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24832 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.fam.\n",
      "2681 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:23 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:24 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24818 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.fam.\n",
      "3920 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:25 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:26 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24809 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.fam.\n",
      "1942 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:26 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:27 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24809 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.fam.\n",
      "4269 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.vcdone.\n",
      "End time: Sat Jul 26 11:09:28 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:29 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24803 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.fam.\n",
      "7102 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (0 cases, 1 control).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:29 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1\n",
      "  --export vcf-fid\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1\n",
      "\n",
      "Start time: Sat Jul 26 11:09:30 2025\n",
      "Note: --export 'vcf-fid' modifier is deprecated.  Use 'vcf' + 'id-paste=fid'.\n",
      "26046 MiB RAM detected, ~24802 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1 sample (0 females, 1 male; 1 founder) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.fam.\n",
      "3094 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.bim.\n",
      "1 binary phenotype loaded (1 case, 0 controls).\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.vcf .done.\n",
      "End time: Sat Jul 26 11:09:31 2025\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2_v1 \\\n",
    "    --recode vcf-fid \\\n",
    "    --out {WORK_DIR}/{ancestry}_SCARB2_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e5abe70f-6f15-45c3-a73e-61dea02b58de",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    ! bgzip -f {WORK_DIR}/{ancestry}_SCARB2_v1.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/{ancestry}_SCARB2_v1.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e5d76cdc-51e8-446a-9a25-23f748f833e4",
   "metadata": {},
   "source": [
    "Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d3ef2f9d-63f7-4c72-959d-508e937bb272",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 2275 lines from VCF file\n",
      "NOTICE: A total of 2268 locus in VCF file passed QC threshold, representing 2093 SNPs (1452 transitions and 641 transversions) and 175 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2093 SNP genotypes (1452 transitions and 641 transversions) and 175 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2268 unique variants in 2268 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2268 unique variants in 2268 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 1207 lines from VCF file\n",
      "NOTICE: A total of 1200 locus in VCF file passed QC threshold, representing 1114 SNPs (758 transitions and 356 transversions) and 86 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1114 SNP genotypes (758 transitions and 356 transversions) and 86 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1200 unique variants in 1200 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1200 unique variants in 1200 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 3017 lines from VCF file\n",
      "NOTICE: A total of 3010 locus in VCF file passed QC threshold, representing 2759 SNPs (1924 transitions and 835 transversions) and 251 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2759 SNP genotypes (1924 transitions and 835 transversions) and 251 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3010 unique variants in 3010 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3010 unique variants in 3010 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 3974 lines from VCF file\n",
      "NOTICE: A total of 3967 locus in VCF file passed QC threshold, representing 3613 SNPs (2513 transitions and 1100 transversions) and 354 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3613 SNP genotypes (2513 transitions and 1100 transversions) and 354 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3967 unique variants in 3967 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3967 unique variants in 3967 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 4703 lines from VCF file\n",
      "NOTICE: A total of 4696 locus in VCF file passed QC threshold, representing 4257 SNPs (2943 transitions and 1314 transversions) and 439 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 4257 SNP genotypes (2943 transitions and 1314 transversions) and 439 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 4696 unique variants in 4696 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 4696 unique variants in 4696 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 2688 lines from VCF file\n",
      "NOTICE: A total of 2681 locus in VCF file passed QC threshold, representing 2468 SNPs (1709 transitions and 759 transversions) and 213 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2468 SNP genotypes (1709 transitions and 759 transversions) and 213 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2681 unique variants in 2681 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2681 unique variants in 2681 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 3927 lines from VCF file\n",
      "NOTICE: A total of 3920 locus in VCF file passed QC threshold, representing 3586 SNPs (2504 transitions and 1082 transversions) and 334 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3586 SNP genotypes (2504 transitions and 1082 transversions) and 334 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3920 unique variants in 3920 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3920 unique variants in 3920 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 1949 lines from VCF file\n",
      "NOTICE: A total of 1942 locus in VCF file passed QC threshold, representing 1790 SNPs (1239 transitions and 551 transversions) and 152 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1790 SNP genotypes (1239 transitions and 551 transversions) and 152 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1942 unique variants in 1942 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1942 unique variants in 1942 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 4276 lines from VCF file\n",
      "NOTICE: A total of 4269 locus in VCF file passed QC threshold, representing 3919 SNPs (2742 transitions and 1177 transversions) and 350 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3919 SNP genotypes (2742 transitions and 1177 transversions) and 350 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 4269 unique variants in 4269 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 4269 unique variants in 4269 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 7109 lines from VCF file\n",
      "NOTICE: A total of 7102 locus in VCF file passed QC threshold, representing 6480 SNPs (4498 transitions and 1982 transversions) and 622 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 6480 SNP genotypes (4498 transitions and 1982 transversions) and 622 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 7102 unique variants in 7102 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 7102 unique variants in 7102 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2_v1.vcf.gz > /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.avinput>\n",
      "NOTICE: Finished reading 3101 lines from VCF file\n",
      "NOTICE: A total of 3094 locus in VCF file passed QC threshold, representing 2843 SNPs (2000 transitions and 843 transversions) and 251 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2843 SNP genotypes (2000 transitions and 843 transversions) and 251 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3094 unique variants in 3094 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 23 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3094 unique variants in 3094 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 3\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    ! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_SCARB2_v1.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "    -out {WORK_DIR}/{ancestry}_SCARB2.annovar \\\n",
    "    -remove -protocol refGene,clinvar_20140902 \\\n",
    "    -operation g,f \\\n",
    "    --nopolish \\\n",
    "    -nastring . \\\n",
    "    -vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "bed17082-116c-4566-ab20-a4d20b93bd8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = f'~/workspace/ws_files/SCARB2/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "0b2a9848-0c64-473f-9346-7fb239a10698",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>76108753</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108753</td>\n",
       "      <td>chr4:76108753:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>76108755</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108755</td>\n",
       "      <td>chr4:76108755:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>76108804</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108804</td>\n",
       "      <td>chr4:76108804:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>chr4:76108828:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>76108848</td>\n",
       "      <td>76108848</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>ART3</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76108848</td>\n",
       "      <td>chr4:76108848:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3915</th>\n",
       "      <td>4</td>\n",
       "      <td>76284281</td>\n",
       "      <td>76284281</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284281</td>\n",
       "      <td>chr4:76284281:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3916</th>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>76284298</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284298</td>\n",
       "      <td>chr4:76284298:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3917</th>\n",
       "      <td>4</td>\n",
       "      <td>76284303</td>\n",
       "      <td>76284303</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284303</td>\n",
       "      <td>chr4:76284303:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3918</th>\n",
       "      <td>4</td>\n",
       "      <td>76284327</td>\n",
       "      <td>76284327</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284327</td>\n",
       "      <td>chr4:76284327:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3919</th>\n",
       "      <td>4</td>\n",
       "      <td>76284397</td>\n",
       "      <td>76284397</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>FAM47E-STBD1</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>76284397</td>\n",
       "      <td>chr4:76284397:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3920 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr     Start       End Ref Alt Func.refGene  Gene.refGene  \\\n",
       "0       4  76108753  76108753   T   C     intronic          ART3   \n",
       "1       4  76108755  76108755   T   C     intronic          ART3   \n",
       "2       4  76108804  76108804   T   C     intronic          ART3   \n",
       "3       4  76108828  76108828   G   T     intronic          ART3   \n",
       "4       4  76108848  76108848   C   T     intronic          ART3   \n",
       "...   ...       ...       ...  ..  ..          ...           ...   \n",
       "3915    4  76284281  76284281   C   A     intronic  FAM47E-STBD1   \n",
       "3916    4  76284298  76284298   C   T     intronic  FAM47E-STBD1   \n",
       "3917    4  76284303  76284303   C   A     intronic  FAM47E-STBD1   \n",
       "3918    4  76284327  76284327   C   T     intronic  FAM47E-STBD1   \n",
       "3919    4  76284397  76284397   A   G     intronic  FAM47E-STBD1   \n",
       "\n",
       "     GeneDetail.refGene ExonicFunc.refGene AAChange.refGene clinvar_20140902  \\\n",
       "0                     .                  .                .                .   \n",
       "1                     .                  .                .                .   \n",
       "2                     .                  .                .                .   \n",
       "3                     .                  .                .                .   \n",
       "4                     .                  .                .                .   \n",
       "...                 ...                ...              ...              ...   \n",
       "3915                  .                  .                .                .   \n",
       "3916                  .                  .                .                .   \n",
       "3917                  .                  .                .                .   \n",
       "3918                  .                  .                .                .   \n",
       "3919                  .                  .                .                .   \n",
       "\n",
       "      Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  Otherinfo5  \\\n",
       "0            0.0          .          .           4    76108753   \n",
       "1            0.0          .          .           4    76108755   \n",
       "2            0.0          .          .           4    76108804   \n",
       "3            0.0          .          .           4    76108828   \n",
       "4            0.0          .          .           4    76108848   \n",
       "...          ...        ...        ...         ...         ...   \n",
       "3915         0.0          .          .           4    76284281   \n",
       "3916         0.0          .          .           4    76284298   \n",
       "3917         0.0          .          .           4    76284303   \n",
       "3918         0.0          .          .           4    76284327   \n",
       "3919         0.0          .          .           4    76284397   \n",
       "\n",
       "             Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9 Otherinfo10  \\\n",
       "0     chr4:76108753:T:C          T          C          .           .   \n",
       "1     chr4:76108755:T:C          T          C          .           .   \n",
       "2     chr4:76108804:T:C          T          C          .           .   \n",
       "3     chr4:76108828:G:T          G          T          .           .   \n",
       "4     chr4:76108848:C:T          C          T          .           .   \n",
       "...                 ...        ...        ...        ...         ...   \n",
       "3915  chr4:76284281:C:A          C          A          .           .   \n",
       "3916  chr4:76284298:C:T          C          T          .           .   \n",
       "3917  chr4:76284303:C:A          C          A          .           .   \n",
       "3918  chr4:76284327:C:T          C          T          .           .   \n",
       "3919  chr4:76284397:A:G          A          G          .           .   \n",
       "\n",
       "     Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0             PR          GT         0/0  \n",
       "1             PR          GT         0/0  \n",
       "2             PR          GT         0/0  \n",
       "3             PR          GT         0/0  \n",
       "4             PR          GT         0/0  \n",
       "...          ...         ...         ...  \n",
       "3915          PR          GT         0/0  \n",
       "3916          PR          GT         0/0  \n",
       "3917          PR          GT         0/0  \n",
       "3918          PR          GT         0/0  \n",
       "3919          PR          GT         0/0  \n",
       "\n",
       "[3920 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/SCARB2_CAH/CAH_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "2466b203-5c6f-48d4-9c2e-889144b4ce31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic      3440\n",
       "intergenic     210\n",
       "UTR3            81\n",
       "downstream      72\n",
       "exonic          60\n",
       "upstream        30\n",
       "UTR5            24\n",
       "splicing         3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "457ee957-f3f8-4298-b769-ae69930eb2f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV    33\n",
       "synonymous SNV       26\n",
       "stopgain              1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "33d085a6-58a5-4fc3-980b-e4a4c603113d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: SAS\n",
      "{'SAS'}\n",
      "Total variants:  2268\n",
      "Intronic:  1972\n",
      "UTR3:  59\n",
      "UTR5:  12\n",
      "Total exonic:  35\n",
      "  Synonymous:  15\n",
      "  Nonsynonymous:  20\n",
      "nonsynonymous_lof:  20\n",
      "\n",
      "\n",
      "WORKING ON: FIN\n",
      "{'FIN'}\n",
      "Total variants:  1200\n",
      "Intronic:  1032\n",
      "UTR3:  23\n",
      "UTR5:  3\n",
      "Total exonic:  16\n",
      "  Synonymous:  7\n",
      "  Nonsynonymous:  9\n",
      "nonsynonymous_lof:  9\n",
      "\n",
      "\n",
      "WORKING ON: MDE\n",
      "{'MDE'}\n",
      "Total variants:  3010\n",
      "Intronic:  2656\n",
      "UTR3:  56\n",
      "UTR5:  14\n",
      "Total exonic:  59\n",
      "  Synonymous:  19\n",
      "  Nonsynonymous:  40\n",
      "nonsynonymous_lof:  41\n",
      "\n",
      "\n",
      "WORKING ON: AAC\n",
      "{'AAC'}\n",
      "Total variants:  3967\n",
      "Intronic:  3493\n",
      "UTR3:  93\n",
      "UTR5:  21\n",
      "Total exonic:  63\n",
      "  Synonymous:  27\n",
      "  Nonsynonymous:  36\n",
      "nonsynonymous_lof:  37\n",
      "\n",
      "\n",
      "WORKING ON: AFR\n",
      "{'AFR'}\n",
      "Total variants:  4696\n",
      "Intronic:  4111\n",
      "UTR3:  95\n",
      "UTR5:  34\n",
      "Total exonic:  75\n",
      "  Synonymous:  35\n",
      "  Nonsynonymous:  40\n",
      "nonsynonymous_lof:  42\n",
      "\n",
      "\n",
      "WORKING ON: CAS\n",
      "{'CAS'}\n",
      "Total variants:  2681\n",
      "Intronic:  2356\n",
      "UTR3:  59\n",
      "UTR5:  18\n",
      "Total exonic:  50\n",
      "  Synonymous:  22\n",
      "  Nonsynonymous:  28\n",
      "nonsynonymous_lof:  29\n",
      "\n",
      "\n",
      "WORKING ON: CAH\n",
      "{'CAH'}\n",
      "Total variants:  3920\n",
      "Intronic:  3440\n",
      "UTR3:  81\n",
      "UTR5:  24\n",
      "Total exonic:  60\n",
      "  Synonymous:  27\n",
      "  Nonsynonymous:  33\n",
      "nonsynonymous_lof:  34\n",
      "\n",
      "\n",
      "WORKING ON: AJ\n",
      "{'AJ'}\n",
      "Total variants:  1942\n",
      "Intronic:  1690\n",
      "UTR3:  46\n",
      "UTR5:  8\n",
      "Total exonic:  30\n",
      "  Synonymous:  11\n",
      "  Nonsynonymous:  19\n",
      "nonsynonymous_lof:  19\n",
      "\n",
      "\n",
      "WORKING ON: AMR\n",
      "{'AMR'}\n",
      "Total variants:  4269\n",
      "Intronic:  3771\n",
      "UTR3:  87\n",
      "UTR5:  18\n",
      "Total exonic:  71\n",
      "  Synonymous:  25\n",
      "  Nonsynonymous:  46\n",
      "nonsynonymous_lof:  46\n",
      "\n",
      "\n",
      "WORKING ON: EUR\n",
      "{'EUR'}\n",
      "Total variants:  7102\n",
      "Intronic:  6189\n",
      "UTR3:  166\n",
      "UTR5:  49\n",
      "Total exonic:  140\n",
      "  Synonymous:  52\n",
      "  Nonsynonymous:  88\n",
      "nonsynonymous_lof:  92\n",
      "\n",
      "\n",
      "WORKING ON: EAS\n",
      "{'EAS'}\n",
      "Total variants:  3094\n",
      "Intronic:  2681\n",
      "UTR3:  84\n",
      "UTR5:  23\n",
      "Total exonic:  59\n",
      "  Synonymous:  22\n",
      "  Nonsynonymous:  37\n",
      "nonsynonymous_lof:  39\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Make lists of variants to keep - all coding, coding nonsynonymous (missense - as they are coded in ANNOVAR), deleterious (CADD_phred > 20)\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "\n",
    "    # Read in ANNOVAR multianno file\n",
    "    gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_SCARB2.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "    \n",
    "    #Print number of variants in the different categories\n",
    "    results = [] \n",
    "\n",
    "    utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "    intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "    exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "    utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "    coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "    coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'nonsynonymous SNV')]\n",
    "    lof = exonic[(exonic['ExonicFunc.refGene'] == 'stopgain') | (exonic['ExonicFunc.refGene'] == 'stoploss') | (exonic['ExonicFunc.refGene'] == 'frameshift deletion') | (exonic['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "    nonsynonymous_lof = pd.concat([coding_nonsynonymous, lof])\n",
    "\n",
    "    print({ancestry})\n",
    "    print('Total variants: ', len(gene))\n",
    "    print(\"Intronic: \", len(intronic))\n",
    "    print('UTR3: ', len(utr3))\n",
    "    print('UTR5: ', len(utr5))\n",
    "    print(\"Total exonic: \", len(exonic))\n",
    "    print('  Synonymous: ', len(coding_synonymous))\n",
    "    print(\"  Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "    print(\"nonsynonymous_lof: \", len(nonsynonymous_lof))\n",
    "    results.append((gene, intronic, utr3, utr5, exonic, coding_synonymous, coding_nonsynonymous, nonsynonymous_lof))\n",
    "    print('\\n')\n",
    "\n",
    "    # Save in PLINK format - coding nonsynonymous \n",
    "    # These are missense variants - other types of nonsynonymous variants (e.g stopgain/loss, or frameshift variants are coded differently in the ExonicFunc.refGene \n",
    "    variants_toKeep = nonsynonymous_lof[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_SCARB2.nonsynonymous_lof.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "\n",
    "\n",
    "    # Save in PLINK format - all coding variants\n",
    "    variants_toKeep2 = exonic[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry}_SCARB2.exonic.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15708b1c-0cf9-41b4-b1d9-cdb080a04cc7",
   "metadata": {},
   "source": [
    "ALL variants\n",
    "\n",
    "assoc\n",
    "\n",
    "glm\n",
    "\n",
    "ASSOC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "ba6e9221-bfe9-4aec-bf5e-b4a4bab3b61b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3967 variants loaded from .bim file.\n",
      "1215 people (532 males, 683 females) loaded from .fam.\n",
      "1133 phenotype values loaded from .fam.\n",
      "--keep: 1207 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1207 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies done.\n",
      "Total genotyping rate in remaining samples is 0.997423.\n",
      "2526 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1441 variants and 1207 people pass filters and QC.\n",
      "Among remaining phenotypes, 369 are cases and 756 are controls.  (82 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "4696 variants loaded from .bim file.\n",
      "3780 people (2087 males, 1693 females) loaded from .fam.\n",
      "3498 phenotype values loaded from .fam.\n",
      "--keep: 3581 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.995675.\n",
      "3240 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1456 variants and 3581 people pass filters and QC.\n",
      "Among remaining phenotypes, 1147 are cases and 2169 are controls.  (265\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "1942 variants loaded from .bim file.\n",
      "3472 people (2146 males, 1326 females) loaded from .fam.\n",
      "1966 phenotype values loaded from .fam.\n",
      "--keep: 2716 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2716 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.99958.\n",
      "913 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1029 variants and 2716 people pass filters and QC.\n",
      "Among remaining phenotypes, 1285 are cases and 408 are controls.  (1023\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "4269 variants loaded from .bim file.\n",
      "3608 people (1735 males, 1873 females) loaded from .fam.\n",
      "3434 phenotype values loaded from .fam.\n",
      "--keep: 3471 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3471 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.998722.\n",
      "3352 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "917 variants and 3471 people pass filters and QC.\n",
      "Among remaining phenotypes, 1917 are cases and 1402 are controls.  (152\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "2681 variants loaded from .bim file.\n",
      "1691 people (737 males, 954 females) loaded from .fam.\n",
      "1408 phenotype values loaded from .fam.\n",
      "--keep: 1518 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1518 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies done.\n",
      "Total genotyping rate in remaining samples is 0.997086.\n",
      "1714 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "967 variants and 1518 people pass filters and QC.\n",
      "Among remaining phenotypes, 734 are cases and 578 are controls.  (206\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3094 variants loaded from .bim file.\n",
      "6662 people (4001 males, 2661 females) loaded from .fam.\n",
      "5116 phenotype values loaded from .fam.\n",
      "--keep: 6414 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 6414 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies done.\n",
      "Total genotyping rate in remaining samples is 0.991202.\n",
      "2145 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "949 variants and 6414 people pass filters and QC.\n",
      "Among remaining phenotypes, 2377 are cases and 2598 are controls.  (1439\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "7102 variants loaded from .bim file.\n",
      "58823 people (33074 males, 25749 females) loaded from .fam.\n",
      "37150 phenotype values loaded from .fam.\n",
      "--keep: 53641 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 53641 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies done.\n",
      "Total genotyping rate in remaining samples is 0.796001.\n",
      "6182 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "920 variants and 53641 people pass filters and QC.\n",
      "Among remaining phenotypes, 24208 are cases and 9662 are controls.  (19771\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "1200 variants loaded from .bim file.\n",
      "144 people (62 males, 82 females) loaded from .fam.\n",
      "118 phenotype values loaded from .fam.\n",
      "--keep: 111 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.998551.\n",
      "268 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "932 variants and 111 people pass filters and QC.\n",
      "Among remaining phenotypes, 80 are cases and 11 are controls.  (20 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3010 variants loaded from .bim file.\n",
      "1493 people (815 males, 678 females) loaded from .fam.\n",
      "1311 phenotype values loaded from .fam.\n",
      "--keep: 1448 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1448 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.997719.\n",
      "2038 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "972 variants and 1448 people pass filters and QC.\n",
      "Among remaining phenotypes, 724 are cases and 551 are controls.  (173\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "2268 variants loaded from .bim file.\n",
      "945 people (606 males, 339 females) loaded from .fam.\n",
      "586 phenotype values loaded from .fam.\n",
      "--keep: 870 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 870 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.997281.\n",
      "1358 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "910 variants and 870 people pass filters and QC.\n",
      "Among remaining phenotypes, 309 are cases and 261 are controls.  (300\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3920 variants loaded from .bim file.\n",
      "1111 people (589 males, 522 females) loaded from .fam.\n",
      "1040 phenotype values loaded from .fam.\n",
      "--keep: 1039 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1039 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies done.\n",
      "Total genotyping rate in remaining samples is 0.998878.\n",
      "2693 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1227 variants and 1039 people pass filters and QC.\n",
      "Among remaining phenotypes, 663 are cases and 318 are controls.  (58 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis using plink assoc for all variants, not adjusting for any covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    ! /home/jupyter/tools/plink \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "    --assoc \\\n",
    "    --allow-no-sex \\\n",
    "    --ci 0.95 \\\n",
    "    --maf 0.01 \\\n",
    "    --out {WORK_DIR}/{ancestry}_SCARB2.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "eca648b0-934f-4af0-884d-30faeba4fdcf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "Variants with p-value < 0.05: (236, 13)\n",
      "WORKING ON: AFR\n",
      "Variants with p-value < 0.05: (70, 13)\n",
      "WORKING ON: AJ\n",
      "Variants with p-value < 0.05: (61, 13)\n",
      "WORKING ON: AMR\n",
      "Variants with p-value < 0.05: (740, 13)\n",
      "WORKING ON: CAS\n",
      "Variants with p-value < 0.05: (28, 13)\n",
      "WORKING ON: EAS\n",
      "Variants with p-value < 0.05: (418, 13)\n",
      "WORKING ON: EUR\n",
      "Variants with p-value < 0.05: (270, 13)\n",
      "WORKING ON: FIN\n",
      "Variants with p-value < 0.05: (5, 13)\n",
      "WORKING ON: MDE\n",
      "Variants with p-value < 0.05: (225, 13)\n",
      "WORKING ON: SAS\n",
      "Variants with p-value < 0.05: (4, 13)\n",
      "WORKING ON: CAH\n",
      "Variants with p-value < 0.05: (40, 13)\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Look at assoc results, only variants with nominal p-value < 0.05\n",
    "    freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_SCARB2.all.assoc', sep='\\s+')\n",
    "    sig_all_nonadj = freq[freq['P']<0.05]\n",
    "    \n",
    "    print(f'Variants with p-value < 0.05: {sig_all_nonadj.shape}')\n",
    "    \n",
    "    #Save FREQ to csv\n",
    "    freq.to_csv(f'{WORK_DIR}/{ancestry}.all_nonadj.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "0165bb23-84bb-45c6-9e17-fd1d9a0a76be",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:21 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24828 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam.\n",
      "3967 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--keep: 1207 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.noNA.txt.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2526 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1441 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:22 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:23 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24812 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam.\n",
      "4696 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--keep: 3581 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.noNA.txt.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3240 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1456 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:24 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:25 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24820 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam.\n",
      "1942 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--keep: 2716 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.noNA.txt.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "913 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1029 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:25 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:26 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24812 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam.\n",
      "4269 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.noNA.txt.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "done.lating allele frequencies... 0%\n",
      "3352 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "917 variants remaining after main filters.\n",
      "Error: Cannot proceed with --glm regression on phenotype 'PHENO1', since\n",
      "variance inflation factor for covariate 'PC2' is too high (VIF_TOO_HIGH). You\n",
      "may want to remove redundant covariates and try again.\n",
      "End time: Sat Jul 26 11:24:27 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:28 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24830 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam.\n",
      "2681 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--keep: 1518 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.noNA.txt.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1714 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "967 variants remaining after main filters.\n",
      "done. logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:28 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:29 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24807 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam.\n",
      "3094 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--keep: 6414 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.noNA.txt.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2145 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "949 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:30 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:31 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24841 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam.\n",
      "7102 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--keep: 53641 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.noNA.txt.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "done.lating allele frequencies... 0%\n",
      "6182 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "920 variants remaining after main filters.\n",
      "done. logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:34 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:35 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24831 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam.\n",
      "1200 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--keep: 111 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.noNA.txt.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "268 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "932 variants remaining after main filters.\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\n",
      "phenotype 'PHENO1'.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:36 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:37 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24841 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam.\n",
      "3010 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--keep: 1448 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.noNA.txt.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2038 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "972 variants remaining after main filters.\n",
      "done. logistic-Firth hybrid regression on phenotype 'PHENO1': 0%\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:37 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:39 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24842 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam.\n",
      "2268 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--keep: 870 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.noNA.txt.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1358 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "910 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:39 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:24:40 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24852 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam.\n",
      "3920 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--keep: 1039 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.noNA.txt.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2693 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1227 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:24:41 2025\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis with covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "    --allow-no-sex \\\n",
    "    --maf 0.01 \\\n",
    "    --ci 0.95 \\\n",
    "    --glm \\\n",
    "    --covar /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.noNA.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --covar-variance-standardize \\\n",
    "    --neg9-pheno-really-missing \\\n",
    "    --out {WORK_DIR}/{ancestry}_SCARB2.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "fcc9659f-6d27-4593-a879-5c3f7394f313",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj\n",
      "\n",
      "Start time: Sat Jul 26 11:25:41 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24813 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam.\n",
      "4269 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3352 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "917 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Sat Jul 26 11:25:42 2025\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_AMR'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/AMR_SCARB2 \\\n",
    "--keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep \\\n",
    "--allow-no-sex \\\n",
    "--maf 0.01 \\\n",
    "--ci 0.95 \\\n",
    "--glm \\\n",
    "--covar /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--neg9-pheno-really-missing \\\n",
    "--out {WORK_DIR}/AMR_SCARB2.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "2fda6de6-0d72-4c0e-a1bb-d0af34e04868",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "There are 248 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AFR\n",
      "There are 209 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "There are 52 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AMR\n",
      "There are 167 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: CAS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n",
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 3 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: EAS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 62 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: EUR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 258 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: FIN\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 0 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: MDE\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 110 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: SAS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 11 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n",
      "WORKING ON: CAH\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_454/1758219659.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 54 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    }
   ],
   "source": [
    "#Process results from plink glm analysis including covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Read in plink glm results\n",
    "    assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_SCARB2.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "\n",
    "    #Filter for additive test only - this is the variant results\n",
    "    assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    significant = assoc_add[assoc_add['P']<0.05]\n",
    "\n",
    "    print(f'There are {len(significant)} variants with p-value < 0.05 in glm')\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    GWsignificant = assoc_add[assoc_add['P']<5e-8]\n",
    "\n",
    "    print(f'There are {len(GWsignificant)} variants with p-value < 5e-8 in glm')\n",
    "    \n",
    "    #Save assoc_add to csv\n",
    "    assoc_add.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7071ece2-65fe-4f12-a936-58e065e54cf4",
   "metadata": {},
   "source": [
    "format assoc file to do meta analysis(AFR/AJ/AMR/EAS/EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "430d5f9c-3fc4-436a-b57e-7ac2a7964ab5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define ancestry groups and their sample sizes\n",
    "ancestry_sample_sizes = {\n",
    "    'AFR': 3316,\n",
    "    'AJ': 1693,\n",
    "    'AMR': 3319,\n",
    "    'EAS': 4975,\n",
    "    'EUR': 33870\n",
    "}\n",
    "\n",
    "# Output container (optional)\n",
    "cleaned_results = {}\n",
    "\n",
    "# Loop over each ancestry\n",
    "for ancestry, sample_size in ancestry_sample_sizes.items():\n",
    "    # Load PLINK association result file\n",
    "    file_path = f'/home/jupyter/workspace/ws_files/SCARB2/SCARB2_{ancestry}/{ancestry}.all_adj.csv'\n",
    "    df = pd.read_csv(file_path)\n",
    "\n",
    "    # Build MARKERNAME\n",
    "    df['MARKERNAME'] = 'chr' + df['#CHROM'].astype(str) + ':' + df['POS'].astype(str)\n",
    "\n",
    "    # Rename columns\n",
    "    df.rename(columns={\n",
    "        '#CHROM': 'CHROMOSOME',\n",
    "        'POS': 'POSITION',\n",
    "        'REF': 'NEA',\n",
    "        'ALT': 'EA',\n",
    "        'A1_FREQ': 'EAF',\n",
    "        'LOG(OR)_SE': 'SE',\n",
    "        'L95': 'OR_95L',\n",
    "        'U95': 'OR_95U'\n",
    "    }, inplace=True)\n",
    "\n",
    "    # Add BETA = log(OR)\n",
    "    df['BETA'] = np.log(df['OR'])\n",
    "\n",
    "    # Add sample size\n",
    "    df['N'] = sample_size\n",
    "\n",
    "    # Keep desired columns\n",
    "    final_cols = [\n",
    "        'MARKERNAME', 'CHROMOSOME', 'POSITION', 'NEA', 'EA', 'EAF',\n",
    "        'BETA', 'OR', 'SE', 'OR_95L', 'OR_95U', 'P', 'N'\n",
    "    ]\n",
    "    df_cleaned = df[final_cols]\n",
    "\n",
    "    # Store or save\n",
    "    cleaned_results[ancestry] = df_cleaned\n",
    "\n",
    "    # Optional: save to file\n",
    "    output_path = f'/home/jupyter/workspace/ws_files/SCARB2/SCARB2_{ancestry}/{ancestry}.cleaned_assoc.txt.gz'\n",
    "    df_cleaned.to_csv(output_path, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "10d2a82a-72e0-43dc-afeb-a7d4db6a2344",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MARKERNAME</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>OR_95L</th>\n",
       "      <th>OR_95U</th>\n",
       "      <th>P</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4:76108828</td>\n",
       "      <td>4</td>\n",
       "      <td>76108828</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.488445</td>\n",
       "      <td>-0.072232</td>\n",
       "      <td>0.930315</td>\n",
       "      <td>0.103227</td>\n",
       "      <td>0.759911</td>\n",
       "      <td>1.13893</td>\n",
       "      <td>0.484090</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4:76108858</td>\n",
       "      <td>4</td>\n",
       "      <td>76108858</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.488095</td>\n",
       "      <td>-0.072979</td>\n",
       "      <td>0.929620</td>\n",
       "      <td>0.103190</td>\n",
       "      <td>0.759399</td>\n",
       "      <td>1.13800</td>\n",
       "      <td>0.479424</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4:76108994</td>\n",
       "      <td>4</td>\n",
       "      <td>76108994</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.191527</td>\n",
       "      <td>-0.026174</td>\n",
       "      <td>0.974166</td>\n",
       "      <td>0.132669</td>\n",
       "      <td>0.751113</td>\n",
       "      <td>1.26346</td>\n",
       "      <td>0.843604</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4:76109007</td>\n",
       "      <td>4</td>\n",
       "      <td>76109007</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.191176</td>\n",
       "      <td>-0.027505</td>\n",
       "      <td>0.972870</td>\n",
       "      <td>0.132660</td>\n",
       "      <td>0.750126</td>\n",
       "      <td>1.26176</td>\n",
       "      <td>0.835750</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4:76109102</td>\n",
       "      <td>4</td>\n",
       "      <td>76109102</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.191176</td>\n",
       "      <td>-0.027505</td>\n",
       "      <td>0.972870</td>\n",
       "      <td>0.132660</td>\n",
       "      <td>0.750126</td>\n",
       "      <td>1.26176</td>\n",
       "      <td>0.835750</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1024</th>\n",
       "      <td>chr4:76281999</td>\n",
       "      <td>4</td>\n",
       "      <td>76281999</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.336720</td>\n",
       "      <td>0.113248</td>\n",
       "      <td>1.119910</td>\n",
       "      <td>0.113801</td>\n",
       "      <td>0.896019</td>\n",
       "      <td>1.39975</td>\n",
       "      <td>0.319653</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1025</th>\n",
       "      <td>chr4:76282260</td>\n",
       "      <td>4</td>\n",
       "      <td>76282260</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.502102</td>\n",
       "      <td>-0.021202</td>\n",
       "      <td>0.979021</td>\n",
       "      <td>0.105906</td>\n",
       "      <td>0.795508</td>\n",
       "      <td>1.20487</td>\n",
       "      <td>0.841327</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1026</th>\n",
       "      <td>chr4:76283245</td>\n",
       "      <td>4</td>\n",
       "      <td>76283245</td>\n",
       "      <td>CAT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.111773</td>\n",
       "      <td>-0.284414</td>\n",
       "      <td>0.752455</td>\n",
       "      <td>0.163238</td>\n",
       "      <td>0.546426</td>\n",
       "      <td>1.03617</td>\n",
       "      <td>0.081451</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1027</th>\n",
       "      <td>chr4:76283347</td>\n",
       "      <td>4</td>\n",
       "      <td>76283347</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.240533</td>\n",
       "      <td>0.015204</td>\n",
       "      <td>1.015320</td>\n",
       "      <td>0.126539</td>\n",
       "      <td>0.792303</td>\n",
       "      <td>1.30110</td>\n",
       "      <td>0.904384</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1028</th>\n",
       "      <td>chr4:76283417</td>\n",
       "      <td>4</td>\n",
       "      <td>76283417</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.011555</td>\n",
       "      <td>-0.461326</td>\n",
       "      <td>0.630447</td>\n",
       "      <td>0.441905</td>\n",
       "      <td>0.265154</td>\n",
       "      <td>1.49899</td>\n",
       "      <td>0.296510</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1029 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         MARKERNAME  CHROMOSOME  POSITION  NEA EA       EAF      BETA  \\\n",
       "0     chr4:76108828           4  76108828    G  T  0.488445 -0.072232   \n",
       "1     chr4:76108858           4  76108858    G  A  0.488095 -0.072979   \n",
       "2     chr4:76108994           4  76108994    G  A  0.191527 -0.026174   \n",
       "3     chr4:76109007           4  76109007    G  A  0.191176 -0.027505   \n",
       "4     chr4:76109102           4  76109102    C  T  0.191176 -0.027505   \n",
       "...             ...         ...       ...  ... ..       ...       ...   \n",
       "1024  chr4:76281999           4  76281999    A  G  0.336720  0.113248   \n",
       "1025  chr4:76282260           4  76282260    C  G  0.502102 -0.021202   \n",
       "1026  chr4:76283245           4  76283245  CAT  C  0.111773 -0.284414   \n",
       "1027  chr4:76283347           4  76283347    C  T  0.240533  0.015204   \n",
       "1028  chr4:76283417           4  76283417    C  T  0.011555 -0.461326   \n",
       "\n",
       "            OR        SE    OR_95L   OR_95U         P     N  \n",
       "0     0.930315  0.103227  0.759911  1.13893  0.484090  1693  \n",
       "1     0.929620  0.103190  0.759399  1.13800  0.479424  1693  \n",
       "2     0.974166  0.132669  0.751113  1.26346  0.843604  1693  \n",
       "3     0.972870  0.132660  0.750126  1.26176  0.835750  1693  \n",
       "4     0.972870  0.132660  0.750126  1.26176  0.835750  1693  \n",
       "...        ...       ...       ...      ...       ...   ...  \n",
       "1024  1.119910  0.113801  0.896019  1.39975  0.319653  1693  \n",
       "1025  0.979021  0.105906  0.795508  1.20487  0.841327  1693  \n",
       "1026  0.752455  0.163238  0.546426  1.03617  0.081451  1693  \n",
       "1027  1.015320  0.126539  0.792303  1.30110  0.904384  1693  \n",
       "1028  0.630447  0.441905  0.265154  1.49899  0.296510  1693  \n",
       "\n",
       "[1029 rows x 13 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.cleaned_assoc.txt.gz', sep='\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fdd2d80c-6b94-421a-8265-a52b9c575293",
   "metadata": {},
   "source": [
    "Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "ed96dc93-d4f7-408f-aba8-99fc7ae2b568",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running plink to extract exonic variants for ancestry: AAC\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:34:53 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24892 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--extract bed1: 4117680 variants excluded.\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "63 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:34:54 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AAC\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AAC\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:34:58 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24839 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--extract bed1: 4117705 variants excluded.\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "38 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 11:34:59 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AAC\n",
      "Running plink to extract exonic variants for ancestry: AFR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:02 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24821 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--extract bed1: 5095236 variants excluded.\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "74 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:03 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AFR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AFR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:07 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24834 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--extract bed1: 5095268 variants excluded.\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "42 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:08 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AFR\n",
      "Running plink to extract exonic variants for ancestry: AJ\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:11 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24824 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--extract bed1: 1752651 variants excluded.\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "30 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:12 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AJ\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AJ\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:15 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24824 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--extract bed1: 1752662 variants excluded.\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "19 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 11:35:16 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AJ\n",
      "Running plink to extract exonic variants for ancestry: AMR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:19 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24830 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--extract bed1: 4434593 variants excluded.\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "71 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:20 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AMR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AMR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:24 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24821 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--extract bed1: 4434618 variants excluded.\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "46 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:24 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AMR\n",
      "Running plink to extract exonic variants for ancestry: CAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:28 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24810 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--extract bed1: 2741479 variants excluded.\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "49 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:29 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:32 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24830 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--extract bed1: 2741499 variants excluded.\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "29 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 11:35:33 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAS\n",
      "Running plink to extract exonic variants for ancestry: EAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:36 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24815 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--extract bed1: 3379711 variants excluded.\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "58 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:37 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: EAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:40 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24817 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--extract bed1: 3379731 variants excluded.\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "38 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 11:35:41 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EAS\n",
      "Running plink to extract exonic variants for ancestry: EUR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:45 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24836 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--extract bed1: 7744952 variants excluded.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "137 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:46 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: EUR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EUR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:50 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24850 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--extract bed1: 7744998 variants excluded.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "91 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:52 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EUR\n",
      "Running plink to extract exonic variants for ancestry: FIN\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:55 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24825 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--extract bed1: 939033 variants excluded.\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "16 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:35:56 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: FIN\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: FIN\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:35:59 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24830 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--extract bed1: 939040 variants excluded.\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "9 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 11:36:00 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: FIN\n",
      "Running plink to extract exonic variants for ancestry: MDE\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:36:03 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24826 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--extract bed1: 3023006 variants excluded.\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "58 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:36:04 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: MDE\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: MDE\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:36:08 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24813 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--extract bed1: 3023024 variants excluded.\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "40 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 11:36:08 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: MDE\n",
      "Running plink to extract exonic variants for ancestry: SAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:36:12 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24836 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--extract bed1: 2227552 variants excluded.\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "35 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:36:12 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: SAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: SAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:36:16 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24814 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--extract bed1: 2227567 variants excluded.\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "20 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Sat Jul 26 11:36:16 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: SAS\n",
      "Running plink to extract exonic variants for ancestry: CAH\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:36:20 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24825 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--extract bed1: 4063100 variants excluded.\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "60 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.exonic.vcf ...\n",
      "done.\n",
      "End time: Sat Jul 26 11:36:21 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAH\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAH\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "\n",
      "Start time: Sat Jul 26 11:36:24 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24824 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--extract bed1: 4063126 variants excluded.\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "34 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Sat Jul 26 11:36:25 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAH\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "#Loop over all the ancestries and the 2 variant classes - run rvtests for all coding and missense variants\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "\n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running plink to extract {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        #Extract relevant variants\n",
    "        ! /home/jupyter/tools/plink2 \\\n",
    "        --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release10_vwb \\\n",
    "        --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "        --extract range {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.variantstoKeep.txt \\\n",
    "        --recode vcf-iid \\\n",
    "        --out {WORK_DIR}/{ancestry}_SCARB2.{variant_class}\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running bgzip and tabix for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## Bgzip and Tabix (zip and index the file)\n",
    "        ! bgzip -f {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.vcf\n",
    "        ! tabix -f -p vcf {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d31c4495-100f-41ff-83ec-7de0e6ea08de",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Run RVtests\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "\n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running RVtests for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## RVtests with covariates \n",
    "        #Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "        ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "        --out {WORK_DIR}/{ancestry}_SCARB2.burden.{variant_class} \\\n",
    "        --kernel skato \\\n",
    "        --inVcf {WORK_DIR}/{ancestry}_SCARB2.{variant_class}.vcf.gz \\\n",
    "        --pheno /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt \\\n",
    "        --pheno-name PHENO \\\n",
    "        --gene SCARB2 \\\n",
    "        --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "        --covar /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt \\\n",
    "        --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "        --freqUpper 0.01\n",
    "# --burden cmc,zeggini,mb,fp,cmcWald --kernel skat,skato \\"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "811dd7a6-7681-4355-8c66-08d1ec2f0880",
   "metadata": {},
   "source": [
    "Look at RVtest results SKAT-O"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "bac609c3-454e-41ff-a0ae-1dee0243f33b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/rpy2/rinterface/__init__.py:1211: UserWarning: Environment variable \"R_SESSION_TMPDIR\" redefined by R and overriding existing variable. Current: \"/tmp/RtmptiJLBA\", R: \"/tmp/Rtmp0H0uj8\"\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "eab6a012-17c6-437f-a2b7-bd7f66bde86a",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/SCARB2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "a096d0af-a990-41f8-be13-c0efa151c238",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t40128\t43\t39\t147695\t0\t0.520947\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "6d2fa0f9-c7b3-4c70-89b9-6cebbbe23afe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t40128\t27\t24\t108275\t0\t0.536357\n"
     ]
    }
   ],
   "source": [
    "#Check EUR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_EUR/EUR_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "8e8b6eea-9f25-4c0a-a375-dedb4251eb6a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t1061\t21\t18\t24584.7\t0\t0.00187003\n"
     ]
    }
   ],
   "source": [
    "#Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AAC/AAC_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "93b0d4ee-de92-47a4-a505-14ad32ba73a7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t1061\t12\t9\t7912.68\t0\t0.0500565\n"
     ]
    }
   ],
   "source": [
    "#Check AAC nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_AAC/AAC_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea460af1-8774-4c17-8066-e3fd791dc18d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# run skat test in AAC ancestry in EXONIC\n",
    "#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_AAC'\n",
    "\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/AAC_SCARB2.SKAT.exonic \\\n",
    "--kernel skat \\\n",
    "--inVcf {WORK_DIR}/AAC_SCARB2.exonic.vcf.gz \\\n",
    "--pheno /home/jupyter/workspace/ws_files/AAC/AAC_covariate_file_noGBA.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene SCARB2 \\\n",
    "--geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "--covar /home/jupyter/workspace/ws_files/AAC/AAC_covariate_file_noGBA.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "4a2e4e86-4939-464c-9efb-85f8de452498",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t1061\t21\t18\t3.83605e+06\t0.000890497\t10000\t10000\t3.83605e+06\t485\t0\t0.0485\n"
     ]
    }
   ],
   "source": [
    "# look at result\n",
    "! cat {WORK_DIR}/AAC_SCARB2.SKAT.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9abed024-39fc-40d7-87d6-843f1b28207d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# run single variant burden analysis\n",
    "# single variant analysis\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--inVcf {WORK_DIR}/AAC_SCARB2.exonic.vcf.gz \\\n",
    "--single score \\\n",
    "--pheno /home/jupyter/workspace/ws_files/AAC/AAC_covariate_file_noGBA.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "--gene SCARB2 \\\n",
    "--out {WORK_DIR}/AAC_SCARB2.burdenTESTSV.exonic \\\n",
    "--covar /home/jupyter/workspace/ws_files/AAC/AAC_covariate_file_noGBA.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01c0fb26-8977-41e2-8ee0-edcec3f3671c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# look at results from single variant analysis\n",
    "! cat {WORK_DIR}/AAC_SCARB2.burdenTESTSV.exonic.SingleScore.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "754a0171-b962-4f05-9336-ec0baa0a8225",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>N_INFORMATIVE</th>\n",
       "      <th>AF</th>\n",
       "      <th>U</th>\n",
       "      <th>V</th>\n",
       "      <th>STAT</th>\n",
       "      <th>DIRECTION</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>SE</th>\n",
       "      <th>PVALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76161743</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-3.094480</td>\n",
       "      <td>77.2455</td>\n",
       "      <td>0.123966</td>\n",
       "      <td>-</td>\n",
       "      <td>-3.122440</td>\n",
       "      <td>8.86837</td>\n",
       "      <td>7.247730e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76163295</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>1.264820</td>\n",
       "      <td>74.1873</td>\n",
       "      <td>0.021564</td>\n",
       "      <td>+</td>\n",
       "      <td>1.328860</td>\n",
       "      <td>9.04931</td>\n",
       "      <td>8.832530e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76163320</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76163352</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76174287</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.001414</td>\n",
       "      <td>51.749700</td>\n",
       "      <td>232.2780</td>\n",
       "      <td>11.529400</td>\n",
       "      <td>+</td>\n",
       "      <td>17.365200</td>\n",
       "      <td>5.11419</td>\n",
       "      <td>6.850250e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76176451</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>47.851200</td>\n",
       "      <td>77.4252</td>\n",
       "      <td>29.573500</td>\n",
       "      <td>+</td>\n",
       "      <td>48.171500</td>\n",
       "      <td>8.85807</td>\n",
       "      <td>5.383610e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179562</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000943</td>\n",
       "      <td>-2.068100</td>\n",
       "      <td>155.2640</td>\n",
       "      <td>0.027547</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.038200</td>\n",
       "      <td>6.25526</td>\n",
       "      <td>8.681790e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179643</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.001414</td>\n",
       "      <td>49.205000</td>\n",
       "      <td>231.8630</td>\n",
       "      <td>10.442100</td>\n",
       "      <td>+</td>\n",
       "      <td>16.540900</td>\n",
       "      <td>5.11876</td>\n",
       "      <td>1.231740e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179654</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>-8.060090</td>\n",
       "      <td>301.0740</td>\n",
       "      <td>0.215778</td>\n",
       "      <td>-</td>\n",
       "      <td>-2.086640</td>\n",
       "      <td>4.49204</td>\n",
       "      <td>6.422760e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179673</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-1.287390</td>\n",
       "      <td>77.6605</td>\n",
       "      <td>0.021341</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.292080</td>\n",
       "      <td>8.84464</td>\n",
       "      <td>8.838530e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179684</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.001885</td>\n",
       "      <td>45.787600</td>\n",
       "      <td>309.8200</td>\n",
       "      <td>6.766840</td>\n",
       "      <td>+</td>\n",
       "      <td>11.519100</td>\n",
       "      <td>4.42818</td>\n",
       "      <td>9.286690e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76180995</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000943</td>\n",
       "      <td>-2.785440</td>\n",
       "      <td>154.3530</td>\n",
       "      <td>0.050266</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.406560</td>\n",
       "      <td>6.27368</td>\n",
       "      <td>8.226020e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76180998</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-1.966890</td>\n",
       "      <td>77.5999</td>\n",
       "      <td>0.049854</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.975600</td>\n",
       "      <td>8.84810</td>\n",
       "      <td>8.233180e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76181016</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76181097</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-0.865668</td>\n",
       "      <td>77.5495</td>\n",
       "      <td>0.009663</td>\n",
       "      <td>-</td>\n",
       "      <td>-0.870068</td>\n",
       "      <td>8.85097</td>\n",
       "      <td>9.216930e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76195736</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.001414</td>\n",
       "      <td>-3.156440</td>\n",
       "      <td>232.3410</td>\n",
       "      <td>0.042881</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.058890</td>\n",
       "      <td>5.11349</td>\n",
       "      <td>8.359490e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76195744</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-1.939470</td>\n",
       "      <td>77.5743</td>\n",
       "      <td>0.048490</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.948710</td>\n",
       "      <td>8.84956</td>\n",
       "      <td>8.257120e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76195754</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>42.267500</td>\n",
       "      <td>77.5263</td>\n",
       "      <td>23.044300</td>\n",
       "      <td>+</td>\n",
       "      <td>42.495000</td>\n",
       "      <td>8.85230</td>\n",
       "      <td>1.583090e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213427</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-1.967690</td>\n",
       "      <td>77.6988</td>\n",
       "      <td>0.049831</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.973890</td>\n",
       "      <td>8.84246</td>\n",
       "      <td>8.233570e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213464</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-1.606190</td>\n",
       "      <td>77.5755</td>\n",
       "      <td>0.033256</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.613810</td>\n",
       "      <td>8.84949</td>\n",
       "      <td>8.552990e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213520</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1061</td>\n",
       "      <td>0.000471</td>\n",
       "      <td>-1.717450</td>\n",
       "      <td>77.7296</td>\n",
       "      <td>0.037947</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.722170</td>\n",
       "      <td>8.84071</td>\n",
       "      <td>8.455490e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  CHROM       POS REF ALT  N_INFORMATIVE        AF          U  \\\n",
       "0   SCARB2      4  76161743   C   T           1061  0.000471  -3.094480   \n",
       "1   SCARB2      4  76163295   A   G           1061  0.000471   1.264820   \n",
       "2   SCARB2      4  76163320   T   C           1061  0.000000        NaN   \n",
       "3   SCARB2      4  76163352   C   T           1061  0.000000        NaN   \n",
       "4   SCARB2      4  76174287   T   G           1061  0.001414  51.749700   \n",
       "5   SCARB2      4  76176451   T   C           1061  0.000471  47.851200   \n",
       "6   SCARB2      4  76179562   A   G           1061  0.000943  -2.068100   \n",
       "7   SCARB2      4  76179643   G   A           1061  0.001414  49.205000   \n",
       "8   SCARB2      4  76179654   T   C           1061  0.001885  -8.060090   \n",
       "9   SCARB2      4  76179673   G   A           1061  0.000471  -1.287390   \n",
       "10  SCARB2      4  76179684   C   T           1061  0.001885  45.787600   \n",
       "11  SCARB2      4  76180995   G   C           1061  0.000943  -2.785440   \n",
       "12  SCARB2      4  76180998   C   A           1061  0.000471  -1.966890   \n",
       "13  SCARB2      4  76181016   G   A           1061  0.000000        NaN   \n",
       "14  SCARB2      4  76181097   G   A           1061  0.000471  -0.865668   \n",
       "15  SCARB2      4  76195736   C   G           1061  0.001414  -3.156440   \n",
       "16  SCARB2      4  76195744   T   G           1061  0.000471  -1.939470   \n",
       "17  SCARB2      4  76195754   G   A           1061  0.000471  42.267500   \n",
       "18  SCARB2      4  76213427   C   T           1061  0.000471  -1.967690   \n",
       "19  SCARB2      4  76213464   C   T           1061  0.000471  -1.606190   \n",
       "20  SCARB2      4  76213520   C   T           1061  0.000471  -1.717450   \n",
       "\n",
       "           V       STAT DIRECTION     EFFECT       SE        PVALUE  \n",
       "0    77.2455   0.123966         -  -3.122440  8.86837  7.247730e-01  \n",
       "1    74.1873   0.021564         +   1.328860  9.04931  8.832530e-01  \n",
       "2        NaN        NaN       NaN        NaN      NaN           NaN  \n",
       "3        NaN        NaN       NaN        NaN      NaN           NaN  \n",
       "4   232.2780  11.529400         +  17.365200  5.11419  6.850250e-04  \n",
       "5    77.4252  29.573500         +  48.171500  8.85807  5.383610e-08  \n",
       "6   155.2640   0.027547         -  -1.038200  6.25526  8.681790e-01  \n",
       "7   231.8630  10.442100         +  16.540900  5.11876  1.231740e-03  \n",
       "8   301.0740   0.215778         -  -2.086640  4.49204  6.422760e-01  \n",
       "9    77.6605   0.021341         -  -1.292080  8.84464  8.838530e-01  \n",
       "10  309.8200   6.766840         +  11.519100  4.42818  9.286690e-03  \n",
       "11  154.3530   0.050266         -  -1.406560  6.27368  8.226020e-01  \n",
       "12   77.5999   0.049854         -  -1.975600  8.84810  8.233180e-01  \n",
       "13       NaN        NaN       NaN        NaN      NaN           NaN  \n",
       "14   77.5495   0.009663         -  -0.870068  8.85097  9.216930e-01  \n",
       "15  232.3410   0.042881         -  -1.058890  5.11349  8.359490e-01  \n",
       "16   77.5743   0.048490         -  -1.948710  8.84956  8.257120e-01  \n",
       "17   77.5263  23.044300         +  42.495000  8.85230  1.583090e-06  \n",
       "18   77.6988   0.049831         -  -1.973890  8.84246  8.233570e-01  \n",
       "19   77.5755   0.033256         -  -1.613810  8.84949  8.552990e-01  \n",
       "20   77.7296   0.037947         -  -1.722170  8.84071  8.455490e-01  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# store csv file\n",
    "\n",
    "SV_AACexonic = pd.read_csv(f'{WORK_DIR}/AAC_SCARB2.burdenTESTSV.exonic.SingleScore.assoc', sep = '\\t')\n",
    "# Drop duplicate rows\n",
    "SV_AACexonic = SV_AACexonic.drop_duplicates()\n",
    "display(SV_AACexonic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "2d1e9bc3-0c40-4bb8-b72d-46f16cc17974",
   "metadata": {},
   "outputs": [],
   "source": [
    "SV_AACexonic.to_csv(f'{WORK_DIR}/AAC_SCARB2.burdenTESTSV.exonic.SingleScore.assoc.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "12705eb6-24e0-49bd-a624-87659ab8fe19",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t2738\t19\t16\t7391.53\t1\t0.783387\n"
     ]
    }
   ],
   "source": [
    "#Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AFR/AFR_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "5c6c1dac-ea6c-4183-9883-6cbedf5d82f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t2738\t10\t9\t24604.4\t1\t0.275785\n"
     ]
    }
   ],
   "source": [
    "#Check AFR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_AFR/AFR_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "50a4247c-4f56-4043-833b-7f4cca5ad3b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t2174\t5\t3\t43729.5\t0\t0.0206808\n"
     ]
    }
   ],
   "source": [
    "#Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AJ/AJ_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "2bb5df49-1e9b-4800-b8c9-cb86ea0c21f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t2174\t4\t2\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check AJ nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_AJ/AJ_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52887a46-7743-4605-aec4-489b4828454b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# run skat test in AJ ancestry in EXONIC\n",
    "#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_AJ'\n",
    "\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/AJ_SCARB2.SKAT.exonic \\\n",
    "--kernel skat \\\n",
    "--inVcf {WORK_DIR}/AJ_SCARB2.exonic.vcf.gz \\\n",
    "--pheno /home/jupyter/workspace/ws_files/AJ/AJ_covariate_file_noGBA.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene SCARB2 \\\n",
    "--geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "--covar /home/jupyter/workspace/ws_files/AJ/AJ_covariate_file_noGBA.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "cdb66fe7-9f65-4737-a2c8-7f6e7fe6a0e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t2174\t5\t3\t1.89553e+07\t0.0206494\t10000\t10000\t1.89553e+07\t209\t0\t0.0209\n"
     ]
    }
   ],
   "source": [
    "# look at result\n",
    "! cat {WORK_DIR}/AJ_SCARB2.SKAT.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e48975a-1b48-40d3-b6b1-9f0a093677f7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# run single variant burden analysis\n",
    "# single variant analysis\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--inVcf {WORK_DIR}/AJ_SCARB2.exonic.vcf.gz \\\n",
    "--single score \\\n",
    "--pheno /home/jupyter/workspace/ws_files/AJ/AJ_covariate_file_noGBA.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "--gene SCARB2 \\\n",
    "--out {WORK_DIR}/AJ_SCARB2.burdenTESTSV.exonic \\\n",
    "--covar /home/jupyter/workspace/ws_files/AJ/AJ_covariate_file_noGBA.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "f381b01a-8c5c-4bd5-a57c-b5a458d6eb57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>N_INFORMATIVE</th>\n",
       "      <th>AF</th>\n",
       "      <th>U</th>\n",
       "      <th>V</th>\n",
       "      <th>STAT</th>\n",
       "      <th>DIRECTION</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>SE</th>\n",
       "      <th>PVALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179654</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2174</td>\n",
       "      <td>0.00989</td>\n",
       "      <td>-221.0640</td>\n",
       "      <td>9123.970</td>\n",
       "      <td>5.35614</td>\n",
       "      <td>-</td>\n",
       "      <td>-5.24879</td>\n",
       "      <td>2.26795</td>\n",
       "      <td>0.020649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76195738</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2174</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76195788</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2174</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213464</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>2174</td>\n",
       "      <td>0.00092</td>\n",
       "      <td>34.9293</td>\n",
       "      <td>862.831</td>\n",
       "      <td>1.41402</td>\n",
       "      <td>+</td>\n",
       "      <td>8.76981</td>\n",
       "      <td>7.37501</td>\n",
       "      <td>0.234390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213496</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2174</td>\n",
       "      <td>0.00138</td>\n",
       "      <td>64.1473</td>\n",
       "      <td>1291.920</td>\n",
       "      <td>3.18508</td>\n",
       "      <td>+</td>\n",
       "      <td>10.75640</td>\n",
       "      <td>6.02708</td>\n",
       "      <td>0.074314</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene  CHROM       POS REF ALT  N_INFORMATIVE       AF         U  \\\n",
       "0  SCARB2      4  76179654   T   C           2174  0.00989 -221.0640   \n",
       "1  SCARB2      4  76195738   G   A           2174  0.00000       NaN   \n",
       "2  SCARB2      4  76195788   T   C           2174  0.00000       NaN   \n",
       "3  SCARB2      4  76213464   C   T           2174  0.00092   34.9293   \n",
       "4  SCARB2      4  76213496   C   G           2174  0.00138   64.1473   \n",
       "\n",
       "          V     STAT DIRECTION    EFFECT       SE    PVALUE  \n",
       "0  9123.970  5.35614         -  -5.24879  2.26795  0.020649  \n",
       "1       NaN      NaN       NaN       NaN      NaN       NaN  \n",
       "2       NaN      NaN       NaN       NaN      NaN       NaN  \n",
       "3   862.831  1.41402         +   8.76981  7.37501  0.234390  \n",
       "4  1291.920  3.18508         +  10.75640  6.02708  0.074314  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# store csv file\n",
    "\n",
    "SV_AJexonic = pd.read_csv(f'{WORK_DIR}/AJ_SCARB2.burdenTESTSV.exonic.SingleScore.assoc', sep = '\\t')\n",
    "# Drop duplicate rows\n",
    "SV_AJexonic = SV_AJexonic.drop_duplicates()\n",
    "display(SV_AJexonic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "a068d9e9-61a8-495d-8d84-093115a7652c",
   "metadata": {},
   "outputs": [],
   "source": [
    "SV_AJexonic.to_csv(f'{WORK_DIR}/AJ_SCARB2.burdenTESTSV.exonic.SingleScore.assoc.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "08f8dda4-f7d4-4f78-9116-4c1847c2f051",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t3265\t18\t14\t14716.2\t0\t0.854888\n"
     ]
    }
   ],
   "source": [
    "#Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_AMR/AMR_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "f7d34e2f-0e9a-4199-a64b-2b2de469a93b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t3265\t14\t11\t15769.7\t0\t0.502164\n"
     ]
    }
   ],
   "source": [
    "#Check AMR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_AMR/AMR_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "b7521970-d0b7-4fa1-b34a-400bbe5a0f7a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t1104\t15\t10\t11529.8\t0\t0.127544\n"
     ]
    }
   ],
   "source": [
    "#Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAS/CAS_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "e92dcdbd-38f8-4663-8e3c-5abba9b87863",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t1104\t10\t7\t11315.1\t0.2\t0.090737\n"
     ]
    }
   ],
   "source": [
    "#Check CAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAS/CAS_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "abec9ad1-35ac-45df-a4c0-013ef4924877",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t4670\t17\t14\t80648\t0\t0.0239711\n"
     ]
    }
   ],
   "source": [
    "#Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_EAS/EAS_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "8a4ab6f1-941f-4cda-a454-544e675b48d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t4670\t12\t12\t80564.9\t0\t0.0192043\n"
     ]
    }
   ],
   "source": [
    "#Check EAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_EAS/EAS_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ec9d2329-cdbb-4048-84b5-e7f42ad6e951",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# run skat test in EAS ancestry in EXONIC\n",
    "#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_EAS'\n",
    "\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/EAS_SCARB2.SKAT.exonic \\\n",
    "--kernel skat \\\n",
    "--inVcf {WORK_DIR}/EAS_SCARB2.exonic.vcf.gz \\\n",
    "--pheno /home/jupyter/workspace/ws_files/EAS/EAS_covariate_file_noGBA.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene SCARB2 \\\n",
    "--geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "--covar /home/jupyter/workspace/ws_files/EAS/EAS_covariate_file_noGBA.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "b883871f-dad1-455e-85ee-96ce86500ca3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t4670\t17\t14\t5.06632e+06\t0.0129632\t10000\t10000\t5.06632e+06\t137\t0\t0.0137\n"
     ]
    }
   ],
   "source": [
    "# look at result\n",
    "! cat {WORK_DIR}/EAS_SCARB2.SKAT.exonic.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07e89bda-584f-49ec-ace4-48a56a0f4e4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# run skat test in EAS ancestry in nonsynonymous_lof\n",
    "#Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_EAS'\n",
    "\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--out {WORK_DIR}/EAS_SCARB2.SKAT.nonsynonymous_lof \\\n",
    "--kernel skat \\\n",
    "--inVcf {WORK_DIR}/EAS_SCARB2.nonsynonymous_lof.vcf.gz \\\n",
    "--pheno /home/jupyter/workspace/ws_files/EAS/EAS_covariate_file_noGBA.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--gene SCARB2 \\\n",
    "--geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "--covar /home/jupyter/workspace/ws_files/EAS/EAS_covariate_file_noGBA.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "f78024d5-9872-4d3a-aed5-f997064d3ef3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\tPvalue\tNumPerm\tActualPerm\tStat\tNumGreater\tNumEqual\tPermPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t4670\t12\t12\t5.0611e+06\t0.0107745\t10000\t10000\t5.0611e+06\t113\t0\t0.0113\n"
     ]
    }
   ],
   "source": [
    "# look at result\n",
    "! cat {WORK_DIR}/EAS_SCARB2.SKAT.nonsynonymous_lof.Skat.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46a90df2-4e37-4477-87c8-719f245b2081",
   "metadata": {},
   "outputs": [],
   "source": [
    "# run single variant burden analysis\n",
    "# single variant analysis\n",
    "! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "--inVcf {WORK_DIR}/EAS_SCARB2.exonic.vcf.gz \\\n",
    "--single score \\\n",
    "--pheno /home/jupyter/workspace/ws_files/EAS/EAS_covariate_file_noGBA.txt \\\n",
    "--pheno-name PHENO \\\n",
    "--geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "--gene SCARB2 \\\n",
    "--out {WORK_DIR}/EAS_SCARB2.burdenTESTSV.exonic \\\n",
    "--covar /home/jupyter/workspace/ws_files/EAS/EAS_covariate_file_noGBA.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--freqUpper 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "9c806d55-f9a7-4637-9795-6dc440f70b41",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>N_INFORMATIVE</th>\n",
       "      <th>AF</th>\n",
       "      <th>U</th>\n",
       "      <th>V</th>\n",
       "      <th>STAT</th>\n",
       "      <th>DIRECTION</th>\n",
       "      <th>EFFECT</th>\n",
       "      <th>SE</th>\n",
       "      <th>PVALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76161720</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000107</td>\n",
       "      <td>0.701884</td>\n",
       "      <td>31.3742</td>\n",
       "      <td>0.015702</td>\n",
       "      <td>+</td>\n",
       "      <td>0.702537</td>\n",
       "      <td>5.606480</td>\n",
       "      <td>0.900280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76163238</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.001285</td>\n",
       "      <td>30.770800</td>\n",
       "      <td>374.1370</td>\n",
       "      <td>2.530750</td>\n",
       "      <td>+</td>\n",
       "      <td>2.582770</td>\n",
       "      <td>1.623530</td>\n",
       "      <td>0.111647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76163358</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000107</td>\n",
       "      <td>1.753440</td>\n",
       "      <td>31.3818</td>\n",
       "      <td>0.097973</td>\n",
       "      <td>+</td>\n",
       "      <td>1.754650</td>\n",
       "      <td>5.605790</td>\n",
       "      <td>0.754276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76166279</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000535</td>\n",
       "      <td>17.439300</td>\n",
       "      <td>156.6110</td>\n",
       "      <td>1.941940</td>\n",
       "      <td>+</td>\n",
       "      <td>3.496890</td>\n",
       "      <td>2.509370</td>\n",
       "      <td>0.163459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76168474</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000964</td>\n",
       "      <td>-2.958860</td>\n",
       "      <td>281.4960</td>\n",
       "      <td>0.031101</td>\n",
       "      <td>-</td>\n",
       "      <td>-0.330087</td>\n",
       "      <td>1.871710</td>\n",
       "      <td>0.860015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76174224</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000749</td>\n",
       "      <td>-2.447310</td>\n",
       "      <td>218.9260</td>\n",
       "      <td>0.027358</td>\n",
       "      <td>-</td>\n",
       "      <td>-0.351049</td>\n",
       "      <td>2.122400</td>\n",
       "      <td>0.868628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76175799</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000321</td>\n",
       "      <td>6.950210</td>\n",
       "      <td>93.6762</td>\n",
       "      <td>0.515663</td>\n",
       "      <td>+</td>\n",
       "      <td>2.329940</td>\n",
       "      <td>3.244600</td>\n",
       "      <td>0.472697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179549</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.001071</td>\n",
       "      <td>-42.886500</td>\n",
       "      <td>312.7170</td>\n",
       "      <td>5.881530</td>\n",
       "      <td>-</td>\n",
       "      <td>-4.306710</td>\n",
       "      <td>1.775830</td>\n",
       "      <td>0.015300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179614</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000428</td>\n",
       "      <td>-5.933910</td>\n",
       "      <td>125.4000</td>\n",
       "      <td>0.280791</td>\n",
       "      <td>-</td>\n",
       "      <td>-1.486000</td>\n",
       "      <td>2.804320</td>\n",
       "      <td>0.596183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179625</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000321</td>\n",
       "      <td>-7.477590</td>\n",
       "      <td>93.9443</td>\n",
       "      <td>0.595187</td>\n",
       "      <td>-</td>\n",
       "      <td>-2.499580</td>\n",
       "      <td>3.239970</td>\n",
       "      <td>0.440420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179654</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000214</td>\n",
       "      <td>5.997120</td>\n",
       "      <td>62.7220</td>\n",
       "      <td>0.573411</td>\n",
       "      <td>+</td>\n",
       "      <td>3.002610</td>\n",
       "      <td>3.965210</td>\n",
       "      <td>0.448907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76179699</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.004176</td>\n",
       "      <td>78.040100</td>\n",
       "      <td>1197.5000</td>\n",
       "      <td>5.085800</td>\n",
       "      <td>+</td>\n",
       "      <td>2.046530</td>\n",
       "      <td>0.907482</td>\n",
       "      <td>0.024123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76195811</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213430</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000214</td>\n",
       "      <td>-10.010100</td>\n",
       "      <td>62.7407</td>\n",
       "      <td>1.597080</td>\n",
       "      <td>-</td>\n",
       "      <td>-5.010310</td>\n",
       "      <td>3.964620</td>\n",
       "      <td>0.206318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213438</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000107</td>\n",
       "      <td>-0.016490</td>\n",
       "      <td>31.3434</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>-</td>\n",
       "      <td>-0.016521</td>\n",
       "      <td>5.609230</td>\n",
       "      <td>0.997650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213451</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SCARB2</td>\n",
       "      <td>4</td>\n",
       "      <td>76213495</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4670</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene  CHROM       POS REF ALT  N_INFORMATIVE        AF          U  \\\n",
       "0   SCARB2      4  76161720   C   T           4670  0.000107   0.701884   \n",
       "1   SCARB2      4  76163238   C   T           4670  0.001285  30.770800   \n",
       "2   SCARB2      4  76163358   G   A           4670  0.000107   1.753440   \n",
       "3   SCARB2      4  76166279   T   C           4670  0.000535  17.439300   \n",
       "4   SCARB2      4  76168474   C   T           4670  0.000964  -2.958860   \n",
       "5   SCARB2      4  76174224   G   A           4670  0.000749  -2.447310   \n",
       "6   SCARB2      4  76175799   G   C           4670  0.000321   6.950210   \n",
       "7   SCARB2      4  76179549   C   T           4670  0.001071 -42.886500   \n",
       "8   SCARB2      4  76179614   G   A           4670  0.000428  -5.933910   \n",
       "9   SCARB2      4  76179625   A   T           4670  0.000321  -7.477590   \n",
       "10  SCARB2      4  76179654   T   C           4670  0.000214   5.997120   \n",
       "11  SCARB2      4  76179699   T   A           4670  0.004176  78.040100   \n",
       "12  SCARB2      4  76195811   A   G           4670  0.000000        NaN   \n",
       "13  SCARB2      4  76213430   C   T           4670  0.000214 -10.010100   \n",
       "14  SCARB2      4  76213438   T   G           4670  0.000107  -0.016490   \n",
       "15  SCARB2      4  76213451   C   T           4670  0.000000        NaN   \n",
       "16  SCARB2      4  76213495   G   A           4670  0.000000        NaN   \n",
       "\n",
       "            V      STAT DIRECTION    EFFECT        SE    PVALUE  \n",
       "0     31.3742  0.015702         +  0.702537  5.606480  0.900280  \n",
       "1    374.1370  2.530750         +  2.582770  1.623530  0.111647  \n",
       "2     31.3818  0.097973         +  1.754650  5.605790  0.754276  \n",
       "3    156.6110  1.941940         +  3.496890  2.509370  0.163459  \n",
       "4    281.4960  0.031101         - -0.330087  1.871710  0.860015  \n",
       "5    218.9260  0.027358         - -0.351049  2.122400  0.868628  \n",
       "6     93.6762  0.515663         +  2.329940  3.244600  0.472697  \n",
       "7    312.7170  5.881530         - -4.306710  1.775830  0.015300  \n",
       "8    125.4000  0.280791         - -1.486000  2.804320  0.596183  \n",
       "9     93.9443  0.595187         - -2.499580  3.239970  0.440420  \n",
       "10    62.7220  0.573411         +  3.002610  3.965210  0.448907  \n",
       "11  1197.5000  5.085800         +  2.046530  0.907482  0.024123  \n",
       "12        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "13    62.7407  1.597080         - -5.010310  3.964620  0.206318  \n",
       "14    31.3434  0.000009         - -0.016521  5.609230  0.997650  \n",
       "15        NaN       NaN       NaN       NaN       NaN       NaN  \n",
       "16        NaN       NaN       NaN       NaN       NaN       NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# store csv file\n",
    "\n",
    "SV_EASexonic = pd.read_csv(f'{WORK_DIR}/EAS_SCARB2.burdenTESTSV.exonic.SingleScore.assoc', sep = '\\t')\n",
    "# Drop duplicate rows\n",
    "SV_EASexonic = SV_EASexonic.drop_duplicates()\n",
    "display(SV_EASexonic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "12ee120b-d998-4bcb-9816-2b5c93c121a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "SV_EASexonic.to_csv(f'{WORK_DIR}/EAS_SCARB2.burdenTESTSV.exonic.SingleScore.assoc.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "52134a36-c841-4552-8efd-6e4e6ea32825",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'~/workspace/ws_files/SCARB2'"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "WORK_DIR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "488c58b4-6e3e-453e-85a2-151484e27a99",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n"
     ]
    }
   ],
   "source": [
    "#Check EAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_FIN/FIN_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "0544c11d-0f99-4ceb-a9ea-c0df7bf30729",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t1230\t21\t11\t10.0292\t1\t1\n"
     ]
    }
   ],
   "source": [
    "#Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_MDE/MDE_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "feddee19-f035-4564-bade-52014ee01c86",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t1230\t15\t7\t143.991\t1\t1\n"
     ]
    }
   ],
   "source": [
    "#Check MDE nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_MDE/MDE_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "60a80024-3dfc-4bfc-ab5a-ef13436db83b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t689\t10\t6\t2310.65\t0\t0.286168\n"
     ]
    }
   ],
   "source": [
    "#Check SAS all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_SAS/SAS_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "214c40d7-40fc-4b4d-9f01-dff713fcc21b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t689\t7\t5\t3066.05\t0.8\t0.150845\n"
     ]
    }
   ],
   "source": [
    "#Check SAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_SAS/SAS_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "0d678142-ab33-46d1-b409-8cc8ec45071c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t900\t16\t11\t152.593\t1\t1\n"
     ]
    }
   ],
   "source": [
    "#Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAH/CAH_SCARB2.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "f755d61b-569b-45bd-9d46-1f68f79ed250",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "SCARB2\t4:76158738-76213824,4:76158738-76213899\t900\t9\t7\t393.54\t1\t1\n"
     ]
    }
   ],
   "source": [
    "#Check CAH nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/SCARB2_CAH/CAH_SCARB2.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "efe0f1c3-6171-48b8-b1f3-39182de67a6d",
   "metadata": {},
   "source": [
    "LD pruning(in EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "4e168d05-a578-4f00-aa54-0f30ab0fdc2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "1de4a1ab-6682-4b75-abc1-65285f2c9035",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "  --exclude /home/jupyter/workspace/ws_files//exclusion_regions_hg38.txt\n",
      "  --geno 0.05\n",
      "  --hwe 1e-5 0.001\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED\n",
      "\n",
      "Start time: Sat Jul 26 12:28:25 2025\n",
      "26046 MiB RAM detected, ~24817 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//SCARB2/SCARB2_EUR/EUR_SCARB2.fam.\n",
      "7102 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//SCARB2/SCARB2_EUR/EUR_SCARB2.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--exclude: 7102 variants remaining.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "done.lating allele frequencies... 0%\n",
      "--geno: 2974 variants removed due to missing genotype data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "3208 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "920 variants remaining after main filters.\n",
      "done.ng /home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED.fam ... \n",
      "done.ng /home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED.bim ... \n",
      "done.ng /home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED.bed ... 0%\n",
      "End time: Sat Jul 26 12:28:26 2025\n"
     ]
    }
   ],
   "source": [
    "# Make sure to use high-quality SNPs\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/SCARB2/SCARB2_EUR/EUR_SCARB2 \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.05 \\\n",
    "--hwe 1e-5 0.001 \\\n",
    "--make-bed \\\n",
    "--keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep \\\n",
    "--exclude {WORK_DIR}/exclusion_regions_hg38.txt \\\n",
    "--out {WORK_DIR}/SCARB2/SCARB2_UNIMPUTED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "e2f3d465-b9bf-4c10-9f0b-7e60b606888e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//SCARB2/prune_SCARB2.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out /home/jupyter/workspace/ws_files//SCARB2/prune_SCARB2\n",
      "\n",
      "Start time: Sat Jul 26 12:29:00 2025\n",
      "26046 MiB RAM detected, ~24777 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED.fam.\n",
      "920 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//SCARB2/SCARB2_UNIMPUTED.bim.\n",
      "1 binary phenotype loaded (24208 cases, 9662 controls).\n",
      "Calculating allele frequencies... done.\n",
      "--indep-pairwise (1 compute thread): 779/920 variants removed.\n",
      "Variant lists written to\n",
      "/home/jupyter/workspace/ws_files//SCARB2/prune_SCARB2.prune.in and\n",
      "/home/jupyter/workspace/ws_files//SCARB2/prune_SCARB2.prune.out .\n",
      "End time: Sat Jul 26 12:29:00 2025\n"
     ]
    }
   ],
   "source": [
    "# Prune out unnecessary SNPs (only need to do this to generate PCs)\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/SCARB2/SCARB2_UNIMPUTED \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out {WORK_DIR}/SCARB2/prune_SCARB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "124269cd-ef43-4efa-9f45-d34c4bdec5db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "141 /home/jupyter/workspace/ws_files//SCARB2/prune_SCARB2.prune.in\n"
     ]
    }
   ],
   "source": [
    "!wc -l {WORK_DIR}/SCARB2/prune_SCARB2.prune.in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "7795d5de-09ba-494f-8710-0622ee57c612",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 3.55E-4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee50655d-831f-4646-9362-e9dcae1c1e61",
   "metadata": {},
   "source": [
    "zoom locus plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b0ae6af-e9c5-4392-9f20-78a46a9e3c59",
   "metadata": {},
   "outputs": [],
   "source": [
    "## locus zoom format\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "\n",
    "    df = pd.read_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')\n",
    "    df['b'] = np.log(df['OR'])\n",
    "    export_ldassoc = df[['#CHROM', 'POS', 'REF', 'ALT', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P']].copy()\n",
    "    export_ldassoc = export_ldassoc.rename(columns = {'#CHROM': 'CHROM', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p'})\n",
    "    # save to files\n",
    "    export_ldassoc.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.formatted.tab', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c66a905d-e8b0-49d0-a825-65fb57c40b79",
   "metadata": {},
   "source": [
    "GCTA cojo analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "cc476082-c7fe-4c80-b275-4fc3f80a616c",
   "metadata": {},
   "outputs": [],
   "source": [
    "## cojo format\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "\n",
    "    sumstats = pd.read_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')\n",
    "    #Format summary statistics for GCTA-COJO\n",
    "    #First get the log odds ratio - this is required for COJO\n",
    "    #1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.\n",
    "    sumstats_formatted = sumstats.copy()\n",
    "    sumstats_formatted['b'] = np.log(sumstats_formatted['OR'])\n",
    "\n",
    "    #Now select just the necessary columns for COJO\n",
    "    sumstats_export = sumstats_formatted[['ID', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P', 'OBS_CT']].copy()\n",
    "\n",
    "    #Rename columns following COJO format\n",
    "    sumstats_export = sumstats_export.rename(columns = {'ID':'SNP', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p', 'OBS_CT':'N'})\n",
    "\n",
    "    #Export\n",
    "    sumstats_export.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.sumstats.ma', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "36b1c3a7-4d18-4d97-8aaa-379fcad2307c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:32:55 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "945 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "2268 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 945 individuals and 2268 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 911 SNPs (1357 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 910 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 910 SNPs: 5.67674 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "910 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 910 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:32:55 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.30 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:32:56 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "144 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "1200 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 144 individuals and 1200 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 958 SNPs (242 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 932 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 932 SNPs: 14.0831 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "932 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 932 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:32:57 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.09 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:32:58 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "1493 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "3010 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 1493 individuals and 3010 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 972 SNPs (2038 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 972 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 972 SNPs: 5.02867 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "972 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 972 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:32:58 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.30 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:32:59 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "1215 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "3967 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 1215 individuals and 3967 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1441 SNPs (2526 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1441 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1441 SNPs: 5.9973 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1441 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1441 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:32:59 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.32 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:33:01 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "3780 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "4696 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 3780 individuals and 4696 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1457 SNPs (3239 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1456 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1456 SNPs: 4.89246 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1456 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1456 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:33:01 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.66 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:33:02 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "1691 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "2681 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 1691 individuals and 2681 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 965 SNPs (1716 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 967 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 967 SNPs: 4.42984 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "965 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 965 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:33:03 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.33 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:33:04 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "3920 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 3920 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1228 SNPs (2692 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1227 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1227 SNPs: 5.5426 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1218 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1218 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:33:04 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.30 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:33:05 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "3472 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "1942 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 3472 individuals and 1942 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 1031 SNPs (911 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 1029 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 1029 SNPs: 7.87004 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "1027 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 1027 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 1026 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AJ/AJ.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:33:06 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.75 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:33:07 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "3608 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "4269 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 3608 individuals and 4269 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 917 SNPs (3352 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 917 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 917 SNPs: 5.63331 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "917 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 917 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 916 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_AMR/AMR.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:33:08 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.87 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:33:09 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "58823 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "7102 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 58823 individuals and 7102 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 921 SNPs (6181 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 920 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 920 SNPs: 5.97379 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "920 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 920 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 919 remaining SNPs to [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EUR/EUR.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:33:17 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 7.81 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 12:33:18 UTC on Sat Jul 26 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000355\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "6662 individuals to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed] in SNP-major format ...\n",
      "Genotype data for 6662 individuals and 3094 SNPs to be included from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS_SCARB2.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 950 SNPs (2144 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 949 SNPs read from [/home/jupyter/workspace/ws_files/SCARB2/SCARB2_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 949 SNPs: 5.15563 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "949 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 949 SNPs to select association signals ... (p cutoff = 0.000355; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 12:33:19 UTC on Sat Jul 26 2025\n",
      "Overall computational time: 0.59 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/SCARB2/SCARB2_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on LD pruning p-value 4.67e-4\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.95.0-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_SCARB2 \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 3.55e-4 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.ldprune.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19442796-b759-4e46-8b06-60a8cb51db25",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
