{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "2e7e509c-28ba-43de-baa7-ff7ca792cc6e",
   "metadata": {},
   "source": [
    "TMEM175 - Single gene analysis in GP2 Neurobooster genotyping data (all ancestries)\n",
    "\n",
    "Project: GP2 TMEM175\n",
    "\n",
    "Version: Python/3.10.15, R/4.3.3\n",
    "\n",
    "Notebook Overview\n",
    "\n",
    "Description Loading Python libraries Set paths Make working directory\n",
    "\n",
    "Installing packages\n",
    "\n",
    "Create a covariate file with GP2 data\n",
    "\n",
    "Annotation of the gene TMEM175\n",
    "\n",
    "Association analysis to compare allele frequencies between cases and controls\n",
    "\n",
    "GLM analysis adjusting for gender, age, PC1-5\n",
    "\n",
    "Burden test(SkatO, Skat, cmc,zeggini,mb,fp,cmcWald)\n",
    "\n",
    "Conditional analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f50e514-345d-424e-9a48-d698a0b04077",
   "metadata": {},
   "source": [
    "Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "66a038a8-3559-4366-b47d-46549a9ea0dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use pathlib for file path manipulation\n",
    "import pathlib\n",
    "\n",
    "# Install numpy\n",
    "import numpy as np\n",
    "\n",
    "# Install Pandas for tabular data\n",
    "import pandas as pd\n",
    "\n",
    "# Install plotnine: a ggplot2-compatible Python plotting package\n",
    "from plotnine import *\n",
    "\n",
    "# Always show all columns in a Pandas DataFrame\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e88e7954-f0cd-43df-819c-3dbbe071c6bd",
   "metadata": {},
   "source": [
    "Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "686b9392-53bb-4326-9666-56a0b3c9628c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 100K\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 25 13:32 clinical_data\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 25 13:32 imputed_genotypes\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 25 13:32 meta_data\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 25 13:32 raw_genotypes\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 25 13:32 raw_genotypes_flipped\n",
      "-r--r--r--. 1 jupyter users 100K Jun 30 20:07 README_release10_01072025.txt\n",
      "dr-xr-xr-x. 1 jupyter users    0 Jul 25 13:32 wgs\n"
     ]
    }
   ],
   "source": [
    "REL10_PATH = pathlib.Path(pathlib.Path.home(), 'workspace/gp2_tier2_eu_release10')\n",
    "!ls -hal {REL10_PATH}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e40346f-a603-411d-adf9-35c901db0b2f",
   "metadata": {},
   "source": [
    "Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "27e36bad-e66a-4fd4-93fd-10b755e65226",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir ~/workspace/ws_files/TMEM175"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4c414251-33fe-4f86-8855-57b217764010",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "09165c4a-2438-4acb-a486-370199147dad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\t\t\t\t       plink_linux_x86_64_20190304.zip\n",
      "annovar.latest.tar.gz\t\t       prettify\n",
      "intel-simplified-software-license.txt  rvtests\n",
      "LICENSE\t\t\t\t       toy.map\n",
      "plink\t\t\t\t       toy.ped\n",
      "plink2\t\t\t\t       vcf_subset\n",
      "plink2_linux_x86_64_latest.zip\n"
     ]
    }
   ],
   "source": [
    "# make sure all tools installed\n",
    "! ls /home/jupyter/tools"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d432868b-40a5-4aad-b2e1-f6e449580bee",
   "metadata": {},
   "source": [
    "install packages"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1e3d47b-feb0-4764-8e68-8464db774cff",
   "metadata": {},
   "source": [
    "RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4a35b7b4-8d63-4d92-af36-a1bbbf99b8b9",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rvtests is not installed\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--2025-07-24 15:16:49--  https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz\n",
      "Resolving github.com (github.com)... 140.82.121.3\n",
      "Connecting to github.com (github.com)|140.82.121.3|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://release-assets.githubusercontent.com/github-production-release-asset/7528862/63fbb780-29ae-11e9-91d3-99ec3c3134cc?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-07-24T16%3A10%3A13Z&rscd=attachment%3B+filename%3Drvtests_linux64.tar.gz&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-07-24T15%3A09%3A25Z&ske=2025-07-24T16%3A10%3A13Z&sks=b&skv=2018-11-09&sig=v%2Bg0ZHemeimaLBK7%2B25gwCWFnKuu9C755inMrILqaMs%3D&jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc1MzM3MDUwOSwibmJmIjoxNzUzMzcwMjA5LCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.vfh749Rj_wPijQF5AxVG41wF8b9XpvmeUN-tyVmDc6Y&response-content-disposition=attachment%3B%20filename%3Drvtests_linux64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
      "--2025-07-24 15:16:49--  https://release-assets.githubusercontent.com/github-production-release-asset/7528862/63fbb780-29ae-11e9-91d3-99ec3c3134cc?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-07-24T16%3A10%3A13Z&rscd=attachment%3B+filename%3Drvtests_linux64.tar.gz&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-07-24T15%3A09%3A25Z&ske=2025-07-24T16%3A10%3A13Z&sks=b&skv=2018-11-09&sig=v%2Bg0ZHemeimaLBK7%2B25gwCWFnKuu9C755inMrILqaMs%3D&jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc1MzM3MDUwOSwibmJmIjoxNzUzMzcwMjA5LCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.vfh749Rj_wPijQF5AxVG41wF8b9XpvmeUN-tyVmDc6Y&response-content-disposition=attachment%3B%20filename%3Drvtests_linux64.tar.gz&response-content-type=application%2Foctet-stream\n",
      "Resolving release-assets.githubusercontent.com (release-assets.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
      "Connecting to release-assets.githubusercontent.com (release-assets.githubusercontent.com)|185.199.108.133|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 38050737 (36M) [application/octet-stream]\n",
      "Saving to: ‘rvtests_linux64.tar.gz’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0% 10.5M 3s\n",
      "    50K .......... .......... .......... .......... ..........  0% 30.4M 2s\n",
      "   100K .......... .......... .......... .......... ..........  0% 14.7M 2s\n",
      "   150K .......... .......... .......... .......... ..........  0%  104M 2s\n",
      "   200K .......... .......... .......... .......... ..........  0% 61.2M 2s\n",
      "   250K .......... .......... .......... .......... ..........  0% 22.6M 2s\n",
      "   300K .......... .......... .......... .......... ..........  0% 49.1M 1s\n",
      "   350K .......... .......... .......... .......... ..........  1%  165M 1s\n",
      "   400K .......... .......... .......... .......... ..........  1% 90.3M 1s\n",
      "   450K .......... .......... .......... .......... ..........  1%  181M 1s\n",
      "   500K .......... .......... .......... .......... ..........  1%  184M 1s\n",
      "   550K .......... .......... .......... .......... ..........  1%  134M 1s\n",
      "   600K .......... .......... .......... .......... ..........  1% 26.9M 1s\n",
      "   650K .......... .......... .......... .......... ..........  1% 51.3M 1s\n",
      "   700K .......... .......... .......... .......... ..........  2%  124M 1s\n",
      "   750K .......... .......... .......... .......... ..........  2%  110M 1s\n",
      "   800K .......... .......... .......... .......... ..........  2%  156M 1s\n",
      "   850K .......... .......... .......... .......... ..........  2%  228M 1s\n",
      "   900K .......... .......... .......... .......... ..........  2%  149M 1s\n",
      "   950K .......... .......... .......... .......... ..........  2%  135M 1s\n",
      "  1000K .......... .......... .......... .......... ..........  2%  190M 1s\n",
      "  1050K .......... .......... .......... .......... ..........  2% 97.2M 1s\n",
      "  1100K .......... .......... .......... .......... ..........  3% 36.4M 1s\n",
      "  1150K .......... .......... .......... .......... ..........  3% 45.4M 1s\n",
      "  1200K .......... .......... .......... .......... ..........  3% 55.4M 1s\n",
      "  1250K .......... .......... .......... .......... ..........  3% 47.2M 1s\n",
      "  1300K .......... .......... .......... .......... ..........  3% 47.5M 1s\n",
      "  1350K .......... .......... .......... .......... ..........  3% 45.6M 1s\n",
      "  1400K .......... .......... .......... .......... ..........  3%  187M 1s\n",
      "  1450K .......... .......... .......... .......... ..........  4% 34.9M 1s\n",
      "  1500K .......... .......... .......... .......... ..........  4% 32.6M 1s\n",
      "  1550K .......... .......... .......... .......... ..........  4% 58.4M 1s\n",
      "  1600K .......... .......... .......... .......... ..........  4% 44.5M 1s\n",
      "  1650K .......... .......... .......... .......... ..........  4% 68.3M 1s\n",
      "  1700K .......... .......... .......... .......... ..........  4% 63.9M 1s\n",
      "  1750K .......... .......... .......... .......... ..........  4% 31.6M 1s\n",
      "  1800K .......... .......... .......... .......... ..........  4% 37.0M 1s\n",
      "  1850K .......... .......... .......... .......... ..........  5% 96.8M 1s\n",
      "  1900K .......... .......... .......... .......... ..........  5% 68.9M 1s\n",
      "  1950K .......... .......... .......... .......... ..........  5% 32.5M 1s\n",
      "  2000K .......... .......... .......... .......... ..........  5% 38.5M 1s\n",
      "  2050K .......... .......... .......... .......... ..........  5% 52.6M 1s\n",
      "  2100K .......... .......... .......... .......... ..........  5% 45.6M 1s\n",
      "  2150K .......... .......... .......... .......... ..........  5%  100M 1s\n",
      "  2200K .......... .......... .......... .......... ..........  6% 41.3M 1s\n",
      "  2250K .......... .......... .......... .......... ..........  6% 56.9M 1s\n",
      "  2300K .......... .......... .......... .......... ..........  6% 39.7M 1s\n",
      "  2350K .......... .......... .......... .......... ..........  6% 37.8M 1s\n",
      "  2400K .......... .......... .......... .......... ..........  6% 90.7M 1s\n",
      "  2450K .......... .......... .......... .......... ..........  6% 51.8M 1s\n",
      "  2500K .......... .......... .......... .......... ..........  6% 40.5M 1s\n",
      "  2550K .......... .......... .......... .......... ..........  6% 52.9M 1s\n",
      "  2600K .......... .......... .......... .......... ..........  7% 42.1M 1s\n",
      "  2650K .......... .......... .......... .......... ..........  7% 65.2M 1s\n",
      "  2700K .......... .......... .......... .......... ..........  7% 40.1M 1s\n",
      "  2750K .......... .......... .......... .......... ..........  7% 39.4M 1s\n",
      "  2800K .......... .......... .......... .......... ..........  7% 48.1M 1s\n",
      "  2850K .......... .......... .......... .......... ..........  7% 48.8M 1s\n",
      "  2900K .......... .......... .......... .......... ..........  7% 70.6M 1s\n",
      "  2950K .......... .......... .......... .......... ..........  8% 29.3M 1s\n",
      "  3000K .......... .......... .......... .......... ..........  8%  187M 1s\n",
      "  3050K .......... .......... .......... .......... ..........  8% 53.2M 1s\n",
      "  3100K .......... .......... .......... .......... ..........  8% 40.3M 1s\n",
      "  3150K .......... .......... .......... .......... ..........  8% 59.8M 1s\n",
      "  3200K .......... .......... .......... .......... ..........  8% 36.6M 1s\n",
      "  3250K .......... .......... .......... .......... ..........  8% 40.9M 1s\n",
      "  3300K .......... .......... .......... .......... ..........  9%  149M 1s\n",
      "  3350K .......... .......... .......... .......... ..........  9% 43.7M 1s\n",
      "  3400K .......... .......... .......... .......... ..........  9% 41.6M 1s\n",
      "  3450K .......... .......... .......... .......... ..........  9% 34.6M 1s\n",
      "  3500K .......... .......... .......... .......... ..........  9% 47.9M 1s\n",
      "  3550K .......... .......... .......... .......... ..........  9% 43.6M 1s\n",
      "  3600K .......... .......... .......... .......... ..........  9% 63.8M 1s\n",
      "  3650K .......... .......... .......... .......... ..........  9% 45.4M 1s\n",
      "  3700K .......... .......... .......... .......... .......... 10% 41.7M 1s\n",
      "  3750K .......... .......... .......... .......... .......... 10% 53.6M 1s\n",
      "  3800K .......... .......... .......... .......... .......... 10% 95.4M 1s\n",
      "  3850K .......... .......... .......... .......... .......... 10% 41.7M 1s\n",
      "  3900K .......... .......... .......... .......... .......... 10% 37.4M 1s\n",
      "  3950K .......... .......... .......... .......... .......... 10% 40.6M 1s\n",
      "  4000K .......... .......... .......... .......... .......... 10% 58.1M 1s\n",
      "  4050K .......... .......... .......... .......... .......... 11% 43.2M 1s\n",
      "  4100K .......... .......... .......... .......... .......... 11% 54.7M 1s\n",
      "  4150K .......... .......... .......... .......... .......... 11% 79.4M 1s\n",
      "  4200K .......... .......... .......... .......... .......... 11% 99.2M 1s\n",
      "  4250K .......... .......... .......... .......... .......... 11% 36.2M 1s\n",
      "  4300K .......... .......... .......... .......... .......... 11%  155M 1s\n",
      "  4350K .......... .......... .......... .......... .......... 11% 21.6M 1s\n",
      "  4400K .......... .......... .......... .......... .......... 11% 51.8M 1s\n",
      "  4450K .......... .......... .......... .......... .......... 12%  178M 1s\n",
      "  4500K .......... .......... .......... .......... .......... 12% 52.4M 1s\n",
      "  4550K .......... .......... .......... .......... .......... 12% 37.4M 1s\n",
      "  4600K .......... .......... .......... .......... .......... 12% 36.5M 1s\n",
      "  4650K .......... .......... .......... .......... .......... 12%  141M 1s\n",
      "  4700K .......... .......... .......... .......... .......... 12% 41.2M 1s\n",
      "  4750K .......... .......... .......... .......... .......... 12% 59.3M 1s\n",
      "  4800K .......... .......... .......... .......... .......... 13% 39.1M 1s\n",
      "  4850K .......... .......... .......... .......... .......... 13% 45.6M 1s\n",
      "  4900K .......... .......... .......... .......... .......... 13% 98.3M 1s\n",
      "  4950K .......... .......... .......... .......... .......... 13% 41.4M 1s\n",
      "  5000K .......... .......... .......... .......... .......... 13% 41.0M 1s\n",
      "  5050K .......... .......... .......... .......... .......... 13% 29.7M 1s\n",
      "  5100K .......... .......... .......... .......... .......... 13% 79.5M 1s\n",
      "  5150K .......... .......... .......... .......... .......... 13% 27.2M 1s\n",
      "  5200K .......... .......... .......... .......... .......... 14%  187M 1s\n",
      "  5250K .......... .......... .......... .......... .......... 14% 55.3M 1s\n",
      "  5300K .......... .......... .......... .......... .......... 14% 30.3M 1s\n",
      "  5350K .......... .......... .......... .......... .......... 14% 88.6M 1s\n",
      "  5400K .......... .......... .......... .......... .......... 14%  148M 1s\n",
      "  5450K .......... .......... .......... .......... .......... 14% 38.7M 1s\n",
      "  5500K .......... .......... .......... .......... .......... 14% 50.8M 1s\n",
      "  5550K .......... .......... .......... .......... .......... 15% 31.3M 1s\n",
      "  5600K .......... .......... .......... .......... .......... 15% 40.1M 1s\n",
      "  5650K .......... .......... .......... .......... .......... 15% 73.5M 1s\n",
      "  5700K .......... .......... .......... .......... .......... 15% 67.9M 1s\n",
      "  5750K .......... .......... .......... .......... .......... 15% 27.9M 1s\n",
      "  5800K .......... .......... .......... .......... .......... 15% 87.9M 1s\n",
      "  5850K .......... .......... .......... .......... .......... 15%  186M 1s\n",
      "  5900K .......... .......... .......... .......... .......... 16% 29.8M 1s\n",
      "  5950K .......... .......... .......... .......... .......... 16% 49.5M 1s\n",
      "  6000K .......... .......... .......... .......... .......... 16% 7.03M 1s\n",
      "  6050K .......... .......... .......... .......... .......... 16%  235M 1s\n",
      "  6100K .......... .......... .......... .......... .......... 16%  194M 1s\n",
      "  6150K .......... .......... .......... .......... .......... 16%  138M 1s\n",
      "  6200K .......... .......... .......... .......... .......... 16%  147M 1s\n",
      "  6250K .......... .......... .......... .......... .......... 16%  193M 1s\n",
      "  6300K .......... .......... .......... .......... .......... 17%  190M 1s\n",
      "  6350K .......... .......... .......... .......... .......... 17%  159M 1s\n",
      "  6400K .......... .......... .......... .......... .......... 17%  180M 1s\n",
      "  6450K .......... .......... .......... .......... .......... 17% 34.5M 1s\n",
      "  6500K .......... .......... .......... .......... .......... 17% 45.5M 1s\n",
      "  6550K .......... .......... .......... .......... .......... 17% 50.0M 1s\n",
      "  6600K .......... .......... .......... .......... .......... 17%  175M 1s\n",
      "  6650K .......... .......... .......... .......... .......... 18% 32.7M 1s\n",
      "  6700K .......... .......... .......... .......... .......... 18% 60.2M 1s\n",
      "  6750K .......... .......... .......... .......... .......... 18% 36.0M 1s\n",
      "  6800K .......... .......... .......... .......... .......... 18% 40.5M 1s\n",
      "  6850K .......... .......... .......... .......... .......... 18% 92.4M 1s\n",
      "  6900K .......... .......... .......... .......... .......... 18% 53.1M 1s\n",
      "  6950K .......... .......... .......... .......... .......... 18% 30.6M 1s\n",
      "  7000K .......... .......... .......... .......... .......... 18% 85.4M 1s\n",
      "  7050K .......... .......... .......... .......... .......... 19%  141M 1s\n",
      "  7100K .......... .......... .......... .......... .......... 19% 28.3M 1s\n",
      "  7150K .......... .......... .......... .......... .......... 19% 40.9M 1s\n",
      "  7200K .......... .......... .......... .......... .......... 19% 46.7M 1s\n",
      "  7250K .......... .......... .......... .......... .......... 19% 37.2M 1s\n",
      "  7300K .......... .......... .......... .......... .......... 19% 61.6M 1s\n",
      "  7350K .......... .......... .......... .......... .......... 19%  169M 1s\n",
      "  7400K .......... .......... .......... .......... .......... 20% 30.9M 1s\n",
      "  7450K .......... .......... .......... .......... .......... 20% 33.6M 1s\n",
      "  7500K .......... .......... .......... .......... .......... 20%  188M 1s\n",
      "  7550K .......... .......... .......... .......... .......... 20% 28.7M 1s\n",
      "  7600K .......... .......... .......... .......... .......... 20%  197M 1s\n",
      "  7650K .......... .......... .......... .......... .......... 20% 14.9M 1s\n",
      "  7700K .......... .......... .......... .......... .......... 20%  235M 1s\n",
      "  7750K .......... .......... .......... .......... .......... 20%  204M 1s\n",
      "  7800K .......... .......... .......... .......... .......... 21%  193M 1s\n",
      "  7850K .......... .......... .......... .......... .......... 21% 43.6M 1s\n",
      "  7900K .......... .......... .......... .......... .......... 21% 48.8M 1s\n",
      "  7950K .......... .......... .......... .......... .......... 21% 41.4M 1s\n",
      "  8000K .......... .......... .......... .......... .......... 21% 37.5M 1s\n",
      "  8050K .......... .......... .......... .......... .......... 21%  164M 1s\n",
      "  8100K .......... .......... .......... .......... .......... 21% 35.4M 1s\n",
      "  8150K .......... .......... .......... .......... .......... 22% 29.6M 1s\n",
      "  8200K .......... .......... .......... .......... .......... 22% 57.2M 1s\n",
      "  8250K .......... .......... .......... .......... .......... 22%  203M 1s\n",
      "  8300K .......... .......... .......... .......... .......... 22% 43.5M 1s\n",
      "  8350K .......... .......... .......... .......... .......... 22% 32.9M 1s\n",
      "  8400K .......... .......... .......... .......... .......... 22% 99.0M 1s\n",
      "  8450K .......... .......... .......... .......... .......... 22% 25.8M 1s\n",
      "  8500K .......... .......... .......... .......... .......... 23%  161M 1s\n",
      "  8550K .......... .......... .......... .......... .......... 23% 57.2M 1s\n",
      "  8600K .......... .......... .......... .......... .......... 23% 62.0M 1s\n",
      "  8650K .......... .......... .......... .......... .......... 23% 40.3M 1s\n",
      "  8700K .......... .......... .......... .......... .......... 23%  189M 1s\n",
      "  8750K .......... .......... .......... .......... .......... 23% 28.5M 1s\n",
      "  8800K .......... .......... .......... .......... .......... 23% 46.5M 1s\n",
      "  8850K .......... .......... .......... .......... .......... 23% 43.9M 1s\n",
      "  8900K .......... .......... .......... .......... .......... 24% 43.8M 1s\n",
      "  8950K .......... .......... .......... .......... .......... 24%  162M 1s\n",
      "  9000K .......... .......... .......... .......... .......... 24% 40.4M 1s\n",
      "  9050K .......... .......... .......... .......... .......... 24% 30.4M 1s\n",
      "  9100K .......... .......... .......... .......... .......... 24% 52.9M 1s\n",
      "  9150K .......... .......... .......... .......... .......... 24% 60.7M 1s\n",
      "  9200K .......... .......... .......... .......... .......... 24% 41.0M 1s\n",
      "  9250K .......... .......... .......... .......... .......... 25% 89.4M 1s\n",
      "  9300K .......... .......... .......... .......... .......... 25% 54.5M 1s\n",
      "  9350K .......... .......... .......... .......... .......... 25% 44.0M 1s\n",
      "  9400K .......... .......... .......... .......... .......... 25% 77.1M 1s\n",
      "  9450K .......... .......... .......... .......... .......... 25% 26.4M 1s\n",
      "  9500K .......... .......... .......... .......... .......... 25% 76.2M 1s\n",
      "  9550K .......... .......... .......... .......... .......... 25% 31.1M 1s\n",
      "  9600K .......... .......... .......... .......... .......... 25%  172M 1s\n",
      "  9650K .......... .......... .......... .......... .......... 26% 25.6M 1s\n",
      "  9700K .......... .......... .......... .......... .......... 26%  171M 1s\n",
      "  9750K .......... .......... .......... .......... .......... 26% 39.6M 1s\n",
      "  9800K .......... .......... .......... .......... .......... 26% 76.3M 1s\n",
      "  9850K .......... .......... .......... .......... .......... 26% 24.8M 1s\n",
      "  9900K .......... .......... .......... .......... .......... 26%  108M 1s\n",
      "  9950K .......... .......... .......... .......... .......... 26%  131M 1s\n",
      " 10000K .......... .......... .......... .......... .......... 27% 27.4M 1s\n",
      " 10050K .......... .......... .......... .......... .......... 27% 70.8M 1s\n",
      " 10100K .......... .......... .......... .......... .......... 27% 86.0M 1s\n",
      " 10150K .......... .......... .......... .......... .......... 27% 39.7M 1s\n",
      " 10200K .......... .......... .......... .......... .......... 27%  555K 1s\n",
      " 10250K .......... .......... .......... .......... .......... 27%  173M 1s\n",
      " 10300K .......... .......... .......... .......... .......... 27%  212M 1s\n",
      " 10350K .......... .......... .......... .......... .......... 27%  210M 1s\n",
      " 10400K .......... .......... .......... .......... .......... 28%  224M 1s\n",
      " 10450K .......... .......... .......... .......... .......... 28%  238M 1s\n",
      " 10500K .......... .......... .......... .......... .......... 28%  125M 1s\n",
      " 10550K .......... .......... .......... .......... .......... 28%  140M 1s\n",
      " 10600K .......... .......... .......... .......... .......... 28%  239M 1s\n",
      " 10650K .......... .......... .......... .......... .......... 28%  303M 1s\n",
      " 10700K .......... .......... .......... .......... .......... 28%  235M 1s\n",
      " 10750K .......... .......... .......... .......... .......... 29%  214M 1s\n",
      " 10800K .......... .......... .......... .......... .......... 29%  160M 1s\n",
      " 10850K .......... .......... .......... .......... .......... 29%  297M 1s\n",
      " 10900K .......... .......... .......... .......... .......... 29%  216M 1s\n",
      " 10950K .......... .......... .......... .......... .......... 29%  216M 1s\n",
      " 11000K .......... .......... .......... .......... .......... 29%  278M 1s\n",
      " 11050K .......... .......... .......... .......... .......... 29%  202M 1s\n",
      " 11100K .......... .......... .......... .......... .......... 30%  191M 1s\n",
      " 11150K .......... .......... .......... .......... .......... 30%  186M 1s\n",
      " 11200K .......... .......... .......... .......... .......... 30%  183M 1s\n",
      " 11250K .......... .......... .......... .......... .......... 30%  187M 1s\n",
      " 11300K .......... .......... .......... .......... .......... 30%  212M 1s\n",
      " 11350K .......... .......... .......... .......... .......... 30%  270M 1s\n",
      " 11400K .......... .......... .......... .......... .......... 30%  250M 1s\n",
      " 11450K .......... .......... .......... .......... .......... 30%  234M 1s\n",
      " 11500K .......... .......... .......... .......... .......... 31%  210M 1s\n",
      " 11550K .......... .......... .......... .......... .......... 31%  177M 1s\n",
      " 11600K .......... .......... .......... .......... .......... 31%  214M 1s\n",
      " 11650K .......... .......... .......... .......... .......... 31%  264M 1s\n",
      " 11700K .......... .......... .......... .......... .......... 31%  196M 1s\n",
      " 11750K .......... .......... .......... .......... .......... 31%  175M 1s\n",
      " 11800K .......... .......... .......... .......... .......... 31%  217M 1s\n",
      " 11850K .......... .......... .......... .......... .......... 32%  204M 1s\n",
      " 11900K .......... .......... .......... .......... .......... 32%  145M 1s\n",
      " 11950K .......... .......... .......... .......... .......... 32%  177M 1s\n",
      " 12000K .......... .......... .......... .......... .......... 32%  252M 1s\n",
      " 12050K .......... .......... .......... .......... .......... 32%  297M 1s\n",
      " 12100K .......... .......... .......... .......... .......... 32%  227M 1s\n",
      " 12150K .......... .......... .......... .......... .......... 32%  313M 1s\n",
      " 12200K .......... .......... .......... .......... .......... 32%  284M 1s\n",
      " 12250K .......... .......... .......... .......... .......... 33%  260M 1s\n",
      " 12300K .......... .......... .......... .......... .......... 33%  211M 1s\n",
      " 12350K .......... .......... .......... .......... .......... 33%  200M 1s\n",
      " 12400K .......... .......... .......... .......... .......... 33%  228M 1s\n",
      " 12450K .......... .......... .......... .......... .......... 33%  291M 1s\n",
      " 12500K .......... .......... .......... .......... .......... 33%  264M 1s\n",
      " 12550K .......... .......... .......... .......... .......... 33%  251M 1s\n",
      " 12600K .......... .......... .......... .......... .......... 34%  169M 1s\n",
      " 12650K .......... .......... .......... .......... .......... 34%  171M 1s\n",
      " 12700K .......... .......... .......... .......... .......... 34%  163M 1s\n",
      " 12750K .......... .......... .......... .......... .......... 34%  187M 1s\n",
      " 12800K .......... .......... .......... .......... .......... 34%  209M 1s\n",
      " 12850K .......... .......... .......... .......... .......... 34%  190M 1s\n",
      " 12900K .......... .......... .......... .......... .......... 34%  190M 1s\n",
      " 12950K .......... .......... .......... .......... .......... 34%  189M 1s\n",
      " 13000K .......... .......... .......... .......... .......... 35%  213M 1s\n",
      " 13050K .......... .......... .......... .......... .......... 35%  218M 1s\n",
      " 13100K .......... .......... .......... .......... .......... 35%  173M 1s\n",
      " 13150K .......... .......... .......... .......... .......... 35%  185M 1s\n",
      " 13200K .......... .......... .......... .......... .......... 35%  209M 1s\n",
      " 13250K .......... .......... .......... .......... .......... 35%  220M 1s\n",
      " 13300K .......... .......... .......... .......... .......... 35%  192M 1s\n",
      " 13350K .......... .......... .......... .......... .......... 36%  200M 1s\n",
      " 13400K .......... .......... .......... .......... .......... 36%  248M 1s\n",
      " 13450K .......... .......... .......... .......... .......... 36%  214M 1s\n",
      " 13500K .......... .......... .......... .......... .......... 36%  176M 1s\n",
      " 13550K .......... .......... .......... .......... .......... 36%  201M 1s\n",
      " 13600K .......... .......... .......... .......... .......... 36%  188M 1s\n",
      " 13650K .......... .......... .......... .......... .......... 36%  218M 1s\n",
      " 13700K .......... .......... .......... .......... .......... 37%  194M 1s\n",
      " 13750K .......... .......... .......... .......... .......... 37%  205M 1s\n",
      " 13800K .......... .......... .......... .......... .......... 37%  184M 1s\n",
      " 13850K .......... .......... .......... .......... .......... 37%  283M 1s\n",
      " 13900K .......... .......... .......... .......... .......... 37%  210M 1s\n",
      " 13950K .......... .......... .......... .......... .......... 37%  304M 1s\n",
      " 14000K .......... .......... .......... .......... .......... 37%  302M 1s\n",
      " 14050K .......... .......... .......... .......... .......... 37%  204M 1s\n",
      " 14100K .......... .......... .......... .......... .......... 38%  190M 1s\n",
      " 14150K .......... .......... .......... .......... .......... 38%  211M 1s\n",
      " 14200K .......... .......... .......... .......... .......... 38%  207M 1s\n",
      " 14250K .......... .......... .......... .......... .......... 38%  189M 1s\n",
      " 14300K .......... .......... .......... .......... .......... 38%  218M 0s\n",
      " 14350K .......... .......... .......... .......... .......... 38%  212M 0s\n",
      " 14400K .......... .......... .......... .......... .......... 38%  261M 0s\n",
      " 14450K .......... .......... .......... .......... .......... 39%  197M 0s\n",
      " 14500K .......... .......... .......... .......... .......... 39%  184M 0s\n",
      " 14550K .......... .......... .......... .......... .......... 39%  212M 0s\n",
      " 14600K .......... .......... .......... .......... .......... 39%  217M 0s\n",
      " 14650K .......... .......... .......... .......... .......... 39%  194M 0s\n",
      " 14700K .......... .......... .......... .......... .......... 39%  185M 0s\n",
      " 14750K .......... .......... .......... .......... .......... 39%  206M 0s\n",
      " 14800K .......... .......... .......... .......... .......... 39%  170M 0s\n",
      " 14850K .......... .......... .......... .......... .......... 40%  282M 0s\n",
      " 14900K .......... .......... .......... .......... .......... 40%  171M 0s\n",
      " 14950K .......... .......... .......... .......... .......... 40%  162M 0s\n",
      " 15000K .......... .......... .......... .......... .......... 40%  169M 0s\n",
      " 15050K .......... .......... .......... .......... .......... 40%  158M 0s\n",
      " 15100K .......... .......... .......... .......... .......... 40%  160M 0s\n",
      " 15150K .......... .......... .......... .......... .......... 40%  173M 0s\n",
      " 15200K .......... .......... .......... .......... .......... 41%  184M 0s\n",
      " 15250K .......... .......... .......... .......... .......... 41%  169M 0s\n",
      " 15300K .......... .......... .......... .......... .......... 41%  169M 0s\n",
      " 15350K .......... .......... .......... .......... .......... 41%  207M 0s\n",
      " 15400K .......... .......... .......... .......... .......... 41%  181M 0s\n",
      " 15450K .......... .......... .......... .......... .......... 41%  213M 0s\n",
      " 15500K .......... .......... .......... .......... .......... 41%  200M 0s\n",
      " 15550K .......... .......... .......... .......... .......... 41%  163M 0s\n",
      " 15600K .......... .......... .......... .......... .......... 42%  154M 0s\n",
      " 15650K .......... .......... .......... .......... .......... 42%  163M 0s\n",
      " 15700K .......... .......... .......... .......... .......... 42%  173M 0s\n",
      " 15750K .......... .......... .......... .......... .......... 42%  149M 0s\n",
      " 15800K .......... .......... .......... .......... .......... 42%  189M 0s\n",
      " 15850K .......... .......... .......... .......... .......... 42%  192M 0s\n",
      " 15900K .......... .......... .......... .......... .......... 42%  178M 0s\n",
      " 15950K .......... .......... .......... .......... .......... 43%  235M 0s\n",
      " 16000K .......... .......... .......... .......... .......... 43%  214M 0s\n",
      " 16050K .......... .......... .......... .......... .......... 43%  192M 0s\n",
      " 16100K .......... .......... .......... .......... .......... 43%  160M 0s\n",
      " 16150K .......... .......... .......... .......... .......... 43%  191M 0s\n",
      " 16200K .......... .......... .......... .......... .......... 43%  215M 0s\n",
      " 16250K .......... .......... .......... .......... .......... 43%  211M 0s\n",
      " 16300K .......... .......... .......... .......... .......... 44%  169M 0s\n",
      " 16350K .......... .......... .......... .......... .......... 44%  170M 0s\n",
      " 16400K .......... .......... .......... .......... .......... 44%  210M 0s\n",
      " 16450K .......... .......... .......... .......... .......... 44%  218M 0s\n",
      " 16500K .......... .......... .......... .......... .......... 44%  163M 0s\n",
      " 16550K .......... .......... .......... .......... .......... 44%  191M 0s\n",
      " 16600K .......... .......... .......... .......... .......... 44%  212M 0s\n",
      " 16650K .......... .......... .......... .......... .......... 44%  214M 0s\n",
      " 16700K .......... .......... .......... .......... .......... 45%  165M 0s\n",
      " 16750K .......... .......... .......... .......... .......... 45%  194M 0s\n",
      " 16800K .......... .......... .......... .......... .......... 45%  215M 0s\n",
      " 16850K .......... .......... .......... .......... .......... 45%  217M 0s\n",
      " 16900K .......... .......... .......... .......... .......... 45%  182M 0s\n",
      " 16950K .......... .......... .......... .......... .......... 45%  192M 0s\n",
      " 17000K .......... .......... .......... .......... .......... 45%  216M 0s\n",
      " 17050K .......... .......... .......... .......... .......... 46%  210M 0s\n",
      " 17100K .......... .......... .......... .......... .......... 46%  171M 0s\n",
      " 17150K .......... .......... .......... .......... .......... 46%  188M 0s\n",
      " 17200K .......... .......... .......... .......... .......... 46%  216M 0s\n",
      " 17250K .......... .......... .......... .......... .......... 46%  215M 0s\n",
      " 17300K .......... .......... .......... .......... .......... 46%  174M 0s\n",
      " 17350K .......... .......... .......... .......... .......... 46%  193M 0s\n",
      " 17400K .......... .......... .......... .......... .......... 46%  213M 0s\n",
      " 17450K .......... .......... .......... .......... .......... 47%  212M 0s\n",
      " 17500K .......... .......... .......... .......... .......... 47%  171M 0s\n",
      " 17550K .......... .......... .......... .......... .......... 47%  194M 0s\n",
      " 17600K .......... .......... .......... .......... .......... 47%  219M 0s\n",
      " 17650K .......... .......... .......... .......... .......... 47%  212M 0s\n",
      " 17700K .......... .......... .......... .......... .......... 47%  185M 0s\n",
      " 17750K .......... .......... .......... .......... .......... 47%  188M 0s\n",
      " 17800K .......... .......... .......... .......... .......... 48%  216M 0s\n",
      " 17850K .......... .......... .......... .......... .......... 48%  218M 0s\n",
      " 17900K .......... .......... .......... .......... .......... 48%  171M 0s\n",
      " 17950K .......... .......... .......... .......... .......... 48%  193M 0s\n",
      " 18000K .......... .......... .......... .......... .......... 48%  180M 0s\n",
      " 18050K .......... .......... .......... .......... .......... 48%  211M 0s\n",
      " 18100K .......... .......... .......... .......... .......... 48%  184M 0s\n",
      " 18150K .......... .......... .......... .......... .......... 48%  188M 0s\n",
      " 18200K .......... .......... .......... .......... .......... 49%  212M 0s\n",
      " 18250K .......... .......... .......... .......... .......... 49%  215M 0s\n",
      " 18300K .......... .......... .......... .......... .......... 49%  172M 0s\n",
      " 18350K .......... .......... .......... .......... .......... 49%  192M 0s\n",
      " 18400K .......... .......... .......... .......... .......... 49%  208M 0s\n",
      " 18450K .......... .......... .......... .......... .......... 49%  220M 0s\n",
      " 18500K .......... .......... .......... .......... .......... 49%  188M 0s\n",
      " 18550K .......... .......... .......... .......... .......... 50%  197M 0s\n",
      " 18600K .......... .......... .......... .......... .......... 50%  216M 0s\n",
      " 18650K .......... .......... .......... .......... .......... 50%  202M 0s\n",
      " 18700K .......... .......... .......... .......... .......... 50%  172M 0s\n",
      " 18750K .......... .......... .......... .......... .......... 50%  192M 0s\n",
      " 18800K .......... .......... .......... .......... .......... 50%  219M 0s\n",
      " 18850K .......... .......... .......... .......... .......... 50%  197M 0s\n",
      " 18900K .......... .......... .......... .......... .......... 50%  178M 0s\n",
      " 18950K .......... .......... .......... .......... .......... 51%  188M 0s\n",
      " 19000K .......... .......... .......... .......... .......... 51%  213M 0s\n",
      " 19050K .......... .......... .......... .......... .......... 51%  218M 0s\n",
      " 19100K .......... .......... .......... .......... .......... 51%  173M 0s\n",
      " 19150K .......... .......... .......... .......... .......... 51%  190M 0s\n",
      " 19200K .......... .......... .......... .......... .......... 51%  211M 0s\n",
      " 19250K .......... .......... .......... .......... .......... 51%  216M 0s\n",
      " 19300K .......... .......... .......... .......... .......... 52%  195M 0s\n",
      " 19350K .......... .......... .......... .......... .......... 52%  121M 0s\n",
      " 19400K .......... .......... .......... .......... .......... 52%  211M 0s\n",
      " 19450K .......... .......... .......... .......... .......... 52%  214M 0s\n",
      " 19500K .......... .......... .......... .......... .......... 52%  174M 0s\n",
      " 19550K .......... .......... .......... .......... .......... 52%  193M 0s\n",
      " 19600K .......... .......... .......... .......... .......... 52%  214M 0s\n",
      " 19650K .......... .......... .......... .......... .......... 53%  213M 0s\n",
      " 19700K .......... .......... .......... .......... .......... 53%  174M 0s\n",
      " 19750K .......... .......... .......... .......... .......... 53%  191M 0s\n",
      " 19800K .......... .......... .......... .......... .......... 53%  216M 0s\n",
      " 19850K .......... .......... .......... .......... .......... 53%  261M 0s\n",
      " 19900K .......... .......... .......... .......... .......... 53%  245M 0s\n",
      " 19950K .......... .......... .......... .......... .......... 53%  302M 0s\n",
      " 20000K .......... .......... .......... .......... .......... 53%  244M 0s\n",
      " 20050K .......... .......... .......... .......... .......... 54%  304M 0s\n",
      " 20100K .......... .......... .......... .......... .......... 54%  265M 0s\n",
      " 20150K .......... .......... .......... .......... .......... 54%  214M 0s\n",
      " 20200K .......... .......... .......... .......... .......... 54%  204M 0s\n",
      " 20250K .......... .......... .......... .......... .......... 54%  196M 0s\n",
      " 20300K .......... .......... .......... .......... .......... 54%  183M 0s\n",
      " 20350K .......... .......... .......... .......... .......... 54%  211M 0s\n",
      " 20400K .......... .......... .......... .......... .......... 55%  201M 0s\n",
      " 20450K .......... .......... .......... .......... .......... 55%  879K 0s\n",
      " 20500K .......... .......... .......... .......... .......... 55%  118M 0s\n",
      " 20550K .......... .......... .......... .......... .......... 55%  134M 0s\n",
      " 20600K .......... .......... .......... .......... .......... 55%  116M 0s\n",
      " 20650K .......... .......... .......... .......... .......... 55%  125M 0s\n",
      " 20700K .......... .......... .......... .......... .......... 55%  182M 0s\n",
      " 20750K .......... .......... .......... .......... .......... 55%  104M 0s\n",
      " 20800K .......... .......... .......... .......... .......... 56%  300M 0s\n",
      " 20850K .......... .......... .......... .......... .......... 56%  209M 0s\n",
      " 20900K .......... .......... .......... .......... .......... 56%  299M 0s\n",
      " 20950K .......... .......... .......... .......... .......... 56%  189M 0s\n",
      " 21000K .......... .......... .......... .......... .......... 56%  202M 0s\n",
      " 21050K .......... .......... .......... .......... .......... 56%  264M 0s\n",
      " 21100K .......... .......... .......... .......... .......... 56%  249M 0s\n",
      " 21150K .......... .......... .......... .......... .......... 57%  216M 0s\n",
      " 21200K .......... .......... .......... .......... .......... 57%  318M 0s\n",
      " 21250K .......... .......... .......... .......... .......... 57%  205M 0s\n",
      " 21300K .......... .......... .......... .......... .......... 57%  240M 0s\n",
      " 21350K .......... .......... .......... .......... .......... 57%  214M 0s\n",
      " 21400K .......... .......... .......... .......... .......... 57%  244M 0s\n",
      " 21450K .......... .......... .......... .......... .......... 57%  243M 0s\n",
      " 21500K .......... .......... .......... .......... .......... 57%  229M 0s\n",
      " 21550K .......... .......... .......... .......... .......... 58%  231M 0s\n",
      " 21600K .......... .......... .......... .......... .......... 58%  236M 0s\n",
      " 21650K .......... .......... .......... .......... .......... 58%  276M 0s\n",
      " 21700K .......... .......... .......... .......... .......... 58%  246M 0s\n",
      " 21750K .......... .......... .......... .......... .......... 58%  230M 0s\n",
      " 21800K .......... .......... .......... .......... .......... 58%  177M 0s\n",
      " 21850K .......... .......... .......... .......... .......... 58%  208M 0s\n",
      " 21900K .......... .......... .......... .......... .......... 59%  112M 0s\n",
      " 21950K .......... .......... .......... .......... .......... 59%  150M 0s\n",
      " 22000K .......... .......... .......... .......... .......... 59%  145M 0s\n",
      " 22050K .......... .......... .......... .......... .......... 59%  133M 0s\n",
      " 22100K .......... .......... .......... .......... .......... 59%  146M 0s\n",
      " 22150K .......... .......... .......... .......... .......... 59%  192M 0s\n",
      " 22200K .......... .......... .......... .......... .......... 59%  192M 0s\n",
      " 22250K .......... .......... .......... .......... .......... 60%  160M 0s\n",
      " 22300K .......... .......... .......... .......... .......... 60%  130M 0s\n",
      " 22350K .......... .......... .......... .......... .......... 60% 99.0M 0s\n",
      " 22400K .......... .......... .......... .......... .......... 60%  128M 0s\n",
      " 22450K .......... .......... .......... .......... .......... 60%  126M 0s\n",
      " 22500K .......... .......... .......... .......... .......... 60%  183M 0s\n",
      " 22550K .......... .......... .......... .......... .......... 60%  133M 0s\n",
      " 22600K .......... .......... .......... .......... .......... 60%  209M 0s\n",
      " 22650K .......... .......... .......... .......... .......... 61%  133M 0s\n",
      " 22700K .......... .......... .......... .......... .......... 61%  104M 0s\n",
      " 22750K .......... .......... .......... .......... .......... 61%  121M 0s\n",
      " 22800K .......... .......... .......... .......... .......... 61%  129M 0s\n",
      " 22850K .......... .......... .......... .......... .......... 61%  146M 0s\n",
      " 22900K .......... .......... .......... .......... .......... 61%  205M 0s\n",
      " 22950K .......... .......... .......... .......... .......... 61% 99.5M 0s\n",
      " 23000K .......... .......... .......... .......... .......... 62%  177M 0s\n",
      " 23050K .......... .......... .......... .......... .......... 62%  120M 0s\n",
      " 23100K .......... .......... .......... .......... .......... 62%  110M 0s\n",
      " 23150K .......... .......... .......... .......... .......... 62% 97.6M 0s\n",
      " 23200K .......... .......... .......... .......... .......... 62%  133M 0s\n",
      " 23250K .......... .......... .......... .......... .......... 62%  120M 0s\n",
      " 23300K .......... .......... .......... .......... .......... 62%  180M 0s\n",
      " 23350K .......... .......... .......... .......... .......... 62%  183M 0s\n",
      " 23400K .......... .......... .......... .......... .......... 63%  200M 0s\n",
      " 23450K .......... .......... .......... .......... .......... 63%  218M 0s\n",
      " 23500K .......... .......... .......... .......... .......... 63%  141M 0s\n",
      " 23550K .......... .......... .......... .......... .......... 63%  100M 0s\n",
      " 23600K .......... .......... .......... .......... .......... 63%  124M 0s\n",
      " 23650K .......... .......... .......... .......... .......... 63%  170M 0s\n",
      " 23700K .......... .......... .......... .......... .......... 63%  174M 0s\n",
      " 23750K .......... .......... .......... .......... .......... 64%  122M 0s\n",
      " 23800K .......... .......... .......... .......... .......... 64%  214M 0s\n",
      " 23850K .......... .......... .......... .......... .......... 64%  197M 0s\n",
      " 23900K .......... .......... .......... .......... .......... 64%  259M 0s\n",
      " 23950K .......... .......... .......... .......... .......... 64%  196M 0s\n",
      " 24000K .......... .......... .......... .......... .......... 64%  217M 0s\n",
      " 24050K .......... .......... .......... .......... .......... 64%  201M 0s\n",
      " 24100K .......... .......... .......... .......... .......... 64%  199M 0s\n",
      " 24150K .......... .......... .......... .......... .......... 65%  239M 0s\n",
      " 24200K .......... .......... .......... .......... .......... 65%  239M 0s\n",
      " 24250K .......... .......... .......... .......... .......... 65%  211M 0s\n",
      " 24300K .......... .......... .......... .......... .......... 65%  203M 0s\n",
      " 24350K .......... .......... .......... .......... .......... 65%  175M 0s\n",
      " 24400K .......... .......... .......... .......... .......... 65%  189M 0s\n",
      " 24450K .......... .......... .......... .......... .......... 65%  213M 0s\n",
      " 24500K .......... .......... .......... .......... .......... 66%  202M 0s\n",
      " 24550K .......... .......... .......... .......... .......... 66%  176M 0s\n",
      " 24600K .......... .......... .......... .......... .......... 66%  195M 0s\n",
      " 24650K .......... .......... .......... .......... .......... 66%  869K 0s\n",
      " 24700K .......... .......... .......... .......... .......... 66%  127M 0s\n",
      " 24750K .......... .......... .......... .......... .......... 66%  135M 0s\n",
      " 24800K .......... .......... .......... .......... .......... 66% 10.2M 0s\n",
      " 24850K .......... .......... .......... .......... .......... 67%  134M 0s\n",
      " 24900K .......... .......... .......... .......... .......... 67%  156M 0s\n",
      " 24950K .......... .......... .......... .......... .......... 67%  131M 0s\n",
      " 25000K .......... .......... .......... .......... .......... 67%  166M 0s\n",
      " 25050K .......... .......... .......... .......... .......... 67%  150M 0s\n",
      " 25100K .......... .......... .......... .......... .......... 67%  102M 0s\n",
      " 25150K .......... .......... .......... .......... .......... 67%  128M 0s\n",
      " 25200K .......... .......... .......... .......... .......... 67%  196M 0s\n",
      " 25250K .......... .......... .......... .......... .......... 68%  198M 0s\n",
      " 25300K .......... .......... .......... .......... .......... 68%  189M 0s\n",
      " 25350K .......... .......... .......... .......... .......... 68%  246M 0s\n",
      " 25400K .......... .......... .......... .......... .......... 68%  269M 0s\n",
      " 25450K .......... .......... .......... .......... .......... 68%  243M 0s\n",
      " 25500K .......... .......... .......... .......... .......... 68%  206M 0s\n",
      " 25550K .......... .......... .......... .......... .......... 68%  165M 0s\n",
      " 25600K .......... .......... .......... .......... .......... 69%  204M 0s\n",
      " 25650K .......... .......... .......... .......... .......... 69%  186M 0s\n",
      " 25700K .......... .......... .......... .......... .......... 69%  168M 0s\n",
      " 25750K .......... .......... .......... .......... .......... 69%  153M 0s\n",
      " 25800K .......... .......... .......... .......... .......... 69%  161M 0s\n",
      " 25850K .......... .......... .......... .......... .......... 69%  158M 0s\n",
      " 25900K .......... .......... .......... .......... .......... 69%  185M 0s\n",
      " 25950K .......... .......... .......... .......... .......... 69%  129M 0s\n",
      " 26000K .......... .......... .......... .......... .......... 70%  208M 0s\n",
      " 26050K .......... .......... .......... .......... .......... 70%  209M 0s\n",
      " 26100K .......... .......... .......... .......... .......... 70%  153M 0s\n",
      " 26150K .......... .......... .......... .......... .......... 70%  149M 0s\n",
      " 26200K .......... .......... .......... .......... .......... 70%  205M 0s\n",
      " 26250K .......... .......... .......... .......... .......... 70%  160M 0s\n",
      " 26300K .......... .......... .......... .......... .......... 70%  173M 0s\n",
      " 26350K .......... .......... .......... .......... .......... 71%  163M 0s\n",
      " 26400K .......... .......... .......... .......... .......... 71%  196M 0s\n",
      " 26450K .......... .......... .......... .......... .......... 71%  155M 0s\n",
      " 26500K .......... .......... .......... .......... .......... 71%  160M 0s\n",
      " 26550K .......... .......... .......... .......... .......... 71%  187M 0s\n",
      " 26600K .......... .......... .......... .......... .......... 71%  181M 0s\n",
      " 26650K .......... .......... .......... .......... .......... 71%  140M 0s\n",
      " 26700K .......... .......... .......... .......... .......... 71%  213M 0s\n",
      " 26750K .......... .......... .......... .......... .......... 72%  168M 0s\n",
      " 26800K .......... .......... .......... .......... .......... 72%  168M 0s\n",
      " 26850K .......... .......... .......... .......... .......... 72%  159M 0s\n",
      " 26900K .......... .......... .......... .......... .......... 72%  179M 0s\n",
      " 26950K .......... .......... .......... .......... .......... 72%  164M 0s\n",
      " 27000K .......... .......... .......... .......... .......... 72%  172M 0s\n",
      " 27050K .......... .......... .......... .......... .......... 72%  171M 0s\n",
      " 27100K .......... .......... .......... .......... .......... 73%  199M 0s\n",
      " 27150K .......... .......... .......... .......... .......... 73%  168M 0s\n",
      " 27200K .......... .......... .......... .......... .......... 73%  169M 0s\n",
      " 27250K .......... .......... .......... .......... .......... 73%  189M 0s\n",
      " 27300K .......... .......... .......... .......... .......... 73%  179M 0s\n",
      " 27350K .......... .......... .......... .......... .......... 73%  144M 0s\n",
      " 27400K .......... .......... .......... .......... .......... 73%  155M 0s\n",
      " 27450K .......... .......... .......... .......... .......... 74%  164M 0s\n",
      " 27500K .......... .......... .......... .......... .......... 74%  175M 0s\n",
      " 27550K .......... .......... .......... .......... .......... 74%  104M 0s\n",
      " 27600K .......... .......... .......... .......... .......... 74%  159M 0s\n",
      " 27650K .......... .......... .......... .......... .......... 74%  175M 0s\n",
      " 27700K .......... .......... .......... .......... .......... 74%  176M 0s\n",
      " 27750K .......... .......... .......... .......... .......... 74%  181M 0s\n",
      " 27800K .......... .......... .......... .......... .......... 74%  127M 0s\n",
      " 27850K .......... .......... .......... .......... .......... 75%  164M 0s\n",
      " 27900K .......... .......... .......... .......... .......... 75%  171M 0s\n",
      " 27950K .......... .......... .......... .......... .......... 75%  196M 0s\n",
      " 28000K .......... .......... .......... .......... .......... 75%  155M 0s\n",
      " 28050K .......... .......... .......... .......... .......... 75%  126M 0s\n",
      " 28100K .......... .......... .......... .......... .......... 75%  187M 0s\n",
      " 28150K .......... .......... .......... .......... .......... 75%  154M 0s\n",
      " 28200K .......... .......... .......... .......... .......... 76%  205M 0s\n",
      " 28250K .......... .......... .......... .......... .......... 76%  194M 0s\n",
      " 28300K .......... .......... .......... .......... .......... 76%  128M 0s\n",
      " 28350K .......... .......... .......... .......... .......... 76%  153M 0s\n",
      " 28400K .......... .......... .......... .......... .......... 76%  217M 0s\n",
      " 28450K .......... .......... .......... .......... .......... 76%  190M 0s\n",
      " 28500K .......... .......... .......... .......... .......... 76%  184M 0s\n",
      " 28550K .......... .......... .......... .......... .......... 76%  113M 0s\n",
      " 28600K .......... .......... .......... .......... .......... 77%  185M 0s\n",
      " 28650K .......... .......... .......... .......... .......... 77%  202M 0s\n",
      " 28700K .......... .......... .......... .......... .......... 77%  173M 0s\n",
      " 28750K .......... .......... .......... .......... .......... 77%  152M 0s\n",
      " 28800K .......... .......... .......... .......... .......... 77%  208M 0s\n",
      " 28850K .......... .......... .......... .......... .......... 77%  923K 0s\n",
      " 28900K .......... .......... .......... .......... .......... 77%  156M 0s\n",
      " 28950K .......... .......... .......... .......... .......... 78%  173M 0s\n",
      " 29000K .......... .......... .......... .......... .......... 78%  177M 0s\n",
      " 29050K .......... .......... .......... .......... .......... 78%  142M 0s\n",
      " 29100K .......... .......... .......... .......... .......... 78%  205M 0s\n",
      " 29150K .......... .......... .......... .......... .......... 78%  139M 0s\n",
      " 29200K .......... .......... .......... .......... .......... 78%  174M 0s\n",
      " 29250K .......... .......... .......... .......... .......... 78%  155M 0s\n",
      " 29300K .......... .......... .......... .......... .......... 78%  193M 0s\n",
      " 29350K .......... .......... .......... .......... .......... 79%  185M 0s\n",
      " 29400K .......... .......... .......... .......... .......... 79%  146M 0s\n",
      " 29450K .......... .......... .......... .......... .......... 79%  175M 0s\n",
      " 29500K .......... .......... .......... .......... .......... 79%  173M 0s\n",
      " 29550K .......... .......... .......... .......... .......... 79%  135M 0s\n",
      " 29600K .......... .......... .......... .......... .......... 79%  179M 0s\n",
      " 29650K .......... .......... .......... .......... .......... 79%  180M 0s\n",
      " 29700K .......... .......... .......... .......... .......... 80%  195M 0s\n",
      " 29750K .......... .......... .......... .......... .......... 80%  148M 0s\n",
      " 29800K .......... .......... .......... .......... .......... 80%  181M 0s\n",
      " 29850K .......... .......... .......... .......... .......... 80%  189M 0s\n",
      " 29900K .......... .......... .......... .......... .......... 80%  207M 0s\n",
      " 29950K .......... .......... .......... .......... .......... 80%  130M 0s\n",
      " 30000K .......... .......... .......... .......... .......... 80%  194M 0s\n",
      " 30050K .......... .......... .......... .......... .......... 81%  162M 0s\n",
      " 30100K .......... .......... .......... .......... .......... 81%  153M 0s\n",
      " 30150K .......... .......... .......... .......... .......... 81%  189M 0s\n",
      " 30200K .......... .......... .......... .......... .......... 81%  199M 0s\n",
      " 30250K .......... .......... .......... .......... .......... 81%  156M 0s\n",
      " 30300K .......... .......... .......... .......... .......... 81%  197M 0s\n",
      " 30350K .......... .......... .......... .......... .......... 81%  153M 0s\n",
      " 30400K .......... .......... .......... .......... .......... 81%  207M 0s\n",
      " 30450K .......... .......... .......... .......... .......... 82%  183M 0s\n",
      " 30500K .......... .......... .......... .......... .......... 82%  214M 0s\n",
      " 30550K .......... .......... .......... .......... .......... 82%  193M 0s\n",
      " 30600K .......... .......... .......... .......... .......... 82%  200M 0s\n",
      " 30650K .......... .......... .......... .......... .......... 82%  190M 0s\n",
      " 30700K .......... .......... .......... .......... .......... 82%  542K 0s\n",
      " 30750K .......... .......... .......... .......... .......... 82%  126M 0s\n",
      " 30800K .......... .......... .......... .......... .......... 83%  235M 0s\n",
      " 30850K .......... .......... .......... .......... .......... 83%  239M 0s\n",
      " 30900K .......... .......... .......... .......... .......... 83%  213M 0s\n",
      " 30950K .......... .......... .......... .......... .......... 83%  145M 0s\n",
      " 31000K .......... .......... .......... .......... .......... 83%  247M 0s\n",
      " 31050K .......... .......... .......... .......... .......... 83% 66.9M 0s\n",
      " 31100K .......... .......... .......... .......... .......... 83%  276M 0s\n",
      " 31150K .......... .......... .......... .......... .......... 83%  128M 0s\n",
      " 31200K .......... .......... .......... .......... .......... 84%  159M 0s\n",
      " 31250K .......... .......... .......... .......... .......... 84%  160M 0s\n",
      " 31300K .......... .......... .......... .......... .......... 84%  162M 0s\n",
      " 31350K .......... .......... .......... .......... .......... 84%  178M 0s\n",
      " 31400K .......... .......... .......... .......... .......... 84%  163M 0s\n",
      " 31450K .......... .......... .......... .......... .......... 84%  175M 0s\n",
      " 31500K .......... .......... .......... .......... .......... 84%  208M 0s\n",
      " 31550K .......... .......... .......... .......... .......... 85%  162M 0s\n",
      " 31600K .......... .......... .......... .......... .......... 85%  191M 0s\n",
      " 31650K .......... .......... .......... .......... .......... 85%  218M 0s\n",
      " 31700K .......... .......... .......... .......... .......... 85%  230M 0s\n",
      " 31750K .......... .......... .......... .......... .......... 85%  217M 0s\n",
      " 31800K .......... .......... .......... .......... .......... 85%  172M 0s\n",
      " 31850K .......... .......... .......... .......... .......... 85%  220M 0s\n",
      " 31900K .......... .......... .......... .......... .......... 85%  182M 0s\n",
      " 31950K .......... .......... .......... .......... .......... 86%  215M 0s\n",
      " 32000K .......... .......... .......... .......... .......... 86%  163M 0s\n",
      " 32050K .......... .......... .......... .......... .......... 86%  176M 0s\n",
      " 32100K .......... .......... .......... .......... .......... 86%  163M 0s\n",
      " 32150K .......... .......... .......... .......... .......... 86%  196M 0s\n",
      " 32200K .......... .......... .......... .......... .......... 86%  244M 0s\n",
      " 32250K .......... .......... .......... .......... .......... 86%  227M 0s\n",
      " 32300K .......... .......... .......... .......... .......... 87%  179M 0s\n",
      " 32350K .......... .......... .......... .......... .......... 87%  184M 0s\n",
      " 32400K .......... .......... .......... .......... .......... 87%  192M 0s\n",
      " 32450K .......... .......... .......... .......... .......... 87%  159M 0s\n",
      " 32500K .......... .......... .......... .......... .......... 87%  176M 0s\n",
      " 32550K .......... .......... .......... .......... .......... 87%  172M 0s\n",
      " 32600K .......... .......... .......... .......... .......... 87%  186M 0s\n",
      " 32650K .......... .......... .......... .......... .......... 88%  158M 0s\n",
      " 32700K .......... .......... .......... .......... .......... 88%  209M 0s\n",
      " 32750K .......... .......... .......... .......... .......... 88%  260M 0s\n",
      " 32800K .......... .......... .......... .......... .......... 88%  211M 0s\n",
      " 32850K .......... .......... .......... .......... .......... 88%  132M 0s\n",
      " 32900K .......... .......... .......... .......... .......... 88%  303M 0s\n",
      " 32950K .......... .......... .......... .......... .......... 88%  212M 0s\n",
      " 33000K .......... .......... .......... .......... .......... 88%  198M 0s\n",
      " 33050K .......... .......... .......... .......... .......... 89%  132M 0s\n",
      " 33100K .......... .......... .......... .......... .......... 89%  222M 0s\n",
      " 33150K .......... .......... .......... .......... .......... 89%  260M 0s\n",
      " 33200K .......... .......... .......... .......... .......... 89%  224M 0s\n",
      " 33250K .......... .......... .......... .......... .......... 89%  159M 0s\n",
      " 33300K .......... .......... .......... .......... .......... 89%  209M 0s\n",
      " 33350K .......... .......... .......... .......... .......... 89%  199M 0s\n",
      " 33400K .......... .......... .......... .......... .......... 90%  215M 0s\n",
      " 33450K .......... .......... .......... .......... .......... 90%  163M 0s\n",
      " 33500K .......... .......... .......... .......... .......... 90%  213M 0s\n",
      " 33550K .......... .......... .......... .......... .......... 90%  219M 0s\n",
      " 33600K .......... .......... .......... .......... .......... 90%  313M 0s\n",
      " 33650K .......... .......... .......... .......... .......... 90%  263M 0s\n",
      " 33700K .......... .......... .......... .......... .......... 90%  227M 0s\n",
      " 33750K .......... .......... .......... .......... .......... 90%  227M 0s\n",
      " 33800K .......... .......... .......... .......... .......... 91%  203M 0s\n",
      " 33850K .......... .......... .......... .......... .......... 91%  215M 0s\n",
      " 33900K .......... .......... .......... .......... .......... 91%  324M 0s\n",
      " 33950K .......... .......... .......... .......... .......... 91%  210M 0s\n",
      " 34000K .......... .......... .......... .......... .......... 91%  210M 0s\n",
      " 34050K .......... .......... .......... .......... .......... 91%  177M 0s\n",
      " 34100K .......... .......... .......... .......... .......... 91%  210M 0s\n",
      " 34150K .......... .......... .......... .......... .......... 92%  204M 0s\n",
      " 34200K .......... .......... .......... .......... .......... 92%  310M 0s\n",
      " 34250K .......... .......... .......... .......... .......... 92%  193M 0s\n",
      " 34300K .......... .......... .......... .......... .......... 92%  209M 0s\n",
      " 34350K .......... .......... .......... .......... .......... 92%  287M 0s\n",
      " 34400K .......... .......... .......... .......... .......... 92%  256M 0s\n",
      " 34450K .......... .......... .......... .......... .......... 92%  198M 0s\n",
      " 34500K .......... .......... .......... .......... .......... 92%  204M 0s\n",
      " 34550K .......... .......... .......... .......... .......... 93%  202M 0s\n",
      " 34600K .......... .......... .......... .......... .......... 93%  217M 0s\n",
      " 34650K .......... .......... .......... .......... .......... 93%  186M 0s\n",
      " 34700K .......... .......... .......... .......... .......... 93%  206M 0s\n",
      " 34750K .......... .......... .......... .......... .......... 93%  209M 0s\n",
      " 34800K .......... .......... .......... .......... .......... 93%  194M 0s\n",
      " 34850K .......... .......... .......... .......... .......... 93%  189M 0s\n",
      " 34900K .......... .......... .......... .......... .......... 94%  199M 0s\n",
      " 34950K .......... .......... .......... .......... .......... 94%  216M 0s\n",
      " 35000K .......... .......... .......... .......... .......... 94%  175M 0s\n",
      " 35050K .......... .......... .......... .......... .......... 94%  173M 0s\n",
      " 35100K .......... .......... .......... .......... .......... 94%  206M 0s\n",
      " 35150K .......... .......... .......... .......... .......... 94%  195M 0s\n",
      " 35200K .......... .......... .......... .......... .......... 94%  130M 0s\n",
      " 35250K .......... .......... .......... .......... .......... 94%  126M 0s\n",
      " 35300K .......... .......... .......... .......... .......... 95%  234M 0s\n",
      " 35350K .......... .......... .......... .......... .......... 95%  185M 0s\n",
      " 35400K .......... .......... .......... .......... .......... 95%  192M 0s\n",
      " 35450K .......... .......... .......... .......... .......... 95%  108M 0s\n",
      " 35500K .......... .......... .......... .......... .......... 95%  187M 0s\n",
      " 35550K .......... .......... .......... .......... .......... 95%  210M 0s\n",
      " 35600K .......... .......... .......... .......... .......... 95%  202M 0s\n",
      " 35650K .......... .......... .......... .......... .......... 96%  194M 0s\n",
      " 35700K .......... .......... .......... .......... .......... 96%  193M 0s\n",
      " 35750K .......... .......... .......... .......... .......... 96%  243M 0s\n",
      " 35800K .......... .......... .......... .......... .......... 96%  303M 0s\n",
      " 35850K .......... .......... .......... .......... .......... 96%  177M 0s\n",
      " 35900K .......... .......... .......... .......... .......... 96%  213M 0s\n",
      " 35950K .......... .......... .......... .......... .......... 96%  199M 0s\n",
      " 36000K .......... .......... .......... .......... .......... 97%  216M 0s\n",
      " 36050K .......... .......... .......... .......... .......... 97%  190M 0s\n",
      " 36100K .......... .......... .......... .......... .......... 97%  205M 0s\n",
      " 36150K .......... .......... .......... .......... .......... 97%  193M 0s\n",
      " 36200K .......... .......... .......... .......... .......... 97%  211M 0s\n",
      " 36250K .......... .......... .......... .......... .......... 97%  153M 0s\n",
      " 36300K .......... .......... .......... .......... .......... 97%  253M 0s\n",
      " 36350K .......... .......... .......... .......... .......... 97%  190M 0s\n",
      " 36400K .......... .......... .......... .......... .......... 98%  219M 0s\n",
      " 36450K .......... .......... .......... .......... .......... 98%  183M 0s\n",
      " 36500K .......... .......... .......... .......... .......... 98%  212M 0s\n",
      " 36550K .......... .......... .......... .......... .......... 98%  198M 0s\n",
      " 36600K .......... .......... .......... .......... .......... 98%  209M 0s\n",
      " 36650K .......... .......... .......... .......... .......... 98%  172M 0s\n",
      " 36700K .......... .......... .......... .......... .......... 98%  218M 0s\n",
      " 36750K .......... .......... .......... .......... .......... 99%  194M 0s\n",
      " 36800K .......... .......... .......... .......... .......... 99%  219M 0s\n",
      " 36850K .......... .......... .......... .......... .......... 99%  181M 0s\n",
      " 36900K .......... .......... .......... .......... .......... 99%  213M 0s\n",
      " 36950K .......... .......... .......... .......... .......... 99%  175M 0s\n",
      " 37000K .......... .......... .......... .......... .......... 99%  210M 0s\n",
      " 37050K .......... .......... .......... .......... .......... 99%  168M 0s\n",
      " 37100K .......... .......... .......... .......... .......... 99%  213M 0s\n",
      " 37150K ........                                              100%  220M=0.7s\n",
      "\n",
      "2025-07-24 15:16:50 (51.7 MB/s) - ‘rvtests_linux64.tar.gz’ saved [38050737/38050737]\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "executable/\n",
      "executable/combineKinship\n",
      "executable/plink2vcf\n",
      "executable/vcf2plink\n",
      "executable/vcfAnnoSummaryLite\n",
      "executable/explainCSI2\n",
      "executable/vcf2kinship\n",
      "executable/vcfIndvSummary\n",
      "executable/createVCFIndex\n",
      "executable/extractVCFIndex\n",
      "executable/explainTabix\n",
      "executable/vcf2ld_neighbor\n",
      "executable/vcfConcordance\n",
      "executable/vcfExtractSite\n",
      "executable/vcfSummaryLite\n",
      "executable/bgenFileInfo\n",
      "executable/vcfPeek\n",
      "executable/vcfVariantSummaryLite\n",
      "executable/kinshipDecompose\n",
      "executable/rvtest\n",
      "executable/vcfPair\n",
      "executable/vcfSummary\n",
      "executable/explainCSI1\n",
      "executable/vcf2ld_window\n",
      "executable/queryVCFIndex\n",
      "executable/bgen2vcf\n",
      "executable/vcf2ld_gene\n",
      "executable/vcf2geno\n",
      "example/\n",
      "example/README.md\n",
      "example/example.vcf\n",
      "example/covar.missing\n",
      "example/pheno\n",
      "example/experimental/\n",
      "example/experimental/1k.fam\n",
      "example/experimental/1k.bed\n",
      "example/experimental/1k.bim\n",
      "example/experimental/1k.1m.bgen\n",
      "example/experimental/cmd.sh\n",
      "example/experimental/1k.1m.vcf.gz\n",
      "example/experimental/1k.ped\n",
      "example/experimental/1k.1m.samples\n",
      "example/example.vcf.gz.tbi\n",
      "example/example.vcf.gz\n",
      "example/cmd.sh\n",
      "example/setFile\n",
      "example/covar\n",
      "README.md\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Install RVTESTS: Option 1 (~15min)\n",
    "if test -e /home/jupyter/tools/rvtests; then\n",
    "\n",
    "echo \"rvtests is already installed\"\n",
    "else\n",
    "echo \"rvtests is not installed\"\n",
    "\n",
    "mkdir /home/jupyter/tools/rvtests\n",
    "cd /home/jupyter/tools/rvtests\n",
    "\n",
    "wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz \n",
    "\n",
    "tar -zxvf rvtests_linux64.tar.gz\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1e14051a-1aa1-43ad-adf0-41378025bda7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# give permission\n",
    "\n",
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b43e2d51-aebf-4a3a-a46d-a42d69c48440",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# making working directory\n",
    "#Loop over all the ancestries\n",
    "for ancestry in {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'} ;\n",
    "do\n",
    "\n",
    "#Make a folder for each ancestry\n",
    "mkdir ~/workspace/ws_files/TMEM175/TMEM175_\"$ancestry\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "acebbf7a-dbcd-4c34-961c-41fceb3a02e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# covariate file has been created in another notebook without GBA carriers"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1fe53cdb-9e99-4412-823a-98d8cd4bf43c",
   "metadata": {},
   "source": [
    "Annotation of the gene\n",
    "\n",
    "Extract the region using PLINK\n",
    "\n",
    "Extract TMEM175 gene\n",
    "\n",
    "TMEM175 coordinates: Chromosome 4:932387-4:958656(GRCh38/hg38)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ba02cf8d-89f9-45c0-b683-355a4661ad83",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:23:46 2025\n",
      "26046 MiB RAM detected, ~24020 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "7209 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:23:55 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:23:56 2025\n",
      "26046 MiB RAM detected, ~24029 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "3883 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:00 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:01 2025\n",
      "26046 MiB RAM detected, ~23952 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "2741 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:04 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:05 2025\n",
      "26046 MiB RAM detected, ~23980 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "1378 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed ...\n",
      "done.\n",
      "End time: Thu Jul 24 15:24:07 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:08 2025\n",
      "26046 MiB RAM detected, ~23984 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "2103 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:11 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:12 2025\n",
      "26046 MiB RAM detected, ~24009 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "3430 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:15 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:16 2025\n",
      "26046 MiB RAM detected, ~23977 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "645 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:17 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:18 2025\n",
      "26046 MiB RAM detected, ~23976 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "3374 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:22 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:23 2025\n",
      "26046 MiB RAM detected, ~23985 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "3520 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:26 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:27 2025\n",
      "26046 MiB RAM detected, ~23984 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "2501 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:30 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 882387\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "  --to-bp 1008656\n",
      "\n",
      "Start time: Thu Jul 24 15:24:31 2025\n",
      "26046 MiB RAM detected, ~23982 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "4429 variants remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed\n",
      "done.%\n",
      "End time: Thu Jul 24 15:24:36 2025\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release10_vwb \\\n",
    "    --chr 4 \\\n",
    "    --from-bp 882387 \\\n",
    "    --to-bp 1008656 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "253a43e7-aee3-4ab2-96b8-0d1e8ddc9c88",
   "metadata": {},
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}/'\n",
    "    \n",
    "    ! head -n 1 {WORK_DIR}/{ancestry}_TMEM175.fam > {WORK_DIR}/{ancestry}_s1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "755dff93-02e5-4873-b1ef-325987e23b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "! head /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_s1.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70ebf7dc-8383-40c2-8069-c44133d0e1ff",
   "metadata": {},
   "source": [
    "Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "987ad69b-4187-4fed-ab4b-6254ed56d0d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --keep {WORK_DIR}/{ancestry}_s1.txt \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175_v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "2096cfec-44dc-4979-805f-f76c50f98f34",
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    ! bgzip -f {WORK_DIR}/{ancestry}_TMEM175_v1.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/{ancestry}_TMEM175_v1.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7db34320-abf3-4260-a3da-ad1855f416ec",
   "metadata": {},
   "source": [
    "Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "584bcf2c-9134-4b80-874d-4e42edcb1161",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 7216 lines from VCF file\n",
      "NOTICE: A total of 7209 locus in VCF file passed QC threshold, representing 6826 SNPs (5193 transitions and 1633 transversions) and 383 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 6826 SNP genotypes (5193 transitions and 1633 transversions) and 383 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 7209 unique variants in 7209 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 7209 unique variants in 7209 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3890 lines from VCF file\n",
      "NOTICE: A total of 3883 locus in VCF file passed QC threshold, representing 3659 SNPs (2735 transitions and 924 transversions) and 224 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3659 SNP genotypes (2735 transitions and 924 transversions) and 224 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3883 unique variants in 3883 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3883 unique variants in 3883 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 2748 lines from VCF file\n",
      "NOTICE: A total of 2741 locus in VCF file passed QC threshold, representing 2588 SNPs (1971 transitions and 617 transversions) and 153 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2588 SNP genotypes (1971 transitions and 617 transversions) and 153 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2741 unique variants in 2741 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2741 unique variants in 2741 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 1385 lines from VCF file\n",
      "NOTICE: A total of 1378 locus in VCF file passed QC threshold, representing 1303 SNPs (972 transitions and 331 transversions) and 75 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1303 SNP genotypes (972 transitions and 331 transversions) and 75 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 1378 unique variants in 1378 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 1378 unique variants in 1378 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 2110 lines from VCF file\n",
      "NOTICE: A total of 2103 locus in VCF file passed QC threshold, representing 1987 SNPs (1521 transitions and 466 transversions) and 116 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 1987 SNP genotypes (1521 transitions and 466 transversions) and 116 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2103 unique variants in 2103 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2103 unique variants in 2103 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3437 lines from VCF file\n",
      "NOTICE: A total of 3430 locus in VCF file passed QC threshold, representing 3245 SNPs (2454 transitions and 791 transversions) and 185 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3245 SNP genotypes (2454 transitions and 791 transversions) and 185 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3430 unique variants in 3430 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3430 unique variants in 3430 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 652 lines from VCF file\n",
      "NOTICE: A total of 645 locus in VCF file passed QC threshold, representing 610 SNPs (456 transitions and 154 transversions) and 35 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 610 SNP genotypes (456 transitions and 154 transversions) and 35 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 645 unique variants in 645 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 645 unique variants in 645 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3381 lines from VCF file\n",
      "NOTICE: A total of 3374 locus in VCF file passed QC threshold, representing 3198 SNPs (2497 transitions and 701 transversions) and 176 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3198 SNP genotypes (2497 transitions and 701 transversions) and 176 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3374 unique variants in 3374 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3374 unique variants in 3374 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 3527 lines from VCF file\n",
      "NOTICE: A total of 3520 locus in VCF file passed QC threshold, representing 3327 SNPs (2494 transitions and 833 transversions) and 193 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 3327 SNP genotypes (2494 transitions and 833 transversions) and 193 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 3520 unique variants in 3520 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 3520 unique variants in 3520 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "DoneCE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 2508 lines from VCF file\n",
      "NOTICE: A total of 2501 locus in VCF file passed QC threshold, representing 2365 SNPs (1807 transitions and 558 transversions) and 136 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 2365 SNP genotypes (1807 transitions and 558 transversions) and 136 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 2501 unique variants in 2501 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 2501 unique variants in 2501 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n",
      "\n",
      "NOTICE: Running with system command <convert2annovar.pl  -includeinfo -allsample -withfreq -format vcf4 /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175_v1.vcf.gz > /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput>\n",
      "NOTICE: Finished reading 4436 lines from VCF file\n",
      "NOTICE: A total of 4429 locus in VCF file passed QC threshold, representing 4173 SNPs (3193 transitions and 980 transversions) and 256 indels/substitutions\n",
      "NOTICE: Finished writing allele frequencies based on 4173 SNP genotypes (3193 transitions and 980 transversions) and 256 indels/substitutions for 1 samples\n",
      "\n",
      "NOTICE: Running with system command </home/jupyter/tools/annovar/table_annovar.pl /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/ -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar -remove -protocol refGene,clinvar_20140902 -operation g,f --nopolish -nastring . -otherinfo>\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene -exonsort -nofirstcodondel /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: Output files are written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene.variant_function, /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/tools/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 4429 unique variants in 4429 input lines\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/tools/annovar/humandb/hg38_refGeneMrna.fa ... Done with 15 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20140902\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20140902 -buildver hg38 -outfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.avinput /home/jupyter/tools/annovar/humandb/>\n",
      "NOTICE: the --dbtype clinvar_20140902 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_clinvar_20140902_dropped, and output file with other variants is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_clinvar_20140902_filtered\n",
      "NOTICE: Processing next batch with 4429 unique variants in 4429 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 44781 and the number of bins to be scanned is 10\n",
      "NOTICE: Scanning filter database /home/jupyter/tools/annovar/humandb/hg38_clinvar_20140902.txt...Done\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt\n",
      "NOTICE: Reading from /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.txt\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: VCF output is written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.annovar.hg38_multianno.vcf\n",
      "\n",
      "gzip: stdout: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "## annotate using ANNOVAR\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    ! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_TMEM175_v1.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "    -out {WORK_DIR}/{ancestry}_TMEM175.annovar \\\n",
    "    -remove -protocol refGene,clinvar_20140902 \\\n",
    "    -operation g,f \\\n",
    "    --nopolish \\\n",
    "    -nastring . \\\n",
    "    -vcfinput"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "7fe092d9-0f00-4f21-8e66-6609254b2d9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>clinvar_20140902</th>\n",
       "      <th>Otherinfo1</th>\n",
       "      <th>Otherinfo2</th>\n",
       "      <th>Otherinfo3</th>\n",
       "      <th>Otherinfo4</th>\n",
       "      <th>Otherinfo5</th>\n",
       "      <th>Otherinfo6</th>\n",
       "      <th>Otherinfo7</th>\n",
       "      <th>Otherinfo8</th>\n",
       "      <th>Otherinfo9</th>\n",
       "      <th>Otherinfo10</th>\n",
       "      <th>Otherinfo11</th>\n",
       "      <th>Otherinfo12</th>\n",
       "      <th>Otherinfo13</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>882397</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882397</td>\n",
       "      <td>chr4:882397:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>882432</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882432</td>\n",
       "      <td>chr4:882432:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>882530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>chr4:882530:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>882566</td>\n",
       "      <td>882566</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882565</td>\n",
       "      <td>chr4:882565:AG:A</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>882611</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>GAK</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>chr4:882611:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2098</th>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>1008337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA,FGFRL1</td>\n",
       "      <td>dist=3773;dist=3289</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>chr4:1008337:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2099</th>\n",
       "      <td>4</td>\n",
       "      <td>1008371</td>\n",
       "      <td>1008371</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA,FGFRL1</td>\n",
       "      <td>dist=3807;dist=3255</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008370</td>\n",
       "      <td>chr4:1008370:TC:T</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2100</th>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA,FGFRL1</td>\n",
       "      <td>dist=3916;dist=3146</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>chr4:1008480:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2101</th>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>1008515</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA,FGFRL1</td>\n",
       "      <td>dist=3951;dist=3111</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008515</td>\n",
       "      <td>chr4:1008515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2102</th>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>1008538</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>IDUA,FGFRL1</td>\n",
       "      <td>dist=3974;dist=3088</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>4</td>\n",
       "      <td>1008538</td>\n",
       "      <td>chr4:1008538:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>PR</td>\n",
       "      <td>GT</td>\n",
       "      <td>0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2103 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr    Start      End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0       4   882397   882397   C   T     intronic          GAK   \n",
       "1       4   882432   882432   G   A     intronic          GAK   \n",
       "2       4   882530   882530   C   T     intronic          GAK   \n",
       "3       4   882566   882566   G   -     intronic          GAK   \n",
       "4       4   882611   882611   T   C     intronic          GAK   \n",
       "...   ...      ...      ...  ..  ..          ...          ...   \n",
       "2098    4  1008337  1008337   C   T   intergenic  IDUA,FGFRL1   \n",
       "2099    4  1008371  1008371   C   -   intergenic  IDUA,FGFRL1   \n",
       "2100    4  1008480  1008480   A   G   intergenic  IDUA,FGFRL1   \n",
       "2101    4  1008515  1008515   C   T   intergenic  IDUA,FGFRL1   \n",
       "2102    4  1008538  1008538   A   G   intergenic  IDUA,FGFRL1   \n",
       "\n",
       "       GeneDetail.refGene ExonicFunc.refGene AAChange.refGene  \\\n",
       "0                       .                  .                .   \n",
       "1                       .                  .                .   \n",
       "2                       .                  .                .   \n",
       "3                       .                  .                .   \n",
       "4                       .                  .                .   \n",
       "...                   ...                ...              ...   \n",
       "2098  dist=3773;dist=3289                  .                .   \n",
       "2099  dist=3807;dist=3255                  .                .   \n",
       "2100  dist=3916;dist=3146                  .                .   \n",
       "2101  dist=3951;dist=3111                  .                .   \n",
       "2102  dist=3974;dist=3088                  .                .   \n",
       "\n",
       "     clinvar_20140902 Otherinfo1 Otherinfo2 Otherinfo3  Otherinfo4  \\\n",
       "0                   .          0          .          .           4   \n",
       "1                   .          0          .          .           4   \n",
       "2                   .          0          .          .           4   \n",
       "3                   .          0          .          .           4   \n",
       "4                   .          0          .          .           4   \n",
       "...               ...        ...        ...        ...         ...   \n",
       "2098                .        0.5          .          .           4   \n",
       "2099                .          0          .          .           4   \n",
       "2100                .          0          .          .           4   \n",
       "2101                .          0          .          .           4   \n",
       "2102                .          0          .          .           4   \n",
       "\n",
       "      Otherinfo5         Otherinfo6 Otherinfo7 Otherinfo8 Otherinfo9  \\\n",
       "0         882397    chr4:882397:C:T          C          T          .   \n",
       "1         882432    chr4:882432:G:A          G          A          .   \n",
       "2         882530    chr4:882530:C:T          C          T          .   \n",
       "3         882565   chr4:882565:AG:A         AG          A          .   \n",
       "4         882611    chr4:882611:T:C          T          C          .   \n",
       "...          ...                ...        ...        ...        ...   \n",
       "2098     1008337   chr4:1008337:C:T          C          T          .   \n",
       "2099     1008370  chr4:1008370:TC:T         TC          T          .   \n",
       "2100     1008480   chr4:1008480:A:G          A          G          .   \n",
       "2101     1008515   chr4:1008515:C:T          C          T          .   \n",
       "2102     1008538   chr4:1008538:A:G          A          G          .   \n",
       "\n",
       "     Otherinfo10 Otherinfo11 Otherinfo12 Otherinfo13  \n",
       "0              .          PR          GT         0/0  \n",
       "1              .          PR          GT         0/0  \n",
       "2              .          PR          GT         0/0  \n",
       "3              .          PR          GT         0/0  \n",
       "4              .          PR          GT         0/0  \n",
       "...          ...         ...         ...         ...  \n",
       "2098           .          PR          GT         0/1  \n",
       "2099           .          PR          GT         0/0  \n",
       "2100           .          PR          GT         0/0  \n",
       "2101           .          PR          GT         0/0  \n",
       "2102           .          PR          GT         0/0  \n",
       "\n",
       "[2103 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Read in ANNOVAR multianno file\n",
    "gene = pd.read_csv(f'{WORK_DIR}/TMEM175_SAS/SAS_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "display(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "8bdada7c-4346-4b93-951e-916f4b46839d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Func.refGene\n",
       "intronic           1701\n",
       "exonic              187\n",
       "intergenic           94\n",
       "UTR3                 59\n",
       "UTR5                 25\n",
       "downstream           24\n",
       "upstream             12\n",
       "exonic;splicing       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[\"Func.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "022711c4-fcd1-445f-8727-01051518039f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ExonicFunc.refGene\n",
       "nonsynonymous SNV         104\n",
       "synonymous SNV             78\n",
       "stopgain                    2\n",
       "frameshift deletion         2\n",
       "nonframeshift deletion      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter exonic variants\n",
    "coding = gene[gene['Func.refGene'] == 'exonic']\n",
    "coding[\"ExonicFunc.refGene\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "ce0ac4de-c844-42f3-9927-77a2063677d0",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EUR\n",
      "{'EUR'}\n",
      "Total variants:  7209\n",
      "Intronic:  5651\n",
      "UTR3:  214\n",
      "UTR5:  82\n",
      "Total exonic:  731\n",
      "  Synonymous:  313\n",
      "  Nonsynonymous:  418\n",
      "nonsynonymous_lof:  437\n",
      "\n",
      "\n",
      "WORKING ON: AMR\n",
      "{'AMR'}\n",
      "Total variants:  3883\n",
      "Intronic:  3134\n",
      "UTR3:  102\n",
      "UTR5:  42\n",
      "Total exonic:  323\n",
      "  Synonymous:  150\n",
      "  Nonsynonymous:  173\n",
      "nonsynonymous_lof:  180\n",
      "\n",
      "\n",
      "WORKING ON: MDE\n",
      "{'MDE'}\n",
      "Total variants:  2741\n",
      "Intronic:  2182\n",
      "UTR3:  77\n",
      "UTR5:  34\n",
      "Total exonic:  257\n",
      "  Synonymous:  123\n",
      "  Nonsynonymous:  134\n",
      "nonsynonymous_lof:  136\n",
      "\n",
      "\n",
      "WORKING ON: AJ\n",
      "{'AJ'}\n",
      "Total variants:  1378\n",
      "Intronic:  1086\n",
      "UTR3:  46\n",
      "UTR5:  7\n",
      "Total exonic:  141\n",
      "  Synonymous:  66\n",
      "  Nonsynonymous:  75\n",
      "nonsynonymous_lof:  79\n",
      "\n",
      "\n",
      "WORKING ON: SAS\n",
      "{'SAS'}\n",
      "Total variants:  2103\n",
      "Intronic:  1701\n",
      "UTR3:  59\n",
      "UTR5:  25\n",
      "Total exonic:  187\n",
      "  Synonymous:  83\n",
      "  Nonsynonymous:  104\n",
      "nonsynonymous_lof:  108\n",
      "\n",
      "\n",
      "WORKING ON: CAH\n",
      "{'CAH'}\n",
      "Total variants:  3430\n",
      "Intronic:  2746\n",
      "UTR3:  95\n",
      "UTR5:  32\n",
      "Total exonic:  282\n",
      "  Synonymous:  128\n",
      "  Nonsynonymous:  154\n",
      "nonsynonymous_lof:  162\n",
      "\n",
      "\n",
      "WORKING ON: FIN\n",
      "{'FIN'}\n",
      "Total variants:  645\n",
      "Intronic:  526\n",
      "UTR3:  22\n",
      "UTR5:  3\n",
      "Total exonic:  54\n",
      "  Synonymous:  30\n",
      "  Nonsynonymous:  24\n",
      "nonsynonymous_lof:  25\n",
      "\n",
      "\n",
      "WORKING ON: EAS\n",
      "{'EAS'}\n",
      "Total variants:  3374\n",
      "Intronic:  2659\n",
      "UTR3:  109\n",
      "UTR5:  51\n",
      "Total exonic:  312\n",
      "  Synonymous:  131\n",
      "  Nonsynonymous:  181\n",
      "nonsynonymous_lof:  190\n",
      "\n",
      "\n",
      "WORKING ON: AAC\n",
      "{'AAC'}\n",
      "Total variants:  3520\n",
      "Intronic:  2811\n",
      "UTR3:  98\n",
      "UTR5:  41\n",
      "Total exonic:  293\n",
      "  Synonymous:  138\n",
      "  Nonsynonymous:  155\n",
      "nonsynonymous_lof:  160\n",
      "\n",
      "\n",
      "WORKING ON: CAS\n",
      "{'CAS'}\n",
      "Total variants:  2501\n",
      "Intronic:  2010\n",
      "UTR3:  70\n",
      "UTR5:  27\n",
      "Total exonic:  231\n",
      "  Synonymous:  101\n",
      "  Nonsynonymous:  130\n",
      "nonsynonymous_lof:  135\n",
      "\n",
      "\n",
      "WORKING ON: AFR\n",
      "{'AFR'}\n",
      "Total variants:  4429\n",
      "Intronic:  3516\n",
      "UTR3:  135\n",
      "UTR5:  50\n",
      "Total exonic:  374\n",
      "  Synonymous:  180\n",
      "  Nonsynonymous:  194\n",
      "nonsynonymous_lof:  200\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Make lists of variants to keep - all coding, coding nonsynonymous (missense - as they are coded in ANNOVAR), deleterious (CADD_phred > 20)\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "\n",
    "    # Read in ANNOVAR multianno file\n",
    "    gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_TMEM175.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "    \n",
    "    #Print number of variants in the different categories\n",
    "    results = [] \n",
    "\n",
    "    utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "    intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "    exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "    utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "    coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "    coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'nonsynonymous SNV')]\n",
    "    lof = exonic[(exonic['ExonicFunc.refGene'] == 'stopgain') | (exonic['ExonicFunc.refGene'] == 'stoploss') | (exonic['ExonicFunc.refGene'] == 'frameshift deletion') | (exonic['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "    nonsynonymous_lof = pd.concat([coding_nonsynonymous, lof])\n",
    "\n",
    "    print({ancestry})\n",
    "    print('Total variants: ', len(gene))\n",
    "    print(\"Intronic: \", len(intronic))\n",
    "    print('UTR3: ', len(utr3))\n",
    "    print('UTR5: ', len(utr5))\n",
    "    print(\"Total exonic: \", len(exonic))\n",
    "    print('  Synonymous: ', len(coding_synonymous))\n",
    "    print(\"  Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "    print(\"nonsynonymous_lof: \", len(nonsynonymous_lof))\n",
    "    results.append((gene, intronic, utr3, utr5, exonic, coding_synonymous, coding_nonsynonymous, nonsynonymous_lof))\n",
    "    print('\\n')\n",
    "\n",
    "    # Save in PLINK format - coding nonsynonymous \n",
    "    # These are missense variants - other types of nonsynonymous variants (e.g stopgain/loss, or frameshift variants are coded differently in the ExonicFunc.refGene \n",
    "    variants_toKeep = nonsynonymous_lof[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_TMEM175.nonsynonymous_lof.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "\n",
    "\n",
    "    # Save in PLINK format - all coding variants\n",
    "    variants_toKeep2 = exonic[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry}_TMEM175.exonic.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d1b6f7b-0d48-4613-bdd1-0ceaba87e4e6",
   "metadata": {},
   "source": [
    "ALL variants\n",
    "\n",
    "assoc\n",
    "\n",
    "glm\n",
    "\n",
    "ASSOC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "0dc007f6-2041-4124-9f14-43e8ae71021d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/jupyter/workspace/Sample_Notebooks\n"
     ]
    }
   ],
   "source": [
    "! pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "fb2e62f8-9333-46c8-a42e-0398c92c8b80",
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','FIN','MDE','SAS','CAH', 'EUR'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    # read cov file\n",
    "    cov = pd.read_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt', sep = '\\t')\n",
    "\n",
    "    ## Make file of sample IDs to keep \n",
    "    samples_toKeep = cov[['FID', 'IID']].copy()\n",
    "    samples_toKeep.to_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep', sep = '\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "467913fa-1c8a-4799-adb2-c4e2cbab39d2",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3520 variants loaded from .bim file.\n",
      "1215 people (532 males, 683 females) loaded from .fam.\n",
      "1133 phenotype values loaded from .fam.\n",
      "--keep: 1207 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1207 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.998418.\n",
      "2916 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "604 variants and 1207 people pass filters and QC.\n",
      "Among remaining phenotypes, 369 are cases and 756 are controls.  (82 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "4429 variants loaded from .bim file.\n",
      "3780 people (2087 males, 1693 females) loaded from .fam.\n",
      "3498 phenotype values loaded from .fam.\n",
      "--keep: 3581 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.997315.\n",
      "3833 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "596 variants and 3581 people pass filters and QC.\n",
      "Among remaining phenotypes, 1147 are cases and 2169 are controls.  (265\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "1378 variants loaded from .bim file.\n",
      "3472 people (2146 males, 1326 females) loaded from .fam.\n",
      "1966 phenotype values loaded from .fam.\n",
      "--keep: 2716 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2716 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.998983.\n",
      "964 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "414 variants and 2716 people pass filters and QC.\n",
      "Among remaining phenotypes, 1285 are cases and 408 are controls.  (1023\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3883 variants loaded from .bim file.\n",
      "3608 people (1735 males, 1873 females) loaded from .fam.\n",
      "3434 phenotype values loaded from .fam.\n",
      "--keep: 3471 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3471 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.997517.\n",
      "3509 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "374 variants and 3471 people pass filters and QC.\n",
      "Among remaining phenotypes, 1917 are cases and 1402 are controls.  (152\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "2501 variants loaded from .bim file.\n",
      "1691 people (737 males, 954 females) loaded from .fam.\n",
      "1408 phenotype values loaded from .fam.\n",
      "--keep: 1518 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1518 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.996579.\n",
      "2109 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "392 variants and 1518 people pass filters and QC.\n",
      "Among remaining phenotypes, 734 are cases and 578 are controls.  (206\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3374 variants loaded from .bim file.\n",
      "6662 people (4001 males, 2661 females) loaded from .fam.\n",
      "5116 phenotype values loaded from .fam.\n",
      "--keep: 6414 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 6414 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.994617.\n",
      "3006 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "368 variants and 6414 people pass filters and QC.\n",
      "Among remaining phenotypes, 2377 are cases and 2598 are controls.  (1439\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "7209 variants loaded from .bim file.\n",
      "58823 people (33074 males, 25749 females) loaded from .fam.\n",
      "37150 phenotype values loaded from .fam.\n",
      "--keep: 53641 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 53641 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies done.\n",
      "Total genotyping rate in remaining samples is 0.75561.\n",
      "6812 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "397 variants and 53641 people pass filters and QC.\n",
      "Among remaining phenotypes, 24208 are cases and 9662 are controls.  (19771\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "645 variants loaded from .bim file.\n",
      "144 people (62 males, 82 females) loaded from .fam.\n",
      "118 phenotype values loaded from .fam.\n",
      "--keep: 111 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.994022.\n",
      "240 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "405 variants and 111 people pass filters and QC.\n",
      "Among remaining phenotypes, 80 are cases and 11 are controls.  (20 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "2741 variants loaded from .bim file.\n",
      "1493 people (815 males, 678 females) loaded from .fam.\n",
      "1311 phenotype values loaded from .fam.\n",
      "--keep: 1448 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1448 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.99619.\n",
      "2344 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "397 variants and 1448 people pass filters and QC.\n",
      "Among remaining phenotypes, 724 are cases and 551 are controls.  (173\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "2103 variants loaded from .bim file.\n",
      "945 people (606 males, 339 females) loaded from .fam.\n",
      "586 phenotype values loaded from .fam.\n",
      "--keep: 870 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 870 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.995001.\n",
      "1630 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "473 variants and 870 people pass filters and QC.\n",
      "Among remaining phenotypes, 309 are cases and 261 are controls.  (300\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all.assoc ...\n",
      "done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "  --ci 0.95\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all\n",
      "\n",
      "26046 MB RAM detected; reserving 13023 MB for main workspace.\n",
      "3430 variants loaded from .bim file.\n",
      "1111 people (589 males, 522 females) loaded from .fam.\n",
      "1040 phenotype values loaded from .fam.\n",
      "--keep: 1039 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1039 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate in remaining samples is 0.997672.\n",
      "2897 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "533 variants and 1039 people pass filters and QC.\n",
      "Among remaining phenotypes, 663 are cases and 318 are controls.  (58 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all.assoc ...\n",
      "done.\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis using plink assoc for all variants, not adjusting for any covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    ! /home/jupyter/tools/plink \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "    --assoc \\\n",
    "    --allow-no-sex \\\n",
    "    --ci 0.95 \\\n",
    "    --maf 0.01 \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175.all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "d77d5d77-8a3a-4aa4-8dc8-22f29724e6be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "Variants with p-value < 0.05: (23, 13)\n",
      "WORKING ON: AFR\n",
      "Variants with p-value < 0.05: (76, 13)\n",
      "WORKING ON: AJ\n",
      "Variants with p-value < 0.05: (150, 13)\n",
      "WORKING ON: AMR\n",
      "Variants with p-value < 0.05: (297, 13)\n",
      "WORKING ON: CAS\n",
      "Variants with p-value < 0.05: (8, 13)\n",
      "WORKING ON: EAS\n",
      "Variants with p-value < 0.05: (203, 13)\n",
      "WORKING ON: EUR\n",
      "Variants with p-value < 0.05: (218, 13)\n",
      "WORKING ON: FIN\n",
      "Variants with p-value < 0.05: (14, 13)\n",
      "WORKING ON: MDE\n",
      "Variants with p-value < 0.05: (107, 13)\n",
      "WORKING ON: SAS\n",
      "Variants with p-value < 0.05: (10, 13)\n",
      "WORKING ON: CAH\n",
      "Variants with p-value < 0.05: (88, 13)\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Look at assoc results, only variants with nominal p-value < 0.05\n",
    "    freq = pd.read_csv(f'{WORK_DIR}/{ancestry}_TMEM175.all.assoc', sep='\\s+')\n",
    "    sig_all_nonadj = freq[freq['P']<0.05]\n",
    "    \n",
    "    print(f'Variants with p-value < 0.05: {sig_all_nonadj.shape}')\n",
    "    \n",
    "    #Save FREQ to csv\n",
    "    freq.to_csv(f'{WORK_DIR}/{ancestry}.all_nonadj.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "50d1f614-100d-436f-aecb-30173af9d96e",
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','FIN','MDE','SAS','CAH', 'EUR'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    # read cov file\n",
    "    cov = pd.read_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt', sep = '\\t')\n",
    "    cov = cov.dropna()\n",
    "    cov.columns = ['FID', 'IID', 'FATID', 'MATID', 'SEX', 'AGE', 'PHENO', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "\n",
    "    ## save files \n",
    "    cov.to_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}.noGBA.noNA.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "786105ac-6dbb-4a85-bccb-d51ea0841785",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA\n",
    "cov = pd.read_csv(f'~/workspace/ws_files/AMR/AMR_covariate_file_noGBA.txt', sep = '\\t')\n",
    "pcs = cov[['PC1', 'PC2', 'PC3', 'PC4', 'PC5']]\n",
    "pca = PCA(n_components=5)\n",
    "pcs_ortho = pca.fit_transform(pcs)\n",
    "for i in range(5):\n",
    "    cov[f'PC{i+1}_ortho'] = pcs_ortho[:, i]\n",
    "\n",
    "# (Optional) Drop old PCs\n",
    "cov = cov.drop(columns=['PC1', 'PC2', 'PC3', 'PC4', 'PC5'])\n",
    "cov = cov.rename(columns={\n",
    "    'PC1_ortho': 'PC1',\n",
    "    'PC2_ortho': 'PC2',\n",
    "    'PC3_ortho': 'PC3',\n",
    "    'PC4_ortho': 'PC4',\n",
    "    'PC5_ortho': 'PC5'\n",
    "})\n",
    "\n",
    "cov.to_csv('~/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "5b818fc8-0020-4ff4-9ae0-7ffeece567f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "cov = pd.read_csv(f'~/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt', sep = '\\t')\n",
    "cov = cov.dropna()\n",
    "cov.to_csv('~/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f2907757-8f1e-4f57-8bc7-2e221216a5d1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:36 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24894 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam.\n",
      "3520 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--keep: 1207 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.noNA.txt.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2916 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "604 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:37 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:38 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24872 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam.\n",
      "4429 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--keep: 3581 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.noNA.txt.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3833 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "596 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:39 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:40 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24857 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam.\n",
      "1378 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--keep: 2716 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.noNA.txt.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "964 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "414 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:41 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:42 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24839 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam.\n",
      "3883 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.noNA.txt.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3509 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "374 variants remaining after main filters.\n",
      "Error: Cannot proceed with --glm regression on phenotype 'PHENO1', since\n",
      "variance inflation factor for covariate 'PC2' is too high (VIF_TOO_HIGH). You\n",
      "may want to remove redundant covariates and try again.\n",
      "End time: Fri Jul 25 08:23:42 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:43 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24850 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam.\n",
      "2501 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--keep: 1518 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.noNA.txt.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2109 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "392 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:44 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:45 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24855 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam.\n",
      "3374 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--keep: 6414 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.noNA.txt.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3006 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "368 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:46 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:47 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24848 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam.\n",
      "7209 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--keep: 53641 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.noNA.txt.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "done.lating allele frequencies... 0%\n",
      "6812 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "397 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:49 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:50 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24842 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam.\n",
      "645 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--keep: 111 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.noNA.txt.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "240 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "405 variants remaining after main filters.\n",
      "Warning: --glm remaining control count is less than 10x predictor count for\n",
      "phenotype 'PHENO1'.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:50 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:52 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24853 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam.\n",
      "2741 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--keep: 1448 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.noNA.txt.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2344 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "397 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:52 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:54 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24855 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam.\n",
      "2103 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--keep: 870 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.noNA.txt.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "1630 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "473 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:54 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.noNA.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:23:55 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24857 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam.\n",
      "3430 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--keep: 1039 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.noNA.txt.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "2897 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "533 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:23:56 2025\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis with covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "    --allow-no-sex \\\n",
    "    --maf 0.01 \\\n",
    "    --ci 0.95 \\\n",
    "    --glm \\\n",
    "    --covar /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.noNA.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --covar-variance-standardize \\\n",
    "    --neg9-pheno-really-missing \\\n",
    "    --out {WORK_DIR}/{ancestry}_TMEM175.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "eba6dcd3-8ef8-4184-a31c-c10bd601f0d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --neg9-pheno-really-missing\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj\n",
      "\n",
      "Start time: Fri Jul 25 08:37:12 2025\n",
      "Note: --allow-no-sex no longer has any effect.  (Missing-sex samples are\n",
      "automatically excluded from association analysis when sex is a covariate, and\n",
      "treated normally otherwise.)\n",
      "26046 MiB RAM detected, ~24828 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam.\n",
      "3883 variants loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "3509 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "374 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.all_adj.PHENO1.glm.logistic.hybrid .\n",
      "End time: Fri Jul 25 08:37:12 2025\n"
     ]
    }
   ],
   "source": [
    "WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_AMR'\n",
    "\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/AMR_TMEM175 \\\n",
    "--keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep \\\n",
    "--allow-no-sex \\\n",
    "--maf 0.01 \\\n",
    "--ci 0.95 \\\n",
    "--glm \\\n",
    "--covar /home/jupyter/workspace/ws_files/AMR/AMR_covariate_file_noGBA.orthoPCs.txt \\\n",
    "--covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--neg9-pheno-really-missing \\\n",
    "--out {WORK_DIR}/AMR_TMEM175.all_adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a4366c98-3678-41a0-9841-0f28b371cc34",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "There are 33 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AFR\n",
      "There are 55 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AJ\n",
      "There are 152 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AMR\n",
      "There are 24 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: CAS\n",
      "There are 3 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EAS\n",
      "There are 55 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: EUR\n",
      "There are 171 variants with p-value < 0.05 in glm\n",
      "There are 33 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: FIN\n",
      "There are 10 variants with p-value < 0.05 in glm\n",
      "There are 2 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: MDE\n",
      "There are 94 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: SAS\n",
      "There are 66 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: CAH\n",
      "There are 67 variants with p-value < 0.05 in glm\n",
      "There are 0 variants with p-value < 5e-8 in glm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_440/3155684219.py:10: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\\s+'`` instead\n"
     ]
    }
   ],
   "source": [
    "#Process results from plink glm analysis including covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Read in plink glm results\n",
    "    assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_TMEM175.all_adj.PHENO1.glm.logistic.hybrid', delim_whitespace=True)\n",
    "\n",
    "    #Filter for additive test only - this is the variant results\n",
    "    assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    significant = assoc_add[assoc_add['P']<0.05]\n",
    "\n",
    "    print(f'There are {len(significant)} variants with p-value < 0.05 in glm')\n",
    "    \n",
    "    #Check if there are any significant (p < 0.05) variants\n",
    "    GWsignificant = assoc_add[assoc_add['P']<5e-8]\n",
    "\n",
    "    print(f'There are {len(GWsignificant)} variants with p-value < 5e-8 in glm')\n",
    "    \n",
    "    #Save assoc_add to csv\n",
    "    assoc_add.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ca66428-efbd-4677-b446-de39ae50db15",
   "metadata": {},
   "source": [
    "format assoc file to do meta analysis(AFR/AJ/AMR/EAS/EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0c96f61d-60b6-4b1d-8c7f-4f7b3c860f43",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define ancestry groups and their sample sizes\n",
    "ancestry_sample_sizes = {\n",
    "    'AFR': 3316,\n",
    "    'AJ': 1693,\n",
    "    'AMR': 3319,\n",
    "    'EAS': 4975,\n",
    "    'EUR': 33870\n",
    "}\n",
    "\n",
    "# Output container (optional)\n",
    "cleaned_results = {}\n",
    "\n",
    "# Loop over each ancestry\n",
    "for ancestry, sample_size in ancestry_sample_sizes.items():\n",
    "    # Load PLINK association result file\n",
    "    file_path = f'/home/jupyter/workspace/ws_files/TMEM175/TMEM175_{ancestry}/{ancestry}.all_adj.csv'\n",
    "    df = pd.read_csv(file_path)\n",
    "\n",
    "    # Build MARKERNAME\n",
    "    df['MARKERNAME'] = 'chr' + df['#CHROM'].astype(str) + ':' + df['POS'].astype(str)\n",
    "\n",
    "    # Rename columns\n",
    "    df.rename(columns={\n",
    "        '#CHROM': 'CHROMOSOME',\n",
    "        'POS': 'POSITION',\n",
    "        'REF': 'NEA',\n",
    "        'ALT': 'EA',\n",
    "        'A1_FREQ': 'EAF',\n",
    "        'LOG(OR)_SE': 'SE',\n",
    "        'L95': 'OR_95L',\n",
    "        'U95': 'OR_95U'\n",
    "    }, inplace=True)\n",
    "\n",
    "    # Add BETA = log(OR)\n",
    "    df['BETA'] = np.log(df['OR'])\n",
    "\n",
    "    # Add sample size\n",
    "    df['N'] = sample_size\n",
    "\n",
    "    # Keep desired columns\n",
    "    final_cols = [\n",
    "        'MARKERNAME', 'CHROMOSOME', 'POSITION', 'NEA', 'EA', 'EAF',\n",
    "        'BETA', 'OR', 'SE', 'OR_95L', 'OR_95U', 'P', 'N'\n",
    "    ]\n",
    "    df_cleaned = df[final_cols]\n",
    "\n",
    "    # Store or save\n",
    "    cleaned_results[ancestry] = df_cleaned\n",
    "\n",
    "    # Optional: save to file\n",
    "    output_path = f'/home/jupyter/workspace/ws_files/TMEM175/TMEM175_{ancestry}/{ancestry}.cleaned_assoc.txt.gz'\n",
    "    df_cleaned.to_csv(output_path, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "07231063-e3a4-4969-8a5f-19e1436e0bdf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MARKERNAME</th>\n",
       "      <th>CHROMOSOME</th>\n",
       "      <th>POSITION</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>BETA</th>\n",
       "      <th>OR</th>\n",
       "      <th>SE</th>\n",
       "      <th>OR_95L</th>\n",
       "      <th>OR_95U</th>\n",
       "      <th>P</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4:882530</td>\n",
       "      <td>4</td>\n",
       "      <td>882530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.249648</td>\n",
       "      <td>0.278139</td>\n",
       "      <td>1.320670</td>\n",
       "      <td>0.130923</td>\n",
       "      <td>1.021770</td>\n",
       "      <td>1.70702</td>\n",
       "      <td>0.033631</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4:882611</td>\n",
       "      <td>4</td>\n",
       "      <td>882611</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.039216</td>\n",
       "      <td>0.259213</td>\n",
       "      <td>1.295910</td>\n",
       "      <td>0.292887</td>\n",
       "      <td>0.729911</td>\n",
       "      <td>2.30081</td>\n",
       "      <td>0.376139</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr4:883018</td>\n",
       "      <td>4</td>\n",
       "      <td>883018</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.036765</td>\n",
       "      <td>0.207095</td>\n",
       "      <td>1.230100</td>\n",
       "      <td>0.293740</td>\n",
       "      <td>0.691686</td>\n",
       "      <td>2.18762</td>\n",
       "      <td>0.480791</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4:883133</td>\n",
       "      <td>4</td>\n",
       "      <td>883133</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.015406</td>\n",
       "      <td>-0.502955</td>\n",
       "      <td>0.604741</td>\n",
       "      <td>0.366620</td>\n",
       "      <td>0.294783</td>\n",
       "      <td>1.24062</td>\n",
       "      <td>0.170104</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr4:883483</td>\n",
       "      <td>4</td>\n",
       "      <td>883483</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.025210</td>\n",
       "      <td>-0.468553</td>\n",
       "      <td>0.625907</td>\n",
       "      <td>0.295532</td>\n",
       "      <td>0.350714</td>\n",
       "      <td>1.11703</td>\n",
       "      <td>0.112862</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>chr4:1007954</td>\n",
       "      <td>4</td>\n",
       "      <td>1007954</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.267157</td>\n",
       "      <td>0.015736</td>\n",
       "      <td>1.015860</td>\n",
       "      <td>0.121593</td>\n",
       "      <td>0.800447</td>\n",
       "      <td>1.28924</td>\n",
       "      <td>0.897053</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>chr4:1008009</td>\n",
       "      <td>4</td>\n",
       "      <td>1008009</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.432489</td>\n",
       "      <td>0.060521</td>\n",
       "      <td>1.062390</td>\n",
       "      <td>0.109869</td>\n",
       "      <td>0.856571</td>\n",
       "      <td>1.31766</td>\n",
       "      <td>0.581740</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>chr4:1008209</td>\n",
       "      <td>4</td>\n",
       "      <td>1008209</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.044210</td>\n",
       "      <td>-0.110261</td>\n",
       "      <td>0.895600</td>\n",
       "      <td>0.245206</td>\n",
       "      <td>0.553853</td>\n",
       "      <td>1.44822</td>\n",
       "      <td>0.652950</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>chr4:1008337</td>\n",
       "      <td>4</td>\n",
       "      <td>1008337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.267157</td>\n",
       "      <td>0.015736</td>\n",
       "      <td>1.015860</td>\n",
       "      <td>0.121593</td>\n",
       "      <td>0.800447</td>\n",
       "      <td>1.28924</td>\n",
       "      <td>0.897053</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>chr4:1008480</td>\n",
       "      <td>4</td>\n",
       "      <td>1008480</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.058246</td>\n",
       "      <td>-0.141322</td>\n",
       "      <td>0.868210</td>\n",
       "      <td>0.221412</td>\n",
       "      <td>0.562547</td>\n",
       "      <td>1.33996</td>\n",
       "      <td>0.523296</td>\n",
       "      <td>1693</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>414 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       MARKERNAME  CHROMOSOME  POSITION NEA EA       EAF      BETA        OR  \\\n",
       "0     chr4:882530           4    882530   C  T  0.249648  0.278139  1.320670   \n",
       "1     chr4:882611           4    882611   T  C  0.039216  0.259213  1.295910   \n",
       "2     chr4:883018           4    883018   G  A  0.036765  0.207095  1.230100   \n",
       "3     chr4:883133           4    883133   G  A  0.015406 -0.502955  0.604741   \n",
       "4     chr4:883483           4    883483   G  A  0.025210 -0.468553  0.625907   \n",
       "..            ...         ...       ...  .. ..       ...       ...       ...   \n",
       "409  chr4:1007954           4   1007954   T  C  0.267157  0.015736  1.015860   \n",
       "410  chr4:1008009           4   1008009   C  T  0.432489  0.060521  1.062390   \n",
       "411  chr4:1008209           4   1008209   C  T  0.044210 -0.110261  0.895600   \n",
       "412  chr4:1008337           4   1008337   C  T  0.267157  0.015736  1.015860   \n",
       "413  chr4:1008480           4   1008480   A  G  0.058246 -0.141322  0.868210   \n",
       "\n",
       "           SE    OR_95L   OR_95U         P     N  \n",
       "0    0.130923  1.021770  1.70702  0.033631  1693  \n",
       "1    0.292887  0.729911  2.30081  0.376139  1693  \n",
       "2    0.293740  0.691686  2.18762  0.480791  1693  \n",
       "3    0.366620  0.294783  1.24062  0.170104  1693  \n",
       "4    0.295532  0.350714  1.11703  0.112862  1693  \n",
       "..        ...       ...      ...       ...   ...  \n",
       "409  0.121593  0.800447  1.28924  0.897053  1693  \n",
       "410  0.109869  0.856571  1.31766  0.581740  1693  \n",
       "411  0.245206  0.553853  1.44822  0.652950  1693  \n",
       "412  0.121593  0.800447  1.28924  0.897053  1693  \n",
       "413  0.221412  0.562547  1.33996  0.523296  1693  \n",
       "\n",
       "[414 rows x 13 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.cleaned_assoc.txt.gz', sep='\\t')\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25733324-6f6f-49d6-87d1-1cbbcaa87a70",
   "metadata": {},
   "source": [
    "create PCs file for meta analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c9481a69-1565-4654-941e-b2cdc81c94d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define ancestries and base path\n",
    "ancestries = ['AFR', 'AJ', 'AMR', 'EAS', 'EUR']\n",
    "base_path = '~/workspace/ws_files/'\n",
    "\n",
    "# List to store each ancestry's PC dataframe\n",
    "dfs = []\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    file_path = f'{base_path}{ancestry}/{ancestry}.noGBA.noNA.txt'\n",
    "    \n",
    "    # Read the file\n",
    "    df = pd.read_csv(file_path, sep='\\t')\n",
    "    \n",
    "    # Keep only PC1–PC5\n",
    "    df_pcs = df[['IID','PC1', 'PC2', 'PC3', 'PC4', 'PC5']].copy()\n",
    "    \n",
    "    # Add group label\n",
    "    df_pcs['GROUP'] = ancestry\n",
    "    \n",
    "    # Store\n",
    "    dfs.append(df_pcs)\n",
    "\n",
    "# Combine all ancestries\n",
    "merged_df = pd.concat(dfs, ignore_index=True)\n",
    "\n",
    "# View merged structure\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79d27448-148c-456b-86f9-64ddf68b4004",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df = merged_df.drop(columns=['PC1', 'PC2', 'PC3', 'PC4', 'PC5'])\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f0a0816-657c-42ca-905e-dcc0e6c3e15c",
   "metadata": {},
   "outputs": [],
   "source": [
    "pcs = pd.read_csv(f'{REL10_PATH}/meta_data/qc_metrics/projected_pcs_vwb.csv')\n",
    "pcs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c534220b-af18-4de6-8928-fe4246a7f664",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge on IID, keeping all rows from merged_df\n",
    "final_df = merged_df.merge(pcs.drop(columns=['FID']), on='IID', how='left')\n",
    "\n",
    "# View result\n",
    "final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2b2196d4-102e-4ec9-95e8-0a53b631f3cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv('~/workspace/ws_files/PCs_meta.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24a699fc-64c6-4aed-a180-71f833c1dc94",
   "metadata": {},
   "source": [
    "Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bc52219f-2f30-45f6-ac4b-696e5f25c79c",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running plink to extract exonic variants for ancestry: AAC\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:38:41 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24836 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--extract bed1: 4117453 variants excluded.\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "290 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:38:44 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AAC\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AAC\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:38:48 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24791 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--extract bed1: 4117584 variants excluded.\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "159 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Fri Jul 25 13:38:49 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AAC\n",
      "Running plink to extract exonic variants for ancestry: AFR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:38:53 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24778 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--extract bed1: 5094943 variants excluded.\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "367 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:38:58 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AFR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AFR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:02 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24819 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--extract bed1: 5095114 variants excluded.\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "196 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Fri Jul 25 13:39:03 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AFR\n",
      "Running plink to extract exonic variants for ancestry: AJ\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:07 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24825 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--extract bed1: 1752544 variants excluded.\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "137 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:39:09 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AJ\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AJ\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:13 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24805 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--extract bed1: 1752605 variants excluded.\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "76 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Fri Jul 25 13:39:14 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AJ\n",
      "Running plink to extract exonic variants for ancestry: AMR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:18 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24827 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--extract bed1: 4434343 variants excluded.\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "321 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:39:21 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: AMR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: AMR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:25 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24800 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--extract bed1: 4434486 variants excluded.\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "178 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Fri Jul 25 13:39:26 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: AMR\n",
      "Running plink to extract exonic variants for ancestry: CAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:30 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24791 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--extract bed1: 2741300 variants excluded.\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "228 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:39:33 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:37 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24822 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--extract bed1: 2741393 variants excluded.\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "135 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Fri Jul 25 13:39:37 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAS\n",
      "Running plink to extract exonic variants for ancestry: EAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:41 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24826 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--extract bed1: 3379462 variants excluded.\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "307 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:39:46 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: EAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:49 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24811 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--extract bed1: 3379582 variants excluded.\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "187 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Fri Jul 25 13:39:50 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EAS\n",
      "Running plink to extract exonic variants for ancestry: EUR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:39:54 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24805 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--extract bed1: 7744368 variants excluded.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "721 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:40:03 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: EUR\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: EUR\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:08 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24829 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--extract bed1: 7744655 variants excluded.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "434 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Fri Jul 25 13:40:11 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: EUR\n",
      "Running plink to extract exonic variants for ancestry: FIN\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:15 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24836 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--extract bed1: 938995 variants excluded.\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "54 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:40:17 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: FIN\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: FIN\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:21 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24800 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--extract bed1: 939024 variants excluded.\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "25 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Fri Jul 25 13:40:21 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: FIN\n",
      "Running plink to extract exonic variants for ancestry: MDE\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:25 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24810 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--extract bed1: 3022809 variants excluded.\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "255 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:40:28 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: MDE\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: MDE\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:32 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24817 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--extract bed1: 3022929 variants excluded.\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "135 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Fri Jul 25 13:40:32 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: MDE\n",
      "Running plink to extract exonic variants for ancestry: SAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:36 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24820 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--extract bed1: 2227403 variants excluded.\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "184 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:40:39 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: SAS\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: SAS\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:42 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24801 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--extract bed1: 2227480 variants excluded.\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "107 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.vcf\n",
      "done.\n",
      "End time: Fri Jul 25 13:40:43 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: SAS\n",
      "Running plink to extract exonic variants for ancestry: CAH\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:47 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24807 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--extract bed1: 4062881 variants excluded.\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "279 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.vcf ...\n",
      "done.\n",
      "End time: Fri Jul 25 13:40:51 2025\n",
      "Running bgzip and tabix for exonic variants for ancestry: CAH\n",
      "Running plink to extract nonsynonymous_lof variants for ancestry: CAH\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.log.\n",
      "Options in effect:\n",
      "  --export vcf-iid\n",
      "  --extract range /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.variantstoKeep.txt\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "\n",
      "Start time: Fri Jul 25 13:40:54 2025\n",
      "Note: --export 'vcf-iid' modifier is deprecated.  Use 'vcf' + 'id-paste=iid'.\n",
      "26046 MiB RAM detected, ~24802 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--extract bed1: 4063000 variants excluded.\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "160 variants remaining after main filters.\n",
      "--export vcf to\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.vcf\n",
      ".done.\n",
      "End time: Fri Jul 25 13:40:55 2025\n",
      "Running bgzip and tabix for nonsynonymous_lof variants for ancestry: CAH\n"
     ]
    }
   ],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "#Loop over all the ancestries and the 2 variant classes - run rvtests for all coding and missense variants\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running plink to extract {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        #Extract relevant variants\n",
    "        ! /home/jupyter/tools/plink2 \\\n",
    "        --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release10_vwb \\\n",
    "        --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "        --extract range {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.variantstoKeep.txt \\\n",
    "        --recode vcf-iid \\\n",
    "        --out {WORK_DIR}/{ancestry}_TMEM175.{variant_class}\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running bgzip and tabix for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## Bgzip and Tabix (zip and index the file)\n",
    "        ! bgzip -f {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.vcf\n",
    "        ! tabix -f -p vcf {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a43b8bc2-e466-4db2-a9bc-4705f5db2289",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Run RVtests\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running RVtests for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## RVtests with covariates \n",
    "        #Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "        ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "        --out {WORK_DIR}/{ancestry}_TMEM175.burden.{variant_class} \\\n",
    "        --kernel skato \\\n",
    "        --inVcf {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.vcf.gz \\\n",
    "        --pheno /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt \\\n",
    "        --pheno-name PHENO \\\n",
    "        --gene TMEM175 \\\n",
    "        --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "        --covar /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt \\\n",
    "        --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "        --freqUpper 0.01\n",
    "# --burden cmc,zeggini,mb,fp,cmcWald --kernel skat,skato \\"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8cb5051-79a8-4744-8076-b4d28a449da6",
   "metadata": {},
   "source": [
    "Look at RVtest results SKAT-O"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fbcb674a-a2b2-4190-b62e-18b20753e9ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "a1c4a90b-dd72-4fcc-99ed-c4c6555e0257",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/rpy2/rinterface/__init__.py:1211: UserWarning: Environment variable \"R_SESSION_TMPDIR\" redefined by R and overriding existing variable. Current: \"/tmp/RtmpmxFdFs\", R: \"/tmp/RtmpwZrfZW\"\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "28543bf4-b735-4ea0-b3b1-98231277340b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t40128\t128\t103\t800191\t0\t0.40194\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "39824701-4d13-4b7f-8c3a-14e2d083f397",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t40128\t69\t56\t76157.3\t1\t0.856905\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f063909c-7ec1-4c74-896f-e33fc3be96eb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1061\t59\t46\t36140.1\t0\t0.628155\n"
     ]
    }
   ],
   "source": [
    "#Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e0009a03-8e35-4726-b587-01b5a61698bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1061\t38\t32\t25101.9\t0\t0.524219\n"
     ]
    }
   ],
   "source": [
    "#Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1472901e-4c3e-4015-87c4-269b26c43dd0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t2738\t76\t61\t37231.7\t1\t0.738165\n"
     ]
    }
   ],
   "source": [
    "#Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "20e8c6a4-8627-4218-aea7-6140c82df65b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t2738\t47\t37\t22926.7\t1\t0.743344\n"
     ]
    }
   ],
   "source": [
    "#Check AFR nonsynonymous_lof results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7ea09aad-a29e-48ab-a1b9-e0fa1fb66cba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t2174\t24\t18\t21568.3\t0\t0.746508\n"
     ]
    }
   ],
   "source": [
    "#Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "02d7903e-a1bd-498e-b5d8-81c4c66dfc2e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t2174\t15\t12\t14410\t1\t0.580701\n"
     ]
    }
   ],
   "source": [
    "#Check AJ nonsynonymous_lof  variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0ab7b024-ee3c-44f5-98e1-de540cc3d912",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t3265\t72\t53\t41022.6\t0\t0.62126\n"
     ]
    }
   ],
   "source": [
    "#Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "fb5dfaa3-c69f-4c7c-9dd9-0285bf734f9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t3265\t42\t29\t5872.81\t1\t0.829141\n"
     ]
    }
   ],
   "source": [
    "#Check AMR nonsynonymous_lof  variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c743ae0d-b2ed-4042-9c01-3eedc58bfc45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1104\t39\t22\t22431.3\t1\t0.179996\n"
     ]
    }
   ],
   "source": [
    "#Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "809cfb25-e8ee-4ca6-90c6-3eb78c5c60d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1104\t26\t12\t7573.21\t0\t0.261768\n"
     ]
    }
   ],
   "source": [
    "#Check CAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "24e3a683-273d-46c5-a60b-cde446ce692b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t4670\t52\t34\t32480.5\t0\t0.85871\n"
     ]
    }
   ],
   "source": [
    "#Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "14fc05db-d9d7-4151-b7a1-271c51417bef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t4670\t34\t21\t28568.2\t0\t0.401021\n"
     ]
    }
   ],
   "source": [
    "#Check EAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "348e9e57-2ad0-4292-ada5-484c1a1ccfb9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1230\t47\t30\t17800.6\t0\t0.42309\n"
     ]
    }
   ],
   "source": [
    "#Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "17856f1c-809d-4d70-bd6e-d1bcbb8cd710",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1230\t24\t19\t18140.9\t0\t0.0663344\n"
     ]
    }
   ],
   "source": [
    "#Check MDE nonsynonymous_lof  variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "1e704c96-e0ed-4a7a-810b-f5c7dcdb9cdc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t689\t25\t13\t4939.3\t0\t0.523074\n"
     ]
    }
   ],
   "source": [
    "#Check SAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f4ba3adc-9c8e-4373-b60d-0d1e37f9128f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t689\t18\t10\t3602.35\t1\t0.546915\n"
     ]
    }
   ],
   "source": [
    "#Check SAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "32151fda-c16d-4ad7-b1b0-ac34c94cee2a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t900\t47\t35\t12161\t1\t0.642748\n"
     ]
    }
   ],
   "source": [
    "#Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.burden.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b7eaa5ec-087f-402b-a635-e6ba4fde4cd0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t900\t30\t24\t15084.2\t0\t0.433118\n"
     ]
    }
   ],
   "source": [
    "#Check CAH nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.burden.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51c9cf4d-fb0b-450f-a33d-dba0905bfa67",
   "metadata": {},
   "source": [
    "add GWA hits as covariates for TMEM175 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "2cd62c6b-3f6a-4adb-a37e-729be792ba8e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:06:54 2025\n",
      "26046 MiB RAM detected, ~24890 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:06:56 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:06:57 2025\n",
      "26046 MiB RAM detected, ~24866 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:06:58 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:06:59 2025\n",
      "26046 MiB RAM detected, ~24861 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:00 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:02 2025\n",
      "26046 MiB RAM detected, ~24852 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:03 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:04 2025\n",
      "26046 MiB RAM detected, ~24860 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:05 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:06 2025\n",
      "26046 MiB RAM detected, ~24858 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:07 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:08 2025\n",
      "26046 MiB RAM detected, ~24864 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:09 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:10 2025\n",
      "26046 MiB RAM detected, ~24865 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:11 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:12 2025\n",
      "26046 MiB RAM detected, ~24863 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:13 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:15 2025\n",
      "26046 MiB RAM detected, ~24869 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:16 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 958159\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "  --to-bp 958159\n",
      "\n",
      "Start time: Fri Jul 25 14:07:17 2025\n",
      "26046 MiB RAM detected, ~24867 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:07:18 2025\n"
     ]
    }
   ],
   "source": [
    "## extract p.m393t using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release10_vwb \\\n",
    "    --chr 4 \\\n",
    "    --from-bp 958159 \\\n",
    "    --to-bp 958159 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_M393T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "7157b2c6-3496-4a12-97a6-190541913b0d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:12:58 2025\n",
      "26046 MiB RAM detected, ~24880 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.bim.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:12:59 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:00 2025\n",
      "26046 MiB RAM detected, ~24846 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.bim.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:00 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:02 2025\n",
      "26046 MiB RAM detected, ~24841 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:02 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:04 2025\n",
      "26046 MiB RAM detected, ~24852 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:04 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:05 2025\n",
      "26046 MiB RAM detected, ~24863 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.bim.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:06 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:07 2025\n",
      "26046 MiB RAM detected, ~24849 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.bim.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:07 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:09 2025\n",
      "26046 MiB RAM detected, ~24848 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.bim.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:09 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:11 2025\n",
      "26046 MiB RAM detected, ~24862 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.bim.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:11 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:12 2025\n",
      "26046 MiB RAM detected, ~24848 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.bim.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:13 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:14 2025\n",
      "26046 MiB RAM detected, ~24842 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.bim.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:14 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T\n",
      "\n",
      "Start time: Fri Jul 25 14:13:16 2025\n",
      "26046 MiB RAM detected, ~24841 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.bim.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_M393T.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:13:16 2025\n"
     ]
    }
   ],
   "source": [
    "## recode p.m393t using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    !/home/jupyter/tools/plink2 \\\n",
    "        --bfile {WORK_DIR}/{ancestry}_M393T \\\n",
    "        --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "        --recode A \\\n",
    "        --out {WORK_DIR}/{ancestry}_M393T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "f8a072d7-d6bf-413a-8143-8f55ebd28282",
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','FIN','MDE','SAS','CAH', 'EUR'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    # read cov file\n",
    "    cov = pd.read_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}_covariate_file_noGBA.txt', sep = '\\t')\n",
    "    M393T = pd.read_csv(f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}/{ancestry}_M393T.raw', sep = '\\t')\n",
    "    cov_M393T = cov.merge(M393T[['IID', 'chr4:958159:T:C_T']], on='IID', how='left')\n",
    "    cov_M393T = cov_M393T.rename(columns={'chr4:958159:T:C_T': 'M393T'})\n",
    "    cov_M393T = cov_M393T.dropna()\n",
    "\n",
    "    ## save files \n",
    "    cov_M393T.to_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}.cov_M393T.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "dff1e458-0be4-4c4b-b3b0-1d80f172e379",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:21 2025\n",
      "26046 MiB RAM detected, ~24828 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.psam.\n",
      "4117743 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AAC/chr4_AAC_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:22 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:24 2025\n",
      "26046 MiB RAM detected, ~24829 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.psam.\n",
      "1752681 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AJ/chr4_AJ_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:24 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:26 2025\n",
      "26046 MiB RAM detected, ~24829 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.psam.\n",
      "7745089 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EUR/chr4_EUR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:27 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:28 2025\n",
      "26046 MiB RAM detected, ~24810 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.psam.\n",
      "4434664 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AMR/chr4_AMR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:29 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:31 2025\n",
      "26046 MiB RAM detected, ~24821 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.psam.\n",
      "2741528 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAS/chr4_CAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:32 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:33 2025\n",
      "26046 MiB RAM detected, ~24822 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.psam.\n",
      "3379769 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/EAS/chr4_EAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:34 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:35 2025\n",
      "26046 MiB RAM detected, ~24821 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.psam.\n",
      "939049 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/FIN/chr4_FIN_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:36 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:37 2025\n",
      "26046 MiB RAM detected, ~24819 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.psam.\n",
      "5095310 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/AFR/chr4_AFR_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:38 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:39 2025\n",
      "26046 MiB RAM detected, ~24820 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.psam.\n",
      "3023064 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/MDE/chr4_MDE_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:40 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:41 2025\n",
      "26046 MiB RAM detected, ~24824 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.psam.\n",
      "4063160 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/CAH/chr4_CAH_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:42 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.log.\n",
      "Options in effect:\n",
      "  --chr 4\n",
      "  --from-bp 950422\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P\n",
      "  --pfile /home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb\n",
      "  --to-bp 950422\n",
      "\n",
      "Start time: Fri Jul 25 14:23:44 2025\n",
      "26046 MiB RAM detected, ~24823 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.psam.\n",
      "2227587 variants loaded from\n",
      "/home/jupyter/workspace/gp2_tier2_eu_release10/imputed_genotypes/SAS/chr4_SAS_release10_vwb.pvar.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "1 variant remaining after main filters.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.bed ...\n",
      "done.\n",
      "End time: Fri Jul 25 14:23:44 2025\n"
     ]
    }
   ],
   "source": [
    "## extract p.Q65P using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    ! /home/jupyter/tools/plink2 \\\n",
    "    --pfile {REL10_PATH}/imputed_genotypes/{ancestry}/chr4_{ancestry}_release10_vwb \\\n",
    "    --chr 4 \\\n",
    "    --from-bp 950422 \\\n",
    "    --to-bp 950422 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_Q65P"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "8c2f17d8-c5d4-4018-85d3-ec643ad60887",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AAC/AAC.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:08 2025\n",
      "26046 MiB RAM detected, ~24811 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1215 samples (683 females, 532 males; 1215 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.bim.\n",
      "1 binary phenotype loaded (370 cases, 763 controls).\n",
      "--keep: 1207 samples remaining.\n",
      "1207 samples (680 females, 527 males; 1207 founders) remaining after main\n",
      "filters.\n",
      "369 cases and 756 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:08 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AJ/AJ.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:09 2025\n",
      "26046 MiB RAM detected, ~24804 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3472 samples (1326 females, 2146 males; 3472 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.bim.\n",
      "1 binary phenotype loaded (1531 cases, 435 controls).\n",
      "--keep: 2716 samples remaining.\n",
      "2716 samples (975 females, 1741 males; 2716 founders) remaining after main\n",
      "filters.\n",
      "1285 cases and 408 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:10 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:11 2025\n",
      "26046 MiB RAM detected, ~24800 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "--export A pass 1/1: writindone.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:11 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AMR/AMR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:13 2025\n",
      "26046 MiB RAM detected, ~24822 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3608 samples (1873 females, 1735 males; 3608 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.bim.\n",
      "1 binary phenotype loaded (1995 cases, 1439 controls).\n",
      "--keep: 3471 samples remaining.\n",
      "3471 samples (1801 females, 1670 males; 3471 founders) remaining after main\n",
      "filters.\n",
      "1917 cases and 1402 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:13 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/CAS/CAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:14 2025\n",
      "26046 MiB RAM detected, ~24817 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1691 samples (954 females, 737 males; 1691 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.bim.\n",
      "1 binary phenotype loaded (782 cases, 626 controls).\n",
      "--keep: 1518 samples remaining.\n",
      "1518 samples (872 females, 646 males; 1518 founders) remaining after main\n",
      "filters.\n",
      "734 cases and 578 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:14 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/EAS/EAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:16 2025\n",
      "26046 MiB RAM detected, ~24824 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "6662 samples (2661 females, 4001 males; 6662 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.bim.\n",
      "1 binary phenotype loaded (2411 cases, 2705 controls).\n",
      "--keep: 6414 samples remaining.\n",
      "6414 samples (2569 females, 3845 males; 6414 founders) remaining after main\n",
      "filters.\n",
      "2377 cases and 2598 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:16 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/FIN/FIN.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:17 2025\n",
      "26046 MiB RAM detected, ~24831 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "144 samples (82 females, 62 males; 144 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.bim.\n",
      "1 binary phenotype loaded (106 cases, 12 controls).\n",
      "--keep: 111 samples remaining.\n",
      "111 samples (66 females, 45 males; 111 founders) remaining after main filters.\n",
      "80 cases and 11 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:18 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/AFR/AFR.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:19 2025\n",
      "26046 MiB RAM detected, ~24822 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "3780 samples (1693 females, 2087 males; 3780 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.bim.\n",
      "1 binary phenotype loaded (1191 cases, 2307 controls).\n",
      "--keep: 3581 samples remaining.\n",
      "3581 samples (1596 females, 1985 males; 3581 founders) remaining after main\n",
      "filters.\n",
      "1147 cases and 2169 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:19 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/MDE/MDE.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:20 2025\n",
      "26046 MiB RAM detected, ~24813 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1493 samples (678 females, 815 males; 1493 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.bim.\n",
      "1 binary phenotype loaded (752 cases, 559 controls).\n",
      "--keep: 1448 samples remaining.\n",
      "1448 samples (659 females, 789 males; 1448 founders) remaining after main\n",
      "filters.\n",
      "724 cases and 551 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:21 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/CAH/CAH.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:22 2025\n",
      "26046 MiB RAM detected, ~24807 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "1111 samples (522 females, 589 males; 1111 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.bim.\n",
      "1 binary phenotype loaded (706 cases, 334 controls).\n",
      "--keep: 1039 samples remaining.\n",
      "1039 samples (485 females, 554 males; 1039 founders) remaining after main\n",
      "filters.\n",
      "663 cases and 318 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:22 2025\n",
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P\n",
      "  --export A\n",
      "  --keep /home/jupyter/workspace/ws_files/SAS/SAS.noGBA.samplestoKeep\n",
      "  --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P\n",
      "\n",
      "Start time: Fri Jul 25 14:24:24 2025\n",
      "26046 MiB RAM detected, ~24801 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "945 samples (339 females, 606 males; 945 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.fam.\n",
      "1 variant loaded from\n",
      "/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.bim.\n",
      "1 binary phenotype loaded (317 cases, 269 controls).\n",
      "--keep: 870 samples remaining.\n",
      "870 samples (309 females, 561 males; 870 founders) remaining after main\n",
      "filters.\n",
      "309 cases and 261 controls remaining after main filters.\n",
      "--export A pass 1/1: writing.done.\n",
      "--export A: /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_Q65P.raw\n",
      "written.\n",
      "End time: Fri Jul 25 14:24:24 2025\n"
     ]
    }
   ],
   "source": [
    "## recode p.m393t using plink\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    !/home/jupyter/tools/plink2 \\\n",
    "        --bfile {WORK_DIR}/{ancestry}_Q65P \\\n",
    "        --keep /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.noGBA.samplestoKeep \\\n",
    "        --recode A \\\n",
    "        --out {WORK_DIR}/{ancestry}_Q65P"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5e8d5903-f54b-41c6-aabc-9f2ec8778d86",
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','FIN','MDE','SAS','CAH', 'EUR'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    # read cov file\n",
    "    cov_M393T = pd.read_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}.cov_M393T.txt', sep = '\\t')\n",
    "    Q65P = pd.read_csv(f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}/{ancestry}_Q65P.raw', sep = '\\t')\n",
    "    cov_Q65P = cov_M393T.merge(Q65P[['IID', 'chr4:950422:A:C_A']], on='IID', how='left')\n",
    "    cov_Q65P = cov_Q65P.rename(columns={'chr4:950422:A:C_A': 'Q65P'})\n",
    "    cov_Q65P = cov_Q65P.dropna()\n",
    "\n",
    "    ## save files \n",
    "    cov_Q65P.to_csv(f'~/workspace/ws_files/{ancestry}/{ancestry}.cov_M393T_Q65P.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e50955a-48fd-415e-9572-714b64ce03bc",
   "metadata": {},
   "source": [
    "Run RVtest with GWAS hits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "2dd28132-ce75-46b7-8eda-bd60f5ef626a",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running RVtests for exonic variants for ancestry: AAC\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:29:59 2025\n",
      "[INFO]\tLoaded [ 1207 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 1014 ] sample phenotypes\n",
      "[WARN]\tTotal [ 193 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BBDP_001037 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001042 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000142 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000150 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000204 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000215 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000270 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000375 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000418 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000470 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 183 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 429 male, 585 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 306 cases, 708 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 1014 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 59 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:00 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: AAC\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:01 2025\n",
      "[INFO]\tLoaded [ 1207 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 1014 ] sample phenotypes\n",
      "[WARN]\tTotal [ 193 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BBDP_001037 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001042 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000142 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000150 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000204 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000215 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000270 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000375 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000418 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000470 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 183 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 429 male, 585 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AAC/AAC.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 306 cases, 708 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 1014 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 38 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:02 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: AFR\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:03 2025\n",
      "[INFO]\tLoaded [ 3581 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 2547 ] sample phenotypes\n",
      "[WARN]\tTotal [ 1034 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BCM_000096 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000173 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000224 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000277 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000286 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000287 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000315 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000402 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000415 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000424 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 1024 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 1328 male, 1219 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 997 cases, 1550 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 2547 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 76 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:04 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: AFR\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:05 2025\n",
      "[INFO]\tLoaded [ 3581 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 2547 ] sample phenotypes\n",
      "[WARN]\tTotal [ 1034 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BCM_000096 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000173 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000224 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000277 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000286 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000287 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000315 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000402 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000415 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000424 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 1024 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 1328 male, 1219 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AFR/AFR.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 997 cases, 1550 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 2547 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 47 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:06 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: AJ\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:08 2025\n",
      "[INFO]\tLoaded [ 2716 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 1421 ] sample phenotypes\n",
      "[WARN]\tTotal [ 1295 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_001477 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000014 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000024 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000047 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000059 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000076 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000161 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000242 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000402 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000468 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 1285 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 925 male, 496 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 1203 cases, 218 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 1421 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 24 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:08 2025\n",
      "[INFO]\tAnalysis took 0 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: AJ\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:10 2025\n",
      "[INFO]\tLoaded [ 2716 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 1421 ] sample phenotypes\n",
      "[WARN]\tTotal [ 1295 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_001477 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000014 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000024 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000047 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000059 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000076 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000161 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000242 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000402 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000468 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 1285 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 925 male, 496 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AJ/AJ.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 1203 cases, 218 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 1421 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 15 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:11 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: AMR\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:12 2025\n",
      "[INFO]\tLoaded [ 3471 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 3104 ] sample phenotypes\n",
      "[WARN]\tTotal [ 367 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BBDP_000167 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000219 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000446 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000452 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000595 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000639 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000807 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001035 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001057 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001064 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 357 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 1461 male, 1643 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 1772 cases, 1332 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 3104 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 72 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:13 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: AMR\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:14 2025\n",
      "[INFO]\tLoaded [ 3471 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 3104 ] sample phenotypes\n",
      "[WARN]\tTotal [ 367 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BBDP_000167 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000219 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000446 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000452 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000595 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000639 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000807 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001035 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001057 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001064 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 357 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 1461 male, 1643 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/AMR/AMR.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 1772 cases, 1332 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 3104 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 42 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:15 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: CAS\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:16 2025\n",
      "[INFO]\tLoaded [ 1518 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 861 ] sample phenotypes\n",
      "[WARN]\tTotal [ 657 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BCM_000079 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000141 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000271 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000483 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000557 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BRFAA_000002 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BRFAA_000048 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000020 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000081 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000087 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 647 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 348 male, 513 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 501 cases, 360 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 861 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 39 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:17 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: CAS\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:18 2025\n",
      "[INFO]\tLoaded [ 1518 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 861 ] sample phenotypes\n",
      "[WARN]\tTotal [ 657 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BCM_000079 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000141 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000271 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000483 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000557 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BRFAA_000002 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BRFAA_000048 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000020 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000081 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000087 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 647 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 348 male, 513 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/CAS/CAS.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 501 cases, 360 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 861 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 26 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:19 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: EAS\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:20 2025\n",
      "[INFO]\tLoaded [ 6414 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 3364 ] sample phenotypes\n",
      "[WARN]\tTotal [ 3050 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_000111 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000640 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001167 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001228 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001431 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001470 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_002088 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_003892 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_003993 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_004078 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 3040 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 1720 male, 1644 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 2025 cases, 1339 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 3364 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 52 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:21 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: EAS\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:22 2025\n",
      "[INFO]\tLoaded [ 6414 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 3364 ] sample phenotypes\n",
      "[WARN]\tTotal [ 3050 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_000111 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000640 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001167 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001228 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001431 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001470 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_002088 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_003892 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_003993 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_004078 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 3040 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 1720 male, 1644 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/EAS/EAS.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 2025 cases, 1339 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 3364 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 34 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:23 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: EUR\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:25 2025\n",
      "[INFO]\tLoaded [ 53641 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 25226 ] sample phenotypes\n",
      "[WARN]\tTotal [ 28415 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_000027 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000346 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000566 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000606 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000742 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000824 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000876 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000893 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001152 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001226 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 28405 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 15143 male, 10083 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 19490 cases, 5736 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 25226 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 128 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:29 2025\n",
      "[INFO]\tAnalysis took 4 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: EUR\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:31 2025\n",
      "[INFO]\tLoaded [ 53641 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 25226 ] sample phenotypes\n",
      "[WARN]\tTotal [ 28415 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_000027 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000346 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000566 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000606 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000742 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000824 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000876 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_000893 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001152 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001226 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 28405 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 15143 male, 10083 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/EUR/EUR.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 19490 cases, 5736 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 25226 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 69 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:34 2025\n",
      "[INFO]\tAnalysis took 3 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: FIN\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:36 2025\n",
      "[INFO]\tLoaded [ 111 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 39 ] sample phenotypes\n",
      "[WARN]\tTotal [ 72 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BBDP_000731 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000901 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001044 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000965 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_002493 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008587 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008956 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ HSPH_000272 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ HSPH_000278 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ HSPH_000631 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 62 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 20 male, 19 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 33 cases, 6 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 39 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 4 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:36 2025\n",
      "[INFO]\tAnalysis took 0 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: FIN\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:37 2025\n",
      "[INFO]\tLoaded [ 111 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 39 ] sample phenotypes\n",
      "[WARN]\tTotal [ 72 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ BBDP_000731 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000901 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001044 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_000965 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CATPD_002493 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008587 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ CORIELL_008956 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ HSPH_000272 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ HSPH_000278 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ HSPH_000631 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 62 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 20 male, 19 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/FIN/FIN.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 33 cases, 6 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 39 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 2 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:38 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: MDE\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:39 2025\n",
      "[INFO]\tLoaded [ 1448 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 1070 ] sample phenotypes\n",
      "[WARN]\tTotal [ 378 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ AAPDGC_000022 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000039 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000130 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000163 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000164 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000167 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AUTH_000029 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AUTH_000045 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AUTH_000057 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000188 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 368 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 560 male, 510 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 592 cases, 478 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 1070 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 47 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:40 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: MDE\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:41 2025\n",
      "[INFO]\tLoaded [ 1448 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 1070 ] sample phenotypes\n",
      "[WARN]\tTotal [ 378 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ AAPDGC_000022 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000039 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000130 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000163 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000164 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AAPDGC_000167 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AUTH_000029 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AUTH_000045 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ AUTH_000057 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000188 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 368 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 560 male, 510 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/MDE/MDE.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 592 cases, 478 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 1070 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 24 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:42 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: SAS\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:43 2025\n",
      "[INFO]\tLoaded [ 870 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 405 ] sample phenotypes\n",
      "[WARN]\tTotal [ 465 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_001174 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001310 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001558 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001714 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001778 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_002119 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_003845 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_004105 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000539 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001159 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 455 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 247 male, 158 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 248 cases, 157 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 405 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 25 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:45 2025\n",
      "[INFO]\tAnalysis took 2 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: SAS\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:47 2025\n",
      "[INFO]\tLoaded [ 870 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 405 ] sample phenotypes\n",
      "[WARN]\tTotal [ 465 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_001174 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001310 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001558 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001714 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_001778 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_002119 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_003845 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ APGS_004105 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_000539 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001159 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 455 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 247 male, 158 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/SAS/SAS.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 248 cases, 157 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 405 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 18 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:48 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for exonic variants for ancestry: CAH\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.GWAhits.exonic]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.exonic.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.GWAhits.exonic\n",
      "    --covar /home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:50 2025\n",
      "[INFO]\tLoaded [ 1039 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 845 ] sample phenotypes\n",
      "[WARN]\tTotal [ 194 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_003594 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001043 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000127 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000230 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000444 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000733 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000798 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BIOFIND_000151 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BLAACPD-KPM_000001 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BLAACPD-RUSH_000020 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 184 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 443 male, 402 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 553 cases, 292 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 845 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 47 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:51 2025\n",
      "[INFO]\tAnalysis took 1 seconds\n",
      "RVTESTS finished successfully\n",
      "Running RVtests for nonsynonymous_lof variants for ancestry: CAH\n",
      "Thank you for using rvtests (version: 20190205, git: c86e589efef15382603300dc7f4c3394c82d69b8)\n",
      "  For documentations, refer to http://zhanxw.github.io/rvtests/\n",
      "  For questions and comments, plase send to Xiaowei Zhan <zhanxw@umich.edu>\n",
      "  For bugs and feature requests, please submit at: https://github.com/zhanxw/rvtests/issues\n",
      "\n",
      "The following parameters are available.  Ones with \"[]\" are in effect:\n",
      "\n",
      "Available Options\n",
      "      Basic Input/Output:\n",
      "                          --inVcf [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.vcf.gz]\n",
      "                          --inBgen [], --inBgenSample [], --inKgg []\n",
      "                          --out [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.GWAhits.nonsynonymous_lof]\n",
      "                          --outputRaw\n",
      "       Specify Covariate:\n",
      "                          --covar [/home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt]\n",
      "                          --covar-name [SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P]\n",
      "                          --sex\n",
      "       Specify Phenotype:\n",
      "                          --pheno [/home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt]\n",
      "                          --inverseNormal, --useResidualAsPhenotype, --mpheno []\n",
      "                          --pheno-name [PHENO], --qtl, --multiplePheno []\n",
      "        Specify Genotype: --dosage [], --multipleAllele\n",
      "    Chromosome X Options: --xLabel [], --xParRegion []\n",
      "           People Filter: --peopleIncludeID [], --peopleIncludeFile []\n",
      "                          --peopleExcludeID [], --peopleExcludeFile []\n",
      "             Site Filter: --rangeList [], --rangeFile [], --siteFile []\n",
      "                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []\n",
      "                          --annoType []\n",
      "         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []\n",
      "       Association Model: --single [], --burden [], --vt [], --kernel [skato]\n",
      "                          --meta []\n",
      "     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []\n",
      "                          --xHemiKinshipEigen [], --boltPlink []\n",
      "                          --boltPlinkNoCheck\n",
      "          Grouping Unit :\n",
      "                          --geneFile [/home/jupyter/workspace/ws_files/refFlat_HG38.txt]\n",
      "                          --gene [TMEM175], --setList [], --setFile [], --set []\n",
      "        Frequency Cutoff: --freqUpper [0.01], --freqLower []\n",
      "            Missing Data: --impute [], --imputePheno, --imputeCov\n",
      "    Conditional Analysis: --condition []\n",
      "     Auxiliary Functions: --noweb [true], --hide-covar [true], --numThread []\n",
      "                          --outputID, --help\n",
      "\n",
      "Effective Options\n",
      "    --inVcf /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.nonsynonymous_lof.vcf.gz\n",
      "    --out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.GWAhits.nonsynonymous_lof\n",
      "    --covar /home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt\n",
      "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P\n",
      "    --pheno /home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt\n",
      "    --pheno-name PHENO\n",
      "    --kernel skato\n",
      "    --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt\n",
      "    --gene TMEM175\n",
      "    --freqUpper 0.01\n",
      "    --noweb\n",
      "    --hide-covar\n",
      "\n",
      "[INFO]\tProgram version: 20190205\n",
      "[INFO]\tAnalysis started at: Fri Jul 25 14:30:52 2025\n",
      "[INFO]\tLoaded [ 1039 ] samples from genotype files\n",
      "[INFO]\tLoaded [ 845 ] sample phenotypes\n",
      "[WARN]\tTotal [ 194 ] samples are dropped from VCF file due to missing phenotype.\n",
      "[WARN]\tDrop sample [ APGS_003594 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BBDP_001043 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000127 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000230 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000444 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000733 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BCM_000798 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BIOFIND_000151 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BLAACPD-KPM_000001 ] from VCF file due to missing phenotype\n",
      "[WARN]\tDrop sample [ BLAACPD-RUSH_000020 ] from VCF file due to missing phenotype\n",
      "[WARN]\tSkip outputting additional [ 184 ] samples due to missing phenotype.\n",
      "[INFO]\tBegin to read covariate file\n",
      "[INFO]\tLoaded 443 male, 402 female and 0 sex-unknown samples from /home/jupyter/workspace/ws_files/CAH/CAH.cov_M393T_Q65P.txt\n",
      "[INFO]\tLoaded 553 cases, 292 controls, and 0 missing phenotypes\n",
      "[WARN]\t-- Enabling binary phenotype mode -- \n",
      "[INFO]\tAnalysis begins with [ 845 ] samples...\n",
      "[INFO]\tSKAT-O test significance will be evaluated using weight = Beta[beta1 = 1.00, beta2 = 25.00]\n",
      "[INFO]\tLoaded [ 1 ] genes.\n",
      "[INFO]\tImpute missing genotype to mean (by default)\n",
      "[INFO]\tSet upper minor allele frequency limit to 0.01\n",
      "[INFO]\tAnalysis started\n",
      "[INFO]\tAnalyzed [ 30 ] variants from [ 1 ] genes/regions\n",
      "[INFO]\tAnalysis ends at: Fri Jul 25 14:30:52 2025\n",
      "[INFO]\tAnalysis took 0 seconds\n",
      "RVTESTS finished successfully\n"
     ]
    }
   ],
   "source": [
    "#Run RVtests\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['exonic', 'nonsynonymous_lof']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "        \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running RVtests for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## RVtests with covariates \n",
    "        #Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "        ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "        --out {WORK_DIR}/{ancestry}_TMEM175.GWAhits.{variant_class} \\\n",
    "        --kernel skato \\\n",
    "        --inVcf {WORK_DIR}/{ancestry}_TMEM175.{variant_class}.vcf.gz \\\n",
    "        --pheno /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.cov_M393T_Q65P.txt \\\n",
    "        --pheno-name PHENO \\\n",
    "        --gene TMEM175 \\\n",
    "        --geneFile /home/jupyter/workspace/ws_files/refFlat_HG38.txt \\\n",
    "        --covar /home/jupyter/workspace/ws_files/{ancestry}/{ancestry}.cov_M393T_Q65P.txt \\\n",
    "        --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5,M393T,Q65P \\\n",
    "        --freqUpper 0.01\n",
    "# --burden cmc,zeggini,mb,fp,cmcWald --kernel skat,skato \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "55bd8eac-501f-4a0c-9816-68253c8b4cae",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/TMEM175\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "d9f94bf9-d4f3-4ec8-bde9-cb4bfad6595d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1014\t59\t45\t7107.84\t0\t0.424299\n"
     ]
    }
   ],
   "source": [
    "#Check AAC all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "ef99c567-d214-4159-baa3-c66b17d9e9ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1014\t38\t31\t3376.87\t0\t0.622341\n"
     ]
    }
   ],
   "source": [
    "#Check AAC nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_AAC/AAC_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "85c67698-e148-4819-bad3-ca69b262d29b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t2547\t76\t60\t2155.27\t1\t0.873784\n"
     ]
    }
   ],
   "source": [
    "#Check AFR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "c480e7f2-9cd8-4399-9387-72acd045c120",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t2547\t47\t37\t5176.09\t1\t0.703047\n"
     ]
    }
   ],
   "source": [
    "#Check AFR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_AFR/AFR_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "c50780c3-f8d6-4c0b-a0ff-1384c44d721d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1421\t24\t16\t4023.79\t0\t0.212791\n"
     ]
    }
   ],
   "source": [
    "#Check AJ all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "f606b059-7251-421d-9416-9f637456bd61",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1421\t15\t10\t3156.23\t0\t0.240471\n"
     ]
    }
   ],
   "source": [
    "#Check AJ nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_AJ/AJ_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "7f8ee6c1-944b-4675-846d-023feb1055f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t3104\t72\t51\t7510.5\t0\t0.5333\n"
     ]
    }
   ],
   "source": [
    "#Check AMR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "0c8b29f2-31ab-4bc2-937a-5e2bba968e8d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t3104\t42\t29\t11430.4\t1\t0.143274\n"
     ]
    }
   ],
   "source": [
    "#Check AMR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_AMR/AMR_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "7f71534f-2394-49c2-b047-6232e799026d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t861\t39\t18\t480.407\t1\t0.688999\n"
     ]
    }
   ],
   "source": [
    "#Check CAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "25eecb59-9865-4df0-98e1-e3663fd5ef0c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t861\t26\t9\tNA\tNA\tNA\n"
     ]
    }
   ],
   "source": [
    "#Check CAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAS/CAS_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "a00963b6-bf05-4538-a0b7-65f2f1a9dcef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t3364\t52\t32\t11949.2\t1\t0.255894\n"
     ]
    }
   ],
   "source": [
    "#Check EAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "5543fad8-d2ce-4c0f-8989-120b15b6afdf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t3364\t34\t19\t20875.8\t1\t0.0442927\n"
     ]
    }
   ],
   "source": [
    "#Check EAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_EAS/EAS_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "0e53cbb2-20f3-4fbb-a3a3-e104b266b3c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t25226\t128\t95\t153490\t0\t0.062384\n"
     ]
    }
   ],
   "source": [
    "#Check EUR all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "5d651c7d-25e4-42ff-b925-6c14a041e714",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t25226\t69\t55\t83041.7\t0\t0.141847\n"
     ]
    }
   ],
   "source": [
    "#Check EUR nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_EUR/EUR_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "f47be47c-5dc2-4c4f-9304-31c38edd697c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1070\t47\t28\t3884.07\t0\t0.358024\n"
     ]
    }
   ],
   "source": [
    "#Check MDE all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "5fd4fe42-07a8-4d04-8b05-ccdb404be6f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t1070\t24\t17\t1097.97\t0\t0.608166\n"
     ]
    }
   ],
   "source": [
    "#Check MDE nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_MDE/MDE_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "60421b16-c8a2-4a18-8717-a649e5b5afda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t405\t25\t10\t528.451\t0\t0.465113\n"
     ]
    }
   ],
   "source": [
    "#Check SAS all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "7afda8d6-7d4c-4314-a2e6-cc9d821162d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t405\t18\t8\t208.184\t0\t0.610274\n"
     ]
    }
   ],
   "source": [
    "#Check SAS nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_SAS/SAS_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "374ff640-9af5-4a6f-bc45-b8a3fc4966bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t845\t47\t34\t10121.7\t1\t0.273611\n"
     ]
    }
   ],
   "source": [
    "#Check CAH all_coding variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.GWAhits.exonic.SkatO.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "a2b0c834-b1fe-40d3-b0e6-efb7ec9bec1d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene\tRANGE\tN_INFORMATIVE\tNumVar\tNumPolyVar\tQ\trho\tPvalue\n",
      "TMEM175\t4:932459-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655,4:932386-958655\t845\t30\t23\t14266.3\t1\t0.00597753\n"
     ]
    }
   ],
   "source": [
    "#Check CAH nonsynonymous_lof variant results\n",
    "! cat {WORK_DIR}/TMEM175_CAH/CAH_TMEM175.GWAhits.nonsynonymous_lof.SkatO.assoc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9509fc00-3c42-4809-92af-3573d33629f5",
   "metadata": {},
   "source": [
    "LD pruning(in EUR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "8c1902e6-8ceb-4a64-a2de-0fffb324467d",
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"~/workspace/ws_files/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "27589137-f7b2-4950-8fff-aac40ab6a97d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//TMEM175/TMEM175_UNIMPUTED.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "  --exclude /home/jupyter/workspace/ws_files//exclusion_regions_hg38.txt\n",
      "  --geno 0.05\n",
      "  --hwe 1e-3\n",
      "  --keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep\n",
      "  --maf 0.01\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/ws_files//TMEM175/TMEM175_UNIMPUTED\n",
      "\n",
      "Start time: Fri Jul 25 15:20:54 2025\n",
      "26046 MiB RAM detected, ~24770 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "58823 samples (25749 females, 33074 males; 58823 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//TMEM175/TMEM175_EUR/EUR_TMEM175.fam.\n",
      "7209 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//TMEM175/TMEM175_EUR/EUR_TMEM175.bim.\n",
      "1 binary phenotype loaded (26778 cases, 10372 controls).\n",
      "--exclude: 7209 variants remaining.\n",
      "--keep: 53641 samples remaining.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) remaining after main\n",
      "filters.\n",
      "24208 cases and 9662 controls remaining after main filters.\n",
      "done.lating allele frequencies... 0%\n",
      "--geno: 3538 variants removed due to missing genotype data.\n",
      "Error: --hwe filter is suspiciously strict for the sample size; you may be\n",
      "filtering out many variants with association signals.\n",
      "If you are performing routine QC, consider using --hwe with a sample-size term\n",
      "(e.g. \"--hwe 1e-5 0.001\" results in a threshold of 1e-6 with 1000 observations,\n",
      "1e-7 with 2000 observations, etc.), and/or specifying 'keep-fewhet' to remove\n",
      "only variants with heterozygosity excess.\n",
      "Alternatively, if the strictness is intentional (e.g. you are preparing data\n",
      "for a method that requires variants to be near Hardy-Weinberg equilibrium),\n",
      "explicitly specify a sample-size term of 0.\n",
      "End time: Fri Jul 25 15:20:54 2025\n"
     ]
    }
   ],
   "source": [
    "# Make sure to use high-quality SNPs\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/TMEM175/TMEM175_EUR/EUR_TMEM175 \\\n",
    "--maf 0.01 \\\n",
    "--geno 0.05 \\\n",
    "--hwe 1e-5 0.001 \\\n",
    "--make-bed \\\n",
    "--keep /home/jupyter/workspace/ws_files/EUR/EUR.noGBA.samplestoKeep \\\n",
    "--exclude {WORK_DIR}/exclusion_regions_hg38.txt \\\n",
    "--out {WORK_DIR}/TMEM175/TMEM175_UNIMPUTED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "bee53553-b5a5-446a-ad90-8f7df04ece5a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.7LM 64-bit Intel (7 Jul 2025)       cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/ws_files//TMEM175/prune_TMEM175.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/ws_files//TMEM175/TMEM175_UNIMPUTED\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out /home/jupyter/workspace/ws_files//TMEM175/prune_TMEM175\n",
      "\n",
      "Start time: Fri Jul 25 15:24:04 2025\n",
      "26046 MiB RAM detected, ~24800 available; reserving 13023 MiB for main\n",
      "workspace.\n",
      "Using up to 4 compute threads.\n",
      "53641 samples (23385 females, 30256 males; 53641 founders) loaded from\n",
      "/home/jupyter/workspace/ws_files//TMEM175/TMEM175_UNIMPUTED.fam.\n",
      "366 variants loaded from\n",
      "/home/jupyter/workspace/ws_files//TMEM175/TMEM175_UNIMPUTED.bim.\n",
      "1 binary phenotype loaded (24208 cases, 9662 controls).\n",
      "Calculating allele frequencies... done.\n",
      "--indep-pairwise (1 compute thread): 259/366 variants removed.\n",
      "Variant lists written to\n",
      "/home/jupyter/workspace/ws_files//TMEM175/prune_TMEM175.prune.in and\n",
      "/home/jupyter/workspace/ws_files//TMEM175/prune_TMEM175.prune.out .\n",
      "End time: Fri Jul 25 15:24:05 2025\n"
     ]
    }
   ],
   "source": [
    "# Prune out unnecessary SNPs (only need to do this to generate PCs)\n",
    "! /home/jupyter/tools/plink2 \\\n",
    "--bfile {WORK_DIR}/TMEM175/TMEM175_UNIMPUTED \\\n",
    "--indep-pairwise 50 5 0.5 \\\n",
    "--out {WORK_DIR}/TMEM175/prune_TMEM175"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "f0f07d15-6a03-4d9b-8400-37166ce78201",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "107 /home/jupyter/workspace/ws_files//TMEM175/prune_TMEM175.prune.in\n"
     ]
    }
   ],
   "source": [
    "!wc -l {WORK_DIR}/TMEM175/prune_TMEM175.prune.in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bb3d8bc-a0b2-4fef-b9a8-ea0e0fc5e376",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 4.67E-4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea48fd74-a7a5-4391-bf2a-513ee36ebf2c",
   "metadata": {},
   "source": [
    "zoom locus plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fe3d77ae-145b-4bcb-b9c6-0e16f5f695d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/pandas/core/arraylike.py:399: RuntimeWarning: divide by zero encountered in log\n"
     ]
    }
   ],
   "source": [
    "## locus zoom format\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    df = pd.read_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')\n",
    "    df['b'] = np.log(df['OR'])\n",
    "    export_ldassoc = df[['#CHROM', 'POS', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P']].copy()\n",
    "    export_ldassoc = export_ldassoc.rename(columns = {'#CHROM': 'CHROM', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p'})\n",
    "    # save to files\n",
    "    export_ldassoc.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.formatted.tab', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6fc54574-d0f4-498c-bbc9-a6aa8f22cfe7",
   "metadata": {},
   "source": [
    "GCTA cojo analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "483554d7-8082-41fe-b956-ad9f95a5d7f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/pandas/core/arraylike.py:399: RuntimeWarning: divide by zero encountered in log\n"
     ]
    }
   ],
   "source": [
    "## cojo format\n",
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "\n",
    "    sumstats = pd.read_csv(f'{WORK_DIR}/{ancestry}.all_adj.csv')\n",
    "    #Format summary statistics for GCTA-COJO\n",
    "    #First get the log odds ratio - this is required for COJO\n",
    "    #1) For a case-control study, the effect size should be log(odds ratio) with its corresponding standard error.\n",
    "    sumstats_formatted = sumstats.copy()\n",
    "    sumstats_formatted['b'] = np.log(sumstats_formatted['OR'])\n",
    "\n",
    "    #Now select just the necessary columns for COJO\n",
    "    sumstats_export = sumstats_formatted[['ID', 'A1', 'OMITTED', 'A1_FREQ', 'b', 'LOG(OR)_SE', 'P', 'OBS_CT']].copy()\n",
    "\n",
    "    #Rename columns following COJO format\n",
    "    sumstats_export = sumstats_export.rename(columns = {'ID':'SNP', 'OMITTED':'A2', 'A1_FREQ':'freq', 'LOG(OR)_SE':'se', 'P':'p', 'OBS_CT':'N'})\n",
    "\n",
    "    #Export\n",
    "    sumstats_export.to_csv(f'{WORK_DIR}/{ancestry}.all_adj.sumstats.ma', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40e0685f-0555-4c82-8cc2-47809fcbcee7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# install GCTA-COJO run in terminal\n",
    "! wget https://yanglab.westlake.edu.cn/software/gcta/bin/gcta-1.94.1-linux-kernel-3-x86_64.zip\n",
    "! unzip gcta-1.95.0-linux-kernel-3-x86_64.zip "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "9fdea434-7b5c-4886-bbea-7eb1cbf70646",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:39:09 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;33mWarning: \u001b[0mWarning: No output file specified. Using default output file 'output'.\n",
      "Accepted options:\n",
      "\u001b[0;31mError: \u001b[0m\n",
      "  invalid option \"--version\".\n",
      "\n",
      "\u001b[0m\u001b[0mAn error occurs, please check the options or data\n"
     ]
    }
   ],
   "source": [
    "! /home/jupyter/tools/gcta-1.95.0-linux-kernel-3-x86_64/gcta64 --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "6f8a5203-3077-46a7-a5d5-f397bdcfee34",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:28 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].\n",
      "1215 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].\n",
      "3520 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 1215 individuals and 3520 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 607 SNPs (2913 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 604 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AAC/AAC.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 604 SNPs: 5.96293 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "603 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 603 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:29 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.59 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:30 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].\n",
      "3472 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].\n",
      "1378 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 3472 individuals and 1378 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 414 SNPs (964 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 414 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 414 SNPs: 8.21794 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "412 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 412 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 411 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AJ/AJ.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:31 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.94 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:32 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].\n",
      "58823 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].\n",
      "7209 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 58823 individuals and 7209 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 396 SNPs (6813 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.96265 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "395 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 395 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 394 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EUR/EUR.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:40 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 7.87 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:42 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].\n",
      "3608 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].\n",
      "3883 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 3608 individuals and 3883 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 377 SNPs (3506 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 374 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AMR/AMR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 374 SNPs: 5.73422 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "373 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 373 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:42 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.73 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:43 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].\n",
      "1691 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].\n",
      "2501 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 1691 individuals and 2501 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 423 SNPs (2078 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 392 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAS/CAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 392 SNPs: 4.36192 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "389 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 389 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:44 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.57 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:45 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].\n",
      "6662 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].\n",
      "3374 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 6662 individuals and 3374 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 368 SNPs (3006 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 368 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 368 SNPs: 5.15033 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "368 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 368 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 1 associated SNPs are selected.\n",
      "Performing joint analysis on all the 1 selected signals ...\n",
      "Saving the 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 1 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 367 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_EAS/EAS.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(0 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:47 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 1.31 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:48 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].\n",
      "144 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].\n",
      "645 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 144 individuals and 645 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 422 SNPs (223 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 405 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 405 SNPs: 18.454 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "405 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 405 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "Finally, 3 associated SNPs are selected.\n",
      "Performing joint analysis on all the 3 selected signals ...\n",
      "Saving the 3 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.ldprune.COJO.jma.cojo] ...\n",
      "Saving the LD structure of 3 independent signals to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.ldprune.COJO.ldr.cojo] ...\n",
      "Saving the conditional analysis results of 355 remaining SNPs to [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_FIN/FIN.all_adj.ldprune.COJO.cma.cojo] ...\n",
      "(47 SNPs eliminated by backward selection and 0 SNPs filtered by collinearity test are not included in the output)\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:49 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.99 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:50 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].\n",
      "3780 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].\n",
      "4429 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 3780 individuals and 4429 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 598 SNPs (3831 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 596 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_AFR/AFR.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 596 SNPs: 4.94626 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "594 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 594 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:51 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.88 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:52 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].\n",
      "1493 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].\n",
      "2741 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 1493 individuals and 2741 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 400 SNPs (2341 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 397 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_MDE/MDE.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 397 SNPs: 5.04974 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "396 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 396 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:53 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.57 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:54 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].\n",
      "1111 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].\n",
      "3430 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 1111 individuals and 3430 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 520 SNPs (2910 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 533 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_CAH/CAH.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 533 SNPs: 5.52068 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "519 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 519 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:55 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.56 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.95.0 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Jul 21 2025 17:30:18, by GCC 8.4\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:43:56 UTC on Fri Jul 25 2025.\n",
      "\u001b[0;32m\u001b[0mHostname: 4b8bf4fabeb0\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175\n",
      "--maf 0.01\n",
      "--cojo-file /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma\n",
      "--cojo-p 0.000467\n",
      "--cojo-slct\n",
      "--out /home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.ldprune.COJO\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].\n",
      "945 individuals to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.fam].\n",
      "Reading PLINK BIM file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].\n",
      "2103 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bim].\n",
      "Reading PLINK BED file from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed] in SNP-major format ...\n",
      "Genotype data for 945 individuals and 2103 SNPs to be included from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS_TMEM175.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.01 ...\n",
      "After filtering SNPs with MAF > 0.01, there are 474 SNPs (1629 SNPs with MAF < 0.01).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma] ...\n",
      "GWAS summary statistics of 473 SNPs read from [/home/jupyter/workspace/ws_files/TMEM175/TMEM175_SAS/SAS.all_adj.sumstats.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 473 SNPs: 5.66068 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "471 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 471 SNPs to select association signals ... (p cutoff = 0.000467; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:43:56 UTC on Fri Jul 25 2025\n",
      "Overall computational time: 0.48 sec.\n"
     ]
    }
   ],
   "source": [
    "ancestries = {'AAC','AFR','AJ','AMR','CAS','EAS','EUR','FIN','MDE','SAS','CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    WORK_DIR = f'~/workspace/ws_files/TMEM175/TMEM175_{ancestry}'\n",
    "    \n",
    "    #Select multiple associated SNPs based on LD pruning p-value 4.67e-4\n",
    "    #Can change the p-value for significance if needed\n",
    "    #bfile is referring to the full dataset in plink binary format, e.g. GP2 - whatever you used to run the GWAS\n",
    "    ! /home/jupyter/tools/gcta-1.95.0-linux-kernel-3-x86_64/gcta64 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_TMEM175 \\\n",
    "    --maf 0.01 \\\n",
    "    --cojo-file {WORK_DIR}/{ancestry}.all_adj.sumstats.ma \\\n",
    "    --cojo-p 4.67e-4 \\\n",
    "    --cojo-slct \\\n",
    "    --out {WORK_DIR}/{ancestry}.all_adj.ldprune.COJO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d858fe4a-0663-4d35-a102-334728d929bb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.17"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
